---
title: T is for Topology
author:
  - name: Tanya Strydom
    id: ts
    orcid: 0000-0001-6067-1349
    corresponding: true
    email: t.strydom@sheffield.ac.uk
    role: 
    - conceptualization: lead
    - methodology: supporting
    affiliation:
      - id: sheffield
        name: School of Biosciences, University of Sheffield, Sheffield, UK
  - name: Jennifer A. Dunne
    id: jad
    orcid: 0000-0002-3789-9561
    corresponding: false
    role: 
    - conceptualization: lead
    - methodology: supporting
    affiliation:
      - id: sfi
        name: Santa Fe Institute, 1399 Hyde Park Road, Santa Fe, NM 87501, USA
  - name: Timothée Poisot
    id: tp
    orcid: 0000-0002-0735-5184
    corresponding: false
    role: 
    - conceptualization: lead
    - methodology: supporting
    affiliations:
      - Université de Montreal
      - Québec Centre for Biodiversity Sciences
  - name: Andrew P. Beckerman
    id: apb
    orcid: 0000-0002-7859-8394
    corresponding: false
    role: 
    - conceptualization: lead
    - methodology: supporting
    affiliations:
      - ref: sheffield
funding: "The author(s) received no specific funding for this work. Well they did I just haven't done the homework"
keywords:
  - food web
  - network construction
abstract: |
  There are many reasons one might want to generate a network and there are many tools on the market that might make that possible. However not all tools are created equally and there is reason to assume that not all networks will suit most purposes. Here the aim is to compare and contrast the different topology generating tools that are on the market and see where they shine and where they fall flat. There probably isn't one model to rule them all but it doesn't mean that we shouldn't be critical when we think about the model we want to use.
plain-language-summary: |
  Although it has been acknowledged that communities consist not only of co-occurring species but that they also interact being able to quantify those interactions and assemble them into interaction networks has been a limiting factor in the integration of network ecology into other fields of ecology. As the field of network ecology has matured there has been an accompanying expansion in the development of theory and tools that are centred around generating networks or predicting the interactions between species. Notably many of these tools have been developed with different underlying philosophies, ideas, and mechanisms as to what structures the interactions between species. It is thus critically important that those wanting to adopt these network generating tools be aware of how the the specific questions being asked maps to the underlying assumptions made when generating networks, as well as the limitations of how the networks/interactions are delimited. Here we provide an overview of the canonical network generating models, comparing and contrasting the underlying assumptions, data requirements, and resulting network predictions made by the different families in an attempt to provide guidance for those interested in adopting the generation of networks into their workflow. [R1. a discussion on the underlying assumptions we are making when we delimit a network]. [R2. an overview of how the different model families differ - ordination space/benchmarking]. [R3. identifying the relevant questions/bodies of theory that the networks generated by different families are suited to answer]. When choosing to construct an interaction network the researcher is faced with many assumptions and considerations that should be made and it is important to be aware of these limitations to avoid constructing (something poetic to capture the idea of falsity/false idols). Being aware of these choices is particularly important as the availability of these tools grows and network ecology starts to be adopted into other aspects of ecology and conservation biology.
date: last-modified
bibliography: references.bib
citation:
  container-title: Some fancy journal
number-sections: true
jupyter: python3
---

It can be argued that the interaction between species (or individuals) is one of the main determinants of the emergent properties that are studied in other fields of ecology, *e.g.,* the range of plant will be determined by the range of its pollinator, and although the importance of species interactions and the resulting networks that they form has been an acknowledged part of the ecological canon since Darwin's 'entangled bank' [@darwinOriginSpeciesMeans1859] (if not even earlier, stemming from Greek Antiquity [@thanosAristotleTheophrastusPlantanimal1994]), the adoption of network ecology into other disciplines of ecology has been limited. This was primarily driven by two limitations; firstly, it is extremely challenging to actually record species interactions in the field [@jordanoSamplingNetworksEcological2016; @jordanoChasingEcologicalInteractions2016], which has resulted in a limitation in the coverage of interaction data [@poisotGlobalKnowledgeGaps2021], secondly has been the need to develop a set of tools and terminology to construct, conceptualise, and analyse these networks. Although measuring interactions in the field remains a challenge the development of both practical tools (*i.e.,* tools that help as record predictions, [e.g. ref maybe @pringleResolvingFoodWebStructure2020]) as well as predictive tools [@morales-castillaInferringBioticInteractions2015; @strydomRoadmapPredictingSpecies2021] is allowing us to begin filling in these 'global gaps'. Additionally, there has been extensive development of tools that focus on quantifying the structure [ref], analysis [@daleGraphsSpatialGraphs2010], properties [@delmasAnalysingEcologicalNetworks2019] of networks. All together these tools means that as a field network ecology can (and should) be integrated into ecology [*e.g.,* @thuillerNavigatingIntegrationBiotic2024] and conservation biology. However (as with any new tool or model), it is important that one has a firm grasp of how networks (particularly synthetic ones) are generated and how the underlying philosophy thereof maps onto the questions being asked. Here we provide; a discussion of the underlying assumptions that are made when we attempt to delimit and describe a food webs, a synthesis of the different families of tools that are used to construct food webs, and a discussion linking network ecology to some of the outstanding questions in ecology.

> Three themes that are aimed at: providing a standardisation of terms that are sued when we are talking about both networks as well as what we mean when we are generating networks. The final theme aims to map network ecology to some of the outstanding questions in ecology

## The anatomy of a food web

Although we specifically focus on food webs (interactions representing feeding links) it is beneficial to take a step back and acknowledge the diversity of form that an interaction network can encapsulate. The idea of an interaction network seems simple, it is the representation of the interactions (edges) between species (nodes), the definition of an 'edge' and a 'node', as well as the scale at which they are aggregated can take many forms. As highlighted in @poisotDescribeUnderstandPredict2016 networks can be constructed at the population (the links between individuals), community (the links between species), or metacommunity (fluxes between locations) level. Even if we are to limit our definition of a network to represent community-level processes there are still many ways to define what is captured by the edges and nodes [insert some e.g.]. It is thus clear that the way that a network is coded (constructed) can influence the resulting observations and conclusions that are made [@proulxNetworkThinkingEcology2005; @brimacombeShortcomingsReusingSpecies2023], and it is important to have a strong grasp of what information a network is attempting to convey.

Even if one were to limit their scope to thinking of interaction networks only in terms of food webs there are still many ways to define the various components of the network one needs to understand the different intentions/assumptions that are made when a food web is constructed. Although the main intention of a food web is to capture and represent the feeding links between species there are many ways to define the nodes (*e.g.,* species or taxonomic group), edges (*e.g.* potential or realised feeding links), the magnitude of the edges (*e.g.,* binary vs probabilistic) and even how the network itself is delimited (does it represent an aggregation of interactions over time?, what is the spatial extent?). All these decisions will have an impact on the resultant structure and potential use-cases of the network.

### How do we define a node?

Although this may seem an elementary question in the context of food webs --- a node should represent a species, the reality is that nodes can often represent an aggregate of different (taxonomic) species - so called 'trophic species', and it is not uncommon that networks can have nodes that represent both taxonomic and trophic species (*e.g.,* there are many that do the basal 'plant/plant ally' node but include at least one REF). Practical implications of how we are aggregating the nodes is that the resolution may not be 'pixel perfect' *i.e.,* we may be unable to assess the co-extinction risk of a species pair [mutualism ref, at least there should be one of them], however there is value in 'low-resolution' nodes because it pains a more generalist picture??

### What is meant by an edge?

As discussed earlier there are many ways to define the links between species --- even feeding links. At its core links within food webs can be thought of a representation of the flow of a resource [ref], realised [@pringleResolvingFoodWebStructure2020] or potential [@dunneNetworkStructureFood2006] feeding links, or the flow of energy [sensu the ADBM @petcheySizeForagingFood2008 ??]. How we quantify links will influence how the resulting structure of the network - and the inferences we will make thereof. For example taking a food web that consists of links representing *potential* feeding links between species (*i.e.,* although species *a* may have the ability to consume species *b* it does not mean that it will be realised 'in the field') will be meaningless if you are interested in understanding feeding preferences within/between communities as the links are 'over distributed'.

In order to understand why the links in a food web can be coded to represent different things it is perhaps meaningful to understand the theory behind how the links between nodes are thought to arise. Here I think we can think about the theory of intervality and structural constraints, trait matching, and foraging theory

### How do we quantify the links between nodes?

Key THMs here is that not only what the link represent but how it is quantified can differ. Big there is binary, probabilistic ('continuous' distribution between 0 and 1) and weighted (continuous but not bound by 0, 1). I think the (when it finally goes to preprint) probabilities manuscript will come in here...

### How do we interpret the resulting network????

Talking about delimiting, the idea of aggregating over time or aggregating over space...

*something, something, introducing that the same problem (different philosophies) is also a thing that you need to think about when generating networks.*

## How do we construct ecological networks?

Arguably the need for methods and tools for constructing interaction networks arises from two different (but still aligned) places of interest within the field of network ecology. On the one side sits the researcher who is interested in generating a set of ecologically plausible networks for the purpose of running further simulations (*e.g.,* extinction simulations) or understanding some higher-level process/concept (*e.g.,* understanding energy flows), importantly these networks do not require any level of species specificity *per se* and it is more the arrangements of the nodes (species) within the context of network structure that is of value. This researcher is contrasted by one that is interested in constructing real-world, location specific, interaction data for a specific collection of species (community). This is driven by the need for researchers to find alternative ways to infer the interactions between species as a way to overcome the inherit challenges of inventorying interaction in the field (@morales-castillaInferringBioticInteractions2015 present a more mechanistic overview, while @strydomRoadmapPredictingSpecies2021 provides a more statistical overview). Of course these two categories are not distinct, mutually exclusive, groups but can rather be viewed as operating on a gradient ranging from a need for generality (*i.e.,* creating a network that, when taken in aggregate, the distribution of links (interactions) between species are ecologically plausible) to a need for specificity (*i.e.,* local-level predictions between specific species).

### Predicting structure or interactions?

These two groups are they themselves made up of different tools that also have their own underlying rules and assumptions that are made when constructing a food web, which will determine and influence the resulting structure or inferred interactions [@petcheySizeForagingFood2008]. Thus it is important to not lose sight of the core philosophy behind the model we use and to ensure that we are using the model best suited to what we want to be accomplishing.

 -   Core mechanistic differences that models will work at --- some are really concerned about (and thus constrained by) structure, others are more mechanistic in nature *i.e.,* species *a* has the capacity to eat species *b* because traits

### Model families

Given the large number of models that have been developed it is perhaps more meaningful to group models into families with the idea that models from the same family will yield similar results because they play by similar rules. These rules referring to the underlying philosophy as to what structures either networks or the interactions within them (see @fig-concept panel A). Although there have been efforts to compare and contrast different models [*e.g.,* @williamsSuccessItsLimits2008 looked at 'structural' models; and @pichlerMachineLearningAlgorithms2020 looked at 'machine learning algorithms'] there still lacks an overall synthesis as to how the different model families differ from each other - both in terms of what they are actually predicting as well as how well they are preforming in the different facets of constructing a network.

![Conceptual figure of the 'network prediction'. Panel A shows where the model families fall in the the context of being models that predict networks or models that predict interactions space. Panel B serves to highlight the characteristics one might like to 'test'/benchmark for a model based on it being either a network or interaction predicting model](images/concept.jpeg){#fig-concept}

![I like the use of the different source indicator items (not too dissimilar from Tall Tom's nature paper but also different). This is from @thuillerNavigatingIntegrationBiotic2024](images/thullier_2023_concept.jpeg)

**Null models:** The interactions between species occurs regardless of the identity of the species (*i.e.,* species have no agency) and links are randomly distributed throughout the network. There is however the assumption that a network will be constrained by the number of links. Type I [@fortunaHabitatLossStructure2006], where interactions happen proportionally to connectance and Type II [@bascompteNestedAssemblyPlantanimal2003], where interactions happen proportionally to the joint degree of the two species involved. These two models are equivalent to the Erdos-Renyi and Configuration models [@newmanNetworksIntroduction2010] respectively (check that though).

**Neutral models:** Based on the theory that interactions occur as the result of the abundance of species (*i.e.,* the species still has no agency but its abundance does?). See @pomeranzInferringPredatorPrey2019

**Resource models:** Based on the idea that networks follow a trophic hierarchy and that species interactions can be determined using a single dimension [the “niche axis”; @allesinaGeneralModelFood2008]. Essentially these models can be viewed as being based on the idea of resource partitioning (niches) along a one-dimensional resource and that the number of links scale with species richness (linear link scaling). That is, there is some sort of hierarchical feeding based on how a 'resource' is partitioned. Broadly this family consists of three core models; the cascade model [@cohenCommunityFoodWebs1990], which rests on the idea that species feed on one another in a hierarchical manner; the niche model [@williamsSimpleRulesYield2000], broadly all species are randomly assigned a 'feeding niche' and all species that fall in this niche can be consumed by that species; and the nested hierarchy model [@cattinPhylogeneticConstraintsAdaptation2004], which adds some component of phylogenetic clustering/signal... so not a single dimension? **TODO**. @williamsSuccessItsLimits2008 provides a broader overview of some of the variations in these models as well as comparison between them regarding their ability to retrieve elements of networks structure (see also @allesinaGeneralModelFood2008).

**Generative models:** (this is maybe a bit of a bold term to use). MaxEnt [@banvilleWhatConstrainsFood2023], (maybe) stochastic block [@xieCompletenessCommunityStructure2017].

**Feeding models:** Broadly this family of models is rooted in feeding theory and allocates the links between species based on energetics, which predicts the diet of a consumer based on energy intake. This means that the model is focused on predicting not only the number of links in a network but also the arrangement of these links based on the diet breadth of a species. The diet breadth model [@beckermanForagingBiologyPredicts2006] as well as its allometrically scaled cousin the allometric diet breadth model (ADBM) [@petcheySizeForagingFood2008] determine links between species based on the energetic content, handling time, and density of species. See also @deangelisModelTropicInteraction1975

> @gravelInferringFoodWeb2013 also poses an interesting cross-over between the adbm and niche model.

**Binary classifiers:** The task of predicting if an interaction will occur between a species pair is treated as a statistical binary classification task, where the task is to correlate 'real world' interaction data with a suitable ecological proxy for which data is more widely available (*e.g.,* traits). Model families often used include generalised linear models [*e.g.,* @caronAddressingEltonianShortfall2022], random forest [*e.g.,* @llewelynPredictingPredatorPrey2023], trait-based k-NN [*e.g.,* @desjardins-proulxEcologicalInteractionsNetflix2017], and Bayesian models [*e.g.,* @eklofSecondaryExtinctionsFood2013; @cirtwillQuantitativeFrameworkInvestigating2019]. See @pichlerMachineLearningAlgorithms2020 for a more detailed overview on the performance of machine learning and statistical approaches for inferring trait-trait relationships.

**Graph embedding:** This family of approaches has been extensively discussed in @strydomGraphEmbeddingTransfer2023 but can be broadly explained as an approach that estimates latent features from observed networks that can be used to predict interactions. @strydomFoodWebReconstruction2022 uses a transfer learning framework (specifically using a random dot product graph for embedding) based around the idea that interactions are evolutionarily conserved and that we can use known networks, and phylogenetic relationships, to predict interactions for a given species pool. **TODO** Log-ratio [@rohrModelingFoodWebs2010]

**Trait matching:** Interactions are determined by a series of 'feeding rules', whereby the interaction between a species pair will only occur if all feeding rules are met. These rules are determined on an *a priori* basis using expert/ecological knowledge to determine the underlying feeding hierarchy using ecological proxies [@morales-castillaInferringBioticInteractions2015]. For example the Paleo Foodweb Inference Model [PFIM, @shawFrameworkReconstructingAncient2024] uses a series of rules for a set of trait categories (such as habitat and body size) to determine if an interaction can occur between a species pair. What sets this family of models apart from **expert knowledge** ones is that there is a formalisation of the feeding rules and thus there is some ability to transfer these rules to different communities.

**Expert knowledge:** Not so much about empirical observations but more the value of 'local' knowledge and having specific individuals sitting around a table and assigning a value of how confident they are that a specific species pair are likely to interact [*e.g.,* @dunneCompilationNetworkAnalyses2008], this has the added advantage that interactions can be scored in a more categorical as opposed to binary fashion, *e.g.,* @maioranoTETRAEUSpecieslevelTrophic2020 score interactions as either obligate (typical food resources) or occasional (opportunistic feeding) interactions. I feel like its worth also mentioning downfalls *a la* @brimacombeShortcomingsReusingSpecies2023...

**Data scavenging:** There are also a lot of published *interaction* data that are publicly available *e.g.,* the Global Biotic Interactions (GloBI) database [@poelenGlobalBioticInteractions2014] and these can also be used to construct an interaction network by mining these sources to look for interactions for specific species pairs. This is done by matching species pairs against those within a dataset of trophic interactions to determine if an interaction is present or absent between the two species [*e.g.,* the WebBuilder tool developed by @grayJoiningDotsAutomated2015]. It is important to note that this methodology is only going to be able to infer observations that have been recorded in the field, and given the relative scarcity [*I say @poisotGlobalKnowledgeGaps2021 but that's more an overview of complete networks but one can also get pairwise interactions from these types of data so I feel like its okay?*] and localised sampling of these types of datasets it is very likely that there will be many false negatives (missing pairwise interactions) using this approach.

**Co-occurrence:** Trying to infer interactions from the co-occurrence patterns of species pairs within the community *e.g.,* the geographical lasso [@ohlmannMappingImprintBiotic2018]. This (for me) seems fundamentally flawed and @blanchetCooccurrenceNotEvidence2020 seems to agree with me at least a little bit.

| Model family       | Theory                                                                                                               | Network predicted                                        | Links predict               | Make '*de novo*' predictions (node/species identity) | Needs (minimum)                                                                                          | Assembly mechanism | Constraints                                       | Interaction   |
|--------|--------|--------|--------|--------|--------|--------|--------|--------|
| null               | Network structure is random                                                                                          | structure                                                |                             | no                                                   | network (species agnostic)                                                                               | random             | link                                              | binary        |
| neutral            | Network structure is random, but species abundance plays a role                                                      | structure                                                |                             | yes                                                  | abundance, number of links                                                                               | mass effect        | link                                              | binary        |
| resource           | Networks are interval, species can be ordered on a 'niche axis'                                                      | structure                                                | flow of biomass (resource?) | no                                                   | richness, connectance                                                                                    | 'random'           | link                                              | binary        |
| generative         | Networks are determined by their structural features                                                                 | structure                                                |                             | no                                                   | network (species agnostic)                                                                               | 'random'           |                                                   | binary        |
| energetic          | Interactions are determined by foraging theory (feeding links)                                                       | interactions                                             | flow of energy              | yes                                                  | body size                                                                                                | deterministic      | energy                                            |               |
| graph embedding    | Interactions can be predicted from the latent traits of networks                                                     | interactions                                             | potential feeding links     | yes                                                  | interactions, phylogenetic tree, list of target species (species pool)                                   |                    |                                                   | probabilistic |
| trait matching     | Interactions can be inferred by a mechanistic framework/relationships                                                | interactions                                             | potential feeding links     | yes                                                  | prior (expert) knowledge of trait hierarchy/relationships, traits, list of target species (species pool) | mechanistic        | trait matching (*sensu* forbidden links in a way) |               |
| binary classifiers | Interactions can be predicted by learning the relationship between interactions and ecologically relevant predictors | interactions                                             | potential feeding links     | yes                                                  | interactions, traits, list of target species (species pool)                                              | statistical        |                                                   |               |
| expert knowledge   | 'Boots on the ground' ecological knowledge and observations                                                          | interactions                                             | potential feeding links     | yes                                                  | list of target species (species pool)                                                                    | mechanistic        | forbidden links                                   |               |
| data scavenging    | Webscraping to create networks from online databases                                                                 | interactions                                             | potential feeding links     | no                                                   | list of target species (species pool)                                                                    |                    |                                                   | binary        |
| co-occurrence      | co-occurrence patterns arise from interactions so we can use these patterns to reverse engineer the interactions     | co-occurrence links? (or am I being a bit too mean here) | association links           |                                                      | co-occurrence (so a species list?)                                                                       |                    |                                                   |               |

: Lets make a table that gives an overview of the different model families and some of their features. *A column that captures naïve vs a priori knowledge of interactions/structure i.e., a 'parameter' of sorts?* {#tbl-history}

#### When to use what?

![I still haven't given up on a sort of venn diagram idea but maybe it going to be more of a venn-flow chart hybrid...](images/model_venn.png){#fig-venn}

![I like these schematics that Charlie Outhwaite presented at the EEB seminar (there was a series of them).](images/outhwaite_schematic.jpeg){#fig-outhwaite}

{{< embed notebooks/model_qualitative.qmd#fig-pca >}}

{{< embed notebooks/model_qualitative.qmd#fig-dendo >}}

#### Model benchmarking

-   'Testing' the performance of a model is going to depend on some of the core limitations of the model itself thus it makes sense to think of two sets benchmarking rules for network and interaction prediction models respectively (see @fig-concept panel B).

-   When it comes to network models we are concerned with the 'preservation' of structure and distribution of links across the network. For interaction models we want to ensure that we are able to retrieve interactions that really exist but also those that cannot exist (*sensu* forbidden links @jordanoSamplingNetworksEcological2016)

> "As long as these predictions are not perfect, some interactions will be predicted at the ‘wrong’ position in the network; these measures cannot describe the structural effect of these mistakes. On the other hand, measures of network structure can have the same value with interactions that fall at drastically different positions; this is in part because a lot of these measures covary with connectance, and in part because as long as these values are not 0 or their respective maximum, there is a large number of network configurations that can have the same value." - @poisotGuidelinesPredictionSpecies2023

##### Benchmarking network models

-   Maybe look at some of the historic papers that compare some of the 'resource models'

-   See also @allesinaGeneralModelFood2008 and the likelihood function that they use for model selection

-   Look at @vermaatMajorDimensionsFoodweb2009

> "Possibly, the most striking caveat of the use of summary statistics is that it cannot tell us whether or not a model is able to fully replicate empirical networks." - @allesinaGeneralModelFood2008

##### Benchmarking interaction models

-   Main concern with predicting interactions is that we want to test the 'quality' of the links we are predicting (both true positives and true negatives), but the inherit sparsity (meaning high class imbalance) means that we also need to look at the balance of these predictions.

-   "Both precision and recall may be useful in cases where there is imbalanced data. However, it may be valuable to prioritize one over the other in cases where the outcome of a false positive or false negative is costly."

-   Caveat regarding the use of real world interaction data both for training and validating predictions? *e.g.,* Poisot, Ouellet, et al. et al 2021 and Catchen et al 2023

-   See @poisotGuidelinesPredictionSpecies2023

    -   skill (ability to make the right prediction; evaluate whether low prevalence can lull us into a false sense of predictive accuracy)

    -   bias (trends towards systematically over-predicting one class)

    -   class imbalance (the relative number of cases representing interactions)

-   "These results suggest that learning from a dataset with very low connectance can be a different task than for more connected networks: it becomes increasingly important to capture the mechanisms that make an interaction exist, and therefore having a slightly more biased training dataset might be beneficial. As connectance increases, the need for biased training sets is less prominent, as learning the rules for which interactions do not exist starts gaining importance"

-   Maybe also looking at how well a model can recover 'missing links' *i.e.,* false negatives *sensu* what we did in @strydomFoodWebReconstruction2022

-   Need to discuss the key differences and implications between predicting a metaweb (*sensu* @dunneNetworkStructureFood2006) and a network realisation. Maybe also @poisotSpeciesWhyEcological2015 that discuss how the local factors are going to play a role.

{{< embed notebooks/model_quantitative.qmd#fig-topology >}}

I really like this way of plotting results from @pichlerMachineLearningAlgorithms2020

![Cool way to conceptualise results from @pichlerMachineLearningAlgorithms2020](images/pichler_result.png){#fig-pichler}

## Link network ecology to the outstanding questions in ecology 


-   Bring up the fact that delimiting a network is in and of itself fuzzy - we tend to think of them in terms of snapshots but in reality the final (empirical) network is often the result of aggregation over multiple timescales.
-   Also the fact that *some* people are concerned about the taxonomic resolution and cascading effects those might have on our understanding of network structure [@pringleResolvingFoodWebStructure2020; @pringleUntanglingFoodWebs2020], we are at risk of losing our ability to distinguish the wood from the tree - are we not (at least at times) concerned more with understanding ecosystem level processes than with needing to understand things *perfectly* at the species level.
    -   I don't think these 'rare'/nuanced links (e.g. carnivorous hippos) are going to rock the boat when we think about networks at the structural level. To say this in a different way maybe it comes down to thinking about the scale of organisation within a network... The classical levels of organisation within ecology (population, community, ...) are also relevant when we think about a networks.

-   Brief history of the development of tools within the context of the two different fields? Sort of where the theory/body of work was based and how that has changed?

-   In certain situations structure is 'enough' but there may be use cases where we are really interested in the node-level interactions *i.e.,* species identity is a thing we care about and need to be able to retrieve specific interactions at specific nodes correctly.

-   What is the purpose of generating a network? Is it an element of a bigger question we are asking, *e.g.,* I want to generate a series of networks to do some extinction simulations/bioenergetic stuff OR are we looking for a 'final product' network that is relevant to a specific location? (this can still be broad in geographic scope).

@cohenStochasticTheoryCommunity1985 states that *"\[Their\] approach is more like gross anatomy than like physiology... that is, the gross anatomy is frozen, rather than in motion."*.

Interestingly @williamsSuccessItsLimits2008 also explicitly talk about *structural* food-web models in their introduction... so how I see it that means that there has always been this inherent acknowledgement that models are functioning at a specific 'network level'.

> "The resolution of food-web data is demonic because it can radically change network topology and associated biological inferences in ways that are unknowable in the absence of better data." - @pringleResolvingFoodWebStructure2020 The counter to this is that structural models are often not working at the species level and thus the structure remains 'unchanged' when you increase the resolution - I don't think that people are that concerned with the structure of real world networks barring connectance and since that scales with species richness anyway your final proportion will probably still remain the same...

> "It makes no sense to describe the interaction structure of nodes which in themselves are poorly defined." — Roslin et al. (2013, p. 2)

## Discussion

-   I think a big take home will (hopefully) be how different approaches do better in different situations and so you as an end user need to take this into consideration and pick accordingly. I think @petcheyFitEfficiencyBiology2011 might have (and share) some thoughts on this (thanks Andrew). I feel like I need to look at @berlowGoldilocksFactorFood2008 but maybe not exactly in this context but vaguely adjacent.

-   An interesting thing to also think about (and arguably it will be addressed based on some of the other thoughts and ideas) is data dependant and data independent 'parametrisation' of the models...

-   Why do interaction models do so badly at predicting structure? Nuance of metaweb vs realisation but also time? At the core of it interaction models are trained on existing interaction data; this is data that are most likely closer to a metaweb than a local realisation even if they are being inventoried at a small scale.

    -   I think this is sort of the crux of the argument presented in @brimacombeApplyingMethodIts2024

> *"we highlight an interesting paradox: the models with the best performance measures are not necessarily the models with the closest reconstructed network structure."* - @poisotGuidelinesPredictionSpecies2023

-   *Do we need network models to predict interactions and interaction models to predict structure?* (lets not think about that too hard or I might just have to sit in silence for a while...)

    -   "Another argument for the joint prediction of networks and interactions is to reduce circularity and biases in the predictions. As an example, models like linear filtering generate probabilities of non-observed interactions existing, but do so based on measured network properties." - @strydomRoadmapPredictingSpecies2021

    -   Aligning (dove-tailing) with this the idea of ensemble modelling as presented by @beckerOptimisingPredictiveModels2022

-   It will be interesting to bring up the idea that if a model is missing a specific pairwise link but doing well at the structural level then when does it matter?

-   Close out with a call to action that we have models that predict networks very well and models that predict interactions very well but nothing that is doing well at predicting both - this is where we should be focusing our attention when it comes to furthering model development. (we need models that will fill the space in the top right quadrant of panel A in @fig-concept)

## References {.unnumbered}

::: {#refs}
:::