---
title: "Unveiling the Complexity of Food Webs: A Comprehensive Overview of Definitions, Scales, and Mechanisms"
author:
  - name: Tanya Strydom
    id: ts
    orcid: 0000-0001-6067-1349
    corresponding: true
    email: t.strydom@sheffield.ac.uk
    role: 
    - conceptualization: lead
    - methodology: supporting
    affiliation:
      - id: sheffield
        name: School of Biosciences, University of Sheffield, Sheffield, UK
  - name: Jennifer A. Dunne
    id: jad
    orcid: 0000-0002-3789-9561
    corresponding: false
    role: 
    - conceptualization: lead
    - methodology: supporting
    affiliation:
      - id: sfi
        name: Santa Fe Institute, 1399 Hyde Park Road, Santa Fe, NM 87501, USA
  - name: Timothée Poisot
    id: tp
    orcid: 0000-0002-0735-5184
    corresponding: false
    role: 
    - conceptualization: lead
    - methodology: supporting
    affiliations:
      - Université de Montreal
      - Québec Centre for Biodiversity Sciences
  - name: Andrew P. Beckerman
    id: apb
    orcid: 0000-0002-7859-8394
    corresponding: false
    role: 
    - conceptualization: lead
    - methodology: supporting
    affiliations:
      - ref: sheffield
funding: "The author(s) received no specific funding for this work. Well they did I just haven't done the homework"
keywords:
  - food web
  - network construction
  - scientific ignorance
abstract: |
  Food webs are a useful abstraction and representation of the feeding links between species in a community and are used to infer many ecosystem level processes. However, the different theories, mechanisms, and criteria that underpin how a food web is defined, and ultimately, constructed means that not all food webs are representing the same ecological process at the same scale. Here we present a synthesis of the different assumptions, scales, and mechanisms that are used to define the different ecological networks , leading to a revision of definitions for different types of networks. Additionally we explicitly link the different network representations to the broader methodological approaches (models) that are used to construct them. In explicitly outlining the assumptions, scales, and mechanisms of network inference allows for a formal categorisation of how to use networks to answer key ecological and conservation questions as wel as defining clear guidelines to prevent unintentional misuse or misinterpretation.
date: last-modified
bibliography: references.bib
number-sections: true
---

At the heart of modern biodiversity science are a set of concepts and theories about species richness, stability, and function [@loreauBiodiversityEcosystemStability2013]. These relate to the abundance, distribution, and services that biodiversity provides, and how biodiversity (as an interconnected set of species) responds to multiple stressors. Documenting interactions between and among species is thus one of the fundamental building blocks of community ecology, providing a powerful abstraction and platform for mathematical and statistical modelling of biodiversity to make predictions, mitigate threats, and manage services [@windsorUsingEcologicalNetworks2023].Such network representations of biodiversity are increasingly argued to be an asset to understanding and predicting the abundance, distribution, dynamics, and services provided by multiple species facing multiple stressors [@simmonsRefocusingMultipleStressor2021]. However, there is a growing discourse around limitations to the interpretation and applied use of networks [@dormannRisePossibleFall2023; @bluthgenWhyNetworkAnalysis2010], primarily as the result of shortcomings regarding their conceptualisation [@bluthgenCriticalEvaluationNetwork2024].

We propose that every network embeds assumptions about the process(es) that determine interactions, and about the levels of organization at which this occurs (*i.e.* the biological, ecological, spatial/temporal scale). The differences in these assumptions ultimately influence the nature and scope of inference that can be made from a given network [@proulxNetworkThinkingEcology2005]. Fundamentally, we are talking about an intersection of the data used to construct the network and the underlying theory as to what drives the occurrence of interactions between species. Although there have been extensive discussions about the challenges relating to data collection and observation [*e.g.,* @bluthgenCriticalEvaluationNetwork2024; @brimacombeShortcomingsReusingSpecies2023; @brimacombePublicationdrivenConsistencyFood2024; @moulatletScalingTrophicSpecialization2024; @pringleResolvingFoodWebStructure2020; @polisComplexTrophicInteractions1991; @saberskiImpactDataResolution2024] we still lack a clear framework framed by the different assumptions and scale dependent processes.

In this perspective we aim to provide an overview of the different **food web** representations, particularly how these relate to the terminology used to define a food web, and how this is influenced by both the processes that determine interactions [@sec-process], as well as how this relates to the way in which we construct the resulting networks [@sec-construct]. This allows us to deliver an overview of fundamental questions in ecology that we think can benefit from network thinking and a proposal that such thinking can accelerate our capacity to predict the impact of multiple stressors on biodiverse communities. Specifically, we finish this perspective with an overview of fundamental questions in ecology that we think can benefit from network thinking and a proposal that such thinking can accelerate our capacity to predict the impact of change on biodiverse communities.

# Setting the Scene: The Not So Basics of Nodes and Edges {#sec-anatomy}

Networks often have multiple uses: an 'object' from which inferences are made (*e.g.,* topological inference about biodiversity, interactions among species, and community structure, [REF]); a platform for evaluating 'downstream' responses to stressors [REF]; and a platform for evaluating mathematical and statistical models of ‘generative processes’ [REF]. Against this backdrop of multiple research agendas, it should come as no surprise that the definition of ‘edges’ and ‘nodes’, as well as the levels of organisation at which they are collated takes many forms [@poisotDescribeUnderstandPredict2016; @moulatletScalingTrophicSpecialization2024], while also encoding a series of assumptions within a network.

## How do we define a node?

Although this may seem elementary that a node should represent a (taxonomic) species, the reality is that nodes often represents non-taxonomic units such as a trophic species (*e.g.,* @yodzisCompartmentationRealAssembled1982; @williamsSimpleRulesYield2000), a feeding guild [*e.g.,* @garcia-callejasNonrandomInteractionsGuilds2023], or a segregation of species by life stages [*e.g.,* @cleggImpactIntraspecificVariation2018]. Such granularity and variation can limit the ability to make (taxonomic) species specific inferences (*e.g.,* does species $a$ eat species $b$?), affect inference made from networks, including estimates of complexity and structure [@beckermanForagingBiologyPredicts2006; @cleggImpactIntraspecificVariation2018] and make it challenging to use networks in ‘downstream analyses’, for example, of extinction or invasions. Despite these implications, there may also be value in having nodes that represent an aggregation of species, as the distribution of the links between them may be more meaningful in terms of understanding energy flow and distribution within the system.

## What is captured by an edge?

Links within food webs can be thought of as a representation of either feeding links between species (be that realised or potential [@dunneNetworkStructureFood2006; @pringleUntanglingFoodWebs2020], or fluxes within a system *e.g.,* energy transfer or material flow as the result of the feeding links between species [@lindemanTrophicDynamicAspectEcology1942].  These correspond to different ‘currencies’ (the feasibility of links or the energy that is moving between nodes). There are also a myriad of ways in which the links themselves can be specified. Links between species can be treated present or absent (i.e., binary), may be defined as probabilities [@banvilleDecipheringProbabilisticSpecies2024; @poisotStructureProbabilisticNetworks2016] or by continuous functions which further quantify the strength of an interaction [@berlowInteractionStrengthsFood2004]. How links are specified will influence the structure of the network. For example, taking a food web that consists of links representing all *potential* feeding links in a collection of species will be meaningless if one is interested in understanding the flow of energy through the network as the links are not environmentally/energetically constrained.

## Network representations {#sec-representation}

Against these definitions, networks fall into two major ‘types’: metawebs, traditionally defined as all the *potential* interactions for a specific species pool [@dunneNetworkStructureFood2006]; and realised networks, which is the subset of interactions in a metaweb that are *realised* for a specific community at a given time and place. The fundamental differences between these two network representations are the spatial scale at which they are constructed and the associated processes that are assumed to drive pattern at these scales.

A metaweb is at its core a list of feasible interactions between pairs of species. The feasibility for a given pair is derived from the complementarity (phylogenetic relationships) of their traits (representing a *global metaweb*), which can be further refined by co-occurrence (representing a *regional metaweb*). By this definition, metawebs provide a means to identify links that are not ecologically plausible, *i.e.,* forbidden links [@jordanoSamplingNetworksEcological2016], or provide an idea of the *complete* diet of a species [@strydomGraphEmbeddingTransfer2023].

In contrast realised networks are relatively localised in space and time, and the links between species are contingent on both the co-occurrence of species, the role of the environment, and mechanisms of diet choice. Fundamentally this means that the presence/absence of a link is the result of the ‘behaviour’ of the species.

This distinction between metawebs and realised webs lead to some further definitions. Links that are absent in a metaweb can conceptually (although not always practically) be treated as being truly absent. However, links that are absent in a realised network cannot be considered as truly absent but rather as absent due to the broader environmental/community context. Furthermore, a realised network is not simply the downscaling of a metaweb to a smaller scale (*e.g.,* moving from the country to the 1x1 km^2^ scale based on fine-scale species co-occurrence).  Instead, realised webs capture processes that determine the realisation of an interaction. Specifically, in realised webs, the definition of an edges shifts from being determined by feasibility to that of choices and consequences that centre around energy. Meaning if one were to take the same community of species and constructed both a metaweb and realised network the two networks might have the same species but would be structurally different, owing to the differences in the 'rules' constraining the presence of links [@caronTraitmatchingModelsPredict2024].

# From Nodes and Edges to Process and Constraints {#sec-process}

In the previous section we discussed how the definition of nodes and edges at representing different biological and ecological and processes associated with them lead to the concept of a metweb and a realised web. Here we expand this discussion, introducing five core constraints across these scales that further expose processes that determine the links among species: evolutionary compatibility, co-occurrence, abundance, diet choice, and non-trophic interactions [@fig-process].


![Aligning the various processes that determine interactions (right column) with the different network representations (left column). First, we start with a **global metaweb** this network captures all possible interactions for a collection of species in the global context. However, within the global environment different species occur in different regions (region one = yellow and region 2 = orange), and it is possible to construct two different metawebs (**regional metawebs**) for each region by taking accounting for the co-occurrence patterns of the difference species - as shown here we have two regions with some species (blue) that are found in both regions and others endemic to either region one (yellow) or region two (orange). However even within a region we do not expect that all interactions to be realised but rather that there are multiple configurations of the regional metaweb over both space and time. The 'state' of the different **realised networks** is ultimately influenced not just by the co-occurrence of a species pair but rather the larger community context such as the abundance of different species, maximisation of energy gain, or indirect/higher order interactions.](images/anatomy.png){#fig-process}

## Processes that determine the feasibility of an interaction {#sec-process-feasibility}

Here we introduce evolutionary compatibility and co-occurrence as processes that ‘act’ at the species pair of interest, that is the possibility of an interaction being present/absent is assessed at the pairwise level. 

Here we introduce evolutionary compatibility and co-occurrence as processes that ‘act’ at the species pair of interest. The scale of inference and set of processes embodied in these two constraints combine to define a ‘list’ of interactions that are viable/feasible and defined as present/absent. It is however possible to build a network from this information.  However, it is important to be aware that the structure of this network is not constrained by any community context and so just because species are able to interact does not mean that they will [@poisotSpeciesWhyEcological2015].

**Evolutionary compatibility**

This constraint is defined by shared (co)evolutionary history between consumers and resources [@segarRoleEvolutionShaping2020; @gomezEcologicalInteractionsAre2010; @dallarivaExploringEvolutionarySignature2016] which, in the more proximal sense, is manifested as 'trait complementarity' between two species [@benadiQuantitativePredictionInteractions2022]. In this body of theory, one species (the consumer) has the 'correct' (multivariate) set of traits that allow it to chase, capture, kill, and consume the other species (the resource) and interactions that are not compatible are defined as forbidden links [@jordanoSamplingNetworksEcological2016]; *i.e.,* not physically possible and will *always* be absent within a network.

Networks arising from this constraint can be binary (possible vs forbidden) or probabilistic [@banvilleDecipheringProbabilisticSpecies2024], *e.g.,* the metaweb constructed by @strydomFoodWebReconstruction2022 uses probabilities to quantify their confidence with regards to the possibility of a specific interaction existing between two species. A network constructed on the basis of evolutionary compatibility is conceptually aligned with a 'global metaweb', and gives us information as to the feasibility of links between species despite the fact that they do not co-occur (as shown in @fig-process).

**(Co)occurrence**

The co-occurrence of species in both time and space is a fundamental requirement for an interaction between two species to occur (at least in terms of feeding links). Although co-occurrence data alone is insufficient for building an accurate and ecologically meaningful representation of *feeding links* [@blanchetCooccurrenceNotEvidence2020], it is still a critical process that determines the realisation of a feeding link and allows us to spatially constrain a global metaweb based on local communities [@dansereauSpatiallyExplicitPredictions2024], in the context of @fig-process this would be the metawebs for regions one and two.

## Processes that modify the behaviour (preference) of species {#sec-process-realisation}

Here we will showcase three processes that will ultimately influence the realisation of an interaction between species and form the conceptual basis for realised networks. As we show in @fig-process a 'truly realised' network is the product of different facets of both the properties of the community (**abundance** and **non-trophic interactions**) as well as the individual (**profitability**). This represents a contextual shift where the presence (realisation) of an interaction is no longer constrained to evaluating the viability between a *pair* of species but rather takes into consideration information about the community and the individual [@quinteroDownscalingMutualisticNetworks2024], and as discussed in @sec-representation, links are now *constrained* by consumer choice.

**Abundance**

The most basic abundance constraint linked to foraging biology is the principle that organisms feeding randomly will consume resources in proportion to their abundance [@stephensForagingTheory1986], and interactions are not necessarily contingent on there being any compatibility between them [@canardEmergenceStructuralPatterns2012; @momalTreebasedInferenceSpecies2020; @pomeranzInferringPredatorPrey2019]. However, a more ecologically sound assumption would be that the abundance of different prey species will influence the distribution of links in a network [@vazquezUnitingPatternProcess2009], by influencing which prey are targeted or preferred by the predator, as abundance influences factors such as the likelihood of two species (individuals) meeting [@poisotSpeciesWhyEcological2015; @banvilleDecipheringProbabilisticSpecies2024]. Thus, if abundance data are combined with a derived metaweb, there is a basic ruleset that can define the distribution (*e.g.,* structure) and potentially the strength of links.  

**Profitability**

It is well established that consumers make more active decisions than eating items in proportion to their abundance [@stephensForagingTheory1986]. Ultimately, consumer choice is underpinned by an energetic cost-benefit framework centred around profitability and defined by traits associated with finding, catching, killing, and consuming a resource [@woottonModularTheoryTrophic2023]. Although energetic constraints can be invoked in a myriad of ways [*e.g.,* @pawarDimensionalityConsumerSearch2012; @portalierMechanicsPredatorPrey2019; @cherifEnvironmentRescueCan2024] we select profitability as a term to capture rules linked to optimal foraging [@pykeOptimalForagingTheory1984] and metabolic theory [@brownMetabolicTheoryEcology2004]; it is a sensible ‘umbrella concept’ for capturing the energetic constraint on of the distribution and strength of interactions.

**Non-trophic interactions**

Perhaps not as intuitive when thinking about the previous constraints, non-trophic interactions [@ingsEcologicalNetworksbeyondFood2009] specifically modify either the realisation or strength of trophic interactions [@golubskiModifyingModifiersWhat2011; @pilosofMultilayerNatureEcological2017; @staniczenkoStructuralDynamicsRobustness2010; @kamaruDisruptionAntplantMutualism2024]. Non-trophic interactions can modify interactions either ‘directly’ e.g., predator a outcompetes predator b; or ‘indirectly’ e.g., mutualistic/facilitative interactions. Altogether they can alter the fine-scale distribution and abundance of species as well as their persistence [@kefiMoreMealIntegrating2012; @kefiNetworkStructureFood2015; @bucheMultitrophicHigherOrderInteractions2024].

# Network construction: a case for models {#sec-construct}

## Why construct networks?

Broadly the desire to construct a network has arisen for two different purposes; building networks that can be used in real-world, applied contexts (have actionable consequences?), and building networks that allow us to interrogate, generate, and reflect upon different ecological theories. The act of constructing a 'real world' network through the empirical collection of interaction data is both costly and challenging to execute [@jordanoChasingEcologicalInteractions2016; @jordanoSamplingNetworksEcological2016], which has led to the development of a suite of modelling approaches that allow us to predict the interaction between two species, or network structure [see @strydomRoadmapPredictingSpecies2021 for a broader discussion], or identify missing interactions (gap fill) within existing empirical datasets [*e.g.,* @bitonInductiveLinkPrediction2024; @stockPairwiseLearningPredicting2021; @dallasPredictingCrypticLinks2017]. However, working with 'real-world networks' is data-hungry and cumbersome and has driven the development of models that construct ecologically plausible but often times species agnostic networks. These models often explicitly model one or a few of of the processes discussed in @sec-process and in doing so allow us to better understand the different constraints determining interactions [@stoufferAllEcologicalModels2019; @songRigorousValidationEcological2024].

## Stochastic networks

Within stochastic networks the assumption is that the interactions between species occurs irrespective of the identity of the species (*i.e.,* species have no agency) and links are *randomly* distributed throughout the network. Alternatively it can be assumed that interactions are still independent of a species' trait but are rater driven by its abundance, *sensu* neutral theory [@hubbellUnifiedNeutralTheory2001].

Practically there is little support that networks are truly stochastic, however null models [*e.g.,* @fortunaHabitatLossStructure2006; @bascompteNestedAssemblyPlantanimal2003] are often used as a 'null hypothesis' that can be used to ask questions about deviations in observed network structure [*e.g.,* @banvilleWhatConstrainsFood2023; @strydomSVDEntropyReveals2021]. Conversely, neutral models [@canardEmpiricalEvaluationNeutral2014; @krishnaNeutralnicheTheoryNestedness2008], and by extension processes, are most likely relevant at local scales [@pomeranzInferringPredatorPrey2019], *i.e.,* should be considered to play a role when constructing realised networks but are superseded by trait-based processes at a metaweb scale.

## Feasibility networks (metawebs)

Metawebs (depending on the aggregation) can help us develop our understanding of the intersection of species interactions and their co-occurrence[@soberonGrinnellianEltonianNiches2007; @gravelBringingEltonGrinnell2019]. Whereby a *global metaweb* presents an approximation of the fundamental Eltonian niche of a species (*i.e.,* its relation to its food source), whereas as *regional metawebs* represent an intersection of Elton and Grinnell. As discussed in @sec-process-feasibility the feasibility of an interaction is typically assessed on a pairwise basis, and is often assessed based on the idea that interactions are governed by a set of 'feeding rules' [@morales-castillaInferringBioticInteractions2015], and are broadly elucidated in two different ways; *mechanistic models*, [*e.g.,* @shawFrameworkReconstructingAncient2024; @dunneCompilationNetworkAnalyses2008; @roopnarineEcologicalModellingPaleocommunity2017] and *pattern finding models* [*e.g.,* @strydomGraphEmbeddingTransfer2023; @pichlerMachineLearningAlgorithms2020; @strydomFoodWebReconstruction2022; @caronAddressingEltonianShortfall2022; @llewelynPredictingPredatorPrey2023; @desjardins-proulxEcologicalInteractionsNetflix2017; @eklofSecondaryExtinctionsFood2013; @cirtwillQuantitativeFrameworkInvestigating2019]. The fundamental difference between these two model groups is that *mechanistic models* rely on expert knowledge and make explicit assumptions on trait-feeding relationships, whereas the *pattern finding models* are dependent on existing interaction datasets from feeding rules can be elucidated. It perhaps also bears repeating that these models are often only presenting a list of feasible interactions and that the resulting network is 'unstructured', as it is uconstrained by any processes or conditions that generate structure. While these networks can be imprinted with external definitions of trophic position and guild identity to deliver hypothetical structure, this structure is not an emergent property of the links and species pairs [@caronTraitmatchingModelsPredict2024].

Feasibility networks are useful for determining all feasible interactions for a specific community, and the models that have been developed in this context have the potential to allow us to construct first draft networks for communities for which we have no interaction data [@strydomFoodWebReconstruction2022], and are valuable not only in data poor regions but also for predicting interactions for 'unobservable' communities *e.g.,* prehistoric networks [@yeakelCollapseEcologicalNetwork2014; @frickeCollapseTerrestrialMammal2022; @dunhillExtinctionCascadesCommunity2024] or future, novel community assemblages. Conceptually this is particularly valuable if we want to understand interactions between novel communitites, as well as the rewiring capacity of species. Additionally, an understanding of the role of interactions between species has allowed us to better determine the distribution of a species by accounting not only for the role of the environment but also the role of species interactions [@higinoMismatchIUCNRange2023; @pollockUnderstandingCooccurrenceModelling2014].

## 'Behavioural' networks

Ultimately realised networks and capture some aspect of how the behavior of a species determines if a link is realised or not and can be modelled in two ways; models that predict realised interactions (whereby he behaiour of a secies is modelled *i.e.,* its diet choice), and models that predict the structure of realised networks (whereby the behvaiour of the system is modelled and assumptions are made with regards to the structure of a network). In terms predicting interactions current models are rooted in feeding theory and allocate the links between species based on energy *e.g.,* diet models [@beckermanForagingBiologyPredicts2006; @petcheySizeForagingFood2008] have been used construct networks based on both profitability (as determined by the handling time, energy content, and predator attack rate) as well as abundance (prey density). [@woottonModularTheoryTrophic2023]. At a 'coarser', funtional level there are models that are based on the compartmentation and aquisition of energy for species at different trophic levels [@allesinaFoodWebModels2009; @krauseCompartmentsRevealedFoodweb2003]. Models that determine structure are based on the idea that networks follow a trophic hierarchy and that network structure can be determined by distributing interactions along single dimension [the “niche axis”; @allesinaGeneralModelFood2008], while parametrising an aspect of the network structure [although see @allesinaFoodWebModels2009 for a parameter-free model].

As behavioural networks are are build on the concept of dynamic processes (*e.g.,* the abundance of species will always be in flux) these networks are valuable for understanding the behaviour of networks over time, or their response to change [@lajaaitiEcologicalNetworksDynamicsJlJulia2024; @delmasSimulationsBiomassDynamics2017; @curtsdotterEcosystemFunctionPredator2019]. However, they are 'costly' to construct (requiring data about the entire community, as it is the behaviour of the system that determines the behaviour of the part) and also lack the larger diet niche context afforded by metawebs. Structural models provide a data-light (the models often only require species richness) but assumption heavy (the resulting network structure is determined by an assumption of network structure) alternative, however they do not make species specific predictions and so cannot be used to determine if an interaction is either possible *or* realised between two species (*i.e.,* one cannot use these models to determine if species $a$ eats species $b$). Although this means this suite of models are unsuitable as tools for predicting species-specific interactions, they have been shown to be sufficient tools to predict the structure of networks [@williamsSuccessItsLimits2008], and are useful in synthetic simulations.

# Making Progress with Networks {#sec-progress}

It is probably both this nuance as well as a lack of clear boundaries and guidelines as to the links between network form and function [although see @delmasAnalysingEcologicalNetworks2019] that has stifled the 'productive use' of networks beyond the inventorying the interactions between species. Although progress with using networks as a means to address questions within larger bodies of ecological theory *e.g.,* invasion biology [@huiHowInvadeEcological2019] and co-existence theory [@garcia-callejasNonrandomInteractionsGuilds2023] has been made we still lack explicit guidelines as to what the appropriate network representation for the task at hand would be, and as highlighted in Box 1, underscores the need to evaluate exactly what process a specific network representation captures as well as its suitability for the question of interest. Below we present a mapping of what we believe are some of the key questions for which interaction networks can be used to the different networks representations that are most suitable, as well as highlight some of the methodological challenges that still need to be improved upon.

## Making use of the different network representations

**Methodological challenges**

1.  Tools that allow us to estimate both the feasibility as well as realisation of links: Currently most approaches to modelling relaised networks fail to explicitly account for any form of evolutionary constraint [although see @vandewalleArthropodFoodWebs2023 and @woottonModularTheoryTrophic2023] and we need to develop either an ensemble modelling approach [@beckerOptimisingPredictiveModels2022; @terryFindingMissingLinks2020] or tools that will allow for the downsampling of metawebs into realised networks [*e.g.,* @roopnarineExtinctionCascadesCatastrophe2006].
2.  Is there something in generalisable models that 'combine' different processes/aspects (*e.g.,* using body size as a catch all) versus limited models that allow you to unpack things bit-by-bit (*i.e.,* process by process). So @woottonModularTheoryTrophic2023 *may* (TBD) span the gamut but it lacks the ability to unpack... Although myabe the terms do?
3.  Modelling interaction strength: Although realised networks are more closely aligned with *explicitly* capturing interaction strength we lack models that allow us to quantify this [@wellsSpeciesInteractionsEstimating2013; @strydomRoadmapPredictingSpecies2021].
4.  How do we validate our predictions?: Progress has been made to assess how well a model recovers pairwise interactions [@strydomRoadmapPredictingSpecies2021; @poisotGuidelinesPredictionSpecies2023], but we still lack clear set of guidelines for benchmarking the ability of models to recover structure [@allesinaGeneralModelFood2008]
5.  Something about making what we do with networks more tractablie in the applied space? *e.g.,* @dansereauOvercomingDisconnectInteraction2024

**Theory challenges**

1.  Core Theory Advancement: Do the decades of insights arrived at for stability-diversity-productivity relationships with tri-trophic or diamond shaped models hold for complex communities (10’s-100s) [@danetResponseDiversityMajor2024]; How will spatial and temporal variation in climate and productivity drive change in complex ecosystems. Necessary to move to predicting changes in biodiversity per se, ecosystem functions and identifying sensitive and robust species and portions of communities.
2.  How will novel communities interact? How will range shifts and invasions result in new/novel community assemblages. And then also the intentional changes of species compositions through rewilding.
3.  Does rewiring happen and does it deliver robustness? Specific sub points to consider here is persistence, especially persistence to perturbation. Again, dynamic networks and network/community assembly and finally extinctions [@dunhillExtinctionCascadesCommunity2024].
4.  When do invasive species enhance or decimate communities? When do reintroductions work? [@woosterAustraliasRecentlyEstablished2024]
5.  Are there temperature threshold to community collapse
6.  Can socioeconomic networks combined with biological networks drive understanding of externalities?
7.  Can paleoecological data from deep time hyperthermal events provide sufficient insight into the targets, pace and recovery times from rapid climate events?

![Here we highlight some of the outstanding questions in both network as well as general ecology, as well as some of the outstanding methodological challenges with regards to constructing food webs (shown in orange) that we are faced with.](images/NetworkFuture.png){#fig-future}

# Concluding remarks

Having a clear understanding of the interplay between network representations and the processes that they are capable of encoding is critical if we are to understand exactly which networks can be used to answer which questions. As we highlight in Box 1 the different network representations have different potential uses and it should be clear that there is no 'best' network representation but rather a network representation that is best suited to its intended purpose. In providing a formalisation regards to the assumptions and mechanisms that need to be explicitly taken into consideration when deciding to use (and construct) networks we hope to prevent the unintentional misuse or misinterpretation of networks as well as provide a starting point from which we can develop a better framework for the applied use of networks to answer questions that are not only pressing within the field but also within broader biodiversity science.

# References {.unnumbered}

::: {#refs}
:::