---
title: "Unveiling the Complexity of Food Webs: A Comprehensive Overview of Definitions, Scales, and Mechanisms"
author:
  - name: Tanya Strydom
    id: ts
    orcid: 0000-0001-6067-1349
    corresponding: true
    email: t.strydom@sheffield.ac.uk
    role: 
    - conceptualization: lead
    - methodology: supporting
    affiliation:
      - id: sheffield
        name: School of Biosciences, University of Sheffield, Sheffield, UK
  - name: Jennifer A. Dunne
    id: jad
    orcid: 0000-0002-3789-9561
    corresponding: false
    role: 
    - conceptualization: lead
    - methodology: supporting
    affiliation:
      - id: sfi
        name: Santa Fe Institute, 1399 Hyde Park Road, Santa Fe, NM 87501, USA
  - name: Timothée Poisot
    id: tp
    orcid: 0000-0002-0735-5184
    corresponding: false
    role: 
    - conceptualization: lead
    - methodology: supporting
    affiliations:
      - Université de Montreal
      - Québec Centre for Biodiversity Sciences
  - name: Andrew P. Beckerman
    id: apb
    orcid: 0000-0002-7859-8394
    corresponding: false
    role: 
    - conceptualization: lead
    - methodology: supporting
    affiliations:
      - ref: sheffield
funding: "The author(s) received no specific funding for this work. Well they did I just haven't done the homework"
keywords:
  - food web
  - network construction
  - scientific ignorance
abstract: |
  Food webs are a useful abstraction and representation of the feeding links between species in a community and are used to infer many ecosystem level processes. However, the different theories, mechanisms, and criteria that underpin how a food web is defined and, ultimately, constructed means that not all food webs are representing the same ecological process. Here we present a synthesis of the different assumptions, scales and mechanisms that are used to define different ecological networks ranging from metawebs (an inventory of all potential interactions) to fully realised networks (interactions that occur within a given community over a certain timescale). Illuminating the assumptions, scales, and mechanisms of network inference allows a formal categorisation of how to use networks to answer key ecological and conservation questions and defines guidelines to prevent unintentional misuse or misinterpretation.
date: last-modified
bibliography: references.bib
citation:
  container-title: TREE (one can dream...)
number-sections: true
---

At the heart of modern biodiversity science are a set of concepts and theories about biodiversity, stability and function. These relate to the abundance, distribution and services that biodiversity provides, and how biodiversity – as an interconnected set of species – responds to multiple stressors. The interaction between species (or individuals) is one of the fundamental building blocks of ecological communities provide a powerful abstraction that can help quantify, conceptualise, and understand biodiversity dynamics, and ultimately, one hopes, make prediction, mitigate change and manage services \[ref\]. Such network representations of biodiversity (including within species diversity) are increasingly argued to be an asset to predictive ecology, climate change mitigation and resource management. Here, it is argued that characterising biodiversity in a network will allow deeper capacity to understand and predict the abundance, distribution, dynamics and services provided by multiple species facing multiple stressors.

However, the way that a network is constructed (encoded) defines an epistemology of the network concept which, we argue, can influence the resulting observations and conclusions about pattern and mechanisms that are made [@proulxNetworkThinkingEcology2005; @brimacombeShortcomingsReusingSpecies2023]. This process of constructing networks has two major pillars: the data and theory, the latter representing an expression of mechanism and process giving rise to patterns that emerge from collating interactions among species. Each of these pillars carries with it a set of practical, semantic and conceptual constraints that not only influence progress in making network ecology more valuable and potentially predictive, but help define the spatial, temporal and evolutionary scale of assumptions we make and predictions we might generate from the networks.

With respect to data, it is extremely challenging to actually record species interactions in the field [@jordanoChasingEcologicalInteractions2016; @jordanoSamplingNetworksEcological2016]. Despite notable herculean efforts (**Woodward? Benguela?** @maioranoTETRAEUSpecieslevelTrophic2020), actual coverage of ‘real world’ interaction data remains sparse [@poisotGlobalKnowledgeGaps2021]. Against this practical challenge, there is additionally high variance in the terminology we use to define networks. Finally, the mathematical and statistical tools we use to construct, conceptualise, analyse and predict with these networks are also highly variable.

1.  what are the underlying assumptions about nodes, edges, scale and process that are made when we attempt to delimit and describe a food webs;
2.  are there families of commonly used tools that map onto assumptions about scales and processes;

The provision of this detail ultimately leads to a set of insights and conclusions about whether, when and under what conditions network representations of biodiversity can contribute to the advancement of ecological theory and generate value in predictive ecology. Specifically, we finish this perspective with an overview of fundamental questions in ecology that we think can benefit from network thinking and a proposal that such thinking can accelerate our capacity to predict the impact of multiple stressors on biodiverse communities.

# Setting the Scene: The Not So Basics of Nodes and Edges {#sec-anatomy}

Defining a food web seems simple; it is the representation of the interactions (edges) between species (nodes), however the definition of 'edges' and 'nodes', as well as the scale at which they are aggregated can take many forms [@poisotDescribeUnderstandPredict2016], which ultimately encodes a series of assumptions and criteria within a network. An awareness of variance in the way a food web can be defined is critical as a network (or its adjacency matrix) is both the 'object' from which inferences are made (*e.g.,* the interactions between species, or how the structure influences ecosystem level processes) as well as the 'product' of either the data collection [@brimacombeShortcomingsReusingSpecies2023] or prediction process [@banvilleDecipheringProbabilisticSpecies2024]. One thus needs to be aware of both the criteria that is used to define nodes and edges, and what processes or mechanisms the aggregation of the two represents, as this will determine what the network can be used for.

### How do we define a node?

Although this may seem an elementary question in the context of food webs --- a node *should* represent a (taxonomic) species, the reality is that nodes can often represent an aggregation of different species - so called 'trophic species' [@yodzisCompartmentationRealAssembled1982; @williamsSimpleRulesYield2000] or segregation of species by life stages [@cleggImpactIntraspecificVariation2018]. Practical implications of how we are aggregating the nodes is that the resolution may not always be 'pixel perfect', which limits the ability to make (taxonomic) species specific inferences *e.g.,* does species $a$ eat species $b$, however there is value in having nodes that represent an aggregation of species, as the distribution of the links between them are more meaningful in terms of understanding energy flow and distribution within the system.

### What is meant by an edge?

At its core, links within food webs can be thought of as a representation of either feeding links between species - be that realised [@pringleUntanglingFoodWebs2020] or potential [@dunneNetworkStructureFood2006], or representative of fluxes within the community/system *e.g.,* energy transfer or material flow [@lindemanTrophicDynamicAspectEcology1942]. How we specify links will influence the resulting structure of the network - and the inferences we will make thereof. For example taking a food web that consists of links representing all *potential* feeding links for a community (*i.e.,* a metaweb) will be meaningless if one is interested in understanding the flow of energy through the network as the links within a metaweb do not represent environmental/energetic constraints, making them poor representations of which interactions are *realised* in a specific location [@caronTraitmatchingModelsPredict2024]. In addition to the various ways of defining the links between species pairs there are also a myriad of ways in which the links themselves can be quantified. Links between species are often treated as being present or absent (*i.e.,* binary) but it is also possible to use probabilities [which quantifies how likely an interaction is to occur, @poisotStructureProbabilisticNetworks2016; @banvilleDecipheringProbabilisticSpecies2024] or continuous measurements [which quantifies the strength of of an interaction, @berlowInteractionStrengthsFood2004].

### Network representations

Broadly, networks can be thought of to fall into two different 'types'; namely metawebs; traditionally defined as all of the *potential* interactions for a specific species pool [@dunneNetworkStructureFood2006], and realised networks; which is the subset of interactions in a metaweb that are *realised* 'on the ground'. The fundamental difference between these two different types of networks is that a metaweb provides insight as to the viability of an interaction between two species occurring and is a means to identify links that are not ecologically plausible, *i.e.,* forbidden links [@jordanoSamplingNetworksEcological2016], or an idea of the *complete* diet of a species [@strydomGraphEmbeddingTransfer2023]. Although metawebs are typically 'constrained' to a collection of species that also co-occur, there is no reason that a metaweb cannot include species that do not co-occur (although this would require some degree of prediction/assumptions to identify those possible interactions). In contrast realised networks are highly localised and contingent on both the co-occurrence of species as well as the influence of the environment, and population and community dynamics on predator choice. In the context of definitions and semantics the links that are represented by a metaweb and a realised network are different; links that are absent in a metaweb can be treated as being truly absent, however links that are absent in a realised network cannot be considered to be truly absent but are rather as absent due to the broader environmental/community context. Importantly, a realised network is *not* simply the downscaling of a metaweb to a smaller scale (*e.g.,* moving from the country to the 1x1 km^2^ scale based on fine-scale species co-occurrence) but represents a shift towards capturing the higher level processes that determine the *realisation* of an interaction. Thus, different network representations are determined and constrained by different sets of assumptions as to what the processes are that determine the presence/absence of an interaction between two species as well as the resulting network structure.

# From Nodes and Edges to Scales, Context, and Processes {#sec-process}

The interplay between network representation and network definition is primarily governed by the process(es) that determine the interaction between species, however these processes are also scale and context dependent. Here we start by introducing the five core processes that determine either the feasibility or realisation of interactions, namely: evolutionary compatibility, co-occurrence, feasibility, abundance, predator choice, and non-trophic interactions; while simultaneously contextualising them within, and linking them to the different network representations [@fig-process]; specifically if the processes captures an all-or-nothing (possibility) vs context dependent (likelihood) determination of interactions between species. Of course these processes do not function in a vacuum and do interact with/influence one another, but it is still beneficial to present them in a categorical manner as these different processes are often the underpinning logic in the development of prediction/network models, the criteria for data collection in the field, and the scale of organisation for which they are relevant (species, population, or community).

![TODO.](images/concept_v2.png){#fig-process}

## The processes that determine species interactions

**Evolutionary compatibility**

There is compelling evidence that the possibility of an interaction occurring between two species is the result of their shared (co)evolutionary history [@segarRoleEvolutionShaping2020; @gomezEcologicalInteractionsAre2010; @dallarivaExploringEvolutionarySignature2016]. In the more proximal sense this is manifested as the 'trait complementarity' between two species, whereby one species (the predator) has the 'correct' set of traits that allow it to chase, capture, kill, and consume the other species (the prey). For species pairs where this condition is not met the link is deemed to be forbidden [@jordanoSamplingNetworksEcological2016]; *i.e.,* not physically possible and will always be absent within the network. In the context of trying to determine the feasibility (*i.e.,* the *possibility*) of an interaction, phylogeny is an excellent predictor [@strydomFoodWebReconstruction2022; @frickeCollapseTerrestrialMammal2022] and allows one to construct what can be considered to be a metaweb. In terms of thinking about the anatomy of an 'feasibility network' one should be aware that it is possible to represent interactions as either binary (feasible/forbidden; *i.e.,* the traditional definition of a metaweb @dunneNetworkStructureFood2006) or as a probability [@banvilleDecipheringProbabilisticSpecies2024], where the probability represents how likely that the interaction between to species is feasible (what is the possibility of this interaction occurring?).

**(Co)occurrence**

Although the outright assumption that because two species are co-occurring it must mean that they are interacting is inherently flawed [@blanchetCooccurrenceNotEvidence2020], it is of course impossible for two species to interact (at least in terms of feeding links) if they are not co-occurring in time and space. Thus co-occurrence data alone is insufficient to build an accurate and ecologically meaningful representation of a food web having information on the co-occurrence of species can further aid us in refining metawebs by allowing us to downsample the network based on the species found in a specific location, or even add additional uncertainty based in how likely species are to co-occur [@dansereauSpatiallyExplicitPredictions2024]. Additionally the interplay between the interaction between a species pair and their co-occurrence is meaningful when one is operating in the space of trying to determine the distribution of a species [@higinoMismatchIUCNRange2023], and forms a key component of some of the next generation species distribution models *e.g.,* joint SDMs [@pollockUnderstandingCooccurrenceModelling2014].

**Abundance**

The abundance of the different species within the community can influence the likelihood of an interaction occurring in a myriad of ways *e.g.,* the likelihood of two species meeting, or the long term availability of prey. There is the argument that structure of networks (and the interactions that they are composed of) are driven by only the abundance of the different species and not their characteristics (traits), *sensu* neutral processes [@canardEmergenceStructuralPatterns2012; @momalTreebasedInferenceSpecies2020]. Alternatively the abundance of species in a community can influence which interactions are ultimately realised [@poisotSpeciesWhyEcological2015; @banvilleDecipheringProbabilisticSpecies2024].

**Predator choice (energetic cost)**

Ultimately, predator choice is underpinned by the energetic cost-benefit of trying to catch, kill, and consume prey, and is well described within both optimal foraging theory \[ref\] and metabolic theory \[ref\], which rests on the idea that the prey a predator chooses to target is one that will have the greatest return on energy with the lowest energetic cost. With a body of evidence that suggests that body size might be the underlying driver, and thus suitable proxy for understanding these processes [@yodzisBodySizeConsumerResource1992] There are additional bodies of work that attempt to include the cost of movement that the environment imposes on an individual [@cherifEnvironmentRescueCan2024] as well as 2D/3D search space [@pawarDimensionalityConsumerSearch2012].

**Indirect/higher order interactions**

The realisation (presence/absence) or strength of trophic interactions themselves can also be modified by other, indirect (non-trophic), interactions [@golubskiModifyingModifiersWhat2011; @pilosofMultilayerNatureEcological2017], this can be either 'directly' through *e.g.,* competition or 'indirectly' *e.g.,* mutualistic/facilitative interactions will alter the fine-scale distribution and abundance of some species [@kefiMoreMealIntegrating2012; @kefiNetworkStructureFood2015] as well as persistence [@bucheMultitrophicHigherOrderInteractions2024].

## Contextualising the processes that determine species interactions

It should be self evident that the different processes discussed above are all ultimately going to influence the realisation of interactions as well as the structure of a network, however they are acting at different scales of organisation. Both the **co-occurrence** and the **evolutionary compatibility** are valid at the scale of the species pair of interest, that is the *possibility* of an interaction being present/absent is assessed at the pairwise level and one is left with a 'list' of interactions that are present/absent. Although it is possible to build a network (*i.e.,* metaweb) from this information it is important to be aware that the structure of this network is not constrained by real-world dynamics or conditions (*i.e.,* community context), and so just because species are able to interact does not mean that they will [@poisotSpeciesWhyEcological2015]. In order to construct a network who's structure is a closer approximation of reality (localised interactions) one needs to take into consideration properties of the community as a whole and not just the two species of interest, which requires more data at the community scale, such as the abundance of species.

*something about 'physical'/landscape scale as well as time scale??*

# Network construction is nuanced

The act of constructing a 'real world' network will ultimately be delimited by its intended use, however the reality is that the empirical collection of interaction data is both costly \[ref\] and challenging to execute in a way that captures the different processes (owing to the different time and spatial scales they may be operating at). Thus we often turn to models to either predict networks, be that the interaction between two species, or its structure [@strydomRoadmapPredictingSpecies2021], or as a means to identify missing interactions (gap fill) within existing empirical dataset [@bitonInductiveLinkPrediction2024; @stockPairwiseLearningPredicting2021], and so for the purpose of this discussion network construction will be synonymous with using a model as a means to represent or predict a network --- it can be argued that even the collection of empirical data is in and of itself a 'model' as it is still only a *representation* of the system. Different models have different underlying philosophies that often only capture one or a few of the processes discussed in @sec-process, has implications for how the resulting network is defined [@sec-anatomy], which will ultimately delimit and define what inferences can be made from the resulting network. Here we will introduce the three different types of network representations, how they link back to the different processes determining interactions [@fig-process], and broadly discuss some of the modelling approaches that are used to construct these different network types. This is paralleled by a hypothetical case study (Box 1) where we showcase the utility/applicability of the different network representation in the context of trying to understand the feeding dynamics of a seasonal community.

::: {#box-hypothetical .callout-note}
# Box 1 - Why we need to aggregate networks at different scales: A hypothetical case study {.unnumbered}

Although it might seem most prudent to be predicting, constructing, and defining networks that are the closest representation of reality there are pros and cons of constructing both realised networks as well as metawebs. Let us take for example a community across time/through seasons. In this community we expect species to be either present or absent depending on the season (*i.e.,* changes in co-occurrence) as well as some species exhibiting seasonal diet shifts, these details would be lost at the scale of the metaweb an it would be valuable to construct either smaller metawebs for the different seasonal communities (thereby capturing the changes in community diversity), or realised networks for each season (to capture diet or ecosystem process shifts). However, these small-scale networks lack the context of the bigger picture that is available at the metaweb - that is it gives us a more holistic idea of the entire diet range of a specific species, which is important when one needs to make conservation-based/applied decisions (*e.g.,* conserving the entire diet of a species and not just seasonal prey items) as well as providing information on interactions that may be possible regardless of the environmental/community context (species may have the capacity to consume certain prey items but do not do so due to local conditions). With this is mind let us see how the different network aggregations can be used

**1: A global metaweb**

Knowledge of the entire diet breadth of a species is valuable especially in terms of understanding how a species will respond to changes in the community - *e.g.,* invasions/rewilding exercises (where does the new species 'fit' within the network?) as well as potential capacity to shift its diet. ALthough this might make sense across space and not time but certain species act as links across the landscape \[Rooney\]

**2: A seasonal metaweb**

Knowledge at the finer scale is also valuable to understand/identify that there are in fact differences between the seasons

**3: A seasonal realised network**

Dynamics are useful because they are a representation of the different configurations/energy flows/ecosystem processes. Also to detect more nuanced shifts in diet - *e.g.,* seasonal diet shifts.

**4: A structural network**

**Data trade off**

Above we highlight the practical uses of the different network configurations but we also need to take into consideration the barriers to construction/associated data needs/cost and acknowledge them. Basically in the ideal world we would have all this information at hand but in reality we might be sitting with seasonal metawebs...
:::

## How do we predict food webs?

There as many ways to predict networks as what there is to define them and along with taking into consideration the points raised in the previous section it is also beneficial to think about the context in which the different models were developed - and how this will influence the networks that they produce...

There is a bit of a 'point of conflict' between those calling for 'pixel perfect', regional scale data [@pringleUntanglingFoodWebs2020; @pringleResolvingFoodWebStructure2020] and for the means to generate networks that are ecologically plausible *representations* (*sensu* structural networks). This represents two challenges; one is that models that represent generalisations of networks often lack the ability to retrieve any species/community specificity which limits their utility for real world, species-driven scenarios *e.g.,* species driven conservation efforts [@dunnSixthMassCoextinction2009], however networks that are constructed through either (most) empirical observations or through predictive means are fundamentally going to represent metawebs, *i.e.,* lack constrained links, a representation of structure, or energy flow...

### Models that predict metawebs (feasible interactions)

This is perhaps the most developed group of models; with a variety of approaches having been developed that typically determine the feasibility of an interaction based on the trait compatibility between predator and prey (*i.e.* their evolutionary compatibility) to determine 'feeding rules' [@morales-castillaInferringBioticInteractions2015]. These feeding rules are broadly elucidated in two different ways; mechanistic feeding rules can be explicitly defined and applied to a community [*e.g.,* @shawFrameworkReconstructingAncient2024; @dunneCompilationNetworkAnalyses2008] or they are inferred from a community for which there is interaction data and the 'rules' are then applied to a different community [*e.g.,* @strydomGraphEmbeddingTransfer2023; @pichlerMachineLearningAlgorithms2020; @strydomFoodWebReconstruction2022; @caronAddressingEltonianShortfall2022; @llewelynPredictingPredatorPrey2023; @desjardins-proulxEcologicalInteractionsNetflix2017; @eklofSecondaryExtinctionsFood2013; @cirtwillQuantitativeFrameworkInvestigating2019]. The fundamental difference between these two model groups is that 'mechanistic models' rely on expert knowledge and make assumptions on trait-feeding relationships, whereas the 'pattern finding' models are dependent on existing datasets from which to elucidate feeding rules. These models are useful for determining all feasible interactions for a specific community, and owing to the availability of datasets [*e.g.,* @poelenGlobalBioticInteractions2014; @poisotMangalMakingEcological2016; @grayJoiningDotsAutomated2015], as well as the development of model testing/benchmarking tools [@poisotGuidelinesPredictionSpecies2023], means that these models can be validated and (with relative confidence) be used to construct first draft networks for communities for which we have no data [@strydomFoodWebReconstruction2022], and are valuable for constructing networks where we lack any interaction data *e.g.,* prehistoric networks [@yeakelCollapseEcologicalNetwork2014; @frickeCollapseTerrestrialMammal2022].

### Models that predict realised networks (realised interactions)

In order to construct realised networks models need to incorporate *both* the feasibility of interactions (*i.e.,* determine the entire diet breadth of a species) as well as then determine which interactions are realised (*i.e.,* incorporate the 'cost' of interactions). As far as we are aware there is no model that explicitly accounts for both of these 'rules' and rather *only* account for processes that determine the realisation of an interaction (*i.e.,* abundance, predator choice, or non-trophic interactions). Although the use of allometry *i.e.,* body size [*e.g.,* @valdovinosBioenergeticFrameworkAboveground2023; @beckermanForagingBiologyPredicts2006] may represent a first step in capturing 'evolutionary compatibility' alongside more energy (predator choice) driven processes accounting for additional traits is still needed [*e.g.,* @vandewalleArthropodFoodWebs2023 show how incorporating prey defensive properties alongside body size improves predictions]. In terms of models that do formalise these processes, diet models [@beckermanForagingBiologyPredicts2006; @petcheySizeForagingFood2008] have been used construct networks based on both predator choice (as determined by the handling time, energy content, and predator attack rate) as well as abundance (prey density). @woottonModularTheoryTrophic2023 developed a model that moves the energy of the system into different modules related to the process of the predator acquiring energy from the prey *i.e.,* compartmentation in food webs [@krauseCompartmentsRevealedFoodweb2003].

### Models that predict structure (interaction agnostic)

Although we identify mechanisms that determine species interactions in @sec-process not all models that are used to predict networks explicitly operate at the 'process' level, but rather represent the *structure* of a network based on a series of *a priori* assumptions as to the distribution of links between species (typically trophic not taxonomic species) by parametrising an aspect of the network structure, (*e.g.,* the niche model [@williamsSimpleRulesYield2000] makes an assumption as to the expected connectance of the network,although see @allesinaFoodWebModels2009 for a parameter-free model) or alternatively uses structural features of an exiting *realised* network (*e.g.,* stochastic block model, @xieCompletenessCommunityStructure2017). Importantly these structural models do not make species specific predictions (they are usually species agnostic and treat nodes as trophic species) and so cannot be used to determine if an interaction is either possible *or* realised between two species (*i.e.,* one cannot use these models to determine if species $a$ eats species $b$). Although this means this suite of models are unsuitable as tools for predicting species-specific interactions, they have been shown to be sufficient tools to predict the structure of networks [@williamsSuccessItsLimits2008], and provide a data-light (the models often only require species richness) but assumption heavy (the resulting network structure is determined by an assumption of network structure) way to construct a network.

# Making Progress with Networks

## Further development of models and tools

There has been a suite of models that have been developed to predict trophic links, however we are lacking in tools that are explicitly taking into consideration estimating both the feasibility as well as realisation of links, *i.e.,* both interactions and structure simultaneously [@strydomRoadmapPredictingSpecies2021]. This could be addressed either through the development of tools that do both (predict both interactions and structure), or to develop an ensemble modelling approach [@beckerOptimisingPredictiveModels2022]. Alternatively the development of tools that will allow for the downsampling of metawebs into realised networks [*e.g.,* @roopnarineExtinctionCascadesCatastrophe2006], although deciding exactly what is driving differences between local networks and the regional metaweb might not be that simple [@saraviaEcologicalNetworkAssembly2022]. Probably also something that aligns with trying to predict interaction strength - because that would be the gold standard [*e.g.,* @wellsSpeciesInteractionsEstimating2013]. Probably also worth just plainly stating that feasibility of developing a model that is both broadly generalisable, but also has local specificity is probably not attainable [@stoufferAllEcologicalModels2019], and more specifically the potential use of models to untangle/identify the different processes that shape interaction networks [@songRigorousValidationEcological2024], *e.g.,* @curtsdotterEcosystemFunctionPredator2019 showcase the use of models to disentangle the drivers of community function and @strydomSVDEntropyReveals2021 who identified that networks are less complex than they could be, suggesting that there are constraints on network assembly. In addition to the more intentional development of models we also need to consider the validation of these models, there have been developments and discussions for assessing how well a model recovers pairwise interactions [@strydomRoadmapPredictingSpecies2021; @poisotGuidelinesPredictionSpecies2023], although the rate of false-negatives that may be present in the testing data still present a challenge [@catchenMissingLinkDiscerning2023], and we still lack clear strategies for benchmarking the ability of models to recover structure [@allesinaGeneralModelFood2008].

## At what scale should we be predicting and using networks?

Look at @hutchinsonSeeingForestTrees2019

We lack a clear agenda (and conceptualisation) as to what the appropriate level of aggregation is for a 'network'. Realistically most empirical networks are more aligned with metawebs as opposed to realised networks as they are often the result of some sort of aggregation of observations across time, this creates a two-fold problem. Firstly, we need to think about how this affects any sort of development of theory that sits closer to the 'realised network' side of the spectrum - how often are we trying to ask and answer questions about realised networks using feasible networks? The second is that this lack of 'direction' as to how we should define a network is (actually) probably one of the biggest barriers that is affecting the use of networks in applied settings... By define I mean both delimiting the time and geographic scale at which a network is aggregated at [@estayEditorialPatternsProcesses2023]. We know that space plays a role - the motility of different species will influence both the dynamics of networks but also serve to link smaller 'subnetworks'/community [@rooneyLandscapeTheoryFood2008; @fortinNetworkEcologyDynamic2021]. And so does time *e.g.,* seasonal rewiring [@brimacombeInferredSeasonalInteraction2021; @laenderCarbonTransferHerbivore2010]. There is also a bit of an interplay with time and data and the different scales that they may be integrated at - co-occurrence may span decades and just because two species have been recorded in the same space does not mean it was at the same timescale [@brimacombeApplyingMethodIts2024].

## Feasible, realised, or sustainable?

When do we determine a link to be 'real'... In the context of metawebs this is perhaps clearer - if all things were equal (*i.e.,* community context is irrelevant) would the predator be able to consume the prey. However in the realised space there is also the question of the long term 'energetic feasibility' of an interaction - just because an interaction is possible in the now is it able to sustain a population in the long term. And what is the scale for that long term - are we thinking at the generational scale? Because ultimately when we are constructing a network we are aggregating not only across space but also across time... This is probably again a Lokta-Volterra space question and something that the dynamic foodweb models [@lajaaitiEcologicalNetworksDynamicsJlJulia2024; @delmasSimulationsBiomassDynamics2017; @curtsdotterEcosystemFunctionPredator2019] are addressing, but again it is integrating this with the feasible/realised axis. And of course the Petchey dilemma of even what networks should we be feeding into these dynamic models.

# The future value of networks

> developing a dictionary of use... that helps navigate between the levels and assumptions

It should be clear that there is a high degree of interrelatedness and overlap between the way a network is constructed (modelled or predicted) and the process(es) it captures, these are encoded (embedded) within the network representation and ultimately influences how the network can and should be used [@petcheyFitEfficiencyBiology2011; @berlowGoldilocksFactorFood2008]. It is probably both this nuance as well as a lack of clear boundaries and guidelines as to the links between network form and function [although see @delmasAnalysingEcologicalNetworks2019] that has stifled the 'productive use' of networks beyond inventorying the interactions between species. Although, progress with using networks as a means to address questions within larger bodies of ecological theory *e.g.,* invasion biology [@huiHowInvadeEcological2019] and co-existence theory [@garcia-callejasNonrandomInteractionsGuilds2023] we still need to have a discussion on what the appropriate network configuration for the task at hand would be. This is highlighted in Box 1, and underscores that we need to evaluate exactly what process a specific network representation captures as well as its suitability for the question of interest. 

> The @terryFindingMissingLinks2020 paper looks at some methods but is specifically looking at a bipartite world...

# References {.unnumbered}

::: {#refs}
:::