---
title: "Unveiling the Complexity of Food Webs: A Comprehensive Overview of Definitions, Scales, and Mechanisms"
author:
  - name: Tanya Strydom
    id: ts
    orcid: 0000-0001-6067-1349
    corresponding: true
    email: t.strydom@sheffield.ac.uk
    role: 
    - conceptualization: lead
    - methodology: supporting
    affiliation:
      - id: sheffield
        name: School of Biosciences, University of Sheffield, Sheffield, UK
  - name: Jennifer A. Dunne
    id: jad
    orcid: 0000-0002-3789-9561
    corresponding: false
    role: 
    - conceptualization: lead
    - methodology: supporting
    affiliation:
      - id: sfi
        name: Santa Fe Institute, 1399 Hyde Park Road, Santa Fe, NM 87501, USA
  - name: Timothée Poisot
    id: tp
    orcid: 0000-0002-0735-5184
    corresponding: false
    role: 
    - conceptualization: lead
    - methodology: supporting
    affiliations:
      - Université de Montreal
      - Québec Centre for Biodiversity Sciences
  - name: Andrew P. Beckerman
    id: apb
    orcid: 0000-0002-7859-8394
    corresponding: false
    role: 
    - conceptualization: lead
    - methodology: supporting
    affiliations:
      - ref: sheffield
funding: "The author(s) received no specific funding for this work. Well they did I just haven't done the homework"
keywords:
  - food web
  - network construction
  - scientific ignorance
abstract: |
  Food webs are a useful abstraction and representation of the feeding links between species in a community and are used to infer many ecosystem level processes. However, the different theories, mechanisms, and criteria that underpin how a food web is defined, and ultimately, constructed means that not all food webs are representing the same ecological process at the same scale. Here we present a synthesis of the different assumptions, scales, and mechanisms that are used to define the different ecological networks , leading to a revision of definitions for different types of networks. Additionally we explicitly link the different network representations to the broader methodological approaches (models) that are used to construct them. In explicitly outlining the assumptions, scales, and mechanisms of network inference allows for a formal categorisation of how to use networks to answer key ecological and conservation questions as wel as defining clear guidelines to prevent unintentional misuse or misinterpretation.
date: last-modified
bibliography: references.bib
number-sections: true
---

At the heart of modern biodiversity science are a set of concepts and theories about species richness, stability, and function [@loreauBiodiversityEcosystemStability2013]. These relate to the abundance, distribution, and services that biodiversity provides, and how biodiversity (as an interconnected set of species) responds to multiple stressors. Documenting interactions between and among species is thus one of the fundamental building blocks of community ecology, providing a powerful abstraction and platform for mathematical and statistical modelling of biodiversity to make predictions, mitigate threats, and manage services [@windsorUsingEcologicalNetworks2023].Such network representations of biodiversity are increasingly argued to be an asset to understanding and predicting the abundance, distribution, dynamics, and services provided by multiple species facing multiple stressors [@simmonsRefocusingMultipleStressor2021]. However, there is a growing discourse around limitations to the interpretation and applied use of networks [@dormannRisePossibleFall2023; @bluthgenWhyNetworkAnalysis2010], primarily as the result of shortcomings regarding their conceptualisation [@bluthgenCriticalEvaluationNetwork2024].

We propose that every network embeds assumptions about the process(es) that determine interactions, and about the levels of organization at which this occurs (*i.e.* the biological, ecological, spatial/temporal scale). The differences in these assumptions ultimately influence the nature and scope of inference that can be made from a given network [@proulxNetworkThinkingEcology2005]. Fundamentally, we are talking about an intersection of the data used to construct the network and the underlying theory as to what drives the occurrence of interactions between species. Although there have been extensive discussions about the challenges relating to data collection and observation [*e.g.,* @bluthgenCriticalEvaluationNetwork2024; @brimacombeShortcomingsReusingSpecies2023; @brimacombePublicationdrivenConsistencyFood2024; @moulatletScalingTrophicSpecialization2024; @pringleResolvingFoodWebStructure2020; @polisComplexTrophicInteractions1991; @saberskiImpactDataResolution2024] we still lack a clear framework framed by the different assumptions and scale dependent processes.

In this perspective we aim to provide an overview of the different **food web** representations, particularly how these relate to the terminology used to define a food web, and how this is influenced by both the processes that determine interactions [@sec-process], as well as how this relates to the way in which we construct the resulting networks [@sec-construct]. This allows us to deliver an overview of fundamental questions in ecology that we think can benefit from network thinking and a proposal that such thinking can accelerate our capacity to predict the impact of multiple stressors on biodiverse communities. Specifically, we finish this perspective with an overview of fundamental questions in ecology that we think can benefit from network thinking and a proposal that such thinking can accelerate our capacity to predict the impact of change on biodiverse communities.

# Setting the Scene: The Not So Basics of Nodes and Edges {#sec-anatomy}

Networks often have multiple uses: an 'object' from which inferences are made (*e.g.,* topological inference about biodiversity, interactions among species, and community structure, \[REF\]); a platform for evaluating 'downstream' responses to stressors \[REF\]; and a platform for evaluating mathematical and statistical models of ‘generative processes’ \[REF\]. Against this backdrop of multiple research agendas, it should come as no surprise that the definition of ‘edges’ and ‘nodes’, as well as the levels of organisation at which they are collated takes many forms [@poisotDescribeUnderstandPredict2016; @moulatletScalingTrophicSpecialization2024], while also encoding a series of assumptions within a network.

## How do we define a node?

Although this may seem elementary that a node should represent a (taxonomic) species, the reality is that nodes often represents non-taxonomic units such as a trophic species (*e.g.,* @yodzisCompartmentationRealAssembled1982; @williamsSimpleRulesYield2000), a feeding guild [*e.g.,* @garcia-callejasNonrandomInteractionsGuilds2023], or a segregation of species by life stages [*e.g.,* @cleggImpactIntraspecificVariation2018]. Such granularity and variation can limit the ability to make (taxonomic) species specific inferences (*e.g.,* does species $a$ eat species $b$?), affect inference made from networks, including estimates of complexity and structure [@beckermanForagingBiologyPredicts2006; @cleggImpactIntraspecificVariation2018] and make it challenging to use networks in ‘downstream analyses’, for example, of extinction or invasions. Despite these implications, there may also be value in having nodes that represent an aggregation of species, as the distribution of the links between them may be more meaningful in terms of understanding energy flow and distribution within the system.

## What is captured by an edge?

Links within food webs can be thought of as a representation of either feeding links between species (be that realised or potential [@dunneNetworkStructureFood2006; @pringleUntanglingFoodWebs2020], or fluxes within a system *e.g.,* energy transfer or material flow as the result of the feeding links between species [@lindemanTrophicDynamicAspectEcology1942; @proulxNetworkThinkingEcology2005]. These correspond to different ‘currencies’ (the feasibility of links or the energy that is moving between nodes). There are also a myriad of ways in which the links themselves can be specified. Links between species can be treated present or absent (*i.e.,* binary), may be defined as probabilities [@banvilleDecipheringProbabilisticSpecies2024; @poisotStructureProbabilisticNetworks2016] or by continuous functions which further quantify the strength of an interaction [@berlowInteractionStrengthsFood2004]. How links are specified will influence the structure of the network. For example, taking a food web that consists of links representing all *potential* feeding links in a collection of species will be meaningless if one is interested in understanding the flow of energy through the network as the links are not environmentally/energetically constrained.

## Network representations {#sec-representation}

Against these definitions, networks fall into two major ‘types’: metawebs, traditionally defined as all the *potential* interactions for a specific species pool [@dunneNetworkStructureFood2006]; and realised networks, which is the subset of interactions in a metaweb that are *realised* for a specific community at a given time and place. The fundamental differences between these two network representations are the spatial scale at which they are constructed and the associated processes that are assumed to drive pattern at these scales.

A metaweb is at its core a list of feasible interactions between pairs of species. The feasibility for a given pair is derived from the complementarity (phylogenetic relationships) of their traits (representing a *global metaweb*), which can be further refined by co-occurrence (representing a *regional metaweb*). By this definition, metawebs provide a means to identify links that are not ecologically plausible, *i.e.,* forbidden links [@jordanoSamplingNetworksEcological2016], or provide an idea of the *complete* diet of a species [@strydomGraphEmbeddingTransfer2023].

In contrast realised networks are relatively localised in space and time, and the links between species are contingent on both the co-occurrence of species, the role of the environment, and mechanisms of diet choice. Fundamentally this means that the presence/absence of a link is the result of the ‘behaviour’ of the species.

This distinction between metawebs and realised webs lead to some further definitions. Links that are absent in a metaweb can conceptually (although not always practically) be treated as being truly absent. However, links that are absent in a realised network cannot be considered as truly absent but rather as absent due to the broader environmental/community context. Furthermore, a realised network is not simply the downscaling of a metaweb to a smaller scale (*e.g.,* moving from the country to the 1x1 km^2^ scale based on fine-scale species co-occurrence). Instead, realised webs capture processes that determine the realisation of an interaction. Specifically, in realised webs, the definition of an edges shifts from being determined by feasibility to that of choices and consequences that centre around energy. Meaning if one were to take the same community of species and constructed both a metaweb and realised network the two networks might have the same species but would be structurally different, owing to the differences in the 'rules' constraining the presence of links [@caronTraitmatchingModelsPredict2024].

# From Nodes and Edges to Process and Constraints {#sec-process}

In the previous section we discussed how the definition of nodes and edges at representing different biological and ecological and processes associated with them lead to the concept of a metweb and a realised web. Here we expand this discussion, introducing five core constraints across these scales that further expose processes that determine the links among species: evolutionary compatibility, co-occurrence, abundance, diet choice, and non-trophic interactions [@fig-process].

![Aligning the various processes that determine interactions (right column) with the different network representations (left column). First, we start with a **global metaweb** this network captures all possible interactions for a collection of species in the global context. However, within the global environment different species occur in different regions (region one = yellow and region 2 = orange), and it is possible to construct two different metawebs (**regional metawebs**) for each region by taking accounting for the co-occurrence patterns of the difference species - as shown here we have two regions with some species (blue) that are found in both regions and others endemic to either region one (yellow) or region two (orange). However even within a region we do not expect that all interactions to be realised but rather that there are multiple configurations of the regional metaweb over both space and time. The 'state' of the different **realised networks** is ultimately influenced not just by the co-occurrence of a species pair but rather the larger community context such as the abundance of different species, maximisation of energy gain, or indirect/higher order interactions.](images/anatomy.png){#fig-process}

## Processes that determine the feasibility of an interaction {#sec-process-feasibility}

Here we introduce evolutionary compatibility and co-occurrence as processes that ‘act’ at the species pair of interest. The scale of inference and set of processes embodied in these two constraints typically combine to define a ‘list’ of interactions that are viable/feasible and defined as present/absent. It is however possible to build a network from this information. However, it is important to be aware that the structure of this network is not constrained by any community context and so just because species are able to interact does not mean that they will [@poisotSpeciesWhyEcological2015].

**Evolutionary compatibility**

This constraint is defined by shared (co)evolutionary history between consumers and resources [@segarRoleEvolutionShaping2020; @gomezEcologicalInteractionsAre2010; @dallarivaExploringEvolutionarySignature2016] which, in the more proximal sense, is manifested as 'trait complementarity' between two species [@benadiQuantitativePredictionInteractions2022]. In this body of theory, one species (the consumer) has the 'correct' (multivariate) set of traits that allow it to chase, capture, kill, and consume the other species (the resource) and interactions that are not compatible are defined as forbidden links [@jordanoSamplingNetworksEcological2016]; *i.e.,* not physically possible and will *always* be absent within a network.

Networks arising from this constraint can be binary (possible vs forbidden) or probabilistic [@banvilleDecipheringProbabilisticSpecies2024], *e.g.,* the metaweb constructed by @strydomFoodWebReconstruction2022 uses probabilities to quantify their confidence with regards to the possibility of a specific interaction existing between two species. A network constructed on the basis of evolutionary compatibility is conceptually aligned with a 'global metaweb', and gives us information as to the feasibility of links between species despite the fact that they do not co-occur (as shown in @fig-process).

**(Co)occurrence**

The co-occurrence of species in both time and space is a fundamental requirement for an interaction between two species to occur (at least in terms of feeding links). Although co-occurrence data alone is insufficient for building an accurate and ecologically meaningful representation of *feeding links* [@blanchetCooccurrenceNotEvidence2020], it is still a critical process that determines the realisation of a feeding link and allows us to spatially constrain a global metaweb based on local communities [@dansereauSpatiallyExplicitPredictions2024], in the context of @fig-process this would be the metawebs for regions one and two.

## Processes that modify the behaviour (preference) of species {#sec-process-realisation}

Here we highlight three processes that influence the *realisation* of an interaction between species and form the conceptual basis for realised networks. As we show in @fig-process, a 'truly realised' network is the product of properties of the community (**abundance** and **non-trophic interactions**) and the individual (**profitability**). This represents a conceptual shift from considering the feasibility for species pairwise interactions to the considering how information about the community and the individual *constrains* network topology defined by consumer choice ([@quinteroDownscalingMutualisticNetworks2024] @sec-representation)

**Abundance**

Perhaps the simplest realising process is the principle that organisms feeding randomly will consume resources in proportion to their abundance [@stephensForagingTheory1986]. On its own, such abundance constrained interactions are not necessarily contingent on there being any compatibility between species [@canardEmergenceStructuralPatterns2012; @momalTreebasedInferenceSpecies2020; @pomeranzInferringPredatorPrey2019]. Here, abundance of different prey species influence the distribution of links in a network [@vazquezUnitingPatternProcess2009] by defining a preference linked to individuals among species meeting [@poisotSpeciesWhyEcological2015; @banvilleDecipheringProbabilisticSpecies2024]. Abundance data, linked to a derived metaweb delivers a foundation ruleset that can define the distribution and strength of links - a formal network topology/structure.

**Profitability**

It is well established that consumers make more active decisions than eating items in proportion to their abundance [@stephensForagingTheory1986]. Ultimately, consumer choice is underpinned by an energetic cost-benefit framework centred around profitability and defined by traits associated with finding, catching, killing, and consuming a resource [@woottonModularTheoryTrophic2023]. Energetic constraints are invoked to construct networks in a myriad of ways [*e.g.,* **ADD DIET BREADTH MODEL 2006** @pawarDimensionalityConsumerSearch2012; @portalierMechanicsPredatorPrey2019; **NOTE Portalier creates a pairwise list, not a network** @cherifEnvironmentRescueCan2024]. In contrast to requiring a combination of foraging rules and a derived metaweb as noted above, many of these methods deliver a realised web directly. Note that we select profitability as a term to capture rules linked to optimal foraging [@pykeOptimalForagingTheory1984] and metabolic theory [@brownMetabolicTheoryEcology2004]; it is a sensible ‘umbrella concept’ for capturing the energetic constraint on of the distribution and strength of interactions.

**Non-trophic interactions**

Non-trophic inteactions (see MIELE et al PLOS) include competition for space, predator interference, refuge provisioning, recruitment facilitation as well as non-trophic effects that increase or decrease mortality. These interactions [@ingsEcologicalNetworksbeyondFood2009] specifically modify either the realisation or strength of trophic interactions [@golubskiModifyingModifiersWhat2011; @pilosofMultilayerNatureEcological2017; @staniczenkoStructuralDynamicsRobustness2010; @kamaruDisruptionAntplantMutualism2024] and represent interactions either direct (e.g., predator a outcompetes predator b) and indirect (e.g., mutualistic/facilitative interactions) mechanisms. They operate on the realisation of a network of interactions by altering the fine-scale distribution and abundance of species as well as their persistence [@kefiMoreMealIntegrating2012; @kefiNetworkStructureFood2015; @bucheMultitrophicHigherOrderInteractions2024].

# Network construction {#sec-construct}

The above five processes are central to understanding the assumptions inherent in building networks. Each of these processes, or combinations thereof, deliver via these assumptions a set of constraints on what a network represents and can be used for. Here we build on the introduction of these five processes to categorise two approaches to constructing networks and in doing so also introduce more detail on methodologies.

## Why construct networks?

Networks are a representation of biodiversity. They are unique in delivering more than just estimates of species richness. A network embodies the organising structure of biodiversity and allows numerous opportunities for 'down-stream' analysis, including the comparison of structures, estimation of energy flux or extinction dynamics and ultimately form the structural inputs to dynamical systems models that faciliate ecological and conservation relevant inference about productivity-diversity-stability-function relationships in space and time.

In a perfect world, we might know about all interactions. However, the empirical collection of interaction data is both costly and challenging to execute [@jordanoChasingEcologicalInteractions2016; @jordanoSamplingNetworksEcological2016]. In the absence of robust empirical data, we construct models that predict the feasibilty of interaction among pairs of species, facilitate interpolation and gap-filling of existing empirical datasets [*e.g.,* @bitonInductiveLinkPrediction2024; @stockPairwiseLearningPredicting2021; @dallasPredictingCrypticLinks2017] or directly predict network structure[see @strydomRoadmapPredictingSpecies2021 for a broader discussion]. Doing so clearly embeds one or more of the processes discussed in @sec-process.

## Construction through induction

Constructing feasible networks and facilitating the interpolation or gap-filling of existing empirical datasets aligns with Evolutionary Compatibility and Co-Occurrence constraints. Methods embedding these scales and assumptions have been critical to the construction of 'first draft' networks for communities for which we have no interaction data [@strydomFoodWebReconstruction2022]. They are central to interpolation in data poor regions and predicting interactions for 'unobservable' communities *e.g.,* prehistoric networks [@yeakelCollapseEcologicalNetwork2014; @frickeCollapseTerrestrialMammal2022; @dunhillExtinctionCascadesCommunity2024] or future, novel community assemblages. They have the capacity to evaluate a role of interactions among species relative to their distribution by accounting for the role of the environment and the role of species interactions [@higinoMismatchIUCNRange2023; @pollockUnderstandingCooccurrenceModelling2014].

All methods in this inference space rest on an assumption that there are a set of 'feeding rules' that underpin feasibility[@morales-castillaInferringBioticInteractions2015], rules defined by evolutionary principles **\[\<- I feel like we are not saying enough about phylogeny here\]**. They represent a class of models that work by inductive reasoning. There appear to be three core ways the feeding rules are defined.

The determination of the feeding rules can be defined *a priori* based expert knowledge opinions. Typically this is done on a trait-based 'matching' basis. An example are the paleo food web models of Shaw et al and of Roopnarine et al **\<- IS THIS SENSIBLE?**[@shawFrameworkReconstructingAncient2024] that specify a series of rules for four different sets of traits and interactions are deemed feasible of all conditions are met.

Rules can also be defined by correlating real world interaction data with a suitable ecological proxy for which data is more widely available (*e.g.,* traits). These rules can be used by a binary classifier to determine if a link is present (see @pichlerMachineLearningAlgorithms2020 for an overview), including generalised linear models [*e.g.,* @caronAddressingEltonianShortfall2022], random forest [*e.g.,* @llewelynPredictingPredatorPrey2023], trait-based k-NN [*e.g.,* @desjardins-proulxEcologicalInteractionsNetflix2017], and Bayesian models [*e.g.,* @eklofSecondaryExtinctionsFood2013; @cirtwillQuantitativeFrameworkInvestigating2019]. Finally, rules have been defined by graph embedding, which uses the structural features of a known network to infer the position of species in the network (see @strydomFoodWebReconstruction2022 for a detailed review of methods).

There are substantial data requirements for these approaches, either reflecting expert knowledge, species traits or existing interaction data.

## Construction through deduction

In contrast to the above approaches centred on feasibility, methods accommodating abundance and profitability typically rely on deductive reasoning and have a unique agenda to those above. Typically, models constructed that embed abundance and profitabilty constraints reference theory that allows inference about the distribution and strength of interactions irrespective of the identity of the species identity (but see below). There are two broad group of models in this deductive category.

### **Species agnostic networks**

First, links are randomly distributed throughout the network [*e.g.,* @fortunaHabitatLossStructure2006; @bascompteNestedAssemblyPlantanimal2003], these models are often used as a 'null hypothesis' to ask questions about network structure [*e.g.,* @banvilleWhatConstrainsFood2023; @strydomSVDEntropyReveals2021].

Second. Interactions that occur between species are due to the abundance of species within the community [@pomeranzInferringPredatorPrey2019; @canardEmpiricalEvaluationNeutral2014; @krishnaNeutralnicheTheoryNestedness2008]

Third. Based on the idea that networks follow a trophic hierarchy and that network structure can be determined by distributing interactions along single dimension (the “niche axis”, @allesinaGeneralModelFood2008). Essentially these models can be viewed as being based on the idea of resource partitioning (niches) along a one-dimensional resource which will result in the standard 'trophic pyramid' to ensure that all species can 'fit' along this resource (which has strong ties back to the idea of intervality) *e.g.,* Cascade model [@cohenCommunityFoodWebs1990], Niche model [@williamsSimpleRulesYield2000], Nested hierarchy model [@cattinPhylogeneticConstraintsAdaptation2004].

These models are data light but there are some decisions that need to be made regarding what the expectations are on network structure.

### Species-specific networks

In terms predicting interactions current models are rooted in feeding theory and allocate the links between species based on energy *e.g.,* diet models [@beckermanForagingBiologyPredicts2006; @petcheySizeForagingFood2008] have been used construct networks based on both profitability (as determined by the handling time, energy content, and predator attack rate) as well as abundance (prey density). [@woottonModularTheoryTrophic2023].

At a 'coarser', functional level there are models that are based on the compartmentation and acquisition of energy for species at different trophic levels [@allesinaFoodWebModels2009; @krauseCompartmentsRevealedFoodweb2003]. Models that determine structure are based on the idea that networks follow a trophic hierarchy and that network structure can be determined by distributing interactions along single dimension [the “niche axis”; @allesinaGeneralModelFood2008], while parametrising an aspect of the network structure [although see @allesinaFoodWebModels2009 for a parameter-free model].

They are 'costly' to construct in real world settings (requiring data about the entire community, as it is the behaviour of the system that determines the behaviour of the part) and also lack the larger diet niche context afforded by metawebs.

# Making Progress with Networks {#sec-progress}

It is probably both this nuance as well as a lack of clear boundaries and guidelines as to the links between network form and function [although see @delmasAnalysingEcologicalNetworks2019] that has stifled the 'productive use' of networks beyond the inventorying the interactions between species. Although progress with using networks as a means to address questions within larger bodies of ecological theory *e.g.,* invasion biology [@huiHowInvadeEcological2019] and co-existence theory [@garcia-callejasNonrandomInteractionsGuilds2023] has been made we still lack explicit guidelines as to what the appropriate network representation for the task at hand would be, and as highlighted in Box 1, underscores the need to evaluate exactly what process a specific network representation captures as well as its suitability for the question of interest. Below we present a mapping of what we believe are some of the key questions for which interaction networks can be used to the different networks representations that are most suitable, as well as highlight some of the methodological challenges that still need to be improved upon.

## Making use of the different network representations

**Methodological challenges**

1.  Tools that allow us to estimate both the feasibility as well as realisation of links: Currently most approaches to modelling realised networks fail to explicitly account for any form of evolutionary constraint [although see @vandewalleArthropodFoodWebs2023 and @woottonModularTheoryTrophic2023] and we need to develop either an ensemble modelling approach [@beckerOptimisingPredictiveModels2022; @terryFindingMissingLinks2020] or tools that will allow for the downsampling of metawebs into realised networks [*e.g.,* @roopnarineExtinctionCascadesCatastrophe2006].
2.  Is there something in generalisable models that 'combine' different processes/aspects (*e.g.,* using body size as a catch all) versus limited models that allow you to unpack things bit-by-bit (*i.e.,* process by process). So @woottonModularTheoryTrophic2023 *may* (TBD) span the gamut but it lacks the ability to unpack... Although myabe the terms do?
3.  Modelling interaction strength: Although realised networks are more closely aligned with *explicitly* capturing interaction strength we lack models that allow us to quantify this [@wellsSpeciesInteractionsEstimating2013; @strydomRoadmapPredictingSpecies2021].
4.  How do we validate our predictions?: Progress has been made to assess how well a model recovers pairwise interactions [@strydomRoadmapPredictingSpecies2021; @poisotGuidelinesPredictionSpecies2023], but we still lack clear set of guidelines for benchmarking the ability of models to recover structure [@allesinaGeneralModelFood2008]
5.  Something about making what we do with networks more tractablie in the applied space? *e.g.,* @dansereauOvercomingDisconnectInteraction2024

**Theory challenges**

1.  Core Theory Advancement: Do the decades of insights arrived at for stability-diversity-productivity relationships with tri-trophic or diamond shaped models hold for complex communities (10’s-100s) [@danetResponseDiversityMajor2024]; How will spatial and temporal variation in climate and productivity drive change in complex ecosystems. Necessary to move to predicting changes in biodiversity per se, ecosystem functions and identifying sensitive and robust species and portions of communities.
2.  How will novel communities interact? How will range shifts and invasions result in new/novel community assemblages. And then also the intentional changes of species compositions through rewilding.
3.  Does rewiring happen and does it deliver robustness? Specific sub points to consider here is persistence, especially persistence to perturbation. Again, dynamic networks and network/community assembly and finally extinctions [@dunhillExtinctionCascadesCommunity2024].
4.  When do invasive species enhance or decimate communities? When do reintroductions work? [@woosterAustraliasRecentlyEstablished2024]
5.  Are there temperature threshold to community collapse
6.  Can socioeconomic networks combined with biological networks drive understanding of externalities?
7.  Can paleoecological data from deep time hyperthermal events provide sufficient insight into the targets, pace and recovery times from rapid climate events?

![Here we highlight some of the outstanding questions in both network as well as general ecology, as well as some of the outstanding methodological challenges with regards to constructing food webs (shown in orange) that we are faced with.](images/NetworkFuture.png){#fig-future}

# Concluding remarks

Having a clear understanding of the interplay between network representations and the processes that they are capable of encoding is critical if we are to understand exactly which networks can be used to answer which questions. As we highlight in Box 1 the different network representations have different potential uses and it should be clear that there is no 'best' network representation but rather a network representation that is best suited to its intended purpose. In providing a formalisation regards to the assumptions and mechanisms that need to be explicitly taken into consideration when deciding to use (and construct) networks we hope to prevent the unintentional misuse or misinterpretation of networks as well as provide a starting point from which we can develop a better framework for the applied use of networks to answer questions that are not only pressing within the field but also within broader biodiversity science.

# References {.unnumbered}

::: {#refs}
:::