---
title: "Unveiling the Complexity of Food Webs: A Comprehensive Overview of Definitions, Scales, and Mechanisms"
author:
  - name: Tanya Strydom
    id: ts
    orcid: 0000-0001-6067-1349
    corresponding: true
    email: t.strydom@sheffield.ac.uk
    role: 
    - conceptualization: lead
    - methodology: supporting
    affiliation:
      - id: sheffield
        name: School of Biosciences, University of Sheffield, Sheffield, UK
  - name: Jennifer A. Dunne
    id: jad
    orcid: 0000-0002-3789-9561
    corresponding: false
    role: 
    - conceptualization: lead
    - methodology: supporting
    affiliation:
      - id: sfi
        name: Santa Fe Institute, 1399 Hyde Park Road, Santa Fe, NM 87501, USA
  - name: Timothée Poisot
    id: tp
    orcid: 0000-0002-0735-5184
    corresponding: false
    role: 
    - conceptualization: lead
    - methodology: supporting
    affiliations:
      - Université de Montreal
      - Québec Centre for Biodiversity Sciences
  - name: Andrew P. Beckerman
    id: apb
    orcid: 0000-0002-7859-8394
    corresponding: false
    role: 
    - conceptualization: lead
    - methodology: supporting
    affiliations:
      - ref: sheffield
funding: "The author(s) received no specific funding for this work. Well they did I just haven't done the homework"
keywords:
  - food web
  - network construction
  - scientific ignorance
abstract: |
  Food webs are a useful abstraction and representation of the feeding links between species in a community and are used to infer many ecosystem level processes. However, the different theories, mechanisms, and criteria that underpin how a food web is defined, and ultimately, constructed means that not all food webs are representing the same ecological process at the same scale. Here we present a synthesis of the different assumptions, scales, and mechanisms that are used to define the different ecological networks , leading to a revision of definitions for different types of networks. Additionally we explicitly link the different network representations to the broader methodological approaches (models) that are used to construct them. In explicitly outlining the assumptions, scales, and mechanisms of network inference allows for a formal categorisation of how to use networks to answer key ecological and conservation questions as wel as defining clear guidelines to prevent unintentional misuse or misinterpretation.
date: last-modified
bibliography: references.bib
number-sections: true
---

At the heart of modern biodiversity science are a set of concepts and theories about species richness, stability, and function [@loreauBiodiversityEcosystemStability2013]. These relate to the abundance, distribution, functions, and services that biodiversity provides. Network representations of biodiversity are increasingly argued to be an asset to understanding and predicting the impacts of multiple, simultaneous stress on these core components of biodiversity [@simmonsRefocusingMultipleStressor2021]. Documenting interactions between and among species is thus one of the fundamental building blocks of community ecology and provide a powerful abstraction and platform for mathematical and statistical modelling of biodiversity to make predictions, and to mitigate and manage threats [@windsorUsingEcologicalNetworks2023].

However, there is a growing discourse around limitations to the interpretation and applied use of networks [@dormannRisePossibleFall2023; @bluthgenWhyNetworkAnalysis2010]. Against this, it is important to evaluate the value and the limitations of the various network conceptualisations of biodiversity [@bluthgenCriticalEvaluationNetwork2024]. In this perspective we aim to provide an overview of different **food web** representations, particularly how each representation embeds assumptions about the processes that determine interactions (@sec-process) about the levels of organization at which this occurs (*i.e.* the biological, ecological, spatial/temporal scale) and and the way in which we construct the resulting networks (@sec-construct). The differences among this tri-partite set of assumptions ultimately influence the nature and scope of inference that can be made from a given network [@proulxNetworkThinkingEcology2005].

Fundamentally, we are talking about an intersection of the type of data used to construct a network and the underlying theory as to what drives the resolution and occurrence of interactions between species in those data. We still lack a clear explanation of the different assumptions and scale dependent processes that underpin network construction alongside extensive discussions about the challenges relating to data collection and observation [*e.g.,* @bluthgenCriticalEvaluationNetwork2024; @brimacombeShortcomingsReusingSpecies2023; @brimacombePublicationdrivenConsistencyFood2024; @moulatletScalingTrophicSpecialization2024; @pringleResolvingFoodWebStructure2020; @polisComplexTrophicInteractions1991; @saberskiImpactDataResolution2024]. Such an understanding should deliver an acceleration in capacity to more effectively predict the impact of multiple stressors on biodiverse communities.

In their recent work, @gauzensTailoringInteractionNetwork2025 showcased a 2+2 decomposition of networks around aggregated versus species level resolution of nodes and around potential and realised links among the nodes. Their review delivers valuable insight into the methodologies used to collect and manage data among the node and link differentiation. It also delivers an overview of the scale and types of questions that are associated with each category of differentiation.

Here we provide a complementary perspective focused on concepts, models, and theory, in contrast to the data driven breakdown in @gauzensTailoringInteractionNetwork2025 (e.g. their Tables 1 and 2). Our approach delivers a hierarchical perspective on network construction based on a gradient from feasibility, capturing the concept of metawebs and Gauzen et al's 'potential' webs, through to realised webs as in Gauzens' et al. In contrast to their 2 + 2 decomposition (their Fig 1), our perspective showcases nested ecological scales and processes that derive from shifts in the assumptions and theories embedded along this gradient. This includes classic ecological 'aggregations' such as functional/phylogenetic groups through to species, populations and individuals, unique perspective on how space and time intersect with node and link resolution, refined insight into which networks are derived by induction vs. deduction and a revealing of a core transition between assumptions about how links are derived based on evolutionary vs. ecological theories.

In the following sections we provide a scene-setting review of nodes and edges (links) in networks before aligning various processes that determine interactions with the different network representations. Ultimately, we provide a unique perspective on the nested hierarchy of processes that govern transitions from meta-webs to realised webs. We finish with a refined and nuanced alignment of models/representations and key questions in biodiveristy science in the anthropocene.

# Setting the Scene: The Not So Basics of Nodes and Edges {#sec-anatomy}

Networks in ecology have multiple uses, representing an 'object' from which inferences can be made. For example, a network is needed to make inference specifically about the structure of communities. The structure of networks - their topology - have a long history reflecting core theory about energy flow \[Lindeman etc\], function \[REF\] and even stability \[REF\]. Networks are thus required as the response variable in evaluating ecological theory and statistical models of 'generative processes' giving rise to such structure \[REF\]. Such structure is now commonly used to compare communities along environmental gradients \[REF\]. Networks and their topology are also used as a platform for evaluating 'downstream' responses to stressors such as evaluating patterns of secondary extinction \[REF\]. Finally, they are commonly used as a platform for implementing mathematical models of community dynamics \[REF\]; delivering inference about stability, function, invasive species, climate change, contaminants, and secondary extinction, to name a few applications \[REF\]. Against this backdrop of multiple research agendas, the definition of 'edges' and 'nodes', and the levels of organisation at which they are defined, take many forms [@poisotDescribeUnderstandPredict2016; @moulatletScalingTrophicSpecialization2024], each of which encode a series of assumptions within a network. Here we introduce a perspective on these baseline assumptions.

## How do we define a node?

Although this may seem elementary that a node should represent a (taxonomic) species, the reality is that nodes often represents non-taxonomic units such as a trophic species (*e.g.,* @yodzisCompartmentationRealAssembled1982; @williamsSimpleRulesYield2000), a feeding guild [*e.g.,* @garcia-callejasNonrandomInteractionsGuilds2023], or a segregation of species by life stages [*e.g.,* @cleggImpactIntraspecificVariation2018]. Such granularity and variation is often defined as aggregation. Such aggregation can limit the ability to make species (taxonomic) specific inferences (*e.g.,* does species $a$ eat species $b$?). It can also affect the estimates of degree distributions and more specifically generality and vulnerability in networks (in/out degree). These metrics are central to inference about the structure and complexity of networks[@beckermanForagingBiologyPredicts2006; @cleggImpactIntraspecificVariation2018]. Finally, aggregation makes it challenging to use networks in 'downstream analyses' of, for example, extinction or invasions as the identity of species and the consequences of their losses can be hidden. Despite these issues, there are justifications for representing nodes as aggregated units. Most prominent relates to when the distribution of the links between aggregated nodes may be more meaningful in terms of understanding or generalising about energy flow and distribution within the system \[REF\].

## What is captured by an edge?

In order to break down the definitions of an edge, it is important to introduce the concept of *potential* versus *realised* links: potential links reflect feasibility while realised links are connected to flux of some currency (typically energy; see below for more detail). Links within food webs are thus a representation of either potential links between species or fluxes within a system *e.g.,* energy transfer or material flow as the result of the feeding links between species [@lindemanTrophicDynamicAspectEcology1942; @proulxNetworkThinkingEcology2005][@dunneNetworkStructureFood2006, @pringleUntanglingFoodWebs2020]. Edges can thus correspond to different 'currencies' [@gauzensTailoringInteractionNetwork2025]. There is also a myriad of ways in which the links themselves can be specified. Links between species can be treated as present or absent (*i.e.,* binary), may be defined as probabilities [@banvilleDecipheringProbabilisticSpecies2025; @poisotStructureProbabilisticNetworks2016] or by continuous functions which further quantify the strength of an interaction [@berlowInteractionStrengthsFood2004]. How links are specified thus requires intersecting both the currency being modelled and their specification. For example, feasibility is unlikely to accommodate flux, but does align with binary or probability representations. Taking a food web that consists of links representing feasible interactions among a collection of species will be meaningless if one is interested in understanding the flow of energy through the network as the links are not environmentally/energetically constrained.

## Network representations {#sec-representation}

Against these definitions of nodes and edges, networks fall into two major 'types': metawebs, traditionally defined as all the *potential* interactions for a specific species pool [@dunneNetworkStructureFood2006]; and realised networks, which is the subset of interactions in a metaweb that are *realised* for a specific community at a given time and place. The fundamental differences between these two network representations are the spatial and temporal scale at which they are constructed, and the associated processes that are assumed to drive pattern at these scales.

A metaweb is, at its core, a list of *feasible* interactions between pairs of species. The feasibility for a given pair is derived from the complementarity (phylogenetic relationships) of their traits, typically aligned with feeding. Feasibility can be further refined by *co-occurrence* leading to the transition from a *global* to *regional metaweb*. Metawebs thus provide a means to identify evolutionarily plausible links, regionally plausible interactions, the set of ecologically possible, *i.e.,* forbidden, links [@jordanoSamplingNetworksEcological2016], and ultimately a definition of the plausible *complete* diet of a species [@strydomGraphEmbeddingTransfer2023].

In contrast, realised networks are typically more localised in space and time, and the links between species are contingent on the co-occurrence of species, the role of the environment, and mechanisms of diet choice. Fundamentally this means that the presence/absence of a link is the result of the 'behaviour' of the species and even when the realised network is presented as a binary matrix, the edges imply a function is available to define the strength of an interaction. A realised network is therefore not simply the downscaling of a metaweb to a smaller scale (*e.g.,* moving from the country to the 1x1 km^2^ scale based on fine-scale species co-occurrence). Instead, realised webs capture processes that determine the realisation of an interaction and flows of energy in a community. Specifically, in realised webs, the definition of an edge shifts from being determined by feasibility to that of choices and consequences that centre around energy. If one were to take the same community of species and constructed both a metaweb and realised network the two networks might have the same species but would be structurally different, owing to the differences in the 'rules' constraining the presence of links. This distinction between metawebs and realised webs leads to a further insight. Links that are absent in a metaweb can conceptually (although not always practically) be treated as being truly absent. However, links that are absent in a realised network cannot be considered as truly absent but rather as absent due to the broader environmental/community context.

# From Nodes and Edges to Process and Constraints {#sec-process}

In the previous section we discussed how the definition of nodes and edges, representing different scales and processes, lead to the concept of a metaweb and a realised web. The fundamental take-homes are that nodes vary in their resolution, edges vary in what kind of process they represent and the intersection of these, defined by meta- vs. realised webs, underpins distinct lines of enquiry and constraints on the type of inference we can make with networks. Here we reveal five core constraints across evolutionary and ecological scales that further delineate the transition from meta- to realised webs, exposing processes that determine the nature of links among nodes: evolutionary compatibility, co-occurrence, abundance, diet choice, and non-trophic interactions [@fig-process].

![Aligning the various processes that determine interactions (right column) with the different network representations (left column). First, we start with a **global metaweb** this network captures all possible interactions for a collection of species in the global context. However, within the global environment different species occur in different regions (region one = yellow and region 2 = orange), and it is possible to construct two different metawebs (**regional metawebs**) for each region by taking accounting for the co-occurrence of the difference species - as shown here we have two regions with some species that are found in both regions (blue) and others endemic to either region one (yellow) or region two (orange). However even within a region we do not expect all interactions to be realised but rather that there are multiple configurations of the regional metaweb over both space and time. The 'state' of the different **realised networks** is ultimately influenced not just by the co-occurrence of a species pair but rather the larger community context such as the abundance of different species, maximisation of energy gain, or indirect/higher order interactions.](images/anatomy.png){#fig-process}

## Processes that determine the feasibility of an interaction {#sec-process-feasibility}

Evolutionary compatibility and co-occurrence are the two principle processes that 'act' at the species pair of interest and define feasibility. The scale of inference and set of processes embodied in these two constraints typically combine to define a 'list' of interactions that are viable/feasible and defined strictly as present/absent. Reflecting on the previous section, nodes are typically species and rules defining edges are defined by trait complementarity (phylogenetic) and/or co-occurrence. Here we provide more insight into each process.

**Evolutionary compatibility**

This constraint is defined by shared (co)evolutionary history between consumers and resources [@segarRoleEvolutionShaping2020; @gomezEcologicalInteractionsAre2010; @dallarivaExploringEvolutionarySignature2016; @rossbergFoodWebsExperts2006] which is manifested as 'trait complementarity' between two species [@benadiQuantitativePredictionInteractions2022]. In this body of theory, the consumer has the 'correct' set of traits that allow it to chase, capture, and consume the resource. Interactions that are not compatible are defined as forbidden links [@jordanoSamplingNetworksEcological2016]; *i.e.,* they are not physically possible and will *always* be absent within a network.

Networks do not properly arise from models based on this constraint. Instead, interacting species pairs are defined and these are represented as binary (possible vs forbidden) or probabilistic [@banvilleDecipheringProbabilisticSpecies2025]. For example, in the metaweb constructed by @strydomFoodWebReconstruction2022 probabilities are quantified as the confidence of a specific being *possible* between two species. A network constructed on the basis of evolutionary compatibility is conceptually aligned with a 'global metaweb', and gives us information as to the global feasibility of links between species pairs despite the fact that they do not co-occur (see @fig-process).

**(Co)occurrence**

The co-occurrence of species in both time and space is a fundamental requirement for an interaction between two species to occur (at least in terms of feeding links). Although co-occurrence data alone is insufficient for building an accurate and ecologically meaningful representation of *feeding links* [@blanchetCooccurrenceNotEvidence2020], it is still a critical process that determines the realisation of a feeding. Knowledge on the co-occurrence of species allows us to spatially constrain a global metaweb to reflect regional metawebs [@dansereauSpatiallyExplicitPredictions2024]. In the context of @fig-process this would be the metawebs for regions one and two.

We reinforce that these two constraints don't deliver a network *per se,* but a list of feasible species pairs. Although it is possible to build a network from the list of interactions generated by these constraints, it is important to be aware that the structure of this network is not constrained by any community context: just because species are able to interact does not mean that they will [@poisotSpeciesWhyEcological2015; @caronTraitmatchingModelsPredict2024].

## Processes that realise networks {#sec-process-realisation}

In contrast to the above, here we highlight three processes that influence the *realisation* of an interaction between species and thus form the conceptual basis for realised networks. As we show in @fig-process, a 'truly realised' network is the product of properties of the community (**abundance** and **non-trophic interactions**) and the individual (**diet choice**). This represents a conceptual shift from considering the feasibility for species pairwise interactions to considering the edge as a representation of energy flow. Such a transition requires information about how the community, the environment and the individual *constrains* network topology as defined by consumer choice (@quinteroDownscalingMutualisticNetworks2024, @sec-representation)

**Abundance**

Abundance as a realising process emerges from a null model for energy acquisition: organisms feeding randomly will consume resources in proportion to their abundance [@stephensForagingTheory1986]. Here, abundance of different prey species influences the distribution of links in a network [@vazquezUnitingPatternProcess2009] by defining a preference linked to individuals among species meeting [@poisotSpeciesWhyEcological2015; @banvilleDecipheringProbabilisticSpecies2025]. Abundance data, linked to a derived metaweb delivers a foundation ruleset that can define the distribution and strength of links. Of note, however, is that such abundance constrained interactions are not necessarily contingent on there being any compatibility between species [@canardEmergenceStructuralPatterns2012; @momalTreebasedInferenceSpecies2020; @pomeranzInferringPredatorPrey2019].

**Diet choice**

It is well established that consumers make more active decisions than eating items in proportion to their abundance [@stephensForagingTheory1986]. Ultimately, consumer choice is underpinned by an energetic cost-benefit framework centred around profitability and defined by traits associated with finding, catching, killing, and consuming a resource [@woottonModularTheoryTrophic2023; @smithBehavioralResponsesMosaic2021]. Energetic constraints are invoked to construct networks in a myriad of ways [*e.g.,* @beckermanForagingBiologyPredicts2006; @pawarDimensionalityConsumerSearch2012; @portalierMechanicsPredatorPrey2019; @cherifEnvironmentRescueCan2024].

In contrast to metaweb 'construction' from a list of pairwise interactions, these methods deliver a realised web directly and as an emergent property of node behaviour. We also here make a distinction, developed below, with models like the Niche Model, where diet choice is implicit in it's probabilistic network generating function, but it is working to replicate the *expected* structure of the network and this structure does not emerge from node-based rules. Note that we select diet choice as a term to capture rules linked to optimal foraging [@pykeOptimalForagingTheory1984] and metabolic theory [@brownMetabolicTheoryEcology2004]; it is a sensible 'umbrella concept' for capturing the energetic constraint on of the distribution and strength of interactions.

**Non-trophic interactions**

We include non-trophic interactions [see @mieleNontrophicInteractionsStrengthen2019] here not as a determinant of links, but a modifier of them - they are the community context above and beyond co-occurrence and abundance. Non-trophic interactions include competition for space, predator interference, refuge provisioning, recruitment facilitation as well as non-trophic effects that increase or decrease mortality. These interactions [@ingsEcologicalNetworksbeyondFood2009] specifically modify either the realisation or strength of trophic interactions [@golubskiModifyingModifiersWhat2011; @pilosofMultilayerNatureEcological2017; @staniczenkoStructuralDynamicsRobustness2010; @kamaruDisruptionAntplantMutualism2024] and represent direct (e.g., predator $a$ outcompetes predator $b$) and indirect (e.g., mutualistic/facilitative interactions) mechanisms. They operate on the realisation of a network by altering the fine-scale distribution and abundance of species and relative contributions of direct and indirect effects to biomass, persistence, stability and the functioning of the communities [@kefiMoreMealIntegrating2012; @kefiNetworkStructureFood2015; @bucheMultitrophicHigherOrderInteractions2024; @mieleNontrophicInteractionsStrengthen2019].

**are these strictly modifiers of realised networks? - because we class them as community context with co-occurrence, a modifier of feasible networks....**

# Network construction {#sec-construct}

The above five processes are central to understanding the assumptions inherent in building different types of networks. Each of the processes, or combinations thereof, deliver a unique set of boundary conditions on what a network represents and can be used for. Here we build on the introduction of these five processes to further categorise the approaches to constructing networks. In doing so also introduce more detail on a variety of methodologies used to construct networks.

## Why construct networks?

Networks are a representation of biodiversity. In a perfect world, we might know about all interactions. However, the empirical collection of interaction data is both costly and challenging to execute [@jordanoChasingEcologicalInteractions2016; @jordanoSamplingNetworksEcological2016; @poisotGlobalKnowledgeGaps2021]. In the absence of robust empirical data, we construct models that facilitate interpolation and gap-filling of existing empirical datasets [*e.g.,* @bitonInductiveLinkPrediction2024; @stockLinearFilteringReveals2017; @dallasPredictingCrypticLinks2017; @poisotNetworkEmbeddingUnveils2023], predict the feasibility of interaction among pairs of species, or directly predict network structure [see @strydomRoadmapPredictingSpecies2021 for a broader discussion].

They are unique in delivering more than just estimates of species richness. As note in the introduction, a network embodies the organising structure of biodiversity and allows numerous opportunities for 'downstream' analysis, including the comparison of structures, estimation of energy flux or extinction dynamics and ultimately form the structural inputs to dynamical systems models that facilitate ecological and conservation relevant inference about productivity-diversity-stability-function relationships [@danetResponseDiversityMajor2024] in space and time. But making such inferences requires careful attention to one or more of the processes discussed in @sec-process.

## Construction through induction {#sec-construct-induct}

Constructing feasible or realised networks can be framed as an 'inductive reasoning' process where insight and generalisation arises from a set of observations and relationships. Inductive reasoning as a foundation for network construction is implemented through node- and network levels. When applied at the node level, species specific networks are created and judge by their association with expected feeding interactions. When applied at the network level, networks are judged by their structural properties per se.

### Species specific networks: construction through node level induction

Constructing feasible networks and facilitating the interpolation or gap-filling of existing empirical datasets on sets of species interactions can be framed as an 'inductive reasoning' process where insight and generalisation arises from a set of observations and relationships about feeding. All methods in this inference space rest on a set of three assumptions: there are a set of 'feeding rules' that underpin interaction feasibility [@morales-castillaInferringBioticInteractions2015]; these rules are phylogenetically conserved [@dallarivaExploringEvolutionarySignature2016; @bramonmoraIdentifyingCommonBackbone2018]; they can be specified by matching the traits between consumer and resource.

Evolutionary compatibility and co-occurrence constraints, the foundation theory for feasible networks, and have delivered insight in many ways. They have been critical to the construction of 'first draft' networks for communities for which we have no interaction data [@strydomFoodWebReconstruction2022]. They are also central to interpolation in data poor regions and predicting interactions for 'unobservable' communities *e.g.,* prehistoric networks [@yeakelCollapseEcologicalNetwork2014; @frickeCollapseTerrestrialMammal2022; @dunhillExtinctionCascadesCommunity2024] or future, novel community assemblages [@vanderputtenPredictingSpeciesDistribution2010]. Furthermore, they have the capacity to evaluate a role of interactions among species relative to their distribution by accounting for the role of the environment and the role of species interactions [@higinoMismatchIUCNRange2023; @pollockUnderstandingCooccurrenceModelling2014; @gravelBringingEltonGrinnell2019].There are substantial data requirements for these approaches including expert knowledge, species traits and phylogenetic relationships and/or interaction data on related species or communities.

Feeding rules are defined in multiple ways. The determination of the feeding rules can be defined *a priori* based expert knowledge opinions. Typically this is done on a 'trait matching' basis. An example are the paleo food web models of @shawFrameworkReconstructingAncient2024 and @roopnarineEcologicalModellingPaleocommunity2017 that specify a series of rules for a set of traits and interactions are deemed feasible if all conditions are met. Alternatively the body size ratio between the consumer and resource is often used [*e.g.,* @gravelInferringFoodWeb2013; @rohrModelingFoodWebs2010], with the idea that consumers will only utilise a resource with a body size is less than or equal to their own. However, work from @vandewalleArthropodFoodWebs2023 seems to suggest that adding morphological traits in addition to body size ratio improves model performance.

Rules are also defined by correlating real world interaction data with suitable ecological proxies for which data is more widely available (*e.g.,* traits) using some sort of binary classifier (see @pichlerMachineLearningAlgorithms2020 for an overview). These include generalised linear models [*e.g.,* @caronAddressingEltonianShortfall2022], random forest [*e.g.,* @llewelynPredictingPredatorPrey2023], trait-based k-NN [*e.g.,* @desjardins-proulxEcologicalInteractionsNetflix2017], and Bayesian models [*e.g.,* @eklofSecondaryExtinctionsFood2013; @cirtwillQuantitativeFrameworkInvestigating2019].

Finally, graph embedding uses the structural features of a known network to infer the position of species in an unknown network through the decomposition of the interaction onto the embedding space. This decomposition relies on a combination of ecological proxies **(e.g. ???)** in conjunction with known interactions to infer the latent values of species **What is a latent value of a species with respect to inferring interactions?**. See @strydomGraphEmbeddingTransfer2023 for a detailed review of methods and @strydomFoodWebReconstruction2022 for a specific example.

### Species agnostic networks: construction through structure induction

Networks in this category are generated rules that create non-random (**note that this is irritated by the Random models paragraph below; do we need that? The stochastic models are the 'real' version of this type of network?**) networks that reflect empirical knowledge of ecological network structures and evaluated by matching predictions to this *expected* structure of the network(s). The determination of links between species is only implicitly linked to properties of the nodes (**see ADBM 5 rules**). This means these networks are usually not species specific. Although these models are data input light, often requiring only species richness and an estimate of the number of expected links, they make clear assumptions regarding what the expectations are for network structure. These are some of the most commonly used network generation tools (e.g. the Niche model REF). There are two sub-categories of these species agnostic networks:

Random network models [*e.g.,* @erdosRandomGraphs1959; @fortunaHabitatLossStructure2006; @bascompteNestedAssemblyPlantanimal2003] represent a 'process free' model. These models are not explicitly tied to a process discussed in @sec-process, rather links are randomly distributed among nodes. Although these models lack real world tractability [@bascompteNetworksEcology2007] they are often used as a 'null hypothesis' to ask questions about network structure [*e.g.,* @banvilleWhatConstrainsFood2023; @strydomSVDEntropyReveals2021].

Stochastic network models use a probabilistic rule-set about diet choice and niche breadth to reflect fundamental ideas of foraging biology. **RATHER THAN ALLESINA, I'D SUGGEST PRESENTING THE 5 RULES IN THE ADBM EXPLANATION OF THE NICHE MODEL** These models that are based on the compartmentation and acquisition of energy for species at different trophic levels [@allesinaFoodWebModels2009; @krauseCompartmentsRevealedFoodweb2003] and that network structure can be determined by distributing interactions along single dimension [the 'niche axis'; @allesinaGeneralModelFood2008]. Typically these models parametrise some aspect of the network structure [although see @allesinaFoodWebModels2009 for a parameter-free model]. These models include the most commonly used network generator, the Niche model [@williamsSimpleRulesYield2000], as well as the original Cascade model [@cohenCommunityFoodWebs1990] and the derived Nested hierarchy model [@cattinPhylogeneticConstraintsAdaptation2004]. These models often form the basis for dynamic models *e.g.,* the allometric trophic network [@broseAllometricScalingEnhances2006; @schneiderAnimalDiversityEcosystem2016a] and bioenergetic food web models [@delmasSimulationsBiomassDynamics2017].

## Construction through deduction

In contrast to the above approaches centred on feasibility, relised networks via methods reflecting abundance and diet choice typically rely on deductive reasoning and have a unique agenda to those above. In contrast to the inductive methods, inference about a realise network follows from a set of premises defining generative processes, often referred to as mechanisms. Typically, models that embed abundance and diet choice constraints reference theory that allows inference about the distribution and strength of interactions. Such models are 'network topology generators' and have a strong representation in research comparing network structures along environmental gradients and delivering inference about extinctions and energy flux. They also provide the structural backbone for dynamical systems modelling to address questions about stability-structure-productivity-function relationships, secondary extinction dynamics, species invasion and climate change. There are two broad group of models in this deductive category.

### Species-specific networks

These models capture the behaviour of the nodes by explicitly taking into account the properties of the different species in the community. Which means that there is a degree of variance in which links are predicted between species unlike the more 'static' predictions made by inductive models. However, these networks are 'costly' to construct in real world settings (requiring data about the entire community, as it is the behaviour of the system that determines the behaviour of the part) and also lack the larger diet niche context afforded by metawebs.

Neutral networks are built on the assumption that foraging decisions are tied *only* to the abundance of species within the community [@canardEmpiricalEvaluationNeutral2014; @krishnaNeutralnicheTheoryNestedness2008]. Here links are soley determined by the relative abundance of the different species in the community. Although it is highly unlikely that abundance is the only determaninant of interactions work by @pomeranzInferringPredatorPrey2019 showcases how these neutral processes can be used in conjunction with inductive models to construct more refined/localised networks.

There is a broader group of models that focus on determining interactions in terms of energetic constraints on diet breadth, often using the ratio of consumer-resource bodysize as a proxy for capturing the energetic constraints of feeding. Models such as those developed by @portalierMechanicsPredatorPrey2019 and @woottonModularTheoryTrophic2023 are similar to the mechanistic approaches discussed in @sec-construct-induct, however instead of determining interactions based on mechanistic feasibility it is rather constrained by the energetic cost of predation. Note that although these models do not place any explicit constraints on the expected structure of the network, the links should still be considered as 'realised' owing to the energetic constraint placed on links. A different subset of diet models [*e.g.,* @beckermanForagingBiologyPredicts2006; @petcheySizeForagingFood2008] use a diet choice approach, however similar to the stochastic network models they also embed assumptions on network structure. Thus these models predict both interactions and network structure simultaneously, although they would benefit in being refined by more explicitly accounting for trait-based (*i.e.,* feasibility) paramterisation [@curtsdotterEcosystemFunctionPredator2019].

# Making Progress with Networks {#sec-progress}

The motivation to leverage network ecology in conservation ecology, environmental risk assessment and natural resource managment stems from a shift away from species/populaiton specific measures of the effects of stress and distrubrance to community level metrics.  These metrics, such as resilience and more generally stability, ecosystem function and biodiversity _per se_, are natural properties of networks.  This suggests that modern conservation, risk assessment and resource management requires robust tools to support decision making.

This is also true in the discipline of ecology.  Major questions remain about stability-diversity-productivity relationships and the impacts of multiple stressors operating at multiple ecological scales.  The network approach to answering these questions specifically allows for questions about how direct and indirect effects among species in a complex community, shaped by their environment, deliver fundamental patterns and understanding of processes in the natural world.  and theories that help define the abundance, distribution, functions and services that biodiversity provides [@loreauBiodiversityEcosystemStability2013] 

In order to effectively use networks to aid us in answering questions about conservation/risk assessment/management and core ecological theorywe need to be mindful that we are mapping the *correct* network representation to the question of interest [@gauzensTailoringInteractionNetwork2025]. Notably, there are certain questions that cannot be answered using specific network representations as the scale of the question of interest is fundamentally misaligned with either the process captured by a specific network representation [@sec-process-feasibility], the underlying data that is used to construct it [@sec-construct] or both of these factors. 

Here we will discuss and map different the different network representations shown in @fig-process the the appropriate research questions and agendas \[see also @tbl-questions\], as well as highlighting some of the key methodological challenges that limiting our conceptualisation of a 'network' as well as the effective practical application of them in real world settings.

| Network Representation | Research Question |
|----|----|
| Global Metaweb | How will novel communities interact *e.g.* invasion and rewilding |
|  | Diet-based conservation |
|  | Rewiring capacity of species |
|  | Ecology-Evolutionary dynamics |
| Regional Metawebs | Applied use potential of questions highlighted for global metawebs at the management scale *e.g.,* a protected area |
|  | Refinement of species distributions |
| Realised webs | Multiple stressor allocation across networks |
|  | Temperature threshold to community collapse |
|  | Extinction and persistence after perturbation |
|  | Stability-diversity-function |
|  | Ecosystem level processes |
|  | Meta communitites and the idea of meta-network-communities |

: Table Caption {#tbl-questions}

## Key Eco-Evo-Conservation Questions

### Global Metawebs

As the interactions in global metawebs are not constrained by the realisation of specific community assemblages (or species co-occurrence) they can provide valuable insights as to what interactions *could* occur between species or the *entire* diet breadth of a specific species. Fundamentally this allows us to ask questions about *hypothetical* communities and interactions, such as thinking about novel communities in future climate change scenarios, or the potential 'position' of an invasive (or re-introduced) within a network [@huiHowInvadeEcological2019] or the rewiring capacity of species, and how this may help inform on the potential persistence of species within new community assemblages [@marjakangasFundamentalInteractionNiches2025]. Finally because metawebs are often built on some proxy of the evolutionary lineages there is scope to deeper interrogate the role of ecological-evolutionary dynamics and the role that both the evolutionary history as well as plasticity influences and shapes the diet breadth of species.

### Regional Metawebs

As regional metawebs are conceptually a spatially constrained global metaweb they could conceptually be used to answer similar questions to those raised above and because they represent a 'real' community are probably more meaningful in the applied space. Regional metawebs have also been used as a way to help refine and constrain species distributions giving us more refined range maps [@garcia-callejasNonrandomInteractionsGuilds2023] or community composition under climate change scenarios, even at global scales [@haoGlobalProjectionTerrestrial2025]. However it should be clear that the links in these regional metawebs are only being constrained by feasibility and species co-occurrence and we must exercise a high degree of caution to not assume that changes/differences between different regional webs in structure (link distribution) are being driven by *e.g.,* environmental factors. At minimum if one is interested in comparing regional metawebs across environmental scales or gradients we need to assert that structural differences are independent from species turnover.

### Realised networks

Realised network are perhaps the most representative of what comes to mind when people think of networks, and more specifically how we can use them to help inform on larger biogeographic processes [@thuillerNavigatingIntegrationBiotic2024]. This is because they represent a shift in the 'currency' that determines interactions that is constrained by a broader community and environmental context that allows us the opportunity to ask questions that revolve around community stability, diversity, function, and complexity. Specifically we are in a position to explicitly link network structure to ecosystem function. The dynamic nature of realised webs mean that they allowing us to think about the propagation of change (across both time and space) which allow us to ask questions about the persistence of communities and how they respond to perturbations or stressors (at both the level of the node as well as modification of links). Although the recent boom in the availability of long-term observation data is allowing us to unpack decades of insights arrived at for stability-diversity-productivity relationships for more complex communities [@danetResponseDiversityMajor2024] or to evaluate the impacts and efficacy of re-introductions [@woosterAustraliasRecentlyEstablished2024] we need to be mindful that empirical interaction data is typically accumulated over time and so it compresses the dynamic component of the interactions between species [@polisComplexTrophicInteractions1991]. Thus we need to apply a degree of caution when using empirical data to construct realised networks - however there is scope to think about developing methods that will allow us to modify metawebs in such a way that their structures become more aligned with realised webs (see the next section).

## Key methodological challenges

**Transitioning between metawebs and realised webs:** Currently most approaches to modelling realised networks fail to explicitly account for any form of evolutionary constraint (although @vandewalleArthropodFoodWebs2023 and @woottonModularTheoryTrophic2023 have added traits to more ) and we need to develop either an ensemble modelling approach [@beckerOptimisingPredictiveModels2022; @terryFindingMissingLinks2020] or. tools that will allow for the downsampling of metawebs into realised networks, [*e.g.,* @roopnarineExtinctionCascadesCatastrophe2006]. Importantly we need to think critically how the creation of either an 'ensemble network' or downsampled metaweb might change the underlying 'currency' of a nework and thus the underling defintiion of the edge *e.g.* the downsampling approach developed by @roopnarineExtinctionCascadesCatastrophe2006 structually constrains the network to structurally look like a realised web, but the links to not represent prey choice *per se*.

**The validation of network structure:** Progress has been made to assess how well a model recovers pairwise interactions [@strydomRoadmapPredictingSpecies2021; @poisotGuidelinesPredictionSpecies2023], but we still lack clear set of guidelines for benchmarking the ability of models to recover structure [@allesinaGeneralModelFood2008]. This makes it challenging to assess if models are capturing network structure accurately, especially if we are interested developing ways in which we can begin to downsample metawebs.

**Making networks more tractible in applied spaces:** There is a disconnect when it comes to effectively using networks in applied spaces [@dansereauOvercomingDisconnectInteraction2024]. We need to make an effort to more efficiently map the from (structure) of a network to its function as well as identify how this can effectively be integrated into policy to make it meaningful and actionable. Additionally we also need a firmer grasp as to what defines a 'network' as a unit, are the logical (environmental) boundaries between networks and how do these relate to 'management' units and scales [@fortinNetworkEcologyDynamic2021].

**Understanding what emperical data represents:** What does it mean when we 'observe' an interaction be that directly (predator actively feeding on prey) or indirectly *e.g.,* gut or isotope analysis. A netwrok constructued with emperical data will most liekley still represents an 'accumulation' of feeding data and this is conceptually more closely aligned with idea of a metaweb. Something about how this is a challenge if we want to use empirical networks to test or ask questions in the realised space.

# Concluding remarks

Having a clear understanding of the interplay between network representations and the processes that they are capable of encoding is critical if we are to understand exactly which networks can be used to answer which questions. As we highlight in Box 1 the different network representations have different potential uses and it should be clear that there is no 'best' network representation but rather a network representation that is best suited to its intended purpose. In providing a formalisation regards to the assumptions and mechanisms that need to be explicitly taken into consideration when deciding to use (and construct) networks we hope to prevent the unintentional misuse or misinterpretation of networks as well as provide a starting point from which we can develop a better framework for the applied use of networks to answer questions that are not only pressing within the field but also within broader biodiversity science.

# References {.unnumbered}

::: {#refs}
:::