---
title: "Unveiling the Complexity of Food Webs: A Comprehensive Overview of Definitions, Scales, and Mechanisms"
author:
  - name: Tanya Strydom
    id: ts
    orcid: 0000-0001-6067-1349
    corresponding: true
    email: t.strydom@sheffield.ac.uk
    role: 
    - conceptualization: lead
    - methodology: supporting
    affiliation:
      - id: sheffield
        name: School of Biosciences, University of Sheffield, Sheffield, UK
  - name: Jennifer A. Dunne
    id: jad
    orcid: 0000-0002-3789-9561
    corresponding: false
    role: 
    - conceptualization: lead
    - methodology: supporting
    affiliation:
      - id: sfi
        name: Santa Fe Institute, 1399 Hyde Park Road, Santa Fe, NM 87501, USA
  - name: Timothée Poisot
    id: tp
    orcid: 0000-0002-0735-5184
    corresponding: false
    role: 
    - conceptualization: lead
    - methodology: supporting
    affiliations:
      - Université de Montreal
      - Québec Centre for Biodiversity Sciences
  - name: Andrew P. Beckerman
    id: apb
    orcid: 0000-0002-7859-8394
    corresponding: false
    role: 
    - conceptualization: lead
    - methodology: supporting
    affiliations:
      - ref: sheffield
funding: "The author(s) received no specific funding for this work. Well they did I just haven't done the homework"
keywords:
  - food web
  - network construction
  - scientific ignorance
abstract: |
  Food webs are a useful abstraction and representation of the feeding links between species in a community and are used to infer many ecosystem level processes. However, the different theories, mechanisms, and criteria that underpin how a food web is defined, and ultimately, constructed means that not all food webs are representing the same ecological process at the same scale. Here we present a synthesis of the different assumptions, scales, and mechanisms that are used to define the different ecological networks , leading to a revision of definitions for different types of networks. Additionally we explicitly link the different network representations to the broader methodological approaches (models) that are used to construct them. In explicitly outlining the assumptions, scales, and mechanisms of network inference allows for a formal categorisation of how to use networks to answer key ecological and conservation questions as wel as defining clear guidelines to prevent unintentional misuse or misinterpretation.
date: last-modified
bibliography: references.bib
number-sections: true
---

At the heart of modern biodiversity science are a set of concepts and theories about species richness, stability, and function [@loreauBiodiversityEcosystemStability2013]. These relate to the abundance, distribution, functions and services that biodiversity provides. Such network representations of biodiversity are increasingly argued to be an asset to understanding and predicting the the impacts of multiple, simultaneous stress on these core components of biodiversity[@simmonsRefocusingMultipleStressor2021]. Documenting interactions between and among species is thus one of the fundamental building blocks of community ecology and provide a powerful abstraction and platform for mathematical and statistical modelling of biodiversity to make predictions and to mitigate and manage threats[@windsorUsingEcologicalNetworks2023]. However, against a growing discourse around limitations to the interpretation and applied use of networks [@dormannRisePossibleFall2023; @bluthgenWhyNetworkAnalysis2010], it is important to. evaluate the value and the limitations of the various network conceptualisations of biodiversity [@bluthgenCriticalEvaluationNetwork2024].

In this perspective we aim to provide an overview of the different **food web** representations, particularly how these relate to the terminology used to define a food web, and how this is influenced by both the processes that determine interactions [@sec-process], as well as how this relates to the way in which we construct the resulting networks [@sec-construct]. We propose that every network embeds assumptions about the process(es) that determine interactions, and about the levels of organization at which this occurs (*i.e.* the biological, ecological, spatial/temporal scale). The differences in these assumptions ultimately influence the nature and scope of inference that can be made from a given network [@proulxNetworkThinkingEcology2005].

Fundamentally, we are talking about an intersection of the type of data used to construct a network and the underlying theory as to what drives the occurrence of interactions between species in those data. Although there have been extensive discussions about the challenges relating to data collection and observation [*e.g.,* @bluthgenCriticalEvaluationNetwork2024; @brimacombeShortcomingsReusingSpecies2023; @brimacombePublicationdrivenConsistencyFood2024; @moulatletScalingTrophicSpecialization2024; @pringleResolvingFoodWebStructure2020; @polisComplexTrophicInteractions1991; @saberskiImpactDataResolution2024] we still lack a clear explanation of the different assumptions and scale dependent processes that underpin network construction. Such an understanding delivers an alignment of different network structures with fundamental questions in ecology and, we suggest, an opportunity to accelerate our capacity to more effectively predict the impact of multiple stressors on biodiverse communities.

# Setting the Scene: The Not So Basics of Nodes and Edges {#sec-anatomy}

Networks in ecology have multiple uses. They are sometimes an 'object' from which inferences are made. For example, a network is needed to make inference specifically about the structure of communities. The structure of networks - their topology - have a long history reflecting core theory about energy flow \[Lindeman etc\], function \[REF\] and even stability \[REF\]. Networks are thus required as the response variable in evaluating ecological theory and statistical models of 'generative processes' giving rise to such structure \[REF\]. Such structure is now commonly used to compare communities along environmental gradients \[REF\]. Networks and their topology are are also used a platform for evaluating 'downstream' responses to stressors such as evaluating patterns of secondary extinction \[REF\]. Finally, they are commonly used as a platform for implementing mathematical models of community dynamics \[REF\] delivering inference about stability, function, invasive species, climate change, contaminants and secondary extinction, to name a few applications \[REF\]. Against this backdrop of multiple research agendas, the definition of 'edges' and 'nodes', and the levels of organisation at which they are defined, take many forms [@poisotDescribeUnderstandPredict2016; @moulatletScalingTrophicSpecialization2024], each of which encode a series of assumptions within a network. Here we introduce a perspective on these baseline assumptions.

## How do we define a node?

Although this may seem elementary that a node should represent a (taxonomic) species, the reality is that nodes often represents non-taxonomic units such as a trophic species (*e.g.,* @yodzisCompartmentationRealAssembled1982; @williamsSimpleRulesYield2000), a feeding guild [*e.g.,* @garcia-callejasNonrandomInteractionsGuilds2023], or a segregation of species by life stages [*e.g.,* @cleggImpactIntraspecificVariation2018]. Such granularity and variation is often defined as aggregation. Such aggregation can limit the ability to make species (taxonomic) specific inferences (*e.g.,* does species $a$ eat species $b$?). It can also affect the estimates of degree distributions and more specifically generality and vulnerability in networks (in/out degree). These metrics are central to inference about the structure and complexity of networks[@beckermanForagingBiologyPredicts2006; @cleggImpactIntraspecificVariation2018]. Finally, aggregation makes it challenging to use networks in 'downstream analyses' of , for example, extinction or invasions as the identity of species and the consequences of their losses can be hidden. Despite these issues, there are justifications for representing nodes as aggregated units. Most prominent relates to when the distribution of the links between aggregated nodes may be more meaningful in terms of understanding or generalising about energy flow and distribution within the system \[REF\].

## What is captured by an edge?

In order to breakdown the definitions of an edge, it is important to introduce the concept of *potential* versus *realised* links: potential links reflect feasibility while realised links are connected to flux of some currency (typically energy; see below for more detail). Links within food webs are thus a representation of either potential links between species or fluxes within a system *e.g.,* energy transfer or material flow as the result of the feeding links between species [@lindemanTrophicDynamicAspectEcology1942; @proulxNetworkThinkingEcology2005][@dunneNetworkStructureFood2006, @pringleUntanglingFoodWebs2020]. Edges can thus correspond to different 'currencies'. There is also a myriad of ways in which the links themselves can be specified. Links between species can be treated as present or absent (*i.e.,* binary), may be defined as probabilities [@banvilleDecipheringProbabilisticSpecies2024; @poisotStructureProbabilisticNetworks2016] or by continuous functions which further quantify the strength of an interaction [@berlowInteractionStrengthsFood2004]. How links are specified thus requires intersecting both the currency being modelled and their specification. For example, feasibility is unlikely to accommodate flux, but does align with binary or probability representations. Taking a food web that consists of links representing feasible interactions among a collection of species will be meaningless if one is interested in understanding the flow of energy through the network as the links are not environmentally/energetically constrained.

## Network representations {#sec-representation}

Against these definitions of nodes and edges, networks fall into two major 'types': metawebs, traditionally defined as all the *potential* interactions for a specific species pool [@dunneNetworkStructureFood2006]; and realised networks, which is the subset of interactions in a metaweb that are *realised* for a specific community at a given time and place. The fundamental differences between these two network representations are the spatial scale at which they are constructed and the associated processes that are assumed to drive pattern at these scales.

A metaweb is, at its core, a list of *feasible* interactions between pairs of species. The feasibility for a given pair is derived from the complementarity (phylogenetic relationships) of their traits, typically aligned with feeding. Feasibility can be further refined by *co-occurrence* leading to the transition from a *global* to *regional metaweb*. Metawebs thus provide a means to identify evolutionarily plausible links, regionally plausible interactions, the set of ecologically possible, *i.e.,* forbidden, links [@jordanoSamplingNetworksEcological2016] and ultimately a definition of the plausible *complete* diet of a species [@strydomGraphEmbeddingTransfer2023].

In contrast, realised networks are typically more localised in space and time, and the links between species are contingent on the co-occurrence of species, the role of the environment, and mechanisms of diet choice. Fundamentally this means that the presence/absence of a link is the result of the 'behaviour' of the species and even when the realised network is presented as a binary matrix, the edges imply a function is available to define the strength of an interaction. A realised network is therefore not simply the downscaling of a metaweb to a smaller scale (*e.g.,* moving from the country to the 1x1 km^2^ scale based on fine-scale species co-occurrence). Instead, realised webs capture processes that determine the realisation of an interaction and flows of energy in a community. Specifically, in realised webs, the definition of an edge shifts from being determined by feasibility to that of choices and consequences that centre around energy. If one were to take the same community of species and constructed both a metaweb and realised network the two networks might have the same species but would be structurally different, owing to the differences in the 'rules' constraining the presence of links. This distinction between metawebs and realised webs lead to a further insight. Links that are absent in a metaweb can conceptually (although not always practically) be treated as being truly absent. However, links that are absent in a realised network cannot be considered as truly absent but rather as absent due to the broader environmental/community context.

# From Nodes and Edges to Process and Constraints {#sec-process}

In the previous section we discussed how the definition of nodes and edges, representing different scales and processes, lead to the concept of a metaweb and a realised web. The fundamental take-homes are that nodes vary in their resolution, edges vary in what kind of process they represent and the intersetion of these, defined by meta- vs. realised webs, underpins distinct lines of enquiry and constraints on the type of inference we can make with networks. Here we reveal five core constraints across evolutionary and ecological scales that further delineate the transition from meta- to realised webs, exposing processes that determine the nature of links among nodes: evolutionary compatibility, co-occurrence, abundance, diet choice, and non-trophic interactions [@fig-process].

![Aligning the various processes that determine interactions (right column) with the different network representations (left column). First, we start with a **global metaweb** this network captures all possible interactions for a collection of species in the global context. However, within the global environment different species occur in different regions (region one = yellow and region 2 = orange), and it is possible to construct two different metawebs (**regional metawebs**) for each region by taking accounting for the co-occurrence of the difference species - as shown here we have two regions with some species that are found in both regions (blue) and others endemic to either region one (yellow) or region two (orange). However even within a region we do not expect all interactions to be realised but rather that there are multiple configurations of the regional metaweb over both space and time. The 'state' of the different **realised networks** is ultimately influenced not just by the co-occurrence of a species pair but rather the larger community context such as the abundance of different species, maximisation of energy gain, or indirect/higher order interactions.](images/anatomy.png){#fig-process}

## Processes that determine the feasibility of an interaction {#sec-process-feasibility}

Evolutionary compatibility and co-occurrence are the two principle processes that 'act' at the species pair of interest and define feasibilty. The scale of inference and set of processes embodied in these two constraints typically combine to define a 'list' of interactions that are viable/feasible and defined strictly as present/absent. Reflecting on the previous section, nodes are typically species and rules defining edges are defined by trait complementarity (phylogenetic) and/or co-ooccurence. Here we provide more insight into each process.

**Evolutionary compatibility**

This constraint is defined by shared (co)evolutionary history between consumers and resources [@segarRoleEvolutionShaping2020; @gomezEcologicalInteractionsAre2010; @dallarivaExploringEvolutionarySignature2016; @rossbergFoodWebsExperts2006] which is manifested as 'trait complementarity' between two species [@benadiQuantitativePredictionInteractions2022]. In this body of theory, the consumer has the 'correct' set of traits that allow it to chase, capture, and consume the resource. Interactions that are not compatible are defined as forbidden links [@jordanoSamplingNetworksEcological2016]; *i.e.,* they are not physically possible and will *always* be absent within a network.

Networks do not properly arise from models based on this constraint. Instead, interacting species pairs are defined and these are represented as binary (possible vs forbidden) or probabilistic [@banvilleDecipheringProbabilisticSpecies2024]. For example, in the metaweb constructed by @strydomFoodWebReconstruction2022 probabilities are quantified as the confidence of a specific being *possible* between two species. A network constructed on the basis of evolutionary compatibility is conceptually aligned with a 'global metaweb', and gives us information as to the global feasibility of links between species pairs despite the fact that they do not co-occur (see @fig-process).

**(Co)occurrence**

The co-occurrence of species in both time and space is a fundamental requirement for an interaction between two species to occur (at least in terms of feeding links). Although co-occurrence data alone is insufficient for building an accurate and ecologically meaningful representation of *feeding links* [@blanchetCooccurrenceNotEvidence2020], it is still a critical process that determines the realisation of a feeding. Knowledge on the co-occurrence of species allows us to spatially constrain a global metaweb to reflect regional metawebs [@dansereauSpatiallyExplicitPredictions2024]. In the context of @fig-process this would be the metawebs for regions one and two.

We reinforce that these two constraints don't deliver a network *per se,* but a list of feasble species pairs. Although it is possible to build a network from the list of interactions generated by these constraints, it is important to be aware that the structure of this network is not constrained by any community context: just because species are able to interact does not mean that they will [@poisotSpeciesWhyEcological2015; @caronTraitmatchingModelsPredict2024].

## Processes that realise networks {#sec-process-realisation}

In contrast to the above, here we highlight three processes that influence the *realisation* of an interaction between species and thus form the conceptual basis for realised networks. As we show in @fig-process, a 'truly realised' network is the product of properties of the community (**abundance** and **non-trophic interactions**) and the individual (**diet choice**). This represents a conceptual shift from considering the feasibility for species pairwise interactions to considering the edge as a representation of energy flow. Such a transition requires information about how the community, the environment and the individual *constrains* network topology as defined by consumer choice (@quinteroDownscalingMutualisticNetworks2024, @sec-representation)

**Abundance**

Abundance as a realising process emerges from a null model for energy aquisition: organisms feeding randomly will consume resources in proportion to their abundance [@stephensForagingTheory1986]. Here, abundance of different prey species influences the distribution of links in a network [@vazquezUnitingPatternProcess2009] by defining a preference linked to individuals among species meeting [@poisotSpeciesWhyEcological2015; @banvilleDecipheringProbabilisticSpecies2024]. Abundance data, linked to a derived metaweb delivers a foundation ruleset that can define the distribution and strength of links. Of note, however, is that such abundance constrained interactions are not necessarily contingent on there being any compatibility between species [@canardEmergenceStructuralPatterns2012; @momalTreebasedInferenceSpecies2020; @pomeranzInferringPredatorPrey2019].

**Diet choice**

It is well established that consumers make more active decisions than eating items in proportion to their abundance [@stephensForagingTheory1986]. Ultimately, consumer choice is underpinned by an energetic cost-benefit framework centred around profitability and defined by traits associated with finding, catching, killing, and consuming a resource [@woottonModularTheoryTrophic2023; @smithBehavioralResponsesMosaic2021]. Energetic constraints are invoked to construct networks in a myriad of ways [*e.g.,* @beckermanForagingBiologyPredicts2006; @pawarDimensionalityConsumerSearch2012; @portalierMechanicsPredatorPrey2019; @cherifEnvironmentRescueCan2024].

In contrast to metaweb 'construction' from a list of pairwise interactions, these methods deliver a realised web directly and as an emergent property of node behaviour. We also here make a distinction, developed below, with models like the Niche Model, where diet choice is implicit in it's probabilistic network generating function, but it is working to replicate the *expected* structure of the network and this structure does not emerge from node-based rules. Note that we select diet choice as a term to capture rules linked to optimal foraging [@pykeOptimalForagingTheory1984] and metabolic theory [@brownMetabolicTheoryEcology2004]; it is a sensible 'umbrella concept' for capturing the energetic constraint on of the distribution and strength of interactions.

**Non-trophic interactions**

Non-trophic interactions [see @mieleNontrophicInteractionsStrengthen2019] include competition for space, predator interference, refuge provisioning, recruitment facilitation as well as non-trophic effects that increase or decrease mortality. These interactions [@ingsEcologicalNetworksbeyondFood2009] specifically modify either the realisation or strength of trophic interactions [@golubskiModifyingModifiersWhat2011; @pilosofMultilayerNatureEcological2017; @staniczenkoStructuralDynamicsRobustness2010; @kamaruDisruptionAntplantMutualism2024] and represent interactions either direct (e.g., predator $a$ outcompetes predator $b$) and indirect (e.g., mutualistic/facilitative interactions) mechanisms. They operate on the realisation of a network of interactions by altering the fine-scale distribution and abundance of species as well as their persistence [@kefiMoreMealIntegrating2012; @kefiNetworkStructureFood2015; @bucheMultitrophicHigherOrderInteractions2024].

# Network construction {#sec-construct}

The above five processes are central to understanding the assumptions inherent in building networks. Each of these processes, or combinations thereof, deliver via these assumptions a set of constraints on what a network represents and can be used for. Here we build on the introduction of these five processes to categorise two approaches to constructing networks and in doing so also introduce more detail on methodologies.

## Why construct networks?

Networks are a representation of biodiversity. They are unique in delivering more than just estimates of species richness. A network embodies the organising structure of biodiversity and allows numerous opportunities for 'downstream' analysis, including the comparison of structures, estimation of energy flux or extinction dynamics and ultimately form the structural inputs to dynamical systems models that facilitate ecological and conservation relevant inference about productivity-diversity-stability-function relationships in space and time.

In a perfect world, we might know about all interactions. However, the empirical collection of interaction data is both costly and challenging to execute [@jordanoChasingEcologicalInteractions2016; @jordanoSamplingNetworksEcological2016; @poisotGlobalKnowledgeGaps2021]. In the absence of robust empirical data, we construct models that facilitate interpolation and gap-filling of existing empirical datasets [*e.g.,* @bitonInductiveLinkPrediction2024; @stockLinearFilteringReveals2017; @dallasPredictingCrypticLinks2017; @poisotNetworkEmbeddingUnveils2023], predict the feasibility of interaction among pairs of species, or directly predict network structure [see @strydomRoadmapPredictingSpecies2021 for a broader discussion]. Doing so clearly embeds one or more of the processes discussed in @sec-process.

## Construction through induction {#sec-construct-induct}

Constructing feasible networks and facilitating the interpolation or gap-filling of existing empirical datasets aligns with evolutionary compatibility and co-occurrence constraints. Methods embedding these scales and assumptions have been critical to the construction of 'first draft' networks for communities for which we have no interaction data [@strydomFoodWebReconstruction2022]. They are central to interpolation in data poor regions and predicting interactions for 'unobservable' communities *e.g.,* prehistoric networks [@yeakelCollapseEcologicalNetwork2014; @frickeCollapseTerrestrialMammal2022; @dunhillExtinctionCascadesCommunity2024] or future, novel community assemblages [@vanderputtenPredictingSpeciesDistribution2010]. They have the capacity to evaluate a role of interactions among species relative to their distribution by accounting for the role of the environment and the role of species interactions [@higinoMismatchIUCNRange2023; @pollockUnderstandingCooccurrenceModelling2014; @gravelBringingEltonGrinnell2019].

All methods in this inference space rest on an assumption that there are a set of 'feeding rules' that underpin interaction feasibility [@morales-castillaInferringBioticInteractions2015]. With the assumption that these rules are phylogenetically conserved [@dallarivaExploringEvolutionarySignature2016; @bramonmoraIdentifyingCommonBackbone2018], and can be specified by matching the traits between consumer and resource. They represent a class of models that work by inductive reasoning. Feeding rules can be defined in multiple ways:

The determination of the feeding rules can be defined *a priori* based expert knowledge opinions. Typically this is done on a 'trait matching' basis. An example are the paleo food web models of @shawFrameworkReconstructingAncient2024 and @roopnarineEcologicalModellingPaleocommunity2017 that specify a series of rules for a set of traits and interactions are deemed feasible if all conditions are met. Alternatively the body size ratio between the consumer and resource is often used [*e.g.,* @gravelInferringFoodWeb2013; @rohrModelingFoodWebs2010], with the idea that consumers will only utilise a resource with a body size is less than or equal to their own. However, work from @vandewalleArthropodFoodWebs2023 seems to suggest that adding morphological traits in addition to body size ratio improves model performance.

Rules can also be defined by correlating real world interaction data with suitable ecological proxies for which data is more widely available (*e.g.,* traits) using some sort of binary classifier (see @pichlerMachineLearningAlgorithms2020 for an overview). These include generalised linear models [*e.g.,* @caronAddressingEltonianShortfall2022], random forest [*e.g.,* @llewelynPredictingPredatorPrey2023], trait-based k-NN [*e.g.,* @desjardins-proulxEcologicalInteractionsNetflix2017], and Bayesian models [*e.g.,* @eklofSecondaryExtinctionsFood2013; @cirtwillQuantitativeFrameworkInvestigating2019].

Finally, graph embedding uses a combination of ecological proxies in conjunction with the known interactions to infer the latent values of species. The structural features of a known network are used to infer the position of species in a network through the decomposition of the interaction onto the embedding space. See @strydomGraphEmbeddingTransfer2023 for a detailed review of methods and @strydomFoodWebReconstruction2022 for a specific example.

There are substantial data requirements for these approaches, either reflecting expert knowledge, species traits or existing interaction data.

## Construction through deduction

In contrast to the above approaches centred on feasibility, methods reflecting abundance and diet choice typically rely on deductive reasoning and have a unique agenda to those above. Typically, models that embed abundance and diet choice constraints reference theory that allows inference about the distribution and strength of interactions. Such models are 'network topology generators' and have a strong representation in research comparing network structures along environmental gradients and delivering inference about extinctions and energy flux. They also provide the structural backbone for dynamical systems modelling to address questions about stability-structure-productivity-function relationships, secondary extinction dynamics, species invasion and climate change.

There are two broad group of models in this deductive category.

### Species agnostic networks

We refer to these networks as species agnostic, as the determination of links between species is not due to the properties of the node but rather due to the property of the *expected* structure of the network. This means these networks are usually not species specific and have a limited use in real-world context. Finally, although these models are data light (often requiring only species richness and the number of expected links) there are assumptions that need to be made regarding what the expectations are for network structure. There are two sub-categories of species agnostic networks:

Random network models [*e.g.,* @erdosRandomGraphs1959; @fortunaHabitatLossStructure2006; @bascompteNestedAssemblyPlantanimal2003] represent a 'process free' model. These models are not explicitly tied to a process discussed in @sec-process, rather links are randomly distributed among nodes. Although these models lack real world tractability [@bascompteNetworksEcology2007] they are often used as a 'null hypothesis' to ask questions about network structure [*e.g.,* @banvilleWhatConstrainsFood2023; @strydomSVDEntropyReveals2021].

Stochastic network models use a probabilistic rule-set about diet choice and niche breadth to reflect fundamental ideas of foraging biology. **RATHER THAN ALLESINA, I'D SUGGEST PRESENTING THE 5 RULES IN THE ADBM EXPLANATION OF THE NICHE MODEL** These models that are based on the compartmentation and acquisition of energy for species at different trophic levels [@allesinaFoodWebModels2009; @krauseCompartmentsRevealedFoodweb2003] and that network structure can be determined by distributing interactions along single dimension [the 'niche axis'; @allesinaGeneralModelFood2008]. Typically these models parametrise some aspect of the network structure [although see @allesinaFoodWebModels2009 for a parameter-free model]. These models include the most commonly used network generator, the Niche model [@williamsSimpleRulesYield2000], as well as the original Cascade model [@cohenCommunityFoodWebs1990] and the derived Nested hierarchy model [@cattinPhylogeneticConstraintsAdaptation2004]. These models often form the basis for dynamic models *e.g.,* the allometric trophic network [@broseAllometricScalingEnhances2006; @schneiderAnimalDiversityEcosystem2016a] and bioenergetic food web models [@delmasSimulationsBiomassDynamics2017].

### Species-specific networks

These models capture the behaviour of the nodes by explicitly taking into account the properties of the different species in the community. Which means that there is a degree of variance in which links are predicted between species unlike the more 'static' predictions made by inductive models. However, these networks are 'costly' to construct in real world settings (requiring data about the entire community, as it is the behaviour of the system that determines the behaviour of the part) and also lack the larger diet niche context afforded by metawebs.

Neutral networks are built on the assumption that foraging decisions are tied *only* to the abundance of species within the community [@canardEmpiricalEvaluationNeutral2014; @krishnaNeutralnicheTheoryNestedness2008]. Here links are soley determined by the relative abundance of the different species in the community. Although it is highly unlikely that abundance is the only determaninant of interactions work by @pomeranzInferringPredatorPrey2019 showcases how these neutral processes can be used in conjunction with inductive models to construct more refined/localised networks.

There is a broader group of models that focus on determining interactions in terms of energetic constraints on diet breadth, often using the ratio of consumer-resource bodysize as a proxy for capturing the energetic constraints of feeding. Models such as those developed by @portalierMechanicsPredatorPrey2019 and @woottonModularTheoryTrophic2023 are similar to the mechanistic approaches discussed in @sec-construct-induct, however instead of determining interactions based on mechanistic feasibility it is rather constrained by the energetic cost of predation. Note that although these models do not place any explicit constraints on the expected structure of the network, the links should still be considered as 'realised' owing to the energetic constraint placed on links. A different subset of diet models [*e.g.,* @beckermanForagingBiologyPredicts2006; @petcheySizeForagingFood2008] use a diet choice approach, however similar to the stochastic network models they also embed assumptions on network structure. Thus these models predict both interactions and network structure simultaneously, although they would benefit in being refined by more explicitly accounting for trait-based (*i.e.,* feasibility) paramterisation [@curtsdotterEcosystemFunctionPredator2019].

# Making Progress with Networks {#sec-progress}

**THIS SECTION NEEDS WORK. THE FIRST TWO SENTENCES SAY TOO MUCH AT ONCE AND NOT ENOUGH**

It is probably both this nuance as well as a lack of clear boundaries and guidelines as to the links between network form and function [although see @delmasAnalysingEcologicalNetworks2019] that has stifled the 'productive use' of networks beyond the inventorying the interactions between species. Although progress with using networks as a means to address questions within larger bodies of ecological theory *e.g.,* invasion biology [@huiHowInvadeEcological2019] and co-existence theory [@garcia-callejasNonrandomInteractionsGuilds2023] has been made we still lack explicit guidelines as to what the appropriate network representation for the task at hand would be, and as highlighted in Box 1, underscores the need to evaluate exactly what process a specific network representation captures as well as its suitability for the question of interest. Below we present a mapping of what we believe are some of the key questions for which interaction networks can be used to the different networks representations that are most suitable, as well as highlight some of the methodological challenges that still need to be improved upon.

## Making use of the different network representations

**Methodological challenges**

1.  Tools that allow us to estimate both the feasibility as well as realisation of links: Currently most approaches to modelling realised networks fail to explicitly account for any form of evolutionary constraint [although see @vandewalleArthropodFoodWebs2023 and @woottonModularTheoryTrophic2023] and we need to develop either an ensemble modelling approach [@beckerOptimisingPredictiveModels2022; @terryFindingMissingLinks2020] or tools that will allow for the downsampling of metawebs into realised networks [*e.g.,* @roopnarineExtinctionCascadesCatastrophe2006].
2.  Is there something in generalisable models that 'combine' different processes/aspects (*e.g.,* using body size as a catch all) versus limited models that allow you to unpack things bit-by-bit (*i.e.,* process by process). So @woottonModularTheoryTrophic2023 *may* (TBD) span the gamut but it lacks the ability to unpack... Although maybe the terms do?
3.  Modelling interaction strength: Although realised networks are more closely aligned with *explicitly* capturing interaction strength we lack models that allow us to quantify this [@wellsSpeciesInteractionsEstimating2013; @strydomRoadmapPredictingSpecies2021].
4.  How do we validate our predictions?: Progress has been made to assess how well a model recovers pairwise interactions [@strydomRoadmapPredictingSpecies2021; @poisotGuidelinesPredictionSpecies2023], but we still lack clear set of guidelines for benchmarking the ability of models to recover structure [@allesinaGeneralModelFood2008]
5.  Something about making what we do with networks more tractable in the applied space? *e.g.,* @dansereauOvercomingDisconnectInteraction2024

**Theory challenges**

1.  Core Theory Advancement: Do the decades of insights arrived at for stability-diversity-productivity relationships with tri-trophic or diamond shaped models hold for complex communities (10's-100s) [@danetResponseDiversityMajor2024]; How will spatial and temporal variation in climate and productivity drive change in complex ecosystems. Necessary to move to predicting changes in biodiversity per se, ecosystem functions and identifying sensitive and robust species and portions of communities.
2.  How will novel communities interact? How will range shifts and invasions result in new/novel community assemblages. And then also the intentional changes of species compositions through rewilding.
3.  Does rewiring happen and does it deliver robustness? Specific sub points to consider here is persistence, especially persistence to perturbation. Again, dynamic networks and network/community assembly and finally extinctions [@dunhillExtinctionCascadesCommunity2024].
4.  When do invasive species enhance or decimate communities? When do reintroductions work? [@woosterAustraliasRecentlyEstablished2024]
5.  Are there temperature threshold to community collapse
6.  Can socioeconomic networks combined with biological networks drive understanding of externalities?
7.  Can paleoecological data from deep time hyperthermal events provide sufficient insight into the targets, pace and recovery times from rapid climate events?

![Here we highlight some of the outstanding questions in both network as well as general ecology, as well as some of the outstanding methodological challenges with regards to constructing food webs (shown in orange) that we are faced with.](images/NetworkFuture.png){#fig-future}

# Concluding remarks

Having a clear understanding of the interplay between network representations and the processes that they are capable of encoding is critical if we are to understand exactly which networks can be used to answer which questions. As we highlight in Box 1 the different network representations have different potential uses and it should be clear that there is no 'best' network representation but rather a network representation that is best suited to its intended purpose. In providing a formalisation regards to the assumptions and mechanisms that need to be explicitly taken into consideration when deciding to use (and construct) networks we hope to prevent the unintentional misuse or misinterpretation of networks as well as provide a starting point from which we can develop a better framework for the applied use of networks to answer questions that are not only pressing within the field but also within broader biodiversity science.

# References {.unnumbered}

::: {#refs}
:::