---
title: Navigating food web prediction; assumptions, rationale, and methods
author:
  - name: Tanya Strydom
    id: ts
    orcid: 0000-0001-6067-1349
    corresponding: true
    email: t.strydom@sheffield.ac.uk
    role: 
    - conceptualization: lead
    - methodology: supporting
    affiliation:
      - id: sheffield
        name: School of Biosciences, University of Sheffield, Sheffield, UK
  - name: Jennifer A. Dunne
    id: jad
    orcid: 0000-0002-3789-9561
    corresponding: false
    role: 
    - conceptualization: lead
    - methodology: supporting
    affiliation:
      - id: sfi
        name: Santa Fe Institute, 1399 Hyde Park Road, Santa Fe, NM 87501, USA
  - name: Timothée Poisot
    id: tp
    orcid: 0000-0002-0735-5184
    corresponding: false
    role: 
    - conceptualization: lead
    - methodology: supporting
    affiliations:
      - Université de Montreal
      - Québec Centre for Biodiversity Sciences
  - name: Andrew P. Beckerman
    id: apb
    orcid: 0000-0002-7859-8394
    corresponding: false
    role: 
    - conceptualization: lead
    - methodology: supporting
    affiliations:
      - ref: sheffield
funding: "The author(s) received no specific funding for this work. Well they did I just haven't done the homework"
keywords:
  - food web
  - network construction
  - scientific ignorance
abstract: |
  TODO
date: last-modified
bibliography: references.bib
citation:
  container-title: TREE (one can dream...)
number-sections: true
---

At the heart of modern biodiversity science are a set of concepts about biodiversity, community structure, productivity, and asynchrony, and how they define the stability, resilience, and dynamics of complex communities. The use of species interaction networks provides a powerful abstraction that one can use to help quantify, conceptualise, and understand these concepts. However, network ecology has its own nuance and idiosyncrasies that not only provide a barrier to entry but causes dissonance even within the field [@dormannRisePossibleFall2023]. This is perhaps particularly pervasive within the space of network prediction...

One of the fundamental challenges that we are faced when working with and studying interaction networks (and, within the context of this manuscript, specifically food webs) is that there is a scarcity of 'real world' interaction data [@hortalSevenShortfallsThat2015; @poisotGlobalKnowledgeGaps2021]. The difficulty of recording interactions in the field [@jordanoChasingEcologicalInteractions2016; @jordanoSamplingNetworksEcological2016] has necessitated that researchers find and develop alternative means to construct and build food webs using **models** [@morales-castillaInferringBioticInteractions2015; @strydomRoadmapPredictingSpecies2021]. Over the past decade, there has been a proliferation of tools and processes for characterising food webs, these models span a wide range of philosophies that rely on different approaches, data, and definitions, which ultimately determine how the food web is constructed and coded. Although the development of these different models have carved out the path for constructing either synthetic, ecologically plausible networks [@poisotSyntheticDatasetsCommunity2016], or providing 'first draft' networks that can be utilised in real world settings [@strydomFoodWebReconstruction2022] we are still lacking in discussions that are explicitly comparing and contrasting how the way one chooses to approach the task of constructing a food web is introducing (and ultimately embedding) specific assumptions and hypotheses [@petcheySizeForagingFood2008]. Most attempts that focus on comparing and contrasting models are focused on the same group of **model families** [@williamsSuccessItsLimits2008; @pichlerMachineLearningAlgorithms2020] and only benchmark the different models as opposed to contextualising them within the bigger framework of understanding the data needs of the different models, as well as how the resulting network is defined and structured. As food webs become a more integrated part of some of the broader fields of ecology [@bhatiaNetworkbasedRestorationStrategies2023; @thuillerNavigatingIntegrationBiotic2024] it is critical that we review these different model families as a whole (not only in isolation), and move away from simply benchmarking the performance of these different model families. This is important because different models impose different constraints upon themselves and will not only delimit and dictate the potential questions one will be able to ask [@petcheyFitEfficiencyBiology2011] but also determine the appropriate research setting for which the model (and resulting network) can be used. For example the use of 'structural food webs' are useful for developing additional theory such as re-wiring of networks [@staniczenkoStructuralDynamicsRobustness2010] but would be meaningless if one's intention is to produce a location-specific network \[do we need an *e.g.,* ref??\]. This will allow us to ensure the right models are being used to answer the right questions, particularly within the context of trying to accelerate cross-cutting research in the face of global change.

When navigating the seas of using and constructing food webs the researcher needs to be able to clearly articulate and define the parameters that are used to define their food web(s) of interest. This will aid them in being able to select the correct model to help them to reach their goal. In order to be able to make informed decisions it is important that one has a strong grasp of exactly what it means to 'code'/define a food web (@sec-network-anatomy), a clear understanding of why one wants to predict a food web (@sec-network-why), and ultimately one needs to be able to asses and evaluate which model family is going to best match up with the goal of network prediction (@sec-network-build). Here we specifically aim to look at not look at only the performance of the different models but also initiate a (thus far lacking) discussion around how the interplay between the language used to define networks and the underlying theory/philosophy should also be a part of the broader discussion when it comes to the task of 'model selection'.

![Panel **A** shows the many ways in which a food web can be defined and described at the node, edge, and even network level. Panel **B** (will) shows how the way in which we predict networks also limited and often focuses only only predicting the structure of a network (the final networks is parametrised by the expected structure of the network) or the interactions between species (the final network is determined by the behaviour of the nodes). These different models also encode different philosophies/hypotheses not only as to what determines how a network will look like but also how the final network itself is encoded *i.e.,* its anatomy. (*aside:* there is the potential to either try and visually summarise how the different model families define a network (so repeating the motifs used in the ANATOMY panel) alternatively it would be cool to try and have a panel C that tries to quantify the different 'data ingredients' you would need to try and construct a network, this would probably be very visually overwhelming though...)](images/concept_2.png){#fig-concept}

## The anatomy of a food web {#sec-network-anatomy}

Defining a food web seems simple, it is the representation of the interactions (edges) between species (nodes), however the definition of 'edges' and 'nodes', as well as the scale at which they are aggregated can take many forms. As highlighted in @poisotDescribeUnderstandPredict2016 networks can be constructed at the population (the links among individuals), community (the links between species), or metacommunity (fluxes between locations) level. Even if one were to limit their scope to thinking of interaction networks only in terms of food webs at the community-level there are still many ways to define the various components of the network [Panel A of @fig-concept], one needs to understand the different intentions/assumptions that are made when a food web is constructed. Although the main intention of a food web is to capture and represent the feeding links between species there are many ways to define the nodes (*e.g.,* species or taxonomic group), edges (*e.g.,* **potential** or **realised feeding links**), the magnitude of the edges (*e.g.,* binary vs probabilistic), and even how the network itself is delimited (does it represent an aggregation of interactions over time?).

### How do we define a node?

Although this may seem an elementary question in the context of food webs --- a node *should* represent a (taxonomic) species, the reality is that nodes can often represent an aggregation of different species - so called 'trophic species' or segregation of species by life stages. Representing nodes as non-taxonomic species can be useful in certain contexts [@williamsSimpleRulesYield2000] and in cases where the adult and larval stages of a species have different diets it may make ecological sense [@cleggImpactIntraspecificVariation2018] meaning that it is not uncommon that networks often have nodes that have different definitions of a 'species' *e.g.* consisting of both taxonomic and trophic species. Practical implications of how we are aggregating the nodes is that the resolution may not always be 'pixel perfect' *i.e.,* we may be unable to assess the co-extinction risk of a species pair, however there is value in having nodes that represent an aggregation of species, as these convey a much more general overview of how the links are distributed within the community.

### What is meant by an edge?

As discussed earlier there are many ways to define the links between species --- even feeding links. At its core links within food webs can be thought of as a representation of either the flow of a resource \[ref\], realised [@pringleUntanglingFoodWebs2020] or potential [@dunneNetworkStructureFood2006] feeding links, or energy transfer and material flow [@lindemanTrophicDynamicAspectEcology1942]. How we specify links will influence the resulting structure of the network - and the inferences we will make thereof. For example taking a food web that consists of links representing *potential* feeding links between species will be meaningless if you are interested in understanding *e.g.,* the flow of energy through the system as the links within the network are over overrepresented. In addition to the various ways of defining the links between species pairs there are also a myriad of ways in which the links themselves can be quantified. Links between species are often treated as being present or absent (*i.e.,* binary) but it is also possible to use probabilities [which quantifies how likely an interaction is to occur, @poisotStructureProbabilisticNetworks2016] or continuous measurements [which quantifies the strength of of an interaction, @berlowInteractionStrengthsFood2004]. Moving away from a purely binary way of representing allows us to quantify a level of (un)certainty of our knowledge of interactions (*i.e.,* moving from being able to ask if are they occurring to quantifying how likely they are to occur) does add an additional level of 'complexity' to the construction and interpretation of networks, but ultimately it allows us to capture more information at different scales (Banville, in prep).

### Putting the parts together; what does it mean?

The ingredients one uses to construct networks from nodes and edges generates a unique representation of the mechanisms (see Box 1 - Mechanisms that determine feeding links) that allow inference and reasoning about the structure, aspects of dynamics (*e.g.,* stability), and potentially the function of communities (*e.g.,* flux). It is thus beneficial to keep in mind that in the process of 'codifying' a network one is already embedding some sort of hypothesis as to the nature of the feeding links between species [@proulxNetworkThinkingEcology2005; @brimacombeShortcomingsReusingSpecies2023]. Here it may be meaningful to contextualise the different 'types' of food webs within the larger research programmes (or even practical needs) that have been driving the construction of them.

::: {#box-mechanisms .callout-note}
## Box 1 - Moving between the feasibility and reality of interactions

Before thinking about the ways in which we can predict networks it is perhaps meaningful to take a step back and think about the different criteria that must be met in order for an interaction to be able to occur between two species, specifically thinking of this in terms of distinguishing between the feasibility or realisation of an interaction and how these are determined (and defined by) different 'rules'/hierarchical mechanisms. If we look at this feasibility-reality continuum (@fig-feasibility) it is clear how the different predictive approaches usually encapsulate one specific 'rule' and fail to account for the rest. This is not to say that this shortcoming should be viewed as a 'bug' but rather a 'feature' of the field as it allows one to engage with, as well as construct networks at different scales, which is particularly valuable if one take into consideration the considerable 'data cost' of predicting well resolved, realised networks in comparison to constructing high-level metawebs. However, it is important that there is an awareness and acknowledgement of where within this feasibility-reality one is working at and how this will impact and limit the contexts in which the resulting network can be used and applied within.

![A schematic overview of the feasibility-reality axis that defines species interactions (and the resulting networks), specifically in the context of the underling mechanisms that determine the 'rules of engagement'. As one moves from feasibility towards reality the way in which the edges between nodes are defined (but not necessarily quantified) begins to shift from representing the *possibility* of an interaction occurring between two species to the quantification of *strength* of the interaction. This is because as we move down this axis we begin to incorporate additional 'rules' that allow us to begin approaching a much more resolved reality of the interactions for a specific location until we reach a point where we are able to encapsulate the energetics of the system. Moving from the most general to the most specific it goes: 1.) the co-occurrence of species, 2.) the mechanistic feasibility of the interaction, 3.) the likelihood of two potentially interacting species to meet, 4.) the energetic 'cost' of interaction (building on the feeding ecology framework) and 5.) the environmental cost of searching for and capturing prey (building on the movement ecology framework).](images/concept_2.png){#fig-feasibility}

> Aside: this is maybe not the IRL mechanisms (e.g. we species must co-occur, then they must meet, then we can determine if the interaction will happen) but it makes modelling sense? IDK... anyway we can shuffles things around

### Mechanisms

**1. Co-occurrence**

Although the assumption that because two species are co-occurring it must mean that they are interacting is presumptive [@blanchetCooccurrenceNotEvidence2020] it is of course impossible for two species to interact (in terms of feeding links) if they are not co-occurring We are co-occurring and so it represents the *basic* requirement for the interaction between a species pair

**2. Mechanistic feasibility**

This is based on the idea of forbidden links introduced by @jordanoSamplingNetworksEcological2016, specifically that there must be some degree of *trait complementarity* that allows a predator to chase, capture, kill, and consume, its prey. This is probably the level that the idea of a metaweb [@dunneNetworkStructureFood2006] is most applicable. Within the network prediction 'field' this is perhaps the most developed space. Predictive models run the gamut including mechanistic models [@morales-castillaInferringBioticInteractions2015], binary classifiers [@pichlerMachineLearningAlgorithms2020], and graph embedding [@strydomGraphEmbeddingTransfer2023]. It is probably worth having a brief interlude here to be really clear that just because an interaction is probabilistic it does not make it quantitative - it is still binary but at least we do have some way of saying how confident we are in the feasibility of an interaction - not how likely it is to occur - per se.

**3. Likelihood of meeting (there must be a better word)**

This is probably the point where we start to shift from a presence/absence way of defining interactions and start moving into the 'qualitative'/weighted space - we are not 'determining' if the interaction is feasible but rather making an assumption on prey selection based on species likelihood of 'meeting'.

> Not sure if there are models that 'only' consider abundance but it is rather more of a building block in some of the models that are more relevant to the next steps.

**4. Prey choice (energy cost-benefit)**

Now we move into the 'prey choice' space - specifically the cost-benefit side of things that falls under feeding ecology. Examples of models [@beckermanForagingBiologyPredicts2006; @petcheySizeForagingFood2008]

**5. Environmental 'energy budget'**

Builds primarily on the ideas presented in @cherifEnvironmentRescueCan, which is essentially a take on movement ecology? What it boils down to is being able to quantify the cost of movement *i.e.,* the physical constraints that the environment imposes on a species... 

> The role of the environment: Although we only explicitly take the environment into consideration in the final 'rule' the role of the environment (at least from a modelling perspective) is perhaps also meaningful at the first step (we expect environmental filtering to be the main determinant of species distributions in the absence of species interactions) I think the way in which we perceive the environment at the extreme of these two ends of the spectrum comes down to a question of scale?? Or at least the degree of information that we need at need at these two points is very different. RE occurrence it is a case of survival and for movement it is a case of performance??

### Notes on the 'ensemble' approach

IDK I just really want to articulate that we can't feasibly be functioning at a 'level 5' at all times because of other real world constraints such as the fact that the amount of data one needs is simply not going to be feasible in most cases... Also that you (probably) need to go through the whole step-wise process because (at least for now) models at the 'higher' levels are overlooking some of the 'lower' levels *e.g.,* the the ADBM model (in its current form) has no way of accounting for forbidden links and so it is probably going to end up predicting some links that are strictly speaking 'forbidden' at the trait level (I guess this kind of thinking will probably open the door for talking about benchmarking should we still want to keep the  model comparison angle).

:::

## Why do we want to predict food webs? {#sec-network-why}

As discussed in @sec-network-anatomy there are many ways to define a food web, meaning that there are equally as many reasons one might be interested in predicting a food web. However we may think of two primary drivers for wanting to predict networks (Panel B @fig-concept), namely an interest in generating a set of ecologically plausible networks (*i.e.,* being able to describe networks using a model) or being able to recover (predict) location specific, 'realised', interactions for a specific species community (*i.e.,* being able to predict/infer the interactions between species). Of course these two categories are not distinct, mutually exclusive, groups but can rather be viewed as operating on a continuum ranging from a need for generality (*i.e.,* creating a network that, when taken in aggregate, the distribution of links (interactions) between nodes (species) are ecologically plausible) to a need for specificity (*i.e.,* local-level predictions between specific species pairs). Although the ability to predict 'real-world' interactions (and the resulting food webs) can have more intuitive 'real world' applications *e.g.,* being able to 'recover' food webs that have since gone extinct [@dunneCompilationNetworkAnalyses2008; @yeakelCollapseEcologicalNetwork2014], using pairwise interactions to understand species distributions [@pollockUnderstandingCooccurrenceModelling2014] or even co-extinction risk [@dunnSixthMassCoextinction2009], a more structural approach to network construction affords one an opportunity to interrogate some of the more high-level mechanisms that are structuring networks (Box 1).

It is perhaps more important that when one is talking about 'why' they want to predict networks to articulate exactly what anatomical part of the food web we are interested in scrutinising.

## How do we predict food webs? {#sec-network-build}

Selecting a model for the task of network prediction should come down to two things; what *aspect* of a food web one is interested in predicting, and what data are available, necessary, and sufficient. As shown in panel B of @fig-concept the interest in a network is (usually) at either the 'structural' or 'interaction' level and the development of models for the task of network prediction often focus on high fidelity (performance) at one of these scales. With this in mind it is beneficial to think of the different model families relative to these two different goals; here we refer to models that are used to predict the structure of a network as **topology generators** and models developed to infer the interactions for a given species pool as **interaction predictors**. It is meaningful to make this distinction because although it is possible to construct a food web given using an *interaction predictor* the models themselves lack any sort of parametrisation of the network structure and so the resulting network is a poor reflection of the actual network structure [@caronTraitmatchingModelsPredict2024]. This is primarily because *interaction predictors* are models that evaluate the feasibility of an interaction between species pairs and not in the context of feasibility at the community level. Models themselves are a reflection of the different goals and intentions of the research program from which they are developed and are often 'described' by a specific mechanism that will determine the resulting structure or interactions (Box 1). Models such as the niche [@williamsSimpleRulesYield2000] or cascade [@cohenCommunityFoodWebs1990] were developed with the intent of being used to understand the *structural* aspects of food webs, specifically how links are distributed amongst species in the community, whereas bayesian [@cirtwillQuantitativeFrameworkInvestigating2019] or trait hierarchy [@shawFrameworkReconstructingAncient2024] models have been developed on the basis that the traits of a species are the underlying mechanism in determining the feasibility of interactions (*i.e.,* species $a$ has the capacity to eat species $b$). Along with predicting different anatomical parts of a food web the different models have varying degrees of data that are needed to 'parametrise' the network. Once these two limitations are assessed and addressed it is then possible to select the model (or model family) that will best be able to capture food web feature that the researcher is most interested in (see Box 2 - Assessing model outputs). It is thus clear that (realistically) there will probably never be a 'best fit' tool that is able to construct a food web that will span the entire range of needs, and rather the responsibility lies with the researcher to be aware of not only the underlying philosophy of the specific toolset (as this could have knock-on effects when using those networks for downstream analyses/simulations; pers. comms. Beckerman, 2024), but also how well the tool is able to retrieve the specific network or interaction properties that is of interest.

> In order for a model to formalise a 'complete' food web it is necessary to formalise two aspects of the network, 'who eats whom' (to determine the links between nodes) as well as the structure of the network (to limit the distribution of links), however most models are inclined to focus on one of the two aspects [panel B of @fig-concept].

> Crucially most topology generators lack some key data on the interaction between species (this can be because of how the model itself defines species or the way in which links are assigned in the network) and interaction predictors lack some sort of parametrisation of network structure [just because two species can interact it does not mean that they will, @poisotSpeciesWhyEcological2015].

> What is the purpose of generating a network? Is it an element of a bigger question we are asking, *e.g.,* I want to generate a series of networks to do some extinction simulations/bioenergetic stuff OR are we looking for a 'final product' network that is relevant to a specific location? (this can still be broad in geographic scope).

### Model families

As there are many food web models to choose from it is perhaps useful to think about the models in terms of model families, a summary of these families is presented in @tbl-families and along with @fig-dendro highlights the differences and similarities of the philosophies and assumptions that determine a network. A more extensive overview of the different models that fall with in the different model families can be found in [SuppMat 1](https://beckslab.github.io/ms_t_is_for_topology/notebooks/model_descriptions-preview.html) and for a more detailed breakdown of the different 'traits' of the model families refer to [SuppMat 2](https://beckslab.github.io/ms_t_is_for_topology/notebooks/model_qualitative-preview.html).

| Model family       | Theory                                                                                                               | Network predicted      | Nodes represent | Links represent         | Interaction   | Key reference                                                                          |
|-----------|-----------|-----------|-----------|-----------|-----------|-----------|
| null               | Links are randomly distributed within a network                                                                      | structural             | agnostic        | feeding links           | binary        |                                                                                        |
| neutral            | Network structure is random, but species abundance determines links between nodes                                    | structural             | species         | feeding links           | binary        |                                                                                        |
| resource           | Networks are interval, species can be ordered on a 'niche axis'                                                      | structural             | trophic species | subdivision of resource | binary        | @williamsSuccessItsLimits2008                                                          |
| generative         | Networks are determined by their structural features                                                                 | structural             | agnostic        | links                   | binary        |                                                                                        |
| energetic          | Interactions are determined by foraging theory (feeding links)                                                       | interaction            | species         | feeding links           | quantitative  |                                                                                        |
| graph embedding    | Interactions can be predicted from the latent traits of networks                                                     | interaction            | species         | potential feeding links | probabilistic | @strydomGraphEmbeddingTransfer2023                                                     |
| trait matching     | Interactions can be inferred by a mechanistic framework/relationships                                                | interaction            | species         | feeding links           | binary        | @morales-castillaInferringBioticInteractions2015                                       |
| binary classifiers | Interactions can be predicted by learning the relationship between interactions and ecologically relevant predictors | interaction            | species         | feeding links           | binary        | @pichlerMachineLearningAlgorithms2020                                                  |
| expert knowledge   | 'Boots on the ground' ecological knowledge and observations                                                          | interaction            | species         | feeding links           | binary        |                                                                                        |
| data scavenging    | Webscraping to create networks from online databases                                                                 | interaction            | species         | feeding links           | binary        | @poisotSyntheticDatasetsCommunity2016 (f you squint?)                                  |
| co-occurrence      | co-occurrence patterns arise from interactions so we can use these patterns to reverse engineer the interactions     | co-occurrence patterns | species         | association links       | binary        | @kuschEcologicalNetworkInference2023 (although more plant-plant *i.e.* non-trophic...) |

: A summary of the different families of tools that can be used to generate food webs, this includes a brief description of the underlying philosophy of the family as well as how the different elements (nodes and edges) of the generated network represents. {#tbl-families}

![Dendrogram of the trait table using a hierarchical clustering model, This is based off of the traits table in SuppMat 2)](images/dendo.png){#fig-dendro}

::: {#box-benchmarking .callout-note}
## Box 2 - Assessing model outputs

Although understanding the underlying philosophy of the different model families is beneficial it is also important to understand in what situations the different families are likely to preform well or poorly. When we are assessing the performance of the different model families it is beneficial to think of benchmarking these assessments based on a broader basis than just its ability to correctly recover network structure or pairwise interactions. When thinking about how to benchmark models it is perhaps beneficial to take a step back and once again assess what are the needs of the researcher (@sec-network-why) and linking this back to what aspects of the network (@sec-network-anatomy) are of importance and assess the performance of a model within those parameters.

**Benchmarking**

Benchmarking how well a model is doing to capture the desired elements of a network is also a task that required some thought and contemplation. Even if we think about the predicting the structure of a network it is possible that two networks may have the same number of nodes and links but that those links may be distributed in very different ways. Thus it is important to think critically about the suite of summary statistics that are used to assess a model, since there is no one 'silver bullet' summary statistic that will be able to assess if a model is able to fully replicate an empirical network [@allesinaGeneralModelFood2008]. One of the main challenges when assessing the ability to retrieve pairwise interactions is that food webs are sparse (that means that there are few links given the number of species) and it is important that we are able to discern between a model that is able to correctly predict interactions that do (true positives) and not (true negatives) occur and one that is simply predicting a lack of interactions [@poisotGuidelinesPredictionSpecies2023]. For more detailed methods as to how benchmarking was done refer to [SuppMat 3](https://beckslab.github.io/ms_t_is_for_topology/notebooks/model_quantitative-preview.html)

![Difference between real and model network property. S1 - S5 represent the different motif structures identified in @stoufferEvidenceExistenceRobust2007 which are S1: Number of linear chains, S2: Number of omnivory motifs, S4: Number of apparent competition motifs, and S5: Number of direct competition motifs](images/topology.png){#fig-topology}

**Data cost**

This includes thinking about the need for additional data sources (such as trait or phylogenetic data), the computational cost, as well as the time it might take to generate a network, *e.g.,* binary classifiers require an (often times) extensive list of additional trait data for the model training process, which limits predictions to communities for which you do have the relevant auxiliary data available.

**Philosophical constraints**

Probably mentioned elsewhere but basically are we constructing networks because we want to make real-world, case-specific predictions *e.g.,* for a conservation area or do we want to just have a set of ecologically plausible networks we can use for theoretical stuffs. Need to discuss the key differences and implications between predicting a **metaweb** (*sensu* @dunneNetworkStructureFood2006) and a network realisation. (In a way the idea of predicting a metaweb vs realisation is what makes me hesitant to use the Mangal networks to test the structural models because do we even know what the Mangal networks represent and what the structural models are predicting...) Maybe also @poisotSpeciesWhyEcological2015 that discuss how the local factors are going to play a role.

Also need to take into consideration inherent constraints that the model imposes on itself and how it will affect our ability to test hypotheses/ask questions using the *e.g.,* from @petcheyFitEfficiencyBiology2011 - models that are constrained by connectance means that we are unable to explain connectance itself and you would need a different approach if understanding connectance is your goal. Another way of phrasing this is thinking about what is needed (input data/parameters), produced (final network characteristics), and desired (end-use).

> An interesting thing to also think about is data dependant and data independent 'parametrisation' of the models...
:::

## Concluding remarks

-   As discussion about the different model families and in what areas they do/do not do well. This will depend probably a fair bit on how @fig-topology end up looking... But it will also be important to tie in some of the other considerations/constraints that are listed in what is currently Box 2

    -   In certain situations structure is 'enough' but there may be use cases where we are really interested in the node-level interactions *i.e.,* species identity is a thing we care about and need to be able to retrieve specific interactions at specific nodes correctly.

-   Why do interaction models do so badly at predicting structure? Nuance of metaweb vs realisation but also time? At the core of it interaction models are trained on existing interaction data; this is data that are most likely closer to a metaweb than a local realisation even if they are being inventoried at a small scale...

    -   We can briefly shoehorn downsampling here maybe??

-   It will be interesting to bring up the idea that if a model is missing a specific pairwise link but doing well overall then when does it matter?

    -   The fact that *some* people are concerned about the taxonomic resolution and cascading effects those might have on our understanding of network structure [@pringleResolvingFoodWebStructure2020; @pringleUntanglingFoodWebs2020], but that puts us in a place where we are at risk of losing our ability to distinguish the wood from the tree - are we not (at least at times) concerned more with understanding ecosystem level processes than with needing to understand things *perfectly* at the species level.

    -   I don't think these 'rare'/nuanced links (e.g. carnivorous hippos) are going to rock the boat when we think about networks at the structural level.

> "The resolution of food-web data is demonic because it can radically change network topology and associated biological inferences in ways that are unknowable in the absence of better data." - @pringleResolvingFoodWebStructure2020 The counter to this is that structural models are often not working at the species level and thus the structure remains 'unchanged' when you increase the resolution - I don't think that people are that concerned with the structure of real world networks barring connectance and since that scales with species richness anyway your final proportion will probably still remain the same...

-   I think a big take home will (hopefully) be how different approaches do better in different situations and so you as an end user need to take this into consideration and pick accordingly. I think @petcheyFitEfficiencyBiology2011 might have (and share) some thoughts on this. I feel like I need to look at @berlowGoldilocksFactorFood2008 but maybe not exactly in this context but vaguely adjacent.

    -   I think this is sort of the crux of the argument presented in @brimacombeApplyingMethodIts2024 as well.

> *"we highlight an interesting paradox: the models with the best performance measures are not necessarily the models with the closest reconstructed network structure."* - @poisotGuidelinesPredictionSpecies2023

-   Do we need network models to predict interactions and interaction models to predict structure?

    -   "Another argument for the joint prediction of networks and interactions is to reduce circularity and biases in the predictions. As an example, models like linear filtering generate probabilities of non-observed interactions existing, but do so based on measured network properties." - @strydomRoadmapPredictingSpecies2021

    -   Aligning (dove-tailing) with this the idea of ensemble modelling as presented by @beckerOptimisingPredictiveModels2022

-   Close out with a call to action that we have models that predict networks very well and models that predict interactions very well but nothing that is doing well at predicting both - this is where we should be focusing our attention when it comes to furthering model development...

-   Do we expect there to be differences when thinking about unipartite vs bipartite networks? Is there underlying ecology/theory that would assume that different mechanisms (and thus models) are relevant in these two 'systems'.

    -   The @terryFindingMissingLinks2020 paper looks at some methods but is specifically looking at a bipartite world...

### Downsampling

do we bring this up? this could be a box... if we have the 'finances' for it... otherwise it should go to the outstanding questions fur sure

-   @dansereauSpatiallyExplicitPredictions2023

-   "That being said, there is a compelling argument for the need to 'combine' these smaller functional units with larger spatial networks [@fortinNetworkEcologyDynamic2021] and that we should also start thinking about the interplay of time and space [@estayEditorialPatternsProcesses2023]. Although deciding exactly what measure might actually be driving differences between local networks and the regional metaweb might not be that simple [@saraviaEcologicalNetworkAssembly2022]."

### Time

We lack a clear agenda (and conceptualisation) as to what the appropriate level of aggregation is for a 'network'. Realistically most empirical networks are more aligned with 'feasibility networks' as opposed to 'realised networks' as they are often the result of some sort of aggregation of observations across time. This 'problem' is two-fold. Firstly we need to think about how this affects any sort of development of theory that sits closer to the 'realised network' side of the spectrum - how often are we trying to ask and answer questions about realised networks using feasible networks? The second is that this lack of 'direction' as to how we should define a network is (actually) probably one of the biggest barriers that is affecting the use of networks in applied settings...

## Glossary {.unnumbered}

| Term                   | Definition                                                                                                                                                                                                                                                                                                      |
|------------------------------------|------------------------------------|
| food web               | a representation of feeding links between species                                                                                                                                                                                                                                                               |
| topology generator     | a model that predicts a network based on assumptions of structure, this network is species agnostic in the sense that it does not necessarily contain information at the node level                                                                                                                             |
| interaction predictor  | a model that predicts species interactions, these interactions can be used to construct a network but there are no *a priori* assumptions as that will constrain the network structure                                                                                                                          |
| model                  | A tool that can be used to construct food webs, where the resulting network is a representation of a real world network. Models typically only capture specific elements of real world networks and are intended to be used in specific settings                                                                |
| model family           | A family of models that share an underlying philosophy when it comes to the mapping, pragmatism, and reduction of a network. Families have the same underlying philosophies and assumptions that determine the links between nodes as well as how these may be encoded                                          |
| metaweb                | A network that represents *all* the potential links between species. Importantly these links will not necessarily all be realised in a specific location for a specific time                                                                                                                                    |
| realised network       | A network that represents the links between species that are occurring. These networks represent a very localised network...                                                                                                                                                                                    |
| potential feeding link | links that indicate that an interaction is ecologically feasible but not realised *per se* (a metaweb would contain potential feeding links)                                                                                                                                                                    |
| realised feeding link  | links that indicate that the interaction is realised 'in the field'. (a realised network contains realised feeding links)                                                                                                                                                                                       |
| confusion matrix       | captures the number of true positives (interaction predicted as present when it is present), false negatives (interaction predicted as absent when it is present), false positives (interaction predicted as present when it is absent), and true negatives (interaction predicted as absent when it is absent) |

## Outstanding questions {.unnumbered}

-   non-consumptive effects

-   can we develop a model that is both a topology generator as well as an interaction predictor?

-   how do we define the spatial and temporal 'boundaries' of a network

## References {.unnumbered}

::: {#refs}
:::