% Options for packages loaded elsewhere
\PassOptionsToPackage{unicode}{hyperref}
\PassOptionsToPackage{hyphens}{url}
\PassOptionsToPackage{dvipsnames,svgnames,x11names}{xcolor}
%
\documentclass[
]{article}

\usepackage{amsmath,amssymb}
\usepackage{iftex}
\ifPDFTeX
  \usepackage[T1]{fontenc}
  \usepackage[utf8]{inputenc}
  \usepackage{textcomp} % provide euro and other symbols
\else % if luatex or xetex
  \usepackage{unicode-math}
  \defaultfontfeatures{Scale=MatchLowercase}
  \defaultfontfeatures[\rmfamily]{Ligatures=TeX,Scale=1}
\fi
\usepackage{lmodern}
\ifPDFTeX\else  
    % xetex/luatex font selection
\fi
% Use upquote if available, for straight quotes in verbatim environments
\IfFileExists{upquote.sty}{\usepackage{upquote}}{}
\IfFileExists{microtype.sty}{% use microtype if available
  \usepackage[]{microtype}
  \UseMicrotypeSet[protrusion]{basicmath} % disable protrusion for tt fonts
}{}
\makeatletter
\@ifundefined{KOMAClassName}{% if non-KOMA class
  \IfFileExists{parskip.sty}{%
    \usepackage{parskip}
  }{% else
    \setlength{\parindent}{0pt}
    \setlength{\parskip}{6pt plus 2pt minus 1pt}}
}{% if KOMA class
  \KOMAoptions{parskip=half}}
\makeatother
\usepackage{xcolor}
\setlength{\emergencystretch}{3em} % prevent overfull lines
\setcounter{secnumdepth}{5}
% Make \paragraph and \subparagraph free-standing
\makeatletter
\ifx\paragraph\undefined\else
  \let\oldparagraph\paragraph
  \renewcommand{\paragraph}{
    \@ifstar
      \xxxParagraphStar
      \xxxParagraphNoStar
  }
  \newcommand{\xxxParagraphStar}[1]{\oldparagraph*{#1}\mbox{}}
  \newcommand{\xxxParagraphNoStar}[1]{\oldparagraph{#1}\mbox{}}
\fi
\ifx\subparagraph\undefined\else
  \let\oldsubparagraph\subparagraph
  \renewcommand{\subparagraph}{
    \@ifstar
      \xxxSubParagraphStar
      \xxxSubParagraphNoStar
  }
  \newcommand{\xxxSubParagraphStar}[1]{\oldsubparagraph*{#1}\mbox{}}
  \newcommand{\xxxSubParagraphNoStar}[1]{\oldsubparagraph{#1}\mbox{}}
\fi
\makeatother


\providecommand{\tightlist}{%
  \setlength{\itemsep}{0pt}\setlength{\parskip}{0pt}}\usepackage{longtable,booktabs,array}
\usepackage{calc} % for calculating minipage widths
% Correct order of tables after \paragraph or \subparagraph
\usepackage{etoolbox}
\makeatletter
\patchcmd\longtable{\par}{\if@noskipsec\mbox{}\fi\par}{}{}
\makeatother
% Allow footnotes in longtable head/foot
\IfFileExists{footnotehyper.sty}{\usepackage{footnotehyper}}{\usepackage{footnote}}
\makesavenoteenv{longtable}
\usepackage{graphicx}
\makeatletter
\newsavebox\pandoc@box
\newcommand*\pandocbounded[1]{% scales image to fit in text height/width
  \sbox\pandoc@box{#1}%
  \Gscale@div\@tempa{\textheight}{\dimexpr\ht\pandoc@box+\dp\pandoc@box\relax}%
  \Gscale@div\@tempb{\linewidth}{\wd\pandoc@box}%
  \ifdim\@tempb\p@<\@tempa\p@\let\@tempa\@tempb\fi% select the smaller of both
  \ifdim\@tempa\p@<\p@\scalebox{\@tempa}{\usebox\pandoc@box}%
  \else\usebox{\pandoc@box}%
  \fi%
}
% Set default figure placement to htbp
\def\fps@figure{htbp}
\makeatother
% definitions for citeproc citations
\NewDocumentCommand\citeproctext{}{}
\NewDocumentCommand\citeproc{mm}{%
  \begingroup\def\citeproctext{#2}\cite{#1}\endgroup}
\makeatletter
 % allow citations to break across lines
 \let\@cite@ofmt\@firstofone
 % avoid brackets around text for \cite:
 \def\@biblabel#1{}
 \def\@cite#1#2{{#1\if@tempswa , #2\fi}}
\makeatother
\newlength{\cslhangindent}
\setlength{\cslhangindent}{1.5em}
\newlength{\csllabelwidth}
\setlength{\csllabelwidth}{3em}
\newenvironment{CSLReferences}[2] % #1 hanging-indent, #2 entry-spacing
 {\begin{list}{}{%
  \setlength{\itemindent}{0pt}
  \setlength{\leftmargin}{0pt}
  \setlength{\parsep}{0pt}
  % turn on hanging indent if param 1 is 1
  \ifodd #1
   \setlength{\leftmargin}{\cslhangindent}
   \setlength{\itemindent}{-1\cslhangindent}
  \fi
  % set entry spacing
  \setlength{\itemsep}{#2\baselineskip}}}
 {\end{list}}
\usepackage{calc}
\newcommand{\CSLBlock}[1]{\hfill\break\parbox[t]{\linewidth}{\strut\ignorespaces#1\strut}}
\newcommand{\CSLLeftMargin}[1]{\parbox[t]{\csllabelwidth}{\strut#1\strut}}
\newcommand{\CSLRightInline}[1]{\parbox[t]{\linewidth - \csllabelwidth}{\strut#1\strut}}
\newcommand{\CSLIndent}[1]{\hspace{\cslhangindent}#1}

\makeatletter
\@ifpackageloaded{caption}{}{\usepackage{caption}}
\AtBeginDocument{%
\ifdefined\contentsname
  \renewcommand*\contentsname{Table of contents}
\else
  \newcommand\contentsname{Table of contents}
\fi
\ifdefined\listfigurename
  \renewcommand*\listfigurename{List of Figures}
\else
  \newcommand\listfigurename{List of Figures}
\fi
\ifdefined\listtablename
  \renewcommand*\listtablename{List of Tables}
\else
  \newcommand\listtablename{List of Tables}
\fi
\ifdefined\figurename
  \renewcommand*\figurename{Figure}
\else
  \newcommand\figurename{Figure}
\fi
\ifdefined\tablename
  \renewcommand*\tablename{Table}
\else
  \newcommand\tablename{Table}
\fi
}
\@ifpackageloaded{float}{}{\usepackage{float}}
\floatstyle{ruled}
\@ifundefined{c@chapter}{\newfloat{codelisting}{h}{lop}}{\newfloat{codelisting}{h}{lop}[chapter]}
\floatname{codelisting}{Listing}
\newcommand*\listoflistings{\listof{codelisting}{List of Listings}}
\makeatother
\makeatletter
\makeatother
\makeatletter
\@ifpackageloaded{caption}{}{\usepackage{caption}}
\@ifpackageloaded{subcaption}{}{\usepackage{subcaption}}
\makeatother

\usepackage{bookmark}

\IfFileExists{xurl.sty}{\usepackage{xurl}}{} % add URL line breaks if available
\urlstyle{same} % disable monospaced font for URLs
\hypersetup{
  pdftitle={Unveiling the Complexity of Food Webs: A Comprehensive Overview of Definitions, Scales, and Mechanisms},
  pdfauthor={Tanya Strydom; Jennifer A. Dunne; Timothée Poisot; Andrew P. Beckerman},
  pdfkeywords={food web, network construction, scientific ignorance},
  colorlinks=true,
  linkcolor={blue},
  filecolor={Maroon},
  citecolor={Blue},
  urlcolor={Blue},
  pdfcreator={LaTeX via pandoc}}



\title{Unveiling the Complexity of Food Webs: A Comprehensive Overview
of Definitions, Scales, and Mechanisms}
\author{Tanya Strydom %
%
\textsuperscript{%
%
1%
}%
; Jennifer A. Dunne %
%
\textsuperscript{%
%
2%
}%
; Timothée Poisot %
%
\textsuperscript{%
3,%
4%
}%
; Andrew P. Beckerman %
%
\textsuperscript{%
%
1%
}%
}
\date{2024-12-06}

\usepackage{setspace}
\usepackage[left]{lineno}
\usepackage[letterpaper]{geometry}

\usepackage[nolists,noheads,markers]{endfloat}
\geometry{margin=2.5cm}

\begin{document}

\thispagestyle{empty}
{\bfseries\sffamily\Large Unveiling the Complexity of Food Webs: A
Comprehensive Overview of Definitions, Scales, and Mechanisms}
\vfil
Tanya Strydom %
%
\textsuperscript{%
%
1%
}%
; Jennifer A. Dunne %
%
\textsuperscript{%
%
2%
}%
; Timothée Poisot %
%
\textsuperscript{%
3,%
4%
}%
; Andrew P. Beckerman %
%
\textsuperscript{%
%
1%
}%

\vfil
{\small
\textbf{Abstract:} Food webs are a useful abstraction and representation
of the feeding links between species in a community and are used to
infer many ecosystem level processes. However, the different theories,
mechanisms, and criteria that underpin how a food web is defined, and
ultimately, constructed means that not all food webs are representing
the same ecological process at the same scale. Here we present a
synthesis of the different assumptions, scales, and mechanisms that are
used to define the different ecological networks , leading to a revision
of definitions for different types of networks. Additionally we
explicitly link the different network representations to the broader
methodological approaches (models) that are used to construct them. In
explicitly outlining the assumptions, scales, and mechanisms of network
inference allows for a formal categorisation of how to use networks to
answer key ecological and conservation questions as wel as defining
clear guidelines to prevent unintentional misuse or misinterpretation.
\vfil
\textbf{Keywords:} %
food web, network construction, %
scientific ignorance%
}
\clearpage
\setcounter{page}{1}
\doublespacing
\linenumbers


At the heart of modern biodiversity science are a set of concepts and
theories about species richness, stability, and function (Loreau \& de
Mazancourt, 2013). These relate to the abundance, distribution, and
services that biodiversity provides, and how biodiversity (as an
interconnected set of species) responds to multiple stressors.
Documenting interactions between and among species is thus one of the
fundamental building blocks of community ecology, providing a powerful
abstraction and platform for mathematical and statistical modelling of
biodiversity to make predictions, mitigate threats, and manage services
(Windsor et al., 2023).Such network representations of biodiversity are
increasingly argued to be an asset to understanding and predicting the
abundance, distribution, dynamics, and services provided by multiple
species facing multiple stressors (Simmons et al., 2021). However, there
is a growing discourse around limitations to the interpretation and
applied use of networks (Blüthgen, 2010; Dormann, 2023), primarily as
the result of shortcomings regarding their conceptualisation (Blüthgen
\& Staab, 2024).

We propose that every network embeds assumptions about the process(es)
that determine interactions, and about the levels of organization at
which this occurs (\emph{i.e.} the biological, ecological,
spatial/temporal scale). The differences in these assumptions ultimately
influence the nature and scope of inference that can be made from a
given network (Proulx et al., 2005). Fundamentally, we are talking about
an intersection of the data used to construct the network and the
underlying theory as to what drives the occurrence of interactions
between species. Although there have been extensive discussions about
the challenges relating to data collection and observation (\emph{e.g.,}
Blüthgen \& Staab, 2024; Brimacombe et al., 2023, 2024; Moulatlet et
al., 2024; Polis, 1991; Pringle \& Hutchinson, 2020; Saberski et al.,
2024) we still lack a clear framework framed by the different
assumptions and scale dependent processes.

In this perspective we aim to provide an overview of the different
\textbf{food web} representations, particularly how these relate to the
terminology used to define a food web, and how this is influenced by
both the processes that determine interactions
Section~\ref{sec-process}, as well as how this relates to the way in
which we construct the resulting networks Section~\ref{sec-construct}.
This allows us to deliver an overview of fundamental questions in
ecology that we think can benefit from network thinking and a proposal
that such thinking can accelerate our capacity to predict the impact of
multiple stressors on biodiverse communities. Specifically, we finish
this perspective with an overview of fundamental questions in ecology
that we think can benefit from network thinking and a proposal that such
thinking can accelerate our capacity to predict the impact of change on
biodiverse communities.

\section{Setting the Scene: The Not So Basics of Nodes and
Edges}\label{sec-anatomy}

Networks often have multiple uses: an `object' from which inferences are
made (\emph{e.g.,} topological inference about biodiversity,
interactions among species, and community structure, {[}REF{]}); a
platform for evaluating `downstream' responses to stressors {[}REF{]};
and a platform for evaluating mathematical and statistical models of
`generative processes' {[}REF{]}. Against this backdrop of multiple
research agendas, it should come as no surprise that the definition of
`edges' and `nodes', as well as the levels of organisation at which they
are collated takes many forms (Moulatlet et al., 2024; Poisot, Stouffer,
et al., 2016), while also encoding a series of assumptions within a
network.

\subsection{How do we define a node?}\label{how-do-we-define-a-node}

Although this may seem elementary that a node should represent a
(taxonomic) species, the reality is that nodes often represents
non-taxonomic units such as a trophic species (\emph{e.g.,} Yodzis
(1982); Williams \& Martinez (2000)), a feeding guild (\emph{e.g.,}
García-Callejas et al., 2023), or a segregation of species by life
stages (\emph{e.g.,} Clegg et al., 2018). Such granularity and variation
can limit the ability to make (taxonomic) species specific inferences
(\emph{e.g.,} does species \(a\) eat species \(b\)?), affect inference
made from networks, including estimates of complexity and structure
(Beckerman et al., 2006; Clegg et al., 2018) and make it challenging to
use networks in `downstream analyses', for example, of extinction or
invasions. Despite these implications, there may also be value in having
nodes that represent an aggregation of species, as the distribution of
the links between them may be more meaningful in terms of understanding
energy flow and distribution within the system.

\subsection{What is captured by an
edge?}\label{what-is-captured-by-an-edge}

Links within food webs can be thought of as a representation of either
feeding links between species (be that realised or potential (Dunne,
2006; Pringle, 2020), or fluxes within a system \emph{e.g.,} energy
transfer or material flow as the result of the feeding links between
species (Lindeman, 1942). These correspond to different `currencies'
(the feasibility of links or the energy that is moving between nodes).
There are also a myriad of ways in which the links themselves can be
specified. Links between species can be treated present or absent (i.e.,
binary), may be defined as probabilities (Banville et al., 2024; Poisot,
Cirtwill, et al., 2016) or by continuous functions which further
quantify the strength of an interaction (Berlow et al., 2004). How links
are specified will influence the structure of the network. For example,
taking a food web that consists of links representing all
\emph{potential} feeding links in a collection of species will be
meaningless if one is interested in understanding the flow of energy
through the network as the links are not environmentally/energetically
constrained.

\subsection{Network representations}\label{sec-representation}

Against these definitions, networks fall into two major `types':
metawebs, traditionally defined as all the \emph{potential} interactions
for a specific species pool (Dunne, 2006); and realised networks, which
is the subset of interactions in a metaweb that are \emph{realised} for
a specific community at a given time and place. The fundamental
differences between these two network representations are the spatial
scale at which they are constructed and the associated processes that
are assumed to drive pattern at these scales.

A metaweb is at its core a list of feasible interactions between pairs
of species. The feasibility for a given pair is derived from the
complementarity (phylogenetic relationships) of their traits
(representing a \emph{global metaweb}), which can be further refined by
co-occurrence (representing a \emph{regional metaweb}). By this
definition, metawebs provide a means to identify links that are not
ecologically plausible, \emph{i.e.,} forbidden links (Jordano, 2016b),
or provide an idea of the \emph{complete} diet of a species (Strydom et
al., 2023).

In contrast realised networks are relatively localised in space and
time, and the links between species are contingent on both the
co-occurrence of species, the role of the environment, and mechanisms of
diet choice. Fundamentally this means that the presence/absence of a
link is the result of the `behaviour' of the species.

This distinction between metawebs and realised webs lead to some further
definitions. Links that are absent in a metaweb can conceptually
(although not always practically) be treated as being truly absent.
However, links that are absent in a realised network cannot be
considered as truly absent but rather as absent due to the broader
environmental/community context. Furthermore, a realised network is not
simply the downscaling of a metaweb to a smaller scale (\emph{e.g.,}
moving from the country to the 1x1 km\textsuperscript{2} scale based on
fine-scale species co-occurrence). Instead, realised webs capture
processes that determine the realisation of an interaction.
Specifically, in realised webs, the definition of an edges shifts from
being determined by feasibility to that of choices and consequences that
centre around energy. Meaning if one were to take the same community of
species and constructed both a metaweb and realised network the two
networks might have the same species but would be structurally
different, owing to the differences in the `rules' constraining the
presence of links (Caron et al., 2024).

\section{From Nodes and Edges to Process and
Constraints}\label{sec-process}

In the previous section we discussed how the definition of nodes and
edges at representing different biological and ecological and processes
associated with them lead to the concept of a metweb and a realised web.
Here we expand this discussion, introducing five core constraints across
these scales that further expose processes that determine the links
among species: evolutionary compatibility, co-occurrence, abundance,
diet choice, and non-trophic interactions Figure~\ref{fig-process}.

\begin{figure}

\centering{

\pandocbounded{\includegraphics[keepaspectratio]{images/anatomy.png}}

}

\caption{\label{fig-process}Aligning the various processes that
determine interactions (right column) with the different network
representations (left column). First, we start with a \textbf{global
metaweb} this network captures all possible interactions for a
collection of species in the global context. However, within the global
environment different species occur in different regions (region one =
yellow and region 2 = orange), and it is possible to construct two
different metawebs (\textbf{regional metawebs}) for each region by
taking accounting for the co-occurrence patterns of the difference
species - as shown here we have two regions with some species (blue)
that are found in both regions and others endemic to either region one
(yellow) or region two (orange). However even within a region we do not
expect that all interactions to be realised but rather that there are
multiple configurations of the regional metaweb over both space and
time. The `state' of the different \textbf{realised networks} is
ultimately influenced not just by the co-occurrence of a species pair
but rather the larger community context such as the abundance of
different species, maximisation of energy gain, or indirect/higher order
interactions.}

\end{figure}%

\subsection{Processes that determine the feasibility of an
interaction}\label{sec-process-feasibility}

Here we introduce evolutionary compatibility and co-occurrence as
processes that `act' at the species pair of interest, that is the
possibility of an interaction being present/absent is assessed at the
pairwise level.

Here we introduce evolutionary compatibility and co-occurrence as
processes that `act' at the species pair of interest. The scale of
inference and set of processes embodied in these two constraints combine
to define a `list' of interactions that are viable/feasible and defined
as present/absent. It is however possible to build a network from this
information. However, it is important to be aware that the structure of
this network is not constrained by any community context and so just
because species are able to interact does not mean that they will
(Poisot et al., 2015).

\textbf{Evolutionary compatibility}

This constraint is defined by shared (co)evolutionary history between
consumers and resources (Dalla Riva \& Stouffer, 2016; Gómez et al.,
2010; Segar et al., 2020) which, in the more proximal sense, is
manifested as `trait complementarity' between two species (Benadi et
al., 2022). In this body of theory, one species (the consumer) has the
`correct' (multivariate) set of traits that allow it to chase, capture,
kill, and consume the other species (the resource) and interactions that
are not compatible are defined as forbidden links (Jordano, 2016b);
\emph{i.e.,} not physically possible and will \emph{always} be absent
within a network.

Networks arising from this constraint can be binary (possible vs
forbidden) or probabilistic (Banville et al., 2024), \emph{e.g.,} the
metaweb constructed by Strydom et al. (2022) uses probabilities to
quantify their confidence with regards to the possibility of a specific
interaction existing between two species. A network constructed on the
basis of evolutionary compatibility is conceptually aligned with a
`global metaweb', and gives us information as to the feasibility of
links between species despite the fact that they do not co-occur (as
shown in Figure~\ref{fig-process}).

\textbf{(Co)occurrence}

The co-occurrence of species in both time and space is a fundamental
requirement for an interaction between two species to occur (at least in
terms of feeding links). Although co-occurrence data alone is
insufficient for building an accurate and ecologically meaningful
representation of \emph{feeding links} (Blanchet et al., 2020), it is
still a critical process that determines the realisation of a feeding
link and allows us to spatially constrain a global metaweb based on
local communities (Dansereau, Barros, et al., 2024), in the context of
Figure~\ref{fig-process} this would be the metawebs for regions one and
two.

\subsection{Processes that modify the behaviour (preference) of
species}\label{sec-process-realisation}

Here we will showcase three processes that will ultimately influence the
realisation of an interaction between species and form the conceptual
basis for realised networks. As we show in Figure~\ref{fig-process} a
`truly realised' network is the product of different facets of both the
properties of the community (\textbf{abundance} and \textbf{non-trophic
interactions}) as well as the individual (\textbf{profitability}). This
represents a contextual shift where the presence (realisation) of an
interaction is no longer constrained to evaluating the viability between
a \emph{pair} of species but rather takes into consideration information
about the community and the individual (Quintero et al., 2024), and as
discussed in Section~\ref{sec-representation}, links are now
\emph{constrained} by consumer choice.

\textbf{Abundance}

The most basic abundance constraint linked to foraging biology is the
principle that organisms feeding randomly will consume resources in
proportion to their abundance (Stephens \& Krebs, 1986), and
interactions are not necessarily contingent on there being any
compatibility between them (E. Canard et al., 2012; Momal et al., 2020;
Pomeranz et al., 2019). However, a more ecologically sound assumption
would be that the abundance of different prey species will influence the
distribution of links in a network (Vázquez et al., 2009), by
influencing which prey are targeted or preferred by the predator, as
abundance influences factors such as the likelihood of two species
(individuals) meeting (Banville et al., 2024; Poisot et al., 2015).
Thus, if abundance data are combined with a derived metaweb, there is a
basic ruleset that can define the distribution (\emph{e.g.,} structure)
and potentially the strength of links.

\textbf{Profitability}

It is well established that consumers make more active decisions than
eating items in proportion to their abundance (Stephens \& Krebs, 1986).
Ultimately, consumer choice is underpinned by an energetic cost-benefit
framework centred around profitability and defined by traits associated
with finding, catching, killing, and consuming a resource (Wootton et
al., 2023). Although energetic constraints can be invoked in a myriad of
ways (Cherif et al., 2024; \emph{e.g.,} Pawar et al., 2012; Portalier et
al., 2019) we select profitability as a term to capture rules linked to
optimal foraging (Pyke, 1984) and metabolic theory (Brown et al., 2004);
it is a sensible `umbrella concept' for capturing the energetic
constraint on of the distribution and strength of interactions.

\textbf{Non-trophic interactions}

Perhaps not as intuitive when thinking about the previous constraints,
non-trophic interactions (Ings et al., 2009) specifically modify either
the realisation or strength of trophic interactions (Golubski \& Abrams,
2011; Kamaru et al., 2024; Pilosof et al., 2017; Staniczenko et al.,
2010). Non-trophic interactions can modify interactions either
`directly' e.g., predator a outcompetes predator b; or `indirectly'
e.g., mutualistic/facilitative interactions. Altogether they can alter
the fine-scale distribution and abundance of species as well as their
persistence (Buche et al., 2024; Kéfi et al., 2012, 2015).

\section{Network construction: a case for models}\label{sec-construct}

\subsection{Why construct networks?}\label{why-construct-networks}

Broadly the desire to construct a network has arisen for two different
purposes; building networks that can be used in real-world, applied
contexts (have actionable consequences?), and building networks that
allow us to interrogate, generate, and reflect upon different ecological
theories. The act of constructing a `real world' network through the
empirical collection of interaction data is both costly and challenging
to execute (Jordano, 2016a, 2016b), which has led to the development of
a suite of modelling approaches that allow us to predict the interaction
between two species, or network structure (see Strydom, Catchen, et al.,
2021 for a broader discussion), or identify missing interactions (gap
fill) within existing empirical datasets (\emph{e.g.,} Biton et al.,
2024; Dallas et al., 2017; Stock, 2021). However, working with
`real-world networks' is data-hungry and cumbersome and has driven the
development of models that construct ecologically plausible but often
times species agnostic networks. These models often explicitly model one
or a few of of the processes discussed in Section~\ref{sec-process} and
in doing so allow us to better understand the different constraints
determining interactions (Song \& Levine, 2024; Stouffer, 2019).

\subsection{Stochastic networks}\label{stochastic-networks}

Within stochastic networks the assumption is that the interactions
between species occurs irrespective of the identity of the species
(\emph{i.e.,} species have no agency) and links are \emph{randomly}
distributed throughout the network. Alternatively it can be assumed that
interactions are still independent of a species' trait but are rater
driven by its abundance, \emph{sensu} neutral theory (Hubbell, 2001).

Practically there is little support that networks are truly stochastic,
however null models (Bascompte et al., 2003; \emph{e.g.,} Fortuna \&
Bascompte, 2006) are often used as a `null hypothesis' that can be used
to ask questions about deviations in observed network structure
(\emph{e.g.,} Banville et al., 2023; Strydom, Dalla Riva, et al., 2021).
Conversely, neutral models (E. F. Canard et al., 2014; Krishna et al.,
2008), and by extension processes, are most likely relevant at local
scales (Pomeranz et al., 2019), \emph{i.e.,} should be considered to
play a role when constructing realised networks but are superseded by
trait-based processes at a metaweb scale.

\subsection{Feasibility networks
(metawebs)}\label{feasibility-networks-metawebs}

Metawebs (depending on the aggregation) can help us develop our
understanding of the intersection of species interactions and their
co-occurrence(Gravel et al., 2019; Soberón, 2007). Whereby a
\emph{global metaweb} presents an approximation of the fundamental
Eltonian niche of a species (\emph{i.e.,} its relation to its food
source), whereas as \emph{regional metawebs} represent an intersection
of Elton and Grinnell. As discussed in
Section~\ref{sec-process-feasibility} the feasibility of an interaction
is typically assessed on a pairwise basis, and is often assessed based
on the idea that interactions are governed by a set of `feeding rules'
(Morales-Castilla et al., 2015), and are broadly elucidated in two
different ways; \emph{mechanistic models}, (Dunne et al., 2008;
Roopnarine, 2017; \emph{e.g.,} Shaw et al., 2024) and \emph{pattern
finding models} (Caron et al., 2022; Cirtwill et al., 2019;
Desjardins-Proulx et al., 2017; Eklöf et al., 2013; Llewelyn et al.,
2023; Pichler et al., 2020; Strydom et al., 2022; \emph{e.g.,} Strydom
et al., 2023). The fundamental difference between these two model groups
is that \emph{mechanistic models} rely on expert knowledge and make
explicit assumptions on trait-feeding relationships, whereas the
\emph{pattern finding models} are dependent on existing interaction
datasets from feeding rules can be elucidated. It perhaps also bears
repeating that these models are often only presenting a list of feasible
interactions and that the resulting network is `unstructured', as it is
uconstrained by any processes or conditions that generate structure.
While these networks can be imprinted with external definitions of
trophic position and guild identity to deliver hypothetical structure,
this structure is not an emergent property of the links and species
pairs (Caron et al., 2024).

Feasibility networks are useful for determining all feasible
interactions for a specific community, and the models that have been
developed in this context have the potential to allow us to construct
first draft networks for communities for which we have no interaction
data (Strydom et al., 2022), and are valuable not only in data poor
regions but also for predicting interactions for `unobservable'
communities \emph{e.g.,} prehistoric networks (Dunhill et al., 2024;
Fricke et al., 2022; Yeakel et al., 2014) or future, novel community
assemblages. Conceptually this is particularly valuable if we want to
understand interactions between novel communitites, as well as the
rewiring capacity of species. Additionally, an understanding of the role
of interactions between species has allowed us to better determine the
distribution of a species by accounting not only for the role of the
environment but also the role of species interactions (Higino et al.,
2023; Pollock et al., 2014).

\subsection{`Behavioural' networks}\label{behavioural-networks}

Ultimately realised networks and capture some aspect of how the behavior
of a species determines if a link is realised or not and can be modelled
in two ways; models that predict realised interactions (whereby he
behaiour of a secies is modelled \emph{i.e.,} its diet choice), and
models that predict the structure of realised networks (whereby the
behvaiour of the system is modelled and assumptions are made with
regards to the structure of a network). In terms predicting interactions
current models are rooted in feeding theory and allocate the links
between species based on energy \emph{e.g.,} diet models (Beckerman et
al., 2006; Petchey et al., 2008) have been used construct networks based
on both profitability (as determined by the handling time, energy
content, and predator attack rate) as well as abundance (prey density).
(Wootton et al., 2023). At a `coarser', funtional level there are models
that are based on the compartmentation and aquisition of energy for
species at different trophic levels (Allesina \& Pascual, 2009; Krause
et al., 2003). Models that determine structure are based on the idea
that networks follow a trophic hierarchy and that network structure can
be determined by distributing interactions along single dimension {[}the
``niche axis''; Allesina et al. (2008){]}, while parametrising an aspect
of the network structure (although see Allesina \& Pascual, 2009 for a
parameter-free model).

As behavioural networks are are build on the concept of dynamic
processes (\emph{e.g.,} the abundance of species will always be in flux)
these networks are valuable for understanding the behaviour of networks
over time, or their response to change (Curtsdotter et al., 2019; Delmas
et al., 2017; Lajaaiti et al., 2024). However, they are `costly' to
construct (requiring data about the entire community, as it is the
behaviour of the system that determines the behaviour of the part) and
also lack the larger diet niche context afforded by metawebs. Structural
models provide a data-light (the models often only require species
richness) but assumption heavy (the resulting network structure is
determined by an assumption of network structure) alternative, however
they do not make species specific predictions and so cannot be used to
determine if an interaction is either possible \emph{or} realised
between two species (\emph{i.e.,} one cannot use these models to
determine if species \(a\) eats species \(b\)). Although this means this
suite of models are unsuitable as tools for predicting species-specific
interactions, they have been shown to be sufficient tools to predict the
structure of networks (Williams \& Martinez, 2008), and are useful in
synthetic simulations.

\section{Making Progress with Networks}\label{sec-progress}

It is probably both this nuance as well as a lack of clear boundaries
and guidelines as to the links between network form and function
(although see Delmas et al., 2019) that has stifled the `productive use'
of networks beyond the inventorying the interactions between species.
Although progress with using networks as a means to address questions
within larger bodies of ecological theory \emph{e.g.,} invasion biology
(Hui \& Richardson, 2019) and co-existence theory (García-Callejas et
al., 2023) has been made we still lack explicit guidelines as to what
the appropriate network representation for the task at hand would be,
and as highlighted in Box 1, underscores the need to evaluate exactly
what process a specific network representation captures as well as its
suitability for the question of interest. Below we present a mapping of
what we believe are some of the key questions for which interaction
networks can be used to the different networks representations that are
most suitable, as well as highlight some of the methodological
challenges that still need to be improved upon.

\subsection{Making use of the different network
representations}\label{making-use-of-the-different-network-representations}

\textbf{Methodological challenges}

\begin{enumerate}
\def\labelenumi{\arabic{enumi}.}
\tightlist
\item
  Tools that allow us to estimate both the feasibility as well as
  realisation of links: Currently most approaches to modelling relaised
  networks fail to explicitly account for any form of evolutionary
  constraint Wootton et al. (2023) and we need to develop either an
  ensemble modelling approach (Becker et al., 2022; Terry \& Lewis,
  2020) or tools that will allow for the downsampling of metawebs into
  realised networks (\emph{e.g.,} Roopnarine, 2006).
\item
  Is there something in generalisable models that `combine' different
  processes/aspects (\emph{e.g.,} using body size as a catch all) versus
  limited models that allow you to unpack things bit-by-bit
  (\emph{i.e.,} process by process). So Wootton et al. (2023) \emph{may}
  (TBD) span the gamut but it lacks the ability to unpack\ldots{}
  Although myabe the terms do?
\item
  Modelling interaction strength: Although realised networks are more
  closely aligned with \emph{explicitly} capturing interaction strength
  we lack models that allow us to quantify this (Strydom, Catchen, et
  al., 2021; Wells \& O'Hara, 2013).
\item
  How do we validate our predictions?: Progress has been made to assess
  how well a model recovers pairwise interactions (Poisot, 2023;
  Strydom, Catchen, et al., 2021), but we still lack clear set of
  guidelines for benchmarking the ability of models to recover structure
  (Allesina et al., 2008)
\item
  Something about making what we do with networks more tractablie in the
  applied space? \emph{e.g.,} Dansereau, Braga, et al. (2024)
\end{enumerate}

\textbf{Theory challenges}

\begin{enumerate}
\def\labelenumi{\arabic{enumi}.}
\tightlist
\item
  Core Theory Advancement: Do the decades of insights arrived at for
  stability-diversity-productivity relationships with tri-trophic or
  diamond shaped models hold for complex communities (10's-100s) (Danet
  et al., 2024); How will spatial and temporal variation in climate and
  productivity drive change in complex ecosystems. Necessary to move to
  predicting changes in biodiversity per se, ecosystem functions and
  identifying sensitive and robust species and portions of communities.
\item
  How will novel communities interact? How will range shifts and
  invasions result in new/novel community assemblages. And then also the
  intentional changes of species compositions through rewilding.
\item
  Does rewiring happen and does it deliver robustness? Specific sub
  points to consider here is persistence, especially persistence to
  perturbation. Again, dynamic networks and network/community assembly
  and finally extinctions (Dunhill et al., 2024).
\item
  When do invasive species enhance or decimate communities? When do
  reintroductions work? (Wooster et al., 2024)
\item
  Are there temperature threshold to community collapse
\item
  Can socioeconomic networks combined with biological networks drive
  understanding of externalities?
\item
  Can paleoecological data from deep time hyperthermal events provide
  sufficient insight into the targets, pace and recovery times from
  rapid climate events?
\end{enumerate}

\begin{figure}

\centering{

\pandocbounded{\includegraphics[keepaspectratio]{images/NetworkFuture.png}}

}

\caption{\label{fig-future}Here we highlight some of the outstanding
questions in both network as well as general ecology, as well as some of
the outstanding methodological challenges with regards to constructing
food webs (shown in orange) that we are faced with.}

\end{figure}%

\section{Concluding remarks}\label{concluding-remarks}

Having a clear understanding of the interplay between network
representations and the processes that they are capable of encoding is
critical if we are to understand exactly which networks can be used to
answer which questions. As we highlight in Box 1 the different network
representations have different potential uses and it should be clear
that there is no `best' network representation but rather a network
representation that is best suited to its intended purpose. In providing
a formalisation regards to the assumptions and mechanisms that need to
be explicitly taken into consideration when deciding to use (and
construct) networks we hope to prevent the unintentional misuse or
misinterpretation of networks as well as provide a starting point from
which we can develop a better framework for the applied use of networks
to answer questions that are not only pressing within the field but also
within broader biodiversity science.

\section*{References}\label{references}
\addcontentsline{toc}{section}{References}

\phantomsection\label{refs}
\begin{CSLReferences}{1}{0}
\bibitem[\citeproctext]{ref-allesinaGeneralModelFood2008}
Allesina, S., Alonso, D., \& Pascual, M. (2008). A {General Model} for
{Food Web Structure}. \emph{Science}, \emph{320}(5876), 658--661.
\url{https://doi.org/10.1126/science.1156269}

\bibitem[\citeproctext]{ref-allesinaFoodWebModels2009}
Allesina, S., \& Pascual, M. (2009). Food web models: A plea for groups.
\emph{Ecology Letters}, \emph{12}(7), 652--662.
\url{https://doi.org/10.1111/j.1461-0248.2009.01321.x}

\bibitem[\citeproctext]{ref-banvilleWhatConstrainsFood2023}
Banville, F., Gravel, D., \& Poisot, T. (2023). What constrains food
webs? {A} maximum entropy framework for predicting their structure with
minimal biases. \emph{PLOS Computational Biology}, \emph{19}(9),
e1011458. \url{https://doi.org/10.1371/journal.pcbi.1011458}

\bibitem[\citeproctext]{ref-banvilleDecipheringProbabilisticSpecies2024}
Banville, F., Strydom, T., Blyth, P., Brimacombe, C., Catchen, M. D.,
Dansereau, G., Higino, G., Malpas, T., Mayall, H., Norman, K., Gravel,
D., \& Poisot, T. (2024). \emph{Deciphering probabilistic species
interaction networks}. EcoEvoRxiv. \url{https://doi.org/10.32942/X28G8Z}

\bibitem[\citeproctext]{ref-bascompteNestedAssemblyPlantanimal2003}
Bascompte, J., Jordano, P., Melian, C. J., \& Olesen, J. M. (2003). The
nested assembly of plant-animal mutualistic networks. \emph{Proceedings
of the National Academy of Sciences}, \emph{100}(16), 9383--9387.
\url{https://doi.org/10.1073/pnas.1633576100}

\bibitem[\citeproctext]{ref-beckerOptimisingPredictiveModels2022}
Becker, D. J., Albery, G. F., Sjodin, A. R., Poisot, T., Bergner, L. M.,
Chen, B., Cohen, L. E., Dallas, T. A., Eskew, E. A., Fagre, A. C.,
Farrell, M. J., Guth, S., Han, B. A., Simmons, N. B., Stock, M.,
Teeling, E. C., \& Carlson, C. J. (2022). Optimising predictive models
to prioritise viral discovery in zoonotic reservoirs. \emph{The Lancet
Microbe}, \emph{3}(8), e625--e637.
\url{https://doi.org/10.1016/S2666-5247(21)00245-7}

\bibitem[\citeproctext]{ref-beckermanForagingBiologyPredicts2006}
Beckerman, A. P., Petchey, O. L., \& Warren, P. H. (2006). Foraging
biology predicts food web complexity. \emph{Proceedings of the National
Academy of Sciences}, \emph{103}(37), 13745--13749.
\url{https://doi.org/10.1073/pnas.0603039103}

\bibitem[\citeproctext]{ref-benadiQuantitativePredictionInteractions2022}
Benadi, G., Dormann, C. F., Fründ, J., Stephan, R., \& Vázquez, D. P.
(2022). Quantitative {Prediction} of {Interactions} in {Bipartite
Networks Based} on {Traits}, {Abundances}, and {Phylogeny}. \emph{The
American Naturalist}, \emph{199}(6), 841--854.
\url{https://doi.org/10.1086/714420}

\bibitem[\citeproctext]{ref-berlowInteractionStrengthsFood2004}
Berlow, E. L., Neutel, A.-M., Cohen, J. E., de Ruiter, P. C., Ebenman,
B., Emmerson, M., Fox, J. W., Jansen, V. A. A., Iwan Jones, J.,
Kokkoris, G. D., Logofet, D. O., McKane, A. J., Montoya, J. M., \&
Petchey, O. (2004). Interaction strengths in food webs: Issues and
opportunities. \emph{Journal of Animal Ecology}, \emph{73}(3), 585--598.
\url{https://doi.org/10.1111/j.0021-8790.2004.00833.x}

\bibitem[\citeproctext]{ref-bitonInductiveLinkPrediction2024}
Biton, B., Puzis, R., \& Pilosof, S. (2024). \emph{Inductive link
prediction boosts data availability and enables cross-community link
prediction in ecological networks}.

\bibitem[\citeproctext]{ref-blanchetCooccurrenceNotEvidence2020}
Blanchet, F. G., Cazelles, K., \& Gravel, D. (2020). Co-occurrence is
not evidence of ecological interactions. \emph{Ecology Letters},
\emph{23}(7), 1050--1063. \url{https://doi.org/10.1111/ele.13525}

\bibitem[\citeproctext]{ref-bluthgenWhyNetworkAnalysis2010}
Blüthgen, N. (2010). Why network analysis is often disconnected from
community ecology: {A} critique and an ecologist's guide. \emph{Basic
and Applied Ecology}, \emph{11}(3), 185--195.
\url{https://doi.org/10.1016/j.baae.2010.01.001}

\bibitem[\citeproctext]{ref-bluthgenCriticalEvaluationNetwork2024}
Blüthgen, N., \& Staab, M. (2024). A {Critical Evaluation} of {Network
Approaches} for {Studying Species Interactions}. \emph{Annual Review of
Ecology, Evolution, and Systematics}, \emph{55}(1), 65--88.
\url{https://doi.org/10.1146/annurev-ecolsys-102722-021904}

\bibitem[\citeproctext]{ref-brimacombePublicationdrivenConsistencyFood2024}
Brimacombe, C., Bodner, K., Gravel, D., Leroux, S. J., Poisot, T., \&
Fortin, M.-J. (2024). Publication-driven consistency in food web
structures: {Implications} for comparative ecology. \emph{Ecology},
\emph{n/a}(n/a), e4467. \url{https://doi.org/10.1002/ecy.4467}

\bibitem[\citeproctext]{ref-brimacombeShortcomingsReusingSpecies2023}
Brimacombe, C., Bodner, K., Michalska-Smith, M., Poisot, T., \& Fortin,
M.-J. (2023). Shortcomings of reusing species interaction networks
created by different sets of researchers. \emph{PLOS Biology},
\emph{21}(4), e3002068.
\url{https://doi.org/10.1371/journal.pbio.3002068}

\bibitem[\citeproctext]{ref-brownMetabolicTheoryEcology2004}
Brown, J. H., Gillooly, J. F., Allen, A. P., Savage, V. M., \& West, G.
B. (2004). Toward a {Metabolic Theory} of {Ecology}. \emph{Ecology},
\emph{85}(7), 1771--1789. \url{https://doi.org/10.1890/03-9000}

\bibitem[\citeproctext]{ref-bucheMultitrophicHigherOrderInteractions2024}
Buche, L., Bartomeus, I., \& Godoy, O. (2024). Multitrophic
{Higher-Order Interactions Modulate Species Persistence}. \emph{The
American Naturalist}, \emph{203}(4), 458--472.
\url{https://doi.org/10.1086/729222}

\bibitem[\citeproctext]{ref-canardEmpiricalEvaluationNeutral2014}
Canard, E. F., Mouquet, N., Mouillot, D., Stanko, M., Miklisova, D., \&
Gravel, D. (2014). Empirical {Evaluation} of {Neutral Interactions} in
{Host-Parasite Networks}. \emph{The American Naturalist}, \emph{183}(4),
468--479. \url{https://doi.org/10.1086/675363}

\bibitem[\citeproctext]{ref-canardEmergenceStructuralPatterns2012}
Canard, E., Mouquet, N., Marescot, L., Gaston, K. J., Gravel, D., \&
Mouillot, D. (2012). Emergence of {Structural Patterns} in {Neutral
Trophic Networks}. \emph{PLOS ONE}, \emph{7}(8), e38295.
\url{https://doi.org/10.1371/journal.pone.0038295}

\bibitem[\citeproctext]{ref-caronTraitmatchingModelsPredict2024}
Caron, D., Brose, U., Lurgi, M., Blanchet, F. G., Gravel, D., \&
Pollock, L. J. (2024). Trait-matching models predict pairwise
interactions across regions, not food web properties. \emph{Global
Ecology and Biogeography}, \emph{33}(4), e13807.
\url{https://doi.org/10.1111/geb.13807}

\bibitem[\citeproctext]{ref-caronAddressingEltonianShortfall2022}
Caron, D., Maiorano, L., Thuiller, W., \& Pollock, L. J. (2022).
Addressing the {Eltonian} shortfall with trait-based interaction models.
\emph{Ecology Letters}, \emph{25}(4), 889--899.
\url{https://doi.org/10.1111/ele.13966}

\bibitem[\citeproctext]{ref-cherifEnvironmentRescueCan2024}
Cherif, M., Brose, U., Hirt, M. R., Ryser, R., Silve, V., Albert, G.,
Arnott, R., Berti, E., Cirtwill, A., Dyer, A., Gauzens, B., Gupta, A.,
Ho, H.-C., Portalier, S. M. J., Wain, D., \& Wootton, K. (2024). The
environment to the rescue: Can physics help predict predator--prey
interactions? \emph{Biological Reviews}, \emph{138}(1).
\url{https://doi.org/10.1111/brv.13105}

\bibitem[\citeproctext]{ref-cirtwillQuantitativeFrameworkInvestigating2019}
Cirtwill, A. R., Eklf, A., Roslin, T., Wootton, K., \& Gravel, D.
(2019). A quantitative framework for investigating the reliability of
empirical network construction. \emph{Methods in Ecology and Evolution},
\emph{10}(6), 902--911. \url{https://doi.org/10.1111/2041-210X.13180}

\bibitem[\citeproctext]{ref-cleggImpactIntraspecificVariation2018}
Clegg, T., Ali, M., \& Beckerman, A. P. (2018). The impact of
intraspecific variation on food web structure. \emph{Ecology},
\emph{99}(12), 2712--2720. \url{https://doi.org/10.1002/ecy.2523}

\bibitem[\citeproctext]{ref-curtsdotterEcosystemFunctionPredator2019}
Curtsdotter, A., Banks, H. T., Banks, J. E., Jonsson, M., Jonsson, T.,
Laubmeier, A. N., Traugott, M., \& Bommarco, R. (2019). Ecosystem
function in predator--prey food webs---confronting dynamic models with
empirical data. \emph{Journal of Animal Ecology}, \emph{88}(2),
196--210. \url{https://doi.org/10.1111/1365-2656.12892}

\bibitem[\citeproctext]{ref-dallarivaExploringEvolutionarySignature2016}
Dalla Riva, G. V., \& Stouffer, D. B. (2016). Exploring the evolutionary
signature of food webs' backbones using functional traits. \emph{Oikos},
\emph{125}(4), 446--456. \url{https://doi.org/10.1111/oik.02305}

\bibitem[\citeproctext]{ref-dallasPredictingCrypticLinks2017}
Dallas, T., Park, A. W., \& Drake, J. M. (2017). Predicting cryptic
links in host-parasite networks. \emph{PLOS Computational Biology},
\emph{13}(5), e1005557.
\url{https://doi.org/10.1371/journal.pcbi.1005557}

\bibitem[\citeproctext]{ref-danetResponseDiversityMajor2024}
Danet, A., Kéfi, S., Johnson, T. F., \& Beckerman, A. P. (2024).
\emph{Response diversity is a major driver of temporal stability in
complex food webs} (p. 2024.08.29.610288). bioRxiv.
\url{https://doi.org/10.1101/2024.08.29.610288}

\bibitem[\citeproctext]{ref-dansereauSpatiallyExplicitPredictions2024}
Dansereau, G., Barros, C., \& Poisot, T. (2024). Spatially explicit
predictions of food web structure from regional-level data.
\emph{Philosophical Transactions of the Royal Society B: Biological
Sciences}, \emph{379}(1909).
\url{https://doi.org/10.1098/rstb.2023.0166}

\bibitem[\citeproctext]{ref-dansereauOvercomingDisconnectInteraction2024}
Dansereau, G., Braga, J., Ficetola, G. F., Galiana, N., Gravel, D.,
Maiorano, L., Montoya, J. M., O'Connor, L., Pollock, L. J., Thuiller,
W., Poisot, T., \& Barros, C. (2024). \emph{Overcoming the disconnect
between interaction networks and biodiversity conservation and
management}.

\bibitem[\citeproctext]{ref-delmasAnalysingEcologicalNetworks2019}
Delmas, E., Besson, M., Brice, M.-H., Burkle, L. A., Riva, G. V. D.,
Fortin, M.-J., Gravel, D., Guimarães, P. R., Hembry, D. H., Newman, E.
A., Olesen, J. M., Pires, M. M., Yeakel, J. D., \& Poisot, T. (2019).
Analysing ecological networks of species interactions. \emph{Biological
Reviews}, \emph{94}(1), 16--36. \url{https://doi.org/10.1111/brv.12433}

\bibitem[\citeproctext]{ref-delmasSimulationsBiomassDynamics2017}
Delmas, E., Brose, U., Gravel, D., Stouffer, D. B., \& Poisot, T.
(2017). Simulations of biomass dynamics in community food webs.
\emph{Methods in Ecology and Evolution}, \emph{8}(7), 881--886.
\url{https://doi.org/10.1111/2041-210X.12713}

\bibitem[\citeproctext]{ref-desjardins-proulxEcologicalInteractionsNetflix2017}
Desjardins-Proulx, P., Laigle, I., Poisot, T., \& Gravel, D. (2017).
Ecological interactions and the {Netflix} problem. \emph{PeerJ},
\emph{5}, e3644. \url{https://doi.org/10.7717/peerj.3644}

\bibitem[\citeproctext]{ref-dormannRisePossibleFall2023}
Dormann, C. F. (2023). The rise, and possible fall, of network ecology.
In \emph{Defining {Agroecology} -- {A Festschrift} for {Teja
Tscharntke}} (pp. 143--159.). Tredition.

\bibitem[\citeproctext]{ref-dunhillExtinctionCascadesCommunity2024}
Dunhill, A. M., Zarzyczny, K., Shaw, J. O., Atkinson, J. W., Little, C.
T. S., \& Beckerman, A. P. (2024). Extinction cascades, community
collapse, and recovery across a {Mesozoic} hyperthermal event.
\emph{Nature Communications}, \emph{15}(1), 8599.
\url{https://doi.org/10.1038/s41467-024-53000-2}

\bibitem[\citeproctext]{ref-dunneNetworkStructureFood2006}
Dunne, J. A. (2006). The {Network Structure} of {Food Webs}. In J. A.
Dunne \& M. Pascual (Eds.), \emph{Ecological networks: {Linking}
structure and dynamics} (pp. 27--86). Oxford University Press.

\bibitem[\citeproctext]{ref-dunneCompilationNetworkAnalyses2008}
Dunne, J. A., Williams, R. J., Martinez, N. D., Wood, R. A., \& Erwin,
D. H. (2008). Compilation and {Network Analyses} of {Cambrian Food
Webs}. \emph{PLOS Biology}, \emph{6}(4), e102.
\url{https://doi.org/10.1371/journal.pbio.0060102}

\bibitem[\citeproctext]{ref-eklofSecondaryExtinctionsFood2013}
Eklöf, A., Tang, S., \& Allesina, S. (2013). Secondary extinctions in
food webs: A {Bayesian} network approach. \emph{Methods in Ecology and
Evolution}, \emph{4}(8), 760--770.
\url{https://doi.org/10.1111/2041-210X.12062}

\bibitem[\citeproctext]{ref-fortunaHabitatLossStructure2006}
Fortuna, M. A., \& Bascompte, J. (2006). Habitat loss and the structure
of plant-animal mutualistic networks: {Mutualistic} networks and habitat
loss. \emph{Ecology Letters}, \emph{9}(3), 281--286.
\url{https://doi.org/10.1111/j.1461-0248.2005.00868.x}

\bibitem[\citeproctext]{ref-frickeCollapseTerrestrialMammal2022}
Fricke, E. C., Hsieh, C., Middleton, O., Gorczynski, D., Cappello, C.
D., Sanisidro, O., Rowan, J., Svenning, J.-C., \& Beaudrot, L. (2022).
Collapse of terrestrial mammal food webs since the {Late Pleistocene}.
\emph{Science}, \emph{377}(6609), 1008--1011.
\url{https://doi.org/10.1126/science.abn4012}

\bibitem[\citeproctext]{ref-garcia-callejasNonrandomInteractionsGuilds2023}
García-Callejas, D., Godoy, O., Buche, L., Hurtado, M., Lanuza, J. B.,
Allen-Perkins, A., \& Bartomeus, I. (2023). Non-random interactions
within and across guilds shape the potential to coexist in multi-trophic
ecological communities. \emph{Ecology Letters}, \emph{26}(6), 831--842.
\url{https://doi.org/10.1111/ele.14206}

\bibitem[\citeproctext]{ref-golubskiModifyingModifiersWhat2011}
Golubski, A. J., \& Abrams, P. A. (2011). Modifying modifiers: What
happens when interspecific interactions interact? \emph{Journal of
Animal Ecology}, \emph{80}(5), 1097--1108.
\url{https://doi.org/10.1111/j.1365-2656.2011.01852.x}

\bibitem[\citeproctext]{ref-gomezEcologicalInteractionsAre2010}
Gómez, J. M., Verdú, M., \& Perfectti, F. (2010). Ecological
interactions are evolutionarily conserved across the entire tree of
life. \emph{Nature}, \emph{465}(7300), 918--921.
\url{https://doi.org/10.1038/nature09113}

\bibitem[\citeproctext]{ref-gravelBringingEltonGrinnell2019}
Gravel, D., Baiser, B., Dunne, J. A., Kopelke, J.-P., Martinez, N. D.,
Nyman, T., Poisot, T., Stouffer, D. B., Tylianakis, J. M., Wood, S. A.,
\& Roslin, T. (2019). Bringing {Elton} and {Grinnell} together: A
quantitative framework to represent the biogeography of ecological
interaction networks. \emph{Ecography}, \emph{42}(3), 401--415.
\url{https://doi.org/10.1111/ecog.04006}

\bibitem[\citeproctext]{ref-higinoMismatchIUCNRange2023}
Higino, G. T., Banville, F., Dansereau, G., Muñoz, N. R. F., Windsor,
F., \& Poisot, T. (2023). Mismatch between {IUCN} range maps and species
interactions data illustrated using the {Serengeti} food web.
\emph{PeerJ}, \emph{11}, e14620.
\url{https://doi.org/10.7717/peerj.14620}

\bibitem[\citeproctext]{ref-hubbellUnifiedNeutralTheory2001}
Hubbell, S. P. (2001). \emph{The {Unified Neutral Theory} of
{Biodiversity} and {Biogeography} ({MPB-32})}. Princeton University
Press. \url{https://www.jstor.org/stable/j.ctt7rj8w}

\bibitem[\citeproctext]{ref-huiHowInvadeEcological2019}
Hui, C., \& Richardson, D. M. (2019). How to {Invade} an {Ecological
Network}. \emph{Trends in Ecology \& Evolution}, \emph{34}(2), 121--131.
\url{https://doi.org/10.1016/j.tree.2018.11.003}

\bibitem[\citeproctext]{ref-ingsEcologicalNetworksbeyondFood2009}
Ings, T. C., Montoya, J. M., Bascompte, J., Blüthgen, N., Brown, L.,
Dormann, C. F., Edwards, F., Figueroa, D., Jacob, U., Jones, J. I.,
Lauridsen, R. B., Ledger, M. E., Lewis, H. M., Olesen, J. M., van Veen,
F. J. F., Warren, P. H., \& Woodward, G. (2009). Ecological
networks--beyond food webs. \emph{The Journal of Animal Ecology},
\emph{78}(1), 253--269.
\url{https://doi.org/10.1111/j.1365-2656.2008.01460.x}

\bibitem[\citeproctext]{ref-jordanoChasingEcologicalInteractions2016}
Jordano, P. (2016a). Chasing {Ecological Interactions}. \emph{PLOS
Biology}, \emph{14}(9), e1002559.
\url{https://doi.org/10.1371/journal.pbio.1002559}

\bibitem[\citeproctext]{ref-jordanoSamplingNetworksEcological2016}
Jordano, P. (2016b). Sampling networks of ecological interactions.
\emph{Functional Ecology}. \url{https://doi.org/10.1111/1365-2435.12763}

\bibitem[\citeproctext]{ref-kamaruDisruptionAntplantMutualism2024}
Kamaru, D. N., Palmer, T. M., Riginos, C., Ford, A. T., Belnap, J.,
Chira, R. M., Githaiga, J. M., Gituku, B. C., Hays, B. R., Kavwele, C.
M., Kibungei, A. K., Lamb, C. T., Maiyo, N. J., Milligan, P. D.,
Mutisya, S., Ng'weno, C. C., Ogutu, M., Pietrek, A. G., Wildt, B. T., \&
Goheen, J. R. (2024). Disruption of an ant-plant mutualism shapes
interactions between lions and their primary prey. \emph{Science},
\emph{383}(6681), 433--438.
\url{https://doi.org/10.1126/science.adg1464}

\bibitem[\citeproctext]{ref-kefiNetworkStructureFood2015}
Kéfi, S., Berlow, E. L., Wieters, E. A., Joppa, L. N., Wood, S. A.,
Brose, U., \& Navarrete, S. A. (2015). Network structure beyond food
webs: Mapping non-trophic and trophic interactions on {Chilean} rocky
shores. \emph{Ecology}, \emph{96}(1), 291--303.
\url{https://doi.org/10.1890/13-1424.1}

\bibitem[\citeproctext]{ref-kefiMoreMealIntegrating2012}
Kéfi, S., Berlow, E. L., Wieters, E. A., Navarrete, S. A., Petchey, O.
L., Wood, S. A., Boit, A., Joppa, L. N., Lafferty, K. D., Williams, R.
J., Martinez, N. D., Menge, B. A., Blanchette, C. A., Iles, A. C., \&
Brose, U. (2012). More than a meal{\ldots{}} integrating non-feeding
interactions into food webs: {More} than a meal {\ldots{}}.
\emph{Ecology Letters}, \emph{15}(4), 291--300.
\url{https://doi.org/10.1111/j.1461-0248.2011.01732.x}

\bibitem[\citeproctext]{ref-krauseCompartmentsRevealedFoodweb2003}
Krause, A. E., Frank, K. A., Mason, D. M., Ulanowicz, R. E., \& Taylor,
W. W. (2003). Compartments revealed in food-web structure.
\emph{Nature}, \emph{426}(6964), 282--285.
\url{https://doi.org/10.1038/nature02115}

\bibitem[\citeproctext]{ref-krishnaNeutralnicheTheoryNestedness2008}
Krishna, A., Guimarães Jr, P. R., Jordano, P., \& Bascompte, J. (2008).
A neutral-niche theory of nestedness in mutualistic networks.
\emph{Oikos}, \emph{117}(11), 1609--1618.
\url{https://doi.org/10.1111/j.1600-0706.2008.16540.x}

\bibitem[\citeproctext]{ref-lajaaitiEcologicalNetworksDynamicsJlJulia2024}
Lajaaiti, I., Bonnici, I., Kéfi, S., Mayall, H., Danet, A., Beckerman,
A. P., Malpas, T., \& Delmas, E. (2024).
\emph{{EcologicalNetworksDynamics}.jl {A Julia} package to simulate the
temporal dynamics of complex ecological networks} (p.
2024.03.20.585899). bioRxiv.
\url{https://doi.org/10.1101/2024.03.20.585899}

\bibitem[\citeproctext]{ref-lindemanTrophicDynamicAspectEcology1942}
Lindeman, R. L. (1942). The {Trophic-Dynamic Aspect} of {Ecology}.
\emph{Ecology}, \emph{23}(4), 399--417.
\url{https://doi.org/10.2307/1930126}

\bibitem[\citeproctext]{ref-llewelynPredictingPredatorPrey2023}
Llewelyn, J., Strona, G., Dickman, C. R., Greenville, A. C., Wardle, G.
M., Lee, M. S. Y., Doherty, S., Shabani, F., Saltré, F., \& Bradshaw, C.
J. A. (2023). Predicting predator--prey interactions in terrestrial
endotherms using random forest. \emph{Ecography}, \emph{2023}(9),
e06619. \url{https://doi.org/10.1111/ecog.06619}

\bibitem[\citeproctext]{ref-loreauBiodiversityEcosystemStability2013}
Loreau, M., \& de Mazancourt, C. (2013). Biodiversity and ecosystem
stability: A synthesis of underlying mechanisms. \emph{Ecology Letters},
\emph{16}(s1), 106--115. \url{https://doi.org/10.1111/ele.12073}

\bibitem[\citeproctext]{ref-momalTreebasedInferenceSpecies2020}
Momal, R., Robin, S., \& Ambroise, C. (2020). Tree-based inference of
species interaction networks from abundance data. \emph{Methods in
Ecology and Evolution}, \emph{11}(5), 621--632.
\url{https://doi.org/10.1111/2041-210X.13380}

\bibitem[\citeproctext]{ref-morales-castillaInferringBioticInteractions2015}
Morales-Castilla, I., Matias, M. G., Gravel, D., \& Araújo, M. B.
(2015). Inferring biotic interactions from proxies. \emph{Trends in
Ecology \& Evolution}, \emph{30}(6), 347--356.
\url{https://doi.org/10.1016/j.tree.2015.03.014}

\bibitem[\citeproctext]{ref-moulatletScalingTrophicSpecialization2024}
Moulatlet, G., Luna, P., Dattilo, W., \& Villalobos, F. (2024).
\emph{The scaling of trophic specialization in interaction networks
across levels of organization}. Authorea.
\url{https://doi.org/10.22541/au.172977303.33335171/v1}

\bibitem[\citeproctext]{ref-pawarDimensionalityConsumerSearch2012}
Pawar, S., Dell, A. I., \& Savage, V. M. (2012). Dimensionality of
consumer search space drives trophic interaction strengths.
\emph{Nature}, \emph{486}(7404), 485--489.
\url{https://doi.org/10.1038/nature11131}

\bibitem[\citeproctext]{ref-petcheySizeForagingFood2008}
Petchey, O. L., Beckerman, A. P., Riede, J. O., \& Warren, P. H. (2008).
Size, foraging, and food web structure. \emph{Proceedings of the
National Academy of Sciences}, \emph{105}(11), 4191--4196.
\url{https://doi.org/10.1073/pnas.0710672105}

\bibitem[\citeproctext]{ref-pichlerMachineLearningAlgorithms2020}
Pichler, M., Boreux, V., Klein, A.-M., Schleuning, M., \& Hartig, F.
(2020). Machine learning algorithms to infer trait-matching and predict
species interactions in ecological networks. \emph{Methods in Ecology
and Evolution}, \emph{11}(2), 281--293.
\url{https://doi.org/10.1111/2041-210X.13329}

\bibitem[\citeproctext]{ref-pilosofMultilayerNatureEcological2017}
Pilosof, S., Porter, M. A., Pascual, M., \& Kéfi, S. (2017). The
multilayer nature of ecological networks. \emph{Nature Ecology \&
Evolution}, \emph{1}(4), 101.
\url{https://doi.org/10.1038/s41559-017-0101}

\bibitem[\citeproctext]{ref-poisotGuidelinesPredictionSpecies2023}
Poisot, T. (2023). Guidelines for the prediction of species interactions
through binary classification. \emph{Methods in Ecology and Evolution},
\emph{14}(5), 1333--1345. \url{https://doi.org/10.1111/2041-210X.14071}

\bibitem[\citeproctext]{ref-poisotStructureProbabilisticNetworks2016}
Poisot, T., Cirtwill, A., Cazelles, K., Gravel, D., Fortin, M.-J., \&
Stouffer, D. (2016). The structure of probabilistic networks.
\emph{Methods in Ecology and Evolution}, \emph{7}(3), 303--312.
\url{https://doi.org/10}

\bibitem[\citeproctext]{ref-poisotSpeciesWhyEcological2015}
Poisot, T., Stouffer, D. B., \& Gravel, D. (2015). Beyond species: Why
ecological interaction networks vary through space and time.
\emph{Oikos}, \emph{124}(3), 243--251.
\url{https://doi.org/10.1111/oik.01719}

\bibitem[\citeproctext]{ref-poisotDescribeUnderstandPredict2016}
Poisot, T., Stouffer, D. B., \& Kéfi, S. (2016). Describe, understand
and predict: Why do we need networks in ecology? \emph{Functional
Ecology}, \emph{30}(12), 1878--1882.
\url{https://www.jstor.org/stable/48582345}

\bibitem[\citeproctext]{ref-polisComplexTrophicInteractions1991}
Polis, G. A. (1991). Complex {Trophic Interactions} in {Deserts}: {An
Empirical Critique} of {Food-Web Theory}. \emph{The American
Naturalist}, \emph{138}(1), 123--155.
\url{https://doi.org/10.1086/285208}

\bibitem[\citeproctext]{ref-pollockUnderstandingCooccurrenceModelling2014}
Pollock, L. J., Tingley, R., Morris, W. K., Golding, N., O'Hara, R. B.,
Parris, K. M., Vesk, P. A., \& McCarthy, M. A. (2014). Understanding
co-occurrence by modelling species simultaneously with a {Joint Species
Distribution Model} ({JSDM}). \emph{Methods in Ecology and Evolution},
\emph{5}(5), 397--406. \url{https://doi.org/10.1111/2041-210X.12180}

\bibitem[\citeproctext]{ref-pomeranzInferringPredatorPrey2019}
Pomeranz, J. P. F., Thompson, R. M., Poisot, T., \& Harding, J. S.
(2019). Inferring predator--prey interactions in food webs.
\emph{Methods in Ecology and Evolution}, \emph{10}(3), 356--367.
\url{https://doi.org/10.1111/2041-210X.13125}

\bibitem[\citeproctext]{ref-portalierMechanicsPredatorPrey2019}
Portalier, S. M. J., Fussmann, G. F., Loreau, M., \& Cherif, M. (2019).
The mechanics of predator--prey interactions: {First} principles of
physics predict predator--prey size ratios. \emph{Functional Ecology},
\emph{33}(2), 323--334. \url{https://doi.org/10.1111/1365-2435.13254}

\bibitem[\citeproctext]{ref-pringleUntanglingFoodWebs2020}
Pringle, R. M. (2020). Untangling {Food Webs}. In \emph{Unsolved
{Problems} in {Ecology}} (pp. 225--238). Princeton University Press.
\url{https://doi.org/10.1515/9780691195322-020}

\bibitem[\citeproctext]{ref-pringleResolvingFoodWebStructure2020}
Pringle, R. M., \& Hutchinson, M. C. (2020). Resolving {Food-Web
Structure}. \emph{Annual Review of Ecology, Evolution and Systematics},
\emph{51}(Volume 51, 2020), 55--80.
\url{https://doi.org/10.1146/annurev-ecolsys-110218-024908}

\bibitem[\citeproctext]{ref-proulxNetworkThinkingEcology2005}
Proulx, S. R., Promislow, D. E. L., \& Phillips, P. C. (2005). Network
thinking in ecology and evolution. \emph{Trends in Ecology \&
Evolution}, \emph{20}(6), 345--353.
\url{https://doi.org/10.1016/j.tree.2005.04.004}

\bibitem[\citeproctext]{ref-pykeOptimalForagingTheory1984}
Pyke, G. (1984). Optimal {Foraging Theory}: {A Critical Review}.
\emph{Annual Review of Ecology, Evolution and Systematic}, \emph{15},
523--575. \url{https://doi.org/10.1146/annurev.ecolsys.15.1.523}

\bibitem[\citeproctext]{ref-quinteroDownscalingMutualisticNetworks2024}
Quintero, E., Arroyo-Correa, B., Isla, J., Rodríguez-Sánchez, F., \&
Jordano, P. (2024). \emph{Downscaling mutualistic networks from species
to individuals reveals consistent interaction niches and roles within
plant populations} (p. 2024.02.02.578595). bioRxiv.
\url{https://doi.org/10.1101/2024.02.02.578595}

\bibitem[\citeproctext]{ref-roopnarineExtinctionCascadesCatastrophe2006}
Roopnarine, P. D. (2006). Extinction {Cascades} and {Catastrophe} in
{Ancient Food Webs}. \emph{Paleobiology}, \emph{32}(1), 1--19.
\url{https://www.jstor.org/stable/4096814}

\bibitem[\citeproctext]{ref-roopnarineEcologicalModellingPaleocommunity2017}
Roopnarine, P. D. (2017). Ecological {Modelling} of {Paleocommunity Food
Webs}. In \emph{Conservation {Paleobiology}: {Using} the {Past} to
{Manage} for the {Future}} (pp. 201--226). University of Chicago Press.

\bibitem[\citeproctext]{ref-saberskiImpactDataResolution2024}
Saberski, E., Lorimer, T., Carpenter, D., Deyle, E., Merz, E., Park, J.,
Pao, G. M., \& Sugihara, G. (2024). The impact of data resolution on
dynamic causal inference in multiscale ecological networks.
\emph{Communications Biology}, \emph{7}(1), 1--10.
\url{https://doi.org/10.1038/s42003-024-07054-z}

\bibitem[\citeproctext]{ref-segarRoleEvolutionShaping2020}
Segar, S. T., Fayle, T. M., Srivastava, D. S., Lewinsohn, T. M., Lewis,
O. T., Novotny, V., Kitching, R. L., \& Maunsell, S. C. (2020). The
{Role} of {Evolution} in {Shaping Ecological Networks}. \emph{Trends in
Ecology \& Evolution}, \emph{35}(5), 454--466.
\url{https://doi.org/10.1016/j.tree.2020.01.004}

\bibitem[\citeproctext]{ref-shawFrameworkReconstructingAncient2024}
Shaw, J. O., Dunhill, A. M., Beckerman, A. P., Dunne, J. A., \& Hull, P.
M. (2024). \emph{A framework for reconstructing ancient food webs using
functional trait data} (p. 2024.01.30.578036). bioRxiv.
\url{https://doi.org/10.1101/2024.01.30.578036}

\bibitem[\citeproctext]{ref-simmonsRefocusingMultipleStressor2021}
Simmons, B. I., Blyth, P. S. A., Blanchard, J. L., Clegg, T., Delmas,
E., Garnier, A., Griffiths, C. A., Jacob, U., Pennekamp, F., Petchey, O.
L., Poisot, T., Webb, T. J., \& Beckerman, A. P. (2021). Refocusing
multiple stressor research around the targets and scales of ecological
impacts. \emph{Nature Ecology \& Evolution}, \emph{5}(11), 1478--1489.
\url{https://doi.org/10.1038/s41559-021-01547-4}

\bibitem[\citeproctext]{ref-soberonGrinnellianEltonianNiches2007}
Soberón, J. (2007). Grinnellian and {Eltonian} niches and geographic
distributions of species. \emph{Ecology Letters}, \emph{10}(12),
1115--1123. \url{https://doi.org/10.1111/j.1461-0248.2007.01107.x}

\bibitem[\citeproctext]{ref-songRigorousValidationEcological2024}
Song, C., \& Levine, J. M. (2024). \emph{Rigorous (in)validation of
ecological models} (p. 2024.09.19.613075). bioRxiv.
\url{https://doi.org/10.1101/2024.09.19.613075}

\bibitem[\citeproctext]{ref-staniczenkoStructuralDynamicsRobustness2010}
Staniczenko, P. P. A., Lewis, O. T., Jones, N. S., \& Reed-Tsochas, F.
(2010). Structural dynamics and robustness of food webs. \emph{Ecology
Letters}, \emph{13}(7), 891--899.
\url{https://doi.org/10.1111/j.1461-0248.2010.01485.x}

\bibitem[\citeproctext]{ref-stephensForagingTheory1986}
Stephens, D. W., \& Krebs, J. R. (1986). \emph{Foraging {Theory}} (Vol.
1). Princeton University Press.
\url{https://doi.org/10.2307/j.ctvs32s6b}

\bibitem[\citeproctext]{ref-stockPairwiseLearningPredicting2021}
Stock, M. (2021). Pairwise learning for predicting pollination
interactions based on traits and phylogeny. \emph{Ecological Modelling},
14.

\bibitem[\citeproctext]{ref-stoufferAllEcologicalModels2019}
Stouffer, D. B. (2019). All ecological models are wrong, but some are
useful. \emph{Journal of Animal Ecology}, \emph{88}(2), 192--195.
\url{https://doi.org/10.1111/1365-2656.12949}

\bibitem[\citeproctext]{ref-strydomFoodWebReconstruction2022}
Strydom, T., Bouskila, S., Banville, F., Barros, C., Caron, D., Farrell,
M. J., Fortin, M.-J., Hemming, V., Mercier, B., Pollock, L. J., Runghen,
R., Dalla Riva, G. V., \& Poisot, T. (2022). Food web reconstruction
through phylogenetic transfer of low-rank network representation.
\emph{Methods in Ecology and Evolution}, \emph{13}(12), 2838--2849.
\url{https://doi.org/10.1111/2041-210X.13835}

\bibitem[\citeproctext]{ref-strydomGraphEmbeddingTransfer2023}
Strydom, T., Bouskila, S., Banville, F., Barros, C., Caron, D., Farrell,
M. J., Fortin, M.-J., Mercier, B., Pollock, L. J., Runghen, R., Dalla
Riva, G. V., \& Poisot, T. (2023). Graph embedding and transfer learning
can help predict potential species interaction networks despite data
limitations. \emph{Methods in Ecology and Evolution}, \emph{14}(12),
2917--2930. \url{https://doi.org/10.1111/2041-210X.14228}

\bibitem[\citeproctext]{ref-strydomRoadmapPredictingSpecies2021}
Strydom, T., Catchen, M. D., Banville, F., Caron, D., Dansereau, G.,
Desjardins-Proulx, P., Forero-Muñoz, N. R., Higino, G., Mercier, B.,
Gonzalez, A., Gravel, D., Pollock, L., \& Poisot, T. (2021). A roadmap
towards predicting species interaction networks (across space and time).
\emph{Philosophical Transactions of the Royal Society B: Biological
Sciences}, \emph{376}(1837), 20210063.
\url{https://doi.org/10.1098/rstb.2021.0063}

\bibitem[\citeproctext]{ref-strydomSVDEntropyReveals2021}
Strydom, T., Dalla Riva, G. V., \& Poisot, T. (2021). {SVD Entropy
Reveals} the {High Complexity} of {Ecological Networks}. \emph{Frontiers
in Ecology and Evolution}, \emph{9}.
\url{https://doi.org/10.3389/fevo.2021.623141}

\bibitem[\citeproctext]{ref-terryFindingMissingLinks2020}
Terry, J. C. D., \& Lewis, O. T. (2020). Finding missing links in
interaction networks. \emph{Ecology}, \emph{101}(7), e03047.
\url{https://doi.org/10.1002/ecy.3047}

\bibitem[\citeproctext]{ref-vandewalleArthropodFoodWebs2023}
Van De Walle, R., Logghe, G., Haas, N., Massol, F., Vandegehuchte, M.
L., \& Bonte, D. (2023). Arthropod food webs predicted from body size
ratios are improved by incorporating prey defensive properties.
\emph{Journal of Animal Ecology}, \emph{92}(4), 913--924.
\url{https://doi.org/10.1111/1365-2656.13905}

\bibitem[\citeproctext]{ref-vazquezUnitingPatternProcess2009}
Vázquez, D. P., Blüthgen, N., Cagnolo, L., \& Chacoff, N. P. (2009).
Uniting pattern and process in plant--animal mutualistic networks: A
review. \emph{Annals of Botany}, \emph{103}(9), 1445--1457.
\url{https://doi.org/10.1093/aob/mcp057}

\bibitem[\citeproctext]{ref-wellsSpeciesInteractionsEstimating2013}
Wells, K., \& O'Hara, R. B. (2013). Species interactions: Estimating
per-individual interaction strength and covariates before simplifying
data into per-species ecological networks. \emph{Methods in Ecology and
Evolution}, \emph{4}(1), 1--8.
\url{https://doi.org/10.1111/j.2041-210x.2012.00249.x}

\bibitem[\citeproctext]{ref-williamsSimpleRulesYield2000}
Williams, R. J., \& Martinez, N. D. (2000). Simple rules yield complex
food webs. \emph{Nature}, \emph{404}(6774), 180--183.
\url{https://doi.org/10.1038/35004572}

\bibitem[\citeproctext]{ref-williamsSuccessItsLimits2008}
Williams, R. J., \& Martinez, N. D. (2008). Success and its limits among
structural models of complex food webs. \emph{Journal of Animal
Ecology}, \emph{77}(3), 512--519.
\url{https://doi.org/10.1111/j.1365-2656.2008.01362.x}

\bibitem[\citeproctext]{ref-windsorUsingEcologicalNetworks2023}
Windsor, F. M., van den Hoogen, J., Crowther, T. W., \& Evans, D. M.
(2023). Using ecological networks to answer questions in global
biogeography and ecology. \emph{Journal of Biogeography}, \emph{50}(1),
57--69. \url{https://doi.org/10.1111/jbi.14447}

\bibitem[\citeproctext]{ref-woosterAustraliasRecentlyEstablished2024}
Wooster, E. I. F., Middleton, O. S., Wallach, A. D., Ramp, D.,
Sanisidro, O., Harris, V. K., Rowan, J., Schowanek, S. D., Gordon, C.
E., Svenning, J.-C., Davis, M., Scharlemann, J. P. W., Nimmo, D. G.,
Lundgren, E. J., \& Sandom, C. J. (2024). Australia's recently
established predators restore complexity to food webs simplified by
extinction. \emph{Current Biology}, \emph{34}(22), 5164--5172.e2.
\url{https://doi.org/10.1016/j.cub.2024.09.049}

\bibitem[\citeproctext]{ref-woottonModularTheoryTrophic2023}
Wootton, K. L., Curtsdotter, A., Roslin, T., Bommarco, R., \& Jonsson,
T. (2023). Towards a modular theory of trophic interactions.
\emph{Functional Ecology}, \emph{37}(1), 26--43.
\url{https://doi.org/10.1111/1365-2435.13954}

\bibitem[\citeproctext]{ref-yeakelCollapseEcologicalNetwork2014}
Yeakel, J. D., Pires, M. M., Rudolf, L., Dominy, N. J., Koch, P. L.,
Guimarães, P. R., \& Gross, T. (2014). Collapse of an ecological network
in {Ancient Egypt}. \emph{PNAS}, \emph{111}(40), 14472--14477.
\url{https://doi.org/10.1073/pnas.1408471111}

\bibitem[\citeproctext]{ref-yodzisCompartmentationRealAssembled1982}
Yodzis, P. (1982). The {Compartmentation} of {Real} and {Assembled
Ecosystems}. \emph{The American Naturalist}, \emph{120}(5), 551--570.
\url{https://doi.org/10.1086/284013}

\end{CSLReferences}





\end{document}
