{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Extended model descriptions\n",
    "\n",
    "Tanya Strydom [![](data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAA2ZpVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADw/eHBhY2tldCBiZWdpbj0i77u/IiBpZD0iVzVNME1wQ2VoaUh6cmVTek5UY3prYzlkIj8+IDx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8iIHg6eG1wdGs9IkFkb2JlIFhNUCBDb3JlIDUuMC1jMDYwIDYxLjEzNDc3NywgMjAxMC8wMi8xMi0xNzozMjowMCAgICAgICAgIj4gPHJkZjpSREYgeG1sbnM6cmRmPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5LzAyLzIyLXJkZi1zeW50YXgtbnMjIj4gPHJkZjpEZXNjcmlwdGlvbiByZGY6YWJvdXQ9IiIgeG1sbnM6eG1wTU09Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC9tbS8iIHhtbG5zOnN0UmVmPSJodHRwOi8vbnMuYWRvYmUuY29tL3hhcC8xLjAvc1R5cGUvUmVzb3VyY2VSZWYjIiB4bWxuczp4bXA9Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC8iIHhtcE1NOk9yaWdpbmFsRG9jdW1lbnRJRD0ieG1wLmRpZDo1N0NEMjA4MDI1MjA2ODExOTk0QzkzNTEzRjZEQTg1NyIgeG1wTU06RG9jdW1lbnRJRD0ieG1wLmRpZDozM0NDOEJGNEZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wTU06SW5zdGFuY2VJRD0ieG1wLmlpZDozM0NDOEJGM0ZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wOkNyZWF0b3JUb29sPSJBZG9iZSBQaG90b3Nob3AgQ1M1IE1hY2ludG9zaCI+IDx4bXBNTTpEZXJpdmVkRnJvbSBzdFJlZjppbnN0YW5jZUlEPSJ4bXAuaWlkOkZDN0YxMTc0MDcyMDY4MTE5NUZFRDc5MUM2MUUwNEREIiBzdFJlZjpkb2N1bWVudElEPSJ4bXAuZGlkOjU3Q0QyMDgwMjUyMDY4MTE5OTRDOTM1MTNGNkRBODU3Ii8+IDwvcmRmOkRlc2NyaXB0aW9uPiA8L3JkZjpSREY+IDwveDp4bXBtZXRhPiA8P3hwYWNrZXQgZW5kPSJyIj8+84NovQAAAR1JREFUeNpiZEADy85ZJgCpeCB2QJM6AMQLo4yOL0AWZETSqACk1gOxAQN+cAGIA4EGPQBxmJA0nwdpjjQ8xqArmczw5tMHXAaALDgP1QMxAGqzAAPxQACqh4ER6uf5MBlkm0X4EGayMfMw/Pr7Bd2gRBZogMFBrv01hisv5jLsv9nLAPIOMnjy8RDDyYctyAbFM2EJbRQw+aAWw/LzVgx7b+cwCHKqMhjJFCBLOzAR6+lXX84xnHjYyqAo5IUizkRCwIENQQckGSDGY4TVgAPEaraQr2a4/24bSuoExcJCfAEJihXkWDj3ZAKy9EJGaEo8T0QSxkjSwORsCAuDQCD+QILmD1A9kECEZgxDaEZhICIzGcIyEyOl2RkgwAAhkmC+eAm0TAAAAABJRU5ErkJggg==)](https://orcid.org/0000-0001-6067-1349)  \n",
    "May 2, 2024\n",
    "\n",
    "# Extended model descriptions\n",
    "\n",
    "## Network models\n",
    "\n",
    "### Null models\n",
    "\n",
    "**Type I** \\[@fortunaHabitatLossStructure2006\\]:\n",
    "\n",
    "**Type II** \\[@bascompteNestedAssemblyPlantanimal2003\\]:\n",
    "\n",
    "### Neutral models\n",
    "\n",
    "### Resource models\n",
    "\n",
    "**Cascade model** \\[@cohenCommunityFoodWebs1990\\]: Much like the name suggests the cascade model rests on the idea that species feed on one another in a hierarchical manner. This rests on the assumption that the links within a network are variably distributed across the network; with the proportion of links decreasing as one moves up the trophic levels (*i.e.,* ‘many’ prey and ‘few’ predators). This is achieved by assigning all species a random rank, this rank will then determine both the predators and prey of that species. A species will have a particular probability of being fed on by any species with a higher ranking than it, this probability is constrained by the specified connectance of the network. Interestingly here ‘species’ are treated as any individual that consume and are consumed by the same ‘species’, *i.e.,* these are not taxonomical species \\[@cohenStochasticTheoryCommunity1985\\]. The original cascade model has altered to be more ‘generalised’ \\[@stoufferQuantitativePatternsStructure2005\\], which altered the probability distribution of the prey that could be consumed by a species.\n",
    "\n",
    "**Niche models** \\[@williamsSimpleRulesYield2000\\]: The niche model introduces the idea that species interactions are based on the ‘feeding niche’ of a species. Broadly, all species are randomly assigned a ‘feeding niche’ range and all species that fall in this range can be consumed by that species (thereby allowing for cannibalism). The niche of each species is randomly assigned and the range of each species’ niche is (in part) constrained by the specified connectance of the network. The niche model has also been modified, although it appears that adding to the ‘complexity’ of the niche model does not improve on its ability to generate a more ecologically ‘correct’ network \\[@williamsSuccessItsLimits2008\\].\n",
    "\n",
    "**Nested hierarchy model** \\[@cattinPhylogeneticConstraintsAdaptation2004\\]\n",
    "\n",
    "## Interaction models\n",
    "\n",
    "### Energetic models\n",
    "\n",
    "**ADBM** \\[@petcheySizeForagingFood2008\\]:\n",
    "\n",
    "**DBM** \\[@beckermanForagingBiologyPredicts2006\\]:\n",
    "\n",
    "### Trait hierarchy\n",
    "\n",
    "**PFIM** \\[@shawFrameworkReconstructingAncient2024\\]:\n",
    "\n",
    "### Graph embedding\n",
    "\n",
    "**Transfer learning/RDPG** \\[@strydomFoodWebReconstruction2022\\]: The products of the embedding process are fed into a transfer learning framework for novel prediction…\n",
    "\n",
    "**Log-ratio** \\[@rohrModelingFoodWebs2010\\]: Interestingly often used in paleo settings (at least that’s what it currently looks like in my mind… \\[\\*e.g.,\\* @yeakelCollapseEcologicalNetwork2014, @piresMegafaunalExtinctionsHuman2020\\])\n",
    "\n",
    "### Other\n",
    "\n",
    "**Matching** \\[@rossbergFoodWebsExperts2006\\]: This one is more of a dynamic model (so BEF) and maybe beyond the scope of this work. I think there is value on only focusing on the ‘static’ models at this point (probably have said this before elsewhere but yeah)\n",
    "\n",
    "## Datasets used\n",
    "\n",
    "### Mangal networks\n",
    "\n",
    "We queried the Mangal \\[@poisotMangalMakingEcological2016\\] database and extracted a total of **TODO** networks. \\[*Some sort of summary as to the geographic/taxonomic range??\\]* Although these networks represent a high volume of interaction data they do not have accompanying ‘metadata’ that we would need for some of the more data-hungry model families (*e.g.,* local abundance), the Mangal networks were used to provide the ‘starting values’ for the random, resource, and generative families. This allows us to generate a large number of different networks that we can use to compare and contrast the performance of the various model families. For each network from Mangal we generated **TODO** versions of that network using each model family.\n",
    "\n",
    "> “These complex food webs differ in their level of resolution and sampling effort, which may introduce noise in the estimation of their properties, especially given their large number of interacting elements. However, because our MaxEnt models are applied on imperfect data, they aim at reproducing the sampled structure of food webs, not their actual structure.” - @banvilleWhatConstrainsFood2023 (something to think about…)\n",
    "\n",
    "### Empirical networks\n",
    "\n",
    "‘Elite’ number of datasets for interaction models\n",
    "\n",
    "Although the availability of empirical interaction data is growing as techniques begin to improve and grow \\[@pringleResolvingFoodWebStructure2020\\], we still lack a way to define what is the ‘ideal’ interaction dataset.\n",
    "\n",
    "New Zealand dataset(s): @pomeranzInferringPredatorPrey2019\n",
    "\n",
    "> Here I think we need to span a variety of domains, at minimum aquatic and terrestrial but maybe there should be a ‘scale’ element as well *i.e.,* a regional and local network. I think there is going to be a ‘turning point’ where structural will take over from mechanistic in terms of performance. More specifically at local scales bioenergetic constraints (and co-occurrence) may play a bigger role in structuring a network whereas at the metaweb level then mechanistic may make more (since by default its about who can potentially interact and obviously not constrained by real-world scenarios) *sensu* @caronTraitmatchingModelsPredict2024. Although having said that I feel that contradicts the idea of backbones (*sensu* Bramon Mora (sp?) et al & Stouffer et al) But that might be where we get the idea of core *structure* vs something like linkage density. So core things like trophic level/chain length will be conserved but connectance might not (I think I understand what I’m trying to say here)\n",
    "\n",
    "I think we should also use the @dunneCompilationNetworkAnalyses2008 work. Because 1) it gives the paleo-centric methods their moment in the sun and 2) I think it also brings up the interesting question of can we use modern structure to predict past ones?\n",
    "\n",
    "## Model benchmarking\n",
    "\n",
    "For now the (still essentially pending) workflow/associated code can be found at the following repository [BecksLab/topology_generators](https://github.com/BecksLab/topology_generators). This will reflect that which is shown in panel *B*.\n",
    "\n",
    "-   Data ‘cost’ (some methods might need a lot lot of supporting data vs something very light weight)\n",
    "-   I think it would be remiss to not also take into consideration computational cost\n",
    "-   Something about the network output - I’m acknowledging my biases and saying that probabilistic (or *maybe* weighted) links are the way\n",
    "\n",
    "### Network models\n",
    "\n",
    "-   connectance, nestedness (Bastolla et al., 2009), modularity (Barber, 2007), asymmetry (Delmas et al., 2018), and Jaccard network dissimilarity (Canard et al., 2014)\n",
    "\n",
    "-   *Shape:* do the models construct tall ‘pencil’ vs flat ‘pancake’ networks (Beckerman 2024, pers comms), generality/vulnerability, chain length (?)\n",
    "\n",
    "-   *Structure:* Predicting ‘structure’ - SVD \\[@strydomSVDEntropyReveals2021\\] but maybe something like nestedness as well (?)\n",
    "\n",
    "-   *Links:* are the number of links preserved (most network predicting models are to some extend link constrained but useful to see)\n",
    "\n",
    "-   *Motifs:* @staniczenkoStructuralDynamicsRobustness2010 uses S1, S2, S4, S5 from @stoufferEvidenceExistenceRobust2007\n",
    "\n",
    "    -   S1: Number of linear chains\n",
    "\n",
    "    -   S2: Number of omnivory motifs\n",
    "\n",
    "    -   S4: Number of apparent competition motifs\n",
    "\n",
    "    -   S5: Number of direct competition motifs\n",
    "\n",
    "### Interaction models\n",
    "\n",
    "-   Based on @poisotGuidelinesPredictionSpecies2023:\n",
    "    -   Precision-Recall (PR-AUC) - performance\n",
    "    -   Matthews correlation coefficient (MCC) - accuracy\n",
    "-   Maybe same measures we use for the network models\n",
    "\n",
    "### Action plan\n",
    "\n",
    "1.  Shortlist/finalise the different topo generators\n",
    "2.  collate/translate into `Julia`\n",
    "    -   *e.g.,* some models wil be in SpeciesInteractionNetworks.jl (new EcoNet); I know (parts of) the transfer learning stuff is and the niche model\n",
    "    -   others will need to be coded out (the more simpler models should be easier)\n",
    "3.  Curate networks for the different datasets/scenarios we select - I feel like there might be some scenarios that we can’t do all models for all datasets but maybe I’m being a pessimist.\n",
    "    -   Need to also think about where one might find the additional data for some of the models…\n",
    "        -   Body size: @herbersteinAnimalTraitsCuratedAnimal2022 - Although maybe Andrew has strong thotsTM RE the one true body size database to rule them all…\n",
    "        -   Other trait sources: @wilmanEltonTraitsSpecieslevelForaging2014 and @jonesPanTHERIASpecieslevelDatabase2009\n",
    "        -   This is where we’ll get the paleo traits from if I’m correct @bambachAutecologyFillingEcospace2007\n",
    "        -   Phylogeny stuff: @uphamInferringMammalTree2019 (what we used for TL but its only mammals…) but I’m sure there will be others\n",
    "    -   Also limitation of scope… *e.g.,* do we even dare to think about including plants/basal producers (see *e.g.,* @valdovinosBioenergeticFrameworkAboveground2023)\n",
    "    -   Taxonomic harmonisation - something to think about and check\n",
    "\n",
    "## References"
   ],
   "id": "3f2dfcb1-e302-4d43-8b23-d089a7243533"
  }
 ],
 "nbformat": 4,
 "nbformat_minor": 5,
 "metadata": {}
}
