{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Extended model descriptions\n",
    "\n",
    "Tanya Strydom [![](data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAA2ZpVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADw/eHBhY2tldCBiZWdpbj0i77u/IiBpZD0iVzVNME1wQ2VoaUh6cmVTek5UY3prYzlkIj8+IDx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8iIHg6eG1wdGs9IkFkb2JlIFhNUCBDb3JlIDUuMC1jMDYwIDYxLjEzNDc3NywgMjAxMC8wMi8xMi0xNzozMjowMCAgICAgICAgIj4gPHJkZjpSREYgeG1sbnM6cmRmPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5LzAyLzIyLXJkZi1zeW50YXgtbnMjIj4gPHJkZjpEZXNjcmlwdGlvbiByZGY6YWJvdXQ9IiIgeG1sbnM6eG1wTU09Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC9tbS8iIHhtbG5zOnN0UmVmPSJodHRwOi8vbnMuYWRvYmUuY29tL3hhcC8xLjAvc1R5cGUvUmVzb3VyY2VSZWYjIiB4bWxuczp4bXA9Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC8iIHhtcE1NOk9yaWdpbmFsRG9jdW1lbnRJRD0ieG1wLmRpZDo1N0NEMjA4MDI1MjA2ODExOTk0QzkzNTEzRjZEQTg1NyIgeG1wTU06RG9jdW1lbnRJRD0ieG1wLmRpZDozM0NDOEJGNEZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wTU06SW5zdGFuY2VJRD0ieG1wLmlpZDozM0NDOEJGM0ZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wOkNyZWF0b3JUb29sPSJBZG9iZSBQaG90b3Nob3AgQ1M1IE1hY2ludG9zaCI+IDx4bXBNTTpEZXJpdmVkRnJvbSBzdFJlZjppbnN0YW5jZUlEPSJ4bXAuaWlkOkZDN0YxMTc0MDcyMDY4MTE5NUZFRDc5MUM2MUUwNEREIiBzdFJlZjpkb2N1bWVudElEPSJ4bXAuZGlkOjU3Q0QyMDgwMjUyMDY4MTE5OTRDOTM1MTNGNkRBODU3Ii8+IDwvcmRmOkRlc2NyaXB0aW9uPiA8L3JkZjpSREY+IDwveDp4bXBtZXRhPiA8P3hwYWNrZXQgZW5kPSJyIj8+84NovQAAAR1JREFUeNpiZEADy85ZJgCpeCB2QJM6AMQLo4yOL0AWZETSqACk1gOxAQN+cAGIA4EGPQBxmJA0nwdpjjQ8xqArmczw5tMHXAaALDgP1QMxAGqzAAPxQACqh4ER6uf5MBlkm0X4EGayMfMw/Pr7Bd2gRBZogMFBrv01hisv5jLsv9nLAPIOMnjy8RDDyYctyAbFM2EJbRQw+aAWw/LzVgx7b+cwCHKqMhjJFCBLOzAR6+lXX84xnHjYyqAo5IUizkRCwIENQQckGSDGY4TVgAPEaraQr2a4/24bSuoExcJCfAEJihXkWDj3ZAKy9EJGaEo8T0QSxkjSwORsCAuDQCD+QILmD1A9kECEZgxDaEZhICIzGcIyEyOl2RkgwAAhkmC+eAm0TAAAAABJRU5ErkJggg==)](https://orcid.org/0000-0001-6067-1349)  \n",
    "April 26, 2024\n",
    "\n",
    "# Extended model descriptions\n",
    "\n",
    "## Network models\n",
    "\n",
    "### Null models\n",
    "\n",
    "**Type I** ([Fortuna and Bascompte 2006](#ref-fortunaHabitatLossStructure2006)):\n",
    "\n",
    "**Type II** ([Bascompte et al. 2003](#ref-bascompteNestedAssemblyPlantanimal2003)):\n",
    "\n",
    "### Neutral models\n",
    "\n",
    "### Resource models\n",
    "\n",
    "**Cascade model** ([Joel E. Cohen, Briand, and Newman 1990](#ref-cohenCommunityFoodWebs1990)): Much like the name suggests the cascade model rests on the idea that species feed on one another in a hierarchical manner. This rests on the assumption that the links within a network are variably distributed across the network; with the proportion of links decreasing as one moves up the trophic levels (*i.e.,* ‘many’ prey and ‘few’ predators). This is achieved by assigning all species a random rank, this rank will then determine both the predators and prey of that species. A species will have a particular probability of being fed on by any species with a higher ranking than it, this probability is constrained by the specified connectance of the network. Interestingly here ‘species’ are treated as any individual that consume and are consumed by the same ‘species’, *i.e.,* these are not taxonomical species ([Joel E. Cohen, Newman, and Steele 1985](#ref-cohenStochasticTheoryCommunity1985)). The original cascade model has altered to be more ‘generalised’ ([D. B. Stouffer et al. 2005](#ref-stoufferQuantitativePatternsStructure2005)), which altered the probability distribution of the prey that could be consumed by a species.\n",
    "\n",
    "**Niche models** ([Williams and Martinez 2000](#ref-williamsSimpleRulesYield2000)): The niche model introduces the idea that species interactions are based on the ‘feeding niche’ of a species. Broadly, all species are randomly assigned a ‘feeding niche’ range and all species that fall in this range can be consumed by that species (thereby allowing for cannibalism). The niche of each species is randomly assigned and the range of each species’ niche is (in part) constrained by the specified connectance of the network. The niche model has also been modified, although it appears that adding to the ‘complexity’ of the niche model does not improve on its ability to generate a more ecologically ‘correct’ network ([Williams and Martinez 2008](#ref-williamsSuccessItsLimits2008)).\n",
    "\n",
    "**Nested hierarchy model** ([Cattin et al. 2004](#ref-cattinPhylogeneticConstraintsAdaptation2004))\n",
    "\n",
    "## Interaction models\n",
    "\n",
    "### Energetic models\n",
    "\n",
    "**ADBM** ([Petchey et al. 2008](#ref-petcheySizeForagingFood2008)):\n",
    "\n",
    "**DBM** ([Beckerman, Petchey, and Warren 2006](#ref-beckermanForagingBiologyPredicts2006)):\n",
    "\n",
    "### Trait hierarchy\n",
    "\n",
    "**PFIM** ([Shaw et al. 2024](#ref-shawFrameworkReconstructingAncient2024)):\n",
    "\n",
    "### Graph embedding\n",
    "\n",
    "**Transfer learning/RDPG** ([Strydom et al. 2022](#ref-strydomFoodWebReconstruction2022)): The products of the embedding process are fed into a transfer learning framework for novel prediction…\n",
    "\n",
    "**Log-ratio** ([Rohr et al. 2010](#ref-rohrModelingFoodWebs2010)): Interestingly often used in paleo settings (at least that’s what it currently looks like in my mind… Pires et al. ([2020](#ref-piresMegafaunalExtinctionsHuman2020)))\n",
    "\n",
    "### Other\n",
    "\n",
    "**Matching** ([Rossberg et al. 2006](#ref-rossbergFoodWebsExperts2006)): This one is more of a dynamic model (so BEF) and maybe beyond the scope of this work. I think there is value on only focusing on the ‘static’ models at this point (probably have said this before elsewhere but yeah)\n",
    "\n",
    "## Datasets used\n",
    "\n",
    "### Mangal networks\n",
    "\n",
    "We queried the Mangal ([Poisot et al. 2016](#ref-poisotMangalMakingEcological2016)) database and extracted a total of **TODO** networks. \\[*Some sort of summary as to the geographic/taxonomic range??\\]* Although these networks represent a high volume of interaction data they do not have accompanying ‘metadata’ that we would need for some of the more data-hungry model families (*e.g.,* local abundance), the Mangal networks were used to provide the ‘starting values’ for the random, resource, and generative families. This allows us to generate a large number of different networks that we can use to compare and contrast the performance of the various model families. For each network from Mangal we generated **TODO** versions of that network using each model family.\n",
    "\n",
    "> “These complex food webs differ in their level of resolution and sampling effort, which may introduce noise in the estimation of their properties, especially given their large number of interacting elements. However, because our MaxEnt models are applied on imperfect data, they aim at reproducing the sampled structure of food webs, not their actual structure.” - Banville, Gravel, and Poisot ([2023](#ref-banvilleWhatConstrainsFood2023)) (something to think about…)\n",
    "\n",
    "### Empirical networks\n",
    "\n",
    "‘Elite’ number of datasets for interaction models\n",
    "\n",
    "Although the availability of empirical interaction data is growing as techniques begin to improve and grow ([Pringle and Hutchinson 2020](#ref-pringleResolvingFoodWebStructure2020)), we still lack a way to define what is the ‘ideal’ interaction dataset.\n",
    "\n",
    "New Zealand dataset(s): Pomeranz et al. ([2019](#ref-pomeranzInferringPredatorPrey2019))\n",
    "\n",
    "> Here I think we need to span a variety of domains, at minimum aquatic and terrestrial but maybe there should be a ‘scale’ element as well *i.e.,* a regional and local network. I think there is going to be a ‘turning point’ where structural will take over from mechanistic in terms of performance. More specifically at local scales bioenergetic constraints (and co-occurrence) may play a bigger role in structuring a network whereas at the metaweb level then mechanistic may make more (since by default its about who can potentially interact and obviously not constrained by real-world scenarios) *sensu* Caron et al. ([2024](#ref-caronTraitmatchingModelsPredict2024)). Although having said that I feel that contradicts the idea of backbones (*sensu* Bramon Mora (sp?) et al & Stouffer et al) But that might be where we get the idea of core *structure* vs something like linkage density. So core things like trophic level/chain length will be conserved but connectance might not (I think I understand what I’m trying to say here)\n",
    "\n",
    "I think we should also use the Dunne et al. ([2008](#ref-dunneCompilationNetworkAnalyses2008)) work. Because 1) it gives the paleo-centric methods their moment in the sun and 2) I think it also brings up the interesting question of can we use modern structure to predict past ones?\n",
    "\n",
    "## Model benchmarking\n",
    "\n",
    "For now the (still essentially pending) workflow/associated code can be found at the following repository [BecksLab/topology_generators](https://github.com/BecksLab/topology_generators). This will reflect that which is shown in panel *B*.\n",
    "\n",
    "-   Data ‘cost’ (some methods might need a lot lot of supporting data vs something very light weight)\n",
    "-   I think it would be remiss to not also take into consideration computational cost\n",
    "-   Something about the network output - I’m acknowledging my biases and saying that probabilistic (or *maybe* weighted) links are the way\n",
    "\n",
    "### Network models\n",
    "\n",
    "-   connectance, nestedness (Bastolla et al., 2009), modularity (Barber, 2007), asymmetry (Delmas et al., 2018), and Jaccard network dissimilarity (Canard et al., 2014)\n",
    "\n",
    "-   *Shape:* do the models construct tall ‘pencil’ vs flat ‘pancake’ networks (Beckerman 2024, pers comms), generality/vulnerability, chain length (?)\n",
    "\n",
    "-   *Structure:* Predicting ‘structure’ - SVD ([Strydom, Dalla Riva, and Poisot 2021](#ref-strydomSVDEntropyReveals2021)) but maybe something like nestedness as well (?)\n",
    "\n",
    "-   *Links:* are the number of links preserved (most network predicting models are to some extend link constrained but useful to see)\n",
    "\n",
    "-   *Motifs:* Staniczenko et al. ([2010](#ref-staniczenkoStructuralDynamicsRobustness2010)) uses S1, S2, S4, S5 from Daniel B. Stouffer et al. ([2007](#ref-stoufferEvidenceExistenceRobust2007))\n",
    "\n",
    "    -   S1: Number of linear chains\n",
    "\n",
    "    -   S2: Number of omnivory motifs\n",
    "\n",
    "    -   S4: Number of apparent competition motifs\n",
    "\n",
    "    -   S5: Number of direct competition motifs\n",
    "\n",
    "### Interaction models\n",
    "\n",
    "-   Based on Poisot ([2023](#ref-poisotGuidelinesPredictionSpecies2023)):\n",
    "    -   Precision-Recall (PR-AUC) - performance\n",
    "    -   Matthews correlation coefficient (MCC) - accuracy\n",
    "-   Maybe same measures we use for the network models\n",
    "\n",
    "### Action plan\n",
    "\n",
    "1.  Shortlist/finalise the different topo generators\n",
    "2.  collate/translate into `Julia`\n",
    "    -   *e.g.,* some models wil be in SpeciesInteractionNetworks.jl (new EcoNet); I know (parts of) the transfer learning stuff is and the niche model\n",
    "    -   others will need to be coded out (the more simpler models should be easier)\n",
    "3.  Curate networks for the different datasets/scenarios we select - I feel like there might be some scenarios that we can’t do all models for all datasets but maybe I’m being a pessimist.\n",
    "    -   Need to also think about where one might find the additional data for some of the models…\n",
    "        -   Body size: Herberstein et al. ([2022](#ref-herbersteinAnimalTraitsCuratedAnimal2022)) - Although maybe Andrew has strong thotsTM RE the one true body size database to rule them all…\n",
    "        -   Other trait sources: Wilman et al. ([2014](#ref-wilmanEltonTraitsSpecieslevelForaging2014)) and Jones et al. ([2009](#ref-jonesPanTHERIASpecieslevelDatabase2009))\n",
    "        -   This is where we’ll get the paleo traits from if I’m correct Bambach, Bush, and Erwin ([2007](#ref-bambachAutecologyFillingEcospace2007))\n",
    "        -   Phylogeny stuff: Upham, Esselstyn, and Jetz ([2019](#ref-uphamInferringMammalTree2019)) (what we used for TL but its only mammals…) but I’m sure there will be others\n",
    "    -   Also limitation of scope… *e.g.,* do we even dare to think about including plants/basal producers (see *e.g.,* Valdovinos et al. ([2023](#ref-valdovinosBioenergeticFrameworkAboveground2023)))\n",
    "    -   Taxonomic harmonisation - something to think about and check\n",
    "\n",
    "## References\n",
    "\n",
    "Bambach, Richard K., Andrew M. Bush, and Douglas H. Erwin. 2007. “Autecology and the Filling of Ecospace: Key Metazoan Radiations.” *Palaeontology* 50 (1): 1–22. <https://doi.org/10.1111/j.1475-4983.2006.00611.x>.\n",
    "\n",
    "Banville, Francis, Dominique Gravel, and Timothée Poisot. 2023. “What Constrains Food Webs? A Maximum Entropy Framework for Predicting Their Structure with Minimal Biases.” *PLOS Computational Biology* 19 (9): e1011458. <https://doi.org/10.1371/journal.pcbi.1011458>.\n",
    "\n",
    "Bascompte, J., P. Jordano, C. J. Melian, and J. M. Olesen. 2003. “The Nested Assembly of Plant-Animal Mutualistic Networks.” *Proceedings of the National Academy of Sciences* 100 (16): 9383–87. <https://doi.org/10.1073/pnas.1633576100>.\n",
    "\n",
    "Beckerman, Andrew P., Owen L. Petchey, and Philip H. Warren. 2006. “Foraging Biology Predicts Food Web Complexity.” *Proceedings of the National Academy of Sciences* 103 (37): 13745–49. <https://doi.org/10.1073/pnas.0603039103>.\n",
    "\n",
    "Caron, Dominique, Ulrich Brose, Miguel Lurgi, F. Guillaume Blanchet, Dominique Gravel, and Laura J. Pollock. 2024. “Trait-Matching Models Predict Pairwise Interactions Across Regions, Not Food Web Properties.” *Global Ecology and Biogeography* 33 (4): e13807. <https://doi.org/10.1111/geb.13807>.\n",
    "\n",
    "Cattin, Marie-France, Louis-Félix Bersier, Carolin Banašek-Richter, Richard Baltensperger, and Jean-Pierre Gabriel. 2004. “Phylogenetic Constraints and Adaptation Explain Food-Web Structure.” *Nature* 427 (6977): 835–39. <https://doi.org/10.1038/nature02327>.\n",
    "\n",
    "Cohen, Joel E, Frederic Briand, and Charles Newman. 1990. *Community Food Webs: Data and Theory*. Biomathematics. Berlin Heidelberg: Springer-Verlag.\n",
    "\n",
    "Cohen, Joel E., C. M. Newman, and John Hyslop Steele. 1985. “A Stochastic Theory of Community Food Webs I. Models and Aggregated Data.” *Proceedings of the Royal Society of London. Series B. Biological Sciences* 224 (1237): 421–48. <https://doi.org/10.1098/rspb.1985.0042>.\n",
    "\n",
    "Dunne, Jennifer A., Richard J. Williams, Neo D. Martinez, Rachel A. Wood, and Douglas H. Erwin. 2008. “Compilation and Network Analyses of Cambrian Food Webs.” *PLOS Biology* 6 (4): e102. <https://doi.org/10.1371/journal.pbio.0060102>.\n",
    "\n",
    "Fortuna, Miguel A., and Jordi Bascompte. 2006. “Habitat Loss and the Structure of Plant-Animal Mutualistic Networks: Mutualistic Networks and Habitat Loss.” *Ecology Letters* 9 (3): 281–86. <https://doi.org/10.1111/j.1461-0248.2005.00868.x>.\n",
    "\n",
    "Herberstein, Marie E., Donald James McLean, Elizabeth Lowe, Jonas O. Wolff, Md Kawsar Khan, Kaitlyn Smith, Andrew P. Allen, et al. 2022. “AnimalTraits - a Curated Animal Trait Database for Body Mass, Metabolic Rate and Brain Size.” *Scientific Data* 9 (1): 265. <https://doi.org/10.1038/s41597-022-01364-9>.\n",
    "\n",
    "Jones, Kate E., Jon Bielby, Marcel Cardillo, Susanne A. Fritz, Justin O’Dell, C. David L. Orme, Kamran Safi, et al. 2009. “PanTHERIA: A Species-Level Database of Life History, Ecology, and Geography of Extant and Recently Extinct Mammals.” *Ecology* 90 (9): 2648–48. <https://doi.org/10.1890/08-1494.1>.\n",
    "\n",
    "Petchey, Owen L., Andrew P. Beckerman, Jens O. Riede, and Philip H. Warren. 2008. “Size, Foraging, and Food Web Structure.” *Proceedings of the National Academy of Sciences* 105 (11): 4191–96. <https://doi.org/10.1073/pnas.0710672105>.\n",
    "\n",
    "Pires, Mathias M., Diego Rindel, Bruno Moscardi, Livia R. Cruz, Paulo R. Guimarães, Sergio F. dos Reis, and S. Ivan Perez. 2020. “Before, During and After Megafaunal Extinctions: Human Impact on Pleistocene-Holocene Trophic Networks in South Patagonia.” *Quaternary Science Reviews* 250 (December): 106696. <https://doi.org/10.1016/j.quascirev.2020.106696>.\n",
    "\n",
    "Poisot, Timothée. 2023. “Guidelines for the Prediction of Species Interactions Through Binary Classification.” *Methods in Ecology and Evolution* 14 (5): 1333–45. <https://doi.org/10.1111/2041-210X.14071>.\n",
    "\n",
    "Poisot, Timothée, Benjamin Baiser, Jennifer Dunne, Sonia Kéfi, François Massol, Nicolas Mouquet, Tamara N. Romanuk, Daniel B. Stouffer, Spencer A. Wood, and Dominique Gravel. 2016. “Mangal – Making Ecological Network Analysis Simple.” *Ecography* 39 (4): 384–90. <https://doi.org/10.1111/ecog.00976>.\n",
    "\n",
    "Pomeranz, Justin P. F., Ross M. Thompson, Timothée Poisot, and Jon S. Harding. 2019. “Inferring Predator–Prey Interactions in Food Webs.” *Methods in Ecology and Evolution* 10 (3): 356–67. <https://doi.org/10.1111/2041-210X.13125>.\n",
    "\n",
    "Pringle, Robert M., and Matthew C. Hutchinson. 2020. “Resolving Food-Web Structure.” *Annual Review of Ecology, Evolution and Systematics* 51 (Volume 51, 2020): 55–80. <https://doi.org/10.1146/annurev-ecolsys-110218-024908>.\n",
    "\n",
    "Rohr, Rudolf Philippe, Heike Scherer, Patrik Kehrli, Christian Mazza, and Louis-Félix Bersier. 2010. “Modeling Food Webs: Exploring Unexplained Structure Using Latent Traits.” *The American Naturalist* 176 (2): 170–77. <https://doi.org/10.1086/653667>.\n",
    "\n",
    "Rossberg, A. G., H. Matsuda, T. Amemiya, and K. Itoh. 2006. “Food Webs: Experts Consuming Families of Experts.” *Journal of Theoretical Biology* 241 (3): 552–63. <https://doi.org/10.1016/j.jtbi.2005.12.021>.\n",
    "\n",
    "Shaw, Jack O., Alexander M. Dunhill, Andrew P. Beckerman, Jennifer A. Dunne, and Pincelli M. Hull. 2024. “A Framework for Reconstructing Ancient Food Webs Using Functional Trait Data.” bioRxiv. <https://doi.org/10.1101/2024.01.30.578036>.\n",
    "\n",
    "Staniczenko, Phillip P. A., Owen T. Lewis, Nick S. Jones, and Felix Reed-Tsochas. 2010. “Structural Dynamics and Robustness of Food Webs.” *Ecology Letters* 13 (7): 891–99. <https://doi.org/10.1111/j.1461-0248.2010.01485.x>.\n",
    "\n",
    "Stouffer, D. B., J. Camacho, R. Guimerà, C. A. Ng, and L. A. Nunes Amaral. 2005. “Quantitative Patterns in the Structure of Model and Empirical Food Webs.” *Ecology* 86 (5): 1301–11. <https://doi.org/10.1890/04-0957>.\n",
    "\n",
    "Stouffer, Daniel B, Juan Camacho, Wenxin Jiang, and Luís A Nunes Amaral. 2007. “Evidence for the Existence of a Robust Pattern of Prey Selection in Food Webs.” *Proceedings of the Royal Society B: Biological Sciences* 274 (1621): 1931–40. <https://doi.org/10.1098/rspb.2007.0571>.\n",
    "\n",
    "Strydom, Tanya, Salomé Bouskila, Francis Banville, Ceres Barros, Dominique Caron, Maxwell J. Farrell, Marie-Josée Fortin, et al. 2022. “Food Web Reconstruction Through Phylogenetic Transfer of Low-Rank Network Representation.” *Methods in Ecology and Evolution* 13 (12): 2838–49. <https://doi.org/10.1111/2041-210X.13835>.\n",
    "\n",
    "Strydom, Tanya, Giulio V. Dalla Riva, and Timothée Poisot. 2021. “SVD Entropy Reveals the High Complexity of Ecological Networks.” *Frontiers in Ecology and Evolution* 9. <https://doi.org/10.3389/fevo.2021.623141>.\n",
    "\n",
    "Upham, Nathan S., Jacob A. Esselstyn, and Walter Jetz. 2019. “Inferring the Mammal Tree: <span class=\"nocase\">Species-level</span> Sets of Phylogenies for Questions in Ecology, Evolution, and Conservation.” *PLOS Biology* 17 (12): e3000494. <https://doi.org/10.1371/journal.pbio.3000494>.\n",
    "\n",
    "Valdovinos, Fernanda S., Kayla R. S. Hale, Sabine Dritz, Paul R. Glaum, Kevin S. McCann, Sophia M. Simon, Elisa Thébault, William C. Wetzel, Kate L. Wootton, and Justin D. Yeakel. 2023. “A Bioenergetic Framework for Aboveground Terrestrial Food Webs.” *Trends in Ecology & Evolution* 38 (3): 301–12. <https://doi.org/10.1016/j.tree.2022.11.004>.\n",
    "\n",
    "Williams, Richard J., and Neo D. Martinez. 2000. “Simple Rules Yield Complex Food Webs.” *Nature* 404 (6774): 180–83. <https://doi.org/10.1038/35004572>.\n",
    "\n",
    "———. 2008. “Success and Its Limits Among Structural Models of Complex Food Webs.” *Journal of Animal Ecology* 77 (3): 512–19. <https://doi.org/10.1111/j.1365-2656.2008.01362.x>.\n",
    "\n",
    "Wilman, Hamish, Jonathan Belmaker, Jennifer Simpson, Carolina de la Rosa, Marcelo M. Rivadeneira, and Walter Jetz. 2014. “EltonTraits 1.0: <span class=\"nocase\">Species-level</span> Foraging Attributes of the World’s Birds and Mammals.” *Ecology* 95 (7): 2027–27. <https://doi.org/10.1890/13-1917.1>.\n",
    "\n",
    "Yeakel, Justin D., Mathias M. Pires, Lars Rudolf, Nathaniel J. Dominy, Paul L. Koch, Paulo R. Guimarães, and Thilo Gross. 2014. “Collapse of an Ecological Network in Ancient Egypt.” *PNAS* 111 (40): 14472–77. <https://doi.org/10.1073/pnas.1408471111>."
   ],
   "id": "4f13a364-14f1-49f7-b03f-4b3fbfd5af31"
  }
 ],
 "nbformat": 4,
 "nbformat_minor": 5,
 "metadata": {}
}
