<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.4.549">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Tanya Strydom">
<meta name="author" content="Jennifer A. Dunne">
<meta name="author" content="Timothée Poisot">
<meta name="author" content="Andrew P. Beckerman">
<meta name="dcterms.date" content="2024-09-27">
<meta name="keywords" content="food web, network construction, scientific ignorance">

<title>Unveiling the Complexity of Food Webs: A Comprehensive Overview of Definitions, Scales, and Mechanisms</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for citations */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
  margin-bottom: 0em;
}
.hanging-indent div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}</style>


<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script src="site_libs/quarto-contrib/glightbox/glightbox.min.js"></script>
<link href="site_libs/quarto-contrib/glightbox/glightbox.min.css" rel="stylesheet">
<link href="site_libs/quarto-contrib/glightbox/lightbox.css" rel="stylesheet">
<script async="" src="https://hypothes.is/embed.js"></script>
<script>
  window.document.addEventListener("DOMContentLoaded", function (_event) {
    document.body.classList.add('hypothesis-enabled');
  });
</script>

  <script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>
  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

<script type="text/javascript">
const typesetMath = (el) => {
  if (window.MathJax) {
    // MathJax Typeset
    window.MathJax.typeset([el]);
  } else if (window.katex) {
    // KaTeX Render
    var mathElements = el.getElementsByClassName("math");
    var macros = [];
    for (var i = 0; i < mathElements.length; i++) {
      var texText = mathElements[i].firstChild;
      if (mathElements[i].tagName == "SPAN") {
        window.katex.render(texText.data, mathElements[i], {
          displayMode: mathElements[i].classList.contains('display'),
          throwOnError: false,
          macros: macros,
          fleqn: false
        });
      }
    }
  }
}
window.Quarto = {
  typesetMath
};
</script>

<meta name="citation_title" content="Unveiling the Complexity of Food Webs: A Comprehensive Overview of Definitions, Scales, and Mechanisms">
<meta name="citation_abstract" content="Food webs are a useful abstraction and representation of the feeding links between species in a community and are used to infer many ecosystem level processes. However, the different theories, mechanisms, and criteria that underpin how a food web is defined and, ultimately, constructed means that not all food webs are representing the same ecological process. Here we present a synthesis of the different assumptions, scales and mechanisms that are used to define different ecological networks ranging from metawebs (an inventory of all potential interactions) to fully realised networks (interactions that occur within a given community over a certain timescale). Illuminating the assumptions, scales, and mechanisms of network inference allows a formal categorisation of how to use networks to answer key ecological and conservation questions and defines guidelines to prevent unintentional misuse or misinterpretation.
">
<meta name="citation_keywords" content="food web,network construction,scientific ignorance">
<meta name="citation_author" content="Tanya Strydom">
<meta name="citation_author" content="Jennifer A. Dunne">
<meta name="citation_author" content="Timothée Poisot">
<meta name="citation_author" content="Andrew P. Beckerman">
<meta name="citation_publication_date" content="2024-09-27">
<meta name="citation_cover_date" content="2024-09-27">
<meta name="citation_year" content="2024">
<meta name="citation_online_date" content="2024-09-27">
<meta name="citation_language" content="en">
<meta name="citation_journal_title" content="TREE (one can dream...)">
<meta name="citation_reference" content="citation_title=A novel method for predicting ecological interactions with an unsupervised machine learning algorithm;,citation_abstract=This gap in knowledge regarding ecological interactions has prompted the development of various predictive approaches. Traditionally, ecological interactions have been inferred using traits. However, the lack of trait information for numerous organisms necessitates using phylogenetic data and statistical insights from interaction matrices for prediction. Previous studies have overlooked the validation of model-predicted interactions. This study used a novel method in predicting ecological interactions using a self-organizing map (SOM), an unsupervised machine learning algorithm. The SOM learns from the input interaction matrix by grouping the nodes into output layers based on their interactions. Subsequently, the trained model predicts the interactions as scores. To distinguish between interactions and non-interactions, we employed F1 score maximization, setting scores above a specific threshold as interactions and the remainder as non-interactions. We applied this method to three unipartite metawebs and one bipartite metaweb and subsequently validated the predicted interactions using two innovative approaches: taxonomic and interaction recovery validation. Our method exhibited outstanding predictive performance, particularly for large networks. Various binary classification performance indicators, including F1 score (0.84–0.97) and accuracy (0.97–0.99), indicated high performance. Moreover, the method generated minimal predicted interactions, signifying low noise in the predictions, particularly for large networks. Taxonomic validation excels in metawebs with a connectance &amp;amp;amp;gt;0.1 but performs poorly in metawebs with very low connectance. In contrast, interaction recovery was most effective in larger metawebs. Our proposed method excels at making highly accurate predictions of ecological interactions with minimal noise, solely utilizing input interaction data without relying on traits or phylogenetic information regarding interacting nodes. These predictions are particularly precise for large networks, underscoring their potential to address knowledge gaps in emerging extensive metawebs. Notably, taxonomic validation and interaction recovery methods are sensitive to connectance and network size, respectively, suggesting prospects for developing robust interaction validation methods.;,citation_author=Sagar Adhurya;,citation_author=Young-Seuk Park;,citation_publication_date=2024;,citation_cover_date=2024;,citation_year=2024;,citation_issue=n/a;,citation_doi=10.1111/2041-210X.14358;,citation_issn=2041-210X;,citation_volume=n/a;,citation_language=en-US;,citation_journal_title=Methods in Ecology and Evolution;">
<meta name="citation_reference" content="citation_title=Food web models: A plea for groups;,citation_abstract=The concept of a group is ubiquitous in biology. It underlies classifications in evolution and ecology, including those used to describe phylogenetic levels, the habitat and functional roles of organisms in ecosystems. Surprisingly, this concept is not explicitly included in simple models for the structure of food webs, the ecological networks formed by consumer–resource interactions. We present here the simplest possible model based on groups, and show that it performs substantially better than current models at predicting the structure of large food webs. Our group-based model can be applied to different types of biological and non-biological networks, and for the first time merges in the same framework two important notions in network theory: that of compartments (sets of highly interacting nodes) and that of roles (sets of nodes that have similar interaction patterns). This model provides a basis to examine the significance of groups in biological networks and to develop more accurate models for ecological network structure. It is especially relevant at a time when a new generation of empirical data is providing increasingly large food webs.;,citation_author=Stefano Allesina;,citation_author=Mercedes Pascual;,citation_publication_date=2009;,citation_cover_date=2009;,citation_year=2009;,citation_issue=7;,citation_doi=10.1111/j.1461-0248.2009.01321.x;,citation_issn=1461-0248;,citation_volume=12;,citation_language=en-US;,citation_journal_title=Ecology Letters;">
<meta name="citation_reference" content="citation_title=A General Model for Food Web Structure;,citation_abstract=A central problem in ecology is determining the processes that shape the complex networks known as food webs formed by species and their feeding relationships. The topology of these networks is a major determinant of ecosystems’ dynamics and is ultimately responsible for their responses to human impacts. Several simple models have been proposed for the intricate food webs observed in nature. We show that the three main models proposed so far fail to fully replicate the empirical data, and we develop a likelihood-based approach for the direct comparison of alternative models based on the full structure of the network. Results drive a new model that is able to generate all the empirical data sets and to do so with the highest likelihood.;,citation_author=Stefano Allesina;,citation_author=David Alonso;,citation_author=Mercedes Pascual;,citation_publication_date=2008-05;,citation_cover_date=2008-05;,citation_year=2008;,citation_issue=5876;,citation_doi=10.1126/science.1156269;,citation_volume=320;,citation_journal_title=Science;,citation_publisher=American Association for the Advancement of Science;">
<meta name="citation_reference" content="citation_title=Autecology and the Filling of Ecospace: Key Metazoan Radiations;,citation_abstract=Abstract: All possible combinations of six tiering positions in relation to the substratum/water interface, six motility levels and six feeding strategies define a complete theoretical ecospace of 216 potential modes of life for marine animals. The number of modes of life actually utilized specifies realized ecospace. Owing to constraints of effectiveness and efficiency the modern marine fauna utilizes only about half the potential number of modes of life, two-thirds of which (62 of 92) are utilized by animals with readily preserved, mineralized hard parts. Realized ecospace has increased markedly since the early evolution of animal ecosystems. The Ediacaran fauna utilized at most 12 modes of life, with just two practised by skeletal organisms. A total of 30 modes of life are recorded in the Early and Middle Cambrian, 19 of which were utilized by skeletal organisms. The other 11 are documented from soft-bodied animals preserved in the Chengjiang and Burgess Shale Konservat-Lagerstätten. The number of modes of life utilized by skeletal organisms increased by more than 50 per cent during the Ordovician radiation to a Late Ordovician total of 30. Between the Late Ordovician and the Recent the number of utilized modes of life has doubled again. The autecological and taxonomic diversity histories of the marine metazoa appear to be broadly parallel, and future studies of theoretical ecospace utilization should provide more detailed tests of pattern and process in the ecological history of the metazoa.;,citation_author=Richard K. Bambach;,citation_author=Andrew M. Bush;,citation_author=Douglas H. Erwin;,citation_publication_date=2007;,citation_cover_date=2007;,citation_year=2007;,citation_issue=1;,citation_doi=10.1111/j.1475-4983.2006.00611.x;,citation_issn=1475-4983;,citation_volume=50;,citation_language=en-US;,citation_journal_title=Palaeontology;">
<meta name="citation_reference" content="citation_title=Deciphering probabilistic species interaction networks;,citation_abstract=Representing species interactions probabilistically (how likely are they to occur?) as opposed to deterministically (are they occurring?) conveys uncertainties in our knowledge of interactions and information on their variability. The sources of uncertainty captured by interaction probabilities depend on the method used to evaluate them: uncertainty of predictive models, subjective assessment of experts, or empirical measurement of interaction spatiotemporal variability. However, guidelines for the estimation and documentation of probabilistic interaction data are lacking. This is concerning because our understanding and analysis of interaction probabilities depend on their sometimes elusive definition and uncertainty sources. We review how probabilistic interactions are defined at different spatial scales, from local interactions to regional networks (metawebs), with a strong emphasis on host-parasite and trophic (predatory and herbivory) interactions. These definitions are based on the distinction between the realization of an interaction at a specific time and space (local) and its biological or ecological feasibility (regional). Using host-parasite interactions in Europe, we illustrate how these two network representations differ in their statistical properties, specifically: how local networks and metawebs differ in their spatial and temporal scaling of probabilistic interactions, but not in their taxonomic scaling. We present two approaches to inferring binary interactions from probabilistic ones that account for these differences and show that systematic biases arise when directly inferring local networks from metawebs. Our results underscore the importance of more rigorous descriptions of probabilistic species interaction networks that specify their type of interaction (local or regional), conditional variables and uncertainty sources.;,citation_author=Francis Banville;,citation_author=Tanya Strydom;,citation_author=Penelope Blyth;,citation_author=Chris Brimacombe;,citation_author=Michael D. Catchen;,citation_author=Gabriel Dansereau;,citation_author=Gracielle Higino;,citation_author=Thomas Malpas;,citation_author=Hana Mayall;,citation_author=Kari Norman;,citation_author=Dominique Gravel;,citation_author=Timothée Poisot;,citation_publication_date=2024-07;,citation_cover_date=2024-07;,citation_year=2024;,citation_doi=10.32942/X28G8Z;,citation_language=en-US;,citation_publisher=EcoEvoRxiv;">
<meta name="citation_reference" content="citation_title=Mangal.jl and EcologicalNetworks.jl: Two complementary packages for analyzing ecological networks in Julia;,citation_abstract=Network ecology is an emerging field of study describing species interactions (e.g. predation, pollination) in a biological community. Ecological networks, in which two species are connected if they can interact, are a mathematical representation of all interactions encountered in a given ecosystem. Anchored in graph theory, the methods developed in network ecology are remarkably rigorous and biologically insightful. Indeed, many ecological and evolutionary processes are driven by species interactions and network structure (i.e. the arrangement of links in ecological networks). The study of ecological networks, from data importation and simulation to data analysis and visualization, requires a coherent and efficient set of numerical tools. With its powerful and dynamic type system, the Julia programming language provides a very good and fast computing environment suitable for ecological research. Julia’s typing system notably ensures that appropriate methods are used when a function is applied to different types of ecological data, which therefore minimizes the risk of committing errors of analysis and interpretation in network ecology. Mangal.jl and EcologicalNetworks.jl are two novel and complementary Julia packages designed to conduct research in network ecology efficiently. Researchers studying a broad range of ecological networks can read the data and metadata of numerous well-documented networks using Mangal.jl, and analyze their properties using EcologicalNetworks.jl.;,citation_author=Francis Banville;,citation_author=Steve Vissault;,citation_author=Timothée Poisot;,citation_publication_date=2021-05;,citation_cover_date=2021-05;,citation_year=2021;,citation_issue=61;,citation_doi=10.21105/joss.02721;,citation_issn=2475-9066;,citation_volume=6;,citation_language=en-US;,citation_journal_title=Journal of Open Source Software;">
<meta name="citation_reference" content="citation_title=What constrains food webs? A maximum entropy framework for predicting their structure with minimal biases;,citation_abstract=Food webs are complex ecological networks whose structure is both ecologically and statistically constrained, with many network properties being correlated with each other. Despite the recognition of these invariable relationships in food webs, the use of the principle of maximum entropy (MaxEnt) in network ecology is still rare. This is surprising considering that MaxEnt is a statistical tool precisely designed for understanding and predicting many types of constrained systems. This principle asserts that the least-biased probability distribution of a system’s property, constrained by prior knowledge about that system, is the one with maximum information entropy. MaxEnt has been proven useful in many ecological modeling problems, but its application in food webs and other ecological networks is limited. Here we show how MaxEnt can be used to derive many food-web properties both analytically and heuristically. First, we show how the joint degree distribution (the joint probability distribution of the numbers of prey and predators for each species in the network) can be derived analytically using the number of species and the number of interactions in food webs. Second, we present a heuristic and flexible approach of finding a network’s adjacency matrix (the network’s representation in matrix format) based on simulated annealing and SVD entropy. We built two heuristic models using the connectance and the joint degree sequence as statistical constraints, respectively. We compared both models’ predictions against corresponding null and neutral models commonly used in network ecology using open access data of terrestrial and aquatic food webs sampled globally (N = 257). We found that the heuristic model constrained by the joint degree sequence was a good predictor of many measures of food-web structure, especially the nestedness and motifs distribution. Specifically, our results suggest that the structure of terrestrial and aquatic food webs is mainly driven by their joint degree distribution.;,citation_author=Francis Banville;,citation_author=Dominique Gravel;,citation_author=Timothée Poisot;,citation_publication_date=2023-09;,citation_cover_date=2023-09;,citation_year=2023;,citation_issue=9;,citation_doi=10.1371/journal.pcbi.1011458;,citation_issn=1553-7358;,citation_volume=19;,citation_language=en-US;,citation_journal_title=PLOS Computational Biology;,citation_publisher=Public Library of Science;">
<meta name="citation_reference" content="citation_title=Using network analysis to explore co-occurrence patterns in soil microbial communities;,citation_abstract=Exploring large environmental datasets generated by high-throughput DNA sequencing technologies requires new analytical approaches to move beyond the basic inventory descriptions of the composition and diversity of natural microbial communities. In order to investigate potential interactions between microbial taxa, network analysis of significant taxon co-occurrence patterns may help to decipher the structure of complex microbial communities across spatial or temporal gradients. Here, we calculated associations between microbial taxa and applied network analysis approaches to a 16S rRNA gene barcoded pyrosequencing dataset containing &amp;amp;amp;gt;160 000 bacterial and archaeal sequences from 151 soil samples from a broad range of ecosystem types. We described the topology of the resulting network and defined operational taxonomic unit categories based on abundance and occupancy (that is, habitat generalists and habitat specialists). Co-occurrence patterns were readily revealed, including general non-random association, common life history strategies at broad taxonomic levels and unexpected relationships between community members. Overall, we demonstrated the potential of exploring inter-taxa correlations to gain a more integrated understanding of microbial community structure and the ecological rules guiding community assembly.;,citation_author=Albert Barberán;,citation_author=Scott T. Bates;,citation_author=Emilio O. Casamayor;,citation_author=Noah Fierer;,citation_publication_date=2012-02;,citation_cover_date=2012-02;,citation_year=2012;,citation_issue=2;,citation_doi=10.1038/ismej.2011.119;,citation_issn=1751-7370;,citation_volume=6;,citation_language=en-US;,citation_journal_title=The ISME Journal;,citation_publisher=Nature Publishing Group;">
<meta name="citation_reference" content="citation_title=A common framework for identifying linkage rules across different types of interactions;,citation_abstract=Species interactions, ranging from antagonisms to mutualisms, form the architecture of biodiversity and determine ecosystem functioning. Understanding the rules responsible for who interacts with whom, as well as the functional consequences of these interspecific interactions, is central to predict community dynamics and stability. Species traits sensu lato may affect different ecological processes by determining species interactions through a two-step process. First, ecological and life-history traits govern species distributions and abundance, and hence determine species co-occurrence and the potential for species to interact. Secondly, morphological or physiological traits between co-occurring potential interaction partners should match for the realization of an interaction. Here, we review recent advances on predicting interactions from species co-occurrence and develop a probabilistic model for inferring trait matching. The models proposed here integrate both neutral and trait-matching constraints, while using only information about known interactions, thereby overcoming problems originating from undersampling of rare interactions (i.e. missing links). They can easily accommodate qualitative or quantitative data and can incorporate trait variation within species, such as values that vary along developmental stages or environmental gradients. We use three case studies to show that the proposed models can detect strong trait matching (e.g. predator–prey system), relaxed trait matching (e.g. herbivore–plant system) and barrier trait matching (e.g. plant–pollinator systems). Only by elucidating which species traits are important in each process (i.e. in determining interaction establishment and frequency), we can advance in explaining how species interact and the consequences of these interactions for ecosystem functioning. A lay summary is available for this article.;,citation_author=Ignasi Bartomeus;,citation_author=Dominique Gravel;,citation_author=Jason M. Tylianakis;,citation_author=Marcelo A. Aizen;,citation_author=Ian A. Dickie;,citation_author=Maud Bernard-Verdier;,citation_publication_date=2016;,citation_cover_date=2016;,citation_year=2016;,citation_issue=12;,citation_doi=10.1111/1365-2435.12666;,citation_issn=1365-2435;,citation_volume=30;,citation_language=en-US;,citation_journal_title=Functional Ecology;">
<meta name="citation_reference" content="citation_title=The nested assembly of plant-animal mutualistic networks;,citation_abstract=Most studies of plant–animal mutualisms involve a small number of species. There is almost no information on the structural organization of species-rich mutualistic networks despite its potential importance for the maintenance of diversity. Here we analyze 52 mutualistic networks and show that they are highly nested; that is, the more specialist species interact only with proper subsets of those species interacting with the more generalists. This assembly pattern generates highly asymmetrical interactions and organizes the community cohesively around a central core of interactions. Thus, mutualistic networks are neither randomly assembled nor organized in compartments arising from tight, parallel specialization. Furthermore, nestedness increases with the complexity (number of interactions) of the network: for a given number of species, communities with more interactions are significantly more nested. Our results indicate a nonrandom pattern of community organization that may be relevant for our understanding of the organization and persistence of biodiversity.;,citation_author=J. Bascompte;,citation_author=P. Jordano;,citation_author=C. J. Melian;,citation_author=J. M. Olesen;,citation_publication_date=2003-08;,citation_cover_date=2003-08;,citation_year=2003;,citation_issue=16;,citation_doi=10.1073/pnas.1633576100;,citation_issn=0027-8424, 1091-6490;,citation_volume=100;,citation_language=en-US;,citation_journal_title=Proceedings of the National Academy of Sciences;">
<meta name="citation_reference" content="citation_title=Foraging biology predicts food web complexity;,citation_abstract=Food webs, the networks of feeding links between species, are central to our understanding of ecosystem structure, stability, and function. One of the key aspects of food web structure is complexity, or connectance, the number of links expressed as a proportion of the total possible number of links. Connectance (complexity) is linked to the stability of webs and is a key parameter in recent models of other aspects of web structure. However, there is still no fundamental biological explanation for connectance in food webs. Here, we propose that constraints on diet breadth, driven by optimal foraging, provide such an explanation. We show that a simple diet breadth model predicts highly constrained values of connectance as an emergent consequence of individual foraging behavior. When combined with features of real food web data, such as taxonomic and trophic aggregation and cumulative sampling of diets, the model predicts well the levels of connectance and scaling of connectance with species richness, seen in real food webs. This result is a previously undescribed synthesis of foraging theory and food web theory, in which network properties emerge from the behavior of individuals and, as such, provides a mechanistic explanation of connectance currently lacking in food web models.;,citation_author=Andrew P. Beckerman;,citation_author=Owen L. Petchey;,citation_author=Philip H. Warren;,citation_publication_date=2006-09;,citation_cover_date=2006-09;,citation_year=2006;,citation_issue=37;,citation_doi=10.1073/pnas.0603039103;,citation_issn=0027-8424, 1091-6490;,citation_volume=103;,citation_language=en-US;,citation_journal_title=Proceedings of the National Academy of Sciences;">
<meta name="citation_reference" content="citation_title=Optimising predictive models to prioritise viral discovery in zoonotic reservoirs;,citation_author=Daniel J. Becker;,citation_author=Gregory F. Albery;,citation_author=Anna R. Sjodin;,citation_author=Timothée Poisot;,citation_author=Laura M. Bergner;,citation_author=Binqi Chen;,citation_author=Lily E. Cohen;,citation_author=Tad A. Dallas;,citation_author=Evan A. Eskew;,citation_author=Anna C. Fagre;,citation_author=Maxwell J. Farrell;,citation_author=Sarah Guth;,citation_author=Barbara A. Han;,citation_author=Nancy B. Simmons;,citation_author=Michiel Stock;,citation_author=Emma C. Teeling;,citation_author=Colin J. Carlson;,citation_publication_date=2022-08;,citation_cover_date=2022-08;,citation_year=2022;,citation_issue=8;,citation_doi=10.1016/S2666-5247(21)00245-7;,citation_issn=2666-5247;,citation_pmid=35036970;,citation_volume=3;,citation_language=en-US;,citation_journal_title=The Lancet Microbe;,citation_publisher=Elsevier;">
<meta name="citation_reference" content="citation_title=The “Goldilocks factor” in food webs;,citation_author=Eric L. Berlow;,citation_author=Ulrich Brose;,citation_author=Neo D. Martinez;,citation_publication_date=2008-03;,citation_cover_date=2008-03;,citation_year=2008;,citation_issue=11;,citation_doi=10.1073/pnas.0800967105;,citation_volume=105;,citation_journal_title=Proceedings of the National Academy of Sciences;,citation_publisher=Proceedings of the National Academy of Sciences;">
<meta name="citation_reference" content="citation_title=Interaction strengths in food webs: Issues and opportunities;,citation_author=Eric L. Berlow;,citation_author=Anje-Margiet Neutel;,citation_author=Joel E. Cohen;,citation_author=Peter C. Ruiter;,citation_author=Bo Ebenman;,citation_author=Mark Emmerson;,citation_author=Jeremy W. Fox;,citation_author=Vincent A. A. Jansen;,citation_author=J. Iwan Jones;,citation_author=Giorgos D. Kokkoris;,citation_author=Dmitrii O. Logofet;,citation_author=Alan J. McKane;,citation_author=Jose M. Montoya;,citation_author=Owen Petchey;,citation_publication_date=2004-05;,citation_cover_date=2004-05;,citation_year=2004;,citation_issue=3;,citation_doi=10.1111/j.0021-8790.2004.00833.x;,citation_issn=0021-8790, 1365-2656;,citation_volume=73;,citation_language=en-US;,citation_journal_title=Journal of Animal Ecology;">
<meta name="citation_reference" content="citation_title=Network-based restoration strategies maximize ecosystem recovery;,citation_abstract=Redressing global patterns of biodiversity loss requires quantitative frameworks that can predict ecosystem collapse and inform restoration strategies. By applying a network-based dynamical approach to synthetic and real-world mutualistic ecosystems, we show that biodiversity recovery following collapse is maximized when extirpated species are reintroduced based solely on their total number of connections in the original interaction network. More complex network-based strategies that prioritize the reintroduction of species that improve “higher order” topological features such as compartmentalization do not provide meaningful performance improvements. These results suggest that it is possible to design nearly optimal restoration strategies that maximize biodiversity recovery for data-poor ecosystems in order to ensure the delivery of critical natural services that fuel economic development, food security, and human health around the globe.;,citation_author=Udit Bhatia;,citation_author=Sarth Dubey;,citation_author=Tarik C. Gouhier;,citation_author=Auroop R. Ganguly;,citation_publication_date=2023-12;,citation_cover_date=2023-12;,citation_year=2023;,citation_issue=1;,citation_doi=10.1038/s42003-023-05622-3;,citation_issn=2399-3642;,citation_volume=6;,citation_language=en-US;,citation_journal_title=Communications Biology;,citation_publisher=Nature Publishing Group;">
<meta name="citation_reference" content="citation_title=Inductive link prediction boosts data availability and enables cross-community link prediction in ecological networks;,citation_abstract=Predicting species interactions within ecological networks is vital for understanding ecosystem functioning and the response of communities to changing environments. Traditional link prediction models often fall short due to sparse and incomplete data and are limited to single networks. Here, we present a novel approach using inductive link prediction (ILP), which leverages structural similarities across diverse ecological networks. Our model pools data across communities, and uses transfer learning to enable prediction within and between different ecological communities. We applied our model to 538 networks across four community types: plant-seed disperser, plant-pollinator, host-parasite, and plant-herbivore. ILP outperforms non-ILP models, particularly in host-parasite and plant-seed disperser networks. However, the efficacy of cross-community predictions varies, with plant-pollinator networks consistently under-performing as train and test sets. Moreover, we developed the first method to computationally estimate the limits of link prediction given a certain proportion of missing links, in which ILP performs better than a non-ILP model. This study underscores the potential of ILP to generalize link prediction across different ecological contexts.;,citation_author=Barry Biton;,citation_author=Rami Puzis;,citation_author=Shai Pilosof;,citation_publication_date=2024-08;,citation_cover_date=2024-08;,citation_year=2024;,citation_language=en-US;,citation_publisher=EcoEvoRxiv;">
<meta name="citation_reference" content="citation_title=Co-occurrence is not evidence of ecological interactions;,citation_abstract=There is a rich amount of information in co-occurrence (presence–absence) data that could be used to understand community assembly. This proposition first envisioned by Forbes (1907) and then Diamond (1975) prompted the development of numerous modelling approaches (e.g. null model analysis, co-occurrence networks and, more recently, joint species distribution models). Both theory and experimental evidence support the idea that ecological interactions may affect co-occurrence, but it remains unclear to what extent the signal of interaction can be captured in observational data. It is now time to step back from the statistical developments and critically assess whether co-occurrence data are really a proxy for ecological interactions. In this paper, we present a series of arguments based on probability, sampling, food web and coexistence theories supporting that significant spatial associations between species (or lack thereof) is a poor proxy for ecological interactions. We discuss appropriate interpretations of co-occurrence, along with potential avenues to extract as much information as possible from such data.;,citation_author=F. Guillaume Blanchet;,citation_author=Kevin Cazelles;,citation_author=Dominique Gravel;,citation_publication_date=2020;,citation_cover_date=2020;,citation_year=2020;,citation_issue=7;,citation_doi=10.1111/ele.13525;,citation_issn=1461-0248;,citation_volume=23;,citation_language=en-US;,citation_journal_title=Ecology Letters;">
<meta name="citation_reference" content="citation_title=Neural ordinary differential equations for ecological and evolutionary time-series analysis;,citation_abstract=Inferring the functional shape of ecological and evolutionary processes from time-series data can be challenging because processes are often not describable with simple equations. The dynamical coupling between variables in time series further complicates the identification of equations through model selection as the inference of a given process is contingent on the accurate depiction of all other processes. We present a novel method, neural ordinary differential equations (NODEs), for learning ecological and evolutionary processes from time-series data by modelling dynamical systems as ordinary differential equations and dynamical functions with artificial neural networks (ANNs). Upon successful training, the ANNs converge to functional shapes that best describe the biological processes underlying the dynamics observed, in a way that is robust to mathematical misspecifications of the dynamical model. We demonstrate NODEs in a population dynamic context and show how they can be used to infer ecological interactions, dynamical causation and equilibrium points. We tested NODEs by analysing well-understood hare and lynx time-series data, which revealed that prey–predator oscillations were mainly driven by the interspecific interaction, as well as intraspecific densitydependence, and characterised by a single equilibrium point at the centre of the oscillation. Our approach is applicable to any system that can be modelled with differential equations, and particularly suitable for linking ecological, evolutionary and environmental dynamics where parametric approaches are too challenging to implement, opening new avenues for theoretical and empirical investigations.;,citation_author=Willem Bonnaffé;,citation_author=Ben C. Sheldon;,citation_author=Tim Coulson;,citation_publication_date=2021;,citation_cover_date=2021;,citation_year=2021;,citation_issue=7;,citation_doi=10.1111/2041-210X.13606;,citation_issn=2041-210X;,citation_volume=12;,citation_language=en-US;,citation_journal_title=Methods in Ecology and Evolution;">
<meta name="citation_reference" content="citation_title=Walk partitions of flow in Ecological Network Analysis: Review and synthesis of methods and indicators;,citation_abstract=Ecological Network Analysis (ENA) has provided insights into the structure, function, and transformation of ecosystems for more than forty years. Key insights from ENA focus on how the patterns of directed weighted transactions among system components (e.g., species, functional groups, economic sectors) create emergent and often unexpected relationships in ecosystems that affect system function and sustainability. Flow analysis, also called throughflow analysis, is one of several core techniques in ENA. Generally, it traces the flux of energy or matter through the network from inputs to outputs. During the forty-years of development, flow analysis has accreted multiple extensions and modifications. In this concept and synthesis paper, we review four flow analyses and show how they are conceptually linked by partitioning flows across subsets of pathways within networks. These flow analyses include: (1) the definition of throughflow, a measure of the total processing power of a network; (2) Leontief’s decomposition based on walk length, indicating the direction and distance of energy or matter flow; (3) Finn’s measure of recycling of matter in networks; and (4) five mode analysis, characterizing flows according to their origin and destination. Presenting these techniques side-by-side with a common conceptual framework reveals overlaps and distinctive elements among the analytic products. This synthesis clarifies the flow analyses tools and their applications to ecological and socio-economic networks and provides example applications. Further, new insights are presented by combining existing flow analyses to calculate novel indices that further characterize the flow structure of networks. For example, both indirect flows in networks and cycling are highly important features in networks. In order to determine the proportion of indirect flows generated through cycling, we can use the ratio of Cycled Flow identified from Finn’s analysis and the indirect flows identified in the Leontief analysis. As ENA matures through additional analysis development and applications, it will continue to provide insights into ecosystems and contribute to the broader area of network science.;,citation_author=Stuart R. Borrett;,citation_author=Ursula M. Scharler;,citation_publication_date=2019-11;,citation_cover_date=2019-11;,citation_year=2019;,citation_doi=10.1016/j.ecolind.2019.105451;,citation_issn=1470-160X;,citation_volume=106;,citation_language=en-US;,citation_journal_title=Ecological Indicators;">
<meta name="citation_reference" content="citation_title=Partitioning time series to improve process-based models with machine learning;,citation_abstract=Process-based community models are required to extrapolate beyond current dynamics and anticipate biodiversity response to global change, but their adoption has been limited in practice because of issues with model parameter estimation and model inaccuracies. While observation data could be used to directly estimate parameters and refine model structures, model nonlinearities, sparse and noisy datasets and the complexity of ecological dynamics complicate this process. Here, we propose an inverse modelling framework designed for addressing these issues, that relies on a segmentation method combined with automatic differentiation, deep learning gradient-based optimizers, minibatching, and a parameter normalization technique. By partitioning the data into time series with a short time horizon, we show that the segmentation method regularizes the inverse problem. Implemented in the Julia package PiecewiseInference.jl, the resulting framework enables the efficient calibration of realistic dynamical community models and supports the use of neural networks to represent complex ecological processes within the model. We evaluate the performance of the inverse modelling framework on simulated food-web dynamics of increasing complexity, including scenarios with partial observations. Our results demonstrate its performance in inferring model parameters and providing forecasts. We further showcase that the regularization induced by the framework, together with its efficiency, are instrumental for the online training of neural network-based parametrizations, alongside the other model parameters. Neural network-based parametrizations not only mitigate model inaccuracies but can also contribute to the refinement of ecological theory through their offline examination. The proposed inverse modelling framework is data-efficient, interpretable and capable of extrapolating beyond observed dynamics, showing promise in improving our ability to understand and forecast biodiversity dynamics.;,citation_author=Victor Boussange;,citation_author=Pau Vilimelis Aceituno;,citation_author=Frank Schaefer;,citation_author=Loic Pellissier;,citation_publication_date=2024-04;,citation_cover_date=2024-04;,citation_year=2024;,citation_doi=10.1101/2022.07.25.501365;,citation_language=en-US;,citation_publisher=bioRxiv;">
<meta name="citation_reference" content="citation_title=Identifying a common backbone of interactions underlying food webs from different ecosystems;,citation_author=Bernat Bramon Mora;,citation_author=Dominique Gravel;,citation_author=Luis J. Gilarranz;,citation_author=Timothée Poisot;,citation_author=Daniel B. Stouffer;,citation_publication_date=2018-12;,citation_cover_date=2018-12;,citation_year=2018;,citation_issue=1;,citation_doi=10.1038/s41467-018-05056-0;,citation_issn=2041-1723;,citation_volume=9;,citation_language=en-US;,citation_journal_title=Nature Communications;,citation_publisher=Nature Publishing Group;">
<meta name="citation_reference" content="citation_title=Applying a method before its proof-of-concept: A cautionary tale using inferred food webs;,citation_author=Chris Brimacombe;,citation_author=Korryn Bodner;,citation_author=Marie-Josee Fortin;,citation_publication_date=2024-04;,citation_cover_date=2024-04;,citation_year=2024;,citation_doi=10.13140/RG.2.2.22076.65927;">
<meta name="citation_reference" content="citation_title=Shortcomings of reusing species interaction networks created by different sets of researchers;,citation_abstract=Given the requisite cost associated with observing species interactions, ecologists often reuse species interaction networks created by different sets of researchers to test their hypotheses regarding how ecological processes drive network topology. Yet, topological properties identified across these networks may not be sufficiently attributable to ecological processes alone as often assumed. Instead, much of the totality of topological differences between networks—topological heterogeneity—could be due to variations in research designs and approaches that different researchers use to create each species interaction network. To evaluate the degree to which this topological heterogeneity is present in available ecological networks, we first compared the amount of topological heterogeneity across 723 species interaction networks created by different sets of researchers with the amount quantified from non-ecological networks known to be constructed following more consistent approaches. Then, to further test whether the topological heterogeneity was due to differences in study designs, and not only to inherent variation within ecological networks, we compared the amount of topological heterogeneity between species interaction networks created by the same sets of researchers (i.e., networks from the same publication) with the amount quantified between networks that were each from a unique publication source. We found that species interaction networks are highly topologically heterogeneous: while species interaction networks from the same publication are much more topologically similar to each other than interaction networks that are from a unique publication, they still show at least twice as much heterogeneity as any category of non-ecological networks that we tested. Altogether, our findings suggest that extra care is necessary to effectively analyze species interaction networks created by different researchers, perhaps by controlling for the publication source of each network.;,citation_author=Chris Brimacombe;,citation_author=Korryn Bodner;,citation_author=Matthew Michalska-Smith;,citation_author=Timothée Poisot;,citation_author=Marie-Josée Fortin;,citation_publication_date=2023-04;,citation_cover_date=2023-04;,citation_year=2023;,citation_issue=4;,citation_doi=10.1371/journal.pbio.3002068;,citation_issn=1545-7885;,citation_volume=21;,citation_language=en-US;,citation_journal_title=PLOS Biology;,citation_publisher=Public Library of Science;">
<meta name="citation_reference" content="citation_title=Emergence of Structural Patterns in Neutral Trophic Networks;,citation_abstract=Interaction networks are central elements of ecological systems and have very complex structures. Historically, much effort has focused on niche-mediated processes to explain these structures, while an emerging consensus posits that both niche and neutral mechanisms simultaneously shape many features of ecological communities. However, the study of interaction networks still lacks a comprehensive neutral theory. Here we present a neutral model of predator-prey interactions and analyze the structural characteristics of the simulated networks. We find that connectance values (complexity) and complexity-diversity relationships of neutral networks are close to those observed in empirical bipartite networks. High nestedness and low modularity values observed in neutral networks fall in the range of those from empirical antagonist bipartite networks. Our results suggest that, as an alternative to niche-mediated processes that induce incompatibility between species (“niche forbidden links”), neutral processes create “neutral forbidden links” due to uneven species abundance distributions and the low probability of interaction between rare species. Neutral trophic networks must be seen as the missing endpoint of a continuum from niche to purely stochastic approaches of community organization.;,citation_author=Elsa Canard;,citation_author=Nicolas Mouquet;,citation_author=Lucile Marescot;,citation_author=Kevin J. Gaston;,citation_author=Dominique Gravel;,citation_author=David Mouillot;,citation_publication_date=2012-08;,citation_cover_date=2012-08;,citation_year=2012;,citation_issue=8;,citation_doi=10.1371/journal.pone.0038295;,citation_issn=1932-6203;,citation_volume=7;,citation_language=en-US;,citation_journal_title=PLOS ONE;,citation_publisher=Public Library of Science;">
<meta name="citation_reference" content="citation_title=Addressing the Eltonian shortfall with trait-based interaction models;,citation_abstract=We have very limited knowledge of how species interact in most communities and ecosystems despite trophic relationships being fundamental for linking biodiversity to ecosystem functioning. A promising approach to fill this gap is to predict interactions based on functional traits, but many questions remain about how well we can predict interactions for different taxa, ecosystems and amounts of input data. Here, we built a new traits-based model of trophic interactions for European vertebrates and found that even models calibrated with 0.1% of the interactions (100 out of 71 k) estimated the full European vertebrate food web reasonably well. However, predators were easier to predict than prey, especially for some clades (e.g. fowl and storks) and local food web connectance was consistently overestimated. Our results demonstrate the ability to rapidly generate food webs when empirical data are lacking—an important step towards a more complete and spatially explicit description of food webs.;,citation_author=Dominique Caron;,citation_author=Luigi Maiorano;,citation_author=Wilfried Thuiller;,citation_author=Laura J. Pollock;,citation_publication_date=2022;,citation_cover_date=2022;,citation_year=2022;,citation_issue=4;,citation_doi=10.1111/ele.13966;,citation_issn=1461-0248;,citation_volume=25;,citation_language=en-US;,citation_journal_title=Ecology Letters;">
<meta name="citation_reference" content="citation_title=Trait-matching models predict pairwise interactions across regions, not food web properties;,citation_abstract=Aim Food webs are essential for understanding how ecosystems function, but empirical data on the interactions that make up these ecological networks are lacking for most taxa in most ecosystems. Trait-based models can fill these data gaps, but their ability to do so has not been widely tested. We test how well these models can extrapolate to new ecological communities both in terms of pairwise predator–prey interactions and higher level food web attributes (i.e. species position, food web-level properties). Location Canada, Europe, Tanzania. Time Period Current. Major Taxa Studied Terrestrial vertebrates. Methods We train trait-based models of pairwise trophic interactions on four independent vertebrate food webs (Canadian tundra, Serengeti, alpine south-eastern Pyrenees and Europe) and evaluate how well these models predict pairwise interactions and network properties of each food web. Results We find that, overall, trait-based models predict most interactions and their absence correctly. Performance was best for training and testing on the same food web (AUC &amp;amp;amp;gt; 0.90) and declined with environmental and phylogenetic distances with the strongest loss of performance for the tundra-Serengeti ecosystems (AUC &amp;gt; 0.75). Network metrics were less well-predicted than single interactions by our models with predicted food webs being more connected, less modular, and with higher mean trophic levels than observed. Main Conclusions Theory predicts that the variability observed in food webs can be explained by differences in trait distributions and trait-matching relationships. Our finding that trait-based models can predict many trophic interactions, even in contrasting environments, adds to the growing body of evidence that there are general constraints on interactions and that trait-based methods can serve as a useful first approximation of food webs in unknown areas. However, food webs are more than the sum of their parts, and predicting network attributes will likely require models that simultaneously predict individual interactions and community constraints.;,citation_author=Dominique Caron;,citation_author=Ulrich Brose;,citation_author=Miguel Lurgi;,citation_author=F. Guillaume Blanchet;,citation_author=Dominique Gravel;,citation_author=Laura J. Pollock;,citation_publication_date=2024;,citation_cover_date=2024;,citation_year=2024;,citation_issue=4;,citation_doi=10.1111/geb.13807;,citation_issn=1466-8238;,citation_volume=33;,citation_language=en-US;,citation_journal_title=Global Ecology and Biogeography;">
<meta name="citation_reference" content="citation_title=The missing link: Discerning true from false negatives when sampling species interaction networks;,citation_abstract=Ecosystems are composed of networks of interacting species. These interactions allow communities of species to persist through time through both neutral and adaptive processes. Despite their importance, a robust understanding of (and ability to predict and forecast) interactions among species remains elusive. This knowledge-gap is largely driven by a shortfall of data—although species occurrence data has rapidly increased in the last decade, species interaction data has not kept pace, largely due to the effort required to sample interactions. This means there are many interactions between species that occur in nature, but we do not know these interactions occur because we have never observed them. These so-called “false-negatives” bias data and hinder inference about the structure and dynamics of interaction networks. Here, we demonstrate the realized rate of false-negatives in data can be quite high, even in thoroughly sampled systems, due to the intrinsic variation in abundances across species in a community. We illustrate how a null model of occurrence detection can be used to estimate the false-negative rate in a given dataset. We also show how to directly incorporate uncertainty due to observation error into model-based predictions of interaction probabilities between species. One hypothesis is that interactions between “rare” species are themselves rare because these species are less likely to encounter one-another than species of higher relative abundance, and that this can (in part) explain the common pattern of nestedness in bipartite interaction networks. However, we demonstrate that across several datasets of spatial or temporally replicated networks, there are positive associations between species co-occurrence and interactions, which suggests these interactions among “rare” species actually exist but simply are not observed. Finally, we assess how false negatives influence various models of network prediction, and recommend directly accounting for observation error in predictive models. We conclude by discussing how the understanding of false-negatives can inform how we design monitoring schemes for species interactions.;,citation_author=Michael D. Catchen;,citation_author=Timothée Poisot;,citation_author=Laura J. Pollock;,citation_author=Andrew Gonzalez;,citation_publication_date=2023-01;,citation_cover_date=2023-01;,citation_year=2023;,citation_language=en-US;,citation_publisher=EcoEvoRxiv;">
<meta name="citation_reference" content="citation_title=Phylogenetic constraints and adaptation explain food-web structure;,citation_abstract=Food webs are descriptions of who eats whom in an ecosystem. Although extremely complex and variable, their structure possesses basic regularities1,2,3,4,5,6. A fascinating question is to find a simple model capturing the underlying processes behind these repeatable patterns. Until now, two models have been devised for the description of trophic interactions within a natural community7,8. Both are essentially based on the concept of ecological niche, with the consumers organized along a single niche dimension; for example, prey size8,9. Unfortunately, they fail to describe adequately recent and high-quality data. Here, we propose a new model built on the hypothesis that any species’ diet is the consequence of phylogenetic constraints and adaptation. Simple rules incorporating both concepts yield food webs whose structure is very close to real data. Consumers are organized in groups forming a nested hierarchy, which better reflects the complexity and multidimensionality of most natural systems.;,citation_author=Marie-France Cattin;,citation_author=Louis-Félix Bersier;,citation_author=Carolin Banašek-Richter;,citation_author=Richard Baltensperger;,citation_author=Jean-Pierre Gabriel;,citation_publication_date=2004-02;,citation_cover_date=2004-02;,citation_year=2004;,citation_issue=6977;,citation_doi=10.1038/nature02327;,citation_issn=1476-4687;,citation_volume=427;,citation_language=en-US;,citation_journal_title=Nature;,citation_publisher=Nature Publishing Group;">
<meta name="citation_reference" content="citation_title=Transient dynamics and food–web complexity in the Lotka–Volterra cascade model;,citation_abstract=How does the long–term behaviour near equilibrium of model food webs correlate with their short–term transient dynamics? Here, simulations of the Lotka–Volterra cascade model of food webs provide the first evidence to answer this question. Transient behaviour is measured by resilience, reactivity, the maximum amplification of a perturbation and the time at which the maximum amplification occurs. Model food webs with a higher probability of local asymptotic stability may be less resilient and may have a larger transient growth of perturbations. Given a fixed connectance, the sizes and durations of transient responses to perturbations increase with the number of species. Given a fixed number of species, as connectance increases, the sizes and durations of transient responses to perturbations may increase or decrease depending on the type of link that is varied. Reactivity is more sensitive to changes in the number of donor–controlled links than to changes in the number of recipient–controlled links, while resilience is more sensitive to changes in the number of recipient–controlled links than to changes in the number of donor–controlled links. Transient behaviour is likely to be one of the important factors affecting the persistence of ecological communities.;,citation_author=X. Chen;,citation_author=J. E. Cohen;,citation_publication_date=2001-04;,citation_cover_date=2001-04;,citation_year=2001;,citation_issue=1469;,citation_doi=10.1098/rspb.2001.1596;,citation_volume=268;,citation_journal_title=Proceedings of the Royal Society of London. Series B: Biological Sciences;,citation_publisher=Royal Society;">
<meta name="citation_reference" content="citation_title=The environment to the rescue: Can physics help predict predator–prey interactions?;,citation_abstract=Understanding the factors that determine the occurrence and strength of ecological interactions under specific abiotic and biotic conditions is fundamental since many aspects of ecological community stability and ecosystem functioning depend on patterns of interactions among species. Current approaches to mapping food webs are mostly based on traits, expert knowledge, experiments, and/or statistical inference. However, they do not offer clear mechanisms explaining how trophic interactions are affected by the interplay between organism characteristics and aspects of the physical environment, such as temperature, light intensity or viscosity. Hence, they cannot yet predict accurately how local food webs will respond to anthropogenic pressures, notably to climate change and species invasions. Herein, we propose a framework that synthesises recent developments in food-web theory, integrating body size and metabolism with the physical properties of ecosystems. We advocate for combination of the movement paradigm with a modular definition of the predation sequence, because movement is central to predator–prey interactions, and a generic, modular model is needed to describe all the possible variation in predator–prey interactions. Pending sufficient empirical and theoretical knowledge, our framework will help predict the food-web impacts of well-studied physical factors, such as temperature and oxygen availability, as well as less commonly considered variables such as wind, turbidity or electrical conductivity. An improved predictive capability will facilitate a better understanding of ecosystem responses to a changing world.;,citation_author=Mehdi Cherif;,citation_author=Ulrich Brose;,citation_author=Myriam R. Hirt;,citation_author=Remo Ryser;,citation_author=Violette Silve;,citation_author=Georg Albert;,citation_author=Russell Arnott;,citation_author=Emilio Berti;,citation_author=Alyssa Cirtwill;,citation_author=Alexander Dyer;,citation_author=Benoit Gauzens;,citation_author=Anhubav Gupta;,citation_author=Hsi-Cheng Ho;,citation_author=Sébastien M. J. Portalier;,citation_author=Danielle Wain;,citation_author=Kate Wootton;,citation_publication_date=2024;,citation_cover_date=2024;,citation_year=2024;,citation_issue=n/a;,citation_doi=10.1111/brv.13105;,citation_issn=1469-185X;,citation_volume=n/a;,citation_language=en-US;,citation_journal_title=Biological Reviews;">
<meta name="citation_reference" content="citation_title=The advantages of the Matthews correlation coefficient (MCC) over F1 score and accuracy in binary classification evaluation;,citation_abstract=To evaluate binary classifications and their confusion matrices, scientific researchers can employ several statistical rates, accordingly to the goal of the experiment they are investigating. Despite being a crucial issue in machine learning, no widespread consensus has been reached on a unified elective chosen measure yet. Accuracy and F1 score computed on confusion matrices have been (and still are) among the most popular adopted metrics in binary classification tasks. However, these statistical measures can dangerously show overoptimistic inflated results, especially on imbalanced datasets.;,citation_author=Davide Chicco;,citation_author=Giuseppe Jurman;,citation_publication_date=2020-01;,citation_cover_date=2020-01;,citation_year=2020;,citation_issue=1;,citation_doi=10.1186/s12864-019-6413-7;,citation_issn=1471-2164;,citation_volume=21;,citation_journal_title=BMC Genomics;">
<meta name="citation_reference" content="citation_title=A quantitative framework for investigating the reliability of empirical network construction;,citation_abstract=1.Descriptions of ecological networks typically assume that the same interspecific interactions occur each time a community is observed. This contrasts with the known stochasticity of ecological communities: community composition, species abundances, and link structure all vary in space and time. Moreover, finite sampling generates variation in the set of interactions actually observed. For interactions that have not been observed, most data sets will not contain enough information for the ecologist to be con_dent that unobserved interactions truly did not occur. 2. Here we develop the conceptual and analytical tools needed to capture uncertainty in the estimation of pairwise interactions. To define the problem, we identify the different contributions to the uncertainty of an interaction. We then outline a framework to quantify the uncertainty around each interaction by combining data on observed co-occurrences with prior knowledge. We illustrate this framework using perhaps the most extensively sampled network to date. 3. We found significant uncertainty in estimates for the probability of most pairwise interactions. This uncertainty can, however, be constrained with informative priors. This uncertainty scaled up to summary measures of network structure such as connectance and nestedness. Even with informative priors, we are likely to miss many interactions that may occur rarely or under different local conditions. 4. Overall, we demonstrate the importance of acknowledging the uncertainty inherent in network studies, and the utility of treating interactions as probabilities in pinpointing areas where more study is needed. Most importantly, we stress that networks are best thought of as systems constructed from random variables, the stochastic nature of which must be acknowledged for an accurate representation. Doing so will fundamentally change networks analyses and yield greater realism. This article is protected by copyright. All rights reserved.;,citation_author=Alyssa R. Cirtwill;,citation_author=Anna Eklf;,citation_author=Tomas Roslin;,citation_author=Kate Wootton;,citation_author=Dominique Gravel;,citation_publication_date=2019;,citation_cover_date=2019;,citation_year=2019;,citation_issue=6;,citation_doi=10.1111/2041-210X.13180;,citation_issn=2041-210X;,citation_volume=10;,citation_language=en-US;,citation_journal_title=Methods in Ecology and Evolution;">
<meta name="citation_reference" content="citation_title=The impact of intraspecific variation on food web structure;,citation_abstract=Accounting for the variation that occurs within species in food webs can theoretically result in significant changes in both network structure and dynamics. However, there has been little work exploring their role with empirical data. In particular, the variation associated with species’ life cycles, which is prevalent and represents both trait variation and taxonomic identity, has received little attention. Here, we characterize the structural consequences of life stage variation in five food webs, including a newly compiled web from the Arabian Gulf. We show that making life stage variation explicit in food webs results in larger food webs that possess consistent structural changes that are separate from the changes in structure that come simply from increasing the number of nodes in the webs. Furthermore, we show that the magnitude of these changes is related to ontogenetic specialism, the degree of overlap in the ecological niches of life stages. These results demonstrate the capacity of intraspecific variation to affect ecological networks and indicate the potential usefulness of stage-structured food webs, which capture size and taxonomic information, to represent variation below the species level.;,citation_author=Tom Clegg;,citation_author=Mohammad Ali;,citation_author=Andrew P. Beckerman;,citation_publication_date=2018;,citation_cover_date=2018;,citation_year=2018;,citation_issue=12;,citation_doi=10.1002/ecy.2523;,citation_issn=1939-9170;,citation_volume=99;,citation_language=en-US;,citation_journal_title=Ecology;">
<meta name="citation_reference" content="citation_title=Community Food Webs: Data and Theory;,citation_abstract=Food webs hold a central place in ecology. They describe which organisms feed on which others in natural habitats. This book describes recently discovered empirical regularities in real food webs: it proposes a novel theory unifying many of these regularities, as well as extensive empirical data. After a general introduction, reviewing the empirical and theoretical discoveries about food webs, the second portion of the book shows that community food webs obey several striking phenomenological regularities. Some of these unify, regardless of habitat. Others differentiate, showing that habitat significantly influences structure. The third portion of the book presents a theoretical analysis of some of the unifying empirical regularities. The fourth portion of the book presents 13 community food webs. Collected from scattered sources and carefully edited, they are the empirical basis for the results in the volume. The largest available set of data on community food webs provides a valuable foundation for future studies of community food webs. The book is intended for graduate students, teachers and researchers primarily in ecology. The theoretical portions of the book provide materials useful to teachers of applied combinatorics, in particular, random graphs. Researchers in random graphs will find here unsolved mathematical problems.;,citation_author=Joel E Cohen;,citation_author=Frederic Briand;,citation_author=Charles Newman;,citation_publication_date=1990;,citation_cover_date=1990;,citation_year=1990;,citation_isbn=978-3-642-83786-9;,citation_language=en-US;,citation_series_title=Biomathematics;">
<meta name="citation_reference" content="citation_title=Food webs and the dimensionality of trophic niche space;,citation_abstract=If the trophic niche of a kind of organism is a connected region in niche space, then it is possible for trophic niche overlaps to be described in a one-dimensional niche space if and only if the trophic niche overlap graph is an interval graph. An analysis of 30 food webs, using the combinatorial theory of interval graphs, suggests that a niche space of dimension 1 suffices, with unexpectedly high frequency and perhaps always, to describe the trophic niche overlaps implied by real food webs in single habitats. Consequently, real food webs fall in a small subset of the set of mathematically possible food webs. That real food webs are compatible with one-dimensional trophic niche spaces, more often than can be explained by chance alone, has not been noticed previously.;,citation_author=Joel E. Cohen;,citation_publication_date=1977-10;,citation_cover_date=1977-10;,citation_year=1977;,citation_issue=10;,citation_doi=10.1073/pnas.74.10.4533;,citation_volume=74;,citation_journal_title=Proceedings of the National Academy of Sciences;,citation_publisher=Proceedings of the National Academy of Sciences;">
<meta name="citation_reference" content="citation_title=A stochastic theory of community food webs I. Models and aggregated data;,citation_abstract=Three recently discovered quantitative empirical generalizations describe major features of the structure of community food webs. These generalizations are: (i) a species scaling law: the mean proportions of basal, intermediate and top species remain invariant at approximately 0.19, 0.53, and 0.29, respectively, over the range of variation in the number of species in a web; (ii) a link scaling law : the mean proportions of trophic links in the categories basal-intermediate, basal-top, intermediate–intermediate, and intermediate–top remain invariant at approximately 0.27, 0.08, 0.30 and 0.35, respectively, over the range of variation in the number of species in a web; and (iii) a link-species scaling law: the ratio of mean trophic links to species remains invariant at approximately 1.86, over the range of variation in the number of species in a web. This paper presents a model, the only successful one among several attempts, in which the first two of these empirical generalizations can be derived as a consequence of the third. The model assumes that species are ordered in a cascade or hierarchy such that a given species can prey on only those species below it and can be preyed on by only those species above it in the hierarchy.;,citation_author=Joel E. Cohen;,citation_author=C. M. Newman;,citation_author=John Hyslop Steele;,citation_publication_date=1985;,citation_cover_date=1985;,citation_year=1985;,citation_issue=1237;,citation_doi=10.1098/rspb.1985.0042;,citation_volume=224;,citation_journal_title=Proceedings of the Royal Society of London. Series B. Biological Sciences;,citation_publisher=Royal Society;">
<meta name="citation_reference" content="citation_title=DeepDive: Estimating global biodiversity patterns through time using deep learning;,citation_abstract=Understanding how biodiversity has changed through time is a central goal of evolutionary biology. However, estimates of past biodiversity are challenged by the inherent incompleteness of the fossil record, even when state-of-the-art statistical methods are applied to adjust estimates while correcting for sampling biases. Here we develop an approach based on stochastic simulations of biodiversity and a deep learning model to infer richness at global or regional scales through time while incorporating spatial, temporal and taxonomic sampling variation. Our method outperforms alternative approaches across simulated datasets, especially at large spatial scales, providing robust palaeodiversity estimates under a wide range of preservation scenarios. We apply our method on two empirical datasets of different taxonomic and temporal scope: the Permian-Triassic record of marine animals and the Cenozoic evolution of proboscideans. Our estimates provide a revised quantitative assessment of two mass extinctions in the marine record and reveal rapid diversification of proboscideans following their expansion out of Africa and a &amp;amp;amp;gt;70% diversity drop in the Pleistocene.;,citation_author=Rebecca B. Cooper;,citation_author=Joseph T. Flannery-Sutherland;,citation_author=Daniele Silvestro;,citation_publication_date=2024-05;,citation_cover_date=2024-05;,citation_year=2024;,citation_issue=1;,citation_doi=10.1038/s41467-024-48434-7;,citation_issn=2041-1723;,citation_volume=15;,citation_language=en-US;,citation_journal_title=Nature Communications;,citation_publisher=Nature Publishing Group;">
<meta name="citation_reference" content="citation_title=Ecosystem function in predator–prey food webs—confronting dynamic models with empirical data;,citation_abstract=Most ecosystem functions and related services involve species interactions across trophic levels, for example, pollination and biological pest control. Despite this, our understanding of ecosystem function in multitrophic communities is poor, and research has been limited to either manipulation in small communities or statistical descriptions in larger ones. Recent advances in food web ecology may allow us to overcome the trade-off between mechanistic insight and ecological realism. Molecular tools now simplify the detection of feeding interactions, and trait-based approaches allow the application of dynamic food web models to real ecosystems. We performed the first test of an allometric food web model’s ability to replicate temporally nonaggregated abundance data from the field and to provide mechanistic insight into the function of predation. We aimed to reproduce and explore the drivers of the population dynamics of the aphid herbivore Rhopalosiphum padi observed in ten Swedish barley fields. We used a dynamic food web model, taking observed interactions and abundances of predators and alternative prey as input data, allowing us to examine the role of predation in aphid population control. The inverse problem methods were used for simultaneous model fit optimization and model parameterization. The model captured &amp;amp;amp;gt;70% of the variation in aphid abundance in five of ten fields, supporting the model-embodied hypothesis that body size can be an important determinant of predation in the arthropod community. We further demonstrate how in-depth model analysis can disentangle the likely drivers of function, such as the community’s abundance and trait composition. Analysing the variability in model performance revealed knowledge gaps, such as the source of episodic aphid mortality, and general method development needs that, if addressed, would further increase model success and enable stronger inference about ecosystem function. The results demonstrate that confronting dynamic food web models with abundance data from the field is a viable approach to evaluate ecological theory and to aid our understanding of function in real ecosystems. However, to realize the full potential of food web models, in ecosystem function research and beyond, trait-based parameterization must be refined and extended to include more traits than body size.;,citation_author=Alva Curtsdotter;,citation_author=H. Thomas Banks;,citation_author=John E. Banks;,citation_author=Mattias Jonsson;,citation_author=Tomas Jonsson;,citation_author=Amanda N. Laubmeier;,citation_author=Michael Traugott;,citation_author=Riccardo Bommarco;,citation_publication_date=2019;,citation_cover_date=2019;,citation_year=2019;,citation_issue=2;,citation_doi=10.1111/1365-2656.12892;,citation_issn=1365-2656;,citation_volume=88;,citation_language=en-US;,citation_journal_title=Journal of Animal Ecology;">
<meta name="citation_reference" content="citation_title=From Graphs to Spatial Graphs;,citation_abstract=Graph theory is a powerful body of mathematical knowledge, based on simple concepts, in which structural units are depicted as nodes with relationships between them depicted as lines. The nodes may have qualitative and quantitative characteristics, and the edges may have properties such as weights and directions. Graph theory provides a flexible conceptual model that can clarify the relationship between structures and processes, including the mechanisms of configuration effects and compositional differences. Graph concepts apply to many ecological and evolutionary phenomena, including interspecific associations, spatial structure, dispersal in landscapes, and relationships within metapopulations and metacommunities. We review applications of graph theory in biology, emphasizing graphs with spatial contexts. We show how spatial graph properties can be used for description and comparison as well as to test specific hypotheses. We suggest that future applications should include explicit spatial elements for landscape studies of ecological, genetic and epidemiological phenomena.;,citation_author=M. R. T. Dale;,citation_author=M.-J. Fortin;,citation_publication_date=2010;,citation_cover_date=2010;,citation_year=2010;,citation_fulltext_html_url=https://www.jstor.org/stable/27896212;,citation_issn=1543-592X;,citation_volume=41;,citation_journal_title=Annual Review of Ecology, Evolution, and Systematics;,citation_publisher=Annual Reviews;">
<meta name="citation_reference" content="citation_title=Exploring the evolutionary signature of food webs’ backbones using functional traits;,citation_abstract=Increasing evidence suggests that an appropriate model for food webs, the network of feeding links in a community of species, should take into account the inherent variability of ecological interactions. Harnessing this variability, we will show that it is useful to interpret empirically observed food webs as realisations of a family of stochastic processes, namely random dot-product graph models. These models provide an ideal extension of food-web models beyond the limitations of current deterministic or partially probabilistic models. As an additional benefit, our RDPG framework enables us to identify the pairwise distance structure given by species’ functional food-web traits: this allows for the natural emergence of ecologically meaningful species groups. Lastly, our results suggest the notion that the evolutionary signature in food webs is already detectable in their stochastic backbones, while the contribution of their fine wiring is arguable. Synthesis Food webs are influenced by many stochastic processes and are constantly evolving. Here, we treat observed food webs as realisations of random dot-product graph models (RDPG), extending food-web modelling beyond the limitations of current deterministic or partially probabilistic models. Our RDPG framework enables us to identify the pairwise-distance structure given by species’ functional food-web traits, which in turn allows for the natural emergence of ecologically meaningful species groups. It also provides a way to measure the phylogenetic signal present in food webs, which we find is strongest in webs’ low-dimensional backbones.;,citation_author=Giulio V Dalla Riva;,citation_author=Daniel B. Stouffer;,citation_publication_date=2016;,citation_cover_date=2016;,citation_year=2016;,citation_issue=4;,citation_doi=10.1111/oik.02305;,citation_issn=1600-0706;,citation_volume=125;,citation_language=en-US;,citation_journal_title=Oikos;">
<meta name="citation_reference" content="citation_title=Spatially explicit predictions of food web structure from regional-level data;,citation_abstract=Knowledge about how ecological networks vary across global scales is currently limited given the complexity of acquiring repeated data for species interactions. Yet, recent developments of metawebs highlight efficient ways to first document possible interactions within regional species pools. Downscaling metawebs towards local network predictions is a promising approach to use current data to investigate the variation of networks across space. However, issues remain in how to represent the spatial variability and uncertainty of species interactions, especially for large scale food webs. Here, we present a probabilistic framework to downscale a metaweb based on the Canadian mammal metaweb and species occurrences from GBIF. We investigate how our approach can be used to represent the variability of networks and communities between ecoregions in Canada. Our results highlighted mismatches in the distribution of species richness and interactions, especially in their within-ecoregion variability, indicating that interactions vary differently than species distributions over continental-scale gradients. Results summarized by ecoregion showed non-constant variation within and between ecologically meaningful biogeographic boundaries and identified contrasting diversity hotspots. Our method offers the potential to bring global predictions down to a more actionable local scale, and increases the diversity of ecological networks that can be projected in space.;,citation_author=Gabriel Dansereau;,citation_author=Ceres Barros;,citation_author=Timothée Poisot;,citation_publication_date=2024;,citation_cover_date=2024;,citation_year=2024;,citation_issue=1909;,citation_doi=10.1098/rstb.2023.0166;,citation_volume=379;,citation_language=en-US;,citation_journal_title=Philosophical Transactions of the Royal Society B: Biological Sciences;,citation_publisher=EcoEvoRxiv;">
<meta name="citation_reference" content="citation_title=On the Origin of Species by Means of Natural Selection, or the Preservation of Favoured Races in the Struggle for Life;,citation_author=Charles Darwin;,citation_publication_date=1859;,citation_cover_date=1859;,citation_year=1859;">
<meta name="citation_reference" content="citation_title=A Model for Tropic Interaction;,citation_abstract=A nonlinear function general enough to include the effects of feeding saturation and intraspecific consumer interference is used to represent the transfer of material or energy from one trophic level to another. The function agrees with some recent experimental data on feeding rates. A model using this feeding rate function is subjected to equilibrium and stability analyses to ascertain its mathematical implications. The analyses lead to several observations; for example, increases in maximum feeding rate may, under certain circumstances, result in decreases in consumer population and mutual interference between consumers is a major stabilizing factor in a nonlinear system. The analyses also suggest that realistic classes of consumer-resourcemodels exist which do not obey Kolmogorov’s Criteria but are nevertheless globally stable.;,citation_author=D. L. DeAngelis;,citation_author=R. A. Goldstein;,citation_author=R. V. O’Neill;,citation_publication_date=1975-07;,citation_cover_date=1975-07;,citation_year=1975;,citation_issue=4;,citation_doi=10.2307/1936298;,citation_issn=0012-9658, 1939-9170;,citation_volume=56;,citation_language=en-US;,citation_journal_title=Ecology;">
<meta name="citation_reference" content="citation_title=Nutrient Dynamics and Food-Web Stability;,citation_author=D. L. DeAngelis;,citation_author=P. J. Mulholland;,citation_author=A. V. Palumbo;,citation_author=A. D. Steinman;,citation_author=M. A. Huston;,citation_author=J. W. Elwood;,citation_publication_date=1989-11;,citation_cover_date=1989-11;,citation_year=1989;,citation_issue=Volume 20, 1989;,citation_doi=10.1146/annurev.es.20.110189.000443;,citation_issn=1543-592X, 1545-2069;,citation_volume=20;,citation_language=en-US;,citation_journal_title=Annual Review of Ecology, Evolution and Systematics;,citation_publisher=Annual Reviews;">
<meta name="citation_reference" content="citation_title=Analysing ecological networks of species interactions;,citation_abstract=Network approaches to ecological questions have been increasingly used, particularly in recent decades. The abstraction of ecological systems – such as communities – through networks of interactions between their components indeed provides a way to summarize this information with single objects. The methodological framework derived from graph theory also provides numerous approaches and measures to analyze these objects and can offer new perspectives on established ecological theories as well as tools to address new challenges. However, prior to using these methods to test ecological hypotheses, it is necessary that we understand, adapt, and use them in ways that both allow us to deliver their full potential and account for their limitations. Here, we attempt to increase the accessibility of network approaches by providing a review of the tools that have been developed so far, with – what we believe to be – their appropriate uses and potential limitations. This is not an exhaustive review of all methods and metrics, but rather, an overview of tools that are robust, informative, and ecologically sound. After providing a brief presentation of species interaction networks and how to build them in order to summarize ecological information of different types, we then classify methods and metrics by the types of ecological questions that they can be used to answer from global to local scales, including methods for hypothesis testing and future perspectives. Specifically, we show how the organization of species interactions in a community yields different network structures (e.g., more or less dense, modular or nested), how different measures can be used to describe and quantify these emerging structures, and how to compare communities based on these differences in structures. Within networks, we illustrate metrics that can be used to describe and compare the functional and dynamic roles of species based on their position in the network and the organization of their interactions as well as associated new methods to test the significance of these results. Lastly, we describe potential fruitful avenues for new methodological developments to address novel ecological questions.;,citation_author=Eva Delmas;,citation_author=Mathilde Besson;,citation_author=Marie-Hélène Brice;,citation_author=Laura A. Burkle;,citation_author=Giulio V. Dalla Riva;,citation_author=Marie-Josée Fortin;,citation_author=Dominique Gravel;,citation_author=Paulo R. Guimarães;,citation_author=David H. Hembry;,citation_author=Erica A. Newman;,citation_author=Jens M. Olesen;,citation_author=Mathias M. Pires;,citation_author=Justin D. Yeakel;,citation_author=Timothée Poisot;,citation_publication_date=2019;,citation_cover_date=2019;,citation_year=2019;,citation_issue=1;,citation_doi=10.1111/brv.12433;,citation_issn=1469-185X;,citation_volume=94;,citation_language=en-US;,citation_journal_title=Biological Reviews;">
<meta name="citation_reference" content="citation_title=Simulations of biomass dynamics in community food webs;,citation_abstract=Food webs are the backbone upon which biomass flows through ecosystems. Dynamical models of biomass can reveal how the structure of food webs is involved in many key ecosystem properties, such as persistence, stability, etc. In this contribution, we present BioEnergeticFoodWebs, an implementation of Yodzis &amp;amp;amp; Innes (The American Naturalist 139, 1151?1175, 1992) bio-energetic model, in the high-performance computing language Julia. We illustrate how this package can be used to conduct numerical experiments in a reproducible and standard way. A reference implementation of this widely used model will ease reproducibility and comparison of results across studies.;,citation_author=Eva Delmas;,citation_author=Ulrich Brose;,citation_author=Dominique Gravel;,citation_author=Daniel B. Stouffer;,citation_author=Timothée Poisot;,citation_publication_date=2017;,citation_cover_date=2017;,citation_year=2017;,citation_issue=7;,citation_doi=10.1111/2041-210X.12713;,citation_issn=2041-210X;,citation_volume=8;,citation_language=en-US;,citation_journal_title=Methods in Ecology and Evolution;">
<meta name="citation_reference" content="citation_title=Ecological interactions and the Netflix problem;,citation_abstract=Species interactions are a key component of ecosystems but we generally have an incomplete picture of who-eats-who in a given community. Different techniques have been devised to predict species interactions using theoretical models or abundances. Here, we explore the K nearest neighbour approach, with a special emphasis on recommendation, along with a supervised machine learning technique. Recommenders are algorithms developed for companies like Netflix to predict whether a customer will like a product given the preferences of similar customers. These machine learning techniques are well-suited to study binary ecological interactions since they focus on positive-only data. By removing a prey from a predator, we find that recommenders can guess the missing prey around 50% of the times on the first try, with up to 881 possibilities. Traits do not improve significantly the results for the K nearest neighbour, although a simple test with a supervised learning approach (random forests) show we can predict interactions with high accuracy using only three traits per species. This result shows that binary interactions can be predicted without regard to the ecological community given only three variables: body mass and two variables for the species’ phylogeny. These techniques are complementary, as recommenders can predict interactions in the absence of traits, using only information about other species’ interactions, while supervised learning algorithms such as random forests base their predictions on traits only but do not exploit other species’ interactions. Further work should focus on developing custom similarity measures specialized for ecology to improve the KNN algorithms and using richer data to capture indirect relationships between species.;,citation_author=Philippe Desjardins-Proulx;,citation_author=Idaline Laigle;,citation_author=Timothée Poisot;,citation_author=Dominique Gravel;,citation_publication_date=2017;,citation_cover_date=2017;,citation_year=2017;,citation_doi=10.7717/peerj.3644;,citation_issn=2167-8359;,citation_pmid=28828250;,citation_volume=5;,citation_language=en-US;,citation_journal_title=PeerJ;">
<meta name="citation_reference" content="citation_title=The rise, and possible fall, of network ecology;,citation_abstract=Ecology, much like any other discipline, has its fashions and fads, and each brings its own buzzwords and jargon. “Network ecology” is an example of such a fashion, which has for a few decades imprinted heavily on ecological publications. However, the topics of network ecology are of course much older, as are most of its methods. This invites the question whether network ecology is on a good path to providing dierent viewpoints and new insights. I here try to outline a somewhat opinionated view of why there is a high risk that this avenue of research may prove to be a cul-de-sac, for two reasons. On the one hand, the word “network” has become an empty buzzword void of specic meaning. On the other, there are six problems that I deem to be “deal breakers” in research on interaction networks: unless they all are resolved, this approach cannot make meaningful contributions. They are: () sampling bias; () ecological meaning of recorded interactions; () data aggregation over individuals; () lack of quantitative expectation; and () ecologically meaningless indices. Together they lead to the biggest problem () confusion in what it all means ecologically. Until these issues are being tackled by improved eld and computational research, there seem to be little progress possible in our understanding of assemblages of interacting species under the header of “network ecology”.;,citation_author=Carsten F Dormann;,citation_publication_date=2023;,citation_cover_date=2023;,citation_year=2023;,citation_language=en-US;,citation_inbook_title=Defining Agroecology – A Festschrift for Teja Tscharntke;">
<meta name="citation_reference" content="citation_title=A probabilistic view of forbidden links: Their prevalence and their consequences for the robustness of mutualistic networks;,citation_abstract=Understanding how species extinctions affect communities of interacting species is an important challenge of ecology. The presence of unfeasible interactions, termed forbidden links, due to physiological or morphological barriers, is likely to decrease the plasticity of interaction networks, affecting their robustness to species extinctions. However, the existence of these forbidden links has been long debated, stressing the need to learn more about their prevalence and putative consequences for community robustness to extinctions. To tackle this gap, we used a dataset of plant-hummingbird interactions collected in Brazil, Costa Rica and Ecuador along elevation gradients. We used a Bayesian hierarchical model, to assess the probabilistic importance of exploitation barriers in determining species interactions. We found evidence for exploitation barriers between flowers and hummingbirds with a shorter bill than the corolla. We also showed that the proportion of forbidden links can change drastically among communities, because of changes in trait distributions. Finally we showed that networks with a high proportion of forbidden links had low robustness, because of constraints on interaction rewiring. Our results suggest that exploitation barriers are not rare and strongly limit interaction rewiring, and thus the rescue of species experiencing partner extinction may be limited.;,citation_author=François Duchenne;,citation_author=Elisa Barreto;,citation_author=Esteban A. Guevara;,citation_author=Holger Beck;,citation_author=Carolina Bello;,citation_author=Rafaela Bobato;,citation_author=Daniela Bôlla;,citation_author=Emanuel Brenes;,citation_author=Nicole Büttner;,citation_author=Ana P. Caron;,citation_author=Nelson Elizondo;,citation_author=Alejandro Restrepo-González;,citation_author=José A. Castro;,citation_author=Miriam Kaehler;,citation_author=Tiago Machado-de-Souza;,citation_author=Miguel Machnicki-Reis;,citation_author=Andrés S. Marcayata-Fajardo;,citation_author=Cauã G. De Menezes;,citation_author=Andrea Nieto;,citation_author=Rafael De Oliveira;,citation_author=Ricardo A. C. De Oliveira;,citation_author=Friederike Richter;,citation_author=Bryan G. Rojas;,citation_author=Luciele L. Romanowski;,citation_author=Wellinton L De Souza;,citation_author=Danila S. Veluza;,citation_author=Ben Weinstein;,citation_author=Rafael Wüest;,citation_author=Thais B. Zanata;,citation_author=Krystal Zuniga;,citation_author=María A. Maglianesi;,citation_author=Tatiana Santander;,citation_author=Isabella G. Varassin;,citation_author=Catherine H. Graham;,citation_publication_date=2024-07;,citation_cover_date=2024-07;,citation_year=2024;,citation_doi=10.1101/2024.07.04.602032;,citation_language=en-US;">
<meta name="citation_reference" content="citation_title=Compilation and Network Analyses of Cambrian Food Webs;,citation_abstract=A rich body of empirically grounded theory has developed about food webs—the networks of feeding relationships among species within habitats. However, detailed food-web data and analyses are lacking for ancient ecosystems, largely because of the low resolution of taxa coupled with uncertain and incomplete information about feeding interactions. These impediments appear insurmountable for most fossil assemblages; however, a few assemblages with excellent soft-body preservation across trophic levels are candidates for food-web data compilation and topological analysis. Here we present plausible, detailed food webs for the Chengjiang and Burgess Shale assemblages from the Cambrian Period. Analyses of degree distributions and other structural network properties, including sensitivity analyses of the effects of uncertainty associated with Cambrian diet designations, suggest that these early Paleozoic communities share remarkably similar topology with modern food webs. Observed regularities reflect a systematic dependence of structure on the numbers of taxa and links in a web. Most aspects of Cambrian food-web structure are well-characterized by a simple “niche model,” which was developed for modern food webs and takes into account this scale dependence. However, a few aspects of topology differ between the ancient and recent webs: longer path lengths between species and more species in feeding loops in the earlier Chengjiang web, and higher variability in the number of links per species for both Cambrian webs. Our results are relatively insensitive to the exclusion of low-certainty or random links. The many similarities between Cambrian and recent food webs point toward surprisingly strong and enduring constraints on the organization of complex feeding interactions among metazoan species. The few differences could reflect a transition to more strongly integrated and constrained trophic organization within ecosystems following the rapid diversification of species, body plans, and trophic roles during the Cambrian radiation. More research is needed to explore the generality of food-web structure through deep time and across habitats, especially to investigate potential mechanisms that could give rise to similar structure, as well as any differences.;,citation_author=Jennifer A. Dunne;,citation_author=Richard J. Williams;,citation_author=Neo D. Martinez;,citation_author=Rachel A. Wood;,citation_author=Douglas H. Erwin;,citation_publication_date=2008-04;,citation_cover_date=2008-04;,citation_year=2008;,citation_issue=4;,citation_doi=10.1371/journal.pbio.0060102;,citation_issn=1545-7885;,citation_volume=6;,citation_language=en-US;,citation_journal_title=PLOS Biology;,citation_publisher=Public Library of Science;">
<meta name="citation_reference" content="citation_title=The Network Structure of Food Webs;,citation_author=Jennifer A Dunne;,citation_editor=Jennifer A Dunne;,citation_editor=Mercedes Pascual;,citation_publication_date=2006;,citation_cover_date=2006;,citation_year=2006;,citation_inbook_title=Ecological networks: Linking structure and dynamics;">
<meta name="citation_reference" content="citation_title=Network structure and robustness of marine food webs;,citation_abstract=Previous studies suggest that food-web theory has yet to account for major differences in food-web properties of marine versus other types of ecosystems. We examined this issue by analyzing the network structure of food webs for the Northeast US Shelf, a Caribbean reef, and Benguela, off South Africa. The values of connectance (links per species2), link density (links per species), mean chain length, and fractions of intermediate, omnivorous, and cannibalistic taxa of these marine webs are somewhat high but still within the ranges observed in other webs. We further compared the marine webs by using the empirically corroborated “niche model” that accounts for observed variation in diversity (taxon number) and complexity (connectance). Our results substantiate previously reported results for estuarine, fresh-water, and terrestrial datasets, which suggests that food webs from different types of ecosystems with variable diversity and complexity share fundamental structural and ordering characteristics. Analyses of potential secondary extinctions resulting from species loss show that the structural robustness of marine food webs is also consistent with trends from other food webs. As expected, given their relatively high connectance, marine food webs appear fairly robust to loss of most-connected taxa as well as random taxa. Still, the short average path length between marine taxa (1.6 links) suggests that effects from perturbations, such as overfishing, can be transmitted more widely throughout marine ecosystems than previously appreciated.;,citation_author=Jennifer A. Dunne;,citation_author=Richard J. Williams;,citation_author=Neo D. Martinez;,citation_publication_date=2004;,citation_cover_date=2004;,citation_year=2004;,citation_doi=10.3354/meps273291;,citation_issn=0171-8630, 1616-1599;,citation_volume=273;,citation_language=en-US;,citation_journal_title=Marine Ecology Progress Series;">
<meta name="citation_reference" content="citation_title=The sixth mass coextinction: Are most endangered species parasites and mutualists?;,citation_abstract=The effects of species declines and extinction on biotic interactions remain poorly understood. The loss of a species is expected to result in the loss of other species that depend on it (coextinction), leading to cascading effects across trophic levels. Such effects are likely to be most severe in mutualistic and parasitic interactions. Indeed, models suggest that coextinction may be the most common form of biodiversity loss. Paradoxically, few historical or contemporary coextinction events have actually been recorded. We review the current knowledge of coextinction by: (i) considering plausible explanations for the discrepancy between predicted and observed coextinction rates; (ii) exploring the potential consequences of coextinctions; (iii) discussing the interactions and synergies between coextinction and other drivers of species loss, particularly climate change; and (iv) suggesting the way forward for understanding the phenomenon of coextinction, which may well be the most insidious threat to global biodiversity.;,citation_author=Robert R. Dunn;,citation_author=Nyeema C. Harris;,citation_author=Robert K. Colwell;,citation_author=Lian Pin Koh;,citation_author=Navjot S. Sodhi;,citation_publication_date=2009-09;,citation_cover_date=2009-09;,citation_year=2009;,citation_issue=1670;,citation_doi=10.1098/rspb.2009.0413;,citation_issn=0962-8452;,citation_pmid=19474041;,citation_volume=276;,citation_language=en-US;,citation_journal_title=Proceedings. Biological Sciences;">
<meta name="citation_reference" content="citation_title=Potential extinction cascades in a desert ecosystem: Linking food web interactions to community viability;,citation_abstract=Desert communities are threatened with species loss due to climate change, and their resistance to such losses is unknown. We constructed a food web of the Mojave Desert terrestrial community (300 nodes, 4080 edges) to empirically examine the potential cascading effects of bird extinctions on this desert network, compared to losses of mammals and lizards. We focused on birds because they are already disappearing from the Mojave, and their relative thermal vulnerabilities are known. We quantified bottom-up secondary extinctions and evaluated the relative resistance of the community to losses of each vertebrate group. The impact of random bird species loss was relatively low compared to the consequences of mammal (causing the greatest number of cascading losses) or reptile loss, and birds were relatively less likely to be in trophic positions that could drive top-down effects in apparent competition and tri-tropic cascade motifs. An avian extinction cascade with year-long resident birds caused more secondary extinctions than the cascade involving all bird species for randomized ordered extinctions. Notably, we also found that relatively high interconnectivity among avian species has formed a subweb, enhancing network resistance to bird losses.;,citation_author=Adam J. Eichenwald;,citation_author=Nina H. Fefferman;,citation_author=J. Michael Reed;,citation_publication_date=2024;,citation_cover_date=2024;,citation_year=2024;,citation_issue=2;,citation_doi=10.1002/ece3.10930;,citation_issn=2045-7758;,citation_volume=14;,citation_language=en-US;,citation_journal_title=Ecology and Evolution;">
<meta name="citation_reference" content="citation_title=The dimensionality of ecological networks;,citation_abstract=How many dimensions (trait-axes) are required to predict whether two species interact? This unanswered question originated with the idea of ecological niches, and yet bears relevance today for understanding what determines network structure. Here, we analyse a set of 200 ecological networks, including food webs, antagonistic and mutualistic networks, and find that the number of dimensions needed to completely explain all interactions is small ( &amp;amp;amp;lt; 10), with model selection favouring less than five. Using 18 high-quality webs including several species traits, we identify which traits contribute the most to explaining network structure. We show that accounting for a few traits dramatically improves our understanding of the structure of ecological networks. Matching traits for resources and consumers, for example, fruit size and bill gape, are the most successful combinations. These results link ecologically important species attributes to large-scale community structure.;,citation_author=Anna Eklöf;,citation_author=Ute Jacob;,citation_author=Jason Kopp;,citation_author=Jordi Bosch;,citation_author=Rocío Castro-Urgal;,citation_author=Natacha P. Chacoff;,citation_author=Bo Dalsgaard;,citation_author=Claudio Sassi;,citation_author=Mauro Galetti;,citation_author=Paulo R. Guimarães;,citation_author=Silvia Beatriz Lomáscolo;,citation_author=Ana M. Martín González;,citation_author=Marco Aurelio Pizo;,citation_author=Romina Rader;,citation_author=Anselm Rodrigo;,citation_author=Jason M. Tylianakis;,citation_author=Diego P. Vázquez;,citation_author=Stefano Allesina;,citation_publication_date=2013;,citation_cover_date=2013;,citation_year=2013;,citation_issue=5;,citation_doi=10.1111/ele.12081;,citation_issn=1461-0248;,citation_volume=16;,citation_language=en-US;,citation_journal_title=Ecology Letters;">
<meta name="citation_reference" content="citation_title=Secondary extinctions in food webs: A Bayesian network approach;,citation_abstract=Summary Ecological communities are composed of populations connected in tangled networks of ecological interactions. Therefore, the extinction of a species can reverberate through the network and cause other (possibly distantly connected) species to go extinct as well. The study of these secondary extinctions is a fertile area of research in ecological network theory. However, to facilitate practical applications, several improvements to the current analytical approaches are needed. In particular, we need to consider that (i) species have different “a priori” probabilities of extinction, (ii) disturbances can simultaneously affect several species, and (iii) extinction risk of consumers likely grows with resource loss. All these points can be included in dynamical models, which are, however, difficult to parameterize. Here we advance the study of secondary extinctions with Bayesian networks. We show how this approach can account for different extinction responses using binary – where each resource has the same importance – and quantitative data – where resources are weighted by their importance. We simulate ecological networks using a popular dynamical model (the Allometric Trophic Network model) and use it to test our method. We find that the Bayesian network model captures the majority of the secondary extinctions produced by the dynamical model and that consumers’ responses to species loss are best modelled using a nonlinear sigmoid function. We also show that an approach based exclusively on food web structure loses power when species at higher trophic levels are preferentially lost. Because the loss of apex predators is unfortunately widespread, the results highlight a serious limitation of studies on network robustness.;,citation_author=Anna Eklöf;,citation_author=Si Tang;,citation_author=Stefano Allesina;,citation_editor=Jessica Metcalf;,citation_publication_date=2013-08;,citation_cover_date=2013-08;,citation_year=2013;,citation_issue=8;,citation_doi=10.1111/2041-210X.12062;,citation_issn=2041-210X, 2041-210X;,citation_volume=4;,citation_language=en-US;,citation_journal_title=Methods in Ecology and Evolution;">
<meta name="citation_reference" content="citation_title=On Random Graphs I;,citation_author=Paul Erdős;,citation_author=Alfréd Rényi;,citation_publication_date=1959;,citation_cover_date=1959;,citation_year=1959;,citation_doi=10.5486/PMD.1959.6.3-4.12;,citation_journal_title=Publicationes Mathematicae;">
<meta name="citation_reference" content="citation_title=Editorial: Patterns and processes in ecological networks over space;,citation_author=Sergio A. Estay;,citation_author=Marie-Josée Fortin;,citation_author=Daniela N. López;,citation_publication_date=2023;,citation_cover_date=2023;,citation_year=2023;,citation_issn=2296-701X;,citation_volume=11;,citation_journal_title=Frontiers in Ecology and Evolution;">
<meta name="citation_reference" content="citation_title=Graphs.jl: An optimized graphs package for the Julia programming language;,citation_author=James Fairbanks;,citation_author=Mathieu Besançon;,citation_author=Simon Schölly;,citation_author=Júlio Hoffiman;,citation_author=Nick Eubank;,citation_author=Stefan Karpinski;,citation_publication_date=2021;,citation_cover_date=2021;,citation_year=2021;">
<meta name="citation_reference" content="citation_title=Network ecology in dynamic landscapes;,citation_abstract=Network ecology is an emerging field that allows researchers to conceptualize and analyse ecological networks and their dynamics. Here, we focus on the dynamics of ecological networks in response to environmental changes. Specifically, we formalize how network topologies constrain the dynamics of ecological systems into a unifying framework in network ecology that we refer to as the “ecological network dynamics framework”. This framework stresses that the interplay between species interaction networks and the spatial layout of habitat patches is key to identifying which network properties (number and weights of nodes and links) and trade-offs among them are needed to maintain species interactions in dynamic landscapes. We conclude that to be functional, ecological networks should be scaled according to species dispersal abilities in response to landscape heterogeneity. Determining how such effective ecological networks change through space and time can help reveal their complex dynamics in a changing world.;,citation_author=Marie-Josée Fortin;,citation_author=Mark R. T. Dale;,citation_author=Chris Brimacombe;,citation_publication_date=2021-04;,citation_cover_date=2021-04;,citation_year=2021;,citation_issue=1949;,citation_doi=10.1098/rspb.2020.1889;,citation_issn=0962-8452, 1471-2954;,citation_volume=288;,citation_language=en-US;,citation_journal_title=Proceedings of the Royal Society B: Biological Sciences;">
<meta name="citation_reference" content="citation_title=Spatial statistics, spatial regression, and graph theory in ecology;,citation_abstract=A critical part of ecological studies is to quantify how landscape spatial heterogeneity affects species’ distributions. With advancements in remote sensing technology and GIS, we now live in a data-rich era allowing us to investigate species–environment relationships in heterogeneous landscapes at multiple spatial scales. However, the degree and type of spatial heterogeneity changes depending on the spatial scale at which species–environment relationships are analysed. Here we present the current spatial analytic methods used in ecological studies to quantify ecological spatial heterogeneity. To determine the key spatial scales at which underlying ecological processes act upon species, we recommend use of spectral decomposition techniques such as wavelet analysis or Moran’s eigenvector maps. Following this, a suite of spatial regression methods can be used to quantify the relative influence of environmental factors on species’ distributions. Finally, spatial graph metrics can be employed to quantify the effects of spatial heterogeneity on landscape connectivity across or within species’ ranges and can be used as additional predictors in spatial regression models. We emphasize how spatial statistics, spatial regression, and spatial graph theory can be used to provide insights into how landscape spatial complexity influences species distributions and to better understand species response to global change.;,citation_author=Marie-Josée Fortin;,citation_author=Patrick M. A. James;,citation_author=Alistair MacKenzie;,citation_author=Stephanie J. Melles;,citation_author=Bronwyn Rayfield;,citation_publication_date=2012-05;,citation_cover_date=2012-05;,citation_year=2012;,citation_doi=10.1016/j.spasta.2012.02.004;,citation_issn=2211-6753;,citation_volume=1;,citation_journal_title=Spatial Statistics;">
<meta name="citation_reference" content="citation_title=Habitat loss and the structure of plant-animal mutualistic networks: Mutualistic networks and habitat loss;,citation_author=Miguel A. Fortuna;,citation_author=Jordi Bascompte;,citation_publication_date=2006-03;,citation_cover_date=2006-03;,citation_year=2006;,citation_issue=3;,citation_doi=10.1111/j.1461-0248.2005.00868.x;,citation_issn=1461023X;,citation_volume=9;,citation_language=en-US;,citation_journal_title=Ecology Letters;">
<meta name="citation_reference" content="citation_title=Collapse of terrestrial mammal food webs since the Late Pleistocene;,citation_abstract=Food webs influence ecosystem diversity and functioning. Contemporary defaunation has reduced food web complexity, but simplification caused by past defaunation is difficult to reconstruct given the sparse paleorecord of predator-prey interactions. We identified changes to terrestrial mammal food webs globally over the past ~130,000 years using extinct and extant mammal traits, geographic ranges, observed predator-prey interactions, and deep learning models. Food webs underwent steep regional declines in complexity through loss of food web links after the arrival and expansion of human populations. We estimate that defaunation has caused a 53% decline in food web links globally. Although extinctions explain much of this effect, range losses for extant species degraded food webs to a similar extent, highlighting the potential for food web restoration via extant species recovery.;,citation_author=Evan C. Fricke;,citation_author=Chia Hsieh;,citation_author=Owen Middleton;,citation_author=Daniel Gorczynski;,citation_author=Caroline D. Cappello;,citation_author=Oscar Sanisidro;,citation_author=John Rowan;,citation_author=Jens-Christian Svenning;,citation_author=Lydia Beaudrot;,citation_publication_date=2022-08;,citation_cover_date=2022-08;,citation_year=2022;,citation_issue=6609;,citation_doi=10.1126/science.abn4012;,citation_volume=377;,citation_journal_title=Science;,citation_publisher=American Association for the Advancement of Science;">
<meta name="citation_reference" content="citation_title=ggVennDiagram: An Intuitive, Easy-to-Use, and Highly Customizable R Package to Generate Venn Diagram;,citation_abstract=Venn diagrams are widely used diagrams to show the set relationships in biomedical studies. In this study, we developed ggVennDiagram, an R package that could automatically generate high-quality Venn diagrams with two to seven sets. The ggVennDiagram is built based on ggplot2, and it integrates the advantages of existing packages, such as venn, RVenn, VennDiagram, and sf. Satisfactory results can be obtained with minimal configurations. Furthermore, we designed comprehensive objects to store the entire data of the Venn diagram, which allowed free access to both intersection values and Venn plot sub-elements, such as set label/edge and region label/filling. Therefore, high customization of every Venn plot sub-element can be fulfilled without increasing the cost of learning when the user is familiar with ggplot2 methods. To date, ggVennDiagram has been cited in more than 10 publications, and its source code repository has been starred by more than 140 GitHub users, suggesting a great potential in applications. The package is an open-source software released under the GPL-3 license, and it is freely available through CRAN (https://cran.r-project.org/package=ggVennDiagram).;,citation_author=Chun-Hui Gao;,citation_author=Guangchuang Yu;,citation_author=Peng Cai;,citation_publication_date=2021;,citation_cover_date=2021;,citation_year=2021;,citation_issn=1664-8021;,citation_volume=12;,citation_journal_title=Frontiers in Genetics;">
<meta name="citation_reference" content="citation_title=Adaptive rewiring aggravates the effects of species loss in ecosystems;,citation_author=David Gilljam;,citation_author=Alva Curtsdotter;,citation_author=Bo Ebenman;,citation_publication_date=2015-09;,citation_cover_date=2015-09;,citation_year=2015;,citation_issue=1;,citation_doi=10.1038/ncomms9412;,citation_issn=2041-1723;,citation_volume=6;,citation_language=en-US;,citation_journal_title=Nature Communications;">
<meta name="citation_reference" content="citation_title=Modifying modifiers: What happens when interspecific interactions interact?;,citation_abstract=1. The strength of the trophic link between any given pair of species in a food web is likely to depend on the presence and/or densities of other species in the community. How these trophic interaction modifications (TIMs) interact with one another to produce a net modifying effect is an important but under-explored issue. 2. We review several specific types of TIMs that are well understood to address whether the magnitude of the net modification changes with the number of modifiers, and whether modifiers usually increase or decrease each other’s effects. 3. Modifications of interactions are generally not independent. It is likely that TIMs interact antagonistically in the majority of cases; the magnitudes of TIMs decrease as more modifiers are added, or new TIMs reduce the magnitudes of modifications that are already present. 4. Individual modifications are likely to have a smaller effect in many-species systems than expected from independent combination of modifications measured in systems with relatively few species. Thus, models that lack explicit TIMs may in some cases yield adequate predictions for species-level perturbations, provided that the net effects of TIMs are implicitly included in measured interaction strengths. 5. Many types of TIMs share structural similarities. Nevertheless, a complete understanding of their effects may require theory that distinguishes different “functional groups” of modifiers and addresses how these are structured according to trophic relationships.;,citation_author=Antonio J. Golubski;,citation_author=Peter A. Abrams;,citation_publication_date=2011;,citation_cover_date=2011;,citation_year=2011;,citation_issue=5;,citation_doi=10.1111/j.1365-2656.2011.01852.x;,citation_issn=1365-2656;,citation_volume=80;,citation_language=en-US;,citation_journal_title=Journal of Animal Ecology;">
<meta name="citation_reference" content="citation_title=Ecological interactions are evolutionarily conserved across the entire tree of life;,citation_abstract=The characteristics that regulate species interactions are largely inherited, so it seems logical to expect that closely related organisms are more likely to have similar ecological interactions than less related ones. Few studies have been done on this relationship, and those that have tend to focus on specialized organisms, such as parasites or insect herbivores. A new analysis of the evolution of host use in a diverse group of interactions comprising both specialist and generalist, acellular, unicellular and multicellular organisms, and including all types of interaction, finds support for the evolutionary conservation of ecological interactions across all species. The same rules seem to drive the evolution of most ecological interactions, and are strongly contributing to the organization of biodiversity on Earth.;,citation_author=José M. Gómez;,citation_author=Miguel Verdú;,citation_author=Francisco Perfectti;,citation_publication_date=2010-06;,citation_cover_date=2010-06;,citation_year=2010;,citation_issue=7300;,citation_doi=10.1038/nature09113;,citation_issn=1476-4687;,citation_volume=465;,citation_language=en-US;,citation_journal_title=Nature;,citation_publisher=Nature Publishing Group;">
<meta name="citation_reference" content="citation_title=Inferring food web structure from predator–prey body size relationships;,citation_abstract=Current global changes make it important to be able to predict which interactions will occur in the emerging ecosystems. Most of the current methods to infer the existence of interactions between two species require a good knowledge of their behaviour or a direct observation of interactions. In this paper, we overcome these limitations by developing a method, inspired from the niche model of food web structure, using the statistical relationship between predator and prey body size to infer the matrix of potential interactions among a pool of species. The novelty of our approach is to infer, for any species of a given species pool, the three species-specific parameters of the niche model. The method applies to both local and metaweb scales. It allows one to evaluate the feeding interactions of a new species entering the community. We find that this method gives robust predictions of the structure of food webs and that its efficiency is increased when the strength of the body–size relationship between predators and preys increases. We finally illustrate the potential of the method to infer the metaweb structure of pelagic fishes of the Mediterranean sea under different global change scenarios.;,citation_author=Dominique Gravel;,citation_author=Timothée Poisot;,citation_author=Camille Albouy;,citation_author=Laure Velez;,citation_author=David Mouillot;,citation_publication_date=2013;,citation_cover_date=2013;,citation_year=2013;,citation_issue=11;,citation_doi=10.1111/2041-210X.12103;,citation_issn=2041-210X;,citation_volume=4;,citation_language=en-US;,citation_journal_title=Methods in Ecology and Evolution;">
<meta name="citation_reference" content="citation_title=FORUM: Ecological networks: The missing links in biomonitoring science;,citation_abstract=Monitoring anthropogenic impacts is essential for managing and conserving ecosystems, yet current biomonitoring approaches lack the tools required to deal with the effects of stressors on species and their interactions in complex natural systems. Ecological networks (trophic or mutualistic) can offer new insights into ecosystem degradation, adding value to current taxonomically constrained schemes. We highlight some examples to show how new network approaches can be used to interpret ecological responses. Synthesis and applications. Augmenting routine biomonitoring data with interaction data derived from the literature, complemented with ground-truthed data from direct observations where feasible, allows us to begin to characterise large numbers of ecological networks across environmental gradients. This process can be accelerated by adopting emerging technologies and novel analytical approaches, enabling biomonitoring to move beyond simple pass/fail schemes and to address the many ecological responses that can only be understood from a network-based perspective.;,citation_author=Clare Gray;,citation_author=Donald J. Baird;,citation_author=Simone Baumgartner;,citation_author=Ute Jacob;,citation_author=Gareth B. Jenkins;,citation_author=Eoin J. O’Gorman;,citation_author=Xueke Lu;,citation_author=Athen Ma;,citation_author=Michael J. O. Pocock;,citation_author=Nele Schuwirth;,citation_author=Murray Thompson;,citation_author=Guy Woodward;,citation_publication_date=2014;,citation_cover_date=2014;,citation_year=2014;,citation_issue=5;,citation_doi=10.1111/1365-2664.12300;,citation_issn=1365-2664;,citation_volume=51;,citation_language=en-US;,citation_journal_title=Journal of Applied Ecology;">
<meta name="citation_reference" content="citation_title=Joining the dots: An automated method for constructing food webs from compendia of published interactions;,citation_abstract=Food webs are important tools for understanding how complex natural communities are structured and how they respond to environmental change. However their full potential has yet to be realised because of the huge amount of resources required to construct them de novo. Consequently, the current catalogue of networks that are suitable for rigorous and comparative analyses and theoretical development still suffers from a lack of standardisation and replication. Here, we present a novel R function, WebBuilder, which automates the construction of food webs from taxonomic lists, and a dataset of trophic interactions. This function works by matching species against those within a dataset of trophic interactions, and “filling in” missing trophic interactions based on these matches. We also present a dataset of over 20,000 freshwater trophic interactions, and use this and four well-characterised freshwater food webs to test the method. The WebBuilder function facilitates the generation of food webs of comparable quality to the most detailed published food webs, but at a fraction of the research effort or cost. Furthermore, it matched and often outperformed a selection of predictive models, which are currently among the best, in terms of capturing key properties of empirical food webs. The method is simple to use, systematic and, perhaps most importantly, reproducible, which will facilitate (re-) analysis and data sharing. Although developed and tested on a sample of freshwater food webs, this method could easily be extended to cover other types of ecological interactions (such as mutualistic interactions).;,citation_author=Clare Gray;,citation_author=David H. Figueroa;,citation_author=Lawrence N. Hudson;,citation_author=Athen Ma;,citation_author=Dan Perkins;,citation_author=Guy Woodward;,citation_publication_date=2015-12;,citation_cover_date=2015-12;,citation_year=2015;,citation_doi=10.1016/j.fooweb.2015.09.001;,citation_issn=2352-2496;,citation_volume=5;,citation_journal_title=Food Webs;">
<meta name="citation_reference" content="citation_title=Babel, or the ecological stability discussions: An inventory and analysis of terminology and a guide for avoiding confusion;,citation_abstract=We present an inventory and analysis of discussions of ecological stability, considering 163 definitions of 70 different stability concepts. Our aim is to derive a strategy that can help to dispel the existing ”confusion of tongues” on the subject of ”stability” and prevent its future recurrence. The strategy consists of three questions that should be kept in mind when communicating about stability properties. These three questions should overcome the three main sources of confusion in terminology. Firstly, which stability properties are being addressed in the stability statement? Our analysis shows that the general term ”stability” is so ambiguous as to be useless.It can be replaced by the stability properties ”staying essentially unchanged” (constancy), ”returning to the reference state (or dynamic) after a temporary disturbance” (resilience), and ”persistence through time of an ecological system” (persistence). Second, to what ecological situation does the statement refer? An ecological situation is defined by a set of features that, taken as a whole, determine the domain of validity of a stability statement. The six most important features form the ”ecological checklist”, which serves to classify ecological situations and thereby provides a system of coordinates for communication. The six points are: variable of interest, level of description, reference state, disturbance, spatial scale and temporal scale. Thirdly, is the statement anchored in the situation in question, or is there unacceptable generalisation by inferring ”stability” of the whole system from a certain stability property in a certain ecological ecological situation? This question separates the scientifically valuable content of a statement from the desire for general statements which is often projected through stability statements.;,citation_author=V. Grimm;,citation_author=Christian Wissel;,citation_publication_date=1997-02;,citation_cover_date=1997-02;,citation_year=1997;,citation_issue=3;,citation_doi=10.1007/s004420050090;,citation_issn=1432-1939;,citation_volume=109;,citation_language=en-US;,citation_journal_title=Oecologia;">
<meta name="citation_reference" content="citation_title=The Structure of Ecological Networks Across Levels of Organization;,citation_abstract=Interactions connect the units of ecological systems, forming networks. Individual-based networks characterize variation in niches among individuals within populations. These individual-based networks merge with each other, forming species-based networks and food webs that describe the architecture of ecological communities. Networks at broader spatiotemporal scales portray the structure of ecological interactions across landscapes and over macroevolutionary time. Here, I review the patterns observed in ecological networks across multiple levels of biological organization. A fundamental challenge is to understand the amount of interdependence as we move from individual-based networks to species-based networks and beyond. Despite the uneven distribution of studies, regularities in network structure emerge across scales due to the fundamental architectural patterns shared by complex networks and the interplay between traits and numerical effects. I illustrate the integration of these organizational scales by exploring the consequences of the emergence of highly connected species for network structures across scales. Expected final online publication date for the Annual Review of Ecology, Evolution, and Systematics, Volume 51 is November 2, 2020. Please see http://www.annualreviews.org/page/journal/pubdates for revised estimates.;,citation_author=Paulo R. Guimarães;,citation_publication_date=2020-11;,citation_cover_date=2020-11;,citation_year=2020;,citation_issue=1;,citation_doi=10.1146/annurev-ecolsys-012220-120819;,citation_issn=1543-592X, 1545-2069;,citation_volume=51;,citation_language=en-US;,citation_journal_title=Annual Review of Ecology, Evolution, and Systematics;">
<meta name="citation_reference" content="citation_title=AnimalTraits - a curated animal trait database for body mass, metabolic rate and brain size;,citation_abstract=Trait databases have become important resources for large-scale comparative studies in ecology and evolution. Here we introduce the AnimalTraits database, a curated database of body mass, metabolic rate and brain size, in standardised units, for terrestrial animals. The database has broad taxonomic breadth, including tetrapods, arthropods, molluscs and annelids from almost 2000 species and 1000 genera. All data recorded in the database are sourced from their original empirical publication, and the original metrics and measurements are included with each record. This allows for subsequent data transformations as required. We have included rich metadata to allow users to filter the dataset. The additional R scripts we provide will assist researchers with aggregating standardised observations into species-level trait values. Our goals are to provide this resource without restrictions, to keep the AnimalTraits database current, and to grow the number of relevant traits in the future.;,citation_author=Marie E. Herberstein;,citation_author=Donald James McLean;,citation_author=Elizabeth Lowe;,citation_author=Jonas O. Wolff;,citation_author=Md Kawsar Khan;,citation_author=Kaitlyn Smith;,citation_author=Andrew P. Allen;,citation_author=Matthew Bulbert;,citation_author=Bruno A. Buzatto;,citation_author=Mark D. B. Eldridge;,citation_author=Daniel Falster;,citation_author=Laura Fernandez Winzer;,citation_author=Simon C. Griffith;,citation_author=Joshua S. Madin;,citation_author=Ajay Narendra;,citation_author=Mark Westoby;,citation_author=Martin J. Whiting;,citation_author=Ian J. Wright;,citation_author=Alexandra J. R. Carthey;,citation_publication_date=2022-06;,citation_cover_date=2022-06;,citation_year=2022;,citation_issue=1;,citation_doi=10.1038/s41597-022-01364-9;,citation_issn=2052-4463;,citation_volume=9;,citation_language=en-US;,citation_journal_title=Scientific Data;,citation_publisher=Nature Publishing Group;">
<meta name="citation_reference" content="citation_title=Mismatch between IUCN range maps and species interactions data illustrated using the Serengeti food web;,citation_abstract=Background Range maps are a useful tool to describe the spatial distribution of species. However, they need to be used with caution, as they essentially represent a rough approximation of a species’ suitable habitats. When stacked together, the resulting communities in each grid cell may not always be realistic, especially when species interactions are taken into account. Here we show the extent of the mismatch between range maps, provided by the International Union for Conservation of Nature (IUCN), and species interactions data. More precisely, we show that local networks built from those stacked range maps often yield unrealistic communities, where species of higher trophic levels are completely disconnected from primary producers. Methodology We used the well-described Serengeti food web of mammals and plants as our case study, and identify areas of data mismatch within predators’ range maps by taking into account food web structure. We then used occurrence data from the Global Biodiversity Information Facility (GBIF) to investigate where data is most lacking. Results We found that most predator ranges comprised large areas without any overlapping distribution of their prey. However, many of these areas contained GBIF occurrences of the predator. Conclusions Our results suggest that the mismatch between both data sources could be due either to the lack of information about ecological interactions or the geographical occurrence of prey. We finally discuss general guidelines to help identify defective data among distributions and interactions data, and we recommend this method as a valuable way to assess whether the occurrence data that are being used, even if incomplete, are ecologically accurate.;,citation_author=Gracielle T. Higino;,citation_author=Francis Banville;,citation_author=Gabriel Dansereau;,citation_author=Norma Rocio Forero Muñoz;,citation_author=Fredric Windsor;,citation_author=Timothée Poisot;,citation_publication_date=2023-02;,citation_cover_date=2023-02;,citation_year=2023;,citation_doi=10.7717/peerj.14620;,citation_issn=2167-8359;,citation_volume=11;,citation_language=en-US;,citation_journal_title=PeerJ;,citation_publisher=PeerJ Inc.;">
<meta name="citation_reference" content="citation_title=Dealing with Class Imbalance using Thresholding;,citation_abstract=We propose thresholding as an approach to deal with class imbalance. We define the concept of thresholding as a process of determining a decision boundary in the presence of a tunable parameter. The threshold is the maximum value of this tunable parameter where the conditions of a certain decision are satisfied. We show that thresholding is applicable not only for linear classifiers but also for non-linear classifiers. We show that this is the implicit assumption for many approaches to deal with class imbalance in linear classifiers. We then extend this paradigm beyond linear classification and show how non-linear classification can be dealt with under this umbrella framework of thresholding. The proposed method can be used for outlier detection in many real-life scenarios like in manufacturing. In advanced manufacturing units, where the manufacturing process has matured over time, the number of instances (or parts) of the product that need to be rejected (based on a strict regime of quality tests) becomes relatively rare and are defined as outliers. How to detect these rare parts or outliers beforehand? How to detect combination of conditions leading to these outliers? These are the questions motivating our research. This paper focuses on prediction of outliers and conditions leading to outliers using classification. We address the problem of outlier detection using classification. The classes are good parts (those passing the quality tests) and bad parts (those failing the quality tests and can be considered as outliers). The rarity of outliers transforms this problem into a class-imbalanced classification problem.;,citation_author=Charmgil Hong;,citation_author=Rumi Ghosh;,citation_author=Soundar Srinivasan;,citation_publication_date=2016-07;,citation_cover_date=2016-07;,citation_year=2016;,citation_fulltext_html_url=https://arxiv.org/abs/1607.02705;,citation_doi=10.48550/arXiv.1607.02705;,citation_publisher=arXiv;">
<meta name="citation_reference" content="citation_title=Seven Shortfalls that Beset Large-Scale Knowledge of Biodiversity;,citation_abstract=Ecologists and evolutionary biologists are increasingly using big-data approaches to tackle questions at large spatial, taxonomic, and temporal scales. However, despite recent efforts to gather two centuries of biodiversity inventories into comprehensive databases, many crucial research questions remain unanswered. Here, we update the concept of knowledge shortfalls and review the tradeoffs between generality and uncertainty. We present seven key shortfalls of current biodiversity data. Four previously proposed shortfalls pinpoint knowledge gaps for species taxonomy (Linnean), distribution (Wallacean), abundance (Prestonian), and evolutionary patterns (Darwinian). We also redefine the Hutchinsonian shortfall to apply to the abiotic tolerances of species and propose new shortfalls relating to limited knowledge of species traits (Raunkiæran) and biotic interactions (Eltonian). We conclude with a general framework for the combined impacts and consequences of shortfalls of large-scale biodiversity knowledge for evolutionary and ecological research and consider ways of overcoming the seven shortfalls and dealing with the uncertainty they generate.;,citation_author=Joaquín Hortal;,citation_author=Francesco Bello;,citation_author=José Alexandre F. Diniz-Filho;,citation_author=Thomas M. Lewinsohn;,citation_author=Jorge M. Lobo;,citation_author=Richard J. Ladle;,citation_publication_date=2015-12;,citation_cover_date=2015-12;,citation_year=2015;,citation_issue=1;,citation_doi=10.1146/annurev-ecolsys-112414-054400;,citation_issn=1543-592X, 1545-2069;,citation_volume=46;,citation_language=en-US;,citation_journal_title=Annual Review of Ecology, Evolution, and Systematics;">
<meta name="citation_reference" content="citation_title=The Unified Neutral Theory of Biodiversity and Biogeography (MPB-32);,citation_abstract=Despite its supreme importance and the threat of its global crash, biodiversity remains poorly understood both empirically and theoretically. This ambitious book presents a new, general neutral theory to explain the origin, maintenance, and loss of biodiversity in a biogeographic context. Until now biogeography (the study of the geographic distribution of species) and biodiversity (the study of species richness and relative species abundance) have had largely disjunct intellectual histories. In this book, Stephen Hubbell develops a formal mathematical theory that unifies these two fields. When a speciation process is incorporated into Robert H. MacArthur and Edward O. Wilson’s now classical theory of island biogeography, the generalized theory predicts the existence of a universal, dimensionless biodiversity number. In the theory, this fundamental biodiversity number, together with the migration or dispersal rate, completely determines the steady-state distribution of species richness and relative species abundance on local to large geographic spatial scales and short-term to evolutionary time scales. Although neutral, Hubbell’s theory is nevertheless able to generate many nonobvious, testable, and remarkably accurate quantitative predictions about biodiversity and biogeography. In many ways Hubbell’s theory is the ecological analog to the neutral theory of genetic drift in genetics. The unified neutral theory of biogeography and biodiversity should stimulate research in new theoretical and empirical directions by ecologists, evolutionary biologists, and biogeographers.;,citation_author=Stephen P. Hubbell;,citation_publication_date=2001;,citation_cover_date=2001;,citation_year=2001;,citation_fulltext_html_url=https://www.jstor.org/stable/j.ctt7rj8w;,citation_isbn=978-0-691-02128-7;">
<meta name="citation_reference" content="citation_title=How to Invade an Ecological Network;,citation_abstract=Invasion science is in a state of paradox, having low predictability despite strong, identifiable covariates of invasion performance. We propose shifting the foundation metaphor of biological invasions from a linear filtering scheme to one that invokes complex adaptive networks. We link invasion performance and invasibility directly to the loss of network stability and indirectly to network topology through constraints from the emergence of the stability criterion in complex systems. We propose the wind vane of an invaded network – the major axis of its adjacency matrix – which reveals how species respond dynamically to invasions. We suggest that invasion ecology should steer away from comparative macroecological studies, to rather explore the ecological network centred on the focal species.;,citation_author=Cang Hui;,citation_author=David M. Richardson;,citation_publication_date=2019-02;,citation_cover_date=2019-02;,citation_year=2019;,citation_issue=2;,citation_doi=10.1016/j.tree.2018.11.003;,citation_issn=0169-5347;,citation_volume=34;,citation_journal_title=Trends in Ecology &amp;amp;amp; Evolution;">
<meta name="citation_reference" content="citation_title=Seeing the forest for the trees: Putting multilayer networks to work for community ecology;,citation_author=Matthew C. Hutchinson;,citation_author=Bernat Bramon Mora;,citation_author=Shai Pilosof;,citation_author=Allison K. Barner;,citation_author=Sonia Kéfi;,citation_author=Elisa Thébault;,citation_author=Pedro Jordano;,citation_author=Daniel B. Stouffer;,citation_editor=Oscar Godoy;,citation_publication_date=2019-02;,citation_cover_date=2019-02;,citation_year=2019;,citation_issue=2;,citation_doi=10.1111/1365-2435.13237;,citation_issn=0269-8463, 1365-2435;,citation_volume=33;,citation_language=en-US;,citation_journal_title=Functional Ecology;">
<meta name="citation_reference" content="citation_title=PanTHERIA: A species-level database of life history, ecology, and geography of extant and recently extinct mammals;,citation_abstract=Analyses of life-history, ecological, and geographic trait differences among species, their causes, correlates, and likely consequences are increasingly important for understanding and conserving biodiversity in the face of rapid global change. Assembling multispecies trait data from diverse literature sources into a single comprehensive data set requires detailed consideration of methods to reliably compile data for particular species, and to derive single estimates from multiple sources based on different techniques and definitions. Here we describe PanTHERIA, a species-level data set compiled for analysis of life history, ecology, and geography of all known extant and recently extinct mammals. PanTHERIA is derived from a database capable of holding multiple geo-referenced values for variables within a species containing 100 740 lines of biological data for extant and recently extinct mammalian species, collected over a period of three years by 20 individuals. PanTHERIA also includes spatial databases of mammalian geographic ranges and global climatic and anthropogenic variables. Here we detail how the data fields are extracted and defined for PanTHERIA using a customized data input format (MammalForm); how data were collected from the literature, species names and sources tracked, error-checking and validation procedures applied, and how data were consolidated into species-level values for each variable. Tables of the consolidated species-level values are made available for each of two recent species-level taxonomic classifications of mammals, as well as associated taxonomic synonymy conversion and data-input files. This study provides a useful guide to prospective researchers on how to structure and codify life-history, ecological, geographic, and taxonomic data and methods to extract meaningful species-level traits. It also provides comprehensive information on traits like size, diet, environmental conditions, and ecology to permit macroecological and macroevolutionary analyses of this important clade. The complete data sets corresponding to abstracts published in the Data Papers section of the journal are published electronically in Ecological Archives at http://esapubs.org/archive. (The accession number for each Data Paper is given directly beneath the title.);,citation_author=Kate E. Jones;,citation_author=Jon Bielby;,citation_author=Marcel Cardillo;,citation_author=Susanne A. Fritz;,citation_author=Justin O’Dell;,citation_author=C. David L. Orme;,citation_author=Kamran Safi;,citation_author=Wes Sechrest;,citation_author=Elizabeth H. Boakes;,citation_author=Chris Carbone;,citation_author=Christina Connolly;,citation_author=Michael J. Cutts;,citation_author=Janine K. Foster;,citation_author=Richard Grenyer;,citation_author=Michael Habib;,citation_author=Christopher A. Plaster;,citation_author=Samantha A. Price;,citation_author=Elizabeth A. Rigby;,citation_author=Janna Rist;,citation_author=Amber Teacher;,citation_author=Olaf R. P. Bininda-Emonds;,citation_author=John L. Gittleman;,citation_author=Georgina M. Mace;,citation_author=Andy Purvis;,citation_publication_date=2009;,citation_cover_date=2009;,citation_year=2009;,citation_issue=9;,citation_doi=10.1890/08-1494.1;,citation_issn=1939-9170;,citation_volume=90;,citation_language=en-US;,citation_journal_title=Ecology;">
<meta name="citation_reference" content="citation_title=Chasing Ecological Interactions;,citation_abstract=Basic research on biodiversity has concentrated on individual species—naming new species, studying distribution patterns, and analyzing their evolutionary relationships. Yet biodiversity is more than a collection of individual species; it is the combination of biological entities and processes that support life on Earth. To understand biodiversity we must catalog it, but we must also assess the ways species interact with other species to provide functional support for the Tree of Life. Ecological interactions may be lost well before the species involved in those interactions go extinct; their ecological functions disappear even though they remain. Here, I address the challenges in studying the functional aspects of species interactions and how basic research is helping us address the fast-paced extinction of species due to human activities.;,citation_author=Pedro Jordano;,citation_publication_date=2016-09;,citation_cover_date=2016-09;,citation_year=2016;,citation_issue=9;,citation_doi=10.1371/journal.pbio.1002559;,citation_issn=1545-7885;,citation_volume=14;,citation_language=en-US;,citation_journal_title=PLOS Biology;,citation_publisher=Public Library of Science;">
<meta name="citation_reference" content="citation_title=Sampling networks of ecological interactions;,citation_author=Pedro Jordano;,citation_publication_date=2016-09;,citation_cover_date=2016-09;,citation_year=2016;,citation_doi=10.1111/1365-2435.12763;,citation_issn=02698463;,citation_language=en-US;,citation_journal_title=Functional Ecology;">
<meta name="citation_reference" content="citation_title=More than a meal… integrating non-feeding interactions into food webs: More than a meal …;,citation_author=Sonia Kéfi;,citation_author=Eric L. Berlow;,citation_author=Evie A. Wieters;,citation_author=Sergio A. Navarrete;,citation_author=Owen L. Petchey;,citation_author=Spencer A. Wood;,citation_author=Alice Boit;,citation_author=Lucas N. Joppa;,citation_author=Kevin D. Lafferty;,citation_author=Richard J. Williams;,citation_author=Neo D. Martinez;,citation_author=Bruce A. Menge;,citation_author=Carol A. Blanchette;,citation_author=Alison C. Iles;,citation_author=Ulrich Brose;,citation_publication_date=2012-04;,citation_cover_date=2012-04;,citation_year=2012;,citation_issue=4;,citation_doi=10.1111/j.1461-0248.2011.01732.x;,citation_issn=1461023X;,citation_volume=15;,citation_language=en-US;,citation_journal_title=Ecology Letters;">
<meta name="citation_reference" content="citation_title=Network structure beyond food webs: Mapping non-trophic and trophic interactions on Chilean rocky shores;,citation_abstract=How multiple types of non-trophic interactions map onto trophic networks in real communities remains largely unknown. We present the first effort, to our knowledge, describing a comprehensive ecological network that includes all known trophic and diverse non-trophic links among &amp;amp;amp;gt;100 coexisting species for the marine rocky intertidal community of the central Chilean coast. Our results suggest that non-trophic interactions exhibit highly nonrandom structures both alone and with respect to food web structure. The occurrence of different types of interactions, relative to all possible links, was well predicted by trophic structure and simple traits of the source and target species. In this community, competition for space and positive interactions related to habitat/refuge provisioning by sessile and/or basal species were by far the most abundant non-trophic interactions. If these patterns are corroborated in other ecosystems, they may suggest potentially important dynamic constraints on the combined architecture of trophic and non-trophic interactions. The nonrandom patterning of non-trophic interactions suggests a path forward for developing a more comprehensive ecological network theory to predict the functioning and resilience of ecological communities.;,citation_author=Sonia Kéfi;,citation_author=Eric L. Berlow;,citation_author=Evie A. Wieters;,citation_author=Lucas N. Joppa;,citation_author=Spencer A. Wood;,citation_author=Ulrich Brose;,citation_author=Sergio A. Navarrete;,citation_publication_date=2015;,citation_cover_date=2015;,citation_year=2015;,citation_issue=1;,citation_doi=10.1890/13-1424.1;,citation_issn=1939-9170;,citation_volume=96;,citation_language=en-US;,citation_journal_title=Ecology;">
<meta name="citation_reference" content="citation_title=Compartments revealed in food-web structure;,citation_abstract=Compartments1 in food webs are subgroups of taxa in which many strong interactions occur within the subgroups and few weak interactions occur between the subgroups2. Theoretically, compartments increase the stability in networks1,2,3,4,5, such as food webs. Compartments have been difficult to detect in empirical food webs because of incompatible approaches6,7,8,9 or insufficient methodological rigour8,10,11. Here we show that a method for detecting compartments from the social networking science12,13,14 identified significant compartments in three of five complex, empirical food webs. Detection of compartments was influenced by food web resolution, such as interactions with weights. Because the method identifies compartmental boundaries in which interactions are concentrated, it is compatible with the definition of compartments. The method is rigorous because it maximizes an explicit function, identifies the number of non-overlapping compartments, assigns membership to compartments, and tests the statistical significance of the results12,13,14. A graphical presentation14 reveals systemic relationships and taxa-specific positions as structured by compartments. From this graphic, we explore two scenarios of disturbance to develop a hypothesis for testing how compartmentalized interactions increase stability in food webs15,16,17.;,citation_author=Ann E. Krause;,citation_author=Kenneth A. Frank;,citation_author=Doran M. Mason;,citation_author=Robert E. Ulanowicz;,citation_author=William W. Taylor;,citation_publication_date=2003-11;,citation_cover_date=2003-11;,citation_year=2003;,citation_issue=6964;,citation_doi=10.1038/nature02115;,citation_issn=1476-4687;,citation_volume=426;,citation_language=en-US;,citation_journal_title=Nature;,citation_publisher=Nature Publishing Group;">
<meta name="citation_reference" content="citation_title=A neutral-niche theory of nestedness in mutualistic networks;,citation_abstract=Recently, there has been a vigorous interest in community ecology about the structure of mutualistic networks and its importance for species persistence and coevolution. However, the mechanisms shaping mutualistic networks have been rarely explored. Here we extend for the first time the neutral theory of biodiversity to a multi trophic system. We focus on nestedness, a distinctive pattern of mutualistic community assembly showing two characteristics, namely, asymmetrical specialization (specialists interacting with generalists) and a generalist core (generalists interacting with generalists). We investigate the importance of relative species abundance (RSA) for the nested assembly of plant–animal mutualistic networks. Our results show that neutral mutualistic communities give rise to networks considerably more nested than real communities. RSA explains 60–70% of nested patterns in two real communities studied here, while 30–40% of nestedness is still unexplained. The nested pattern in real communities is better explained when we introduce interaction-specific species traits such as forbidden links and intensity of dependence (relative importance of fruits for the diet of a frugivore) in our analysis. The fact that neutral mutualistic communities exhibit a perfectly nested structure and do not show a random or compartmentalized structure, underlines the importance of RSA in the assembly of mutualistic networks.;,citation_author=Abhay Krishna;,citation_author=Paulo R. Guimarães Jr;,citation_author=Pedro Jordano;,citation_author=Jordi Bascompte;,citation_publication_date=2008;,citation_cover_date=2008;,citation_year=2008;,citation_issue=11;,citation_doi=10.1111/j.1600-0706.2008.16540.x;,citation_issn=1600-0706;,citation_volume=117;,citation_language=en-US;,citation_journal_title=Oikos;">
<meta name="citation_reference" content="citation_title=Ecological network inference is not consistent across scales or approaches;,citation_abstract=Several methods of ecological network inference have been proposed, but their consistency and applicability for use across ecologically relevant scales require further investigation. Here, we infer ecological networks using two data sets (YFDP, FIA) describing distributional and attribute information at local, regional, and continental scales for woody species across North America. We accomplish this inference using four different methodologies (COOCCUR, NETASSOC, HMSC, NDD-RIM), incorporating biological data along an occurrence-performance spectrum while accounting (or not) for various confounding parameters. We contrast 1-1 associations at each evaluated scale to quantify consistency amongst inference approaches. We also assess consistency across scales within each inference approach. Ultimately, we find that inferred networks are inconsistent across scales and methodologies, particularly at continental scales. We highlight how such inconsistencies between network inference methods may be linked to using occurrence or performance information and incorporating or not confounding factors. Finally, we argue that identifying the “best” inference method is non-trivial. Thus, to facilitate the choice of inference methods for a given purpose, we suggest aligning specific research questions and the scale applicability of the method when interpreting the inferred links, network topology, and ecological processes governing network assembly.;,citation_author=Erik Kusch;,citation_author=Malyon Bimler;,citation_author=James A. Lutz;,citation_author=Alejandro Ordonez;,citation_publication_date=2023-07;,citation_cover_date=2023-07;,citation_year=2023;,citation_doi=10.1101/2023.07.13.548816;,citation_language=en-US;,citation_publisher=bioRxiv;">
<meta name="citation_reference" content="citation_title=A Novel Simulation Framework for Validation of Ecological Network Inference;,citation_abstract=Understanding how the differential magnitude and sign of ecological interactions vary across space is vital to assessing ecosystem resilience to biodiversity loss and predict community assemblies. This necessity for ecological network knowledge and their labour-intensive sampling requirements has spurred the creation of ecological network inference methodology. Recent research has identified inconsistencies in networks inferred using different approaches thus necessitating quantification of inference performance to facilitate choice of network inference approach.Here we develop a data simulation method to generate data products fit for network inference and subsequently quantify the validity of two well-established ecological interaction network inference methods – HMSC and COOCCUR. The simulation framework we present here can be parameterised using real-world information (e.g., biological interactions observed in-situ and bioclimatic niche preferences) thus representing network inference capabilities in real-world applications. Using this framework, it is thus possible to evaluate the performance of any ecological network inference approach.We identify a concerningly large range in accuracy of inferred networks as compared to true, realisable association networks. These differences in inference accuracy are governed by a paradigm of input data types and environmental parameter estimation as previously suggested. To establish a workflow for quantification of network inference reliability, we suggest analysis procedures with which to explore inference and detection probabilities of association types of different identity and sign with respect to bioclimatic niche preferences and association strength of association partner-species.With this study, we provide the groundwork with which to validate and compare ecological network inference methods, and ultimately vastly increase our ability to understand and predict species biodiversity across space and time.;,citation_author=Erik Kusch;,citation_author=Anna C. Vinton;,citation_publication_date=2023-10;,citation_cover_date=2023-10;,citation_year=2023;,citation_doi=10.1101/2023.08.05.552122;,citation_language=en-US;,citation_publisher=bioRxiv;">
<meta name="citation_reference" content="citation_title=Species traits as drivers of food web structure;,citation_abstract=The use of functional traits to describe community structure is a promising approach to reveal generalities across organisms and ecosystems. Plant ecologists have demonstrated the importance of traits in explaining community structure, competitive interactions as well as ecosystem functioning. The application of trait-based methods to more complex communities such as food webs is however more challenging owing to the diversity of animal characteristics and of interactions. The objective of this study was to determine how functional structure is related to food web structure. We consider that food web structure is the result of 1) the match between consumer and resource traits, which determine the occurence of a trophic interaction between them, and 2) the distribution of functional traits in the community. We implemented a statistical approach to assess whether or not 35 466 pairwise interactions between soil organisms are constrained by trait-matching and then used a Procrustes analysis to investigate correlations between functional indices and network properties across 48 sites. We found that the occurrence of trophic interactions is well predicted by matching the traits of the resource with those of the consumer. Taxonomy and body mass of both species were the most important traits for the determination of an interaction. As a consequence, functional evenness and the variance of certain traits in the community were correlated to trophic complementarity between species, while trait identity, more than diversity, was related to network topology. The analysis was however limited by trait data availability, and a coarse resolution of certain taxonomic groups in our dataset. These limitations explain the importance of taxonomy, as well as the complexity of the statistical model needed. Our results outline the important implications of trait composition on ecological networks, opening promising avenues of research into the relationship between functional diversity and ecosystem functioning in multi-trophic systems.;,citation_author=Idaline Laigle;,citation_author=Isabelle Aubin;,citation_author=Christoph Digel;,citation_author=Ulrich Brose;,citation_author=Isabelle Boulangeat;,citation_author=Dominique Gravel;,citation_publication_date=2018;,citation_cover_date=2018;,citation_year=2018;,citation_issue=2;,citation_doi=10.1111/oik.04712;,citation_issn=1600-0706;,citation_volume=127;,citation_language=en-US;,citation_journal_title=Oikos;">
<meta name="citation_reference" content="citation_title=EcologicalNetworksDynamics.jl A Julia package to simulate the temporal dynamics of complex ecological networks;,citation_abstract=Species interactions play a crucial role in shaping biodiversity, species coexistence, population dynamics, community stability and ecosystem functioning. Our understanding of the role of the diversity of species interactions driving these species, community and ecosystem features is limited because current approaches often focus only on trophic interactions. This is why a new modelling framework that includes a greater diversity of interactions between species is crucially needed.We developed a modular, user-friendly, and extensible Julia package that delivers the core functionality of the bio-energetic food web model. Moreover, it embeds several ecological interaction types alongside the capacity to manipulate external drivers of ecological dynamics like temperature. These new features represent important processes known to influence biodiversity, coexistence, functioning and stability in natural communities. Specifically, they include: a) an explicit multiple nutrient intake model for producers, b) competition among producers, c) temperature dependence implemented via the Boltzmann-Arhennius rule, and d) the ability to model several non-trophic interactions including competition for space, plant facilitation, predator interference and refuge provisioning.The inclusion of the various features provides users with the ability to ask questions about multiple simultaneous processes and stressor impacts, and thus develop theory relevant to real world scenarios facing complex ecological communities in the Anthropocene. It will allow researchers to quantify the relative importance of different mechanisms to stability and functioning of complex communities.The package was build for theoreticians seeking to explore the effects of different types of species interactions on the dynamics of complex ecological communities, but also for empiricists seeking to confront their empirical findings with theoretical expectations. The package provides a straightforward framework to model explicitly complex ecological communities or provide tools to generate those communities from few parameters.;,citation_author=Ismaël Lajaaiti;,citation_author=Iago Bonnici;,citation_author=Sonia Kéfi;,citation_author=Hana Mayall;,citation_author=Alain Danet;,citation_author=Andrew P. Beckerman;,citation_author=Thomas Malpas;,citation_author=Eva Delmas;,citation_publication_date=2024-03;,citation_cover_date=2024-03;,citation_year=2024;,citation_doi=10.1101/2024.03.20.585899;,citation_language=en-US;,citation_publisher=bioRxiv;">
<meta name="citation_reference" content="citation_title=Complexity and stability of ecological networks: A review of the theory;,citation_abstract=Our planet is changing at paces never observed before. Species extinction is happening at faster rates than ever, greatly exceeding the five mass extinctions in the fossil record. Nevertheless, our lives are strongly based on services provided by ecosystems, thus the responses to global change of our natural heritage are of immediate concern. Understanding the relationship between complexity and stability of ecosystems is of key importance for the maintenance of the balance of human growth and the conservation of all the natural services that ecosystems provide. Mathematical network models can be used to simplify the vast complexity of the real world, to formally describe and investigate ecological phenomena, and to understand ecosystems propensity of returning to its functioning regime after a stress or a perturbation. The use of ecological-network models to study the relationship between complexity and stability of natural ecosystems is the focus of this review. The concept of ecological networks and their characteristics are first introduced, followed by central and occasionally contrasting definitions of complexity and stability. The literature on the relationship between complexity and stability in different types of models and in real ecosystems is then reviewed, highlighting the theoretical debate and the lack of consensual agreement. The summary of the importance of this line of research for the successful management and conservation of biodiversity and ecosystem services concludes the review.;,citation_author=Pietro Landi;,citation_author=Henintsoa O. Minoarivelo;,citation_author=Åke Brännström;,citation_author=Cang Hui;,citation_author=Ulf Dieckmann;,citation_publication_date=2018-10;,citation_cover_date=2018-10;,citation_year=2018;,citation_issue=4;,citation_doi=10.1007/s10144-018-0628-3;,citation_issn=1438-390X;,citation_volume=60;,citation_language=en-US;,citation_journal_title=Population Ecology;">
<meta name="citation_reference" content="citation_title=The Trophic-Dynamic Aspect of Ecology;,citation_author=Raymond L. Lindeman;,citation_publication_date=1942;,citation_cover_date=1942;,citation_year=1942;,citation_fulltext_html_url=https://www.jstor.org/stable/1930126;,citation_issue=4;,citation_doi=10.2307/1930126;,citation_issn=0012-9658;,citation_volume=23;,citation_journal_title=Ecology;,citation_publisher=Ecological Society of America;">
<meta name="citation_reference" content="citation_title=Predicting predator–prey interactions in terrestrial endotherms using random forest;,citation_abstract=Species interactions play a fundamental role in ecosystems. However, few ecological communities have complete data describing such interactions, which is an obstacle to understanding how ecosystems function and respond to perturbations. Because it is often impractical to collect empirical data for all interactions in a community, various methods have been developed to infer interactions. Machine learning is increasingly being used for making interaction predictions, with random forest being one of the most frequently used of these methods. However, performance of random forest in inferring predator-prey interactions in terrestrial vertebrates and its sensitivity to training data quality remain untested. We examined predator–prey interactions in two diverse, primarily terrestrial vertebrate classes: birds and mammals. Combining data from a global interaction dataset and a specific community (Simpson Desert, Australia), we tested how well random forest predicted predator–prey interactions for mammals and birds using species’ ecomorphological and phylogenetic traits. We also tested how variation in training data quality – manipulated by removing records and switching interaction records to non-interactions – affected model performance. We found that random forest could predict predator–prey interactions for birds and mammals using ecomorphological or phylogenetic traits, correctly predicting up to 88 and 67% of interactions and non-interactions in the global and community-specific datasets, respectively. These predictions were accurate even when there were no records in the training data for focal species. In contrast, false non-interactions for focal predators in training data strongly degraded model performance. Our results demonstrate that random forest can identify predator–prey interactions for birds and mammals that have few or no interaction records. Furthermore, our study provides guidance on how to prepare training data to optimise machine learning classifiers for predicting species interactions, which could help ecologists 1) address knowledge gaps and explore network-related questions in data-poor situations, and 2) predict interactions for range-expanding species.;,citation_author=John Llewelyn;,citation_author=Giovanni Strona;,citation_author=Christopher R. Dickman;,citation_author=Aaron C. Greenville;,citation_author=Glenda M. Wardle;,citation_author=Michael S. Y. Lee;,citation_author=Seamus Doherty;,citation_author=Farzin Shabani;,citation_author=Frédérik Saltré;,citation_author=Corey J. A. Bradshaw;,citation_publication_date=2023;,citation_cover_date=2023;,citation_year=2023;,citation_issue=9;,citation_doi=10.1111/ecog.06619;,citation_issn=1600-0587;,citation_volume=2023;,citation_language=en-US;,citation_journal_title=Ecography;">
<meta name="citation_reference" content="citation_title=TETRA-EU 1.0: A species-level trophic metaweb of European tetrapods;,citation_abstract=Motivation Documenting potential interactions between species represents a major step towards understanding and predicting the spatial and temporal structure of multi-trophic communities and their functioning. The metaweb concept summarizes the potential trophic (and non-trophic) interactions in a given species pool. As such, it generalizes the regional species pool of community ecology by incorporating the potential relationships between species from different trophic levels along with their functional characteristics. However, although this concept is very attractive theoretically, it has rarely been used to understand the structure of an ecological network, mostly because of data availability. Here, we provide a continental-scale, species-level metaweb for all tetrapods (mammals, breeding birds, reptiles and amphibians) occurring in Europe and in the Northern Mediterranean basin. This metaweb is based on data extracted from the scientific literature, including published papers, books and grey literature. Main type of variable contained For each species considered, we built the network of potential two-way trophic interactions. Spatial location and grain We considered all species occurring in the entire European subcontinent, from Macaronesia (including only the islands belonging politically to Spain and Portugal) to the Ural Mountains (west to east) and from Fennoscandia and U.K. islands to the Mediterranean (north to south). We included Turkey, geographically part of Asia, to provide a complete picture of the north-eastern Mediterranean coast. Time period The data represent information published and/or collected during the last 50 years. Major taxa studied and level of measurement We focused our metaweb on terrestrial tetrapods occurring in the study area. Only species introduced in historical times and currently naturalized were considered; new introductions were excluded. In total, we included 288 mammals, 509 regularly breeding birds, 250 reptiles and 104 amphibians. Software format Data are supplied as semicolon-separated text files.;,citation_author=Luigi Maiorano;,citation_author=Alessandro Montemaggiori;,citation_author=Gentile Francesco Ficetola;,citation_author=Louise O’Connor;,citation_author=Wilfried Thuiller;,citation_publication_date=2020-09;,citation_cover_date=2020-09;,citation_year=2020;,citation_issue=9;,citation_doi=10.1111/geb.13138;,citation_issn=1466-8238;,citation_volume=29;,citation_language=en-US;,citation_journal_title=Global Ecology and Biogeography;">
<meta name="citation_reference" content="citation_title=The propagation of disturbances in ecological networks;,citation_author=Lucas P. Martins;,citation_author=David Garcia-Callejas;,citation_author=Hao Ran Lai;,citation_author=Kate L. Wootton;,citation_author=Jason M. Tylianakis;,citation_publication_date=2024-02;,citation_cover_date=2024-02;,citation_year=2024;,citation_issue=0;,citation_doi=10.1016/j.tree.2024.01.009;,citation_issn=0169-5347;,citation_pmid=38402007;,citation_volume=0;,citation_language=en-US;,citation_journal_title=Trends in Ecology &amp;amp;amp; Evolution;,citation_publisher=Elsevier;">
<meta name="citation_reference" content="citation_title=Comparison of the predicted and observed secondary structure of T4 phage lysozyme;,citation_abstract=Predictions of the secondary structure of T4 phage lysozyme, made by a number of investigators on the basis of the amino acid sequence, are compared with the structure of the protein determined experimentally by X-ray crystallography. Within the amino terminal half of the molecule the locations of helices predicted by a number of methods agree moderately well with the observed structure, however within the carboxyl half of the molecule the overall agreement is poor. For eleven different helix predictions, the coefficients giving the correlation between prediction and observation range from 0.14 to 0.42. The accuracy of the predictions for both \beta-sheet regions and for turns are generally lower than for the helices, and in a number of instances the agreement between prediction and observation is no better than would be expected for a random selection of residues. The structural predictions for T4 phage lysozyme are much less successful than was the case for adenylate kinase (Schulz et al. (1974) Nature 250, 140–142). No one method of prediction is clearly superior to all others, and although empirical predictions based on larger numbers of known protein structure tend to be more accurate than those based on a limited sample, the improvement in accuracy is not dramatic, suggesting that the accuracy of current empirical predictive methods will not be substantially increased simply by the inclusion of more data from additional protein structure determinations.;,citation_author=B. W. Matthews;,citation_publication_date=1975-10;,citation_cover_date=1975-10;,citation_year=1975;,citation_issue=2;,citation_doi=10.1016/0005-2795(75)90109-9;,citation_issn=0005-2795;,citation_volume=405;,citation_journal_title=Biochimica et Biophysica Acta (BBA) - Protein Structure;">
<meta name="citation_reference" content="citation_title=Analysis of Ecological Communities;,citation_author=Bruce McCune;,citation_author=James Grace;,citation_publication_date=2002;,citation_cover_date=2002;,citation_year=2002;,citation_isbn=0-9721290-0-6;">
<meta name="citation_reference" content="citation_title=Network Motifs: Simple Building Blocks of Complex Networks;,citation_abstract=Complex networks are studied across many fields of science. To uncover their structural design principles, we defined “network motifs,” patterns of interconnections occurring in complex networks at numbers that are significantly higher than those in randomized networks. We found such motifs in networks from biochemistry, neurobiology, ecology, and engineering. The motifs shared by ecological food webs were distinct from the motifs shared by the genetic networks of Escherichia coli and Saccharomyces cerevisiae or from those found in the World Wide Web. Similar motifs were found in networks that perform information processing, even though they describe elements as different as biomolecules within a cell and synaptic connections between neurons in Caenorhabditis elegans. Motifs may thus define universal classes of networks. This approach may uncover the basic building blocks of most networks.;,citation_author=R. Milo;,citation_author=S. Shen-Orr;,citation_author=S. Itzkovitz;,citation_author=N. Kashtan;,citation_author=D. Chklovskii;,citation_author=U. Alon;,citation_publication_date=2002-10;,citation_cover_date=2002-10;,citation_year=2002;,citation_issue=5594;,citation_doi=10.1126/science.298.5594.824;,citation_volume=298;,citation_journal_title=Science;,citation_publisher=American Association for the Advancement of Science;">
<meta name="citation_reference" content="citation_title=Tree-based inference of species interaction networks from abundance data;,citation_abstract=The behaviour of ecological systems mainly relies on the interactions between the species it involves. We consider the problem of inferring the species interaction network from abundance data. To be relevant, any network inference methodology needs to handle count data and to account for possible environmental effects. It also needs to distinguish between direct interactions and indirect associations and graphical models provide a convenient framework for this purpose. A simulation study shows that the proposed methodology compares well with state-of-the-art approaches, even when the underlying graph strongly differs from a tree. The analysis of two datasets highlights the influence of covariates on the inferred network. Accounting for covariates is critical to avoid spurious edges. The proposed approach could be extended to perform network comparison or to look for missing species.;,citation_author=Raphaëlle Momal;,citation_author=Stéphane Robin;,citation_author=Christophe Ambroise;,citation_publication_date=2020;,citation_cover_date=2020;,citation_year=2020;,citation_issue=5;,citation_doi=10.1111/2041-210X.13380;,citation_issn=2041-210X;,citation_volume=11;,citation_language=en-US;,citation_journal_title=Methods in Ecology and Evolution;">
<meta name="citation_reference" content="citation_title=Inferring biotic interactions from proxies;,citation_abstract=Inferring biotic interactions from functional, phylogenetic and geographical proxies remains one great challenge in ecology. We propose a conceptual framework to infer the backbone of biotic interaction networks within regional species pools. First, interacting groups are identified to order links and remove forbidden interactions between species. Second, additional links are removed by examination of the geographical context in which species co-occur. Third, hypotheses are proposed to establish interaction probabilities between species. We illustrate the framework using published food-webs in terrestrial and marine systems. We conclude that preliminary descriptions of the web of life can be made by careful integration of data with theory.;,citation_author=Ignacio Morales-Castilla;,citation_author=Miguel G. Matias;,citation_author=Dominique Gravel;,citation_author=Miguel B. Araújo;,citation_publication_date=2015-06;,citation_cover_date=2015-06;,citation_year=2015;,citation_issue=6;,citation_doi=10.1016/j.tree.2015.03.014;,citation_volume=30;,citation_journal_title=Trends in Ecology &amp;amp;amp; Evolution;">
<meta name="citation_reference" content="citation_title=Networks. An introduction;,citation_author=Mark E. J. Newman;,citation_publication_date=2010;,citation_cover_date=2010;,citation_year=2010;">
<meta name="citation_reference" content="citation_title=Natural - synthetic - artificial!;,citation_abstract=The terms “natural, “synthetic” and “artificial” are discussed in relation to synthetic and artificial chromosomes and genomes, synthetic and artificial cells and artificial life.;,citation_author=Peter E. Nielsen;,citation_publication_date=2010-07;,citation_cover_date=2010-07;,citation_year=2010;,citation_issue=1;,citation_doi=10.4161/adna.1.1.12934;,citation_issn=1949-095X;,citation_pmid=21687528;,citation_volume=1;,citation_journal_title=Artificial DNA: PNA &amp;amp;amp; XNA;,citation_publisher=Taylor &amp;amp; Francis;">
<meta name="citation_reference" content="citation_title=Mapping the imprint of biotic interactions on \beta-diversity;,citation_abstract=Investigating how trophic interactions influence the \beta-diversity of meta-communities is of paramount importance to understanding the processes shaping biodiversity distribution. Here, we apply a statistical method for inferring the strength of spatial dependencies between pairs of species groups. Using simulated community data generated from a multi-trophic model, we showed that this method can approximate biotic interactions in multi-trophic communities based on \beta-diversity patterns across groups. When applied to soil multi-trophic communities along an elevational gradient in the French Alps, we found that fungi make a major contribution to the structuring of \beta-diversity across trophic groups. We also demonstrated that there were strong spatial dependencies between groups known to interact specifically (e.g. plant-symbiotic fungi, bacteria-nematodes) and that the influence of environment was less important than previously reported in the literature. Our method paves the way for a better understanding and mapping of multi-trophic communities through space and time.;,citation_author=Marc Ohlmann;,citation_author=Florent Mazel;,citation_author=Loïc Chalmandrier;,citation_author=Stéphane Bec;,citation_author=Eric Coissac;,citation_author=Ludovic Gielly;,citation_author=Johan Pansu;,citation_author=Vincent Schilling;,citation_author=Pierre Taberlet;,citation_author=Lucie Zinger;,citation_author=Jérome Chave;,citation_author=Wilfried Thuiller;,citation_publication_date=2018;,citation_cover_date=2018;,citation_year=2018;,citation_issue=11;,citation_doi=10.1111/ele.13143;,citation_issn=1461-0248;,citation_volume=21;,citation_language=en-US;,citation_journal_title=Ecology Letters;">
<meta name="citation_reference" content="citation_title=Using latent variable models to identify large networks of species-to-species associations at different spatial scales;,citation_abstract=We present a hierarchical latent variable model that partitions variation in species occurrences and co-occurrences simultaneously at multiple spatial scales. We illustrate how the parameterized model can be used to predict the occurrences of a species by using as predictors not only the environmental covariates, but also the occurrences of all other species, at all spatial scales. We leverage recent progress in Bayesian latent variable models to implement a computationally effective algorithm that enables one to consider large communities and extensive sampling schemes. We exemplify the framework with a community of 98 fungal species sampled in c. 22 500 dead wood units in 230 plots in 29 beech forests. The networks identified by correlations and partial correlations were consistent, as were networks for natural and managed forests, but networks at different spatial scales were dissimilar. Accounting for the occurrences of the other species roughly doubled the predictive powers of the models compared to accounting for environmental covariates only .;,citation_author=Otso Ovaskainen;,citation_author=Nerea Abrego;,citation_author=Panu Halme;,citation_author=David Dunson;,citation_publication_date=2016;,citation_cover_date=2016;,citation_year=2016;,citation_issue=5;,citation_doi=10.1111/2041-210X.12501;,citation_issn=2041-210X;,citation_volume=7;,citation_language=en-US;,citation_journal_title=Methods in Ecology and Evolution;">
<meta name="citation_reference" content="citation_title=Statistical mechanics of networks;,citation_abstract=We study the family of network models derived by requiring the expected properties of a graph ensemble to match a given set of measurements of a real-world network, while maximizing the entropy of the ensemble. Models of this type play the same role in the study of networks as is played by the Boltzmann distribution in classical statistical mechanics; they offer the best prediction of network properties subject to the constraints imposed by a given set of observations. We give exact solutions of models within this class that incorporate arbitrary degree distributions and arbitrary but independent edge probabilities. We also discuss some more complex examples with correlated edges that can be solved approximately or exactly by adapting various familiar methods, including mean-field theory, perturbation theory, and saddle-point expansions.;,citation_author=Juyong Park;,citation_author=M. E. J. Newman;,citation_publication_date=2004-12;,citation_cover_date=2004-12;,citation_year=2004;,citation_issue=6;,citation_doi=10.1103/PhysRevE.70.066117;,citation_volume=70;,citation_journal_title=Physical Review E;,citation_publisher=American Physical Society;">
<meta name="citation_reference" content="citation_title=Dimensionality of consumer search space drives trophic interaction strengths;,citation_abstract=Trophic interactions govern biomass fluxes in ecosystems, and stability in food webs. Knowledge of how trophic interaction strengths are affected by differences among habitats is crucial for understanding variation in ecological systems. Here we show how substantial variation in consumption-rate data, and hence trophic interaction strengths, arises because consumers tend to encounter resources more frequently in three dimensions (3D) (for example, arboreal and pelagic zones) than two dimensions (2D) (for example, terrestrial and benthic zones). By combining new theory with extensive data (376 species, with body masses ranging from 5.24  10-14 kg to 800 kg), we find that consumption rates scale sublinearly with consumer body mass (exponent of approximately 0.85) for 2D interactions, but superlinearly (exponent of approximately 1.06) for 3D interactions. These results contradict the currently widespread assumption of a single exponent (of approximately 0.75) in consumer–resource and food-web research. Further analysis of 2,929 consumer–resource interactions shows that dimensionality of consumer search space is probably a major driver of species coexistence, and the stability and abundance of populations.;,citation_author=Samraat Pawar;,citation_author=Anthony I. Dell;,citation_author=Van M. Savage;,citation_publication_date=2012-06;,citation_cover_date=2012-06;,citation_year=2012;,citation_issue=7404;,citation_doi=10.1038/nature11131;,citation_issn=1476-4687;,citation_volume=486;,citation_language=en-US;,citation_journal_title=Nature;,citation_publisher=Nature Publishing Group;">
<meta name="citation_reference" content="citation_title=Scalable network reconstruction in subquadratic time;,citation_abstract=Network reconstruction consists in determining the unobserved pairwise couplings between $N$ nodes given only observational data on the resulting behavior that is conditioned on those couplings – typically a time-series or independent samples from a graphical model. A major obstacle to the scalability of algorithms proposed for this problem is a seemingly unavoidable quadratic complexity of $\Omega(N^2)$, corresponding to the requirement of each possible pairwise coupling being contemplated at least once, despite the fact that most networks of interest are sparse, with a number of non-zero couplings that is only $O(N)$. Here we present a general algorithm applicable to a broad range of reconstruction problems that significantly outperforms this quadratic baseline. Our algorithm relies on a stochastic second neighbor search (Dong et al., 2011) that produces the best edge candidates with high probability, thus bypassing an exhaustive quadratic search. If we rely on the conjecture that the second-neighbor search finishes in log-linear time (Baron &amp;amp;amp; Darling, 2020; 2022), we demonstrate theoretically that our algorithm finishes in subquadratic time, with a data-dependent complexity loosely upper bounded by $O(N^{3/2}\log N)$, but with a more typical log-linear complexity of $O(N\log^2N)$. In practice, we show that our algorithm achieves a performance that is many orders of magnitude faster than the quadratic baseline – in a manner consistent with our theoretical analysis – allows for easy parallelization, and thus enables the reconstruction of networks with hundreds of thousands and even millions of nodes and edges.;,citation_author=Tiago P. Peixoto;,citation_publication_date=2024-05;,citation_cover_date=2024-05;,citation_year=2024;,citation_fulltext_html_url=https://arxiv.org/abs/2401.01404;,citation_language=en-US;,citation_publisher=arXiv;">
<meta name="citation_reference" content="citation_title=Fit, efficiency, and biology: Some thoughts on judging food web models;,citation_abstract=Revealing the processes that determine who eats whom, and thereby the structure of food webs, is a long running challenge in ecological research. Recent advances include development of new methods for measuring fit of models to observed food web data, and thereby testing which are the “best” food web models. The best model could be considered the most efficient with relatively few parameters and high explanatory power. Another recent advance involves adding some additional biology to food web models in the form of foraging theory based on maximisation of energy intake as the predictor of species’ diets in food webs. While it is interesting to compare efficiency among food web models, we believe that such comparisons at least should be interpreted with caution, since they do not account for any differences in motivation, formulation, and potential that might also exist among models. Furthermore, we see an important but somewhat neglected role for experimental tests of models of food web structure.;,citation_author=Owen L. Petchey;,citation_author=Andrew P. Beckerman;,citation_author=Jens O. Riede;,citation_author=Philip H. Warren;,citation_publication_date=2011-06;,citation_cover_date=2011-06;,citation_year=2011;,citation_issue=1;,citation_doi=10.1016/j.jtbi.2011.03.019;,citation_issn=0022-5193;,citation_volume=279;,citation_journal_title=Journal of Theoretical Biology;">
<meta name="citation_reference" content="citation_title=Size, foraging, and food web structure;,citation_abstract=Understanding what structures ecological communities is vital to answering questions about extinctions, environmental change, trophic cascades, and ecosystem functioning. Optimal foraging theory was conceived to increase such understanding by providing a framework with which to predict species interactions and resulting community structure. Here, we use an optimal foraging model and allometries of foraging variables to predict the structure of real food webs. The qualitative structure of the resulting model provides a more mechanistic basis for the phenomenological rules of previous models. Quantitative analyses show that the model predicts up to 65% of the links in real food webs. The deterministic nature of the model allows analysis of the model’s successes and failures in predicting particular interactions. Predacious and herbivorous feeding interactions are better predicted than pathogenic, parasitoid, and parasitic interactions. Results also indicate that accurate prediction and modeling of some food webs will require incorporating traits other than body size and diet choice models specific to different types of feeding interaction. The model results support the hypothesis that individual behavior, subject to natural selection, determines individual diets and that food web structure is the sum of these individual decisions.;,citation_author=Owen L. Petchey;,citation_author=Andrew P. Beckerman;,citation_author=Jens O. Riede;,citation_author=Philip H. Warren;,citation_publication_date=2008-03;,citation_cover_date=2008-03;,citation_year=2008;,citation_issue=11;,citation_doi=10.1073/pnas.0710672105;,citation_issn=0027-8424, 1091-6490;,citation_volume=105;,citation_language=en-US;,citation_journal_title=Proceedings of the National Academy of Sciences;">
<meta name="citation_reference" content="citation_title=Advancing ecological networks: Moving beyond binary classification to probabilistic interactions;,citation_abstract=Leveraging trophic interactions to deduce macro-ecological patterns has become a prevalent method, taking advantage of the extensive databases on binary trophic interactions (i.e., prey-predator relationships). However, this binary approach oversimplifies complex ecological dynamics and fails to capture the nuanced structure of food webs. The challenge lies in the scarcity and limited availability of data on non-binary interactions, which are crucial for a more comprehensive understanding of ecological networks. This study explores the use of binary classifiers, particularly the XGBOOST algorithm to address the limitations of traditional binary approaches to prey-predator relationships. By predicting predation probabilities among nine mammalian predators using species traits, my findings demonstrate the classifiers’ robust predictive capabilities to binary predictions but also a good correlation between probabilistic predation derived from binary classifiers and observed prey preferences. It also highlighted the importance of selecting informative species traits for predicting interaction, with performance contrastingly superior to null models. Despite a small sample size, this work provides insightful results and sets a foundation for future research to expand these models to broader ecological networks, emphasizing the need for comprehensive prey preference data.;,citation_author=Martin Philippe-Lesaffre;,citation_publication_date=2024-02;,citation_cover_date=2024-02;,citation_year=2024;,citation_doi=10.1101/2024.02.07.579289;,citation_language=en-US;,citation_publisher=bioRxiv;">
<meta name="citation_reference" content="citation_title=Machine learning algorithms to infer trait-matching and predict species interactions in ecological networks;,citation_abstract=Ecologists have long suspected that species are more likely to interact if their traits match in a particular way. For example, a pollination interaction may be more likely if the proportions of a bee’s tongue fit a plant’s flower shape. Empirical estimates of the importance of trait-matching for determining species interactions, however, vary significantly among different types of ecological networks. Here, we show that ambiguity among empirical trait-matching studies may have arisen at least in parts from using overly simple statistical models. Using simulated and real data, we contrast conventional generalized linear models (GLM) with more flexible Machine Learning (ML) models (Random Forest, Boosted Regression Trees, Deep Neural Networks, Convolutional Neural Networks, Support Vector Machines, naïve Bayes, and k-Nearest-Neighbor), testing their ability to predict species interactions based on traits, and infer trait combinations causally responsible for species interactions. We found that the best ML models can successfully predict species interactions in plant–pollinator networks, outperforming GLMs by a substantial margin. Our results also demonstrate that ML models can better identify the causally responsible trait-matching combinations than GLMs. In two case studies, the best ML models successfully predicted species interactions in a global plant–pollinator database and inferred ecologically plausible trait-matching rules for a plant–hummingbird network from Costa Rica, without any prior assumptions about the system. We conclude that flexible ML models offer many advantages over traditional regression models for understanding interaction networks. We anticipate that these results extrapolate to other ecological network types. More generally, our results highlight the potential of machine learning and artificial intelligence for inference in ecology, beyond standard tasks such as image or pattern recognition.;,citation_author=Maximilian Pichler;,citation_author=Virginie Boreux;,citation_author=Alexandra-Maria Klein;,citation_author=Matthias Schleuning;,citation_author=Florian Hartig;,citation_publication_date=2020;,citation_cover_date=2020;,citation_year=2020;,citation_issue=2;,citation_doi=10.1111/2041-210X.13329;,citation_issn=2041-210X;,citation_volume=11;,citation_language=en-US;,citation_journal_title=Methods in Ecology and Evolution;">
<meta name="citation_reference" content="citation_title=The multilayer nature of ecological networks;,citation_abstract=Although networks provide a powerful approach to study a large variety of ecological systems, their formulation does not typically account for multiple interaction types, interactions that vary in space and time, and interconnected systems such as networks of networks. The emergent field of ’multilayer networks’ provides a natural framework for extending analyses of ecological systems to include such multiple layers of complexity, as it specifically allows one to differentiate and model ’intralayer’ and ’interlayer’ connectivity. The framework provides a set of concepts and tools that can be adapted and applied to ecology, facilitating research on high-dimensional, heterogeneous systems in nature. Here, we formally define ecological multilayer networks based on a review of previous, related approaches; illustrate their application and potential with analyses of existing data; and discuss limitations, challenges, and future applications. The integration of multilayer network theory into ecology offers largely untapped potential to investigate ecological complexity and provide new theoretical and empirical insights into the architecture and dynamics of ecological systems.;,citation_author=Shai Pilosof;,citation_author=Mason A. Porter;,citation_author=Mercedes Pascual;,citation_author=Sonia Kéfi;,citation_publication_date=2017-03;,citation_cover_date=2017-03;,citation_year=2017;,citation_issue=4;,citation_doi=10.1038/s41559-017-0101;,citation_issn=2397-334X;,citation_pmid=28812678;,citation_volume=1;,citation_language=en-US;,citation_journal_title=Nature Ecology &amp;amp;amp; Evolution;">
<meta name="citation_reference" content="citation_title=Before, during and after megafaunal extinctions: Human impact on Pleistocene-Holocene trophic networks in South Patagonia;,citation_abstract=Worldwide extinctions of large terrestrial vertebrates in the late Pleistocene provide insight on how humans reshape ecological communities. Understanding the ecological causes and consequences of megafaunal extinctions requires integrating approaches to reconstruct the ecological communities from the past. Here, we combined archeological and paleontological evidence with network analyses to understand the changes in ecological communities from late Pleistocene to the Holocene in South Patagonia, the last continental region where the encounter between humans and extinct megafauna occurred. The zooarcheological record suggests humans would have interacted mainly with large-bodied species, which comprise a small subset of the available prey. Accordingly, using network reconstructions and structural analyses, we found that human arrival would have produced minor changes in the overall structure of trophic networks. However, those few novel interactions established by humans would have created multiple indirect paths among megafaunal species. Indirect paths are the route for indirect effects such as competition and increase the vulnerability of interaction networks to perturbations. After the extinctions of most of the megafauna, the impoverished network became structurally simpler and densely connected. Our reconstructions of past trophic networks show that multiple indirect effects, potentially contributing to extinctions, can emerge even from a limited number of novel interactions and illustrate how network organization can affect and be affected by extinctions events.;,citation_author=Mathias M. Pires;,citation_author=Diego Rindel;,citation_author=Bruno Moscardi;,citation_author=Livia R. Cruz;,citation_author=Paulo R. Guimarães;,citation_author=Sergio F. Reis;,citation_author=S. Ivan Perez;,citation_publication_date=2020-12;,citation_cover_date=2020-12;,citation_year=2020;,citation_doi=10.1016/j.quascirev.2020.106696;,citation_issn=0277-3791;,citation_volume=250;,citation_journal_title=Quaternary Science Reviews;">
<meta name="citation_reference" content="citation_title=Global biotic interactions: An open infrastructure to share and analyze species-interaction datasets;,citation_abstract=An intricate network of interactions between organisms and their environment form the ecosystems that sustain life on earth. With a detailed understanding of these interactions, ecologists and biologists can make better informed predictions about the ways different environmental factors will impact ecosystems. Despite the abundance of research data on biotic and abiotic interactions, no comprehensive and easily accessible data collection is available that spans taxonomic, geospatial, and temporal domains. Biotic-interaction datasets are effectively siloed, inhibiting cross-dataset comparisons. In order to pool resources and bring to light individual datasets, specialized research tools are needed to aggregate, normalize, and integrate existing datasets with standard taxonomies, ontologies, vocabularies, and structured data repositories. Global Biotic Interactions (GloBI) provides such tools by way of an open, community-driven infrastructure designed to lower the barrier for researchers to perform ecological systems analysis and modeling. GloBI provides a tool that (a) ingests, normalizes, and aggregates datasets, (b) integrates interoperable data with accepted ontologies (e.g., OBO Relations Ontology, Uberon, and Environment Ontology), vocabularies (e.g., Coastal and Marine Ecological Classification Standard), and taxonomies (e.g., Integrated Taxonomic Information System and National Center for Biotechnology Information Taxonomy Database), (c) makes data accessible through an application programming interface (API) and various data archives (Darwin Core, Turtle, and Neo4j), and (d) houses a data collection of about 700,000 species interactions across about 50,000 taxa, covering over 1100 references from 19 data sources. GloBI has taken an open-source and open-data approach in order to make integrated species-interaction data maximally accessible and to encourage users to provide feedback, contribute data, and improve data access methods. The GloBI collection of datasets is currently used in the Encyclopedia of Life (EOL) and Gulf of Mexico Species Interactions (GoMexSI).;,citation_author=Jorrit H. Poelen;,citation_author=James D. Simons;,citation_author=Chris J. Mungall;,citation_publication_date=2014-11;,citation_cover_date=2014-11;,citation_year=2014;,citation_doi=10.1016/j.ecoinf.2014.08.005;,citation_issn=1574-9541;,citation_volume=24;,citation_journal_title=Ecological Informatics;">
<meta name="citation_reference" content="citation_title=Describe, understand and predict: Why do we need networks in ecology?;,citation_author=Timothée Poisot;,citation_author=Daniel B. Stouffer;,citation_author=Sonia Kéfi;,citation_publication_date=2016;,citation_cover_date=2016;,citation_year=2016;,citation_fulltext_html_url=https://www.jstor.org/stable/48582345;,citation_issue=12;,citation_issn=0269-8463;,citation_volume=30;,citation_journal_title=Functional Ecology;,citation_publisher=[British Ecological Society, Wiley];">
<meta name="citation_reference" content="citation_title=Global knowledge gaps in species interaction networks data;,citation_abstract=Ecological networks are increasingly studied at large spatial scales, expanding their focus from a conceptual tool for community ecology into one that also addresses questions in biogeography and macroecology. This effort is supported by increased access to standardized information on ecological networks, in the form of openly accessible databases. Yet, there has been no systematic evaluation of the fitness for purpose of these data to explore synthesis questions at very large spatial scales. In particular, because the sampling of ecological networks is a difficult task, they are likely to not have a good representation of the diversity of Earth’s bioclimatic conditions, likely to be spatially aggregated, and therefore unlikely to achieve broad representativeness. In this paper, we analyse over 1300 ecological networks in the mangal.io database, and discuss their coverage of climates, and the geographic areas in which there is a deficit of data on ecological networks. Taken together, our results suggest that while some information about the global structure of ecological networks is available, it remains fragmented over space, with further differences by types of ecological interactions. This causes great concerns both for our ability to transfer knowledge from one region to the next, but also to forecast the structural change in networks under climate change.;,citation_author=Timothée Poisot;,citation_author=Gabriel Bergeron;,citation_author=Kevin Cazelles;,citation_author=Tad Dallas;,citation_author=Dominique Gravel;,citation_author=Andrew MacDonald;,citation_author=Benjamin Mercier;,citation_author=Clément Violet;,citation_author=Steve Vissault;,citation_publication_date=2021;,citation_cover_date=2021;,citation_year=2021;,citation_issue=7;,citation_doi=10.1111/jbi.14127;,citation_issn=1365-2699;,citation_volume=48;,citation_language=en-US;,citation_journal_title=Journal of Biogeography;">
<meta name="citation_reference" content="citation_title=Guidelines for the prediction of species interactions through binary classification;,citation_abstract=The prediction of species interactions is gaining momentum as a way to circumvent limitations in data volume. Yet, ecological networks are challenging to predict because they are typically small and sparse. Dealing with extreme class imbalance is a challenge for most binary classifiers, and there are currently no guidelines as to how predictive models can be trained for this specific problem. Using simple mathematical arguments and numerical experiments in which a variety of classifiers (for supervised learning) are trained on simulated networks, we develop a series of guidelines related to the choice of measures to use for model selection, and the ways to assemble the training dataset. Neither classifier accuracy nor the area under the receiver operating characteristic curve (ROC-AUC) are informative measures for the performance of interaction prediction. The area under the precision-recall curve (PR-AUC) is a fairer assessment of performance. In some cases, even standard measures can lead to selecting a more biased classifier because the effect of connectance is strong. The amount of correction to apply to the training dataset depends on network connectance, on the measure to be optimized, and only weakly on the classifier. These results reveal that training machines to predict networks is a challenging task, and that in virtually all cases, the composition of the training set needs to be fine-tuned before performing the actual training. We discuss these consequences in the context of the low volume of data.;,citation_author=Timothée Poisot;,citation_publication_date=2023;,citation_cover_date=2023;,citation_year=2023;,citation_issue=5;,citation_doi=10.1111/2041-210X.14071;,citation_issn=2041-210X;,citation_volume=14;,citation_language=en-US;,citation_journal_title=Methods in Ecology and Evolution;">
<meta name="citation_reference" content="citation_title=Mangal – making ecological network analysis simple;,citation_abstract=The study of ecological networks is severely limited by 1) the difficulty to access data, 2) the lack of a standardized way to link meta-data with interactions, and 3) the disparity of formats in which ecological networks themselves are stored and represented. To overcome these limitations, we have designed a data specification for ecological networks. We implemented a database respecting this standard, and released an R package (rmangal) allowing users to programmatically access, curate, and deposit data on ecological interactions. In this article, we show how these tools, in conjunction with other frameworks for the programmatic manipulation of open ecological data, streamlines the analysis process and improves replicability and reproducibility of ecological network studies.;,citation_author=Timothée Poisot;,citation_author=Benjamin Baiser;,citation_author=Jennifer Dunne;,citation_author=Sonia Kéfi;,citation_author=François Massol;,citation_author=Nicolas Mouquet;,citation_author=Tamara N. Romanuk;,citation_author=Daniel B. Stouffer;,citation_author=Spencer A. Wood;,citation_author=Dominique Gravel;,citation_publication_date=2016;,citation_cover_date=2016;,citation_year=2016;,citation_issue=4;,citation_doi=10.1111/ecog.00976;,citation_issn=1600-0587;,citation_volume=39;,citation_language=en-US;,citation_journal_title=Ecography;">
<meta name="citation_reference" content="citation_title=Beyond species: Why ecological interaction networks vary through space and time;,citation_abstract=Community ecology is tasked with the considerable challenge of predicting the structure, and properties, of emerging ecosystems. It requires the ability to understand how and why species interact, as this will allow the development of mechanism-based predictive models, and as such to better characterize how ecological mechanisms act locally on the existence of inter-specific interactions. Here we argue that the current conceptualization of species interaction networks is ill-suited for this task. Instead, we propose that future research must start to account for the intrinsic variability of species interactions, then scale up from here onto complex networks. This can be accomplished simply by recognizing that there exists intra-specific variability, in traits or properties related to the establishment of species interactions. By shifting the scale towards population-based processes, we show that this new approach will improve our predictive ability and mechanistic understanding of how species interact over large spatial or temporal scales. Synthesis Although species interactions are the backbone of ecological communities, we have little insights on how (and why) they vary through space and time. In this article, we build on existing empirical literature to show that the same species may happen to interact in different ways when their local abundances vary, their trait distribution changes, or when the environment affects either of these factors. We discuss how these findings can be integrated in existing frameworks for the analysis and simulation of species interactions.;,citation_author=Timothée Poisot;,citation_author=Daniel B. Stouffer;,citation_author=Dominique Gravel;,citation_publication_date=2015;,citation_cover_date=2015;,citation_year=2015;,citation_issue=3;,citation_doi=10.1111/oik.01719;,citation_issn=1600-0706;,citation_volume=124;,citation_language=en-US;,citation_journal_title=Oikos;">
<meta name="citation_reference" content="citation_title=The structure of probabilistic networks;,citation_abstract=Summary There is a growing realization among community ecologists that interactions between species vary across space and time and that this variation needs to be quantified. Our current numerical framework to analyse the structure of species interactions, based on graph-theoretical approaches, usually do not consider the variability of interactions. As this variability has been show to hold valuable ecological information, there is a need to adapt the current measures of network structure so that they can exploit it. We present analytical expressions of key measures of network structured, adapted so that they account for the variability of ecological interactions. We do so by modelling each interaction as a Bernoulli event; using basic calculus allows expressing the expected value, and when mathematically tractable, its variance. When applied to non-probabilistic data, the measures we present give the same results as their non-probabilistic formulations, meaning that they can be generally applied. We present three case studies that highlight how these measures can be used, in re-analysing data that experimentally measured the variability of interactions, to alleviate the computational demands of permutation-based approaches, and to use the frequency at which interactions are observed over several locations to infer the structure of local networks. We provide a free and open-source implementation of these measures. We discuss how both sampling and data representation of ecological networks can be adapted to allow the application of a fully probabilistic numerical network approach.;,citation_author=Timothée Poisot;,citation_author=Alyssa Cirtwill;,citation_author=Kévin Cazelles;,citation_author=Dominique Gravel;,citation_author=Marie-Josée Fortin;,citation_author=Daniel Stouffer;,citation_publication_date=2016;,citation_cover_date=2016;,citation_year=2016;,citation_issue=3;,citation_doi=10;,citation_volume=7;,citation_journal_title=Methods in Ecology and Evolution;">
<meta name="citation_reference" content="citation_title=Synthetic datasets and community tools for the rapid testing of ecological hypotheses;,citation_abstract=The increased availability of both open ecological data, and software to interact with it, allows the fast collection and integration of information at all spatial and taxonomic scales. This offers the opportunity to address macroecological questions in a cost-effective way. In this contribution, we illustrate this approach by forecasting the structure of a stream food web at the global scale. In so doing, we highlight the most salient issues needing to be addressed before this approach can be used with a high degree of confidence.;,citation_author=Timothée Poisot;,citation_author=Dominique Gravel;,citation_author=Shawn Leroux;,citation_author=Spencer A. Wood;,citation_author=Marie-Josée Fortin;,citation_author=Benjamin Baiser;,citation_author=Alyssa R. Cirtwill;,citation_author=Miguel B. Araújo;,citation_author=Daniel B. Stouffer;,citation_publication_date=2016;,citation_cover_date=2016;,citation_year=2016;,citation_issue=4;,citation_doi=10.1111/ecog.01941;,citation_issn=1600-0587;,citation_volume=39;,citation_language=en-US;,citation_journal_title=Ecography;">
<meta name="citation_reference" content="citation_title=When is an ecological network complex? Connectance drives degree distribution and emerging network properties;,citation_abstract=Connectance and degree distributions are important components of the structure of ecological networks. In this contribution, we use a statistical argument and simple network generating models to show that properties of the degree distribution are driven by network connectance. We discuss the consequences of this finding for (1) the generation of random networks in null-model analyses, and (2) the interpretation of network structure and ecosystem properties in relationship with degree distribution.;,citation_author=Timothée Poisot;,citation_author=Dominique Gravel;,citation_publication_date=2014-02;,citation_cover_date=2014-02;,citation_year=2014;,citation_doi=10.7717/peerj.251;,citation_issn=2167-8359;,citation_volume=2;,citation_language=en-US;,citation_journal_title=PeerJ;">
<meta name="citation_reference" content="citation_title=Understanding co-occurrence by modelling species simultaneously with a Joint Species Distribution Model (JSDM);,citation_abstract=A primary goal of ecology is to understand the fundamental processes underlying the geographic distributions of species. Two major strands of ecology – habitat modelling and community ecology – approach this problem differently. Habitat modellers often use species distribution models (SDMs) to quantify the relationship between species’ and their environments without considering potential biotic interactions. Community ecologists, on the other hand, tend to focus on biotic interactions and, in observational studies, use co-occurrence patterns to identify ecological processes. Here, we describe a joint species distribution model (JSDM) that integrates these distinct observational approaches by incorporating species co-occurrence data into a SDM. JSDMs estimate distributions of multiple species simultaneously and allow decomposition of species co-occurrence patterns into components describing shared environmental responses and residual patterns of co-occurrence. We provide a general description of the model, a tutorial and code for fitting the model in R. We demonstrate this modelling approach using two case studies: frogs and eucalypt trees in Victoria, Australia. Overall, shared environmental correlations were stronger than residual correlations for both frogs and eucalypts, but there were cases of strong residual correlation. Frog species generally had positive residual correlations, possibly due to the fact these species occurred in similar habitats that were not fully described by the environmental variables included in the JSDM. Eucalypt species that interbreed had similar environmental responses but had negative residual co-occurrence. One explanation is that interbreeding species may not form stable assemblages despite having similar environmental affinities. Environmental and residual correlations estimated from JSDMs can help indicate whether co-occurrence is driven by shared environmental responses or other ecological or evolutionary process (e.g. biotic interactions), or if important predictor variables are missing. JSDMs take into account the fact that distributions of species might be related to each other and thus overcome a major limitation of modelling species distributions independently.;,citation_author=Laura J. Pollock;,citation_author=Reid Tingley;,citation_author=William K. Morris;,citation_author=Nick Golding;,citation_author=Robert B. O’Hara;,citation_author=Kirsten M. Parris;,citation_author=Peter A. Vesk;,citation_author=Michael A. McCarthy;,citation_publication_date=2014;,citation_cover_date=2014;,citation_year=2014;,citation_issue=5;,citation_doi=10.1111/2041-210X.12180;,citation_issn=2041-210X;,citation_volume=5;,citation_language=en-US;,citation_journal_title=Methods in Ecology and Evolution;">
<meta name="citation_reference" content="citation_title=Inferring predator–prey interactions in food webs;,citation_abstract=Food webs are a powerful way to represent the diversity, structure, and function of ecological systems. However, the accurate description of food webs requires significant effort in time and resources, limiting their widespread use in ecological studies. Newly published methods allow for the inference of feeding interactions using proxy variables. Here, we compare the accuracy of two recently described methods, as well as describe a composite model of the two, for the inference of feeding interactions using a large, well-described dataset. Both niche and neutral processes are involved in determining whether or not two species will form a feeding link in communities. Three different models for determining niche constraints of feeding interactions are compared, and all three models are extended by incorporating neutral processes, based on relative abundances. The three models compared here infer niche processes through (a) phylogenetic relationships, (b) local species trait distributions (e.g., body size), and (c) a composite of phylogeny and local traits. We show that all three methods perform well at predicting individual species interactions, and that these individual predictions scale up to the network level, resulting in food web structure of inferred networks being similar to their empirical counterparts. Our results indicate that inferring food web structure using phylogenies can be an efficient way of getting summary webs with minimal data, and offers a conservative test of changes in food web structure, particularly when there is low species turnover between sites. Inferences made using traits require more data, but allows for greater understanding of the mechanisms underlying trophic interactions. A composite model of the two methods provides a framework for investigating the importance of how phylogeny, trait distributions, and relative abundances, affect species interactions, and network structure.;,citation_author=Justin P. F. Pomeranz;,citation_author=Ross M. Thompson;,citation_author=Timothée Poisot;,citation_author=Jon S. Harding;,citation_publication_date=2019;,citation_cover_date=2019;,citation_year=2019;,citation_issue=3;,citation_doi=10.1111/2041-210X.13125;,citation_issn=2041-210X;,citation_volume=10;,citation_language=en-US;,citation_journal_title=Methods in Ecology and Evolution;">
<meta name="citation_reference" content="citation_title=Resolving Food-Web Structure;,citation_abstract=Food webs are a major focus and organizing theme of ecology, but the data used to assemble them are deficient. Early debates over food-web data focused on taxonomic resolution and completeness, lack of which had produced spurious inferences. Recent data are widely believed to be much better and are used extensively in theoretical and meta-analytic research on network ecology. Confidence in these data rests on the assumptions (a) that empiricists correctly identified consumers and their foods and (b) that sampling methods were adequate to detect a near-comprehensive fraction of the trophic interactions between species. Abundant evidence indicates that these assumptions are often invalid, suggesting that most topological food-web data may remain unreliable for inferences about network structure and underlying ecological and evolutionary processes. Morphologically cryptic species are ubiquitous across taxa and regions, and many trophic interactions routinely evade detection by conventional methods. Molecular methods have diagnosed the severity of these problems and are a necessary part of the cure.;,citation_author=Robert M. Pringle;,citation_author=Matthew C. Hutchinson;,citation_publication_date=2020-11;,citation_cover_date=2020-11;,citation_year=2020;,citation_issue=Volume 51, 2020;,citation_doi=10.1146/annurev-ecolsys-110218-024908;,citation_issn=1543-592X, 1545-2069;,citation_volume=51;,citation_language=en-US;,citation_journal_title=Annual Review of Ecology, Evolution and Systematics;,citation_publisher=Annual Reviews;">
<meta name="citation_reference" content="citation_title=Untangling Food Webs;,citation_abstract=Untangling Food Webs was published in Unsolved Problems in Ecology on page 225.;,citation_author=Robert M. Pringle;,citation_publication_date=2020-06;,citation_cover_date=2020-06;,citation_year=2020;,citation_doi=10.1515/9780691195322-020;,citation_isbn=978-0-691-19532-2;,citation_language=en-US;,citation_inbook_title=Unsolved Problems in Ecology;">
<meta name="citation_reference" content="citation_title=Network thinking in ecology and evolution;,citation_abstract=Although pairwise interactions have always had a key role in ecology and evolutionary biology, the recent increase in the amount and availability of biological data has placed a new focus on the complex networks embedded in biological systems. The increased availability of computational tools to store and retrieve biological data has facilitated wide access to these data, not just by biologists but also by specialists from the social sciences, computer science, physics and mathematics. This fusion of interests has led to a burst of research on the properties and consequences of network structure in biological systems. Although traditional measures of network structure and function have started us off on the right foot, an important next step is to create biologically realistic models of network formation, evolution, and function. Here, we review recent applications of network thinking to the evolution of networks at the gene and protein level and to the dynamics and stability of communities. These studies have provided new insights into the organization and function of biological systems by applying existing techniques of network analysis. The current challenge is to recognize the commonalities in evolutionary and ecological applications of network thinking to create a predictive science of biological networks.;,citation_author=Stephen R. Proulx;,citation_author=Daniel E. L. Promislow;,citation_author=Patrick C. Phillips;,citation_publication_date=2005-06;,citation_cover_date=2005-06;,citation_year=2005;,citation_issue=6;,citation_doi=10.1016/j.tree.2005.04.004;,citation_issn=0169-5347;,citation_pmid=16701391;,citation_volume=20;,citation_language=en-US;,citation_journal_title=Trends in Ecology &amp;amp;amp; Evolution;">
<meta name="citation_reference" content="citation_title=Testing the Cascade Model for Food Webs;,citation_abstract=The cascade model is a standard model for food web data. This article describes a method for testing the cascade model that complements existing tests. Like existing tests, the new test is conditional on the number of links observed in the predation matrix. One important feature of the cascade model is that an ordering of species is required, but in practice the true order is unknown. This feature has important implications for tests of the model, both for the new test and for existing tests. The analysis here does not require a single assumed order. Some real food webs are used for illustration.;,citation_author=John M. Roberts;,citation_publication_date=2003;,citation_cover_date=2003;,citation_year=2003;,citation_fulltext_html_url=https://www.jstor.org/stable/1400428;,citation_issue=2;,citation_issn=1085-7117;,citation_volume=8;,citation_journal_title=Journal of Agricultural, Biological, and Environmental Statistics;,citation_publisher=[International Biometric Society, Springer];">
<meta name="citation_reference" content="citation_title=Modeling Food Webs: Exploring Unexplained Structure Using Latent Traits.;,citation_abstract=Several stochastic models have tried to capture the architecture of food webs. This approach is interesting, but it is limited by the fact that different assumptions can yield similar results. To overcome this limitation, we develop a purely statistical approach. Body size in terms of an optimal ratio between prey and predator is used as explanatory variable. In 12 observed food webs, this model predicts, on average, 20% of interactions. To analyze the unexplained part, we introduce a latent term: each species is described by two latent traits, foraging and vulnerability, that represent nonmeasured characteristics of species once the optimal body size has been accounted for. The model now correctly predicts an average of 73% of links. The key features of our approach are that latent traits quantify the structure that is left unexplained by the explanatory variable and that this quantification allows a test of whether independent biological information, such as microhabitat use, camouflage, or phylogeny, explains this structure. We illustrate this method with phylogeny and find that it is linked to one or both latent traits in nine of 12 food webs. Our approach opens the door to the formulation of more complex models that can be applied to any kind of biological network.;,citation_author=Rudolf Philippe Rohr;,citation_author=Heike Scherer;,citation_author=Patrik Kehrli;,citation_author=Christian Mazza;,citation_author=Louis-Félix Bersier;,citation_publication_date=2010-08;,citation_cover_date=2010-08;,citation_year=2010;,citation_issue=2;,citation_doi=10.1086/653667;,citation_issn=0003-0147;,citation_volume=176;,citation_journal_title=The American Naturalist;,citation_publisher=The University of Chicago Press;">
<meta name="citation_reference" content="citation_title=A landscape theory for food web architecture;,citation_abstract=Ecologists have long searched for structures and processes that impart stability in nature. In particular, food web ecology has held promise in tackling this issue. Empirical patterns in food webs have consistently shown that the distributions of species and interactions in nature are more likely to be stable than randomly constructed systems with the same number of species and interactions. Food web ecology still faces two fundamental challenges, however. First, the quantity and quality of food web data required to document both the species richness and the interaction strengths among all species within food webs is largely prohibitive. Second, where food webs have been well documented, spatial and temporal variation in food web structure has been ignored. Conversely, research that has addressed spatial and temporal variation in ecosystems has generally ignored the full complexity of food web architecture. Here, we incorporate empirical patterns, largely from macroecology and behavioural ecology, into a spatially implicit food web structure to construct a simple landscape theory of food web architecture. Such an approach both captures important architectural features of food webs and allows for an exploration of food web structure across a range of spatial scales. Finally, we demonstrated that food webs are hierarchically organized along the spatial and temporal niche axes of species and their utilization of food resources in ways that stabilize ecosystems.;,citation_author=Neil Rooney;,citation_author=Kevin S. McCann;,citation_author=John C. Moore;,citation_publication_date=2008;,citation_cover_date=2008;,citation_year=2008;,citation_issue=8;,citation_doi=10.1111/j.1461-0248.2008.01193.x;,citation_issn=1461-0248;,citation_volume=11;,citation_language=en-US;,citation_journal_title=Ecology Letters;">
<meta name="citation_reference" content="citation_title=Extinction Cascades and Catastrophe in Ancient Food Webs;,citation_abstract=A model is developed to explore the potential responses of paleocommunities to disruptions of primary production during times of mass extinction and ecological crisis. Disruptions of primary production are expected to generate bottom-up cascades of secondary extinction, and these are predictable given species richnesses, functional diversity, and trophic link distributions. If, however, consumers are permitted to compensate for the loss of trophic resources by increasing the intensities of their remaining biotic interactions, top-down driven catastrophic increases of secondary extinction emerge from the model. Both bottom-up and top-down effects are themselves controlled by the geometry of the food webs. The general Phanerozoic trends of increasing taxonomic and ecological diversities, as well as the varying strengths of biotic interactions, have led to food webs of increasing complexity. The frequency of catastrophic secondary extinction increases as food web complexity increases, but increased complexity also serves to dampen the magnitude of the secondary extinctions. When intraguild competitive interactions are included in the model, competitively inferior taxa are observed to possess greater probabilities of survival if the guilds are embedded in simple subnetworks of the overall food web. The result is the emergence of post-extinction guilds dominated by those inferior taxa. These results are congruent with empirical observations of &amp;amp;amp;quot;disaster taxa&amp;quot; dominance after some mass extinction events, and provide a mechanism for the reorganization of ecosystems that is observed after those events. The model makes the testable prediction that dominance by disaster taxa, however, should be observed only when bottom-up disruptions have caused ecosystems to collapse catastrophically.;,citation_author=Peter D. Roopnarine;,citation_publication_date=2006;,citation_cover_date=2006;,citation_year=2006;,citation_fulltext_html_url=https://www.jstor.org/stable/4096814;,citation_issue=1;,citation_issn=0094-8373;,citation_volume=32;,citation_journal_title=Paleobiology;,citation_publisher=Paleontological Society;">
<meta name="citation_reference" content="citation_title=The use of DNA barcodes in food web construction—terrestrial and aquatic ecologists unite!;,citation_abstract=By depicting who eats whom, food webs offer descriptions of how groupings in nature (typically species or populations) are linked to each other. For asking questions on how food webs are built and work, we need descriptions of food webs at different levels of resolution. DNA techniques provide opportunities for highly resolved webs. In this paper, we offer an exposé of how DNA-based techniques, and DNA barcodes in particular, have recently been used to construct food web structure in both terrestrial and aquatic systems. We highlight how such techniques can be applied to simultaneously improve the taxonomic resolution of the nodes of the web (i.e., the species), and the links between them (i.e., who eats whom). We end by proposing how DNA barcodes and DNA information may allow new approaches to the construction of larger interaction webs, and overcome some hurdles to achieving adequate sample size. Most importantly, we propose that the joint adoption and development of these techniques may serve to unite approaches to food web studies in aquatic and terrestrial systems—revealing the extent to which food webs in these environments are structured similarly to or differently from each other, and how they are linked by dispersal.;,citation_author=Tomas Roslin;,citation_author=Sanna Majaneva;,citation_publication_date=2016-09;,citation_cover_date=2016-09;,citation_year=2016;,citation_issue=9;,citation_doi=10.1139/gen-2015-0229;,citation_issn=0831-2796;,citation_volume=59;,citation_journal_title=Genome;,citation_publisher=NRC Research Press;">
<meta name="citation_reference" content="citation_title=Food webs: Experts consuming families of experts;,citation_abstract=Food webs of habitats as diverse as lakes or desert valleys are known to exhibit common “food-web patterns”, but the detailed mechanisms generating these structures have remained unclear. By employing a stochastic, dynamical model, we show that many aspects of the structure of predatory food webs can be understood as the traces of an evolutionary history where newly evolving species avoid direct competition with their relatives. The tendency to avoid sharing natural enemies (apparent competition) with related species is considerably weaker. Thus, “experts consuming families of experts” can be identified as the main underlying food-web pattern. We report the results of a systematic, quantitative model validation showing that the model is surprisingly accurate.;,citation_author=A. G. Rossberg;,citation_author=H. Matsuda;,citation_author=T. Amemiya;,citation_author=K. Itoh;,citation_publication_date=2006-08;,citation_cover_date=2006-08;,citation_year=2006;,citation_issue=3;,citation_doi=10.1016/j.jtbi.2005.12.021;,citation_issn=0022-5193;,citation_volume=241;,citation_journal_title=Journal of Theoretical Biology;">
<meta name="citation_reference" content="citation_title=Ecological network assembly: How the regional metaweb influences local food webs;,citation_abstract=Local food webs result from a sequence of colonisations and extinctions by species from the regional pool or metaweb, that is, the assembly process. Assembly is theorised to be a selective process: whether or not certain species or network structures can persist is partly determined by local processes including habitat filtering and dynamical constraints. Consequently, local food web structure should reflect these processes. The goal of this study was to test evidence for these selective processes by comparing the structural properties of real food webs to the expected distribution given the metaweb. We were particularly interested in ecological dynamics; if the network properties commonly associated with dynamical stability are indeed the result of stability constraints, then they should deviate from expectation in the direction predicted by theory. To create a null expectation, we used the novel approach of randomly assembling model webs by drawing species and interactions from the empirical metaweb. The assembly model permitted colonisation and extinction, and required a consumer species to have at least one prey, but had no habitat type nor population dynamical constraints. Three datasets were used: (a) the marine Antarctic metaweb, with two local food webs; (b) the 50 lakes of the Adirondacks; and (c) the arthropod community from Florida Keys’ classic defaunation experiment. Contrary to our expectations, we found that there were almost no differences between empirical webs and those resulting from the null assembly model. Few empirical food webs showed significant differences with network properties, motif representations and topological roles. Network properties associated with stability did not deviate from expectation in the direction predicted by theory. Our results suggest that-for the commonly used metrics we considered-local food web structure is not strongly influenced by dynamical nor habitat restrictions. Instead, the structure is inherited from the metaweb. This suggests that the network properties typically attributed as causes or consequences of ecological stability are instead a by-product of the assembly process (i.e. spandrels), and may potentially be too coarse to detect the true signal of dynamical constraint.;,citation_author=Leonardo A. Saravia;,citation_author=Tomás I Marina;,citation_author=Nadiah P. Kristensen;,citation_author=Marleen De Troch;,citation_author=Fernando R. Momo;,citation_publication_date=2022;,citation_cover_date=2022;,citation_year=2022;,citation_issue=3;,citation_doi=10.1111/1365-2656.13652;,citation_issn=0021-8790;,citation_volume=91;,citation_language=en-US;,citation_journal_title=Journal of Animal Ecology;,citation_publisher=Wiley;">
<meta name="citation_reference" content="citation_title=The Role of Evolution in Shaping Ecological Networks;,citation_abstract=The structure of ecological networks reflects the evolutionary history of their biotic components, and their dynamics are strongly driven by ecoevolutionary processes. Here, we present an appraisal of recent relevant research, in which the pervasive role of evolution within ecological networks is manifest. Although evolutionary processes are most evident at macroevolutionary scales, they are also important drivers of local network structure and dynamics. We propose components of a blueprint for further research, emphasising process-based models, experimental evolution, and phenotypic variation, across a range of distinct spatial and temporal scales. Evolutionary dimensions are required to advance our understanding of foundational properties of community assembly and to enhance our capability of predicting how networks will respond to impending changes.;,citation_author=Simon T. Segar;,citation_author=Tom M. Fayle;,citation_author=Diane S. Srivastava;,citation_author=Thomas M. Lewinsohn;,citation_author=Owen T. Lewis;,citation_author=Vojtech Novotny;,citation_author=Roger L. Kitching;,citation_author=Sarah C. Maunsell;,citation_publication_date=2020-05;,citation_cover_date=2020-05;,citation_year=2020;,citation_issue=5;,citation_doi=10.1016/j.tree.2020.01.004;,citation_issn=0169-5347;,citation_volume=35;,citation_journal_title=Trends in Ecology &amp;amp;amp; Evolution;">
<meta name="citation_reference" content="citation_title=A framework for reconstructing ancient food webs using functional trait data;,citation_abstract=Food webs provide quantitative insights into the structure and dynamics of ecological communities. Previous work has shown their utility in understanding community responses to modern and ancient perturbations, including anthropogenic change and mass extinctions. However, few ancient food webs have been reconstructed due to difficulties assessing trophic interactions amongst extinct species derived from an incomplete fossil record. We present and assess the Paleo Food web Inference Model (PFIM). PFIM uses functional trait data–predictive of interactions in modern ecosystems and commonly available for fossil organisms–to reconstruct ancient food webs. We test the model by (i) applying it to four modern ecosystems with empirical constrained food webs to directly compare PFIM-constructed networks to their empirical counterparts, (ii) by carefully comparing discrepancies between PFIM-inferred and empirical webs in one of those systems, and (iii) by comparing networks describing feasible trophic interactions (&amp;amp;amp;quot;feasible webs&amp;quot;) with networks to which we superimpose characteristic interaction distributions derived from modern theory (&quot;realized webs&quot;). As a proof of concept, we then apply the method to faunal data from two Cambrian fossil deposits to reconstruct ancient trophic systems. PFIM-inferred feasible food webs successfully predict ~70% of trophic interactions across four modern systems. Furthermore, inferred food webs with enforced interaction distributions (i.e., realized webs) accurately predict ~90% of interactions. Comparisons with a global database of trophic interactions and other food web models, suggest that under sampling of empirical webs accounts for up to 21% of the remaining differences between PFIM and empirical food webs. Food webs can be reasonably approximated by inferring trophic interactions based upon life habit traits. This study provides the foundation to use trait-based inference models across the fossil record to examine ancient food webs and community evolution.;,citation_author=Jack O. Shaw;,citation_author=Alexander M. Dunhill;,citation_author=Andrew P. Beckerman;,citation_author=Jennifer A. Dunne;,citation_author=Pincelli M. Hull;,citation_publication_date=2024-01;,citation_cover_date=2024-01;,citation_year=2024;,citation_doi=10.1101/2024.01.30.578036;,citation_language=en-US;,citation_publisher=bioRxiv;">
<meta name="citation_reference" content="citation_title=On Lumping Species in Food Webs;,citation_abstract=Species in community food webs are commonly aggregated or lumped on the basis of overlap in predators and prey. This note reports an unexpected result of lumping species in observed food webs and simulated food webs. The main result is that it is much easier to lump species in observed webs than in simulated webs. The reason is that there is much more overlap in predators and prey for species in observed webs than in simulated webs. This points to a fundamental deficiency in web assembly models that assume a random distribution of links.;,citation_author=Andrew R. Solow;,citation_author=Andrew R. Beet;,citation_publication_date=1998;,citation_cover_date=1998;,citation_year=1998;,citation_issue=6;,citation_doi=10.1890/0012-9658(1998)079[2013:OLSIFW]2.0.CO;2;,citation_issn=1939-9170;,citation_volume=79;,citation_language=en-US;,citation_journal_title=Ecology;">
<meta name="citation_reference" content="citation_title=Rigorous (in)validation of ecological models;,citation_abstract=The complexity of ecosystems poses a formidable challenge for confidently invalidating ecological models, as current practices struggle to distinguish model inadequacies from the confounding effects of unobserved biotic or abiotic factors. The prevailing inability to falsify models has resulted in an accumulation of models but not an accumulation of confidence. Here, we introduce a new approach rooted in queueing theory, termed the covariance criteria, that establishes a rigorous test for model validity based on covariance relationships between observable quantities. These criteria set a high bar for models to pass by specifying necessary conditions that must hold regardless of unobserved factors. We demonstrate the broad applicability and discriminatory power of the covariance criteria by applying them to three long-standing challenges in ecological theory: resolving competing models of predator-prey functional responses, disentangling ecological and evolutionary dynamics in systems with rapid evolution, and detecting the often-elusive influence of higher-order species interactions. Across these diverse case studies, the covariance criteria consistently rule out inadequate models, while building strong confidence in those that provide strategically useful approximations. The covariance criteria approach is mathematically rigorous, computationally efficient, and often non-parametric, making it immediately applicable to existing data and models.;,citation_author=Chuliang Song;,citation_author=Jonathan M. Levine;,citation_publication_date=2024-09;,citation_cover_date=2024-09;,citation_year=2024;,citation_doi=10.1101/2024.09.19.613075;,citation_language=en-US;,citation_publisher=bioRxiv;">
<meta name="citation_reference" content="citation_title=Structural dynamics and robustness of food webs;,citation_abstract=Ecology Letters (2010) 13: 891–899 Abstract Food web structure plays an important role when determining robustness to cascading secondary extinctions. However, existing food web models do not take into account likely changes in trophic interactions (“rewiring”) following species loss. We investigated structural dynamics in 12 empirically documented food webs by simulating primary species loss using three realistic removal criteria, and measured robustness in terms of subsequent secondary extinctions. In our model, novel trophic interactions can be established between predators and food items not previously consumed following the loss of competing predator species. By considering the increase in robustness conferred through rewiring, we identify a new category of species – overlap species – which promote robustness as shown by comparing simulations incorporating structural dynamics to those with static topologies. The fraction of overlap species in a food web is highly correlated with this increase in robustness; whereas species richness and connectance are uncorrelated with increased robustness. Our findings underline the importance of compensatory mechanisms that may buffer ecosystems against environmental change, and highlight the likely role of particular species that are expected to facilitate this buffering.;,citation_author=Phillip P. A. Staniczenko;,citation_author=Owen T. Lewis;,citation_author=Nick S. Jones;,citation_author=Felix Reed-Tsochas;,citation_publication_date=2010;,citation_cover_date=2010;,citation_year=2010;,citation_issue=7;,citation_doi=10.1111/j.1461-0248.2010.01485.x;,citation_issn=1461-0248;,citation_volume=13;,citation_language=en-US;,citation_journal_title=Ecology Letters;">
<meta name="citation_reference" content="citation_title=Pairwise learning for predicting pollination interactions based on traits and phylogeny;,citation_abstract=Mutualistic bee–plant interaction networks are a vital part of terrestrial ecosystems. They frequently arise through co-evolutionary processes, which match the traits of both partners, facilitating their interaction. Insights in these interaction mechanisms are vital to be able to manage changing ecosystems. This entails the need for models to predict species interaction networks in general and pollination networks in particular. We show how kernel-based pairwise learning can predict bee–plant interactions based on the traits and the phylogeny of the plant and bee species. The traits and the phylogeny of the plant and bee species proved to be highly predictive. Although the traits were slightly more informative compared to the phylogeny, the best results were obtained by combining both the traits and the phylogeny in the model Notably, the model performance varied greatly depending on whether the goal was to pinpoint missing interactions in the network or to predict interactions for new bee species, new plant species, or both. This issue highlights the importance of proper stratification when fitting biological network prediction models. Our model, however, showed the capacity to generalize beyond the original dataset provided by FlorAbeilles. The model was validated by predicting potentially interacting plant species for the invasive bee species Megachile sculpturalis. Four out of the five plant species identified by the model could be validated based on literature.;,citation_author=Michiel Stock;,citation_publication_date=2021;,citation_cover_date=2021;,citation_year=2021;,citation_language=en-US;,citation_journal_title=Ecological Modelling;">
<meta name="citation_reference" content="citation_title=All ecological models are wrong, but some are useful;,citation_abstract=In Focus: Curtsdotter, A., Banks, H. T., Banks, J. E., Jonsson, M., Jonsson, T., Laubmeier, A. N., … Bommarco, R. (2019). Ecosystem function in predator-prey food webs—Confronting dynamic models with empirical data. Journal of Animal Ecology, 88, https://doi.org/10.1111/1365-2656.12892 Species’ population dynamics are influenced by a variety of abiotic and biotic factors. Curtsdotter et al. (2019) used a food web model to investigate the role of predator–prey interactions in the population dynamics of the bird cherry-oat aphid Rhopalosiphum padi. Their analysis hinged on linking the observed population dynamics to a mathematical description of the multi-species system via inverse methods—an approach less utilized in ecology but that allows one to search a wide space of possible parameterizations and identify best-fit model parameters. By scrutinizing the fit of this model to observed aphid population dynamics in 10 separate barley fields, they identified fields in which predation was the key driving force; in others, they found that accurate predictions depended on the existence of an unpredictable and unidentified extrinsic driver of aphid mortality. By scrutinizing areas where the model gave poor or biologically counterintuitive fits, their study provides a path forward to better link ecological theory to ecosystem function.;,citation_author=Daniel B. Stouffer;,citation_publication_date=2019;,citation_cover_date=2019;,citation_year=2019;,citation_issue=2;,citation_doi=10.1111/1365-2656.12949;,citation_issn=1365-2656;,citation_volume=88;,citation_language=en-US;,citation_journal_title=Journal of Animal Ecology;">
<meta name="citation_reference" content="citation_title=Evidence for the existence of a robust pattern of prey selection in food webs;,citation_abstract=Food webs aim to provide a thorough representation of the trophic interactions found in an ecosystem. The complexity of empirical food webs, however, is leading many ecologists to focus dynamic ecosystem studies on smaller microcosm or mesocosm studies based upon community modules, which comprise three to five species and the interactions likely to have ecological relevance. We provide here a structural counterpart to community modules. We investigate food-web “motifs” which are n-species connected subgraphs found within the food web. Remarkably, we find that the over- and under-representation of three-species motifs in empirical food webs can be understood through comparison to a static food-web model, the niche model. Our result conclusively demonstrates that predation upon species with some “characteristic” niche value is the prey selection mechanism consistent with the structural properties of empirical food webs.;,citation_author=Daniel B Stouffer;,citation_author=Juan Camacho;,citation_author=Wenxin Jiang;,citation_author=Luís A Nunes Amaral;,citation_publication_date=2007-06;,citation_cover_date=2007-06;,citation_year=2007;,citation_issue=1621;,citation_doi=10.1098/rspb.2007.0571;,citation_volume=274;,citation_journal_title=Proceedings of the Royal Society B: Biological Sciences;,citation_publisher=Royal Society;">
<meta name="citation_reference" content="citation_title=Quantitative Patterns in the Structure of Model and Empirical Food Webs;,citation_abstract=Understanding the structure of food webs, and the mechanisms that give rise to this structure, is one of the most challenging problems in ecology. We analyze from a statistical physics perspective the network structure of model food webs and of 15 community food webs from a variety of environments, including freshwater, marine, estuarine, and terrestrial environments. We perform a theoretical analysis of two recently proposed models for food webs, the niche model of R. J. Williams and N. D. Martinez and the nested-hierarchy model of M.-F. Cattin et al. We find that the two models generate distributions of numbers of prey, predators, and links that are described by the same analytical expressions. Our analytical treatment reveals that a model’s capacity to reproduce empirical data is principally determined by its ability to satisfy two conditions: (1) the species’ niche values form a totally ordered set and (2) each species has a specific exponentially decaying probability of preying on a given fraction of the species with lower niche values. To test this hypothesis, we generalize the cascade model of J. E. Cohen and C. M. Newman so that it satisfies condition 2 and find that the new model is able to reproduce the properties of empirical food webs, validating our hypothesis. We use our analytical predictions as a guide to the analysis of 15 of the most complete empirical food webs available. We demonstrate that the quantitative unifying patterns that describe the properties of the food-web models considered earlier also describe the majority of the empirical webs considered. We find good agreement between the empirical data and the models for the average distance between species and the average clustering coefficient. Our results strongly support two hypotheses: first, that any model satisfying the two conditions we identify will accurately reproduce a number of the statistical properties of empirical food webs, and second, that the empirical distributions of number of prey and number of predators follow universal functional forms that, without free parameters, match our analytical predictions.;,citation_author=D. B. Stouffer;,citation_author=J. Camacho;,citation_author=R. Guimerà;,citation_author=C. A. Ng;,citation_author=L. A. Nunes Amaral;,citation_publication_date=2005;,citation_cover_date=2005;,citation_year=2005;,citation_issue=5;,citation_doi=10.1890/04-0957;,citation_issn=1939-9170;,citation_volume=86;,citation_language=en-US;,citation_journal_title=Ecology;">
<meta name="citation_reference" content="citation_title=A robust measure of food web intervality;,citation_abstract=Intervality of a food web is related to the number of trophic dimensions characterizing the niches in a community. We introduce here a mathematically robust measure for food web intervality. It has previously been noted that empirical food webs are not strictly interval; however, upon comparison to suitable null hypotheses, we conclude that empirical food webs actually do exhibit a strong bias toward contiguity of prey, that is, toward intervality. Further, our results strongly suggest that empirically observed species and their diets can be mapped onto a single dimension. This finding validates a critical assumption in the recently proposed static niche model and provides guidance for ongoing efforts to develop dynamic models of ecosystems.;,citation_author=Daniel B. Stouffer;,citation_author=Juan Camacho;,citation_author=Luís A. Nunes Amaral;,citation_publication_date=2006-12;,citation_cover_date=2006-12;,citation_year=2006;,citation_issue=50;,citation_doi=10.1073/pnas.0603844103;,citation_volume=103;,citation_journal_title=Proceedings of the National Academy of Sciences;,citation_publisher=Proceedings of the National Academy of Sciences;">
<meta name="citation_reference" content="citation_title=Food web reconstruction through phylogenetic transfer of low-rank network representation;,citation_abstract=Despite their importance in many ecological processes, collecting data and information on ecological interactions is an exceedingly challenging task. For this reason, large parts of the world have a data deficit when it comes to species interactions and how the resulting networks are structured. As data collection alone is unlikely to be sufficient, community ecologists must adopt predictive methods. We present a methodological framework that uses graph embedding and transfer learning to assemble a predicted list of trophic interactions of a species pool for which their interactions are unknown. Specifically, we “learn” the information (latent traits) of species from a known interaction network and infer the latent traits of another species pool for which we have no a priori interaction data based on their phylogenetic relatedness to species from the known network. The latent traits can then be used to predict interactions and construct an interaction network. Here we assembled a metaweb for Canadian mammals derived from interactions in the European food web, despite only 4% of common species being shared between the two locations. The results of the predictive model are compared against databases of recorded pairwise interactions, showing that we correctly recover 91% of known interactions. The framework itself is robust even when the known network is incomplete or contains spurious interactions making it an ideal candidate as a tool for filling gaps when it comes to species interactions. We provide guidance on how this framework can be adapted by substituting some approaches or predictors in order to make it more generally applicable.;,citation_author=Tanya Strydom;,citation_author=Salomé Bouskila;,citation_author=Francis Banville;,citation_author=Ceres Barros;,citation_author=Dominique Caron;,citation_author=Maxwell J. Farrell;,citation_author=Marie-Josée Fortin;,citation_author=Victoria Hemming;,citation_author=Benjamin Mercier;,citation_author=Laura J. Pollock;,citation_author=Rogini Runghen;,citation_author=Giulio V. Dalla Riva;,citation_author=Timothée Poisot;,citation_publication_date=2022;,citation_cover_date=2022;,citation_year=2022;,citation_issue=12;,citation_doi=10.1111/2041-210X.13835;,citation_issn=2041-210X;,citation_volume=13;,citation_language=en-US;,citation_journal_title=Methods in Ecology and Evolution;">
<meta name="citation_reference" content="citation_title=Graph embedding and transfer learning can help predict potential species interaction networks despite data limitations;,citation_abstract=Metawebs (networks of potential interactions within a species pool) are a powerful abstraction to understand how large-scale species interaction networks are structured. Because metawebs are typically expressed at large spatial and taxonomic scales, assembling them is a tedious and costly process; predictive methods can help circumvent the limitations in data deficiencies, by providing a first approximation of metawebs. One way to improve our ability to predict metawebs is to maximize available information by using graph embeddings, as opposed to an exhaustive list of species interactions. Graph embedding is an emerging field in machine learning that holds great potential for ecological problems. Here, we outline how the challenges associated with inferring metawebs line-up with the advantages of graph embeddings; followed by a discussion as to how the choice of the species pool has consequences on the reconstructed network, specifically as to the role of human-made (or arbitrarily assigned) boundaries and how these may influence ecological hypotheses.;,citation_author=Tanya Strydom;,citation_author=Salomé Bouskila;,citation_author=Francis Banville;,citation_author=Ceres Barros;,citation_author=Dominique Caron;,citation_author=Maxwell J. Farrell;,citation_author=Marie-Josée Fortin;,citation_author=Benjamin Mercier;,citation_author=Laura J. Pollock;,citation_author=Rogini Runghen;,citation_author=Giulio V. Dalla Riva;,citation_author=Timothée Poisot;,citation_publication_date=2023;,citation_cover_date=2023;,citation_year=2023;,citation_issue=12;,citation_doi=10.1111/2041-210X.14228;,citation_issn=2041-210X;,citation_volume=14;,citation_language=en-US;,citation_journal_title=Methods in Ecology and Evolution;">
<meta name="citation_reference" content="citation_title=A roadmap towards predicting species interaction networks (across space and time);,citation_abstract=Networks of species interactions underpin numerous ecosystem processes, but comprehensively sampling these interactions is difficult. Interactions intrinsically vary across space and time, and given the number of species that compose ecological communities, it can be tough to distinguish between a true negative (where two species never interact) from a false negative (where two species have not been observed interacting even though they actually do). Assessing the likelihood of interactions between species is an imperative for several fields of ecology. This means that to predict interactions between species—and to describe the structure, variation, and change of the ecological networks they form—we need to rely on modelling tools. Here, we provide a proof-of-concept, where we show how a simple neural network model makes accurate predictions about species interactions given limited data. We then assess the challenges and opportunities associated with improving interaction predictions, and provide a conceptual roadmap forward towards predictive models of ecological networks that is explicitly spatial and temporal. We conclude with a brief primer on the relevant methods and tools needed to start building these models, which we hope will guide this research programme forward. This article is part of the theme issue “Infectious disease macroecology: parasite diversity and dynamics across the globe”.;,citation_author=Tanya Strydom;,citation_author=Michael D. Catchen;,citation_author=Francis Banville;,citation_author=Dominique Caron;,citation_author=Gabriel Dansereau;,citation_author=Philippe Desjardins-Proulx;,citation_author=Norma R. Forero-Muñoz;,citation_author=Gracielle Higino;,citation_author=Benjamin Mercier;,citation_author=Andrew Gonzalez;,citation_author=Dominique Gravel;,citation_author=Laura Pollock;,citation_author=Timothée Poisot;,citation_publication_date=2021-11;,citation_cover_date=2021-11;,citation_year=2021;,citation_issue=1837;,citation_doi=10.1098/rstb.2021.0063;,citation_volume=376;,citation_journal_title=Philosophical Transactions of the Royal Society B: Biological Sciences;,citation_publisher=Royal Society;">
<meta name="citation_reference" content="citation_title=SVD Entropy Reveals the High Complexity of Ecological Networks;,citation_abstract=Quantifying the complexity of ecological networks has remained elusive. Primarily, complexity has been defined on the basis of the structural (or behavioural) complexity of the system. These definitions ignore the notion of ’physical complexity’, which can measure the amount of information contained in an ecological network, and how difficult it would be to compress. We present relative rank deficiency and SVD entropy as measures of ’external’ and ’internal’ complexity respectively. Using bipartite ecological networks, we find that they all show a very high, almost maximal, physical complexity. Pollination networks, in particular, are more complex when compared to other types of interactions. In addition, we find that SVD entropy relates to other structural measures of complexity (nestedness, connectance, and spectral radius), but does not inform about the resilience of a network when using simulated extinction cascades, which has previously been reported for structural measures of complexity. We argue that SVD entropy provides a fundamentally more ’correct’ measure of network complexity and should be added to the toolkit of descriptors of ecological networks moving forward.;,citation_author=Tanya Strydom;,citation_author=Giulio V. Dalla Riva;,citation_author=Timothée Poisot;,citation_publication_date=2021;,citation_cover_date=2021;,citation_year=2021;,citation_doi=10.3389/fevo.2021.623141;,citation_issn=2296-701X;,citation_volume=9;,citation_language=en-US;,citation_journal_title=Frontiers in Ecology and Evolution;,citation_publisher=Frontiers;">
<meta name="citation_reference" content="citation_title=Finding missing links in interaction networks;,citation_abstract=Documenting which species interact within ecological communities is challenging and labor intensive. As a result, many interactions remain unrecorded, potentially distorting our understanding of network structure and dynamics. We test the utility of four structural models and a new coverage-deficit model for predicting missing links in both simulated and empirical bipartite networks. We find they can perform well, although the predictive power of structural models varies with the underlying network structure. The accuracy of predictions can be improved by ensembling multiple models. Augmenting observed networks with most-likely missing links improves estimates of qualitative network metrics. Tools to identify likely missing links can be simple to implement, allowing the prioritization of research effort and more robust assessment of network properties.;,citation_author=J. Christopher D. Terry;,citation_author=Owen T. Lewis;,citation_publication_date=2020;,citation_cover_date=2020;,citation_year=2020;,citation_issue=7;,citation_doi=10.1002/ecy.3047;,citation_issn=1939-9170;,citation_volume=101;,citation_language=en-US;,citation_journal_title=Ecology;">
<meta name="citation_reference" content="citation_title=Aristotle and Theophrastus on plant-animal interactions;,citation_abstract=Aristotle and Theophrastus, the last great philosophers and scientists of Greek Classical Antiquity, are the founding fathers of Zoology and Botany, respectively; they should also be honoured as the co-founders of Biology. They were close friends and life-long collaborators who evidently decided to pursue an organized study of the living world, probably in Lesbos at 344 BC (the landmark for the creation of the Science of Biology). The product of their division of labour, the voluminous zoological and botanical works of Aristotle and Theophrastus, respectively, were actually used contemporaneously as university textbooks by the students of the Lyceum.;,citation_author=Costas A. Thanos;,citation_editor=M. Arianoutsou;,citation_editor=R. H. Groves;,citation_publication_date=1994;,citation_cover_date=1994;,citation_year=1994;,citation_doi=10.1007/978-94-011-0908-6_1;,citation_isbn=978-94-011-0908-6;,citation_language=en-US;,citation_inbook_title=Plant-animal interactions in Mediterranean-type ecosystems;">
<meta name="citation_reference" content="citation_title=Food webs: Reconciling the structure and function of biodiversity;,citation_author=Ross M. Thompson;,citation_author=Ulrich Brose;,citation_author=Jennifer Dunne;,citation_author=Robert O. Hall;,citation_author=Sally Hladyz;,citation_author=Roger L. Kitching;,citation_author=Neo D. Martinez;,citation_author=Heidi Rantala;,citation_author=Tamara N. Romanuk;,citation_author=Daniel B. Stouffer;,citation_author=Jason M. Tylianakis;,citation_publication_date=2012-12;,citation_cover_date=2012-12;,citation_year=2012;,citation_issue=12;,citation_doi=10.1016/j.tree.2012.08.005;,citation_volume=27;,citation_journal_title=Trends in Ecology &amp;amp;amp; Evolution;">
<meta name="citation_reference" content="citation_title=Navigating the integration of biotic interactions in biogeography;,citation_abstract=Biotic interactions are widely recognised as the backbone of ecological communities, but how best to study them is a subject of intense debate, especially at macro-ecological scales. While some researchers claim that biotic interactions need to be observed directly, others use proxies and statistical approaches to infer them. Despite this ambiguity, studying and predicting the influence of biotic interactions on biogeographic patterns is a thriving area of research with crucial implications for conservation. Three distinct approaches are currently being explored. The first approach involves empirical observation and measurement of biotic interactions’ effects on species demography in laboratory or field settings. While these findings contribute to theory and to understanding species’ demographies, they can be challenging to generalise on a larger scale. The second approach centers on inferring biotic associations from observed co-occurrences in space and time. The goal is to distinguish the environmental and biotic effects on species distributions. The third approach constructs extensive potential interaction networks, known as metanetworks, by leveraging existing knowledge about species ecology and interactions. This approach analyses local realisations of these networks using occurrence data and allows understanding large distributions of multi-taxa assemblages. In this piece, we appraise these three approaches, highlighting their respective strengths and limitations. Instead of seeing them as conflicting, we advocate for their integration to enhance our understanding and expand applications in the emerging field of interaction biogeography. This integration shows promise for ecosystem understanding and management in the Anthropocene era.;,citation_author=Wilfried Thuiller;,citation_author=Irene Calderón-Sanou;,citation_author=Loïc Chalmandrier;,citation_author=Pierre Gaüzère;,citation_author=Louise M. J. O’Connor;,citation_author=Marc Ohlmann;,citation_author=Giovanni Poggiato;,citation_author=Tamara Münkemüller;,citation_publication_date=2024;,citation_cover_date=2024;,citation_year=2024;,citation_issue=4;,citation_doi=10.1111/jbi.14734;,citation_issn=1365-2699;,citation_volume=51;,citation_language=en-US;,citation_journal_title=Journal of Biogeography;">
<meta name="citation_reference" content="citation_title=A practical guide to selecting models for exploration, inference, and prediction in ecology;,citation_abstract=Selecting among competing statistical models is a core challenge in science. However, the many possible approaches and techniques for model selection, and the conflicting recommendations for their use, can be confusing. We contend that much confusion surrounding statistical model selection results from failing to first clearly specify the purpose of the analysis. We argue that there are three distinct goals for statistical modeling in ecology: data exploration, inference, and prediction. Once the modeling goal is clearly articulated, an appropriate model selection procedure is easier to identify. We review model selection approaches and highlight their strengths and weaknesses relative to each of the three modeling goals. We then present examples of modeling for exploration, inference, and prediction using a time series of butterfly population counts. These show how a model selection approach flows naturally from the modeling goal, leading to different models selected for different purposes, even with exactly the same data set. This review illustrates best practices for ecologists and should serve as a reminder that statistical recipes cannot substitute for critical thinking or for the use of independent data to test hypotheses and validate predictions.;,citation_author=Andrew T. Tredennick;,citation_author=Giles Hooker;,citation_author=Stephen P. Ellner;,citation_author=Peter B. Adler;,citation_publication_date=2021;,citation_cover_date=2021;,citation_year=2021;,citation_issue=6;,citation_doi=10.1002/ecy.3336;,citation_issn=1939-9170;,citation_volume=102;,citation_language=en-US;,citation_journal_title=Ecology;">
<meta name="citation_reference" content="citation_title=Inferring the mammal tree: Species-level sets of phylogenies for questions in ecology, evolution, and conservation;,citation_abstract=Big, time-scaled phylogenies are fundamental to connecting evolutionary processes to modern biodiversity patterns. Yet inferring reliable phylogenetic trees for thousands of species involves numerous trade-offs that have limited their utility to comparative biologists. To establish a robust evolutionary timescale for all approximately 6,000 living species of mammals, we developed credible sets of trees that capture root-to-tip uncertainty in topology and divergence times. Our “backbone-and-patch” approach to tree building applies a newly assembled 31-gene supermatrix to two levels of Bayesian inference: (1) backbone relationships and ages among major lineages, using fossil node or tip dating, and (2) species-level “patch” phylogenies with nonoverlapping in-groups that each correspond to one representative lineage in the backbone. Species unsampled for DNA are either excluded (“DNA-only” trees) or imputed within taxonomic constraints using branch lengths drawn from local birth–death models (“completed” trees). Joining time-scaled patches to backbones results in species-level trees of extant Mammalia with all branches estimated under the same modeling framework, thereby facilitating rate comparisons among lineages as disparate as marsupials and placentals. We compare our phylogenetic trees to previous estimates of mammal-wide phylogeny and divergence times, finding that (1) node ages are broadly concordant among studies, and (2) recent (tip-level) rates of speciation are estimated more accurately in our study than in previous “supertree” approaches, in which unresolved nodes led to branch-length artifacts. Credible sets of mammalian phylogenetic history are now available for download at http://vertlife.org/phylosubsets, enabling investigations of long-standing questions in comparative biology.;,citation_author=Nathan S. Upham;,citation_author=Jacob A. Esselstyn;,citation_author=Walter Jetz;,citation_publication_date=2019-12;,citation_cover_date=2019-12;,citation_year=2019;,citation_issue=12;,citation_doi=10.1371/journal.pbio.3000494;,citation_issn=1545-7885;,citation_volume=17;,citation_language=en-US;,citation_journal_title=PLOS Biology;,citation_publisher=Public Library of Science;">
<meta name="citation_reference" content="citation_title=A bioenergetic framework for aboveground terrestrial food webs;,citation_abstract=Bioenergetic approaches have been greatly influential for understanding community functioning and stability and predicting effects of environmental changes on biodiversity. These approaches use allometric relationships to establish species’ trophic interactions and consumption rates and have been successfully applied to aquatic ecosystems. Terrestrial ecosystems, where body mass is less predictive of plant–consumer interactions, present inherent challenges that these models have yet to meet. Here, we discuss the processes governing terrestrial plant–consumer interactions and develop a bioenergetic framework integrating those processes. Our framework integrates bioenergetics specific to terrestrial plants and their consumers within a food web approach while also considering mutualistic interactions. Such a framework is poised to advance our understanding of terrestrial food webs and to predict their responses to environmental changes.;,citation_author=Fernanda S. Valdovinos;,citation_author=Kayla R. S. Hale;,citation_author=Sabine Dritz;,citation_author=Paul R. Glaum;,citation_author=Kevin S. McCann;,citation_author=Sophia M. Simon;,citation_author=Elisa Thébault;,citation_author=William C. Wetzel;,citation_author=Kate L. Wootton;,citation_author=Justin D. Yeakel;,citation_publication_date=2023-03;,citation_cover_date=2023-03;,citation_year=2023;,citation_issue=3;,citation_doi=10.1016/j.tree.2022.11.004;,citation_issn=0169-5347;,citation_volume=38;,citation_journal_title=Trends in Ecology &amp;amp;amp; Evolution;">
<meta name="citation_reference" content="citation_title=Uniting pattern and process in plant–animal mutualistic networks: A review;,citation_abstract=Ecologists and evolutionary biologists are becoming increasingly interested in networks as a framework to study plant–animal mutualisms within their ecological context. Although such focus on networks has brought about important insights into the structure of these interactions, relatively little is still known about the mechanisms behind these patterns.The aim in this paper is to offer an overview of the mechanisms influencing the structure of plant–animal mutualistic networks. A brief summary is presented of the salient network patterns, the potential mechanisms are discussed and the studies that have evaluated them are reviewed. This review shows that researchers of plant–animal mutualisms have made substantial progress in the understanding of the processes behind the patterns observed in mutualistic networks. At the same time, we are still far from a thorough, integrative mechanistic understanding. We close with specific suggestions for directions of future research, which include developing methods to evaluate the relative importance of mechanisms influencing network patterns and focusing research efforts on selected representative study systems throughout the world.;,citation_author=Diego P. Vázquez;,citation_author=Nico Blüthgen;,citation_author=Luciano Cagnolo;,citation_author=Natacha P. Chacoff;,citation_publication_date=2009-06;,citation_cover_date=2009-06;,citation_year=2009;,citation_issue=9;,citation_doi=10.1093/aob/mcp057;,citation_issn=0305-7364;,citation_volume=103;,citation_journal_title=Annals of Botany;">
<meta name="citation_reference" content="citation_title=Major dimensions in food-web structure properties.;,citation_author=Jan E Vermaat;,citation_author=Jennifer A Dunne;,citation_author=Alison J Gilbert;,citation_publication_date=2009;,citation_cover_date=2009;,citation_year=2009;,citation_fulltext_html_url=https://www.ncbi.nlm.nih.gov/pubmed/19294932;,citation_issue=1;,citation_volume=90;,citation_journal_title=Ecology;">
<meta name="citation_reference" content="citation_title=The Probabilistic Niche Model Reveals the Niche Structure and Role of Body Size in a Complex Food Web;,citation_abstract=The niche model has been widely used to model the structure of complex food webs, and yet the ecological meaning of the single niche dimension has not been explored. In the niche model, each species has three traits, niche position, diet position and feeding range. Here, a new probabilistic niche model, which allows the maximum likelihood set of trait values to be estimated for each species, is applied to the food web of the Benguela fishery. We also developed the allometric niche model, in which body size is used as the niche dimension. About 80% of the links in the empirical data are predicted by the probabilistic niche model, a significant improvement over recent models. As in the niche model, species are uniformly distributed on the niche axis. Feeding ranges are exponentially distributed, but diet positions are not uniformly distributed below the predator. Species traits are strongly correlated with body size, but the allometric niche model performs significantly worse than the probabilistic niche model. The best-fit parameter set provides a significantly better model of the structure of the Benguela food web than was previously available. The methodology allows the identification of a number of taxa that stand out as outliers either in the model’s poor performance at predicting their predators or prey or in their parameter values. While important, body size alone does not explain the structure of the one-dimensional niche.;,citation_author=Richard J. Williams;,citation_author=Ananthi Anandanadesan;,citation_author=Drew Purves;,citation_publication_date=2010-08;,citation_cover_date=2010-08;,citation_year=2010;,citation_issue=8;,citation_doi=10.1371/journal.pone.0012092;,citation_issn=1932-6203;,citation_volume=5;,citation_language=en-US;,citation_journal_title=PLOS ONE;,citation_publisher=Public Library of Science;">
<meta name="citation_reference" content="citation_title=Simple rules yield complex food webs;,citation_abstract=Several of the most ambitious theories in ecology1,2,3,4,5,6,7,8,9,10,11,12,13,14 describe food webs that document the structure of strong and weak trophic links9 that is responsible for ecological dynamics among diverse assemblages of species4,11,12,13. Early mechanism-based theory asserted that food webs have little omnivory and several properties that are independent of species richness1,2,3,4,6. This theory was overturned by empirical studies that found food webs to be much more complex5,7,8,9,14,15,16,17,18, but these studies did not provide mechanistic explanations for the complexity9. Here we show that a remarkably simple model fills this scientific void by successfully predicting key structural properties of the most complex and comprehensive food webs in the primary literature. These properties include the fractions of species at top, intermediate and basal trophic levels, the means and variabilities of generality, vulnerability and food-chain length, and the degrees of cannibalism, omnivory, looping and trophic similarity. Using only two empirical parameters, species number and connectance, our “niche model” extends the existing “cascade model”3,19 and improves its fit ten-fold by constraining species to consume a contiguous sequence of prey in a one-dimensional trophic niche20.;,citation_author=Richard J. Williams;,citation_author=Neo D. Martinez;,citation_publication_date=2000-03;,citation_cover_date=2000-03;,citation_year=2000;,citation_issue=6774;,citation_doi=10.1038/35004572;,citation_issn=1476-4687;,citation_volume=404;,citation_language=en-US;,citation_journal_title=Nature;,citation_publisher=Nature Publishing Group;">
<meta name="citation_reference" content="citation_title=Success and its limits among structural models of complex food webs;,citation_abstract=1 Following the development of the relatively successful niche model, several other simple structural food web models have been proposed. These models predict the detailed structure of complex food webs given only two input parameters, the numbers of species and the number of feeding links among them. 2 The models claim different degrees of success but have not been compared consistently with each other or with the empirical data. We compared the performance of five structural models rigorously against 10 empirical food webs from a variety of aquatic and terrestrial habitats containing 25–92 species and 68–997 links. 3 All models include near-hierarchical ordering of species’ consumption and have identical distributions of the number of prey of each consumer species, but differ in the extent to which species’ diets are required to be contiguous and the rules used to assign feeding links. 4 The models perform similarly on a range of food-web properties, including the fraction of top, intermediate and basal species, the standard deviations of generality and connectivity and the fraction of herbivores and omnivores. 5 For other properties, including the standard deviation of vulnerability, the fraction of cannibals and species in loops, mean trophic level, path length, clustering coefficient, maximum similarity and diet discontinuity, there are significant differences in the performance of the different models. 6 While the empirical data do not support the niche model’s assumption of diet contiguity, models which relax this assumption all have worse overall performance than the niche model. All the models under-estimate severely the fraction of species that are herbivores and exhibit other important failures that need to be addressed in future research.;,citation_author=Richard J. Williams;,citation_author=Neo D. Martinez;,citation_publication_date=2008;,citation_cover_date=2008;,citation_year=2008;,citation_issue=3;,citation_doi=10.1111/j.1365-2656.2008.01362.x;,citation_issn=1365-2656;,citation_volume=77;,citation_language=en-US;,citation_journal_title=Journal of Animal Ecology;">
<meta name="citation_reference" content="citation_title=EltonTraits 1.0: Species-level foraging attributes of the world’s birds and mammals;,citation_abstract=Species are characterized by physiological, behavioral, and ecological attributes that are all subject to varying evolutionary and ecological constraints and jointly determine species’ role and function in ecosystems. Attributes such as diet, foraging strata, foraging time, and body size, in particular, characterize a large portion of the “Eltonian” niches of species. Here we present a global species-level compilation of these key attributes for all 9993 and 5400 extant bird and mammal species derived from key literature sources. Global handbooks and monographs allowed the consistent sourcing of attributes for most species. For diet and foraging stratum we followed a defined protocol to translate the verbal descriptions into standardized, semiquantitative information about relative importance of different categories. Together with body size (continuous) and activity time (categorical) this enables a much finer distinction of species’ foraging ecology than typical categorical guild assignments allow. Attributes lacking information for specific species are flagged, and interpolated values based on taxonomy are provided instead. The presented data set is limited by, among others, these select cases missing observed data, by errors and uncertainty in the expert assessment as presented in the literature, and by the lack of intraspecific information. However, the standardized and transparent nature and complete global coverage of the data set should support an array of potential studies in biogeography, community ecology, macroevolution, global change biology, and conservation. Potential uses include comparative work involving these traits as focal or secondary variables, ecological research on the trait or trophic structure of communities, or conservation science concerned with the loss of function among species or in ecosystems in a changing world. We hope that this publication will spur the sharing, collaborative curation, and extension of data to the benefit of a more integrative, rigorous, and global biodiversity science.;,citation_author=Hamish Wilman;,citation_author=Jonathan Belmaker;,citation_author=Jennifer Simpson;,citation_author=Carolina Rosa;,citation_author=Marcelo M. Rivadeneira;,citation_author=Walter Jetz;,citation_publication_date=2014;,citation_cover_date=2014;,citation_year=2014;,citation_issue=7;,citation_doi=10.1890/13-1917.1;,citation_issn=1939-9170;,citation_volume=95;,citation_language=en-US;,citation_journal_title=Ecology;">
<meta name="citation_reference" content="citation_title=Using ecological networks to answer questions in global biogeography and ecology;,citation_abstract=Ecological networks have classically been studied at site and landscape scales, yet recent efforts have been made to collate these data into global repositories. This offers an opportunity to integrate and upscale knowledge about ecological interactions from local to global scales to gain enhanced insights from the mechanistic information provided by these data. By drawing on existing research investigating patterns in ecological interactions at continental to global scales, we show how data on ecological networks, collected at appropriate scales, can be used to generate an improved understanding of many aspects of ecology and biogeography—for example, species distribution modelling, restoration ecology and conservation. We argue that by understanding the patterns in the structure and function of ecological networks across scales, it is possible to enhance our understanding of the natural world.;,citation_author=Fredric M. Windsor;,citation_author=Johan Hoogen;,citation_author=Thomas W. Crowther;,citation_author=Darren M. Evans;,citation_publication_date=2023;,citation_cover_date=2023;,citation_year=2023;,citation_issue=1;,citation_doi=10.1111/jbi.14447;,citation_issn=1365-2699;,citation_volume=50;,citation_language=en-US;,citation_journal_title=Journal of Biogeography;">
<meta name="citation_reference" content="citation_title=Measurement of Interaction Strength in Nature;,citation_abstract=Understanding and predicting the dynamics of multispecies systems generally require estimates of interaction strength among species. Measuring interaction strength is difficult because of the large number of interactions in any natural system, long-term feedback, multiple pathways of effects between species pairs, and possible nonlinearities in interaction-strength functions. Presently, the few studies that extensively estimate interaction strength suggest that distributions of interaction strength tend to be skewed toward few strong and many weak interactions. Modeling studies indicate that such skewed patterns tend to promote system stability and arise during assembly of persistent communities. Methods for estimating interaction strength efficiently from traits of organisms, such as allometric relationships, show some promise. Methods for estimating community response to environmental perturbations without an estimate of interaction strength may also be of use. Spatial and temporal scale may affect patterns of interaction strength, but these effects require further investigation and new multispecies modeling frameworks. Future progress will be aided by development of long-term multispecies time series of natural communities, by experimental tests of different methods for estimating interaction strength, and by increased understanding of nonlinear functional forms.;,citation_author=J. Timothy Wootton;,citation_author=Mark Emmerson;,citation_publication_date=2005;,citation_cover_date=2005;,citation_year=2005;,citation_issue=1;,citation_doi=10.1146/annurev.ecolsys.36.091704.175535;,citation_volume=36;,citation_journal_title=Annual Review of Ecology, Evolution, and Systematics;">
<meta name="citation_reference" content="citation_title=Towards a modular theory of trophic interactions;,citation_abstract=Species traits and environmental conditions determine the occurrence and strength of trophic interactions. If we understand the relationship between these factors and trophic interactions, we can make more accurate predictions and build better trophic-interaction models. We can compare traits and conditions by considering their effect on different parts (steps) of a trophic interaction, such as the steps search and pursuit. By linking traits to relevant steps, we can use these relationships to build trophic-interaction models. Currently, this is done ad hoc, defining steps based on the species and traits of interest. This makes it difficult to compare across traits and species and gain an overarching understanding of how traits and the environment drive trophic interactions. We present a comprehensive approach for the explicit choice of interaction steps and species traits or environmental conditions, which is readily integrated into existing models. The core of this framework is that it is modular; we present eight steps that occur in all trophic interactions and use them to build a modular, general dynamic model. When applying the framework, one explicitly selects only the most relevant steps and uses those to build a specific model. To build our modular framework, we revisit and expand the functional and numerical response functions, dividing the trophic interaction into eight steps: (1) search, (2) prey detection, (3) attack decision, (4) pursuit, (5) subjugation, (6) ingestion, (7) digestion and (8) nutrient allocation. Together these steps form a general dynamical model where trophic interactions can be explicitly parameterized for multiple traits and environmental factors. We then concretize this approach by outlining how a specific community can be modelled by selecting key modules (steps) and parameterizing them for relevant factors. This we exemplify for a community of terrestrial arthropods using empirical data on body size and temperature responses. With species interactions at the core of community dynamics, our modular approach allows for quantification and comparisons of the importance of different steps, traits, and abiotic factors across ecosystems and trophic-interaction types, and provides a powerful tool for trait-based prediction of food-web structure and dynamics. A free Plain Language Summary can be found within the Supporting Information of this article.;,citation_author=Kate L. Wootton;,citation_author=Alva Curtsdotter;,citation_author=Tomas Roslin;,citation_author=Riccardo Bommarco;,citation_author=Tomas Jonsson;,citation_publication_date=2023;,citation_cover_date=2023;,citation_year=2023;,citation_issue=1;,citation_doi=10.1111/1365-2435.13954;,citation_issn=1365-2435;,citation_volume=37;,citation_language=en-US;,citation_journal_title=Functional Ecology;">
<meta name="citation_reference" content="citation_title=Completeness of Community Structure in Networks;,citation_abstract=By defining a new measure to community structure, exclusive modularity, and based on cavity method of statistical physics, we develop a mathematically principled method to determine the completeness of community structure, which represents whether a partition that is either annotated by experts or given by a community-detection algorithm, carries complete information about community structure in the network. Our results demonstrate that the expert partition is surprisingly incomplete in some networks such as the famous political blogs network, indicating that the relation between meta-data and community structure in real-world networks needs to be re-examined. As a byproduct we find that the exclusive modularity, which introduces a null model based on the degree-corrected stochastic block model, is of independent interest. We discuss its applications as principled ways of detecting hidden structures, finding hierarchical structures without removing edges, and obtaining low-dimensional embedding of networks.;,citation_author=Jia-Rong Xie;,citation_author=Pan Zhang;,citation_author=Hai-Feng Zhang;,citation_author=Bing-Hong Wang;,citation_publication_date=2017-07;,citation_cover_date=2017-07;,citation_year=2017;,citation_issue=1;,citation_doi=10.1038/s41598-017-05585-6;,citation_issn=2045-2322;,citation_volume=7;,citation_language=en-US;,citation_journal_title=Scientific Reports;,citation_publisher=Nature Publishing Group;">
<meta name="citation_reference" content="citation_title=Collapse of an ecological network in Ancient Egypt;,citation_author=Justin D. Yeakel;,citation_author=Mathias M. Pires;,citation_author=Lars Rudolf;,citation_author=Nathaniel J. Dominy;,citation_author=Paul L. Koch;,citation_author=Paulo R. Guimarães;,citation_author=Thilo Gross;,citation_publication_date=2014-10;,citation_cover_date=2014-10;,citation_year=2014;,citation_issue=40;,citation_doi=10.1073/pnas.1408471111;,citation_volume=111;,citation_journal_title=PNAS;">
<meta name="citation_reference" content="citation_title=Body Size and Consumer-Resource Dynamics;,citation_abstract=&amp;amp;amp;quot;Plausible&amp;quot; consumer-resource models, based on energetic reasoning and allometric empiricism, are formulated and their dynamics investigated. Most of the parameters in these models are determined by the body sizes and metabolic categories (endotherm, vertebrate ectotherm, invertebrate ectotherm, or plant) of the populations in question. The remaining parameters have clear biological meanings. The intent of these models is to provide maximum realism from minimum data when needed because of constraints of time or of resources for empirical research. The models also demonstrate the influence of physiological power and body size on consumer-resource dynamics. For a given metabolic type, we define ecological scope as the ratio of maximal ingestion rate to respiration rate. Invertebrate ectotherms have greater ecological scope than do vertebrate ectotherms, which have greater scope than do endotherms. Greater ecological scope implies ability to subsist on scarcer resources and a tendency to more stable dynamics. Changes in dynamics associated with increased resource carrying capacity K (the &quot;paradox of enrichment&quot;) are investigated. Robust limit cycles (sustained oscillations) require a Type III functional response, but even in this case the cycles quickly &quot;implode&quot; (with respect to changes in K) to extremely small densities if the resource-consumer body mass ratio is too high. Estimates are given for resource-consumer body mass ratios that permit robust limit cycles, and some pertinent data are discussed. An analytic expression for the periods of limit cycles is derived. In terms of body mass, the periods scale as (consumer body mass)1/8 (resource body mass)1/8.;,citation_author=P. Yodzis;,citation_author=S. Innes;,citation_publication_date=1992-06;,citation_cover_date=1992-06;,citation_year=1992;,citation_issue=6;,citation_doi=10.1086/285380;,citation_issn=0003-0147;,citation_volume=139;,citation_journal_title=The American Naturalist;,citation_publisher=The University of Chicago Press;">
<meta name="citation_reference" content="citation_title=The Compartmentation of Real and Assembled Ecosystems;,citation_abstract=The dominant cliques of graph theory provide an unambiguous compartmentation for ecosystems. It is suggested here that it might make ecological sense to regard dominant cliques as something like &amp;amp;amp;quot;trophic guilds.&amp;quot; The dominant clique structure of 40 real food webs collected from the published literature by Briand (1982) is investigated. It is found that food webs from environments classifiedd by Briand as &quot;fluctuating&quot; differ significantly, both in number and in average size of dominant cliques, from food webs from environments classified by Briand as &quot;constant.&quot; In addition, food web intervality (Cohen 1978) is found to be associated with poverty in dominant cliques. Adding constraints on dominant clique number to &quot;assembled&quot; food web models (Yodzis 1981) yields mathematical food web universes which predict the observed frequency of interval food webs.;,citation_author=P. Yodzis;,citation_publication_date=1982-11;,citation_cover_date=1982-11;,citation_year=1982;,citation_issue=5;,citation_doi=10.1086/284013;,citation_issn=0003-0147;,citation_volume=120;,citation_journal_title=The American Naturalist;,citation_publisher=The University of Chicago Press;">
</head>

<body>

<header id="title-block-header" class="quarto-title-block default toc-left page-columns page-full">
  <div class="quarto-title-banner page-columns page-full">
    <div class="quarto-title column-body">
      <h1 class="title">Unveiling the Complexity of Food Webs: A Comprehensive Overview of Definitions, Scales, and Mechanisms</h1>
          </div>

    
    <div class="quarto-title-meta-container">
      <div class="quarto-title-meta-column-start">
            <div class="quarto-title-meta-author">
          <div class="quarto-title-meta-heading">Authors</div>
          <div class="quarto-title-meta-heading">Affiliations</div>
          
                <div class="quarto-title-meta-contents">
            <p class="author">Tanya Strydom <a href="mailto:t.strydom@sheffield.ac.uk" class="quarto-title-author-email"><i class="bi bi-envelope"></i></a> <a href="https://orcid.org/0000-0001-6067-1349" class="quarto-title-author-orcid"> <img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAA2ZpVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADw/eHBhY2tldCBiZWdpbj0i77u/IiBpZD0iVzVNME1wQ2VoaUh6cmVTek5UY3prYzlkIj8+IDx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8iIHg6eG1wdGs9IkFkb2JlIFhNUCBDb3JlIDUuMC1jMDYwIDYxLjEzNDc3NywgMjAxMC8wMi8xMi0xNzozMjowMCAgICAgICAgIj4gPHJkZjpSREYgeG1sbnM6cmRmPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5LzAyLzIyLXJkZi1zeW50YXgtbnMjIj4gPHJkZjpEZXNjcmlwdGlvbiByZGY6YWJvdXQ9IiIgeG1sbnM6eG1wTU09Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC9tbS8iIHhtbG5zOnN0UmVmPSJodHRwOi8vbnMuYWRvYmUuY29tL3hhcC8xLjAvc1R5cGUvUmVzb3VyY2VSZWYjIiB4bWxuczp4bXA9Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC8iIHhtcE1NOk9yaWdpbmFsRG9jdW1lbnRJRD0ieG1wLmRpZDo1N0NEMjA4MDI1MjA2ODExOTk0QzkzNTEzRjZEQTg1NyIgeG1wTU06RG9jdW1lbnRJRD0ieG1wLmRpZDozM0NDOEJGNEZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wTU06SW5zdGFuY2VJRD0ieG1wLmlpZDozM0NDOEJGM0ZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wOkNyZWF0b3JUb29sPSJBZG9iZSBQaG90b3Nob3AgQ1M1IE1hY2ludG9zaCI+IDx4bXBNTTpEZXJpdmVkRnJvbSBzdFJlZjppbnN0YW5jZUlEPSJ4bXAuaWlkOkZDN0YxMTc0MDcyMDY4MTE5NUZFRDc5MUM2MUUwNEREIiBzdFJlZjpkb2N1bWVudElEPSJ4bXAuZGlkOjU3Q0QyMDgwMjUyMDY4MTE5OTRDOTM1MTNGNkRBODU3Ii8+IDwvcmRmOkRlc2NyaXB0aW9uPiA8L3JkZjpSREY+IDwveDp4bXBtZXRhPiA8P3hwYWNrZXQgZW5kPSJyIj8+84NovQAAAR1JREFUeNpiZEADy85ZJgCpeCB2QJM6AMQLo4yOL0AWZETSqACk1gOxAQN+cAGIA4EGPQBxmJA0nwdpjjQ8xqArmczw5tMHXAaALDgP1QMxAGqzAAPxQACqh4ER6uf5MBlkm0X4EGayMfMw/Pr7Bd2gRBZogMFBrv01hisv5jLsv9nLAPIOMnjy8RDDyYctyAbFM2EJbRQw+aAWw/LzVgx7b+cwCHKqMhjJFCBLOzAR6+lXX84xnHjYyqAo5IUizkRCwIENQQckGSDGY4TVgAPEaraQr2a4/24bSuoExcJCfAEJihXkWDj3ZAKy9EJGaEo8T0QSxkjSwORsCAuDQCD+QILmD1A9kECEZgxDaEZhICIzGcIyEyOl2RkgwAAhkmC+eAm0TAAAAABJRU5ErkJggg=="></a></p>
          </div>
                <div class="quarto-title-meta-contents">
                    <p class="affiliation">
                        School of Biosciences, University of Sheffield, Sheffield, UK
                      </p>
                  </div>
                      <div class="quarto-title-meta-contents">
            <p class="author">Jennifer A. Dunne <a href="https://orcid.org/0000-0002-3789-9561" class="quarto-title-author-orcid"> <img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAA2ZpVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADw/eHBhY2tldCBiZWdpbj0i77u/IiBpZD0iVzVNME1wQ2VoaUh6cmVTek5UY3prYzlkIj8+IDx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8iIHg6eG1wdGs9IkFkb2JlIFhNUCBDb3JlIDUuMC1jMDYwIDYxLjEzNDc3NywgMjAxMC8wMi8xMi0xNzozMjowMCAgICAgICAgIj4gPHJkZjpSREYgeG1sbnM6cmRmPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5LzAyLzIyLXJkZi1zeW50YXgtbnMjIj4gPHJkZjpEZXNjcmlwdGlvbiByZGY6YWJvdXQ9IiIgeG1sbnM6eG1wTU09Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC9tbS8iIHhtbG5zOnN0UmVmPSJodHRwOi8vbnMuYWRvYmUuY29tL3hhcC8xLjAvc1R5cGUvUmVzb3VyY2VSZWYjIiB4bWxuczp4bXA9Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC8iIHhtcE1NOk9yaWdpbmFsRG9jdW1lbnRJRD0ieG1wLmRpZDo1N0NEMjA4MDI1MjA2ODExOTk0QzkzNTEzRjZEQTg1NyIgeG1wTU06RG9jdW1lbnRJRD0ieG1wLmRpZDozM0NDOEJGNEZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wTU06SW5zdGFuY2VJRD0ieG1wLmlpZDozM0NDOEJGM0ZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wOkNyZWF0b3JUb29sPSJBZG9iZSBQaG90b3Nob3AgQ1M1IE1hY2ludG9zaCI+IDx4bXBNTTpEZXJpdmVkRnJvbSBzdFJlZjppbnN0YW5jZUlEPSJ4bXAuaWlkOkZDN0YxMTc0MDcyMDY4MTE5NUZFRDc5MUM2MUUwNEREIiBzdFJlZjpkb2N1bWVudElEPSJ4bXAuZGlkOjU3Q0QyMDgwMjUyMDY4MTE5OTRDOTM1MTNGNkRBODU3Ii8+IDwvcmRmOkRlc2NyaXB0aW9uPiA8L3JkZjpSREY+IDwveDp4bXBtZXRhPiA8P3hwYWNrZXQgZW5kPSJyIj8+84NovQAAAR1JREFUeNpiZEADy85ZJgCpeCB2QJM6AMQLo4yOL0AWZETSqACk1gOxAQN+cAGIA4EGPQBxmJA0nwdpjjQ8xqArmczw5tMHXAaALDgP1QMxAGqzAAPxQACqh4ER6uf5MBlkm0X4EGayMfMw/Pr7Bd2gRBZogMFBrv01hisv5jLsv9nLAPIOMnjy8RDDyYctyAbFM2EJbRQw+aAWw/LzVgx7b+cwCHKqMhjJFCBLOzAR6+lXX84xnHjYyqAo5IUizkRCwIENQQckGSDGY4TVgAPEaraQr2a4/24bSuoExcJCfAEJihXkWDj3ZAKy9EJGaEo8T0QSxkjSwORsCAuDQCD+QILmD1A9kECEZgxDaEZhICIzGcIyEyOl2RkgwAAhkmC+eAm0TAAAAABJRU5ErkJggg=="></a></p>
          </div>
                <div class="quarto-title-meta-contents">
                    <p class="affiliation">
                        Santa Fe Institute, 1399 Hyde Park Road, Santa Fe, NM 87501, USA
                      </p>
                  </div>
                      <div class="quarto-title-meta-contents">
            <p class="author">Timothée Poisot <a href="https://orcid.org/0000-0002-0735-5184" class="quarto-title-author-orcid"> <img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAA2ZpVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADw/eHBhY2tldCBiZWdpbj0i77u/IiBpZD0iVzVNME1wQ2VoaUh6cmVTek5UY3prYzlkIj8+IDx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8iIHg6eG1wdGs9IkFkb2JlIFhNUCBDb3JlIDUuMC1jMDYwIDYxLjEzNDc3NywgMjAxMC8wMi8xMi0xNzozMjowMCAgICAgICAgIj4gPHJkZjpSREYgeG1sbnM6cmRmPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5LzAyLzIyLXJkZi1zeW50YXgtbnMjIj4gPHJkZjpEZXNjcmlwdGlvbiByZGY6YWJvdXQ9IiIgeG1sbnM6eG1wTU09Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC9tbS8iIHhtbG5zOnN0UmVmPSJodHRwOi8vbnMuYWRvYmUuY29tL3hhcC8xLjAvc1R5cGUvUmVzb3VyY2VSZWYjIiB4bWxuczp4bXA9Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC8iIHhtcE1NOk9yaWdpbmFsRG9jdW1lbnRJRD0ieG1wLmRpZDo1N0NEMjA4MDI1MjA2ODExOTk0QzkzNTEzRjZEQTg1NyIgeG1wTU06RG9jdW1lbnRJRD0ieG1wLmRpZDozM0NDOEJGNEZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wTU06SW5zdGFuY2VJRD0ieG1wLmlpZDozM0NDOEJGM0ZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wOkNyZWF0b3JUb29sPSJBZG9iZSBQaG90b3Nob3AgQ1M1IE1hY2ludG9zaCI+IDx4bXBNTTpEZXJpdmVkRnJvbSBzdFJlZjppbnN0YW5jZUlEPSJ4bXAuaWlkOkZDN0YxMTc0MDcyMDY4MTE5NUZFRDc5MUM2MUUwNEREIiBzdFJlZjpkb2N1bWVudElEPSJ4bXAuZGlkOjU3Q0QyMDgwMjUyMDY4MTE5OTRDOTM1MTNGNkRBODU3Ii8+IDwvcmRmOkRlc2NyaXB0aW9uPiA8L3JkZjpSREY+IDwveDp4bXBtZXRhPiA8P3hwYWNrZXQgZW5kPSJyIj8+84NovQAAAR1JREFUeNpiZEADy85ZJgCpeCB2QJM6AMQLo4yOL0AWZETSqACk1gOxAQN+cAGIA4EGPQBxmJA0nwdpjjQ8xqArmczw5tMHXAaALDgP1QMxAGqzAAPxQACqh4ER6uf5MBlkm0X4EGayMfMw/Pr7Bd2gRBZogMFBrv01hisv5jLsv9nLAPIOMnjy8RDDyYctyAbFM2EJbRQw+aAWw/LzVgx7b+cwCHKqMhjJFCBLOzAR6+lXX84xnHjYyqAo5IUizkRCwIENQQckGSDGY4TVgAPEaraQr2a4/24bSuoExcJCfAEJihXkWDj3ZAKy9EJGaEo8T0QSxkjSwORsCAuDQCD+QILmD1A9kECEZgxDaEZhICIzGcIyEyOl2RkgwAAhkmC+eAm0TAAAAABJRU5ErkJggg=="></a></p>
          </div>
                <div class="quarto-title-meta-contents">
                    <p class="affiliation">
                        Université de Montreal
                      </p>
                    <p class="affiliation">
                        Québec Centre for Biodiversity Sciences
                      </p>
                  </div>
                      <div class="quarto-title-meta-contents">
            <p class="author">Andrew P. Beckerman <a href="https://orcid.org/0000-0002-7859-8394" class="quarto-title-author-orcid"> <img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAA2ZpVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADw/eHBhY2tldCBiZWdpbj0i77u/IiBpZD0iVzVNME1wQ2VoaUh6cmVTek5UY3prYzlkIj8+IDx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8iIHg6eG1wdGs9IkFkb2JlIFhNUCBDb3JlIDUuMC1jMDYwIDYxLjEzNDc3NywgMjAxMC8wMi8xMi0xNzozMjowMCAgICAgICAgIj4gPHJkZjpSREYgeG1sbnM6cmRmPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5LzAyLzIyLXJkZi1zeW50YXgtbnMjIj4gPHJkZjpEZXNjcmlwdGlvbiByZGY6YWJvdXQ9IiIgeG1sbnM6eG1wTU09Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC9tbS8iIHhtbG5zOnN0UmVmPSJodHRwOi8vbnMuYWRvYmUuY29tL3hhcC8xLjAvc1R5cGUvUmVzb3VyY2VSZWYjIiB4bWxuczp4bXA9Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC8iIHhtcE1NOk9yaWdpbmFsRG9jdW1lbnRJRD0ieG1wLmRpZDo1N0NEMjA4MDI1MjA2ODExOTk0QzkzNTEzRjZEQTg1NyIgeG1wTU06RG9jdW1lbnRJRD0ieG1wLmRpZDozM0NDOEJGNEZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wTU06SW5zdGFuY2VJRD0ieG1wLmlpZDozM0NDOEJGM0ZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wOkNyZWF0b3JUb29sPSJBZG9iZSBQaG90b3Nob3AgQ1M1IE1hY2ludG9zaCI+IDx4bXBNTTpEZXJpdmVkRnJvbSBzdFJlZjppbnN0YW5jZUlEPSJ4bXAuaWlkOkZDN0YxMTc0MDcyMDY4MTE5NUZFRDc5MUM2MUUwNEREIiBzdFJlZjpkb2N1bWVudElEPSJ4bXAuZGlkOjU3Q0QyMDgwMjUyMDY4MTE5OTRDOTM1MTNGNkRBODU3Ii8+IDwvcmRmOkRlc2NyaXB0aW9uPiA8L3JkZjpSREY+IDwveDp4bXBtZXRhPiA8P3hwYWNrZXQgZW5kPSJyIj8+84NovQAAAR1JREFUeNpiZEADy85ZJgCpeCB2QJM6AMQLo4yOL0AWZETSqACk1gOxAQN+cAGIA4EGPQBxmJA0nwdpjjQ8xqArmczw5tMHXAaALDgP1QMxAGqzAAPxQACqh4ER6uf5MBlkm0X4EGayMfMw/Pr7Bd2gRBZogMFBrv01hisv5jLsv9nLAPIOMnjy8RDDyYctyAbFM2EJbRQw+aAWw/LzVgx7b+cwCHKqMhjJFCBLOzAR6+lXX84xnHjYyqAo5IUizkRCwIENQQckGSDGY4TVgAPEaraQr2a4/24bSuoExcJCfAEJihXkWDj3ZAKy9EJGaEo8T0QSxkjSwORsCAuDQCD+QILmD1A9kECEZgxDaEZhICIzGcIyEyOl2RkgwAAhkmC+eAm0TAAAAABJRU5ErkJggg=="></a></p>
          </div>
                <div class="quarto-title-meta-contents">
                    <p class="affiliation">
                        School of Biosciences, University of Sheffield, Sheffield, UK
                      </p>
                  </div>
                    </div>
        
        <div class="quarto-title-meta">

                      
                <div>
            <div class="quarto-title-meta-heading">Published</div>
            <div class="quarto-title-meta-contents">
              <p class="date">September 27, 2024</p>
            </div>
          </div>
          
                
              </div>
      </div>
      <div class="quarto-title-meta-column-end quarto-other-formats-target">
      <div class="quarto-alternate-formats"><div class="quarto-title-meta-heading">Other Formats</div><div class="quarto-title-meta-contents"><p><a href="index.docx"><i class="bi bi-file-word"></i>MS Word</a></p></div><div class="quarto-title-meta-contents"><p><a href="index.pdf"><i class="bi bi-file-pdf"></i>PDF (preprint)</a></p></div></div></div>
    </div>

    <div>
      <div class="abstract">
        <div class="block-title">Abstract</div>
        <p>Food webs are a useful abstraction and representation of the feeding links between species in a community and are used to infer many ecosystem level processes. However, the different theories, mechanisms, and criteria that underpin how a food web is defined and, ultimately, constructed means that not all food webs are representing the same ecological process. Here we present a synthesis of the different assumptions, scales and mechanisms that are used to define different ecological networks ranging from metawebs (an inventory of all potential interactions) to fully realised networks (interactions that occur within a given community over a certain timescale). Illuminating the assumptions, scales, and mechanisms of network inference allows a formal categorisation of how to use networks to answer key ecological and conservation questions and defines guidelines to prevent unintentional misuse or misinterpretation.</p>
      </div>
    </div>

    <div>
      <div class="keywords">
        <div class="block-title">Keywords</div>
        <p>food web, network construction, scientific ignorance</p>
      </div>
    </div>

    <div class="quarto-other-links-text-target">
    <div class="quarto-code-links"><div class="quarto-title-meta-heading">Code Links</div><div class="quarto-title-meta-contents"><span><a href="https://github.com/BecksLab/topology_generators"><i class="bi bi-github"></i>GitHub</a></span></div></div></div>  </div>
</header><div id="quarto-content" class="page-columns page-rows-contents page-layout-article toc-left">
<div id="quarto-sidebar-toc-left" class="sidebar toc-left">
  <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#sec-anatomy" id="toc-sec-anatomy" class="nav-link active" data-scroll-target="#sec-anatomy"><span class="header-section-number">1</span> Setting the Scene: The Not So Basics of Nodes and Edges</a>
  <ul class="collapse">
  <li><a href="#how-do-we-define-a-node" id="toc-how-do-we-define-a-node" class="nav-link" data-scroll-target="#how-do-we-define-a-node"><span class="header-section-number">1.0.1</span> How do we define a node?</a></li>
  <li><a href="#what-is-meant-by-an-edge" id="toc-what-is-meant-by-an-edge" class="nav-link" data-scroll-target="#what-is-meant-by-an-edge"><span class="header-section-number">1.0.2</span> What is meant by an edge?</a></li>
  <li><a href="#network-representations" id="toc-network-representations" class="nav-link" data-scroll-target="#network-representations"><span class="header-section-number">1.0.3</span> Network representations</a></li>
  </ul></li>
  <li><a href="#sec-process" id="toc-sec-process" class="nav-link" data-scroll-target="#sec-process"><span class="header-section-number">2</span> From Nodes and Edges to Scales, Context, and Processes</a>
  <ul class="collapse">
  <li><a href="#the-processes-that-determine-species-interactions" id="toc-the-processes-that-determine-species-interactions" class="nav-link" data-scroll-target="#the-processes-that-determine-species-interactions"><span class="header-section-number">2.1</span> The processes that determine species interactions</a></li>
  <li><a href="#contextualising-the-processes-that-determine-species-interactions" id="toc-contextualising-the-processes-that-determine-species-interactions" class="nav-link" data-scroll-target="#contextualising-the-processes-that-determine-species-interactions"><span class="header-section-number">2.2</span> Contextualising the processes that determine species interactions</a></li>
  </ul></li>
  <li><a href="#network-construction-is-nuanced" id="toc-network-construction-is-nuanced" class="nav-link" data-scroll-target="#network-construction-is-nuanced"><span class="header-section-number">3</span> Network construction is nuanced</a>
  <ul class="collapse">
  <li><a href="#how-do-we-predict-food-webs" id="toc-how-do-we-predict-food-webs" class="nav-link" data-scroll-target="#how-do-we-predict-food-webs"><span class="header-section-number">3.1</span> How do we predict food webs?</a>
  <ul class="collapse">
  <li><a href="#models-that-predict-metawebs-feasible-interactions" id="toc-models-that-predict-metawebs-feasible-interactions" class="nav-link" data-scroll-target="#models-that-predict-metawebs-feasible-interactions"><span class="header-section-number">3.1.1</span> Models that predict metawebs (feasible interactions)</a></li>
  <li><a href="#models-that-predict-realised-networks-realised-interactions" id="toc-models-that-predict-realised-networks-realised-interactions" class="nav-link" data-scroll-target="#models-that-predict-realised-networks-realised-interactions"><span class="header-section-number">3.1.2</span> Models that predict realised networks (realised interactions)</a></li>
  <li><a href="#models-that-predict-structure-interaction-agnostic" id="toc-models-that-predict-structure-interaction-agnostic" class="nav-link" data-scroll-target="#models-that-predict-structure-interaction-agnostic"><span class="header-section-number">3.1.3</span> Models that predict structure (interaction agnostic)</a></li>
  </ul></li>
  </ul></li>
  <li><a href="#making-progress-with-networks" id="toc-making-progress-with-networks" class="nav-link" data-scroll-target="#making-progress-with-networks"><span class="header-section-number">4</span> Making Progress with Networks</a>
  <ul class="collapse">
  <li><a href="#further-development-of-models-and-tools" id="toc-further-development-of-models-and-tools" class="nav-link" data-scroll-target="#further-development-of-models-and-tools"><span class="header-section-number">4.1</span> Further development of models and tools</a>
  <ul class="collapse">
  <li><a href="#at-what-scale-should-we-be-predicting-and-using-networks" id="toc-at-what-scale-should-we-be-predicting-and-using-networks" class="nav-link" data-scroll-target="#at-what-scale-should-we-be-predicting-and-using-networks"><span class="header-section-number">4.1.1</span> At what scale should we be predicting and using networks?</a></li>
  <li><a href="#feasible-realised-or-sustainable" id="toc-feasible-realised-or-sustainable" class="nav-link" data-scroll-target="#feasible-realised-or-sustainable"><span class="header-section-number">4.1.2</span> Feasible, realised, or sustainable?</a></li>
  </ul></li>
  <li><a href="#how-should-we-use-different-networks" id="toc-how-should-we-use-different-networks" class="nav-link" data-scroll-target="#how-should-we-use-different-networks"><span class="header-section-number">4.2</span> How should we use different networks?</a></li>
  </ul></li>
  <li><a href="#concluding-remarks" id="toc-concluding-remarks" class="nav-link" data-scroll-target="#concluding-remarks"><span class="header-section-number">5</span> Concluding remarks</a></li>
  <li><a href="#references" id="toc-references" class="nav-link" data-scroll-target="#references">References</a></li>
  </ul>
<div class="quarto-alternate-notebooks"><h2>Notebooks</h2><ul><li><a href="notebooks/model_descriptions-preview.html"><i class="bi bi-journal-code"></i>SuppMat 1: Model Families Extended Description</a></li><li><a href="notebooks/model_qualitative-preview.html"><i class="bi bi-journal-code"></i>SuppMat 2: Looking at the 'traits' of the different model families</a></li><li><a href="notebooks/model_quantitative-preview.html"><i class="bi bi-journal-code"></i>SuppMat 3: Benchmarking the different model families</a></li></ul></div></nav>
</div>
<div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
</div>
<main class="content quarto-banner-title-block" id="quarto-document-content">



  


<p>At the heart of modern biodiversity science are a set of concepts and theories about biodiversity, stability and function. These relate to the abundance, distribution and services that biodiversity provides, and how biodiversity – as an interconnected set of species – responds to multiple stressors. The interaction between species (or individuals) is one of the fundamental building blocks of ecological communities provide a powerful abstraction that can help quantify, conceptualise, and understand biodiversity dynamics, and ultimately, one hopes, make prediction, mitigate change and manage services [ref]. Such network representations of biodiversity (including within species diversity) are increasingly argued to be an asset to predictive ecology, climate change mitigation and resource management. Here, it is argued that characterising biodiversity in a network will allow deeper capacity to understand and predict the abundance, distribution, dynamics and services provided by multiple species facing multiple stressors.</p>
<p>However, the way that a network is constructed (encoded) defines an epistemology of the network concept which, we argue, can influence the resulting observations and conclusions about pattern and mechanisms that are made <span class="citation" data-cites="proulxNetworkThinkingEcology2005 brimacombeShortcomingsReusingSpecies2023">(<a href="#ref-proulxNetworkThinkingEcology2005" role="doc-biblioref">Proulx, Promislow, and Phillips 2005</a>; <a href="#ref-brimacombeShortcomingsReusingSpecies2023" role="doc-biblioref">Brimacombe et al. 2023</a>)</span>. This process of constructing networks has two major pillars: the data and theory, the latter representing an expression of mechanism and process giving rise to patterns that emerge from collating interactions among species. Each of these pillars carries with it a set of practical, semantic and conceptual constraints that not only influence progress in making network ecology more valuable and potentially predictive, but help define the spatial, temporal and evolutionary scale of assumptions we make and predictions we might generate from the networks.</p>
<p>With respect to data, it is extremely challenging to actually record species interactions in the field <span class="citation" data-cites="jordanoChasingEcologicalInteractions2016 jordanoSamplingNetworksEcological2016">(<a href="#ref-jordanoChasingEcologicalInteractions2016" role="doc-biblioref">Jordano 2016a</a>, <a href="#ref-jordanoSamplingNetworksEcological2016" role="doc-biblioref">2016b</a>)</span>. Despite notable herculean efforts (<strong>Woodward? Benguela?</strong> <span class="citation" data-cites="maioranoTETRAEUSpecieslevelTrophic2020">Maiorano et al. (<a href="#ref-maioranoTETRAEUSpecieslevelTrophic2020" role="doc-biblioref">2020</a>)</span>), actual coverage of ‘real world’ interaction data remains sparse <span class="citation" data-cites="poisotGlobalKnowledgeGaps2021">(<a href="#ref-poisotGlobalKnowledgeGaps2021" role="doc-biblioref">Poisot et al. 2021</a>)</span>. Against this practical challenge, there is additionally high variance in the terminology we use to define networks. Finally, the mathematical and statistical tools we use to construct, conceptualise, analyse and predict with these networks are also highly variable.</p>
<ol type="1">
<li>what are the underlying assumptions about nodes, edges, scale and process that are made when we attempt to delimit and describe a food webs;</li>
<li>are there families of commonly used tools that map onto assumptions about scales and processes;</li>
</ol>
<p>The provision of this detail ultimately leads to a set of insights and conclusions about whether, when and under what conditions network representations of biodiversity can contribute to the advancement of ecological theory and generate value in predictive ecology. Specifically, we finish this perspective with an overview of fundamental questions in ecology that we think can benefit from network thinking and a proposal that such thinking can accelerate our capacity to predict the impact of multiple stressors on biodiverse communities.</p>
<section id="sec-anatomy" class="level1" data-number="1">
<h1 data-number="1"><span class="header-section-number">1</span> Setting the Scene: The Not So Basics of Nodes and Edges</h1>
<p>Defining a food web seems simple; it is the representation of the interactions (edges) between species (nodes), however the definition of ‘edges’ and ‘nodes’, as well as the scale at which they are aggregated can take many forms <span class="citation" data-cites="poisotDescribeUnderstandPredict2016">(<a href="#ref-poisotDescribeUnderstandPredict2016" role="doc-biblioref">Poisot, Stouffer, and Kéfi 2016</a>)</span>, which ultimately encodes a series of assumptions and criteria within a network. An awareness of variance in the way a food web can be defined is critical as a network (or its adjacency matrix) is both the ‘object’ from which inferences are made (<em>e.g.,</em> the interactions between species, or how the structure influences ecosystem level processes) as well as the ‘product’ of either the data collection <span class="citation" data-cites="brimacombeShortcomingsReusingSpecies2023">(<a href="#ref-brimacombeShortcomingsReusingSpecies2023" role="doc-biblioref">Brimacombe et al. 2023</a>)</span> or prediction process <span class="citation" data-cites="banvilleDecipheringProbabilisticSpecies2024">(<a href="#ref-banvilleDecipheringProbabilisticSpecies2024" role="doc-biblioref">Banville et al. 2024</a>)</span>. One thus needs to be aware of both the criteria that is used to define nodes and edges, and what processes or mechanisms the aggregation of the two represents, as this will ultimately determine and delimit the way in which a network can and should be used.</p>
<section id="how-do-we-define-a-node" class="level3" data-number="1.0.1">
<h3 data-number="1.0.1" class="anchored" data-anchor-id="how-do-we-define-a-node"><span class="header-section-number">1.0.1</span> How do we define a node?</h3>
<p>Although this may seem an elementary question in the context of food webs — a node <em>should</em> represent a (taxonomic) species, the reality is that nodes can often represent an aggregation of different species - so called ‘trophic species’ <span class="citation" data-cites="yodzisCompartmentationRealAssembled1982 williamsSimpleRulesYield2000">(<a href="#ref-yodzisCompartmentationRealAssembled1982" role="doc-biblioref">Yodzis 1982</a>; <a href="#ref-williamsSimpleRulesYield2000" role="doc-biblioref">Williams and Martinez 2000</a>)</span> or segregation of species by life stages <span class="citation" data-cites="cleggImpactIntraspecificVariation2018">(<a href="#ref-cleggImpactIntraspecificVariation2018" role="doc-biblioref">Clegg, Ali, and Beckerman 2018</a>)</span>. Practical implications of how we are aggregating the nodes is that the resolution may not always be ‘pixel perfect’, which limits the ability to make (taxonomic) species specific inferences <em>e.g.,</em> does species <span class="math inline">\(a\)</span> eat species <span class="math inline">\(b\)</span>, however there is value in having nodes that represent an aggregation of species, as the distribution of the links between them are more meaningful in terms of understanding energy flow and distribution within the system.</p>
</section>
<section id="what-is-meant-by-an-edge" class="level3" data-number="1.0.2">
<h3 data-number="1.0.2" class="anchored" data-anchor-id="what-is-meant-by-an-edge"><span class="header-section-number">1.0.2</span> What is meant by an edge?</h3>
<p>At its core, links within food webs can be thought of as a representation of either feeding links between species - be that realised <span class="citation" data-cites="pringleUntanglingFoodWebs2020">(<a href="#ref-pringleUntanglingFoodWebs2020" role="doc-biblioref">Pringle 2020</a>)</span> or potential <span class="citation" data-cites="dunneNetworkStructureFood2006">(<a href="#ref-dunneNetworkStructureFood2006" role="doc-biblioref">Jennifer A. Dunne 2006</a>)</span>, or representative of fluxes within the community/system <em>e.g.,</em> energy transfer or material flow <span class="citation" data-cites="lindemanTrophicDynamicAspectEcology1942">(<a href="#ref-lindemanTrophicDynamicAspectEcology1942" role="doc-biblioref">Lindeman 1942</a>)</span>. How we specify links will influence the resulting structure of the network - and the inferences we will make thereof. For example taking a food web that consists of links representing all <em>potential</em> feeding links for a community (<em>i.e.,</em> a metaweb) will be meaningless if one is interested in understanding the flow of energy through the network as the links within a metaweb do not represent environmental/energetic constraints, making them poor representations of which interactions are <em>realised</em> in a specific location <span class="citation" data-cites="caronTraitmatchingModelsPredict2024">(<a href="#ref-caronTraitmatchingModelsPredict2024" role="doc-biblioref">Caron et al. 2024</a>)</span>. In addition to the various ways of defining the links between species pairs there are also a myriad of ways in which the links themselves can be quantified. Links between species are often treated as being present or absent (<em>i.e.,</em> binary) but it is also possible to use probabilities <span class="citation" data-cites="poisotStructureProbabilisticNetworks2016 banvilleDecipheringProbabilisticSpecies2024">(which quantifies how likely an interaction is to occur, <a href="#ref-poisotStructureProbabilisticNetworks2016" role="doc-biblioref">Poisot, Cirtwill, et al. 2016</a>; <a href="#ref-banvilleDecipheringProbabilisticSpecies2024" role="doc-biblioref">Banville et al. 2024</a>)</span> or continuous measurements <span class="citation" data-cites="berlowInteractionStrengthsFood2004">(which quantifies the strength of of an interaction, <a href="#ref-berlowInteractionStrengthsFood2004" role="doc-biblioref">Berlow et al. 2004</a>)</span>.</p>
</section>
<section id="network-representations" class="level3" data-number="1.0.3">
<h3 data-number="1.0.3" class="anchored" data-anchor-id="network-representations"><span class="header-section-number">1.0.3</span> Network representations</h3>
<p>Broadly, networks can be thought of to fall into two different ‘types’; namely metawebs; traditionally defined as all of the <em>potential</em> interactions for a specific species pool <span class="citation" data-cites="dunneNetworkStructureFood2006">(<a href="#ref-dunneNetworkStructureFood2006" role="doc-biblioref">Jennifer A. Dunne 2006</a>)</span>, and realised networks; which is the subset of interactions in a metaweb that are <em>realised</em> ‘on the ground’. The fundamental difference between these two different types of networks is that a metaweb provides insight as to the viability of an interaction between two species occurring and is a means to identify links that are not ecologically plausible, <em>i.e.,</em> forbidden links <span class="citation" data-cites="jordanoSamplingNetworksEcological2016">(<a href="#ref-jordanoSamplingNetworksEcological2016" role="doc-biblioref">Jordano 2016b</a>)</span>, or an idea of the <em>complete</em> diet of a species <span class="citation" data-cites="strydomGraphEmbeddingTransfer2023">(<a href="#ref-strydomGraphEmbeddingTransfer2023" role="doc-biblioref">Strydom et al. 2023</a>)</span>. Although metawebs are typically ‘constrained’ to a collection of species that also co-occur, there is no reason that a metaweb cannot include species that do not co-occur (although this would require some degree of prediction/assumptions to identify those possible interactions). In contrast realised networks are highly localised and contingent on both the co-occurrence of species as well as the influence of the environment, and population and community dynamics on predator choice. In the context of definitions and semantics the links that are represented by a metaweb and a realised network are different; links that are absent in a metaweb can be treated as being truly absent, however links that are absent in a realised network cannot be considered to be truly absent but are rather as absent due to the broader environmental/community context. Importantly, a realised network is <em>not</em> simply the downscaling of a metaweb to a smaller scale (<em>e.g.,</em> moving from the country to the 1x1 km<sup>2</sup> scale based on fine-scale species co-occurrence) but represents a shift towards capturing the higher level processes that determine the <em>realisation</em> of an interaction. Thus, different network representations are determined and constrained by different sets of assumptions as to what the processes are that determine the presence/absence of an interaction between two species as well as the resulting network structure.</p>
</section>
</section>
<section id="sec-process" class="level1" data-number="2">
<h1 data-number="2"><span class="header-section-number">2</span> From Nodes and Edges to Scales, Context, and Processes</h1>
<p>The interplay between network representation and network definition is primarily governed by the process(es) that determine the interaction between species, however these processes are also scale and context dependent. Here we start by introducing the five core processes that determine either the feasibility or realisation of interactions, namely: evolutionary compatibility, co-occurrence, feasibility, abundance, predator choice, and non-trophic interactions; while simultaneously contextualising them within, and linking them to the different network representations <a href="#fig-process" class="quarto-xref">Figure&nbsp;1</a>; specifically if the processes captures an all-or-nothing (possibility) vs context dependent (likelihood) determination of interactions between species. Of course these processes do not function in a vacuum and do interact with/influence one another, but it is still beneficial to present them in a categorical manner as these different processes are often the underpinning logic in the development of prediction/network models, the criteria for data collection in the field, and the scale of organisation for which they are relevant (species, population, or community).</p>
<div id="fig-process" class="quarto-figure quarto-figure-center quarto-float anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-process-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="images/concept_v2.png" class="lightbox" data-gallery="quarto-lightbox-gallery-1" data-glightbox="description: .lightbox-desc-1"><img src="images/concept_v2.png" class="img-fluid figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-process-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;1: TODO.
</figcaption>
</figure>
</div>
<section id="the-processes-that-determine-species-interactions" class="level2" data-number="2.1">
<h2 data-number="2.1" class="anchored" data-anchor-id="the-processes-that-determine-species-interactions"><span class="header-section-number">2.1</span> The processes that determine species interactions</h2>
<p><strong>Evolutionary compatibility</strong></p>
<p>There is compelling evidence that the possibility of an interaction occurring between two species is the result of their shared (co)evolutionary history <span class="citation" data-cites="segarRoleEvolutionShaping2020 gomezEcologicalInteractionsAre2010 dallarivaExploringEvolutionarySignature2016">(<a href="#ref-segarRoleEvolutionShaping2020" role="doc-biblioref">Segar et al. 2020</a>; <a href="#ref-gomezEcologicalInteractionsAre2010" role="doc-biblioref">Gómez, Verdú, and Perfectti 2010</a>; <a href="#ref-dallarivaExploringEvolutionarySignature2016" role="doc-biblioref">Dalla Riva and Stouffer 2016</a>)</span>. In the more proximal sense this is manifested as the ‘trait complementarity’ between two species, whereby one species (the predator) has the ‘correct’ set of traits that allow it to chase, capture, kill, and consume the other species (the prey). For species pairs where this condition is not met the link is deemed to be forbidden <span class="citation" data-cites="jordanoSamplingNetworksEcological2016">(<a href="#ref-jordanoSamplingNetworksEcological2016" role="doc-biblioref">Jordano 2016b</a>)</span>; <em>i.e.,</em> not physically possible and will always be absent within the network. In the context of trying to determine the feasibility (<em>i.e.,</em> the <em>possibility</em>) of an interaction, phylogeny is an excellent predictor <span class="citation" data-cites="strydomFoodWebReconstruction2022 frickeCollapseTerrestrialMammal2022">(<a href="#ref-strydomFoodWebReconstruction2022" role="doc-biblioref">Strydom et al. 2022</a>; <a href="#ref-frickeCollapseTerrestrialMammal2022" role="doc-biblioref">Fricke et al. 2022</a>)</span> and allows one to construct what can be considered to be a metaweb. In terms of thinking about the anatomy of an ‘feasibility network’ one should be aware that it is possible to represent interactions as either binary (feasible/forbidden; <em>i.e.,</em> the traditional definition of a metaweb <span class="citation" data-cites="dunneNetworkStructureFood2006">Jennifer A. Dunne (<a href="#ref-dunneNetworkStructureFood2006" role="doc-biblioref">2006</a>)</span>) or as a probability <span class="citation" data-cites="banvilleDecipheringProbabilisticSpecies2024">(<a href="#ref-banvilleDecipheringProbabilisticSpecies2024" role="doc-biblioref">Banville et al. 2024</a>)</span>, where the probability represents how likely that the interaction between to species is feasible (what is the possibility of this interaction occurring?).</p>
<p><strong>(Co)occurrence</strong></p>
<p>Although the outright assumption that because two species are co-occurring it must mean that they are interacting is inherently flawed <span class="citation" data-cites="blanchetCooccurrenceNotEvidence2020">(<a href="#ref-blanchetCooccurrenceNotEvidence2020" role="doc-biblioref">Blanchet, Cazelles, and Gravel 2020</a>)</span>, it is of course impossible for two species to interact (at least in terms of feeding links) if they are not co-occurring in time and space. Thus co-occurrence data alone is insufficient to build an accurate and ecologically meaningful representation of a food web having information on the co-occurrence of species can further aid us in refining metawebs by allowing us to downsample the network based on the species found in a specific location, or even add additional uncertainty based in how likely species are to co-occur <span class="citation" data-cites="dansereauSpatiallyExplicitPredictions2024">(<a href="#ref-dansereauSpatiallyExplicitPredictions2024" role="doc-biblioref">Dansereau, Barros, and Poisot 2024</a>)</span>. Additionally the interplay between the interaction between a species pair and their co-occurrence is meaningful when one is operating in the space of trying to determine the distribution of a species <span class="citation" data-cites="higinoMismatchIUCNRange2023">(<a href="#ref-higinoMismatchIUCNRange2023" role="doc-biblioref">Higino et al. 2023</a>)</span>, and forms a key component of some of the next generation species distribution models <em>e.g.,</em> joint SDMs <span class="citation" data-cites="pollockUnderstandingCooccurrenceModelling2014">(<a href="#ref-pollockUnderstandingCooccurrenceModelling2014" role="doc-biblioref">Pollock et al. 2014</a>)</span>.</p>
<p><strong>Abundance</strong></p>
<p>The abundance of the different species within the community can influence the likelihood of an interaction occurring in a myriad of ways. There is the argument that networks (and the interactions that make them up) are driven by only the abundance of the different species and not the characteristics (traits), <em>sensu</em> neutral processes and have been formalised with the neutral model <span class="citation" data-cites="canardEmergenceStructuralPatterns2012">(<a href="#ref-canardEmergenceStructuralPatterns2012" role="doc-biblioref">Canard et al. 2012</a>)</span>, as well as statistical tools <span class="citation" data-cites="momalTreebasedInferenceSpecies2020">(<a href="#ref-momalTreebasedInferenceSpecies2020" role="doc-biblioref">Momal, Robin, and Ambroise 2020</a>)</span>. Alternatively the abundance of species in a community can influence which interactions are ultimately realised <span class="citation" data-cites="poisotSpeciesWhyEcological2015 banvilleDecipheringProbabilisticSpecies2024">(<a href="#ref-poisotSpeciesWhyEcological2015" role="doc-biblioref">Poisot, Stouffer, and Gravel 2015</a>; <a href="#ref-banvilleDecipheringProbabilisticSpecies2024" role="doc-biblioref">Banville et al. 2024</a>)</span>.</p>
<p><strong>Predator choice (energetic cost)</strong></p>
<p>Ultimately, predator choice is underpinned by the energetic cost-benefit of trying to catch, kill, and consume prey, and is well described within both optimal foraging theory [ref] and metabolic theory [ref], which rests on the idea that the prey a predator chooses to target is one that will have the greatest return on energy with the lowest energetic cost. With a body of evidence that suggests that body size might be the underlying driver, and thus suitable proxy for understanding these processes <span class="citation" data-cites="yodzisBodySizeConsumerResource1992">(<a href="#ref-yodzisBodySizeConsumerResource1992" role="doc-biblioref">Yodzis and Innes 1992</a>)</span> There are additional bodies of work that attempt to include the cost of movement that the environment imposes on an individual <span class="citation" data-cites="cherifEnvironmentRescueCan2024">(<a href="#ref-cherifEnvironmentRescueCan2024" role="doc-biblioref">Cherif et al. 2024</a>)</span> as well as 2D/3D search space <span class="citation" data-cites="pawarDimensionalityConsumerSearch2012">(<a href="#ref-pawarDimensionalityConsumerSearch2012" role="doc-biblioref">Pawar, Dell, and Savage 2012</a>)</span>.</p>
<p><strong>Indirect interactions</strong></p>
<p>The realisation (presence/absence) or strength of trophic interactions themselves can also be modified by other, indirect (non-trophic), interactions <span class="citation" data-cites="golubskiModifyingModifiersWhat2011 pilosofMultilayerNatureEcological2017">(<a href="#ref-golubskiModifyingModifiersWhat2011" role="doc-biblioref">Golubski and Abrams 2011</a>; <a href="#ref-pilosofMultilayerNatureEcological2017" role="doc-biblioref">Pilosof et al. 2017</a>)</span>, this can be either ‘directly’ through <em>e.g.,</em> competition or ‘indirectly’ <em>e.g.,</em> mutualistic/facilitative interactions will alter the fine-scale distribution and abundance of some species <span class="citation" data-cites="kefiMoreMealIntegrating2012 kefiNetworkStructureFood2015">(<a href="#ref-kefiMoreMealIntegrating2012" role="doc-biblioref">Kéfi et al. 2012</a>, <a href="#ref-kefiNetworkStructureFood2015" role="doc-biblioref">2015</a>)</span>.</p>
</section>
<section id="contextualising-the-processes-that-determine-species-interactions" class="level2" data-number="2.2">
<h2 data-number="2.2" class="anchored" data-anchor-id="contextualising-the-processes-that-determine-species-interactions"><span class="header-section-number">2.2</span> Contextualising the processes that determine species interactions</h2>
<p>It should be self evident that the different processes discussed above are all ultimately going to influence the realisation of interactions as well as the structure of a network, however they are acting at different scales of organisation. Both the <strong>co-occurrence</strong> and the <strong>evolutionary compatibility</strong> are valid at the scale of the species pair of interest, that is the <em>possibility</em> of an interaction being present/absent is assessed at the pairwise level and one is left with a ‘list’ of interactions that are present/absent. Although it is possible to build a network (<em>i.e.,</em> metaweb) from this information it is important to be aware that the structure of this network is not constrained by real-world dynamics or conditions (<em>i.e.,</em> community context), and so just because species are able to interact does not mean that they will <span class="citation" data-cites="poisotSpeciesWhyEcological2015">(<a href="#ref-poisotSpeciesWhyEcological2015" role="doc-biblioref">Poisot, Stouffer, and Gravel 2015</a>)</span>. In order to construct a network who’s structure is a closer approximation of reality (localised interactions) one needs to take into consideration properties of the community as a whole and not just the two species of interest, which requires more data at the community scale, such as the abundance of species.</p>
<p><em>something about ‘physical’/landscape scale as well as time scale??</em></p>
</section>
</section>
<section id="network-construction-is-nuanced" class="level1" data-number="3">
<h1 data-number="3"><span class="header-section-number">3</span> Network construction is nuanced</h1>
<p>The act of constructing a ‘real world’ network will ultimately be delimited by its intended use, however the reality is that the empirical collection of interaction data is both costly [ref] and challenging to execute in a way that captures the different processes (owing to the different time and spatial scales they may be operating at). Thus we often turn to models to either predict networks, be that the interaction between two species, or its structure <span class="citation" data-cites="strydomRoadmapPredictingSpecies2021">(<a href="#ref-strydomRoadmapPredictingSpecies2021" role="doc-biblioref">Strydom et al. 2021</a>)</span>, or as a means to identify missing interactions (gap fill) within existing empirical dataset <span class="citation" data-cites="bitonInductiveLinkPrediction2024 stockPairwiseLearningPredicting2021">(<a href="#ref-bitonInductiveLinkPrediction2024" role="doc-biblioref">Biton, Puzis, and Pilosof 2024</a>; <a href="#ref-stockPairwiseLearningPredicting2021" role="doc-biblioref">Stock 2021</a>)</span>, and so for the purpose of this discussion network construction will be synonymous with using a model as a means to represent or predict a network — it can be argued that even the collection of empirical data is in and of itself a ‘model’ as it is still only a <em>representation</em> of the system. Different models have different underlying philosophies that often only capture one or a few of the processes discussed in <a href="#sec-process" class="quarto-xref">Section&nbsp;2</a>, has implications for how the resulting network is defined <a href="#sec-anatomy" class="quarto-xref">Section&nbsp;1</a>, which will ultimately delimit and define what inferences can be made from the resulting network. Here we will introduce the three different types of network representations, how they link back to the different processes determining interactions <a href="#fig-process" class="quarto-xref">Figure&nbsp;1</a>, and broadly discuss some of the modelling approaches that are used to construct these different network types. This is paralleled by a hypothetical case study (Box 1) where we showcase the utility/applicability of the different network representation in the context of trying to understand the feeding dynamics of a seasonal community.</p>
<div id="box-hypothetical" class="callout callout-style-simple callout-note no-icon callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Box 1 - Why we need to aggregate networks at different scales: A hypothetical case study
</div>
</div>
<div class="callout-body-container callout-body">
<p>Although it might seem most prudent to be predicting, constructing, and defining networks that are the closest representation of reality there are pros and cons of constructing both realised networks as well as metawebs. Let us take for example a community across time/through seasons. In this community we expect species to be either present or absent depending on the season (<em>i.e.,</em> changes in co-occurrence) as well as some species exhibiting seasonal diet shifts, these details would be lost at the scale of the metaweb an it would be valuable to construct either smaller metawebs for the different seasonal communities (thereby capturing the changes in community diversity), or realised networks for each season (to capture diet or ecosystem process shifts). However, these small-scale networks lack the context of the bigger picture that is available at the metaweb - that is it gives us a more holistic idea of the entire diet range of a specific species, which is important when one needs to make conservation-based/applied decisions (<em>e.g.,</em> conserving the entire diet of a species and not just seasonal prey items) as well as providing information on interactions that may be possible regardless of the environmental/community context (species may have the capacity to consume certain prey items but do not do so due to local conditions). With this is mind let us see how the different network aggregations can be used</p>
<p><strong>1: A global metaweb</strong></p>
<p>Knowledge of the entire diet breadth of a species is valuable especially in terms of understanding how a species will respond to changes in the community - <em>e.g.,</em> invasions/rewilding exercises (where does the new species ‘fit’ within the network?) as well as potential capacity to shift its diet. ALthough this might make sense across space and not time but certain species act as links across the landscape [Rooney]</p>
<p><strong>2: A seasonal metaweb</strong></p>
<p>Knowledge at the finer scale is also valuable to understand/identify that there are in fact differences between the seasons</p>
<p><strong>3: A seasonal realised network</strong></p>
<p>Dynamics are useful because they are a representation of the different configurations/energy flows/ecosystem processes. Also to detect more nuanced shifts in diet - <em>e.g.,</em> seasonal diet shifts.</p>
<p><strong>4: A structural network</strong></p>
<p><strong>Data trade off</strong></p>
<p>Above we highlight the practical uses of the different network configurations but we also need to take into consideration the barriers to construction/associated data needs/cost and acknowledge them. Basically in the ideal world we would have all this information at hand but in reality we might be sitting with seasonal metawebs…</p>
</div>
</div>
<section id="how-do-we-predict-food-webs" class="level2" data-number="3.1">
<h2 data-number="3.1" class="anchored" data-anchor-id="how-do-we-predict-food-webs"><span class="header-section-number">3.1</span> How do we predict food webs?</h2>
<p>There as many ways to predict networks as what there is to define them and along with taking into consideration the points raised in the previous section it is also beneficial to think about the context in which the different models were developed - and how this will influence the networks that they produce…</p>
<p>There is a bit of a ‘point of conflict’ between those calling for ‘pixel perfect’, regional scale data <span class="citation" data-cites="pringleUntanglingFoodWebs2020 pringleResolvingFoodWebStructure2020">(<a href="#ref-pringleUntanglingFoodWebs2020" role="doc-biblioref">Pringle 2020</a>; <a href="#ref-pringleResolvingFoodWebStructure2020" role="doc-biblioref">Pringle and Hutchinson 2020</a>)</span> and for the means to generate networks that are ecologically plausible <em>representations</em> (<em>sensu</em> structural networks). This represents two challenges; one is that models that represent generalisations of networks often lack the ability to retrieve any species/community specificity which limits their utility for real world, species-driven scenarios <em>e.g.,</em> species driven conservation efforts <span class="citation" data-cites="dunnSixthMassCoextinction2009">(<a href="#ref-dunnSixthMassCoextinction2009" role="doc-biblioref">Dunn et al. 2009</a>)</span>, however networks that are constructed through either (most) empirical observations or through predictive means are fundamentally going to represent metawebs, <em>i.e.,</em> lack constrained links, a representation of structure, or energy flow…</p>
<section id="models-that-predict-metawebs-feasible-interactions" class="level3" data-number="3.1.1">
<h3 data-number="3.1.1" class="anchored" data-anchor-id="models-that-predict-metawebs-feasible-interactions"><span class="header-section-number">3.1.1</span> Models that predict metawebs (feasible interactions)</h3>
<p>This is perhaps the most developed group of models; with a variety of approaches having been developed that typically determine the feasibility of an interaction based on the trait compatibility between predator and prey (<em>i.e.</em> their evolutionary compatibility) to determine ‘feeding rules’ <span class="citation" data-cites="morales-castillaInferringBioticInteractions2015">(<a href="#ref-morales-castillaInferringBioticInteractions2015" role="doc-biblioref">Morales-Castilla et al. 2015</a>)</span>. These feeding rules are broadly elucidated in two different ways; mechanistic feeding rules can be explicitly defined and applied to a community <span class="citation" data-cites="shawFrameworkReconstructingAncient2024 dunneCompilationNetworkAnalyses2008">(<em>e.g.,</em> <a href="#ref-shawFrameworkReconstructingAncient2024" role="doc-biblioref">Shaw et al. 2024</a>; <a href="#ref-dunneCompilationNetworkAnalyses2008" role="doc-biblioref">Jennifer A. Dunne et al. 2008</a>)</span> or they are inferred from a community for which there is interaction data and the ‘rules’ are then applied to a different community <span class="citation" data-cites="strydomGraphEmbeddingTransfer2023 pichlerMachineLearningAlgorithms2020 strydomFoodWebReconstruction2022 caronAddressingEltonianShortfall2022 llewelynPredictingPredatorPrey2023 desjardins-proulxEcologicalInteractionsNetflix2017 eklofSecondaryExtinctionsFood2013 cirtwillQuantitativeFrameworkInvestigating2019">(<em>e.g.,</em> <a href="#ref-strydomGraphEmbeddingTransfer2023" role="doc-biblioref">Strydom et al. 2023</a>, <a href="#ref-strydomFoodWebReconstruction2022" role="doc-biblioref">2022</a>; <a href="#ref-pichlerMachineLearningAlgorithms2020" role="doc-biblioref">Pichler et al. 2020</a>; <a href="#ref-caronAddressingEltonianShortfall2022" role="doc-biblioref">Caron et al. 2022</a>; <a href="#ref-llewelynPredictingPredatorPrey2023" role="doc-biblioref">Llewelyn et al. 2023</a>; <a href="#ref-desjardins-proulxEcologicalInteractionsNetflix2017" role="doc-biblioref">Desjardins-Proulx et al. 2017</a>; <a href="#ref-eklofSecondaryExtinctionsFood2013" role="doc-biblioref">Eklöf, Tang, and Allesina 2013</a>; <a href="#ref-cirtwillQuantitativeFrameworkInvestigating2019" role="doc-biblioref">Cirtwill et al. 2019</a>)</span>. The fundamental difference between these two model groups is that ‘mechanistic models’ rely on expert knowledge and make assumptions on trait-feeding relationships, whereas the ‘pattern finding’ models are dependent on existing datasets from which to elucidate feeding rules. These models are useful for determining all feasible interactions for a specific community, and owing to the availability of datasets <span class="citation" data-cites="poelenGlobalBioticInteractions2014 poisotMangalMakingEcological2016 grayJoiningDotsAutomated2015">(<em>e.g.,</em> <a href="#ref-poelenGlobalBioticInteractions2014" role="doc-biblioref">Poelen, Simons, and Mungall 2014</a>; <a href="#ref-poisotMangalMakingEcological2016" role="doc-biblioref">Poisot, Baiser, et al. 2016</a>; <a href="#ref-grayJoiningDotsAutomated2015" role="doc-biblioref">Gray et al. 2015</a>)</span>, as well as the development of model testing/benchmarking tools <span class="citation" data-cites="poisotGuidelinesPredictionSpecies2023">(<a href="#ref-poisotGuidelinesPredictionSpecies2023" role="doc-biblioref">Poisot 2023</a>)</span>, means that these models can be validated and (with relative confidence) be used to construct first draft networks for communities for which we have no data <span class="citation" data-cites="strydomFoodWebReconstruction2022">(<a href="#ref-strydomFoodWebReconstruction2022" role="doc-biblioref">Strydom et al. 2022</a>)</span>, and are valuable for constructing networks where we lack any interaction data <em>e.g.,</em> prehistoric networks <span class="citation" data-cites="yeakelCollapseEcologicalNetwork2014 frickeCollapseTerrestrialMammal2022">(<a href="#ref-yeakelCollapseEcologicalNetwork2014" role="doc-biblioref">Yeakel et al. 2014</a>; <a href="#ref-frickeCollapseTerrestrialMammal2022" role="doc-biblioref">Fricke et al. 2022</a>)</span>.</p>
</section>
<section id="models-that-predict-realised-networks-realised-interactions" class="level3" data-number="3.1.2">
<h3 data-number="3.1.2" class="anchored" data-anchor-id="models-that-predict-realised-networks-realised-interactions"><span class="header-section-number">3.1.2</span> Models that predict realised networks (realised interactions)</h3>
<p>In order to construct realised networks models need to incorporate <em>both</em> the feasibility of interactions (<em>i.e.,</em> determine the entire diet breadth of a species) as well as then determine which interactions are realised (<em>i.e.,</em> incorporate the ‘cost’ of interactions). As far as we are aware there is no model that explicitly accounts for both of these ‘rules’ and rather <em>only</em> account for processes that determine the realisation of an interaction (<em>i.e.,</em> abundance, predator choice, or non-trophic interactions). Although the use of allometric scaling <em>i.e.,</em> body size <span class="citation" data-cites="valdovinosBioenergeticFrameworkAboveground2023 beckermanForagingBiologyPredicts2006">(<em>e.g.,</em> <a href="#ref-valdovinosBioenergeticFrameworkAboveground2023" role="doc-biblioref">Valdovinos et al. 2023</a>; <a href="#ref-beckermanForagingBiologyPredicts2006" role="doc-biblioref">Beckerman, Petchey, and Warren 2006</a>)</span> may represent a first step in capturing evolutionary compatibility one still needs to account for other feeding traits. In terms of models that do formalise these processes, diet models <span class="citation" data-cites="beckermanForagingBiologyPredicts2006 petcheySizeForagingFood2008">(<a href="#ref-beckermanForagingBiologyPredicts2006" role="doc-biblioref">Beckerman, Petchey, and Warren 2006</a>; <a href="#ref-petcheySizeForagingFood2008" role="doc-biblioref">Petchey et al. 2008</a>)</span> have been used construct networks based on both predator choice (as determined by the handling time, energy content, and predator attack rate) as well as abundance (prey density). <span class="citation" data-cites="woottonModularTheoryTrophic2023">Wootton et al. (<a href="#ref-woottonModularTheoryTrophic2023" role="doc-biblioref">2023</a>)</span> developed a model that moves the energy of the system into different modules related to the process of the predator acquiring energy from the prey <em>i.e.,</em> compartmentation in food webs <span class="citation" data-cites="krauseCompartmentsRevealedFoodweb2003">(<a href="#ref-krauseCompartmentsRevealedFoodweb2003" role="doc-biblioref">Krause et al. 2003</a>)</span>.</p>
</section>
<section id="models-that-predict-structure-interaction-agnostic" class="level3" data-number="3.1.3">
<h3 data-number="3.1.3" class="anchored" data-anchor-id="models-that-predict-structure-interaction-agnostic"><span class="header-section-number">3.1.3</span> Models that predict structure (interaction agnostic)</h3>
<p>Although we identify mechanisms that determine species interactions in <a href="#sec-process" class="quarto-xref">Section&nbsp;2</a> not all models that are used to predict networks explicitly operate at the ‘process’ level, but rather represent the <em>structure</em> of a network based on a series of <em>a priori</em> assumptions as to the distribution of links between species (typically trophic not taxonomic species) by parametrising an aspect of the network structure, (<em>e.g.,</em> the niche model <span class="citation" data-cites="williamsSimpleRulesYield2000">(<a href="#ref-williamsSimpleRulesYield2000" role="doc-biblioref">Williams and Martinez 2000</a>)</span> makes an assumption as to the expected connectance of the network,although see <span class="citation" data-cites="allesinaFoodWebModels2009">Allesina and Pascual (<a href="#ref-allesinaFoodWebModels2009" role="doc-biblioref">2009</a>)</span> for a parameter-free model) or alternatively uses structural features of an exiting <em>realised</em> network (<em>e.g.,</em> stochastic block model, <span class="citation" data-cites="xieCompletenessCommunityStructure2017">Xie et al. (<a href="#ref-xieCompletenessCommunityStructure2017" role="doc-biblioref">2017</a>)</span>). Importantly these structural models do not make species specific predictions (they are usually species agnostic and treat nodes as trophic species) and so cannot be used to determine if an interaction is either possible <em>or</em> realised between two species (<em>i.e.,</em> one cannot use these models to determine if species <span class="math inline">\(a\)</span> eats species <span class="math inline">\(b\)</span>). Although this means this suite of models are unsuitable as tools for predicting species-specific interactions, they have been shown to be sufficient tools to predict the structure of networks <span class="citation" data-cites="williamsSuccessItsLimits2008">(<a href="#ref-williamsSuccessItsLimits2008" role="doc-biblioref">Williams and Martinez 2008</a>)</span>, and provide a data-light (the models often only require species richness) but assumption heavy (the resulting network structure is determined by an assumption of network structure) way to construct a network.</p>
</section>
</section>
</section>
<section id="making-progress-with-networks" class="level1" data-number="4">
<h1 data-number="4"><span class="header-section-number">4</span> Making Progress with Networks</h1>
<section id="further-development-of-models-and-tools" class="level2" data-number="4.1">
<h2 data-number="4.1" class="anchored" data-anchor-id="further-development-of-models-and-tools"><span class="header-section-number">4.1</span> Further development of models and tools</h2>
<p>There has been a suite of models that have been developed to predict trophic links, however we are lacking in tools that are explicitly taking into consideration estimating both the feasibility as well as realisation of links, <em>i.e.,</em> both interactions and structure simultaneously <span class="citation" data-cites="strydomRoadmapPredictingSpecies2021">(<a href="#ref-strydomRoadmapPredictingSpecies2021" role="doc-biblioref">Strydom et al. 2021</a>)</span>. This could be addressed either through the development of tools that do both (predict both interactions and structure), or to develop an ensemble modelling approach <span class="citation" data-cites="beckerOptimisingPredictiveModels2022">(<a href="#ref-beckerOptimisingPredictiveModels2022" role="doc-biblioref">Becker et al. 2022</a>)</span>. Alternatively the development of tools that will allow for the downsampling of metawebs into realised networks <span class="citation" data-cites="roopnarineExtinctionCascadesCatastrophe2006">(<em>e.g.,</em> <a href="#ref-roopnarineExtinctionCascadesCatastrophe2006" role="doc-biblioref">Roopnarine 2006</a>)</span>, although deciding exactly what is driving differences between local networks and the regional metaweb might not be that simple <span class="citation" data-cites="saraviaEcologicalNetworkAssembly2022">(<a href="#ref-saraviaEcologicalNetworkAssembly2022" role="doc-biblioref">Saravia et al. 2022</a>)</span>. Probably also something that aligns with trying to predict interaction strength - because that would be the gold standard. Probably also worth just plainly stating that feasibility of developing a model that is both broadly generalisable, but also has local specificity is probably not attainable <span class="citation" data-cites="stoufferAllEcologicalModels2019">(<a href="#ref-stoufferAllEcologicalModels2019" role="doc-biblioref">Stouffer 2019</a>)</span>, and more specifically the potential use in models untangling/identifying the different processes that shape interaction networks <span class="citation" data-cites="songRigorousValidationEcological2024">(<a href="#ref-songRigorousValidationEcological2024" role="doc-biblioref">Song and Levine 2024</a>)</span>, <em>e.g.,</em> <span class="citation" data-cites="curtsdotterEcosystemFunctionPredator2019">Curtsdotter et al. (<a href="#ref-curtsdotterEcosystemFunctionPredator2019" role="doc-biblioref">2019</a>)</span> showcasing the use of models to disentangle the drivers of community function and <span class="citation" data-cites="strydomSVDEntropyReveals2021">Strydom, Dalla Riva, and Poisot (<a href="#ref-strydomSVDEntropyReveals2021" role="doc-biblioref">2021</a>)</span> who identified that networks are less complex than they could be, suggesting that there are constraints on network assembly. In addition to the more intentional development of models we also need to consider the validation of these models, there have been developments and discussions for assessing how well a model recovers pairwise interactions <span class="citation" data-cites="strydomRoadmapPredictingSpecies2021 poisotGuidelinesPredictionSpecies2023">(<a href="#ref-strydomRoadmapPredictingSpecies2021" role="doc-biblioref">Strydom et al. 2021</a>; <a href="#ref-poisotGuidelinesPredictionSpecies2023" role="doc-biblioref">Poisot 2023</a>)</span>, although the rate of false-negatives that may be present in the testing data still present a challenge <span class="citation" data-cites="catchenMissingLinkDiscerning2023">(<a href="#ref-catchenMissingLinkDiscerning2023" role="doc-biblioref">Catchen et al. 2023</a>)</span>, and we still lack clear strategies for benchmarking the ability of models to recover structure <span class="citation" data-cites="allesinaGeneralModelFood2008">(<a href="#ref-allesinaGeneralModelFood2008" role="doc-biblioref">Allesina, Alonso, and Pascual 2008</a>)</span>.</p>
<section id="at-what-scale-should-we-be-predicting-and-using-networks" class="level3" data-number="4.1.1">
<h3 data-number="4.1.1" class="anchored" data-anchor-id="at-what-scale-should-we-be-predicting-and-using-networks"><span class="header-section-number">4.1.1</span> At what scale should we be predicting and using networks?</h3>
<p>Look at <span class="citation" data-cites="hutchinsonSeeingForestTrees2019">Hutchinson et al. (<a href="#ref-hutchinsonSeeingForestTrees2019" role="doc-biblioref">2019</a>)</span></p>
<p>We lack a clear agenda (and conceptualisation) as to what the appropriate level of aggregation is for a ‘network’. Realistically most empirical networks are more aligned with metawebs as opposed to realised networks as they are often the result of some sort of aggregation of observations across time, this creates a two-fold problem. Firstly, we need to think about how this affects any sort of development of theory that sits closer to the ‘realised network’ side of the spectrum - how often are we trying to ask and answer questions about realised networks using feasible networks? The second is that this lack of ‘direction’ as to how we should define a network is (actually) probably one of the biggest barriers that is affecting the use of networks in applied settings… By define I mean both delimiting the time and geographic scale at which a network is aggregated at <span class="citation" data-cites="estayEditorialPatternsProcesses2023">(<a href="#ref-estayEditorialPatternsProcesses2023" role="doc-biblioref">Estay, Fortin, and López 2023</a>)</span>. This is important because it can influence the inferences made, <em>e.g.,</em> the large body of work (landscape theory for food web architecture) that showcases how different species use the landscape will influence network dynamics <span class="citation" data-cites="rooneyLandscapeTheoryFood2008">(<a href="#ref-rooneyLandscapeTheoryFood2008" role="doc-biblioref">Rooney, McCann, and Moore 2008</a>)</span>. There is also a bit of an interplay with time and data and the different scales that they may be integrated at - co-occurrence may span decades and just because two species have been recorded in the same space does not mean it was at the same timescale <span class="citation" data-cites="brimacombeApplyingMethodIts2024">(<a href="#ref-brimacombeApplyingMethodIts2024" role="doc-biblioref">Brimacombe, Bodner, and Fortin 2024</a>)</span></p>
</section>
<section id="feasible-realised-or-sustainable" class="level3" data-number="4.1.2">
<h3 data-number="4.1.2" class="anchored" data-anchor-id="feasible-realised-or-sustainable"><span class="header-section-number">4.1.2</span> Feasible, realised, or sustainable?</h3>
<p>When do we determine a link to be ‘real’… In the context of metawebs this is perhaps clearer - if all things were equal (<em>i.e.,</em> community context is irrelevant) would the predator be able to consume the prey. However in the realised space there is also the question of the long term ‘energetic feasibility’ of an interaction - just because an interaction is possible in the now is it able to sustain a population in the long term. And what is the scale for that long term - are we thinking at the generational scale? Because ultimately when we are constructing a network we are aggregating not only across space but also across time… This is probably again a Lokta-Volterra space question and something that the dynamic foodweb model <span class="citation" data-cites="lajaaitiEcologicalNetworksDynamicsJlJulia2024 delmasSimulationsBiomassDynamics2017 curtsdotterEcosystemFunctionPredator2019">(<a href="#ref-lajaaitiEcologicalNetworksDynamicsJlJulia2024" role="doc-biblioref">Lajaaiti et al. 2024</a>; <a href="#ref-delmasSimulationsBiomassDynamics2017" role="doc-biblioref">Delmas et al. 2017</a>; <a href="#ref-curtsdotterEcosystemFunctionPredator2019" role="doc-biblioref">Curtsdotter et al. 2019</a>)</span> is addressing, but again it is integrating this with the feasible/realised axis.</p>
</section>
</section>
<section id="how-should-we-use-different-networks" class="level2" data-number="4.2">
<h2 data-number="4.2" class="anchored" data-anchor-id="how-should-we-use-different-networks"><span class="header-section-number">4.2</span> How should we use different networks?</h2>
<p>What for and how we can use networks is perhaps one of the biggest ‘gaps’ we have in network ecology (Tim’s EBV ms), and there is a serious need to start drawing clear, ecological links between network form and function <span class="citation" data-cites="delmasAnalysingEcologicalNetworks2019">(although see <a href="#ref-delmasAnalysingEcologicalNetworks2019" role="doc-biblioref">Delmas et al. 2019</a>)</span>. That being said one of the most important things we can do is to be aware of the parameter space that is possible given a specific definition of a network and operate within those parameters. Here we can maybe tie it back to scale - specifically the idea that the fact that metawebs operate at evolutionary scales they are not suitable for ‘dynamic’ processes, although they do have the potential to think about novel species entering the network/community…</p>
</section>
</section>
<section id="concluding-remarks" class="level1" data-number="5">
<h1 data-number="5"><span class="header-section-number">5</span> Concluding remarks</h1>
<p>I think a big take home will (hopefully) be how different approaches do better in different situations and so you as an end user need to take this into consideration and pick accordingly. I think <span class="citation" data-cites="petcheyFitEfficiencyBiology2011">Petchey et al. (<a href="#ref-petcheyFitEfficiencyBiology2011" role="doc-biblioref">2011</a>)</span> might have (and share) some thoughts on this. I feel like I need to look at <span class="citation" data-cites="berlowGoldilocksFactorFood2008">Berlow, Brose, and Martinez (<a href="#ref-berlowGoldilocksFactorFood2008" role="doc-biblioref">2008</a>)</span> but maybe not exactly in this context but vaguely adjacent. This is sort of the crux of the argument presented in <span class="citation" data-cites="brimacombeApplyingMethodIts2024">Brimacombe, Bodner, and Fortin (<a href="#ref-brimacombeApplyingMethodIts2024" role="doc-biblioref">2024</a>)</span> as well.</p>
<p>Do we expect there to be differences when thinking about unipartite vs bipartite networks? Is there underlying ecology/theory that would assume that different mechanisms (and thus models) are relevant in these two ‘systems’.</p>
<ul>
<li>The <span class="citation" data-cites="terryFindingMissingLinks2020">Terry and Lewis (<a href="#ref-terryFindingMissingLinks2020" role="doc-biblioref">2020</a>)</span> paper looks at some methods but is specifically looking at a bipartite world…</li>
</ul>
</section>
<section id="references" class="level1 unnumbered">
<h1 class="unnumbered">References</h1>
<div id="refs" class="references csl-bib-body hanging-indent" data-entry-spacing="0" role="list">
<div id="ref-allesinaGeneralModelFood2008" class="csl-entry" role="listitem">
Allesina, Stefano, David Alonso, and Mercedes Pascual. 2008. <span>“A <span>General Model</span> for <span>Food Web Structure</span>.”</span> <em>Science</em> 320 (5876): 658–61. <a href="https://doi.org/10.1126/science.1156269">https://doi.org/10.1126/science.1156269</a>.
</div>
<div id="ref-allesinaFoodWebModels2009" class="csl-entry" role="listitem">
Allesina, Stefano, and Mercedes Pascual. 2009. <span>“Food Web Models: A Plea for Groups.”</span> <em>Ecology Letters</em> 12 (7): 652–62. <a href="https://doi.org/10.1111/j.1461-0248.2009.01321.x">https://doi.org/10.1111/j.1461-0248.2009.01321.x</a>.
</div>
<div id="ref-banvilleDecipheringProbabilisticSpecies2024" class="csl-entry" role="listitem">
Banville, Francis, Tanya Strydom, Penelope Blyth, Chris Brimacombe, Michael D. Catchen, Gabriel Dansereau, Gracielle Higino, et al. 2024. <span>“Deciphering Probabilistic Species Interaction Networks.”</span> EcoEvoRxiv. <a href="https://doi.org/10.32942/X28G8Z">https://doi.org/10.32942/X28G8Z</a>.
</div>
<div id="ref-beckerOptimisingPredictiveModels2022" class="csl-entry" role="listitem">
Becker, Daniel J., Gregory F. Albery, Anna R. Sjodin, Timothée Poisot, Laura M. Bergner, Binqi Chen, Lily E. Cohen, et al. 2022. <span>“Optimising Predictive Models to Prioritise Viral Discovery in Zoonotic Reservoirs.”</span> <em>The Lancet Microbe</em> 3 (8): e625–37. <a href="https://doi.org/10.1016/S2666-5247(21)00245-7">https://doi.org/10.1016/S2666-5247(21)00245-7</a>.
</div>
<div id="ref-beckermanForagingBiologyPredicts2006" class="csl-entry" role="listitem">
Beckerman, Andrew P., Owen L. Petchey, and Philip H. Warren. 2006. <span>“Foraging Biology Predicts Food Web Complexity.”</span> <em>Proceedings of the National Academy of Sciences</em> 103 (37): 13745–49. <a href="https://doi.org/10.1073/pnas.0603039103">https://doi.org/10.1073/pnas.0603039103</a>.
</div>
<div id="ref-berlowGoldilocksFactorFood2008" class="csl-entry" role="listitem">
Berlow, Eric L., Ulrich Brose, and Neo D. Martinez. 2008. <span>“The <span>‘<span>Goldilocks</span> Factor’</span> in Food Webs.”</span> <em>Proceedings of the National Academy of Sciences</em> 105 (11): 4079–80. <a href="https://doi.org/10.1073/pnas.0800967105">https://doi.org/10.1073/pnas.0800967105</a>.
</div>
<div id="ref-berlowInteractionStrengthsFood2004" class="csl-entry" role="listitem">
Berlow, Eric L., Anje-Margiet Neutel, Joel E. Cohen, Peter C. de Ruiter, Bo Ebenman, Mark Emmerson, Jeremy W. Fox, et al. 2004. <span>“Interaction Strengths in Food Webs: Issues and Opportunities.”</span> <em>Journal of Animal Ecology</em> 73 (3): 585–98. <a href="https://doi.org/10.1111/j.0021-8790.2004.00833.x">https://doi.org/10.1111/j.0021-8790.2004.00833.x</a>.
</div>
<div id="ref-bitonInductiveLinkPrediction2024" class="csl-entry" role="listitem">
Biton, Barry, Rami Puzis, and Shai Pilosof. 2024. <span>“Inductive Link Prediction Boosts Data Availability and Enables Cross-Community Link Prediction in Ecological Networks,”</span> August.
</div>
<div id="ref-blanchetCooccurrenceNotEvidence2020" class="csl-entry" role="listitem">
Blanchet, F. Guillaume, Kevin Cazelles, and Dominique Gravel. 2020. <span>“Co-Occurrence Is Not Evidence of Ecological Interactions.”</span> <em>Ecology Letters</em> 23 (7): 1050–63. <a href="https://doi.org/10.1111/ele.13525">https://doi.org/10.1111/ele.13525</a>.
</div>
<div id="ref-brimacombeApplyingMethodIts2024" class="csl-entry" role="listitem">
Brimacombe, Chris, Korryn Bodner, and Marie-Josee Fortin. 2024. <span>“Applying a Method Before Its Proof-of-Concept: <span>A</span> Cautionary Tale Using Inferred Food Webs.”</span> <a href="https://doi.org/10.13140/RG.2.2.22076.65927">https://doi.org/10.13140/RG.2.2.22076.65927</a>.
</div>
<div id="ref-brimacombeShortcomingsReusingSpecies2023" class="csl-entry" role="listitem">
Brimacombe, Chris, Korryn Bodner, Matthew Michalska-Smith, Timothée Poisot, and Marie-Josée Fortin. 2023. <span>“Shortcomings of Reusing Species Interaction Networks Created by Different Sets of Researchers.”</span> <em>PLOS Biology</em> 21 (4): e3002068. <a href="https://doi.org/10.1371/journal.pbio.3002068">https://doi.org/10.1371/journal.pbio.3002068</a>.
</div>
<div id="ref-canardEmergenceStructuralPatterns2012" class="csl-entry" role="listitem">
Canard, Elsa, Nicolas Mouquet, Lucile Marescot, Kevin J. Gaston, Dominique Gravel, and David Mouillot. 2012. <span>“Emergence of <span>Structural Patterns</span> in <span>Neutral Trophic Networks</span>.”</span> <em>PLOS ONE</em> 7 (8): e38295. <a href="https://doi.org/10.1371/journal.pone.0038295">https://doi.org/10.1371/journal.pone.0038295</a>.
</div>
<div id="ref-caronTraitmatchingModelsPredict2024" class="csl-entry" role="listitem">
Caron, Dominique, Ulrich Brose, Miguel Lurgi, F. Guillaume Blanchet, Dominique Gravel, and Laura J. Pollock. 2024. <span>“Trait-Matching Models Predict Pairwise Interactions Across Regions, Not Food Web Properties.”</span> <em>Global Ecology and Biogeography</em> 33 (4): e13807. <a href="https://doi.org/10.1111/geb.13807">https://doi.org/10.1111/geb.13807</a>.
</div>
<div id="ref-caronAddressingEltonianShortfall2022" class="csl-entry" role="listitem">
Caron, Dominique, Luigi Maiorano, Wilfried Thuiller, and Laura J. Pollock. 2022. <span>“Addressing the <span>Eltonian</span> Shortfall with Trait-Based Interaction Models.”</span> <em>Ecology Letters</em> 25 (4): 889–99. <a href="https://doi.org/10.1111/ele.13966">https://doi.org/10.1111/ele.13966</a>.
</div>
<div id="ref-catchenMissingLinkDiscerning2023" class="csl-entry" role="listitem">
Catchen, Michael D., Timothée Poisot, Laura J. Pollock, and Andrew Gonzalez. 2023. <span>“The Missing Link: Discerning True from False Negatives When Sampling Species Interaction Networks,”</span> January.
</div>
<div id="ref-cherifEnvironmentRescueCan2024" class="csl-entry" role="listitem">
Cherif, Mehdi, Ulrich Brose, Myriam R. Hirt, Remo Ryser, Violette Silve, Georg Albert, Russell Arnott, et al. 2024. <span>“The Environment to the Rescue: Can Physics Help Predict Predator–Prey Interactions?”</span> <em>Biological Reviews</em> n/a (n/a). <a href="https://doi.org/10.1111/brv.13105">https://doi.org/10.1111/brv.13105</a>.
</div>
<div id="ref-cirtwillQuantitativeFrameworkInvestigating2019" class="csl-entry" role="listitem">
Cirtwill, Alyssa R., Anna Eklf, Tomas Roslin, Kate Wootton, and Dominique Gravel. 2019. <span>“A Quantitative Framework for Investigating the Reliability of Empirical Network Construction.”</span> <em>Methods in Ecology and Evolution</em> 10 (6): 902–11. <a href="https://doi.org/10.1111/2041-210X.13180">https://doi.org/10.1111/2041-210X.13180</a>.
</div>
<div id="ref-cleggImpactIntraspecificVariation2018" class="csl-entry" role="listitem">
Clegg, Tom, Mohammad Ali, and Andrew P. Beckerman. 2018. <span>“The Impact of Intraspecific Variation on Food Web Structure.”</span> <em>Ecology</em> 99 (12): 2712–20. <a href="https://doi.org/10.1002/ecy.2523">https://doi.org/10.1002/ecy.2523</a>.
</div>
<div id="ref-curtsdotterEcosystemFunctionPredator2019" class="csl-entry" role="listitem">
Curtsdotter, Alva, H. Thomas Banks, John E. Banks, Mattias Jonsson, Tomas Jonsson, Amanda N. Laubmeier, Michael Traugott, and Riccardo Bommarco. 2019. <span>“Ecosystem Function in Predator–Prey Food Webs—Confronting Dynamic Models with Empirical Data.”</span> <em>Journal of Animal Ecology</em> 88 (2): 196–210. <a href="https://doi.org/10.1111/1365-2656.12892">https://doi.org/10.1111/1365-2656.12892</a>.
</div>
<div id="ref-dallarivaExploringEvolutionarySignature2016" class="csl-entry" role="listitem">
Dalla Riva, Giulio V, and Daniel B. Stouffer. 2016. <span>“Exploring the Evolutionary Signature of Food Webs’ Backbones Using Functional Traits.”</span> <em>Oikos</em> 125 (4): 446–56. <a href="https://doi.org/10.1111/oik.02305">https://doi.org/10.1111/oik.02305</a>.
</div>
<div id="ref-dansereauSpatiallyExplicitPredictions2024" class="csl-entry" role="listitem">
Dansereau, Gabriel, Ceres Barros, and Timothée Poisot. 2024. <span>“Spatially Explicit Predictions of Food Web Structure from Regional-Level Data.”</span> <em>Philosophical Transactions of the Royal Society B: Biological Sciences</em> 379 (1909). <a href="https://doi.org/10.1098/rstb.2023.0166">https://doi.org/10.1098/rstb.2023.0166</a>.
</div>
<div id="ref-delmasAnalysingEcologicalNetworks2019" class="csl-entry" role="listitem">
Delmas, Eva, Mathilde Besson, Marie-Hélène Brice, Laura A. Burkle, Giulio V. Dalla Riva, Marie-Josée Fortin, Dominique Gravel, et al. 2019. <span>“Analysing Ecological Networks of Species Interactions.”</span> <em>Biological Reviews</em> 94 (1): 16–36. <a href="https://doi.org/10.1111/brv.12433">https://doi.org/10.1111/brv.12433</a>.
</div>
<div id="ref-delmasSimulationsBiomassDynamics2017" class="csl-entry" role="listitem">
Delmas, Eva, Ulrich Brose, Dominique Gravel, Daniel B. Stouffer, and Timothée Poisot. 2017. <span>“Simulations of Biomass Dynamics in Community Food Webs.”</span> <em>Methods in Ecology and Evolution</em> 8 (7): 881–86. <a href="https://doi.org/10.1111/2041-210X.12713">https://doi.org/10.1111/2041-210X.12713</a>.
</div>
<div id="ref-desjardins-proulxEcologicalInteractionsNetflix2017" class="csl-entry" role="listitem">
Desjardins-Proulx, Philippe, Idaline Laigle, Timothée Poisot, and Dominique Gravel. 2017. <span>“Ecological Interactions and the <span>Netflix</span> Problem.”</span> <em>PeerJ</em> 5: e3644. <a href="https://doi.org/10.7717/peerj.3644">https://doi.org/10.7717/peerj.3644</a>.
</div>
<div id="ref-dunnSixthMassCoextinction2009" class="csl-entry" role="listitem">
Dunn, Robert R., Nyeema C. Harris, Robert K. Colwell, Lian Pin Koh, and Navjot S. Sodhi. 2009. <span>“The Sixth Mass Coextinction: Are Most Endangered Species Parasites and Mutualists?”</span> <em>Proceedings. Biological Sciences</em> 276 (1670): 3037–45. <a href="https://doi.org/10.1098/rspb.2009.0413">https://doi.org/10.1098/rspb.2009.0413</a>.
</div>
<div id="ref-dunneNetworkStructureFood2006" class="csl-entry" role="listitem">
Dunne, Jennifer A. 2006. <span>“The <span>Network Structure</span> of <span>Food Webs</span>.”</span> In <em>Ecological Networks: <span>Linking</span> Structure and Dynamics</em>, edited by Jennifer A Dunne and Mercedes Pascual, 27–86. Oxford University Press.
</div>
<div id="ref-dunneCompilationNetworkAnalyses2008" class="csl-entry" role="listitem">
Dunne, Jennifer A., Richard J. Williams, Neo D. Martinez, Rachel A. Wood, and Douglas H. Erwin. 2008. <span>“Compilation and <span>Network Analyses</span> of <span>Cambrian Food Webs</span>.”</span> <em>PLOS Biology</em> 6 (4): e102. <a href="https://doi.org/10.1371/journal.pbio.0060102">https://doi.org/10.1371/journal.pbio.0060102</a>.
</div>
<div id="ref-eklofSecondaryExtinctionsFood2013" class="csl-entry" role="listitem">
Eklöf, Anna, Si Tang, and Stefano Allesina. 2013. <span>“Secondary Extinctions in Food Webs: A <span>Bayesian</span> Network Approach.”</span> Edited by Jessica Metcalf. <em>Methods in Ecology and Evolution</em> 4 (8): 760–70. <a href="https://doi.org/10.1111/2041-210X.12062">https://doi.org/10.1111/2041-210X.12062</a>.
</div>
<div id="ref-estayEditorialPatternsProcesses2023" class="csl-entry" role="listitem">
Estay, Sergio A., Marie-Josée Fortin, and Daniela N. López. 2023. <span>“Editorial: <span>Patterns</span> and Processes in Ecological Networks over Space.”</span> <em>Frontiers in Ecology and Evolution</em> 11.
</div>
<div id="ref-frickeCollapseTerrestrialMammal2022" class="csl-entry" role="listitem">
Fricke, Evan C., Chia Hsieh, Owen Middleton, Daniel Gorczynski, Caroline D. Cappello, Oscar Sanisidro, John Rowan, Jens-Christian Svenning, and Lydia Beaudrot. 2022. <span>“Collapse of Terrestrial Mammal Food Webs Since the <span>Late Pleistocene</span>.”</span> <em>Science</em> 377 (6609): 1008–11. <a href="https://doi.org/10.1126/science.abn4012">https://doi.org/10.1126/science.abn4012</a>.
</div>
<div id="ref-golubskiModifyingModifiersWhat2011" class="csl-entry" role="listitem">
Golubski, Antonio J., and Peter A. Abrams. 2011. <span>“Modifying Modifiers: What Happens When Interspecific Interactions Interact?”</span> <em>Journal of Animal Ecology</em> 80 (5): 1097–1108. <a href="https://doi.org/10.1111/j.1365-2656.2011.01852.x">https://doi.org/10.1111/j.1365-2656.2011.01852.x</a>.
</div>
<div id="ref-gomezEcologicalInteractionsAre2010" class="csl-entry" role="listitem">
Gómez, José M., Miguel Verdú, and Francisco Perfectti. 2010. <span>“Ecological Interactions Are Evolutionarily Conserved Across the Entire Tree of Life.”</span> <em>Nature</em> 465 (7300): 918–21. <a href="https://doi.org/10.1038/nature09113">https://doi.org/10.1038/nature09113</a>.
</div>
<div id="ref-grayJoiningDotsAutomated2015" class="csl-entry" role="listitem">
Gray, Clare, David H. Figueroa, Lawrence N. Hudson, Athen Ma, Dan Perkins, and Guy Woodward. 2015. <span>“Joining the Dots: <span>An</span> Automated Method for Constructing Food Webs from Compendia of Published Interactions.”</span> <em>Food Webs</em> 5 (December): 11–20. <a href="https://doi.org/10.1016/j.fooweb.2015.09.001">https://doi.org/10.1016/j.fooweb.2015.09.001</a>.
</div>
<div id="ref-higinoMismatchIUCNRange2023" class="csl-entry" role="listitem">
Higino, Gracielle T., Francis Banville, Gabriel Dansereau, Norma Rocio Forero Muñoz, Fredric Windsor, and Timothée Poisot. 2023. <span>“Mismatch Between <span>IUCN</span> Range Maps and Species Interactions Data Illustrated Using the <span>Serengeti</span> Food Web.”</span> <em>PeerJ</em> 11 (February): e14620. <a href="https://doi.org/10.7717/peerj.14620">https://doi.org/10.7717/peerj.14620</a>.
</div>
<div id="ref-hutchinsonSeeingForestTrees2019" class="csl-entry" role="listitem">
Hutchinson, Matthew C., Bernat Bramon Mora, Shai Pilosof, Allison K. Barner, Sonia Kéfi, Elisa Thébault, Pedro Jordano, and Daniel B. Stouffer. 2019. <span>“Seeing the Forest for the Trees: <span>Putting</span> Multilayer Networks to Work for Community Ecology.”</span> Edited by Oscar Godoy. <em>Functional Ecology</em> 33 (2): 206–17. <a href="https://doi.org/10.1111/1365-2435.13237">https://doi.org/10.1111/1365-2435.13237</a>.
</div>
<div id="ref-jordanoChasingEcologicalInteractions2016" class="csl-entry" role="listitem">
Jordano, Pedro. 2016a. <span>“Chasing <span>Ecological Interactions</span>.”</span> <em>PLOS Biology</em> 14 (9): e1002559. <a href="https://doi.org/10.1371/journal.pbio.1002559">https://doi.org/10.1371/journal.pbio.1002559</a>.
</div>
<div id="ref-jordanoSamplingNetworksEcological2016" class="csl-entry" role="listitem">
———. 2016b. <span>“Sampling Networks of Ecological Interactions.”</span> <em>Functional Ecology</em>, September. <a href="https://doi.org/10.1111/1365-2435.12763">https://doi.org/10.1111/1365-2435.12763</a>.
</div>
<div id="ref-kefiNetworkStructureFood2015" class="csl-entry" role="listitem">
Kéfi, Sonia, Eric L. Berlow, Evie A. Wieters, Lucas N. Joppa, Spencer A. Wood, Ulrich Brose, and Sergio A. Navarrete. 2015. <span>“Network Structure Beyond Food Webs: Mapping Non-Trophic and Trophic Interactions on <span>Chilean</span> Rocky Shores.”</span> <em>Ecology</em> 96 (1): 291–303. <a href="https://doi.org/10.1890/13-1424.1">https://doi.org/10.1890/13-1424.1</a>.
</div>
<div id="ref-kefiMoreMealIntegrating2012" class="csl-entry" role="listitem">
Kéfi, Sonia, Eric L. Berlow, Evie A. Wieters, Sergio A. Navarrete, Owen L. Petchey, Spencer A. Wood, Alice Boit, et al. 2012. <span>“More Than a Meal<span>…</span> Integrating Non-Feeding Interactions into Food Webs: <span>More</span> Than a Meal <span>…</span>.”</span> <em>Ecology Letters</em> 15 (4): 291–300. <a href="https://doi.org/10.1111/j.1461-0248.2011.01732.x">https://doi.org/10.1111/j.1461-0248.2011.01732.x</a>.
</div>
<div id="ref-krauseCompartmentsRevealedFoodweb2003" class="csl-entry" role="listitem">
Krause, Ann E., Kenneth A. Frank, Doran M. Mason, Robert E. Ulanowicz, and William W. Taylor. 2003. <span>“Compartments Revealed in Food-Web Structure.”</span> <em>Nature</em> 426 (6964): 282–85. <a href="https://doi.org/10.1038/nature02115">https://doi.org/10.1038/nature02115</a>.
</div>
<div id="ref-lajaaitiEcologicalNetworksDynamicsJlJulia2024" class="csl-entry" role="listitem">
Lajaaiti, Ismaël, Iago Bonnici, Sonia Kéfi, Hana Mayall, Alain Danet, Andrew P. Beckerman, Thomas Malpas, and Eva Delmas. 2024. <span>“<span>EcologicalNetworksDynamics</span>.jl <span>A Julia</span> Package to Simulate the Temporal Dynamics of Complex Ecological Networks.”</span> bioRxiv. <a href="https://doi.org/10.1101/2024.03.20.585899">https://doi.org/10.1101/2024.03.20.585899</a>.
</div>
<div id="ref-lindemanTrophicDynamicAspectEcology1942" class="csl-entry" role="listitem">
Lindeman, Raymond L. 1942. <span>“The <span>Trophic-Dynamic Aspect</span> of <span>Ecology</span>.”</span> <em>Ecology</em> 23 (4): 399–417. <a href="https://doi.org/10.2307/1930126">https://doi.org/10.2307/1930126</a>.
</div>
<div id="ref-llewelynPredictingPredatorPrey2023" class="csl-entry" role="listitem">
Llewelyn, John, Giovanni Strona, Christopher R. Dickman, Aaron C. Greenville, Glenda M. Wardle, Michael S. Y. Lee, Seamus Doherty, Farzin Shabani, Frédérik Saltré, and Corey J. A. Bradshaw. 2023. <span>“Predicting Predator–Prey Interactions in Terrestrial Endotherms Using Random Forest.”</span> <em>Ecography</em> 2023 (9): e06619. <a href="https://doi.org/10.1111/ecog.06619">https://doi.org/10.1111/ecog.06619</a>.
</div>
<div id="ref-maioranoTETRAEUSpecieslevelTrophic2020" class="csl-entry" role="listitem">
Maiorano, Luigi, Alessandro Montemaggiori, Gentile Francesco Ficetola, Louise O’Connor, and Wilfried Thuiller. 2020. <span>“<span>TETRA-EU</span> 1.0: <span>A</span> Species-Level Trophic Metaweb of <span>European</span> Tetrapods.”</span> <em>Global Ecology and Biogeography</em> 29 (9): 1452–57. <a href="https://doi.org/10.1111/geb.13138">https://doi.org/10.1111/geb.13138</a>.
</div>
<div id="ref-momalTreebasedInferenceSpecies2020" class="csl-entry" role="listitem">
Momal, Raphaëlle, Stéphane Robin, and Christophe Ambroise. 2020. <span>“Tree-Based Inference of Species Interaction Networks from Abundance Data.”</span> <em>Methods in Ecology and Evolution</em> 11 (5): 621–32. <a href="https://doi.org/10.1111/2041-210X.13380">https://doi.org/10.1111/2041-210X.13380</a>.
</div>
<div id="ref-morales-castillaInferringBioticInteractions2015" class="csl-entry" role="listitem">
Morales-Castilla, Ignacio, Miguel G. Matias, Dominique Gravel, and Miguel B. Araújo. 2015. <span>“Inferring Biotic Interactions from Proxies.”</span> <em>Trends in Ecology &amp; Evolution</em> 30 (6): 347–56. <a href="https://doi.org/10.1016/j.tree.2015.03.014">https://doi.org/10.1016/j.tree.2015.03.014</a>.
</div>
<div id="ref-pawarDimensionalityConsumerSearch2012" class="csl-entry" role="listitem">
Pawar, Samraat, Anthony I. Dell, and Van M. Savage. 2012. <span>“Dimensionality of Consumer Search Space Drives Trophic Interaction Strengths.”</span> <em>Nature</em> 486 (7404): 485–89. <a href="https://doi.org/10.1038/nature11131">https://doi.org/10.1038/nature11131</a>.
</div>
<div id="ref-petcheySizeForagingFood2008" class="csl-entry" role="listitem">
Petchey, Owen L., Andrew P. Beckerman, Jens O. Riede, and Philip H. Warren. 2008. <span>“Size, Foraging, and Food Web Structure.”</span> <em>Proceedings of the National Academy of Sciences</em> 105 (11): 4191–96. <a href="https://doi.org/10.1073/pnas.0710672105">https://doi.org/10.1073/pnas.0710672105</a>.
</div>
<div id="ref-petcheyFitEfficiencyBiology2011" class="csl-entry" role="listitem">
———. 2011. <span>“Fit, Efficiency, and Biology: <span>Some</span> Thoughts on Judging Food Web Models.”</span> <em>Journal of Theoretical Biology</em> 279 (1): 169–71. <a href="https://doi.org/10.1016/j.jtbi.2011.03.019">https://doi.org/10.1016/j.jtbi.2011.03.019</a>.
</div>
<div id="ref-pichlerMachineLearningAlgorithms2020" class="csl-entry" role="listitem">
Pichler, Maximilian, Virginie Boreux, Alexandra-Maria Klein, Matthias Schleuning, and Florian Hartig. 2020. <span>“Machine Learning Algorithms to Infer Trait-Matching and Predict Species Interactions in Ecological Networks.”</span> <em>Methods in Ecology and Evolution</em> 11 (2): 281–93. <a href="https://doi.org/10.1111/2041-210X.13329">https://doi.org/10.1111/2041-210X.13329</a>.
</div>
<div id="ref-pilosofMultilayerNatureEcological2017" class="csl-entry" role="listitem">
Pilosof, Shai, Mason A. Porter, Mercedes Pascual, and Sonia Kéfi. 2017. <span>“The Multilayer Nature of Ecological Networks.”</span> <em>Nature Ecology &amp; Evolution</em> 1 (4): 101. <a href="https://doi.org/10.1038/s41559-017-0101">https://doi.org/10.1038/s41559-017-0101</a>.
</div>
<div id="ref-poelenGlobalBioticInteractions2014" class="csl-entry" role="listitem">
Poelen, Jorrit H., James D. Simons, and Chris J. Mungall. 2014. <span>“Global Biotic Interactions: <span>An</span> Open Infrastructure to Share and Analyze Species-Interaction Datasets.”</span> <em>Ecological Informatics</em> 24 (November): 148–59. <a href="https://doi.org/10.1016/j.ecoinf.2014.08.005">https://doi.org/10.1016/j.ecoinf.2014.08.005</a>.
</div>
<div id="ref-poisotGuidelinesPredictionSpecies2023" class="csl-entry" role="listitem">
Poisot, Timothée. 2023. <span>“Guidelines for the Prediction of Species Interactions Through Binary Classification.”</span> <em>Methods in Ecology and Evolution</em> 14 (5): 1333–45. <a href="https://doi.org/10.1111/2041-210X.14071">https://doi.org/10.1111/2041-210X.14071</a>.
</div>
<div id="ref-poisotMangalMakingEcological2016" class="csl-entry" role="listitem">
Poisot, Timothée, Benjamin Baiser, Jennifer Dunne, Sonia Kéfi, François Massol, Nicolas Mouquet, Tamara N. Romanuk, Daniel B. Stouffer, Spencer A. Wood, and Dominique Gravel. 2016. <span>“Mangal – Making Ecological Network Analysis Simple.”</span> <em>Ecography</em> 39 (4): 384–90. <a href="https://doi.org/10.1111/ecog.00976">https://doi.org/10.1111/ecog.00976</a>.
</div>
<div id="ref-poisotGlobalKnowledgeGaps2021" class="csl-entry" role="listitem">
Poisot, Timothée, Gabriel Bergeron, Kevin Cazelles, Tad Dallas, Dominique Gravel, Andrew MacDonald, Benjamin Mercier, Clément Violet, and Steve Vissault. 2021. <span>“Global Knowledge Gaps in Species Interaction Networks Data.”</span> <em>Journal of Biogeography</em> 48 (7): 1552–63. <a href="https://doi.org/10.1111/jbi.14127">https://doi.org/10.1111/jbi.14127</a>.
</div>
<div id="ref-poisotStructureProbabilisticNetworks2016" class="csl-entry" role="listitem">
Poisot, Timothée, Alyssa Cirtwill, Kévin Cazelles, Dominique Gravel, Marie-Josée Fortin, and Daniel Stouffer. 2016. <span>“The Structure of Probabilistic Networks.”</span> <em>Methods in Ecology and Evolution</em> 7 (3): 303–12. <a href="https://doi.org/10">https://doi.org/10</a>.
</div>
<div id="ref-poisotSpeciesWhyEcological2015" class="csl-entry" role="listitem">
Poisot, Timothée, Daniel B. Stouffer, and Dominique Gravel. 2015. <span>“Beyond Species: Why Ecological Interaction Networks Vary Through Space and Time.”</span> <em>Oikos</em> 124 (3): 243–51. <a href="https://doi.org/10.1111/oik.01719">https://doi.org/10.1111/oik.01719</a>.
</div>
<div id="ref-poisotDescribeUnderstandPredict2016" class="csl-entry" role="listitem">
Poisot, Timothée, Daniel B. Stouffer, and Sonia Kéfi. 2016. <span>“Describe, Understand and Predict: Why Do We Need Networks in Ecology?”</span> <em>Functional Ecology</em> 30 (12): 1878–82. <a href="https://www.jstor.org/stable/48582345">https://www.jstor.org/stable/48582345</a>.
</div>
<div id="ref-pollockUnderstandingCooccurrenceModelling2014" class="csl-entry" role="listitem">
Pollock, Laura J., Reid Tingley, William K. Morris, Nick Golding, Robert B. O’Hara, Kirsten M. Parris, Peter A. Vesk, and Michael A. McCarthy. 2014. <span>“Understanding Co-Occurrence by Modelling Species Simultaneously with a <span>Joint Species Distribution Model</span> (<span>JSDM</span>).”</span> <em>Methods in Ecology and Evolution</em> 5 (5): 397–406. <a href="https://doi.org/10.1111/2041-210X.12180">https://doi.org/10.1111/2041-210X.12180</a>.
</div>
<div id="ref-pringleUntanglingFoodWebs2020" class="csl-entry" role="listitem">
Pringle, Robert M. 2020. <span>“Untangling <span>Food Webs</span>.”</span> In <em>Unsolved <span>Problems</span> in <span>Ecology</span></em>, 225–38. Princeton University Press. <a href="https://doi.org/10.1515/9780691195322-020">https://doi.org/10.1515/9780691195322-020</a>.
</div>
<div id="ref-pringleResolvingFoodWebStructure2020" class="csl-entry" role="listitem">
Pringle, Robert M., and Matthew C. Hutchinson. 2020. <span>“Resolving <span>Food-Web Structure</span>.”</span> <em>Annual Review of Ecology, Evolution and Systematics</em> 51 (Volume 51, 2020): 55–80. <a href="https://doi.org/10.1146/annurev-ecolsys-110218-024908">https://doi.org/10.1146/annurev-ecolsys-110218-024908</a>.
</div>
<div id="ref-proulxNetworkThinkingEcology2005" class="csl-entry" role="listitem">
Proulx, Stephen R., Daniel E. L. Promislow, and Patrick C. Phillips. 2005. <span>“Network Thinking in Ecology and Evolution.”</span> <em>Trends in Ecology &amp; Evolution</em> 20 (6): 345–53. <a href="https://doi.org/10.1016/j.tree.2005.04.004">https://doi.org/10.1016/j.tree.2005.04.004</a>.
</div>
<div id="ref-rooneyLandscapeTheoryFood2008" class="csl-entry" role="listitem">
Rooney, Neil, Kevin S. McCann, and John C. Moore. 2008. <span>“A Landscape Theory for Food Web Architecture.”</span> <em>Ecology Letters</em> 11 (8): 867–81. <a href="https://doi.org/10.1111/j.1461-0248.2008.01193.x">https://doi.org/10.1111/j.1461-0248.2008.01193.x</a>.
</div>
<div id="ref-roopnarineExtinctionCascadesCatastrophe2006" class="csl-entry" role="listitem">
Roopnarine, Peter D. 2006. <span>“Extinction <span>Cascades</span> and <span>Catastrophe</span> in <span>Ancient Food Webs</span>.”</span> <em>Paleobiology</em> 32 (1): 1–19. <a href="https://www.jstor.org/stable/4096814">https://www.jstor.org/stable/4096814</a>.
</div>
<div id="ref-saraviaEcologicalNetworkAssembly2022" class="csl-entry" role="listitem">
Saravia, Leonardo A., Tomás I Marina, Nadiah P. Kristensen, Marleen De Troch, and Fernando R. Momo. 2022. <span>“Ecological Network Assembly: <span>How</span> the Regional Metaweb Influences Local Food Webs.”</span> <em>Journal of Animal Ecology</em> 91 (3): 630–42. <a href="https://doi.org/10.1111/1365-2656.13652">https://doi.org/10.1111/1365-2656.13652</a>.
</div>
<div id="ref-segarRoleEvolutionShaping2020" class="csl-entry" role="listitem">
Segar, Simon T., Tom M. Fayle, Diane S. Srivastava, Thomas M. Lewinsohn, Owen T. Lewis, Vojtech Novotny, Roger L. Kitching, and Sarah C. Maunsell. 2020. <span>“The <span>Role</span> of <span>Evolution</span> in <span>Shaping Ecological Networks</span>.”</span> <em>Trends in Ecology &amp; Evolution</em> 35 (5): 454–66. <a href="https://doi.org/10.1016/j.tree.2020.01.004">https://doi.org/10.1016/j.tree.2020.01.004</a>.
</div>
<div id="ref-shawFrameworkReconstructingAncient2024" class="csl-entry" role="listitem">
Shaw, Jack O., Alexander M. Dunhill, Andrew P. Beckerman, Jennifer A. Dunne, and Pincelli M. Hull. 2024. <span>“A Framework for Reconstructing Ancient Food Webs Using Functional Trait Data.”</span> bioRxiv. <a href="https://doi.org/10.1101/2024.01.30.578036">https://doi.org/10.1101/2024.01.30.578036</a>.
</div>
<div id="ref-songRigorousValidationEcological2024" class="csl-entry" role="listitem">
Song, Chuliang, and Jonathan M. Levine. 2024. <span>“Rigorous (in)validation of Ecological Models.”</span> bioRxiv. <a href="https://doi.org/10.1101/2024.09.19.613075">https://doi.org/10.1101/2024.09.19.613075</a>.
</div>
<div id="ref-stockPairwiseLearningPredicting2021" class="csl-entry" role="listitem">
Stock, Michiel. 2021. <span>“Pairwise Learning for Predicting Pollination Interactions Based on Traits and Phylogeny.”</span> <em>Ecological Modelling</em>, 14.
</div>
<div id="ref-stoufferAllEcologicalModels2019" class="csl-entry" role="listitem">
Stouffer, Daniel B. 2019. <span>“All Ecological Models Are Wrong, but Some Are Useful.”</span> <em>Journal of Animal Ecology</em> 88 (2): 192–95. <a href="https://doi.org/10.1111/1365-2656.12949">https://doi.org/10.1111/1365-2656.12949</a>.
</div>
<div id="ref-strydomFoodWebReconstruction2022" class="csl-entry" role="listitem">
Strydom, Tanya, Salomé Bouskila, Francis Banville, Ceres Barros, Dominique Caron, Maxwell J. Farrell, Marie-Josée Fortin, et al. 2022. <span>“Food Web Reconstruction Through Phylogenetic Transfer of Low-Rank Network Representation.”</span> <em>Methods in Ecology and Evolution</em> 13 (12): 2838–49. <a href="https://doi.org/10.1111/2041-210X.13835">https://doi.org/10.1111/2041-210X.13835</a>.
</div>
<div id="ref-strydomGraphEmbeddingTransfer2023" class="csl-entry" role="listitem">
Strydom, Tanya, Salomé Bouskila, Francis Banville, Ceres Barros, Dominique Caron, Maxwell J. Farrell, Marie-Josée Fortin, et al. 2023. <span>“Graph Embedding and Transfer Learning Can Help Predict Potential Species Interaction Networks Despite Data Limitations.”</span> <em>Methods in Ecology and Evolution</em> 14 (12): 2917–30. <a href="https://doi.org/10.1111/2041-210X.14228">https://doi.org/10.1111/2041-210X.14228</a>.
</div>
<div id="ref-strydomRoadmapPredictingSpecies2021" class="csl-entry" role="listitem">
Strydom, Tanya, Michael D. Catchen, Francis Banville, Dominique Caron, Gabriel Dansereau, Philippe Desjardins-Proulx, Norma R. Forero-Muñoz, et al. 2021. <span>“A Roadmap Towards Predicting Species Interaction Networks (Across Space and Time).”</span> <em>Philosophical Transactions of the Royal Society B: Biological Sciences</em> 376 (1837): 20210063. <a href="https://doi.org/10.1098/rstb.2021.0063">https://doi.org/10.1098/rstb.2021.0063</a>.
</div>
<div id="ref-strydomSVDEntropyReveals2021" class="csl-entry" role="listitem">
Strydom, Tanya, Giulio V. Dalla Riva, and Timothée Poisot. 2021. <span>“<span>SVD Entropy Reveals</span> the <span>High Complexity</span> of <span>Ecological Networks</span>.”</span> <em>Frontiers in Ecology and Evolution</em> 9. <a href="https://doi.org/10.3389/fevo.2021.623141">https://doi.org/10.3389/fevo.2021.623141</a>.
</div>
<div id="ref-terryFindingMissingLinks2020" class="csl-entry" role="listitem">
Terry, J. Christopher D., and Owen T. Lewis. 2020. <span>“Finding Missing Links in Interaction Networks.”</span> <em>Ecology</em> 101 (7): e03047. <a href="https://doi.org/10.1002/ecy.3047">https://doi.org/10.1002/ecy.3047</a>.
</div>
<div id="ref-valdovinosBioenergeticFrameworkAboveground2023" class="csl-entry" role="listitem">
Valdovinos, Fernanda S., Kayla R. S. Hale, Sabine Dritz, Paul R. Glaum, Kevin S. McCann, Sophia M. Simon, Elisa Thébault, William C. Wetzel, Kate L. Wootton, and Justin D. Yeakel. 2023. <span>“A Bioenergetic Framework for Aboveground Terrestrial Food Webs.”</span> <em>Trends in Ecology &amp; Evolution</em> 38 (3): 301–12. <a href="https://doi.org/10.1016/j.tree.2022.11.004">https://doi.org/10.1016/j.tree.2022.11.004</a>.
</div>
<div id="ref-williamsSimpleRulesYield2000" class="csl-entry" role="listitem">
Williams, Richard J., and Neo D. Martinez. 2000. <span>“Simple Rules Yield Complex Food Webs.”</span> <em>Nature</em> 404 (6774): 180–83. <a href="https://doi.org/10.1038/35004572">https://doi.org/10.1038/35004572</a>.
</div>
<div id="ref-williamsSuccessItsLimits2008" class="csl-entry" role="listitem">
———. 2008. <span>“Success and Its Limits Among Structural Models of Complex Food Webs.”</span> <em>Journal of Animal Ecology</em> 77 (3): 512–19. <a href="https://doi.org/10.1111/j.1365-2656.2008.01362.x">https://doi.org/10.1111/j.1365-2656.2008.01362.x</a>.
</div>
<div id="ref-woottonModularTheoryTrophic2023" class="csl-entry" role="listitem">
Wootton, Kate L., Alva Curtsdotter, Tomas Roslin, Riccardo Bommarco, and Tomas Jonsson. 2023. <span>“Towards a Modular Theory of Trophic Interactions.”</span> <em>Functional Ecology</em> 37 (1): 26–43. <a href="https://doi.org/10.1111/1365-2435.13954">https://doi.org/10.1111/1365-2435.13954</a>.
</div>
<div id="ref-xieCompletenessCommunityStructure2017" class="csl-entry" role="listitem">
Xie, Jia-Rong, Pan Zhang, Hai-Feng Zhang, and Bing-Hong Wang. 2017. <span>“Completeness of <span>Community Structure</span> in <span>Networks</span>.”</span> <em>Scientific Reports</em> 7 (1): 5269. <a href="https://doi.org/10.1038/s41598-017-05585-6">https://doi.org/10.1038/s41598-017-05585-6</a>.
</div>
<div id="ref-yeakelCollapseEcologicalNetwork2014" class="csl-entry" role="listitem">
Yeakel, Justin D., Mathias M. Pires, Lars Rudolf, Nathaniel J. Dominy, Paul L. Koch, Paulo R. Guimarães, and Thilo Gross. 2014. <span>“Collapse of an Ecological Network in <span>Ancient Egypt</span>.”</span> <em>PNAS</em> 111 (40): 14472–77. <a href="https://doi.org/10.1073/pnas.1408471111">https://doi.org/10.1073/pnas.1408471111</a>.
</div>
<div id="ref-yodzisCompartmentationRealAssembled1982" class="csl-entry" role="listitem">
Yodzis, P. 1982. <span>“The <span>Compartmentation</span> of <span>Real</span> and <span>Assembled Ecosystems</span>.”</span> <em>The American Naturalist</em> 120 (5): 551–70. <a href="https://doi.org/10.1086/284013">https://doi.org/10.1086/284013</a>.
</div>
<div id="ref-yodzisBodySizeConsumerResource1992" class="csl-entry" role="listitem">
Yodzis, P., and S. Innes. 1992. <span>“Body <span>Size</span> and <span>Consumer-Resource Dynamics</span>.”</span> <em>The American Naturalist</em> 139 (6): 1151–75. <a href="https://doi.org/10.1086/285380">https://doi.org/10.1086/285380</a>.
</div>
</div>
<div class="hidden" aria-hidden="true">
<span class="glightbox-desc lightbox-desc-1">Figure&nbsp;1: TODO.</span>
</div>
</section>

<div id="quarto-appendix" class="default"><section class="quarto-appendix-contents" id="quarto-citation"><h2 class="anchored quarto-appendix-heading">Citation</h2><div><div class="quarto-appendix-secondary-label">BibTeX citation:</div><pre class="sourceCode code-with-copy quarto-appendix-bibtex"><code class="sourceCode bibtex">@online{strydom2024,
  author = {Strydom, Tanya and A. Dunne, Jennifer and Poisot, Timothée
    and P. Beckerman, Andrew},
  title = {Unveiling the {Complexity} of {Food} {Webs:} {A}
    {Comprehensive} {Overview} of {Definitions,} {Scales,} and
    {Mechanisms}},
  date = {2024-09-27},
  langid = {en},
  abstract = {Food webs are a useful abstraction and representation of
    the feeding links between species in a community and are used to
    infer many ecosystem level processes. However, the different
    theories, mechanisms, and criteria that underpin how a food web is
    defined and, ultimately, constructed means that not all food webs
    are representing the same ecological process. Here we present a
    synthesis of the different assumptions, scales and mechanisms that
    are used to define different ecological networks ranging from
    metawebs (an inventory of all potential interactions) to fully
    realised networks (interactions that occur within a given community
    over a certain timescale). Illuminating the assumptions, scales, and
    mechanisms of network inference allows a formal categorisation of
    how to use networks to answer key ecological and conservation
    questions and defines guidelines to prevent unintentional misuse or
    misinterpretation.}
}
</code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre><div class="quarto-appendix-secondary-label">For attribution, please cite this work as:</div><div id="ref-strydom2024" class="csl-entry quarto-appendix-citeas" role="listitem">
Strydom, Tanya, Jennifer A. Dunne, Timothée Poisot, and Andrew P.
Beckerman. 2024. <span>“Unveiling the Complexity of Food Webs: A
Comprehensive Overview of Definitions, Scales, and Mechanisms.”</span>
TREE (One Can Dream...). September 27, 2024.
</div></div></section></div></main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->
<script>var lightboxQuarto = GLightbox({"openEffect":"zoom","selector":".lightbox","closeEffect":"zoom","descPosition":"bottom","loop":false});
window.onload = () => {
  lightboxQuarto.on('slide_before_load', (data) => {
    const { slideIndex, slideNode, slideConfig, player, trigger } = data;
    const href = trigger.getAttribute('href');
    if (href !== null) {
      const imgEl = window.document.querySelector(`a[href="${href}"] img`);
      if (imgEl !== null) {
        const srcAttr = imgEl.getAttribute("src");
        if (srcAttr && srcAttr.startsWith("data:")) {
          slideConfig.href = srcAttr;
        }
      }
    } 
  });

  lightboxQuarto.on('slide_after_load', (data) => {
    const { slideIndex, slideNode, slideConfig, player, trigger } = data;
    if (window.Quarto?.typesetMath) {
      window.Quarto.typesetMath(slideNode);
    }
  });

};
          </script>




</body></html>