<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.4.549">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Tanya Strydom">
<meta name="author" content="Andrew P. Beckerman">
<meta name="dcterms.date" content="2024-03-18">
<meta name="keywords" content="food web, network construction">

<title>T is for Topology</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for citations */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
  margin-bottom: 0em;
}
.hanging-indent div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}</style>


<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script src="site_libs/quarto-contrib/glightbox/glightbox.min.js"></script>
<link href="site_libs/quarto-contrib/glightbox/glightbox.min.css" rel="stylesheet">
<link href="site_libs/quarto-contrib/glightbox/lightbox.css" rel="stylesheet">
<script async="" src="https://hypothes.is/embed.js"></script>
<script>
  window.document.addEventListener("DOMContentLoaded", function (_event) {
    document.body.classList.add('hypothesis-enabled');
  });
</script>


<meta name="citation_title" content="T is for Topology">
<meta name="citation_abstract" content="There are many reasons one might want to generate a network and there are many tools on the market that might make that possible. However not all tools are created equally and there is reason to assume that not all networks will suit most purposes. Here the aim is to compare and contrast the different topology generating tools that are on the market and see where they shine and where they fall flat. There probably isn&amp;amp;#039;t one model to rule them all but it doesn't mean that we shouldn't be critical when we think about the model we want to use.
">
<meta name="citation_keywords" content="food web,network construction">
<meta name="citation_author" content="Tanya Strydom">
<meta name="citation_author" content="Andrew P. Beckerman">
<meta name="citation_publication_date" content="2024-03-18">
<meta name="citation_cover_date" content="2024-03-18">
<meta name="citation_year" content="2024">
<meta name="citation_online_date" content="2024-03-18">
<meta name="citation_language" content="en">
<meta name="citation_journal_title" content="Some fancy journal">
<meta name="citation_reference" content="citation_title=A General Model for Food Web Structure;,citation_abstract=A central problem in ecology is determining the processes that shape the complex networks known as food webs formed by species and their feeding relationships. The topology of these networks is a major determinant of ecosystems’ dynamics and is ultimately responsible for their responses to human impacts. Several simple models have been proposed for the intricate food webs observed in nature. We show that the three main models proposed so far fail to fully replicate the empirical data, and we develop a likelihood-based approach for the direct comparison of alternative models based on the full structure of the network. Results drive a new model that is able to generate all the empirical data sets and to do so with the highest likelihood.;,citation_author=Stefano Allesina;,citation_author=David Alonso;,citation_author=Mercedes Pascual;,citation_publication_date=2008-05;,citation_cover_date=2008-05;,citation_year=2008;,citation_issue=5876;,citation_doi=10.1126/science.1156269;,citation_volume=320;,citation_journal_title=Science;,citation_publisher=American Association for the Advancement of Science;">
<meta name="citation_reference" content="citation_title=Autecology and the Filling of Ecospace: Key Metazoan Radiations;,citation_abstract=Abstract: All possible combinations of six tiering positions in relation to the substratum/water interface, six motility levels and six feeding strategies define a complete theoretical ecospace of 216 potential modes of life for marine animals. The number of modes of life actually utilized specifies realized ecospace. Owing to constraints of effectiveness and efficiency the modern marine fauna utilizes only about half the potential number of modes of life, two-thirds of which (62 of 92) are utilized by animals with readily preserved, mineralized hard parts. Realized ecospace has increased markedly since the early evolution of animal ecosystems. The Ediacaran fauna utilized at most 12 modes of life, with just two practised by skeletal organisms. A total of 30 modes of life are recorded in the Early and Middle Cambrian, 19 of which were utilized by skeletal organisms. The other 11 are documented from soft-bodied animals preserved in the Chengjiang and Burgess Shale Konservat-Lagerstätten. The number of modes of life utilized by skeletal organisms increased by more than 50 per cent during the Ordovician radiation to a Late Ordovician total of 30. Between the Late Ordovician and the Recent the number of utilized modes of life has doubled again. The autecological and taxonomic diversity histories of the marine metazoa appear to be broadly parallel, and future studies of theoretical ecospace utilization should provide more detailed tests of pattern and process in the ecological history of the metazoa.;,citation_author=Richard K. Bambach;,citation_author=Andrew M. Bush;,citation_author=Douglas H. Erwin;,citation_publication_date=2007;,citation_cover_date=2007;,citation_year=2007;,citation_issue=1;,citation_doi=10.1111/j.1475-4983.2006.00611.x;,citation_issn=1475-4983;,citation_volume=50;,citation_language=en-US;,citation_journal_title=Palaeontology;">
<meta name="citation_reference" content="citation_title=Mangal.jl and EcologicalNetworks.jl: Two complementary packages for analyzing ecological networks in Julia;,citation_abstract=Network ecology is an emerging field of study describing species interactions (e.g. predation, pollination) in a biological community. Ecological networks, in which two species are connected if they can interact, are a mathematical representation of all interactions encountered in a given ecosystem. Anchored in graph theory, the methods developed in network ecology are remarkably rigorous and biologically insightful. Indeed, many ecological and evolutionary processes are driven by species interactions and network structure (i.e. the arrangement of links in ecological networks). The study of ecological networks, from data importation and simulation to data analysis and visualization, requires a coherent and efficient set of numerical tools. With its powerful and dynamic type system, the Julia programming language provides a very good and fast computing environment suitable for ecological research. Julia’s typing system notably ensures that appropriate methods are used when a function is applied to different types of ecological data, which therefore minimizes the risk of committing errors of analysis and interpretation in network ecology. Mangal.jl and EcologicalNetworks.jl are two novel and complementary Julia packages designed to conduct research in network ecology efficiently. Researchers studying a broad range of ecological networks can read the data and metadata of numerous well-documented networks using Mangal.jl, and analyze their properties using EcologicalNetworks.jl.;,citation_author=Francis Banville;,citation_author=Steve Vissault;,citation_author=Timothée Poisot;,citation_publication_date=2021-05;,citation_cover_date=2021-05;,citation_year=2021;,citation_issue=61;,citation_doi=10.21105/joss.02721;,citation_issn=2475-9066;,citation_volume=6;,citation_language=en-US;,citation_journal_title=Journal of Open Source Software;">
<meta name="citation_reference" content="citation_title=What constrains food webs? A maximum entropy framework for predicting their structure with minimal biases;,citation_abstract=Food webs are complex ecological networks whose structure is both ecologically and statistically constrained, with many network properties being correlated with each other. Despite the recognition of these invariable relationships in food webs, the use of the principle of maximum entropy (MaxEnt) in network ecology is still rare. This is surprising considering that MaxEnt is a statistical tool precisely designed for understanding and predicting many types of constrained systems. This principle asserts that the least-biased probability distribution of a system’s property, constrained by prior knowledge about that system, is the one with maximum information entropy. MaxEnt has been proven useful in many ecological modeling problems, but its application in food webs and other ecological networks is limited. Here we show how MaxEnt can be used to derive many food-web properties both analytically and heuristically. First, we show how the joint degree distribution (the joint probability distribution of the numbers of prey and predators for each species in the network) can be derived analytically using the number of species and the number of interactions in food webs. Second, we present a heuristic and flexible approach of finding a network’s adjacency matrix (the network’s representation in matrix format) based on simulated annealing and SVD entropy. We built two heuristic models using the connectance and the joint degree sequence as statistical constraints, respectively. We compared both models’ predictions against corresponding null and neutral models commonly used in network ecology using open access data of terrestrial and aquatic food webs sampled globally (N = 257). We found that the heuristic model constrained by the joint degree sequence was a good predictor of many measures of food-web structure, especially the nestedness and motifs distribution. Specifically, our results suggest that the structure of terrestrial and aquatic food webs is mainly driven by their joint degree distribution.;,citation_author=Francis Banville;,citation_author=Dominique Gravel;,citation_author=Timothée Poisot;,citation_publication_date=2023-09;,citation_cover_date=2023-09;,citation_year=2023;,citation_issue=9;,citation_doi=10.1371/journal.pcbi.1011458;,citation_issn=1553-7358;,citation_volume=19;,citation_language=en-US;,citation_journal_title=PLOS Computational Biology;,citation_publisher=Public Library of Science;">
<meta name="citation_reference" content="citation_title=A common framework for identifying linkage rules across different types of interactions;,citation_abstract=Species interactions, ranging from antagonisms to mutualisms, form the architecture of biodiversity and determine ecosystem functioning. Understanding the rules responsible for who interacts with whom, as well as the functional consequences of these interspecific interactions, is central to predict community dynamics and stability. Species traits sensu lato may affect different ecological processes by determining species interactions through a two-step process. First, ecological and life-history traits govern species distributions and abundance, and hence determine species co-occurrence and the potential for species to interact. Secondly, morphological or physiological traits between co-occurring potential interaction partners should match for the realization of an interaction. Here, we review recent advances on predicting interactions from species co-occurrence and develop a probabilistic model for inferring trait matching. The models proposed here integrate both neutral and trait-matching constraints, while using only information about known interactions, thereby overcoming problems originating from undersampling of rare interactions (i.e. missing links). They can easily accommodate qualitative or quantitative data and can incorporate trait variation within species, such as values that vary along developmental stages or environmental gradients. We use three case studies to show that the proposed models can detect strong trait matching (e.g. predator–prey system), relaxed trait matching (e.g. herbivore–plant system) and barrier trait matching (e.g. plant–pollinator systems). Only by elucidating which species traits are important in each process (i.e. in determining interaction establishment and frequency), we can advance in explaining how species interact and the consequences of these interactions for ecosystem functioning. A lay summary is available for this article.;,citation_author=Ignasi Bartomeus;,citation_author=Dominique Gravel;,citation_author=Jason M. Tylianakis;,citation_author=Marcelo A. Aizen;,citation_author=Ian A. Dickie;,citation_author=Maud Bernard-Verdier;,citation_publication_date=2016;,citation_cover_date=2016;,citation_year=2016;,citation_issue=12;,citation_doi=10.1111/1365-2435.12666;,citation_issn=1365-2435;,citation_volume=30;,citation_language=en-US;,citation_journal_title=Functional Ecology;">
<meta name="citation_reference" content="citation_title=The nested assembly of plant-animal mutualistic networks;,citation_abstract=Most studies of plant–animal mutualisms involve a small number of species. There is almost no information on the structural organization of species-rich mutualistic networks despite its potential importance for the maintenance of diversity. Here we analyze 52 mutualistic networks and show that they are highly nested; that is, the more specialist species interact only with proper subsets of those species interacting with the more generalists. This assembly pattern generates highly asymmetrical interactions and organizes the community cohesively around a central core of interactions. Thus, mutualistic networks are neither randomly assembled nor organized in compartments arising from tight, parallel specialization. Furthermore, nestedness increases with the complexity (number of interactions) of the network: for a given number of species, communities with more interactions are significantly more nested. Our results indicate a nonrandom pattern of community organization that may be relevant for our understanding of the organization and persistence of biodiversity.;,citation_author=J. Bascompte;,citation_author=P. Jordano;,citation_author=C. J. Melian;,citation_author=J. M. Olesen;,citation_publication_date=2003-08;,citation_cover_date=2003-08;,citation_year=2003;,citation_issue=16;,citation_doi=10.1073/pnas.1633576100;,citation_issn=0027-8424, 1091-6490;,citation_volume=100;,citation_language=en-US;,citation_journal_title=Proceedings of the National Academy of Sciences;">
<meta name="citation_reference" content="citation_title=Foraging biology predicts food web complexity;,citation_abstract=Food webs, the networks of feeding links between species, are central to our understanding of ecosystem structure, stability, and function. One of the key aspects of food web structure is complexity, or connectance, the number of links expressed as a proportion of the total possible number of links. Connectance (complexity) is linked to the stability of webs and is a key parameter in recent models of other aspects of web structure. However, there is still no fundamental biological explanation for connectance in food webs. Here, we propose that constraints on diet breadth, driven by optimal foraging, provide such an explanation. We show that a simple diet breadth model predicts highly constrained values of connectance as an emergent consequence of individual foraging behavior. When combined with features of real food web data, such as taxonomic and trophic aggregation and cumulative sampling of diets, the model predicts well the levels of connectance and scaling of connectance with species richness, seen in real food webs. This result is a previously undescribed synthesis of foraging theory and food web theory, in which network properties emerge from the behavior of individuals and, as such, provides a mechanistic explanation of connectance currently lacking in food web models.;,citation_author=Andrew P. Beckerman;,citation_author=Owen L. Petchey;,citation_author=Philip H. Warren;,citation_publication_date=2006-09;,citation_cover_date=2006-09;,citation_year=2006;,citation_issue=37;,citation_doi=10.1073/pnas.0603039103;,citation_issn=0027-8424, 1091-6490;,citation_volume=103;,citation_language=en-US;,citation_journal_title=Proceedings of the National Academy of Sciences;">
<meta name="citation_reference" content="citation_title=Optimising predictive models to prioritise viral discovery in zoonotic reservoirs;,citation_author=Daniel J. Becker;,citation_author=Gregory F. Albery;,citation_author=Anna R. Sjodin;,citation_author=Timothée Poisot;,citation_author=Laura M. Bergner;,citation_author=Binqi Chen;,citation_author=Lily E. Cohen;,citation_author=Tad A. Dallas;,citation_author=Evan A. Eskew;,citation_author=Anna C. Fagre;,citation_author=Maxwell J. Farrell;,citation_author=Sarah Guth;,citation_author=Barbara A. Han;,citation_author=Nancy B. Simmons;,citation_author=Michiel Stock;,citation_author=Emma C. Teeling;,citation_author=Colin J. Carlson;,citation_publication_date=2022-08;,citation_cover_date=2022-08;,citation_year=2022;,citation_issue=8;,citation_doi=10.1016/S2666-5247(21)00245-7;,citation_issn=2666-5247;,citation_pmid=35036970;,citation_volume=3;,citation_language=en-US;,citation_journal_title=The Lancet Microbe;,citation_publisher=Elsevier;">
<meta name="citation_reference" content="citation_title=The “Goldilocks factor” in food webs;,citation_author=Eric L. Berlow;,citation_author=Ulrich Brose;,citation_author=Neo D. Martinez;,citation_publication_date=2008-03;,citation_cover_date=2008-03;,citation_year=2008;,citation_issue=11;,citation_doi=10.1073/pnas.0800967105;,citation_volume=105;,citation_journal_title=Proceedings of the National Academy of Sciences;,citation_publisher=Proceedings of the National Academy of Sciences;">
<meta name="citation_reference" content="citation_title=Addressing the Eltonian shortfall with trait-based interaction models;,citation_abstract=We have very limited knowledge of how species interact in most communities and ecosystems despite trophic relationships being fundamental for linking biodiversity to ecosystem functioning. A promising approach to fill this gap is to predict interactions based on functional traits, but many questions remain about how well we can predict interactions for different taxa, ecosystems and amounts of input data. Here, we built a new traits-based model of trophic interactions for European vertebrates and found that even models calibrated with 0.1% of the interactions (100 out of 71 k) estimated the full European vertebrate food web reasonably well. However, predators were easier to predict than prey, especially for some clades (e.g. fowl and storks) and local food web connectance was consistently overestimated. Our results demonstrate the ability to rapidly generate food webs when empirical data are lacking—an important step towards a more complete and spatially explicit description of food webs.;,citation_author=Dominique Caron;,citation_author=Luigi Maiorano;,citation_author=Wilfried Thuiller;,citation_author=Laura J. Pollock;,citation_publication_date=2022;,citation_cover_date=2022;,citation_year=2022;,citation_issue=4;,citation_doi=10.1111/ele.13966;,citation_issn=1461-0248;,citation_volume=25;,citation_language=en-US;,citation_journal_title=Ecology Letters;">
<meta name="citation_reference" content="citation_title=Trophic interaction models predict interactions across space, not food webs;,citation_abstract=Aim: Trophic interactions are central to our understanding of essential ecosystem functions as well as their stability. Predicting these interactions has become increasingly common due to the lack of empirical data on trophic interactions for most taxa in most ecosystems. We aim to determine whether and how accurately we can extrapolate to new communities both in terms of pairwise predator-prey interactions and higher level food web attributes (i.e., species position, food web-level properties). Location: Canada, Europe, Tanzania. Time period: Current. Major taxa studied: Terrestrial vertebrates Methods: We use a trait-based model of pairwise trophic interactions, calibrated independently on four different terrestrial vertebrate food webs (Canadian tundra, Serengeti, alpine south-eastern Pyrenees, and entire Europe) and assess the ability of each calibrated instance of the model to predict alternative food webs. We test how well predictions recover individual predator-prey interactions as well as higher level food web properties across geographical locations. Results: We find that, given enough phylogenetic and environmental similarities between food webs, trait-based models predict most interactions and their absence correctly (AUC &amp;amp;amp;gt; 0.82), even across highly contrasting environments. However, network metrics were less well-predicted than single interactions by our models. Predicted food webs were more connected, less modular, and had higher mean trophic levels than observed. Main conclusions: Theory predicts that the variability observed in food webs can be explained by differences in trait distributions and trait-matching relationships. Trait-based models can predict potential interactions amongst species in an ecosystem when calibrated using food web data from reasonably similar ecosystems. This suggests that food webs vary spatially primarily through changes in trait distributions. These models, however, are less good at predicting system level food web properties. We thus highlight the need for methodological advances to simultaneously address trophic interactions and the structure of food webs across time and space.;,citation_author=Dominique Caron;,citation_author=Ulrich Brose;,citation_author=Miguel Lurgi;,citation_author=Guillaume Blanchet;,citation_author=Dominique Gravel;,citation_author=Laura J. Pollock;,citation_publication_date=2023-05;,citation_cover_date=2023-05;,citation_year=2023;,citation_doi=10.32942/X29K55;,citation_language=en-US;,citation_publisher=EcoEvoRxiv;">
<meta name="citation_reference" content="citation_title=Phylogenetic constraints and adaptation explain food-web structure;,citation_abstract=Food webs are descriptions of who eats whom in an ecosystem. Although extremely complex and variable, their structure possesses basic regularities1,2,3,4,5,6. A fascinating question is to find a simple model capturing the underlying processes behind these repeatable patterns. Until now, two models have been devised for the description of trophic interactions within a natural community7,8. Both are essentially based on the concept of ecological niche, with the consumers organized along a single niche dimension; for example, prey size8,9. Unfortunately, they fail to describe adequately recent and high-quality data. Here, we propose a new model built on the hypothesis that any species’ diet is the consequence of phylogenetic constraints and adaptation. Simple rules incorporating both concepts yield food webs whose structure is very close to real data. Consumers are organized in groups forming a nested hierarchy, which better reflects the complexity and multidimensionality of most natural systems.;,citation_author=Marie-France Cattin;,citation_author=Louis-Félix Bersier;,citation_author=Carolin Banašek-Richter;,citation_author=Richard Baltensperger;,citation_author=Jean-Pierre Gabriel;,citation_publication_date=2004-02;,citation_cover_date=2004-02;,citation_year=2004;,citation_issue=6977;,citation_doi=10.1038/nature02327;,citation_issn=1476-4687;,citation_volume=427;,citation_language=en-US;,citation_journal_title=Nature;,citation_publisher=Nature Publishing Group;">
<meta name="citation_reference" content="citation_title=Transient dynamics and food–web complexity in the Lotka–Volterra cascade model;,citation_abstract=How does the long–term behaviour near equilibrium of model food webs correlate with their short–term transient dynamics? Here, simulations of the Lotka–Volterra cascade model of food webs provide the first evidence to answer this question. Transient behaviour is measured by resilience, reactivity, the maximum amplification of a perturbation and the time at which the maximum amplification occurs. Model food webs with a higher probability of local asymptotic stability may be less resilient and may have a larger transient growth of perturbations. Given a fixed connectance, the sizes and durations of transient responses to perturbations increase with the number of species. Given a fixed number of species, as connectance increases, the sizes and durations of transient responses to perturbations may increase or decrease depending on the type of link that is varied. Reactivity is more sensitive to changes in the number of donor–controlled links than to changes in the number of recipient–controlled links, while resilience is more sensitive to changes in the number of recipient–controlled links than to changes in the number of donor–controlled links. Transient behaviour is likely to be one of the important factors affecting the persistence of ecological communities.;,citation_author=X. Chen;,citation_author=J. E. Cohen;,citation_publication_date=2001-04;,citation_cover_date=2001-04;,citation_year=2001;,citation_issue=1469;,citation_doi=10.1098/rspb.2001.1596;,citation_volume=268;,citation_journal_title=Proceedings of the Royal Society of London. Series B: Biological Sciences;,citation_publisher=Royal Society;">
<meta name="citation_reference" content="citation_title=Community Food Webs: Data and Theory;,citation_abstract=Food webs hold a central place in ecology. They describe which organisms feed on which others in natural habitats. This book describes recently discovered empirical regularities in real food webs: it proposes a novel theory unifying many of these regularities, as well as extensive empirical data. After a general introduction, reviewing the empirical and theoretical discoveries about food webs, the second portion of the book shows that community food webs obey several striking phenomenological regularities. Some of these unify, regardless of habitat. Others differentiate, showing that habitat significantly influences structure. The third portion of the book presents a theoretical analysis of some of the unifying empirical regularities. The fourth portion of the book presents 13 community food webs. Collected from scattered sources and carefully edited, they are the empirical basis for the results in the volume. The largest available set of data on community food webs provides a valuable foundation for future studies of community food webs. The book is intended for graduate students, teachers and researchers primarily in ecology. The theoretical portions of the book provide materials useful to teachers of applied combinatorics, in particular, random graphs. Researchers in random graphs will find here unsolved mathematical problems.;,citation_author=Joel E Cohen;,citation_author=Frederic Briand;,citation_author=Charles Newman;,citation_publication_date=1990;,citation_cover_date=1990;,citation_year=1990;,citation_isbn=978-3-642-83786-9;,citation_language=en-US;,citation_series_title=Biomathematics;">
<meta name="citation_reference" content="citation_title=Food webs and the dimensionality of trophic niche space;,citation_abstract=If the trophic niche of a kind of organism is a connected region in niche space, then it is possible for trophic niche overlaps to be described in a one-dimensional niche space if and only if the trophic niche overlap graph is an interval graph. An analysis of 30 food webs, using the combinatorial theory of interval graphs, suggests that a niche space of dimension 1 suffices, with unexpectedly high frequency and perhaps always, to describe the trophic niche overlaps implied by real food webs in single habitats. Consequently, real food webs fall in a small subset of the set of mathematically possible food webs. That real food webs are compatible with one-dimensional trophic niche spaces, more often than can be explained by chance alone, has not been noticed previously.;,citation_author=Joel E. Cohen;,citation_publication_date=1977-10;,citation_cover_date=1977-10;,citation_year=1977;,citation_issue=10;,citation_doi=10.1073/pnas.74.10.4533;,citation_volume=74;,citation_journal_title=Proceedings of the National Academy of Sciences;,citation_publisher=Proceedings of the National Academy of Sciences;">
<meta name="citation_reference" content="citation_title=A stochastic theory of community food webs I. Models and aggregated data;,citation_abstract=Three recently discovered quantitative empirical generalizations describe major features of the structure of community food webs. These generalizations are: (i) a species scaling law: the mean proportions of basal, intermediate and top species remain invariant at approximately 0.19, 0.53, and 0.29, respectively, over the range of variation in the number of species in a web; (ii) a link scaling law : the mean proportions of trophic links in the categories basal-intermediate, basal-top, intermediate–intermediate, and intermediate–top remain invariant at approximately 0.27, 0.08, 0.30 and 0.35, respectively, over the range of variation in the number of species in a web; and (iii) a link-species scaling law: the ratio of mean trophic links to species remains invariant at approximately 1.86, over the range of variation in the number of species in a web. This paper presents a model, the only successful one among several attempts, in which the first two of these empirical generalizations can be derived as a consequence of the third. The model assumes that species are ordered in a cascade or hierarchy such that a given species can prey on only those species below it and can be preyed on by only those species above it in the hierarchy.;,citation_author=Joel E. Cohen;,citation_author=C. M. Newman;,citation_author=John Hyslop Steele;,citation_publication_date=1985;,citation_cover_date=1985;,citation_year=1985;,citation_issue=1237;,citation_doi=10.1098/rspb.1985.0042;,citation_volume=224;,citation_journal_title=Proceedings of the Royal Society of London. Series B. Biological Sciences;,citation_publisher=Royal Society;">
<meta name="citation_reference" content="citation_title=A Model for Tropic Interaction;,citation_abstract=A nonlinear function general enough to include the effects of feeding saturation and intraspecific consumer interference is used to represent the transfer of material or energy from one trophic level to another. The function agrees with some recent experimental data on feeding rates. A model using this feeding rate function is subjected to equilibrium and stability analyses to ascertain its mathematical implications. The anaylses lead to several observations; for example, increases in maximum feeding rate may, under certain circumstances; result in decreases in consumer population and mutal interference between consumers is a major stabilizing factor in a nonlinear system. The analyses also suggest that realistic classes of consumer—resource models exist which do not obey Kolmogorov’s Criteria but are nevertheless globally stable.;,citation_author=D. L. DeAngelis;,citation_author=R. A. Goldstein;,citation_author=R. V. O’Neill;,citation_publication_date=1975-07;,citation_cover_date=1975-07;,citation_year=1975;,citation_issue=4;,citation_doi=10.2307/1936298;,citation_issn=0012-9658, 1939-9170;,citation_volume=56;,citation_language=en-US;,citation_journal_title=Ecology;">
<meta name="citation_reference" content="citation_title=Analysing ecological networks of species interactions;,citation_abstract=Network approaches to ecological questions have been increasingly used, particularly in recent decades. The abstraction of ecological systems – such as communities – through networks of interactions between their components indeed provides a way to summarize this information with single objects. The methodological framework derived from graph theory also provides numerous approaches and measures to analyze these objects and can offer new perspectives on established ecological theories as well as tools to address new challenges. However, prior to using these methods to test ecological hypotheses, it is necessary that we understand, adapt, and use them in ways that both allow us to deliver their full potential and account for their limitations. Here, we attempt to increase the accessibility of network approaches by providing a review of the tools that have been developed so far, with – what we believe to be – their appropriate uses and potential limitations. This is not an exhaustive review of all methods and metrics, but rather, an overview of tools that are robust, informative, and ecologically sound. After providing a brief presentation of species interaction networks and how to build them in order to summarize ecological information of different types, we then classify methods and metrics by the types of ecological questions that they can be used to answer from global to local scales, including methods for hypothesis testing and future perspectives. Specifically, we show how the organization of species interactions in a community yields different network structures (e.g., more or less dense, modular or nested), how different measures can be used to describe and quantify these emerging structures, and how to compare communities based on these differences in structures. Within networks, we illustrate metrics that can be used to describe and compare the functional and dynamic roles of species based on their position in the network and the organization of their interactions as well as associated new methods to test the significance of these results. Lastly, we describe potential fruitful avenues for new methodological developments to address novel ecological questions.;,citation_author=Eva Delmas;,citation_author=Mathilde Besson;,citation_author=Marie-Hélène Brice;,citation_author=Laura A. Burkle;,citation_author=Giulio V. Dalla Riva;,citation_author=Marie-Josée Fortin;,citation_author=Dominique Gravel;,citation_author=Paulo R. Guimarães;,citation_author=David H. Hembry;,citation_author=Erica A. Newman;,citation_author=Jens M. Olesen;,citation_author=Mathias M. Pires;,citation_author=Justin D. Yeakel;,citation_author=Timothée Poisot;,citation_publication_date=2019;,citation_cover_date=2019;,citation_year=2019;,citation_issue=1;,citation_doi=10.1111/brv.12433;,citation_issn=1469-185X;,citation_volume=94;,citation_language=en-US;,citation_journal_title=Biological Reviews;">
<meta name="citation_reference" content="citation_title=Compilation and Network Analyses of Cambrian Food Webs;,citation_abstract=A rich body of empirically grounded theory has developed about food webs—the networks of feeding relationships among species within habitats. However, detailed food-web data and analyses are lacking for ancient ecosystems, largely because of the low resolution of taxa coupled with uncertain and incomplete information about feeding interactions. These impediments appear insurmountable for most fossil assemblages; however, a few assemblages with excellent soft-body preservation across trophic levels are candidates for food-web data compilation and topological analysis. Here we present plausible, detailed food webs for the Chengjiang and Burgess Shale assemblages from the Cambrian Period. Analyses of degree distributions and other structural network properties, including sensitivity analyses of the effects of uncertainty associated with Cambrian diet designations, suggest that these early Paleozoic communities share remarkably similar topology with modern food webs. Observed regularities reflect a systematic dependence of structure on the numbers of taxa and links in a web. Most aspects of Cambrian food-web structure are well-characterized by a simple “niche model,” which was developed for modern food webs and takes into account this scale dependence. However, a few aspects of topology differ between the ancient and recent webs: longer path lengths between species and more species in feeding loops in the earlier Chengjiang web, and higher variability in the number of links per species for both Cambrian webs. Our results are relatively insensitive to the exclusion of low-certainty or random links. The many similarities between Cambrian and recent food webs point toward surprisingly strong and enduring constraints on the organization of complex feeding interactions among metazoan species. The few differences could reflect a transition to more strongly integrated and constrained trophic organization within ecosystems following the rapid diversification of species, body plans, and trophic roles during the Cambrian radiation. More research is needed to explore the generality of food-web structure through deep time and across habitats, especially to investigate potential mechanisms that could give rise to similar structure, as well as any differences.;,citation_author=Jennifer A. Dunne;,citation_author=Richard J. Williams;,citation_author=Neo D. Martinez;,citation_author=Rachel A. Wood;,citation_author=Douglas H. Erwin;,citation_publication_date=2008-04;,citation_cover_date=2008-04;,citation_year=2008;,citation_issue=4;,citation_doi=10.1371/journal.pbio.0060102;,citation_issn=1545-7885;,citation_volume=6;,citation_language=en-US;,citation_journal_title=PLOS Biology;,citation_publisher=Public Library of Science;">
<meta name="citation_reference" content="citation_title=The Network Structure of Food Webs;,citation_author=Jennifer A Dunne;,citation_editor=Jennifer A Dunne;,citation_editor=Mercedes Pascual;,citation_publication_date=2006;,citation_cover_date=2006;,citation_year=2006;,citation_inbook_title=Ecological networks: Linking structure and dynamics;">
<meta name="citation_reference" content="citation_title=Network structure and robustness of marine food webs;,citation_abstract=Previous studies suggest that food-web theory has yet to account for major differences in food-web properties of marine versus other types of ecosystems. We examined this issue by analyzing the network structure of food webs for the Northeast US Shelf, a Caribbean reef, and Benguela, off South Africa. The values of connectance (links per species2), link density (links per species), mean chain length, and fractions of intermediate, omnivorous, and cannibalistic taxa of these marine webs are somewhat high but still within the ranges observed in other webs. We further compared the marine webs by using the empirically corroborated “niche model” that accounts for observed variation in diversity (taxon number) and complexity (connectance). Our results substantiate previously reported results for estuarine, fresh-water, and terrestrial datasets, which suggests that food webs from different types of ecosystems with variable diversity and complexity share fundamental structural and ordering characteristics. Analyses of potential secondary extinctions resulting from species loss show that the structural robustness of marine food webs is also consistent with trends from other food webs. As expected, given their relatively high connectance, marine food webs appear fairly robust to loss of most-connected taxa as well as random taxa. Still, the short average path length between marine taxa (1.6 links) suggests that effects from perturbations, such as overfishing, can be transmitted more widely throughout marine ecosystems than previously appreciated.;,citation_author=Jennifer A. Dunne;,citation_author=Richard J. Williams;,citation_author=Neo D. Martinez;,citation_publication_date=2004;,citation_cover_date=2004;,citation_year=2004;,citation_doi=10.3354/meps273291;,citation_issn=0171-8630, 1616-1599;,citation_volume=273;,citation_language=en-US;,citation_journal_title=Marine Ecology Progress Series;">
<meta name="citation_reference" content="citation_title=Potential extinction cascades in a desert ecosystem: Linking food web interactions to community viability;,citation_abstract=Desert communities are threatened with species loss due to climate change, and their resistance to such losses is unknown. We constructed a food web of the Mojave Desert terrestrial community (300 nodes, 4080 edges) to empirically examine the potential cascading effects of bird extinctions on this desert network, compared to losses of mammals and lizards. We focused on birds because they are already disappearing from the Mojave, and their relative thermal vulnerabilities are known. We quantified bottom-up secondary extinctions and evaluated the relative resistance of the community to losses of each vertebrate group. The impact of random bird species loss was relatively low compared to the consequences of mammal (causing the greatest number of cascading losses) or reptile loss, and birds were relatively less likely to be in trophic positions that could drive top-down effects in apparent competition and tri-tropic cascade motifs. An avian extinction cascade with year-long resident birds caused more secondary extinctions than the cascade involving all bird species for randomized ordered extinctions. Notably, we also found that relatively high interconnectivity among avian species has formed a subweb, enhancing network resistance to bird losses.;,citation_author=Adam J. Eichenwald;,citation_author=Nina H. Fefferman;,citation_author=J. Michael Reed;,citation_publication_date=2024;,citation_cover_date=2024;,citation_year=2024;,citation_issue=2;,citation_doi=10.1002/ece3.10930;,citation_issn=2045-7758;,citation_volume=14;,citation_language=en-US;,citation_journal_title=Ecology and Evolution;">
<meta name="citation_reference" content="citation_title=On Random Graphs I;,citation_author=Paul Erdős;,citation_author=Alfréd Rényi;,citation_publication_date=1959;,citation_cover_date=1959;,citation_year=1959;,citation_doi=10.5486/PMD.1959.6.3-4.12;,citation_journal_title=Publicationes Mathematicae;">
<meta name="citation_reference" content="citation_title=Graphs.jl: An optimized graphs package for the Julia programming language;,citation_author=James Fairbanks;,citation_author=Mathieu Besançon;,citation_author=Simon Schölly;,citation_author=Júlio Hoffiman;,citation_author=Nick Eubank;,citation_author=Stefan Karpinski;,citation_publication_date=2021;,citation_cover_date=2021;,citation_year=2021;">
<meta name="citation_reference" content="citation_title=Habitat loss and the structure of plant-animal mutualistic networks: Mutualistic networks and habitat loss;,citation_author=Miguel A. Fortuna;,citation_author=Jordi Bascompte;,citation_publication_date=2006-03;,citation_cover_date=2006-03;,citation_year=2006;,citation_issue=3;,citation_doi=10.1111/j.1461-0248.2005.00868.x;,citation_issn=1461023X;,citation_volume=9;,citation_language=en-US;,citation_journal_title=Ecology Letters;">
<meta name="citation_reference" content="citation_title=ggVennDiagram: An Intuitive, Easy-to-Use, and Highly Customizable R Package to Generate Venn Diagram;,citation_abstract=Venn diagrams are widely used diagrams to show the set relationships in biomedical studies. In this study, we developed ggVennDiagram, an R package that could automatically generate high-quality Venn diagrams with two to seven sets. The ggVennDiagram is built based on ggplot2, and it integrates the advantages of existing packages, such as venn, RVenn, VennDiagram, and sf. Satisfactory results can be obtained with minimal configurations. Furthermore, we designed comprehensive objects to store the entire data of the Venn diagram, which allowed free access to both intersection values and Venn plot sub-elements, such as set label/edge and region label/filling. Therefore, high customization of every Venn plot sub-element can be fulfilled without increasing the cost of learning when the user is familiar with ggplot2 methods. To date, ggVennDiagram has been cited in more than 10 publications, and its source code repository has been starred by more than 140 GitHub users, suggesting a great potential in applications. The package is an open-source software released under the GPL-3 license, and it is freely available through CRAN (https://cran.r-project.org/package=ggVennDiagram).;,citation_author=Chun-Hui Gao;,citation_author=Guangchuang Yu;,citation_author=Peng Cai;,citation_publication_date=2021;,citation_cover_date=2021;,citation_year=2021;,citation_issn=1664-8021;,citation_volume=12;,citation_journal_title=Frontiers in Genetics;">
<meta name="citation_reference" content="citation_title=Inferring food web structure from predator–prey body size relationships;,citation_abstract=Current global changes make it important to be able to predict which interactions will occur in the emerging ecosystems. Most of the current methods to infer the existence of interactions between two species require a good knowledge of their behaviour or a direct observation of interactions. In this paper, we overcome these limitations by developing a method, inspired from the niche model of food web structure, using the statistical relationship between predator and prey body size to infer the matrix of potential interactions among a pool of species. The novelty of our approach is to infer, for any species of a given species pool, the three species-specific parameters of the niche model. The method applies to both local and metaweb scales. It allows one to evaluate the feeding interactions of a new species entering the community. We find that this method gives robust predictions of the structure of food webs and that its efficiency is increased when the strength of the body–size relationship between predators and preys increases. We finally illustrate the potential of the method to infer the metaweb structure of pelagic fishes of the Mediterranean sea under different global change scenarios.;,citation_author=Dominique Gravel;,citation_author=Timothée Poisot;,citation_author=Camille Albouy;,citation_author=Laure Velez;,citation_author=David Mouillot;,citation_publication_date=2013;,citation_cover_date=2013;,citation_year=2013;,citation_issue=11;,citation_doi=10.1111/2041-210X.12103;,citation_issn=2041-210X;,citation_volume=4;,citation_language=en-US;,citation_journal_title=Methods in Ecology and Evolution;">
<meta name="citation_reference" content="citation_title=AnimalTraits - a curated animal trait database for body mass, metabolic rate and brain size;,citation_abstract=Trait databases have become important resources for large-scale comparative studies in ecology and evolution. Here we introduce the AnimalTraits database, a curated database of body mass, metabolic rate and brain size, in standardised units, for terrestrial animals. The database has broad taxonomic breadth, including tetrapods, arthropods, molluscs and annelids from almost 2000 species and 1000 genera. All data recorded in the database are sourced from their original empirical publication, and the original metrics and measurements are included with each record. This allows for subsequent data transformations as required. We have included rich metadata to allow users to filter the dataset. The additional R scripts we provide will assist researchers with aggregating standardised observations into species-level trait values. Our goals are to provide this resource without restrictions, to keep the AnimalTraits database current, and to grow the number of relevant traits in the future.;,citation_author=Marie E. Herberstein;,citation_author=Donald James McLean;,citation_author=Elizabeth Lowe;,citation_author=Jonas O. Wolff;,citation_author=Md Kawsar Khan;,citation_author=Kaitlyn Smith;,citation_author=Andrew P. Allen;,citation_author=Matthew Bulbert;,citation_author=Bruno A. Buzatto;,citation_author=Mark D. B. Eldridge;,citation_author=Daniel Falster;,citation_author=Laura Fernandez Winzer;,citation_author=Simon C. Griffith;,citation_author=Joshua S. Madin;,citation_author=Ajay Narendra;,citation_author=Mark Westoby;,citation_author=Martin J. Whiting;,citation_author=Ian J. Wright;,citation_author=Alexandra J. R. Carthey;,citation_publication_date=2022-06;,citation_cover_date=2022-06;,citation_year=2022;,citation_issue=1;,citation_doi=10.1038/s41597-022-01364-9;,citation_issn=2052-4463;,citation_volume=9;,citation_language=en-US;,citation_journal_title=Scientific Data;,citation_publisher=Nature Publishing Group;">
<meta name="citation_reference" content="citation_title=PanTHERIA: A species-level database of life history, ecology, and geography of extant and recently extinct mammals;,citation_abstract=Analyses of life-history, ecological, and geographic trait differences among species, their causes, correlates, and likely consequences are increasingly important for understanding and conserving biodiversity in the face of rapid global change. Assembling multispecies trait data from diverse literature sources into a single comprehensive data set requires detailed consideration of methods to reliably compile data for particular species, and to derive single estimates from multiple sources based on different techniques and definitions. Here we describe PanTHERIA, a species-level data set compiled for analysis of life history, ecology, and geography of all known extant and recently extinct mammals. PanTHERIA is derived from a database capable of holding multiple geo-referenced values for variables within a species containing 100 740 lines of biological data for extant and recently extinct mammalian species, collected over a period of three years by 20 individuals. PanTHERIA also includes spatial databases of mammalian geographic ranges and global climatic and anthropogenic variables. Here we detail how the data fields are extracted and defined for PanTHERIA using a customized data input format (MammalForm); how data were collected from the literature, species names and sources tracked, error-checking and validation procedures applied, and how data were consolidated into species-level values for each variable. Tables of the consolidated species-level values are made available for each of two recent species-level taxonomic classifications of mammals, as well as associated taxonomic synonymy conversion and data-input files. This study provides a useful guide to prospective researchers on how to structure and codify life-history, ecological, geographic, and taxonomic data and methods to extract meaningful species-level traits. It also provides comprehensive information on traits like size, diet, environmental conditions, and ecology to permit macroecological and macroevolutionary analyses of this important clade. The complete data sets corresponding to abstracts published in the Data Papers section of the journal are published electronically in Ecological Archives at http://esapubs.org/archive. (The accession number for each Data Paper is given directly beneath the title.);,citation_author=Kate E. Jones;,citation_author=Jon Bielby;,citation_author=Marcel Cardillo;,citation_author=Susanne A. Fritz;,citation_author=Justin O’Dell;,citation_author=C. David L. Orme;,citation_author=Kamran Safi;,citation_author=Wes Sechrest;,citation_author=Elizabeth H. Boakes;,citation_author=Chris Carbone;,citation_author=Christina Connolly;,citation_author=Michael J. Cutts;,citation_author=Janine K. Foster;,citation_author=Richard Grenyer;,citation_author=Michael Habib;,citation_author=Christopher A. Plaster;,citation_author=Samantha A. Price;,citation_author=Elizabeth A. Rigby;,citation_author=Janna Rist;,citation_author=Amber Teacher;,citation_author=Olaf R. P. Bininda-Emonds;,citation_author=John L. Gittleman;,citation_author=Georgina M. Mace;,citation_author=Andy Purvis;,citation_publication_date=2009;,citation_cover_date=2009;,citation_year=2009;,citation_issue=9;,citation_doi=10.1890/08-1494.1;,citation_issn=1939-9170;,citation_volume=90;,citation_language=en-US;,citation_journal_title=Ecology;">
<meta name="citation_reference" content="citation_title=Chasing Ecological Interactions;,citation_abstract=Basic research on biodiversity has concentrated on individual species—naming new species, studying distribution patterns, and analyzing their evolutionary relationships. Yet biodiversity is more than a collection of individual species; it is the combination of biological entities and processes that support life on Earth. To understand biodiversity we must catalog it, but we must also assess the ways species interact with other species to provide functional support for the Tree of Life. Ecological interactions may be lost well before the species involved in those interactions go extinct; their ecological functions disappear even though they remain. Here, I address the challenges in studying the functional aspects of species interactions and how basic research is helping us address the fast-paced extinction of species due to human activities.;,citation_author=Pedro Jordano;,citation_publication_date=2016-09;,citation_cover_date=2016-09;,citation_year=2016;,citation_issue=9;,citation_doi=10.1371/journal.pbio.1002559;,citation_issn=1545-7885;,citation_volume=14;,citation_language=en-US;,citation_journal_title=PLOS Biology;,citation_publisher=Public Library of Science;">
<meta name="citation_reference" content="citation_title=Sampling networks of ecological interactions;,citation_author=Pedro Jordano;,citation_publication_date=2016-09;,citation_cover_date=2016-09;,citation_year=2016;,citation_doi=10.1111/1365-2435.12763;,citation_issn=02698463;,citation_language=en-US;,citation_journal_title=Functional Ecology;">
<meta name="citation_reference" content="citation_title=A neutral-niche theory of nestedness in mutualistic networks;,citation_abstract=Recently, there has been a vigorous interest in community ecology about the structure of mutualistic networks and its importance for species persistence and coevolution. However, the mechanisms shaping mutualistic networks have been rarely explored. Here we extend for the first time the neutral theory of biodiversity to a multi trophic system. We focus on nestedness, a distinctive pattern of mutualistic community assembly showing two characteristics, namely, asymmetrical specialization (specialists interacting with generalists) and a generalist core (generalists interacting with generalists). We investigate the importance of relative species abundance (RSA) for the nested assembly of plant–animal mutualistic networks. Our results show that neutral mutualistic communities give rise to networks considerably more nested than real communities. RSA explains 60–70% of nested patterns in two real communities studied here, while 30–40% of nestedness is still unexplained. The nested pattern in real communities is better explained when we introduce interaction-specific species traits such as forbidden links and intensity of dependence (relative importance of fruits for the diet of a frugivore) in our analysis. The fact that neutral mutualistic communities exhibit a perfectly nested structure and do not show a random or compartmentalized structure, underlines the importance of RSA in the assembly of mutualistic networks.;,citation_author=Abhay Krishna;,citation_author=Paulo R. Guimarães Jr;,citation_author=Pedro Jordano;,citation_author=Jordi Bascompte;,citation_publication_date=2008;,citation_cover_date=2008;,citation_year=2008;,citation_issue=11;,citation_doi=10.1111/j.1600-0706.2008.16540.x;,citation_issn=1600-0706;,citation_volume=117;,citation_language=en-US;,citation_journal_title=Oikos;">
<meta name="citation_reference" content="citation_title=Analysis of Ecological Communities;,citation_author=Bruce McCune;,citation_author=James Grace;,citation_publication_date=2002;,citation_cover_date=2002;,citation_year=2002;,citation_isbn=0-9721290-0-6;">
<meta name="citation_reference" content="citation_title=Inferring biotic interactions from proxies;,citation_abstract=Inferring biotic interactions from functional, phylogenetic and geographical proxies remains one great challenge in ecology. We propose a conceptual framework to infer the backbone of biotic interaction networks within regional species pools. First, interacting groups are identified to order links and remove forbidden interactions between species. Second, additional links are removed by examination of the geographical context in which species co-occur. Third, hypotheses are proposed to establish interaction probabilities between species. We illustrate the framework using published food-webs in terrestrial and marine systems. We conclude that preliminary descriptions of the web of life can be made by careful integration of data with theory.;,citation_author=Ignacio Morales-Castilla;,citation_author=Miguel G. Matias;,citation_author=Dominique Gravel;,citation_author=Miguel B. Araújo;,citation_publication_date=2015-06;,citation_cover_date=2015-06;,citation_year=2015;,citation_issue=6;,citation_doi=10.1016/j.tree.2015.03.014;,citation_volume=30;,citation_journal_title=Trends in Ecology &amp;amp;amp; Evolution;">
<meta name="citation_reference" content="citation_title=Networks. An introduction;,citation_author=Mark E. J. Newman;,citation_publication_date=2010;,citation_cover_date=2010;,citation_year=2010;">
<meta name="citation_reference" content="citation_title=Natural - synthetic - artificial!;,citation_abstract=The terms “natural, “synthetic” and “artificial” are discussed in relation to synthetic and artificial chromosomes and genomes, synthetic and artificial cells and artificial life.;,citation_author=Peter E. Nielsen;,citation_publication_date=2010-07;,citation_cover_date=2010-07;,citation_year=2010;,citation_issue=1;,citation_doi=10.4161/adna.1.1.12934;,citation_issn=1949-095X;,citation_pmid=21687528;,citation_volume=1;,citation_journal_title=Artificial DNA: PNA &amp;amp;amp; XNA;,citation_publisher=Taylor &amp;amp; Francis;">
<meta name="citation_reference" content="citation_title=Statistical mechanics of networks;,citation_abstract=We study the family of network models derived by requiring the expected properties of a graph ensemble to match a given set of measurements of a real-world network, while maximizing the entropy of the ensemble. Models of this type play the same role in the study of networks as is played by the Boltzmann distribution in classical statistical mechanics; they offer the best prediction of network properties subject to the constraints imposed by a given set of observations. We give exact solutions of models within this class that incorporate arbitrary degree distributions and arbitrary but independent edge probabilities. We also discuss some more complex examples with correlated edges that can be solved approximately or exactly by adapting various familiar methods, including mean-field theory, perturbation theory, and saddle-point expansions.;,citation_author=Juyong Park;,citation_author=M. E. J. Newman;,citation_publication_date=2004-12;,citation_cover_date=2004-12;,citation_year=2004;,citation_issue=6;,citation_doi=10.1103/PhysRevE.70.066117;,citation_volume=70;,citation_journal_title=Physical Review E;,citation_publisher=American Physical Society;">
<meta name="citation_reference" content="citation_title=Fit, efficiency, and biology: Some thoughts on judging food web models;,citation_abstract=Revealing the processes that determine who eats whom, and thereby the structure of food webs, is a long running challenge in ecological research. Recent advances include development of new methods for measuring fit of models to observed food web data, and thereby testing which are the “best” food web models. The best model could be considered the most efficient with relatively few parameters and high explanatory power. Another recent advance involves adding some additional biology to food web models in the form of foraging theory based on maximisation of energy intake as the predictor of species’ diets in food webs. While it is interesting to compare efficiency among food web models, we believe that such comparisons at least should be interpreted with caution, since they do not account for any differences in motivation, formulation, and potential that might also exist among models. Furthermore, we see an important but somewhat neglected role for experimental tests of models of food web structure.;,citation_author=Owen L. Petchey;,citation_author=Andrew P. Beckerman;,citation_author=Jens O. Riede;,citation_author=Philip H. Warren;,citation_publication_date=2011-06;,citation_cover_date=2011-06;,citation_year=2011;,citation_issue=1;,citation_doi=10.1016/j.jtbi.2011.03.019;,citation_issn=0022-5193;,citation_volume=279;,citation_journal_title=Journal of Theoretical Biology;">
<meta name="citation_reference" content="citation_title=Size, foraging, and food web structure;,citation_abstract=Understanding what structures ecological communities is vital to answering questions about extinctions, environmental change, trophic cascades, and ecosystem functioning. Optimal foraging theory was conceived to increase such understanding by providing a framework with which to predict species interactions and resulting community structure. Here, we use an optimal foraging model and allometries of foraging variables to predict the structure of real food webs. The qualitative structure of the resulting model provides a more mechanistic basis for the phenomenological rules of previous models. Quantitative analyses show that the model predicts up to 65% of the links in real food webs. The deterministic nature of the model allows analysis of the model’s successes and failures in predicting particular interactions. Predacious and herbivorous feeding interactions are better predicted than pathogenic, parasitoid, and parasitic interactions. Results also indicate that accurate prediction and modeling of some food webs will require incorporating traits other than body size and diet choice models specific to different types of feeding interaction. The model results support the hypothesis that individual behavior, subject to natural selection, determines individual diets and that food web structure is the sum of these individual decisions.;,citation_author=Owen L. Petchey;,citation_author=Andrew P. Beckerman;,citation_author=Jens O. Riede;,citation_author=Philip H. Warren;,citation_publication_date=2008-03;,citation_cover_date=2008-03;,citation_year=2008;,citation_issue=11;,citation_doi=10.1073/pnas.0710672105;,citation_issn=0027-8424, 1091-6490;,citation_volume=105;,citation_language=en-US;,citation_journal_title=Proceedings of the National Academy of Sciences;">
<meta name="citation_reference" content="citation_title=Before, during and after megafaunal extinctions: Human impact on Pleistocene-Holocene trophic networks in South Patagonia;,citation_abstract=Worldwide extinctions of large terrestrial vertebrates in the late Pleistocene provide insight on how humans reshape ecological communities. Understanding the ecological causes and consequences of megafaunal extinctions requires integrating approaches to reconstruct the ecological communities from the past. Here, we combined archeological and paleontological evidence with network analyses to understand the changes in ecological communities from late Pleistocene to the Holocene in South Patagonia, the last continental region where the encounter between humans and extinct megafauna occurred. The zooarcheological record suggests humans would have interacted mainly with large-bodied species, which comprise a small subset of the available prey. Accordingly, using network reconstructions and structural analyses, we found that human arrival would have produced minor changes in the overall structure of trophic networks. However, those few novel interactions established by humans would have created multiple indirect paths among megafaunal species. Indirect paths are the route for indirect effects such as competition and increase the vulnerability of interaction networks to perturbations. After the extinctions of most of the megafauna, the impoverished network became structurally simpler and densely connected. Our reconstructions of past trophic networks show that multiple indirect effects, potentially contributing to extinctions, can emerge even from a limited number of novel interactions and illustrate how network organization can affect and be affected by extinctions events.;,citation_author=Mathias M. Pires;,citation_author=Diego Rindel;,citation_author=Bruno Moscardi;,citation_author=Livia R. Cruz;,citation_author=Paulo R. Guimarães;,citation_author=Sergio F. Reis;,citation_author=S. Ivan Perez;,citation_publication_date=2020-12;,citation_cover_date=2020-12;,citation_year=2020;,citation_doi=10.1016/j.quascirev.2020.106696;,citation_issn=0277-3791;,citation_volume=250;,citation_journal_title=Quaternary Science Reviews;">
<meta name="citation_reference" content="citation_title=Global knowledge gaps in species interaction networks data;,citation_abstract=Ecological networks are increasingly studied at large spatial scales, expanding their focus from a conceptual tool for community ecology into one that also addresses questions in biogeography and macroecology. This effort is supported by increased access to standardized information on ecological networks, in the form of openly accessible databases. Yet, there has been no systematic evaluation of the fitness for purpose of these data to explore synthesis questions at very large spatial scales. In particular, because the sampling of ecological networks is a difficult task, they are likely to not have a good representation of the diversity of Earth’s bioclimatic conditions, likely to be spatially aggregated, and therefore unlikely to achieve broad representativeness. In this paper, we analyse over 1300 ecological networks in the mangal.io database, and discuss their coverage of climates, and the geographic areas in which there is a deficit of data on ecological networks. Taken together, our results suggest that while some information about the global structure of ecological networks is available, it remains fragmented over space, with further differences by types of ecological interactions. This causes great concerns both for our ability to transfer knowledge from one region to the next, but also to forecast the structural change in networks under climate change.;,citation_author=Timothée Poisot;,citation_author=Gabriel Bergeron;,citation_author=Kevin Cazelles;,citation_author=Tad Dallas;,citation_author=Dominique Gravel;,citation_author=Andrew MacDonald;,citation_author=Benjamin Mercier;,citation_author=Clément Violet;,citation_author=Steve Vissault;,citation_publication_date=2021;,citation_cover_date=2021;,citation_year=2021;,citation_issue=n/a;,citation_doi=10.1111/jbi.14127;,citation_issn=1365-2699;,citation_volume=n/a;,citation_language=en-US;,citation_journal_title=Journal of Biogeography;">
<meta name="citation_reference" content="citation_title=Guidelines for the prediction of species interactions through binary classification;,citation_abstract=The prediction of species interactions is gaining momentum as a way to circumvent limitations in data volume. Yet, ecological networks are challenging to predict because they are typically small and sparse. Dealing with extreme class imbalance is a challenge for most binary classifiers, and there are currently no guidelines as to how predictive models can be trained for this specific problem. Using simple mathematical arguments and numerical experiments in which a variety of classifiers (for supervised learning) are trained on simulated networks, we develop a series of guidelines related to the choice of measures to use for model selection, and the ways to assemble the training dataset. Neither classifier accuracy nor the area under the receiver operating characteristic curve (ROC-AUC) are informative measures for the performance of interaction prediction. The area under the precision-recall curve (PR-AUC) is a fairer assessment of performance. In some cases, even standard measures can lead to selecting a more biased classifier because the effect of connectance is strong. The amount of correction to apply to the training dataset depends on network connectance, on the measure to be optimized, and only weakly on the classifier. These results reveal that training machines to predict networks is a challenging task, and that in virtually all cases, the composition of the training set needs to be fine-tuned before performing the actual training. We discuss these consequences in the context of the low volume of data.;,citation_author=Timothée Poisot;,citation_publication_date=2023;,citation_cover_date=2023;,citation_year=2023;,citation_issue=5;,citation_doi=10.1111/2041-210X.14071;,citation_issn=2041-210X;,citation_volume=14;,citation_language=en-US;,citation_journal_title=Methods in Ecology and Evolution;">
<meta name="citation_reference" content="citation_title=Mangal – making ecological network analysis simple;,citation_abstract=The study of ecological networks is severely limited by 1) the difficulty to access data, 2) the lack of a standardized way to link meta-data with interactions, and 3) the disparity of formats in which ecological networks themselves are stored and represented. To overcome these limitations, we have designed a data specification for ecological networks. We implemented a database respecting this standard, and released an R package (rmangal) allowing users to programmatically access, curate, and deposit data on ecological interactions. In this article, we show how these tools, in conjunction with other frameworks for the programmatic manipulation of open ecological data, streamlines the analysis process and improves replicability and reproducibility of ecological network studies.;,citation_author=Timothée Poisot;,citation_author=Benjamin Baiser;,citation_author=Jennifer Dunne;,citation_author=Sonia Kéfi;,citation_author=François Massol;,citation_author=Nicolas Mouquet;,citation_author=Tamara N. Romanuk;,citation_author=Daniel B. Stouffer;,citation_author=Spencer A. Wood;,citation_author=Dominique Gravel;,citation_publication_date=2016;,citation_cover_date=2016;,citation_year=2016;,citation_issue=4;,citation_doi=10.1111/ecog.00976;,citation_issn=1600-0587;,citation_volume=39;,citation_language=en-US;,citation_journal_title=Ecography;">
<meta name="citation_reference" content="citation_title=Beyond species: Why ecological interaction networks vary through space and time;,citation_abstract=Community ecology is tasked with the considerable challenge of predicting the structure, and properties, of emerging ecosystems. It requires the ability to understand how and why species interact, as this will allow the development of mechanism-based predictive models, and as such to better characterize how ecological mechanisms act locally on the existence of inter-specific interactions. Here we argue that the current conceptualization of species interaction networks is ill-suited for this task. Instead, we propose that future research must start to account for the intrinsic variability of species interactions, then scale up from here onto complex networks. This can be accomplished simply by recognizing that there exists intra-specific variability, in traits or properties related to the establishment of species interactions. By shifting the scale towards population-based processes, we show that this new approach will improve our predictive ability and mechanistic understanding of how species interact over large spatial or temporal scales. Synthesis Although species interactions are the backbone of ecological communities, we have little insights on how (and why) they vary through space and time. In this article, we build on existing empirical literature to show that the same species may happen to interact in different ways when their local abundances vary, their trait distribution changes, or when the environment affects either of these factors. We discuss how these findings can be integrated in existing frameworks for the analysis and simulation of species interactions.;,citation_author=Timothée Poisot;,citation_author=Daniel B. Stouffer;,citation_author=Dominique Gravel;,citation_publication_date=2015;,citation_cover_date=2015;,citation_year=2015;,citation_issue=3;,citation_doi=10.1111/oik.01719;,citation_issn=1600-0706;,citation_volume=124;,citation_language=en-US;,citation_journal_title=Oikos;">
<meta name="citation_reference" content="citation_title=Inferring predator–prey interactions in food webs;,citation_abstract=Food webs are a powerful way to represent the diversity, structure, and function of ecological systems. However, the accurate description of food webs requires significant effort in time and resources, limiting their widespread use in ecological studies. Newly published methods allow for the inference of feeding interactions using proxy variables. Here, we compare the accuracy of two recently described methods, as well as describe a composite model of the two, for the inference of feeding interactions using a large, well-described dataset. Both niche and neutral processes are involved in determining whether or not two species will form a feeding link in communities. Three different models for determining niche constraints of feeding interactions are compared, and all three models are extended by incorporating neutral processes, based on relative abundances. The three models compared here infer niche processes through (a) phylogenetic relationships, (b) local species trait distributions (e.g., body size), and (c) a composite of phylogeny and local traits. We show that all three methods perform well at predicting individual species interactions, and that these individual predictions scale up to the network level, resulting in food web structure of inferred networks being similar to their empirical counterparts. Our results indicate that inferring food web structure using phylogenies can be an efficient way of getting summary webs with minimal data, and offers a conservative test of changes in food web structure, particularly when there is low species turnover between sites. Inferences made using traits require more data, but allows for greater understanding of the mechanisms underlying trophic interactions. A composite model of the two methods provides a framework for investigating the importance of how phylogeny, trait distributions, and relative abundances, affect species interactions, and network structure.;,citation_author=Justin P. F. Pomeranz;,citation_author=Ross M. Thompson;,citation_author=Timothée Poisot;,citation_author=Jon S. Harding;,citation_publication_date=2019;,citation_cover_date=2019;,citation_year=2019;,citation_issue=3;,citation_doi=10.1111/2041-210X.13125;,citation_issn=2041-210X;,citation_volume=10;,citation_language=en-US;,citation_journal_title=Methods in Ecology and Evolution;">
<meta name="citation_reference" content="citation_title=Testing the Cascade Model for Food Webs;,citation_abstract=The cascade model is a standard model for food web data. This article describes a method for testing the cascade model that complements existing tests. Like existing tests, the new test is conditional on the number of links observed in the predation matrix. One important feature of the cascade model is that an ordering of species is required, but in practice the true order is unknown. This feature has important implications for tests of the model, both for the new test and for existing tests. The analysis here does not require a single assumed order. Some real food webs are used for illustration.;,citation_author=John M. Roberts;,citation_publication_date=2003;,citation_cover_date=2003;,citation_year=2003;,citation_fulltext_html_url=https://www.jstor.org/stable/1400428;,citation_issue=2;,citation_issn=1085-7117;,citation_volume=8;,citation_journal_title=Journal of Agricultural, Biological, and Environmental Statistics;,citation_publisher=[International Biometric Society, Springer];">
<meta name="citation_reference" content="citation_title=Modeling Food Webs: Exploring Unexplained Structure Using Latent Traits.;,citation_abstract=Several stochastic models have tried to capture the architecture of food webs. This approach is interesting, but it is limited by the fact that different assumptions can yield similar results. To overcome this limitation, we develop a purely statistical approach. Body size in terms of an optimal ratio between prey and predator is used as explanatory variable. In 12 observed food webs, this model predicts, on average, 20% of interactions. To analyze the unexplained part, we introduce a latent term: each species is described by two latent traits, foraging and vulnerability, that represent nonmeasured characteristics of species once the optimal body size has been accounted for. The model now correctly predicts an average of 73% of links. The key features of our approach are that latent traits quantify the structure that is left unexplained by the explanatory variable and that this quantification allows a test of whether independent biological information, such as microhabitat use, camouflage, or phylogeny, explains this structure. We illustrate this method with phylogeny and find that it is linked to one or both latent traits in nine of 12 food webs. Our approach opens the door to the formulation of more complex models that can be applied to any kind of biological network.;,citation_author=Rudolf Philippe Rohr;,citation_author=Heike Scherer;,citation_author=Patrik Kehrli;,citation_author=Christian Mazza;,citation_author=Louis-Félix Bersier;,citation_publication_date=2010-08;,citation_cover_date=2010-08;,citation_year=2010;,citation_issue=2;,citation_doi=10.1086/653667;,citation_issn=0003-0147;,citation_volume=176;,citation_journal_title=The American Naturalist;,citation_publisher=The University of Chicago Press;">
<meta name="citation_reference" content="citation_title=Extinction Cascades and Catastrophe in Ancient Food Webs;,citation_abstract=A model is developed to explore the potential responses of paleocommunities to disruptions of primary production during times of mass extinction and ecological crisis. Disruptions of primary production are expected to generate bottom-up cascades of secondary extinction, and these are predictable given species richnesses, functional diversity, and trophic link distributions. If, however, consumers are permitted to compensate for the loss of trophic resources by increasing the intensities of their remaining biotic interactions, top-down driven catastrophic increases of secondary extinction emerge from the model. Both bottom-up and top-down effects are themselves controlled by the geometry of the food webs. The general Phanerozoic trends of increasing taxonomic and ecological diversities, as well as the varying strengths of biotic interactions, have led to food webs of increasing complexity. The frequency of catastrophic secondary extinction increases as food web complexity increases, but increased complexity also serves to dampen the magnitude of the secondary extinctions. When intraguild competitive interactions are included in the model, competitively inferior taxa are observed to possess greater probabilities of survival if the guilds are embedded in simple subnetworks of the overall food web. The result is the emergence of post-extinction guilds dominated by those inferior taxa. These results are congruent with empirical observations of &amp;amp;amp;quot;disaster taxa&amp;quot; dominance after some mass extinction events, and provide a mechanism for the reorganization of ecosystems that is observed after those events. The model makes the testable prediction that dominance by disaster taxa, however, should be observed only when bottom-up disruptions have caused ecosystems to collapse catastrophically.;,citation_author=Peter D. Roopnarine;,citation_publication_date=2006;,citation_cover_date=2006;,citation_year=2006;,citation_fulltext_html_url=https://www.jstor.org/stable/4096814;,citation_issue=1;,citation_issn=0094-8373;,citation_volume=32;,citation_journal_title=Paleobiology;,citation_publisher=Paleontological Society;">
<meta name="citation_reference" content="citation_title=Food webs: Experts consuming families of experts;,citation_abstract=Food webs of habitats as diverse as lakes or desert valleys are known to exhibit common “food-web patterns”, but the detailed mechanisms generating these structures have remained unclear. By employing a stochastic, dynamical model, we show that many aspects of the structure of predatory food webs can be understood as the traces of an evolutionary history where newly evolving species avoid direct competition with their relatives. The tendency to avoid sharing natural enemies (apparent competition) with related species is considerably weaker. Thus, “experts consuming families of experts” can be identified as the main underlying food-web pattern. We report the results of a systematic, quantitative model validation showing that the model is surprisingly accurate.;,citation_author=A. G. Rossberg;,citation_author=H. Matsuda;,citation_author=T. Amemiya;,citation_author=K. Itoh;,citation_publication_date=2006-08;,citation_cover_date=2006-08;,citation_year=2006;,citation_issue=3;,citation_doi=10.1016/j.jtbi.2005.12.021;,citation_issn=0022-5193;,citation_volume=241;,citation_journal_title=Journal of Theoretical Biology;">
<meta name="citation_reference" content="citation_title=Ecological network assembly: How the regional metaweb influences local food webs;,citation_abstract=Local food webs result from a sequence of colonisations and extinctions by species from the regional pool or metaweb, that is, the assembly process. Assembly is theorised to be a selective process: whether or not certain species or network structures can persist is partly determined by local processes including habitat filtering and dynamical constraints. Consequently, local food web structure should reflect these processes. The goal of this study was to test evidence for these selective processes by comparing the structural properties of real food webs to the expected distribution given the metaweb. We were particularly interested in ecological dynamics; if the network properties commonly associated with dynamical stability are indeed the result of stability constraints, then they should deviate from expectation in the direction predicted by theory. To create a null expectation, we used the novel approach of randomly assembling model webs by drawing species and interactions from the empirical metaweb. The assembly model permitted colonisation and extinction, and required a consumer species to have at least one prey, but had no habitat type nor population dynamical constraints. Three datasets were used: (a) the marine Antarctic metaweb, with two local food webs; (b) the 50 lakes of the Adirondacks; and (c) the arthropod community from Florida Keys’ classic defaunation experiment. Contrary to our expectations, we found that there were almost no differences between empirical webs and those resulting from the null assembly model. Few empirical food webs showed significant differences with network properties, motif representations and topological roles. Network properties associated with stability did not deviate from expectation in the direction predicted by theory. Our results suggest that-for the commonly used metrics we considered-local food web structure is not strongly influenced by dynamical nor habitat restrictions. Instead, the structure is inherited from the metaweb. This suggests that the network properties typically attributed as causes or consequences of ecological stability are instead a by-product of the assembly process (i.e. spandrels), and may potentially be too coarse to detect the true signal of dynamical constraint.;,citation_author=Leonardo A. Saravia;,citation_author=Tomás I Marina;,citation_author=Nadiah P. Kristensen;,citation_author=Marleen De Troch;,citation_author=Fernando R. Momo;,citation_publication_date=2022;,citation_cover_date=2022;,citation_year=2022;,citation_issue=3;,citation_doi=10.1111/1365-2656.13652;,citation_issn=0021-8790;,citation_volume=91;,citation_language=en-US;,citation_journal_title=Journal of Animal Ecology;,citation_publisher=Wiley;">
<meta name="citation_reference" content="citation_title=A framework for reconstructing ancient food webs using functional trait data;,citation_abstract=Food webs provide quantitative insights into the structure and dynamics of ecological communities. Previous work has shown their utility in understanding community responses to modern and ancient perturbations, including anthropogenic change and mass extinctions. However, few ancient food webs have been reconstructed due to difficulties assessing trophic interactions amongst extinct species derived from an incomplete fossil record. We present and assess the Paleo Food web Inference Model (PFIM). PFIM uses functional trait data–predictive of interactions in modern ecosystems and commonly available for fossil organisms–to reconstruct ancient food webs. We test the model by (i) applying it to four modern ecosystems with empirical constrained food webs to directly compare PFIM-constructed networks to their empirical counterparts, (ii) by carefully comparing discrepancies between PFIM-inferred and empirical webs in one of those systems, and (iii) by comparing networks describing feasible trophic interactions (&amp;amp;amp;quot;feasible webs&amp;quot;) with networks to which we superimpose characteristic interaction distributions derived from modern theory (&quot;realized webs&quot;). As a proof of concept, we then apply the method to faunal data from two Cambrian fossil deposits to reconstruct ancient trophic systems. PFIM-inferred feasible food webs successfully predict ~70% of trophic interactions across four modern systems. Furthermore, inferred food webs with enforced interaction distributions (i.e., realized webs) accurately predict ~90% of interactions. Comparisons with a global database of trophic interactions and other food web models, suggest that under sampling of empirical webs accounts for up to 21% of the remaining differences between PFIM and empirical food webs. Food webs can be reasonably approximated by inferring trophic interactions based upon life habit traits. This study provides the foundation to use trait-based inference models across the fossil record to examine ancient food webs and community evolution.;,citation_author=Jack O. Shaw;,citation_author=Alexander M. Dunhill;,citation_author=Andrew P. Beckerman;,citation_author=Jennifer A. Dunne;,citation_author=Pincelli M. Hull;,citation_publication_date=2024-01;,citation_cover_date=2024-01;,citation_year=2024;,citation_doi=10.1101/2024.01.30.578036;,citation_language=en-US;,citation_publisher=bioRxiv;">
<meta name="citation_reference" content="citation_title=On Lumping Species in Food Webs;,citation_abstract=Species in community food webs are commonly aggregated or lumped on the basis of overlap in predators and prey. This note reports an unexpected result of lumping species in observed food webs and simulated food webs. The main result is that it is much easier to lump species in observed webs than in simulated webs. The reason is that there is much more overlap in predators and prey for species in observed webs than in simulated webs. This points to a fundamental deficiency in web assembly models that assume a random distribution of links.;,citation_author=Andrew R. Solow;,citation_author=Andrew R. Beet;,citation_publication_date=1998;,citation_cover_date=1998;,citation_year=1998;,citation_issue=6;,citation_doi=10.1890/0012-9658(1998)079[2013:OLSIFW]2.0.CO;2;,citation_issn=1939-9170;,citation_volume=79;,citation_language=en-US;,citation_journal_title=Ecology;">
<meta name="citation_reference" content="citation_title=Quantitative Patterns in the Structure of Model and Empirical Food Webs;,citation_abstract=Understanding the structure of food webs, and the mechanisms that give rise to this structure, is one of the most challenging problems in ecology. We analyze from a statistical physics perspective the network structure of model food webs and of 15 community food webs from a variety of environments, including freshwater, marine, estuarine, and terrestrial environments. We perform a theoretical analysis of two recently proposed models for food webs, the niche model of R. J. Williams and N. D. Martinez and the nested-hierarchy model of M.-F. Cattin et al. We find that the two models generate distributions of numbers of prey, predators, and links that are described by the same analytical expressions. Our analytical treatment reveals that a model’s capacity to reproduce empirical data is principally determined by its ability to satisfy two conditions: (1) the species’ niche values form a totally ordered set and (2) each species has a specific exponentially decaying probability of preying on a given fraction of the species with lower niche values. To test this hypothesis, we generalize the cascade model of J. E. Cohen and C. M. Newman so that it satisfies condition 2 and find that the new model is able to reproduce the properties of empirical food webs, validating our hypothesis. We use our analytical predictions as a guide to the analysis of 15 of the most complete empirical food webs available. We demonstrate that the quantitative unifying patterns that describe the properties of the food-web models considered earlier also describe the majority of the empirical webs considered. We find good agreement between the empirical data and the models for the average distance between species and the average clustering coefficient. Our results strongly support two hypotheses: first, that any model satisfying the two conditions we identify will accurately reproduce a number of the statistical properties of empirical food webs, and second, that the empirical distributions of number of prey and number of predators follow universal functional forms that, without free parameters, match our analytical predictions.;,citation_author=D. B. Stouffer;,citation_author=J. Camacho;,citation_author=R. Guimerà;,citation_author=C. A. Ng;,citation_author=L. A. Nunes Amaral;,citation_publication_date=2005;,citation_cover_date=2005;,citation_year=2005;,citation_issue=5;,citation_doi=10.1890/04-0957;,citation_issn=1939-9170;,citation_volume=86;,citation_language=en-US;,citation_journal_title=Ecology;">
<meta name="citation_reference" content="citation_title=A robust measure of food web intervality;,citation_abstract=Intervality of a food web is related to the number of trophic dimensions characterizing the niches in a community. We introduce here a mathematically robust measure for food web intervality. It has previously been noted that empirical food webs are not strictly interval; however, upon comparison to suitable null hypotheses, we conclude that empirical food webs actually do exhibit a strong bias toward contiguity of prey, that is, toward intervality. Further, our results strongly suggest that empirically observed species and their diets can be mapped onto a single dimension. This finding validates a critical assumption in the recently proposed static niche model and provides guidance for ongoing efforts to develop dynamic models of ecosystems.;,citation_author=Daniel B. Stouffer;,citation_author=Juan Camacho;,citation_author=Luís A. Nunes Amaral;,citation_publication_date=2006-12;,citation_cover_date=2006-12;,citation_year=2006;,citation_issue=50;,citation_doi=10.1073/pnas.0603844103;,citation_volume=103;,citation_journal_title=Proceedings of the National Academy of Sciences;,citation_publisher=Proceedings of the National Academy of Sciences;">
<meta name="citation_reference" content="citation_title=Food web reconstruction through phylogenetic transfer of low-rank network representation;,citation_abstract=Despite their importance in many ecological processes, collecting data and information on ecological interactions is an exceedingly challenging task. For this reason, large parts of the world have a data deficit when it comes to species interactions and how the resulting networks are structured. As data collection alone is unlikely to be sufficient, community ecologists must adopt predictive methods. We present a methodological framework that uses graph embedding and transfer learning to assemble a predicted list of trophic interactions of a species pool for which their interactions are unknown. Specifically, we “learn” the information (latent traits) of species from a known interaction network and infer the latent traits of another species pool for which we have no a priori interaction data based on their phylogenetic relatedness to species from the known network. The latent traits can then be used to predict interactions and construct an interaction network. Here we assembled a metaweb for Canadian mammals derived from interactions in the European food web, despite only 4% of common species being shared between the two locations. The results of the predictive model are compared against databases of recorded pairwise interactions, showing that we correctly recover 91% of known interactions. The framework itself is robust even when the known network is incomplete or contains spurious interactions making it an ideal candidate as a tool for filling gaps when it comes to species interactions. We provide guidance on how this framework can be adapted by substituting some approaches or predictors in order to make it more generally applicable.;,citation_author=Tanya Strydom;,citation_author=Salomé Bouskila;,citation_author=Francis Banville;,citation_author=Ceres Barros;,citation_author=Dominique Caron;,citation_author=Maxwell J. Farrell;,citation_author=Marie-Josée Fortin;,citation_author=Victoria Hemming;,citation_author=Benjamin Mercier;,citation_author=Laura J. Pollock;,citation_author=Rogini Runghen;,citation_author=Giulio V. Dalla Riva;,citation_author=Timothée Poisot;,citation_publication_date=2022;,citation_cover_date=2022;,citation_year=2022;,citation_issue=12;,citation_doi=10.1111/2041-210X.13835;,citation_issn=2041-210X;,citation_volume=13;,citation_language=en-US;,citation_journal_title=Methods in Ecology and Evolution;">
<meta name="citation_reference" content="citation_title=Graph embedding and transfer learning can help predict potential species interaction networks despite data limitations;,citation_abstract=Metawebs (networks of potential interactions within a species pool) are a powerful abstraction to understand how large-scale species interaction networks are structured. Because metawebs are typically expressed at large spatial and taxonomic scales, assembling them is a tedious and costly process; predictive methods can help circumvent the limitations in data deficiencies, by providing a first approximation of metawebs. One way to improve our ability to predict metawebs is to maximize available information by using graph embeddings, as opposed to an exhaustive list of species interactions. Graph embedding is an emerging field in machine learning that holds great potential for ecological problems. Here, we outline how the challenges associated with inferring metawebs line-up with the advantages of graph embeddings; followed by a discussion as to how the choice of the species pool has consequences on the reconstructed network, specifically as to the role of human-made (or arbitrarily assigned) boundaries and how these may influence ecological hypotheses.;,citation_author=Tanya Strydom;,citation_author=Salomé Bouskila;,citation_author=Francis Banville;,citation_author=Ceres Barros;,citation_author=Dominique Caron;,citation_author=Maxwell J. Farrell;,citation_author=Marie-Josée Fortin;,citation_author=Benjamin Mercier;,citation_author=Laura J. Pollock;,citation_author=Rogini Runghen;,citation_author=Giulio V. Dalla Riva;,citation_author=Timothée Poisot;,citation_publication_date=2023;,citation_cover_date=2023;,citation_year=2023;,citation_issue=12;,citation_doi=10.1111/2041-210X.14228;,citation_issn=2041-210X;,citation_volume=14;,citation_language=en-US;,citation_journal_title=Methods in Ecology and Evolution;">
<meta name="citation_reference" content="citation_title=A roadmap towards predicting species interaction networks (across space and time);,citation_abstract=Networks of species interactions underpin numerous ecosystem processes, but comprehensively sampling these interactions is difficult. Interactions intrinsically vary across space and time, and given the number of species that compose ecological communities, it can be tough to distinguish between a true negative (where two species never interact) from a false negative (where two species have not been observed interacting even though they actually do). Assessing the likelihood of interactions between species is an imperative for several fields of ecology. This means that to predict interactions between species—and to describe the structure, variation, and change of the ecological networks they form—we need to rely on modelling tools. Here, we provide a proof-of-concept, where we show how a simple neural network model makes accurate predictions about species interactions given limited data. We then assess the challenges and opportunities associated with improving interaction predictions, and provide a conceptual roadmap forward towards predictive models of ecological networks that is explicitly spatial and temporal. We conclude with a brief primer on the relevant methods and tools needed to start building these models, which we hope will guide this research programme forward. This article is part of the theme issue “Infectious disease macroecology: parasite diversity and dynamics across the globe”.;,citation_author=Tanya Strydom;,citation_author=Michael D. Catchen;,citation_author=Francis Banville;,citation_author=Dominique Caron;,citation_author=Gabriel Dansereau;,citation_author=Philippe Desjardins-Proulx;,citation_author=Norma R. Forero-Muñoz;,citation_author=Gracielle Higino;,citation_author=Benjamin Mercier;,citation_author=Andrew Gonzalez;,citation_author=Dominique Gravel;,citation_author=Laura Pollock;,citation_author=Timothée Poisot;,citation_publication_date=2021-11;,citation_cover_date=2021-11;,citation_year=2021;,citation_issue=1837;,citation_doi=10.1098/rstb.2021.0063;,citation_volume=376;,citation_journal_title=Philosophical Transactions of the Royal Society B: Biological Sciences;,citation_publisher=Royal Society;">
<meta name="citation_reference" content="citation_title=SVD Entropy Reveals the High Complexity of Ecological Networks;,citation_abstract=Quantifying the complexity of ecological networks has remained elusive. Primarily, complexity has been defined on the basis of the structural (or behavioural) complexity of the system. These definitions ignore the notion of ’physical complexity’, which can measure the amount of information contained in an ecological network, and how difficult it would be to compress. We present relative rank deficiency and SVD entropy as measures of ’external’ and ’internal’ complexity respectively. Using bipartite ecological networks, we find that they all show a very high, almost maximal, physical complexity. Pollination networks, in particular, are more complex when compared to other types of interactions. In addition, we find that SVD entropy relates to other structural measures of complexity (nestedness, connectance, and spectral radius), but does not inform about the resilience of a network when using simulated extinction cascades, which has previously been reported for structural measures of complexity. We argue that SVD entropy provides a fundamentally more ’correct’ measure of network complexity and should be added to the toolkit of descriptors of ecological networks moving forward.;,citation_author=Tanya Strydom;,citation_author=Giulio V. Dalla Riva;,citation_author=Timothée Poisot;,citation_publication_date=2021;,citation_cover_date=2021;,citation_year=2021;,citation_doi=10.3389/fevo.2021.623141;,citation_issn=2296-701X;,citation_volume=9;,citation_language=en-US;,citation_journal_title=Frontiers in Ecology and Evolution;,citation_publisher=Frontiers;">
<meta name="citation_reference" content="citation_title=A practical guide to selecting models for exploration, inference, and prediction in ecology;,citation_abstract=Selecting among competing statistical models is a core challenge in science. However, the many possible approaches and techniques for model selection, and the conflicting recommendations for their use, can be confusing. We contend that much confusion surrounding statistical model selection results from failing to first clearly specify the purpose of the analysis. We argue that there are three distinct goals for statistical modeling in ecology: data exploration, inference, and prediction. Once the modeling goal is clearly articulated, an appropriate model selection procedure is easier to identify. We review model selection approaches and highlight their strengths and weaknesses relative to each of the three modeling goals. We then present examples of modeling for exploration, inference, and prediction using a time series of butterfly population counts. These show how a model selection approach flows naturally from the modeling goal, leading to different models selected for different purposes, even with exactly the same data set. This review illustrates best practices for ecologists and should serve as a reminder that statistical recipes cannot substitute for critical thinking or for the use of independent data to test hypotheses and validate predictions.;,citation_author=Andrew T. Tredennick;,citation_author=Giles Hooker;,citation_author=Stephen P. Ellner;,citation_author=Peter B. Adler;,citation_publication_date=2021;,citation_cover_date=2021;,citation_year=2021;,citation_issue=6;,citation_doi=10.1002/ecy.3336;,citation_issn=1939-9170;,citation_volume=102;,citation_language=en-US;,citation_journal_title=Ecology;">
<meta name="citation_reference" content="citation_title=Inferring the mammal tree: Species-level sets of phylogenies for questions in ecology, evolution, and conservation;,citation_abstract=Big, time-scaled phylogenies are fundamental to connecting evolutionary processes to modern biodiversity patterns. Yet inferring reliable phylogenetic trees for thousands of species involves numerous trade-offs that have limited their utility to comparative biologists. To establish a robust evolutionary timescale for all approximately 6,000 living species of mammals, we developed credible sets of trees that capture root-to-tip uncertainty in topology and divergence times. Our “backbone-and-patch” approach to tree building applies a newly assembled 31-gene supermatrix to two levels of Bayesian inference: (1) backbone relationships and ages among major lineages, using fossil node or tip dating, and (2) species-level “patch” phylogenies with nonoverlapping in-groups that each correspond to one representative lineage in the backbone. Species unsampled for DNA are either excluded (“DNA-only” trees) or imputed within taxonomic constraints using branch lengths drawn from local birth–death models (“completed” trees). Joining time-scaled patches to backbones results in species-level trees of extant Mammalia with all branches estimated under the same modeling framework, thereby facilitating rate comparisons among lineages as disparate as marsupials and placentals. We compare our phylogenetic trees to previous estimates of mammal-wide phylogeny and divergence times, finding that (1) node ages are broadly concordant among studies, and (2) recent (tip-level) rates of speciation are estimated more accurately in our study than in previous “supertree” approaches, in which unresolved nodes led to branch-length artifacts. Credible sets of mammalian phylogenetic history are now available for download at http://vertlife.org/phylosubsets, enabling investigations of long-standing questions in comparative biology.;,citation_author=Nathan S. Upham;,citation_author=Jacob A. Esselstyn;,citation_author=Walter Jetz;,citation_publication_date=2019-12;,citation_cover_date=2019-12;,citation_year=2019;,citation_issue=12;,citation_doi=10.1371/journal.pbio.3000494;,citation_issn=1545-7885;,citation_volume=17;,citation_language=en-US;,citation_journal_title=PLOS Biology;,citation_publisher=Public Library of Science;">
<meta name="citation_reference" content="citation_title=A bioenergetic framework for aboveground terrestrial food webs;,citation_abstract=Bioenergetic approaches have been greatly influential for understanding community functioning and stability and predicting effects of environmental changes on biodiversity. These approaches use allometric relationships to establish species’ trophic interactions and consumption rates and have been successfully applied to aquatic ecosystems. Terrestrial ecosystems, where body mass is less predictive of plant–consumer interactions, present inherent challenges that these models have yet to meet. Here, we discuss the processes governing terrestrial plant–consumer interactions and develop a bioenergetic framework integrating those processes. Our framework integrates bioenergetics specific to terrestrial plants and their consumers within a food web approach while also considering mutualistic interactions. Such a framework is poised to advance our understanding of terrestrial food webs and to predict their responses to environmental changes.;,citation_author=Fernanda S. Valdovinos;,citation_author=Kayla R. S. Hale;,citation_author=Sabine Dritz;,citation_author=Paul R. Glaum;,citation_author=Kevin S. McCann;,citation_author=Sophia M. Simon;,citation_author=Elisa Thébault;,citation_author=William C. Wetzel;,citation_author=Kate L. Wootton;,citation_author=Justin D. Yeakel;,citation_publication_date=2023-03;,citation_cover_date=2023-03;,citation_year=2023;,citation_issue=3;,citation_doi=10.1016/j.tree.2022.11.004;,citation_issn=0169-5347;,citation_volume=38;,citation_journal_title=Trends in Ecology &amp;amp;amp; Evolution;">
<meta name="citation_reference" content="citation_title=Simple rules yield complex food webs;,citation_abstract=Several of the most ambitious theories in ecology1,2,3,4,5,6,7,8,9,10,11,12,13,14 describe food webs that document the structure of strong and weak trophic links9 that is responsible for ecological dynamics among diverse assemblages of species4,11,12,13. Early mechanism-based theory asserted that food webs have little omnivory and several properties that are independent of species richness1,2,3,4,6. This theory was overturned by empirical studies that found food webs to be much more complex5,7,8,9,14,15,16,17,18, but these studies did not provide mechanistic explanations for the complexity9. Here we show that a remarkably simple model fills this scientific void by successfully predicting key structural properties of the most complex and comprehensive food webs in the primary literature. These properties include the fractions of species at top, intermediate and basal trophic levels, the means and variabilities of generality, vulnerability and food-chain length, and the degrees of cannibalism, omnivory, looping and trophic similarity. Using only two empirical parameters, species number and connectance, our “niche model” extends the existing “cascade model”3,19 and improves its fit ten-fold by constraining species to consume a contiguous sequence of prey in a one-dimensional trophic niche20.;,citation_author=Richard J. Williams;,citation_author=Neo D. Martinez;,citation_publication_date=2000-03;,citation_cover_date=2000-03;,citation_year=2000;,citation_issue=6774;,citation_doi=10.1038/35004572;,citation_issn=1476-4687;,citation_volume=404;,citation_language=en-US;,citation_journal_title=Nature;,citation_publisher=Nature Publishing Group;">
<meta name="citation_reference" content="citation_title=Success and its limits among structural models of complex food webs;,citation_abstract=1 Following the development of the relatively successful niche model, several other simple structural food web models have been proposed. These models predict the detailed structure of complex food webs given only two input parameters, the numbers of species and the number of feeding links among them. 2 The models claim different degrees of success but have not been compared consistently with each other or with the empirical data. We compared the performance of five structural models rigorously against 10 empirical food webs from a variety of aquatic and terrestrial habitats containing 25–92 species and 68–997 links. 3 All models include near-hierarchical ordering of species’ consumption and have identical distributions of the number of prey of each consumer species, but differ in the extent to which species’ diets are required to be contiguous and the rules used to assign feeding links. 4 The models perform similarly on a range of food-web properties, including the fraction of top, intermediate and basal species, the standard deviations of generality and connectivity and the fraction of herbivores and omnivores. 5 For other properties, including the standard deviation of vulnerability, the fraction of cannibals and species in loops, mean trophic level, path length, clustering coefficient, maximum similarity and diet discontinuity, there are significant differences in the performance of the different models. 6 While the empirical data do not support the niche model’s assumption of diet contiguity, models which relax this assumption all have worse overall performance than the niche model. All the models under-estimate severely the fraction of species that are herbivores and exhibit other important failures that need to be addressed in future research.;,citation_author=Richard J. Williams;,citation_author=Neo D. Martinez;,citation_publication_date=2008;,citation_cover_date=2008;,citation_year=2008;,citation_issue=3;,citation_doi=10.1111/j.1365-2656.2008.01362.x;,citation_issn=1365-2656;,citation_volume=77;,citation_language=en-US;,citation_journal_title=Journal of Animal Ecology;">
<meta name="citation_reference" content="citation_title=EltonTraits 1.0: Species-level foraging attributes of the world’s birds and mammals;,citation_abstract=Species are characterized by physiological, behavioral, and ecological attributes that are all subject to varying evolutionary and ecological constraints and jointly determine species’ role and function in ecosystems. Attributes such as diet, foraging strata, foraging time, and body size, in particular, characterize a large portion of the “Eltonian” niches of species. Here we present a global species-level compilation of these key attributes for all 9993 and 5400 extant bird and mammal species derived from key literature sources. Global handbooks and monographs allowed the consistent sourcing of attributes for most species. For diet and foraging stratum we followed a defined protocol to translate the verbal descriptions into standardized, semiquantitative information about relative importance of different categories. Together with body size (continuous) and activity time (categorical) this enables a much finer distinction of species’ foraging ecology than typical categorical guild assignments allow. Attributes lacking information for specific species are flagged, and interpolated values based on taxonomy are provided instead. The presented data set is limited by, among others, these select cases missing observed data, by errors and uncertainty in the expert assessment as presented in the literature, and by the lack of intraspecific information. However, the standardized and transparent nature and complete global coverage of the data set should support an array of potential studies in biogeography, community ecology, macroevolution, global change biology, and conservation. Potential uses include comparative work involving these traits as focal or secondary variables, ecological research on the trait or trophic structure of communities, or conservation science concerned with the loss of function among species or in ecosystems in a changing world. We hope that this publication will spur the sharing, collaborative curation, and extension of data to the benefit of a more integrative, rigorous, and global biodiversity science.;,citation_author=Hamish Wilman;,citation_author=Jonathan Belmaker;,citation_author=Jennifer Simpson;,citation_author=Carolina Rosa;,citation_author=Marcelo M. Rivadeneira;,citation_author=Walter Jetz;,citation_publication_date=2014;,citation_cover_date=2014;,citation_year=2014;,citation_issue=7;,citation_doi=10.1890/13-1917.1;,citation_issn=1939-9170;,citation_volume=95;,citation_language=en-US;,citation_journal_title=Ecology;">
<meta name="citation_reference" content="citation_title=Collapse of an ecological network in Ancient Egypt;,citation_author=Justin D. Yeakel;,citation_author=Mathias M. Pires;,citation_author=Lars Rudolf;,citation_author=Nathaniel J. Dominy;,citation_author=Paul L. Koch;,citation_author=Paulo R. Guimarães;,citation_author=Thilo Gross;,citation_publication_date=2014-10;,citation_cover_date=2014-10;,citation_year=2014;,citation_issue=40;,citation_doi=10.1073/pnas.1408471111;,citation_volume=111;,citation_journal_title=PNAS;">
</head>

<body>

<header id="title-block-header" class="quarto-title-block default toc-left page-columns page-full">
  <div class="quarto-title-banner page-columns page-full">
    <div class="quarto-title column-body">
      <h1 class="title">T is for Topology</h1>
          </div>

    
    <div class="quarto-title-meta-container">
      <div class="quarto-title-meta-column-start">
            <div class="quarto-title-meta-author">
          <div class="quarto-title-meta-heading">Authors</div>
          <div class="quarto-title-meta-heading">Affiliation</div>
          
                <div class="quarto-title-meta-contents">
            <p class="author">Tanya Strydom <a href="mailto:t.strydom@sheffield.ac.uk" class="quarto-title-author-email"><i class="bi bi-envelope"></i></a> <a href="https://orcid.org/0000-0001-6067-1349" class="quarto-title-author-orcid"> <img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAA2ZpVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADw/eHBhY2tldCBiZWdpbj0i77u/IiBpZD0iVzVNME1wQ2VoaUh6cmVTek5UY3prYzlkIj8+IDx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8iIHg6eG1wdGs9IkFkb2JlIFhNUCBDb3JlIDUuMC1jMDYwIDYxLjEzNDc3NywgMjAxMC8wMi8xMi0xNzozMjowMCAgICAgICAgIj4gPHJkZjpSREYgeG1sbnM6cmRmPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5LzAyLzIyLXJkZi1zeW50YXgtbnMjIj4gPHJkZjpEZXNjcmlwdGlvbiByZGY6YWJvdXQ9IiIgeG1sbnM6eG1wTU09Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC9tbS8iIHhtbG5zOnN0UmVmPSJodHRwOi8vbnMuYWRvYmUuY29tL3hhcC8xLjAvc1R5cGUvUmVzb3VyY2VSZWYjIiB4bWxuczp4bXA9Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC8iIHhtcE1NOk9yaWdpbmFsRG9jdW1lbnRJRD0ieG1wLmRpZDo1N0NEMjA4MDI1MjA2ODExOTk0QzkzNTEzRjZEQTg1NyIgeG1wTU06RG9jdW1lbnRJRD0ieG1wLmRpZDozM0NDOEJGNEZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wTU06SW5zdGFuY2VJRD0ieG1wLmlpZDozM0NDOEJGM0ZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wOkNyZWF0b3JUb29sPSJBZG9iZSBQaG90b3Nob3AgQ1M1IE1hY2ludG9zaCI+IDx4bXBNTTpEZXJpdmVkRnJvbSBzdFJlZjppbnN0YW5jZUlEPSJ4bXAuaWlkOkZDN0YxMTc0MDcyMDY4MTE5NUZFRDc5MUM2MUUwNEREIiBzdFJlZjpkb2N1bWVudElEPSJ4bXAuZGlkOjU3Q0QyMDgwMjUyMDY4MTE5OTRDOTM1MTNGNkRBODU3Ii8+IDwvcmRmOkRlc2NyaXB0aW9uPiA8L3JkZjpSREY+IDwveDp4bXBtZXRhPiA8P3hwYWNrZXQgZW5kPSJyIj8+84NovQAAAR1JREFUeNpiZEADy85ZJgCpeCB2QJM6AMQLo4yOL0AWZETSqACk1gOxAQN+cAGIA4EGPQBxmJA0nwdpjjQ8xqArmczw5tMHXAaALDgP1QMxAGqzAAPxQACqh4ER6uf5MBlkm0X4EGayMfMw/Pr7Bd2gRBZogMFBrv01hisv5jLsv9nLAPIOMnjy8RDDyYctyAbFM2EJbRQw+aAWw/LzVgx7b+cwCHKqMhjJFCBLOzAR6+lXX84xnHjYyqAo5IUizkRCwIENQQckGSDGY4TVgAPEaraQr2a4/24bSuoExcJCfAEJihXkWDj3ZAKy9EJGaEo8T0QSxkjSwORsCAuDQCD+QILmD1A9kECEZgxDaEZhICIzGcIyEyOl2RkgwAAhkmC+eAm0TAAAAABJRU5ErkJggg=="></a></p>
          </div>
                <div class="quarto-title-meta-contents">
                    <p class="affiliation">
                        School of Biosciences, University of Sheffield
                      </p>
                  </div>
                      <div class="quarto-title-meta-contents">
            <p class="author">Andrew P. Beckerman <a href="https://orcid.org/0000-0002-7859-8394" class="quarto-title-author-orcid"> <img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAA2ZpVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADw/eHBhY2tldCBiZWdpbj0i77u/IiBpZD0iVzVNME1wQ2VoaUh6cmVTek5UY3prYzlkIj8+IDx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8iIHg6eG1wdGs9IkFkb2JlIFhNUCBDb3JlIDUuMC1jMDYwIDYxLjEzNDc3NywgMjAxMC8wMi8xMi0xNzozMjowMCAgICAgICAgIj4gPHJkZjpSREYgeG1sbnM6cmRmPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5LzAyLzIyLXJkZi1zeW50YXgtbnMjIj4gPHJkZjpEZXNjcmlwdGlvbiByZGY6YWJvdXQ9IiIgeG1sbnM6eG1wTU09Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC9tbS8iIHhtbG5zOnN0UmVmPSJodHRwOi8vbnMuYWRvYmUuY29tL3hhcC8xLjAvc1R5cGUvUmVzb3VyY2VSZWYjIiB4bWxuczp4bXA9Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC8iIHhtcE1NOk9yaWdpbmFsRG9jdW1lbnRJRD0ieG1wLmRpZDo1N0NEMjA4MDI1MjA2ODExOTk0QzkzNTEzRjZEQTg1NyIgeG1wTU06RG9jdW1lbnRJRD0ieG1wLmRpZDozM0NDOEJGNEZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wTU06SW5zdGFuY2VJRD0ieG1wLmlpZDozM0NDOEJGM0ZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wOkNyZWF0b3JUb29sPSJBZG9iZSBQaG90b3Nob3AgQ1M1IE1hY2ludG9zaCI+IDx4bXBNTTpEZXJpdmVkRnJvbSBzdFJlZjppbnN0YW5jZUlEPSJ4bXAuaWlkOkZDN0YxMTc0MDcyMDY4MTE5NUZFRDc5MUM2MUUwNEREIiBzdFJlZjpkb2N1bWVudElEPSJ4bXAuZGlkOjU3Q0QyMDgwMjUyMDY4MTE5OTRDOTM1MTNGNkRBODU3Ii8+IDwvcmRmOkRlc2NyaXB0aW9uPiA8L3JkZjpSREY+IDwveDp4bXBtZXRhPiA8P3hwYWNrZXQgZW5kPSJyIj8+84NovQAAAR1JREFUeNpiZEADy85ZJgCpeCB2QJM6AMQLo4yOL0AWZETSqACk1gOxAQN+cAGIA4EGPQBxmJA0nwdpjjQ8xqArmczw5tMHXAaALDgP1QMxAGqzAAPxQACqh4ER6uf5MBlkm0X4EGayMfMw/Pr7Bd2gRBZogMFBrv01hisv5jLsv9nLAPIOMnjy8RDDyYctyAbFM2EJbRQw+aAWw/LzVgx7b+cwCHKqMhjJFCBLOzAR6+lXX84xnHjYyqAo5IUizkRCwIENQQckGSDGY4TVgAPEaraQr2a4/24bSuoExcJCfAEJihXkWDj3ZAKy9EJGaEo8T0QSxkjSwORsCAuDQCD+QILmD1A9kECEZgxDaEZhICIzGcIyEyOl2RkgwAAhkmC+eAm0TAAAAABJRU5ErkJggg=="></a></p>
          </div>
                <div class="quarto-title-meta-contents">
                    <p class="affiliation">
                        School of Biosciences, University of Sheffield
                      </p>
                  </div>
                    </div>
        
        <div class="quarto-title-meta">

                      
                <div>
            <div class="quarto-title-meta-heading">Published</div>
            <div class="quarto-title-meta-contents">
              <p class="date">March 18, 2024</p>
            </div>
          </div>
          
                
              </div>
      </div>
      <div class="quarto-title-meta-column-end quarto-other-formats-target">
      <div class="quarto-alternate-formats"><div class="quarto-title-meta-heading">Other Formats</div><div class="quarto-title-meta-contents"><p><a href="index.docx"><i class="bi bi-file-word"></i>MS Word</a></p></div><div class="quarto-title-meta-contents"><p><a href="index.pdf"><i class="bi bi-file-pdf"></i>PDF</a></p></div></div></div>
    </div>

    <div>
      <div class="abstract">
        <div class="block-title">Abstract</div>
        <p>There are many reasons one might want to generate a network and there are many tools on the market that might make that possible. However not all tools are created equally and there is reason to assume that not all networks will suit most purposes. Here the aim is to compare and contrast the different topology generating tools that are on the market and see where they shine and where they fall flat. There probably isn’t one model to rule them all but it doesn’t mean that we shouldn’t be critical when we think about the model we want to use.</p>
      </div>
    </div>

    <div>
      <div class="keywords">
        <div class="block-title">Keywords</div>
        <p>food web, network construction</p>
      </div>
    </div>

    <div class="quarto-other-links-text-target">
    <div class="quarto-code-links"><div class="quarto-title-meta-heading">Code Links</div><div class="quarto-title-meta-contents"><span><a href="https://github.com/BecksLab/topology_generators"><i class="bi bi-github"></i>GitHub</a></span></div></div></div>  </div>
</header><div id="quarto-content" class="page-columns page-rows-contents page-layout-article toc-left">
<div id="quarto-sidebar-toc-left" class="sidebar toc-left">
  <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#introduction" id="toc-introduction" class="nav-link active" data-scroll-target="#introduction"><span class="header-section-number">1</span> Introduction</a>
  <ul class="collapse">
  <li><a href="#model-families" id="toc-model-families" class="nav-link" data-scroll-target="#model-families"><span class="header-section-number">1.1</span> Model families</a></li>
  <li><a href="#model-benchmarking" id="toc-model-benchmarking" class="nav-link" data-scroll-target="#model-benchmarking"><span class="header-section-number">1.2</span> Model benchmarking</a></li>
  </ul></li>
  <li><a href="#sec-data-methods" id="toc-sec-data-methods" class="nav-link" data-scroll-target="#sec-data-methods"><span class="header-section-number">2</span> Data &amp; Methods</a>
  <ul class="collapse">
  <li><a href="#selecting-models" id="toc-selecting-models" class="nav-link" data-scroll-target="#selecting-models"><span class="header-section-number">2.1</span> Selecting models</a></li>
  <li><a href="#datasets-used" id="toc-datasets-used" class="nav-link" data-scroll-target="#datasets-used"><span class="header-section-number">2.2</span> Datasets used</a></li>
  <li><a href="#model-comparison" id="toc-model-comparison" class="nav-link" data-scroll-target="#model-comparison"><span class="header-section-number">2.3</span> Model comparison</a></li>
  </ul></li>
  <li><a href="#results" id="toc-results" class="nav-link" data-scroll-target="#results"><span class="header-section-number">3</span> Results</a>
  <ul class="collapse">
  <li><a href="#quantitative-stuff" id="toc-quantitative-stuff" class="nav-link" data-scroll-target="#quantitative-stuff"><span class="header-section-number">3.1</span> Quantitative stuff</a></li>
  </ul></li>
  <li><a href="#discussion" id="toc-discussion" class="nav-link" data-scroll-target="#discussion"><span class="header-section-number">4</span> Discussion</a></li>
  <li><a href="#references" id="toc-references" class="nav-link" data-scroll-target="#references">References</a></li>
  </ul>
<div class="quarto-alternate-notebooks"><h2>Notebooks</h2><ul><li><a href="notebooks/model_descriptions-preview.html"><i class="bi bi-journal-code"></i>Extended Model Description</a></li><li><a href="notebooks/model_quantitative.embed-preview.html"><i class="bi bi-journal-code"></i>Quantitative approach to topology generators</a></li></ul></div></nav>
</div>
<div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
</div>
<main class="content quarto-banner-title-block" id="quarto-document-content">



  


<section id="introduction" class="level2" data-number="1">
<h2 data-number="1" class="anchored" data-anchor-id="introduction"><span class="header-section-number">1</span> Introduction</h2>
<ul>
<li><p>In order to construct a ‘perfect’ network <em>i.e.,</em> one which <em>perfectly</em> captures the dynamics for a specific community one needs to consider and account for many different moving parts (<em>e.g.,</em>). So when developing a model it makes sense that you prioritise the aspect of the prediction/construction task that has the most value for your research goal, acknowledging that a model might fall short in others. The thing is that with the growing suite of approaches to generating networks it is important that we don’t lose sight of the core philosophy behind the model we use and to ensure that we are using the model best suited to what we want to be accomplishing.</p></li>
<li><p>It is perhaps useful to start with asking why do we want/need models to generate networks. This can be broadly thought of to fall into two categories. Build networks because we want to build concepts vs build networks because we want specificity. Broadly this means that we either want to construct/predict a collection of interactions (generate networks) or a network of interactions (predict interactions).</p></li>
</ul>
<p>Arguably the need for methods and tools for constructing interaction networks arises from two different (but still aligned) places of interest within the field of network ecology. On the one side sits the researcher who is interested in generating a set of ecologically plausible but not necessarily realised ‘in the field’ networks for the purpose of running further simulations (<em>e.g.,</em> extinction sim) or understanding some higher-level process/concept (<em>e.g.,</em> energetics). This researcher is contrasted by one that is interested in constructing real-world, location specific, interaction data in lieu of inventorying interactions based n observations made in the field <span class="citation" data-cites="strydomRoadmapPredictingSpecies2021">(<a href="#ref-strydomRoadmapPredictingSpecies2021" role="doc-biblioref">Strydom et al. 2021</a>)</span>. Of course these two categories are not two distinct, mutually exclusive, groups but can rather be viewed as operating on a gradient ranging from a need for generality (<em>i.e.,</em> creating a network that, when taken in aggregate, the distribution of links (interactions) between species are ecologically plausible) to a need for specificity (local-level predictions between specific species).</p>
<ul>
<li><p>Brief history of the development of tools within the context of the two different fields? Sort of where the theory/body of work was based and how that has changed?</p>
<ul>
<li><p>Core mechanistic differences that models will work at — some are really concerned about (and thus constrained by) structure, others are more mechanistic in nature <em>i.e.,</em> species <em>a</em> has the capacity to eat species <em>b</em> because traits</p></li>
<li><p>In certain situations structure is ‘enough’ but there may be use cases where we are really interested in the node-level interactions <em>i.e.,</em> species identity is a thing we care about and need to be able to retrieve specific interactions at specific nodes correctly.</p></li>
<li><p>What is the purpose of generating a network? Is it an element of a bigger question we are asking, <em>e.g.,</em> I want to generate a series of networks to do some extinction simulations/bioenergetic stuff OR are we looking for a ‘final product’ network that is relevant to a specific location? (this can still be broad in geographic scope).</p></li>
</ul></li>
<li><p>A breakdown of wanting to generate a network; statement of need and core philosophies</p></li>
<li><p>A breakdown of wanting to predict an interaction; statement of need <span class="citation" data-cites="jordanoSamplingNetworksEcological2016 jordanoChasingEcologicalInteractions2016 poisotGlobalKnowledgeGaps2021">(<a href="#ref-jordanoSamplingNetworksEcological2016" role="doc-biblioref">Jordano 2016b</a>, <a href="#ref-jordanoChasingEcologicalInteractions2016" role="doc-biblioref">2016a</a>; <a href="#ref-poisotGlobalKnowledgeGaps2021" role="doc-biblioref">Poisot et al. 2021</a>)</span> and core philosophies (trait-matching, coexistence)</p></li>
<li><p>Stands to reason then that we have developed methods that specialise in one or the other. Which comes at a cost of ‘performance’ in other aspects. Knowing how the different model families stack up to each other is thus valuable and that is kinda what we are trying to achieve.</p></li>
</ul>
<p><span class="citation" data-cites="cohenStochasticTheoryCommunity1985">Joel E. Cohen, Newman, and Steele (<a href="#ref-cohenStochasticTheoryCommunity1985" role="doc-biblioref">1985</a>)</span> states that <em>“[Their] approach is more like gross anatomy than like physiology… that is, the gross anatomy is frozen, rather than in motion.”</em>.</p>
<p>Interestingly <span class="citation" data-cites="williamsSuccessItsLimits2008">Williams and Martinez (<a href="#ref-williamsSuccessItsLimits2008" role="doc-biblioref">2008</a>)</span> also explicitly talk about <em>structural</em> food-web models in their introduction… so how I see it that means that there has always been this inherent acknowledgement that models are functioning at a specific ‘network level’.</p>
<section id="model-families" class="level3" data-number="1.1">
<h3 data-number="1.1" class="anchored" data-anchor-id="model-families"><span class="header-section-number">1.1</span> Model families</h3>
<p>Given the large number of models that have been developed it is perhaps more meaningful to group models into families with the idea that models from the same family will yield similar results because they play by similar rules. These rules referring to the underlying philosophy as to what structures either networks or the interactions within them (see <a href="#fig-concept" class="quarto-xref">Figure&nbsp;1</a> panel A).</p>
<div id="fig-concept" class="quarto-figure quarto-figure-center quarto-float anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-concept-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="images/concept.jpeg" class="lightbox" data-gallery="quarto-lightbox-gallery-1" data-glightbox="description: .lightbox-desc-1"><img src="images/concept.jpeg" class="img-fluid figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-concept-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;1: Conceptual figure of the ‘network prediction’. Panel A shows where the model families fall in the the context of being models that predict networks or models that predict interactions space. Panel B serves to highlight the characteristics one might like to ‘test’/benchmark for a model based on it being either a network or interaction predicting model
</figcaption>
</figure>
</div>
<p><strong>Null models:</strong> The interactions between species occurs regardless of the identity of the species (<em>i.e.,</em> species have no agency) and links are randomly distributed throughout the network. There is however the assumption that a network will be constrained by the number of links. Type I <span class="citation" data-cites="fortunaHabitatLossStructure2006">(<a href="#ref-fortunaHabitatLossStructure2006" role="doc-biblioref">Fortuna and Bascompte 2006</a>)</span>, where interactions happen proportionally to connectance and Type II <span class="citation" data-cites="bascompteNestedAssemblyPlantanimal2003">(<a href="#ref-bascompteNestedAssemblyPlantanimal2003" role="doc-biblioref">Bascompte et al. 2003</a>)</span>, where interactions happen proportionally to the joint degree of the two species involved. These two models are equivalent to the Erdos-Renyi and Configuration models <span class="citation" data-cites="newmanNetworksIntroduction2010">(<a href="#ref-newmanNetworksIntroduction2010" role="doc-biblioref">Newman 2010</a>)</span> (check that though).</p>
<p><strong>Neutral models:</strong> Based on the theory that interactions occur as the result of the abundance of species (<em>i.e.,</em> the species still has no agency but its abundance does?). See <span class="citation" data-cites="pomeranzInferringPredatorPrey2019">Pomeranz et al. (<a href="#ref-pomeranzInferringPredatorPrey2019" role="doc-biblioref">2019</a>)</span></p>
<p><strong>Resource models:</strong> In the context of network generating models this is perhaps the most well known family of models. Essentially these models can be viewed as being based on the idea of resource partitioning and that the number of links scale with species richness (maybe not directly that but these models are link constrained). That is there is some sort of hierarchical feeding based on how a ‘resource’ is partitioned. This includes the cascade model <span class="citation" data-cites="cohenCommunityFoodWebs1990">(<a href="#ref-cohenCommunityFoodWebs1990" role="doc-biblioref">Joel E. Cohen, Briand, and Newman 1990</a>)</span>, which much like the name suggests the cascade model rests on the idea that species feed on one another in a hierarchical manner. This rests on the assumption that the links within a network are variably distributed across the network; with the proportion of links decreasing as one moves up the trophic levels (<em>i.e.,</em> ‘many’ prey and ‘few’ predators). The niche model <span class="citation" data-cites="williamsSimpleRulesYield2000">(<a href="#ref-williamsSimpleRulesYield2000" role="doc-biblioref">Williams and Martinez 2000</a>)</span> introduces the idea that species interactions are based on the ‘feeding niche’ of a species. Broadly, all species are randomly assigned a ‘feeding niche’ and all species that fall in this niche can be consumed by that species. Finally, the nested hierarchy model<span class="citation" data-cites="cattinPhylogeneticConstraintsAdaptation2004">(<a href="#ref-cattinPhylogeneticConstraintsAdaptation2004" role="doc-biblioref">Cattin et al. 2004</a>)</span> <strong>TODO</strong>.</p>
<p><strong>Energetic models:</strong> Broadly this family of models is rooted in feeding theory and allocates the links between species based on energetics. This means that the model is focused on predicting not only the number of links in a network but also the arrangement of these links based on the diet breadth of a species. The diet breadth model <span class="citation" data-cites="beckermanForagingBiologyPredicts2006">(<a href="#ref-beckermanForagingBiologyPredicts2006" role="doc-biblioref">Beckerman, Petchey, and Warren 2006</a>)</span> as well as its allometrically scaled cousin the allometric diet breadth model (ADBM) <span class="citation" data-cites="petcheySizeForagingFood2008">(<a href="#ref-petcheySizeForagingFood2008" role="doc-biblioref">Petchey et al. 2008</a>)</span> determine links between species based on the energetic content, handling time, and density of species.</p>
<blockquote class="blockquote">
<p><span class="citation" data-cites="gravelInferringFoodWeb2013">Gravel et al. (<a href="#ref-gravelInferringFoodWeb2013" role="doc-biblioref">2013</a>)</span> also poses an interesting cross-over between the adbm and niche model.</p>
</blockquote>
<p><strong>Binary classifiers:</strong></p>
<p><strong>Graph embedding:</strong> This family of approaches has been extensively discussed in <span class="citation" data-cites="strydomGraphEmbeddingTransfer2023">Strydom et al. (<a href="#ref-strydomGraphEmbeddingTransfer2023" role="doc-biblioref">2023</a>)</span> but can be broadly explained as an approach that estimates latent features from observed networks that can be used to predict interactions. <span class="citation" data-cites="strydomFoodWebReconstruction2022">Strydom et al. (<a href="#ref-strydomFoodWebReconstruction2022" role="doc-biblioref">2022</a>)</span> presents a specific use case that is based on transfer learning and the idea that interactions are evolutionarily conserved and that we can use known networks, and this evolutionary relationship to predict interactions for a given species pool. <strong>TODO</strong> Log-ratio <span class="citation" data-cites="rohrModelingFoodWebs2010">(<a href="#ref-rohrModelingFoodWebs2010" role="doc-biblioref">Rohr et al. 2010</a>)</span></p>
<p><strong>Trait hierarchy:</strong> Here I envision models that present an <em>a priori</em> trait hierarchy that determines feeding links between species. That is, there is an element of ‘expert knowledge’ that also comes into play… Something like PFIM <span class="citation" data-cites="shawFrameworkReconstructingAncient2024">(<a href="#ref-shawFrameworkReconstructingAncient2024" role="doc-biblioref">Shaw et al. 2024</a>)</span> is what I imagine fitting in here…</p>
<p><strong>GLM</strong> <span class="citation" data-cites="caronAddressingEltonianShortfall2022">(<a href="#ref-caronAddressingEltonianShortfall2022" role="doc-biblioref">Caron et al. 2022</a>)</span>: To confirm</p>
<p><strong>Expert knowledge:</strong> (boots on the ground ecology)</p>
</section>
<section id="model-benchmarking" class="level3" data-number="1.2">
<h3 data-number="1.2" class="anchored" data-anchor-id="model-benchmarking"><span class="header-section-number">1.2</span> Model benchmarking</h3>
<ul>
<li><p>‘Testing’ the performance of a model is going to depend on some of the core limitations of the model itself thus it makes sense to think of two sets benchmarking rules for network and interaction prediction models respectively (see <a href="#fig-concept" class="quarto-xref">Figure&nbsp;1</a> panel B).</p></li>
<li><p>When it comes to network models we are concerned with the ‘preservation’ of structure and distribution of links across the network. For interaction models we want to ensure that we are able to retrieve interactions that really exist but also those that cannot exist (<em>sensu</em> forbidden links <span class="citation" data-cites="jordanoSamplingNetworksEcological2016">Jordano (<a href="#ref-jordanoSamplingNetworksEcological2016" role="doc-biblioref">2016b</a>)</span>)</p></li>
</ul>
<section id="benchmarking-network-models" class="level4" data-number="1.2.1">
<h4 data-number="1.2.1" class="anchored" data-anchor-id="benchmarking-network-models"><span class="header-section-number">1.2.1</span> Benchmarking network models</h4>
<ul>
<li>Maybe look at some of the historic papers that compare some of the ‘resource models’</li>
</ul>
</section>
<section id="benchmarking-interaction-models" class="level4" data-number="1.2.2">
<h4 data-number="1.2.2" class="anchored" data-anchor-id="benchmarking-interaction-models"><span class="header-section-number">1.2.2</span> Benchmarking interaction models</h4>
<ul>
<li><p>See <span class="citation" data-cites="poisotGuidelinesPredictionSpecies2023">Poisot (<a href="#ref-poisotGuidelinesPredictionSpecies2023" role="doc-biblioref">2023</a>)</span></p></li>
<li><p>Need to discuss the key differences and implications between predicting a metaweb (<em>sensu</em> <span class="citation" data-cites="dunneNetworkStructureFood2006">Jennifer A. Dunne (<a href="#ref-dunneNetworkStructureFood2006" role="doc-biblioref">2006</a>)</span>) and a network realisation. Maybe also <span class="citation" data-cites="poisotSpeciesWhyEcological2015">Poisot, Stouffer, and Gravel (<a href="#ref-poisotSpeciesWhyEcological2015" role="doc-biblioref">2015</a>)</span> that discuss how the local factors are going to play a role.</p></li>
</ul>
</section>
</section>
</section>
<section id="sec-data-methods" class="level2" data-number="2">
<h2 data-number="2" class="anchored" data-anchor-id="sec-data-methods"><span class="header-section-number">2</span> Data &amp; Methods</h2>
<section id="selecting-models" class="level3" data-number="2.1">
<h3 data-number="2.1" class="anchored" data-anchor-id="selecting-models"><span class="header-section-number">2.1</span> Selecting models</h3>
<p>This section depends on if we go the family route and where we introduce them. But a more extended description of each model can be found in the <code>Extended Model Description</code> notebook (I’m trying to work out how to embed this…)</p>
<p>I know tables are awful but in this case they may make more sense. Also I don’t think I’m at the point where I can say that the table is complete/comprehensive but it getting there Not sure about putting in some papers that have used the model - totes happy to drop those I think…</p>
<div id="tbl-history" class="quarto-float anchored">
<figure class="quarto-float quarto-float-tbl figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-tbl" id="tbl-history-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Table&nbsp;1: Lets make a table that gives an overview of the different topology generators that we will look at. Here I take ‘data-driven’ to refer to the need for ‘real world’ data. This can probably be approached in a different way though maybe?
</figcaption>
<div aria-describedby="tbl-history-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<table class="table">
<colgroup>
<col style="width: 16%">
<col style="width: 16%">
<col style="width: 16%">
<col style="width: 16%">
<col style="width: 16%">
<col style="width: 16%">
</colgroup>
<thead>
<tr class="header">
<th>Model</th>
<th>Core Mechanism</th>
<th>Predicts</th>
<th>Specificity</th>
<th>Interaction</th>
<th>Data-driven</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>random</td>
<td>random</td>
<td>networks</td>
<td>species agnostic</td>
<td>binary</td>
<td>no</td>
</tr>
<tr class="even">
<td>cascade</td>
<td>structural</td>
<td>networks</td>
<td>species agnostic</td>
<td>binary</td>
<td>no</td>
</tr>
<tr class="odd">
<td>niche</td>
<td>structural</td>
<td>networks</td>
<td>species agnostic</td>
<td>binary</td>
<td>no</td>
</tr>
<tr class="even">
<td>nested hierarchical</td>
<td>structural</td>
<td>networks</td>
<td>species agnostic</td>
<td>binary</td>
<td>no</td>
</tr>
<tr class="odd">
<td>ADBM</td>
<td>mechanistic</td>
<td>interactions</td>
<td>energetics</td>
<td>quantitative</td>
<td></td>
</tr>
<tr class="even">
<td>log-ratio</td>
<td></td>
<td>interactions</td>
<td></td>
<td></td>
<td></td>
</tr>
<tr class="odd">
<td>PFIM</td>
<td>mechanistic</td>
<td>interactions</td>
<td>trait based</td>
<td></td>
<td></td>
</tr>
<tr class="even">
<td>graph embedding</td>
<td>embedding</td>
<td>interactions</td>
<td>evolutionary</td>
<td>probabilistic</td>
<td>yes</td>
</tr>
<tr class="odd">
<td>trait model</td>
<td>mechanistic</td>
<td>interactions</td>
<td>trait based</td>
<td></td>
<td>yes</td>
</tr>
<tr class="even">
<td>matching</td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
</tr>
</tbody>
</table>
</div>
</figure>
</div>
</section>
<section id="datasets-used" class="level3" data-number="2.2">
<h3 data-number="2.2" class="anchored" data-anchor-id="datasets-used"><span class="header-section-number">2.2</span> Datasets used</h3>
<ul>
<li><p>For network models makes sense to drop datasets from Mangal</p></li>
<li><p>‘Elite’ number of datasets for interaction models</p></li>
</ul>
<blockquote class="blockquote">
<p>Here I think we need to span a variety of domains, at minimum aquatic and terrestrial but maybe there should be a ‘scale’ element as well <em>i.e.,</em> a regional and local network. I think there is going to be a ‘turning point’ where structural will take over from mechanistic in terms of performance. More specifically at local scales bioenergetic constraints (and co-occurrence) may play a bigger role in structuring a network whereas at the metaweb level then mechanistic may make more (since by default its about who can potentially interact and obviously not constrained by real-world scenarios) <em>sensu</em> <span class="citation" data-cites="caronTrophicInteractionModels2023">Caron et al. (<a href="#ref-caronTrophicInteractionModels2023" role="doc-biblioref">2023</a>)</span>. Although having said that I feel that contradicts the idea of backbones (<em>sensu</em> Bramon Mora (sp?) et al &amp; Stouffer et al) But that might be where we get the idea of core <em>structure</em> vs something like linkage density. So core things like trophic level/chain length will be conserved but connectance might not (I think I understand what I’m trying to say here)</p>
</blockquote>
<p>I think we should also use the Dunne <span class="citation" data-cites="dunneCompilationNetworkAnalyses2008">(<a href="#ref-dunneCompilationNetworkAnalyses2008" role="doc-biblioref">Jennifer A. Dunne et al. 2008</a>)</span>. Because 1) it gives the paleo-centric methods their moment in the sun and 2) I think it also brings up the interesting question of can we use modern structure to predict past ones?</p>
</section>
<section id="model-comparison" class="level3" data-number="2.3">
<h3 data-number="2.3" class="anchored" data-anchor-id="model-comparison"><span class="header-section-number">2.3</span> Model comparison</h3>
<p>For now the (still essentially pending) workflow/associated code can be found at the following repository <a href="https://github.com/BecksLab/topology_generators">BecksLab/topology_generators</a>. This will reflect that which is shown in panel <em>B</em> of <a href="#fig-concept" class="quarto-xref">Figure&nbsp;1</a>.</p>
<ul>
<li>Data ‘cost’ (some methods might need a lot lot of supporting data vs something very light weight)</li>
<li>I think it would be remiss to not also take into consideration computational cost</li>
<li>Something about the network output - I’m acknowledging my biases and saying that probabilistic (or <em>maybe</em> weighted) links are the way</li>
</ul>
<section id="network-models" class="level4" data-number="2.3.1">
<h4 data-number="2.3.1" class="anchored" data-anchor-id="network-models"><span class="header-section-number">2.3.1</span> Network models</h4>
<p>Want to compare real vs predicted and then get something that looks like <a href="#fig-topology" class="quarto-xref">Figure&nbsp;2</a></p>
<ul>
<li><em>Shape:</em> do the models construct tall ‘pencil’ vs flat ‘pancake’ networks (Beckerman 2024, pers comms), generality/vulnerability, chain length (?)</li>
<li><em>Structure:</em> Predicting ‘structure’ - SVD <span class="citation" data-cites="strydomSVDEntropyReveals2021">(<a href="#ref-strydomSVDEntropyReveals2021" role="doc-biblioref">Strydom, Dalla Riva, and Poisot 2021</a>)</span> but maybe something like nestedness as well (?)</li>
<li><em>Links:</em> are the number of links preserved (most network predicting are to some extend link constrained but useful to see)</li>
</ul>
</section>
<section id="interaction-models" class="level4" data-number="2.3.2">
<h4 data-number="2.3.2" class="anchored" data-anchor-id="interaction-models"><span class="header-section-number">2.3.2</span> Interaction models</h4>
<ul>
<li>Look at what <span class="citation" data-cites="poisotGuidelinesPredictionSpecies2023">Poisot (<a href="#ref-poisotGuidelinesPredictionSpecies2023" role="doc-biblioref">2023</a>)</span> has to say</li>
<li>% links correctly retrieved</li>
<li>TSS/ROC-AUC</li>
<li>Maybe same measures we use for the network models</li>
</ul>
</section>
<section id="pva-action-plan" class="level4" data-number="2.3.3">
<h4 data-number="2.3.3" class="anchored" data-anchor-id="pva-action-plan"><span class="header-section-number">2.3.3</span> PVA (action plan)</h4>
<ol type="1">
<li>Shortlist/finalise the different topo generators</li>
<li>collate/translate into <code>Julia</code>
<ul>
<li><em>e.g.,</em> some models wil be in SpeciesInteractionNetworks.jl (new EcoNet); I know (parts of) the transfer learning stuff is and the niche model</li>
<li>others will need to be coded out (the more simpler models should be easier)</li>
</ul></li>
<li>Curate networks for the different datasets/scenarios we select - I feel like there might be some scenarios that we can’t do all models for all datasets but maybe I’m being a pessimist.
<ul>
<li>Need to also think about where one might find the additional data for some of the models…
<ul>
<li>Body size: <span class="citation" data-cites="herbersteinAnimalTraitsCuratedAnimal2022">Herberstein et al. (<a href="#ref-herbersteinAnimalTraitsCuratedAnimal2022" role="doc-biblioref">2022</a>)</span> - Although maybe Andrew has strong thotsTM RE the one true body size database to rule them all…</li>
<li>Other trait sources: <span class="citation" data-cites="wilmanEltonTraitsSpecieslevelForaging2014">Wilman et al. (<a href="#ref-wilmanEltonTraitsSpecieslevelForaging2014" role="doc-biblioref">2014</a>)</span> and <span class="citation" data-cites="jonesPanTHERIASpecieslevelDatabase2009">Jones et al. (<a href="#ref-jonesPanTHERIASpecieslevelDatabase2009" role="doc-biblioref">2009</a>)</span></li>
<li>This is where we’ll get the paleo traits from if I’m correct <span class="citation" data-cites="bambachAutecologyFillingEcospace2007">Bambach, Bush, and Erwin (<a href="#ref-bambachAutecologyFillingEcospace2007" role="doc-biblioref">2007</a>)</span></li>
<li>Phylogeny stuff: <span class="citation" data-cites="uphamInferringMammalTree2019">Upham, Esselstyn, and Jetz (<a href="#ref-uphamInferringMammalTree2019" role="doc-biblioref">2019</a>)</span> (what we used for TL but its only mammals…) but I’m sure there will be others</li>
</ul></li>
<li>Also limitation of scope… <em>e.g.,</em> do we even dare to think about including plants/basal producers (see <em>e.g.,</em> <span class="citation" data-cites="valdovinosBioenergeticFrameworkAboveground2023">Valdovinos et al. (<a href="#ref-valdovinosBioenergeticFrameworkAboveground2023" role="doc-biblioref">2023</a>)</span>)</li>
<li>Taxonomic harmonisation - something to think about and check</li>
</ul></li>
</ol>
</section>
</section>
</section>
<section id="results" class="level2" data-number="3">
<h2 data-number="3" class="anchored" data-anchor-id="results"><span class="header-section-number">3</span> Results</h2>
<p><span class="citation" data-cites="cohenStochasticTheoryCommunity1985">Joel E. Cohen, Newman, and Steele (<a href="#ref-cohenStochasticTheoryCommunity1985" role="doc-biblioref">1985</a>)</span> actually tells us that the cascade model only really works for communities that range from 3-33 species… and <span class="citation" data-cites="williamsSuccessItsLimits2008">Williams and Martinez (<a href="#ref-williamsSuccessItsLimits2008" role="doc-biblioref">2008</a>)</span> also highlights how structural models really only work for small communities</p>
<section id="quantitative-stuff" class="level3" data-number="3.1">
<h3 data-number="3.1" class="anchored" data-anchor-id="quantitative-stuff"><span class="header-section-number">3.1</span> Quantitative stuff</h3>
<div class="quarto-embed-nb-cell">
<div id="cell-fig-topology" class="cell">
<div class="cell-output cell-output-display">
<div id="fig-topology" class="quarto-figure quarto-figure-center quarto-float anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-topology-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="index_files/figure-html/notebooks-model_quantitative-fig-topology-output-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-2" data-glightbox="description: .lightbox-desc-2"><img src="index_files/figure-html/notebooks-model_quantitative-fig-topology-output-1.png" class="img-fluid figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-topology-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;2: Real vs observed values for network summary statistics. Note here that ‘basal’ is calculated as the proportion of species that have a generality value of zero <em>i.e.,</em> are basal AFAIK
</figcaption>
</figure>
</div>
</div>
</div>
<a class="quarto-notebook-link" id="nblink-1" href="notebooks/model_quantitative.embed-preview.html#cell-fig-topology">Source: Quantitative approach to topology generators</a></div>
<p>This might actually be an awful way to try and summarise the data but rolling with it for now…</p>
</section>
</section>
<section id="discussion" class="level2" data-number="4">
<h2 data-number="4" class="anchored" data-anchor-id="discussion"><span class="header-section-number">4</span> Discussion</h2>
<p>I think a big take home will (hopefully) be how different approaches do better in different situations and so you as an end user need to take this into consideration and pick accordingly. I think <span class="citation" data-cites="petcheyFitEfficiencyBiology2011">Petchey et al. (<a href="#ref-petcheyFitEfficiencyBiology2011" role="doc-biblioref">2011</a>)</span> might have (and share) some thoughts on this (thanks Andrew). I feel like I need to look at <span class="citation" data-cites="berlowGoldilocksFactorFood2008">Berlow, Brose, and Martinez (<a href="#ref-berlowGoldilocksFactorFood2008" role="doc-biblioref">2008</a>)</span> but maybe not exactly in this context but vaguely adjacent.</p>
<p>An interesting thing to also think about (and arguably it will be addressed based on some of the other thoughts and ideas) is data dependant and data independent ‘parametrisation’ of the models…</p>
<p>I probably think about this point too much but a point of discussion that I think will be interesting to bring up the idea that if a model is missing a specific pairwise link but doing well at the structural level then when does it matter? I think this is covered with the whole node vs graph level performance but I kind of just want to bring it up here again because also one of those things that I think about a bit too much probably…</p>
<p>Thinking very long term here and maybe a bit beyond the scope but also thinking about a multi- model approach? So in other words using one model to build an initial network but maybe a second one to constrain it a bit better. I blame this thought on the over-connected PFIM food webs…</p>
<p>Linking to previous point but framing this as a call to action that we have models that predict networks very well and models that predict interactions very well but nothing that is doing well at predicting both - this is where we should be focusing our attention when it comes to furthering model development.</p>
</section>
<section id="references" class="level2 unnumbered">
<h2 class="unnumbered anchored" data-anchor-id="references">References</h2>
<div id="refs" class="references csl-bib-body hanging-indent" data-entry-spacing="0" role="list">
<div id="ref-bambachAutecologyFillingEcospace2007" class="csl-entry" role="listitem">
Bambach, Richard K., Andrew M. Bush, and Douglas H. Erwin. 2007. <span>“Autecology and the <span>Filling</span> of <span>Ecospace</span>: <span>Key Metazoan Radiations</span>.”</span> <em>Palaeontology</em> 50 (1): 1–22. <a href="https://doi.org/10.1111/j.1475-4983.2006.00611.x">https://doi.org/10.1111/j.1475-4983.2006.00611.x</a>.
</div>
<div id="ref-bascompteNestedAssemblyPlantanimal2003" class="csl-entry" role="listitem">
Bascompte, J., P. Jordano, C. J. Melian, and J. M. Olesen. 2003. <span>“The Nested Assembly of Plant-Animal Mutualistic Networks.”</span> <em>Proceedings of the National Academy of Sciences</em> 100 (16): 9383–87. <a href="https://doi.org/10.1073/pnas.1633576100">https://doi.org/10.1073/pnas.1633576100</a>.
</div>
<div id="ref-beckermanForagingBiologyPredicts2006" class="csl-entry" role="listitem">
Beckerman, Andrew P., Owen L. Petchey, and Philip H. Warren. 2006. <span>“Foraging Biology Predicts Food Web Complexity.”</span> <em>Proceedings of the National Academy of Sciences</em> 103 (37): 13745–49. <a href="https://doi.org/10.1073/pnas.0603039103">https://doi.org/10.1073/pnas.0603039103</a>.
</div>
<div id="ref-berlowGoldilocksFactorFood2008" class="csl-entry" role="listitem">
Berlow, Eric L., Ulrich Brose, and Neo D. Martinez. 2008. <span>“The <span>‘<span>Goldilocks</span> Factor’</span> in Food Webs.”</span> <em>Proceedings of the National Academy of Sciences</em> 105 (11): 4079–80. <a href="https://doi.org/10.1073/pnas.0800967105">https://doi.org/10.1073/pnas.0800967105</a>.
</div>
<div id="ref-caronTrophicInteractionModels2023" class="csl-entry" role="listitem">
Caron, Dominique, Ulrich Brose, Miguel Lurgi, Guillaume Blanchet, Dominique Gravel, and Laura J. Pollock. 2023. <span>“Trophic Interaction Models Predict Interactions Across Space, Not Food Webs.”</span> EcoEvoRxiv. <a href="https://doi.org/10.32942/X29K55">https://doi.org/10.32942/X29K55</a>.
</div>
<div id="ref-caronAddressingEltonianShortfall2022" class="csl-entry" role="listitem">
Caron, Dominique, Luigi Maiorano, Wilfried Thuiller, and Laura J. Pollock. 2022. <span>“Addressing the <span>Eltonian</span> Shortfall with Trait-Based Interaction Models.”</span> <em>Ecology Letters</em> 25 (4): 889–99. <a href="https://doi.org/10.1111/ele.13966">https://doi.org/10.1111/ele.13966</a>.
</div>
<div id="ref-cattinPhylogeneticConstraintsAdaptation2004" class="csl-entry" role="listitem">
Cattin, Marie-France, Louis-Félix Bersier, Carolin Banašek-Richter, Richard Baltensperger, and Jean-Pierre Gabriel. 2004. <span>“Phylogenetic Constraints and Adaptation Explain Food-Web Structure.”</span> <em>Nature</em> 427 (6977): 835–39. <a href="https://doi.org/10.1038/nature02327">https://doi.org/10.1038/nature02327</a>.
</div>
<div id="ref-cohenCommunityFoodWebs1990" class="csl-entry" role="listitem">
Cohen, Joel E, Frederic Briand, and Charles Newman. 1990. <em>Community <span>Food Webs</span>: <span>Data</span> and <span>Theory</span></em>. Biomathematics. Berlin Heidelberg: Springer-Verlag.
</div>
<div id="ref-cohenStochasticTheoryCommunity1985" class="csl-entry" role="listitem">
Cohen, Joel E., C. M. Newman, and John Hyslop Steele. 1985. <span>“A Stochastic Theory of Community Food Webs <span>I</span>. <span>Models</span> and Aggregated Data.”</span> <em>Proceedings of the Royal Society of London. Series B. Biological Sciences</em> 224 (1237): 421–48. <a href="https://doi.org/10.1098/rspb.1985.0042">https://doi.org/10.1098/rspb.1985.0042</a>.
</div>
<div id="ref-dunneNetworkStructureFood2006" class="csl-entry" role="listitem">
Dunne, Jennifer A. 2006. <span>“The <span>Network Structure</span> of <span>Food Webs</span>.”</span> In <em>Ecological Networks: <span>Linking</span> Structure and Dynamics</em>, edited by Jennifer A Dunne and Mercedes Pascual, 27–86. Oxford University Press.
</div>
<div id="ref-dunneCompilationNetworkAnalyses2008" class="csl-entry" role="listitem">
Dunne, Jennifer A., Richard J. Williams, Neo D. Martinez, Rachel A. Wood, and Douglas H. Erwin. 2008. <span>“Compilation and <span>Network Analyses</span> of <span>Cambrian Food Webs</span>.”</span> <em>PLOS Biology</em> 6 (4): e102. <a href="https://doi.org/10.1371/journal.pbio.0060102">https://doi.org/10.1371/journal.pbio.0060102</a>.
</div>
<div id="ref-fortunaHabitatLossStructure2006" class="csl-entry" role="listitem">
Fortuna, Miguel A., and Jordi Bascompte. 2006. <span>“Habitat Loss and the Structure of Plant-Animal Mutualistic Networks: <span>Mutualistic</span> Networks and Habitat Loss.”</span> <em>Ecology Letters</em> 9 (3): 281–86. <a href="https://doi.org/10.1111/j.1461-0248.2005.00868.x">https://doi.org/10.1111/j.1461-0248.2005.00868.x</a>.
</div>
<div id="ref-gravelInferringFoodWeb2013" class="csl-entry" role="listitem">
Gravel, Dominique, Timothée Poisot, Camille Albouy, Laure Velez, and David Mouillot. 2013. <span>“Inferring Food Web Structure from Predator–Prey Body Size Relationships.”</span> <em>Methods in Ecology and Evolution</em> 4 (11): 1083–90. <a href="https://doi.org/10.1111/2041-210X.12103">https://doi.org/10.1111/2041-210X.12103</a>.
</div>
<div id="ref-herbersteinAnimalTraitsCuratedAnimal2022" class="csl-entry" role="listitem">
Herberstein, Marie E., Donald James McLean, Elizabeth Lowe, Jonas O. Wolff, Md Kawsar Khan, Kaitlyn Smith, Andrew P. Allen, et al. 2022. <span>“<span>AnimalTraits</span> - a Curated Animal Trait Database for Body Mass, Metabolic Rate and Brain Size.”</span> <em>Scientific Data</em> 9 (1): 265. <a href="https://doi.org/10.1038/s41597-022-01364-9">https://doi.org/10.1038/s41597-022-01364-9</a>.
</div>
<div id="ref-jonesPanTHERIASpecieslevelDatabase2009" class="csl-entry" role="listitem">
Jones, Kate E., Jon Bielby, Marcel Cardillo, Susanne A. Fritz, Justin O’Dell, C. David L. Orme, Kamran Safi, et al. 2009. <span>“<span>PanTHERIA</span>: A Species-Level Database of Life History, Ecology, and Geography of Extant and Recently Extinct Mammals.”</span> <em>Ecology</em> 90 (9): 2648–48. <a href="https://doi.org/10.1890/08-1494.1">https://doi.org/10.1890/08-1494.1</a>.
</div>
<div id="ref-jordanoChasingEcologicalInteractions2016" class="csl-entry" role="listitem">
Jordano, Pedro. 2016a. <span>“Chasing <span>Ecological Interactions</span>.”</span> <em>PLOS Biology</em> 14 (9): e1002559. <a href="https://doi.org/10.1371/journal.pbio.1002559">https://doi.org/10.1371/journal.pbio.1002559</a>.
</div>
<div id="ref-jordanoSamplingNetworksEcological2016" class="csl-entry" role="listitem">
———. 2016b. <span>“Sampling Networks of Ecological Interactions.”</span> <em>Functional Ecology</em>, September. <a href="https://doi.org/10.1111/1365-2435.12763">https://doi.org/10.1111/1365-2435.12763</a>.
</div>
<div id="ref-newmanNetworksIntroduction2010" class="csl-entry" role="listitem">
Newman, Mark E. J. 2010. <em>Networks. <span>An</span> Introduction</em>. New York, NY: Oxford University Press.
</div>
<div id="ref-petcheySizeForagingFood2008" class="csl-entry" role="listitem">
Petchey, Owen L., Andrew P. Beckerman, Jens O. Riede, and Philip H. Warren. 2008. <span>“Size, Foraging, and Food Web Structure.”</span> <em>Proceedings of the National Academy of Sciences</em> 105 (11): 4191–96. <a href="https://doi.org/10.1073/pnas.0710672105">https://doi.org/10.1073/pnas.0710672105</a>.
</div>
<div id="ref-petcheyFitEfficiencyBiology2011" class="csl-entry" role="listitem">
———. 2011. <span>“Fit, Efficiency, and Biology: <span>Some</span> Thoughts on Judging Food Web Models.”</span> <em>Journal of Theoretical Biology</em> 279 (1): 169–71. <a href="https://doi.org/10.1016/j.jtbi.2011.03.019">https://doi.org/10.1016/j.jtbi.2011.03.019</a>.
</div>
<div id="ref-poisotGuidelinesPredictionSpecies2023" class="csl-entry" role="listitem">
Poisot, Timothée. 2023. <span>“Guidelines for the Prediction of Species Interactions Through Binary Classification.”</span> <em>Methods in Ecology and Evolution</em> 14 (5): 1333–45. <a href="https://doi.org/10.1111/2041-210X.14071">https://doi.org/10.1111/2041-210X.14071</a>.
</div>
<div id="ref-poisotGlobalKnowledgeGaps2021" class="csl-entry" role="listitem">
Poisot, Timothée, Gabriel Bergeron, Kevin Cazelles, Tad Dallas, Dominique Gravel, Andrew MacDonald, Benjamin Mercier, Clément Violet, and Steve Vissault. 2021. <span>“Global Knowledge Gaps in Species Interaction Networks Data.”</span> <em>Journal of Biogeography</em> n/a (n/a). <a href="https://doi.org/10.1111/jbi.14127">https://doi.org/10.1111/jbi.14127</a>.
</div>
<div id="ref-poisotSpeciesWhyEcological2015" class="csl-entry" role="listitem">
Poisot, Timothée, Daniel B. Stouffer, and Dominique Gravel. 2015. <span>“Beyond Species: Why Ecological Interaction Networks Vary Through Space and Time.”</span> <em>Oikos</em> 124 (3): 243–51. <a href="https://doi.org/10.1111/oik.01719">https://doi.org/10.1111/oik.01719</a>.
</div>
<div id="ref-pomeranzInferringPredatorPrey2019" class="csl-entry" role="listitem">
Pomeranz, Justin P. F., Ross M. Thompson, Timothée Poisot, and Jon S. Harding. 2019. <span>“Inferring Predator–Prey Interactions in Food Webs.”</span> <em>Methods in Ecology and Evolution</em> 10 (3): 356–67. <a href="https://doi.org/10.1111/2041-210X.13125">https://doi.org/10.1111/2041-210X.13125</a>.
</div>
<div id="ref-rohrModelingFoodWebs2010" class="csl-entry" role="listitem">
Rohr, Rudolf Philippe, Heike Scherer, Patrik Kehrli, Christian Mazza, and Louis-Félix Bersier. 2010. <span>“Modeling <span>Food Webs</span>: <span>Exploring Unexplained Structure Using Latent Traits</span>.”</span> <em>The American Naturalist</em> 176 (2): 170–77. <a href="https://doi.org/10.1086/653667">https://doi.org/10.1086/653667</a>.
</div>
<div id="ref-shawFrameworkReconstructingAncient2024" class="csl-entry" role="listitem">
Shaw, Jack O., Alexander M. Dunhill, Andrew P. Beckerman, Jennifer A. Dunne, and Pincelli M. Hull. 2024. <span>“A Framework for Reconstructing Ancient Food Webs Using Functional Trait Data.”</span> bioRxiv. <a href="https://doi.org/10.1101/2024.01.30.578036">https://doi.org/10.1101/2024.01.30.578036</a>.
</div>
<div id="ref-strydomFoodWebReconstruction2022" class="csl-entry" role="listitem">
Strydom, Tanya, Salomé Bouskila, Francis Banville, Ceres Barros, Dominique Caron, Maxwell J. Farrell, Marie-Josée Fortin, et al. 2022. <span>“Food Web Reconstruction Through Phylogenetic Transfer of Low-Rank Network Representation.”</span> <em>Methods in Ecology and Evolution</em> 13 (12): 2838–49. <a href="https://doi.org/10.1111/2041-210X.13835">https://doi.org/10.1111/2041-210X.13835</a>.
</div>
<div id="ref-strydomGraphEmbeddingTransfer2023" class="csl-entry" role="listitem">
Strydom, Tanya, Salomé Bouskila, Francis Banville, Ceres Barros, Dominique Caron, Maxwell J. Farrell, Marie-Josée Fortin, et al. 2023. <span>“Graph Embedding and Transfer Learning Can Help Predict Potential Species Interaction Networks Despite Data Limitations.”</span> <em>Methods in Ecology and Evolution</em> 14 (12): 2917–30. <a href="https://doi.org/10.1111/2041-210X.14228">https://doi.org/10.1111/2041-210X.14228</a>.
</div>
<div id="ref-strydomRoadmapPredictingSpecies2021" class="csl-entry" role="listitem">
Strydom, Tanya, Michael D. Catchen, Francis Banville, Dominique Caron, Gabriel Dansereau, Philippe Desjardins-Proulx, Norma R. Forero-Muñoz, et al. 2021. <span>“A Roadmap Towards Predicting Species Interaction Networks (Across Space and Time).”</span> <em>Philosophical Transactions of the Royal Society B: Biological Sciences</em> 376 (1837): 20210063. <a href="https://doi.org/10.1098/rstb.2021.0063">https://doi.org/10.1098/rstb.2021.0063</a>.
</div>
<div id="ref-strydomSVDEntropyReveals2021" class="csl-entry" role="listitem">
Strydom, Tanya, Giulio V. Dalla Riva, and Timothée Poisot. 2021. <span>“<span>SVD Entropy Reveals</span> the <span>High Complexity</span> of <span>Ecological Networks</span>.”</span> <em>Frontiers in Ecology and Evolution</em> 9. <a href="https://doi.org/10.3389/fevo.2021.623141">https://doi.org/10.3389/fevo.2021.623141</a>.
</div>
<div id="ref-uphamInferringMammalTree2019" class="csl-entry" role="listitem">
Upham, Nathan S., Jacob A. Esselstyn, and Walter Jetz. 2019. <span>“Inferring the Mammal Tree: <span class="nocase">Species-level</span> Sets of Phylogenies for Questions in Ecology, Evolution, and Conservation.”</span> <em>PLOS Biology</em> 17 (12): e3000494. <a href="https://doi.org/10.1371/journal.pbio.3000494">https://doi.org/10.1371/journal.pbio.3000494</a>.
</div>
<div id="ref-valdovinosBioenergeticFrameworkAboveground2023" class="csl-entry" role="listitem">
Valdovinos, Fernanda S., Kayla R. S. Hale, Sabine Dritz, Paul R. Glaum, Kevin S. McCann, Sophia M. Simon, Elisa Thébault, William C. Wetzel, Kate L. Wootton, and Justin D. Yeakel. 2023. <span>“A Bioenergetic Framework for Aboveground Terrestrial Food Webs.”</span> <em>Trends in Ecology &amp; Evolution</em> 38 (3): 301–12. <a href="https://doi.org/10.1016/j.tree.2022.11.004">https://doi.org/10.1016/j.tree.2022.11.004</a>.
</div>
<div id="ref-williamsSimpleRulesYield2000" class="csl-entry" role="listitem">
Williams, Richard J., and Neo D. Martinez. 2000. <span>“Simple Rules Yield Complex Food Webs.”</span> <em>Nature</em> 404 (6774): 180–83. <a href="https://doi.org/10.1038/35004572">https://doi.org/10.1038/35004572</a>.
</div>
<div id="ref-williamsSuccessItsLimits2008" class="csl-entry" role="listitem">
———. 2008. <span>“Success and Its Limits Among Structural Models of Complex Food Webs.”</span> <em>Journal of Animal Ecology</em> 77 (3): 512–19. <a href="https://doi.org/10.1111/j.1365-2656.2008.01362.x">https://doi.org/10.1111/j.1365-2656.2008.01362.x</a>.
</div>
<div id="ref-wilmanEltonTraitsSpecieslevelForaging2014" class="csl-entry" role="listitem">
Wilman, Hamish, Jonathan Belmaker, Jennifer Simpson, Carolina de la Rosa, Marcelo M. Rivadeneira, and Walter Jetz. 2014. <span>“<span>EltonTraits</span> 1.0: <span class="nocase">Species-level</span> Foraging Attributes of the World’s Birds and Mammals.”</span> <em>Ecology</em> 95 (7): 2027–27. <a href="https://doi.org/10.1890/13-1917.1">https://doi.org/10.1890/13-1917.1</a>.
</div>
</div>
<div class="hidden" aria-hidden="true">
<span class="glightbox-desc lightbox-desc-1">Figure&nbsp;1: Conceptual figure of the ‘network prediction’. Panel A shows where the model families fall in the the context of being models that predict networks or models that predict interactions space. Panel B serves to highlight the characteristics one might like to ‘test’/benchmark for a model based on it being either a network or interaction predicting model</span>
<span class="glightbox-desc lightbox-desc-2">Figure&nbsp;2: Real vs observed values for network summary statistics. Note here that ‘basal’ is calculated as the proportion of species that have a generality value of zero <em>i.e.,</em> are basal AFAIK</span>
</div>
</section>

<div id="quarto-appendix" class="default"><section class="quarto-appendix-contents" id="quarto-citation"><h2 class="anchored quarto-appendix-heading">Citation</h2><div><div class="quarto-appendix-secondary-label">BibTeX citation:</div><pre class="sourceCode code-with-copy quarto-appendix-bibtex"><code class="sourceCode bibtex">@online{strydom2024,
  author = {Strydom, Tanya and P. Beckerman, Andrew},
  title = {T Is for {Topology}},
  date = {2024-03-18},
  langid = {en},
  abstract = {There are many reasons one might want to generate a
    network and there are many tools on the market that might make that
    possible. However not all tools are created equally and there is
    reason to assume that not all networks will suit most purposes. Here
    the aim is to compare and contrast the different topology generating
    tools that are on the market and see where they shine and where they
    fall flat. There probably isn’t one model to rule them all but it
    doesn’t mean that we shouldn’t be critical when we think about the
    model we want to use.}
}
</code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre><div class="quarto-appendix-secondary-label">For attribution, please cite this work as:</div><div id="ref-strydom2024" class="csl-entry quarto-appendix-citeas" role="listitem">
Strydom, Tanya, and Andrew P. Beckerman. 2024. <span>“T Is for
Topology.”</span> Some Fancy Journal. March 18, 2024.
</div></div></section></div></main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->
<script>var lightboxQuarto = GLightbox({"selector":".lightbox","openEffect":"zoom","loop":false,"descPosition":"bottom","closeEffect":"zoom"});
window.onload = () => {
  lightboxQuarto.on('slide_before_load', (data) => {
    const { slideIndex, slideNode, slideConfig, player, trigger } = data;
    const href = trigger.getAttribute('href');
    if (href !== null) {
      const imgEl = window.document.querySelector(`a[href="${href}"] img`);
      if (imgEl !== null) {
        const srcAttr = imgEl.getAttribute("src");
        if (srcAttr && srcAttr.startsWith("data:")) {
          slideConfig.href = srcAttr;
        }
      }
    } 
  });

  lightboxQuarto.on('slide_after_load', (data) => {
    const { slideIndex, slideNode, slideConfig, player, trigger } = data;
    if (window.Quarto?.typesetMath) {
      window.Quarto.typesetMath(slideNode);
    }
  });

};
          </script>




</body></html>